<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 10 Community typing (clustering) | Orchestrating Microbiome Analysis</title>
  <meta name="description" content="Chapter 10 Community typing (clustering) | Orchestrating Microbiome Analysis" />
  <meta name="generator" content="bookdown 0.32 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 10 Community typing (clustering) | Orchestrating Microbiome Analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="microbiome/OMA" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 10 Community typing (clustering) | Orchestrating Microbiome Analysis" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="microbiome-community.html"/>
<link rel="next" href="differential-abundance.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.1/htmlwidgets.js"></script>
<script src="libs/plotly-binding-4.10.1/plotly.js"></script>
<script src="libs/typedarray-0.1/typedarray.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="libs/plotly-htmlwidgets-css-2.11.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="libs/plotly-main-2.11.1/plotly-latest.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Orchestrating Microbiome Analysis</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="part"><span><b>I Introduction</b></span></li>
<li class="chapter" data-level="1" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="containers.html"><a href="containers.html"><i class="fa fa-check"></i><b>2</b> Microbiome Data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="containers.html"><a href="containers.html#data-science-framework"><i class="fa fa-check"></i><b>2.1</b> Data science framework</a></li>
<li class="chapter" data-level="2.2" data-path="containers.html"><a href="containers.html#data-containers"><i class="fa fa-check"></i><b>2.2</b> Data containers</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="containers.html"><a href="containers.html#assay-data"><i class="fa fa-check"></i><b>2.2.1</b> Assay data</a></li>
<li class="chapter" data-level="2.2.2" data-path="containers.html"><a href="containers.html#coldata"><i class="fa fa-check"></i><b>2.2.2</b> colData</a></li>
<li class="chapter" data-level="2.2.3" data-path="containers.html"><a href="containers.html#rowdata"><i class="fa fa-check"></i><b>2.2.3</b> rowData</a></li>
<li class="chapter" data-level="2.2.4" data-path="containers.html"><a href="containers.html#rowtree"><i class="fa fa-check"></i><b>2.2.4</b> rowTree</a></li>
<li class="chapter" data-level="2.2.5" data-path="containers.html"><a href="containers.html#alt-exp"><i class="fa fa-check"></i><b>2.2.5</b> Alternative experiments</a></li>
<li class="chapter" data-level="2.2.6" data-path="containers.html"><a href="containers.html#mae"><i class="fa fa-check"></i><b>2.2.6</b> MultiAssayExperiments</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="containers.html"><a href="containers.html#loading-experimental-microbiome-data"><i class="fa fa-check"></i><b>2.3</b> Loading experimental microbiome data</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="containers.html"><a href="containers.html#s-workflow"><i class="fa fa-check"></i><b>2.3.1</b> 16S workflow</a></li>
<li class="chapter" data-level="2.3.2" data-path="containers.html"><a href="containers.html#import-from-external-files"><i class="fa fa-check"></i><b>2.3.2</b> Import from external files</a></li>
<li class="chapter" data-level="2.3.3" data-path="containers.html"><a href="containers.html#conversions-between-data-formats-in-r"><i class="fa fa-check"></i><b>2.3.3</b> Conversions between data formats in R</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="containers.html"><a href="containers.html#example-data"><i class="fa fa-check"></i><b>2.4</b> Demonstration data</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="containers.html"><a href="containers.html#package-data"><i class="fa fa-check"></i><b>2.4.1</b> Package data</a></li>
<li class="chapter" data-level="2.4.2" data-path="containers.html"><a href="containers.html#experimenthub-data"><i class="fa fa-check"></i><b>2.4.2</b> ExperimentHub data</a></li>
<li class="chapter" data-level="2.4.3" data-path="containers.html"><a href="containers.html#other-data-sources"><i class="fa fa-check"></i><b>2.4.3</b> Other data sources</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="containers.html"><a href="containers.html#session-info"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="packages.html"><a href="packages.html"><i class="fa fa-check"></i><b>3</b> Packages</a>
<ul>
<li class="chapter" data-level="3.1" data-path="packages.html"><a href="packages.html#package-installation"><i class="fa fa-check"></i><b>3.1</b> Package installation</a></li>
<li class="chapter" data-level="3.2" data-path="packages.html"><a href="packages.html#some-available-packages"><i class="fa fa-check"></i><b>3.2</b> Some available packages</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="packages.html"><a href="packages.html#mia-family-of-methods"><i class="fa fa-check"></i><b>3.2.1</b> mia family of methods</a></li>
<li class="chapter" data-level="3.2.2" data-path="packages.html"><a href="packages.html#tree-based-methods"><i class="fa fa-check"></i><b>3.2.2</b> Tree-based methods</a></li>
<li class="chapter" data-level="3.2.3" data-path="packages.html"><a href="packages.html#differential-abundance"><i class="fa fa-check"></i><b>3.2.3</b> Differential abundance</a></li>
<li class="chapter" data-level="3.2.4" data-path="packages.html"><a href="packages.html#manipulation"><i class="fa fa-check"></i><b>3.2.4</b> Manipulation</a></li>
<li class="chapter" data-level="3.2.5" data-path="packages.html"><a href="packages.html#data"><i class="fa fa-check"></i><b>3.2.5</b> Data</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="packages.html"><a href="packages.html#session-info-1"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="part"><span><b>II Focus Topics</b></span></li>
<li class="chapter" data-level="4" data-path="datamanipulation.html"><a href="datamanipulation.html"><i class="fa fa-check"></i><b>4</b> Data Manipulation</a>
<ul>
<li class="chapter" data-level="4.1" data-path="datamanipulation.html"><a href="datamanipulation.html#tidying-and-subsetting"><i class="fa fa-check"></i><b>4.1</b> Tidying and subsetting</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="datamanipulation.html"><a href="datamanipulation.html#tidy-data"><i class="fa fa-check"></i><b>4.1.1</b> Tidy data</a></li>
<li class="chapter" data-level="4.1.2" data-path="datamanipulation.html"><a href="datamanipulation.html#subsetting"><i class="fa fa-check"></i><b>4.1.2</b> Subsetting</a></li>
<li class="chapter" data-level="4.1.3" data-path="datamanipulation.html"><a href="datamanipulation.html#splitting"><i class="fa fa-check"></i><b>4.1.3</b> Splitting</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="datamanipulation.html"><a href="datamanipulation.html#merge-data"><i class="fa fa-check"></i><b>4.2</b> Merge data</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="datamanipulation.html"><a href="datamanipulation.html#additional-functions"><i class="fa fa-check"></i><b>4.2.1</b> Additional functions</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>5</b> Exploration and quality Control</a>
<ul>
<li class="chapter" data-level="5.1" data-path="quality-control.html"><a href="quality-control.html#abundance"><i class="fa fa-check"></i><b>5.1</b> Abundance</a></li>
<li class="chapter" data-level="5.2" data-path="quality-control.html"><a href="quality-control.html#prevalence"><i class="fa fa-check"></i><b>5.2</b> Prevalence</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="quality-control.html"><a href="quality-control.html#prevalence-analysis"><i class="fa fa-check"></i><b>5.2.1</b> Prevalence analysis</a></li>
<li class="chapter" data-level="5.2.2" data-path="quality-control.html"><a href="quality-control.html#rare-taxa"><i class="fa fa-check"></i><b>5.2.2</b> Rare taxa</a></li>
<li class="chapter" data-level="5.2.3" data-path="quality-control.html"><a href="quality-control.html#plotting-prevalence"><i class="fa fa-check"></i><b>5.2.3</b> Plotting prevalence</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="quality-control.html"><a href="quality-control.html#qc"><i class="fa fa-check"></i><b>5.3</b> Quality control</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="quality-control.html"><a href="quality-control.html#top-taxa"><i class="fa fa-check"></i><b>5.3.1</b> Top taxa</a></li>
<li class="chapter" data-level="5.3.2" data-path="quality-control.html"><a href="quality-control.html#library-size-read-count"><i class="fa fa-check"></i><b>5.3.2</b> Library size / read count</a></li>
<li class="chapter" data-level="5.3.3" data-path="quality-control.html"><a href="quality-control.html#contaminant-sequences"><i class="fa fa-check"></i><b>5.3.3</b> Contaminant sequences</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="quality-control.html"><a href="quality-control.html#session-info-2"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="taxonomic-information.html"><a href="taxonomic-information.html"><i class="fa fa-check"></i><b>6</b> Taxonomic Information</a>
<ul>
<li class="chapter" data-level="6.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#assigning-taxonomic-information."><i class="fa fa-check"></i><b>6.1</b> Assigning taxonomic information.</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#dada2"><i class="fa fa-check"></i><b>6.1.1</b> dada2</a></li>
<li class="chapter" data-level="6.1.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#decipher"><i class="fa fa-check"></i><b>6.1.2</b> DECIPHER</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#functions-to-access-taxonomic-information"><i class="fa fa-check"></i><b>6.2</b> Functions to access taxonomic information</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#generate-a-taxonomic-tree-on-the-fly"><i class="fa fa-check"></i><b>6.2.1</b> Generate a taxonomic tree on the fly</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="taxonomic-information.html"><a href="taxonomic-information.html#data-agglomeration"><i class="fa fa-check"></i><b>6.3</b> Data agglomeration</a></li>
<li class="chapter" data-level="6.4" data-path="taxonomic-information.html"><a href="taxonomic-information.html#data-transformation"><i class="fa fa-check"></i><b>6.4</b> Data transformation</a></li>
<li class="chapter" data-level="6.5" data-path="taxonomic-information.html"><a href="taxonomic-information.html#pick-specific"><i class="fa fa-check"></i><b>6.5</b> Pick specific</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#abundances-of-all-taxa-in-specific-sample"><i class="fa fa-check"></i><b>6.5.1</b> Abundances of all taxa in specific sample</a></li>
<li class="chapter" data-level="6.5.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#abundances-of-specific-taxa-in-all-samples"><i class="fa fa-check"></i><b>6.5.2</b> Abundances of specific taxa in all samples</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="taxonomic-information.html"><a href="taxonomic-information.html#session-info-3"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="community-diversity.html"><a href="community-diversity.html"><i class="fa fa-check"></i><b>7</b> Community diversity</a>
<ul>
<li class="chapter" data-level="7.1" data-path="community-diversity.html"><a href="community-diversity.html#estimation"><i class="fa fa-check"></i><b>7.1</b> Estimation</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="community-diversity.html"><a href="community-diversity.html#richness"><i class="fa fa-check"></i><b>7.1.1</b> Richness</a></li>
<li class="chapter" data-level="7.1.2" data-path="community-diversity.html"><a href="community-diversity.html#estimate-diversity"><i class="fa fa-check"></i><b>7.1.2</b> Diversity</a></li>
<li class="chapter" data-level="7.1.3" data-path="community-diversity.html"><a href="community-diversity.html#faith-diversity"><i class="fa fa-check"></i><b>7.1.3</b> Faith phylogenetic diversity</a></li>
<li class="chapter" data-level="7.1.4" data-path="community-diversity.html"><a href="community-diversity.html#evenness"><i class="fa fa-check"></i><b>7.1.4</b> Evenness</a></li>
<li class="chapter" data-level="7.1.5" data-path="community-diversity.html"><a href="community-diversity.html#dominance"><i class="fa fa-check"></i><b>7.1.5</b> Dominance</a></li>
<li class="chapter" data-level="7.1.6" data-path="community-diversity.html"><a href="community-diversity.html#rarity"><i class="fa fa-check"></i><b>7.1.6</b> Rarity</a></li>
<li class="chapter" data-level="7.1.7" data-path="community-diversity.html"><a href="community-diversity.html#divergence"><i class="fa fa-check"></i><b>7.1.7</b> Divergence</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="community-diversity.html"><a href="community-diversity.html#visualization"><i class="fa fa-check"></i><b>7.2</b> Visualization</a></li>
<li class="chapter" data-level="" data-path="community-diversity.html"><a href="community-diversity.html#session-info-4"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="community-similarity.html"><a href="community-similarity.html"><i class="fa fa-check"></i><b>8</b> Community similarity</a>
<ul>
<li class="chapter" data-level="8.1" data-path="community-similarity.html"><a href="community-similarity.html#explained-variance"><i class="fa fa-check"></i><b>8.1</b> Explained variance</a></li>
<li class="chapter" data-level="8.2" data-path="community-similarity.html"><a href="community-similarity.html#community-comparisons-by-beta-diversity-analysis"><i class="fa fa-check"></i><b>8.2</b> Community comparisons by beta diversity analysis</a></li>
<li class="chapter" data-level="8.3" data-path="community-similarity.html"><a href="community-similarity.html#other-ordination-methods"><i class="fa fa-check"></i><b>8.3</b> Other ordination methods</a></li>
<li class="chapter" data-level="8.4" data-path="community-similarity.html"><a href="community-similarity.html#visualizing-the-most-dominant-genus-on-pcoa"><i class="fa fa-check"></i><b>8.4</b> Visualizing the most dominant genus on PCoA</a>
<ul>
<li class="chapter" data-level="8.4.1" data-path="community-similarity.html"><a href="community-similarity.html#testing-differences-in-community-composition-between-sample-groups"><i class="fa fa-check"></i><b>8.4.1</b> Testing differences in community composition between sample groups</a></li>
<li class="chapter" data-level="8.4.2" data-path="community-similarity.html"><a href="community-similarity.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>8.4.2</b> Checking the homogeneity condition</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="community-similarity.html"><a href="community-similarity.html#further-reading"><i class="fa fa-check"></i><b>8.5</b> Further reading</a></li>
<li class="chapter" data-level="" data-path="community-similarity.html"><a href="community-similarity.html#session-info-5"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="microbiome-community.html"><a href="microbiome-community.html"><i class="fa fa-check"></i><b>9</b> Community composition</a>
<ul>
<li class="chapter" data-level="9.1" data-path="microbiome-community.html"><a href="microbiome-community.html#visual-composition"><i class="fa fa-check"></i><b>9.1</b> Visualizing taxonomic composition</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="microbiome-community.html"><a href="microbiome-community.html#composition-barplot"><i class="fa fa-check"></i><b>9.1.1</b> Composition barplot</a></li>
<li class="chapter" data-level="9.1.2" data-path="microbiome-community.html"><a href="microbiome-community.html#composition-heatmap"><i class="fa fa-check"></i><b>9.1.2</b> Composition heatmap</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="microbiome-community.html"><a href="microbiome-community.html#session-info-6"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="clustering.html"><a href="clustering.html"><i class="fa fa-check"></i><b>10</b> Community typing (clustering)</a>
<ul>
<li class="chapter" data-level="10.1" data-path="clustering.html"><a href="clustering.html#hiearchical-clustering"><i class="fa fa-check"></i><b>10.1</b> Hiearchical clustering</a></li>
<li class="chapter" data-level="10.2" data-path="clustering.html"><a href="clustering.html#k-means-clustering"><i class="fa fa-check"></i><b>10.2</b> K-means clustering</a></li>
<li class="chapter" data-level="10.3" data-path="clustering.html"><a href="clustering.html#dirichlet-multinomial-mixtures-dmm"><i class="fa fa-check"></i><b>10.3</b> Dirichlet Multinomial Mixtures (DMM)</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="clustering.html"><a href="clustering.html#pcoa-for-asv-level-data-with-bray-curtis-with-dmm-clusters-shown-with-colors"><i class="fa fa-check"></i><b>10.3.1</b> PCoA for ASV-level data with Bray-Curtis; with DMM clusters shown with colors</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="clustering.html"><a href="clustering.html#community-detection"><i class="fa fa-check"></i><b>10.4</b> Community Detection</a></li>
<li class="chapter" data-level="10.5" data-path="clustering.html"><a href="clustering.html#biclustering"><i class="fa fa-check"></i><b>10.5</b> Biclustering</a>
<ul>
<li class="chapter" data-level="10.5.1" data-path="clustering.html"><a href="clustering.html#taxa-vs-samples"><i class="fa fa-check"></i><b>10.5.1</b> Taxa vs samples</a></li>
<li class="chapter" data-level="10.5.2" data-path="clustering.html"><a href="clustering.html#taxa-vs-biomolecules"><i class="fa fa-check"></i><b>10.5.2</b> Taxa vs biomolecules</a></li>
<li class="chapter" data-level="10.5.3" data-path="clustering.html"><a href="clustering.html#taxa-vs-taxa"><i class="fa fa-check"></i><b>10.5.3</b> Taxa vs taxa</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="clustering.html"><a href="clustering.html#additional-community-typing"><i class="fa fa-check"></i><b>10.6</b> Additional Community Typing</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="packages.html"><a href="packages.html#differential-abundance"><i class="fa fa-check"></i><b>11</b> Differential abundance</a>
<ul>
<li class="chapter" data-level="11.1" data-path="differential-abundance.html"><a href="differential-abundance.html"><i class="fa fa-check"></i><b>11.1</b> Differential abundance analysis</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="differential-abundance.html"><a href="differential-abundance.html#examples-and-tools"><i class="fa fa-check"></i><b>11.1.1</b> Examples and tools</a></li>
<li class="chapter" data-level="11.1.2" data-path="differential-abundance.html"><a href="differential-abundance.html#prevalence-filtering"><i class="fa fa-check"></i><b>11.1.2</b> Prevalence Filtering</a></li>
<li class="chapter" data-level="11.1.3" data-path="differential-abundance.html"><a href="differential-abundance.html#aldex2"><i class="fa fa-check"></i><b>11.1.3</b> ALDEx2</a></li>
<li class="chapter" data-level="11.1.4" data-path="differential-abundance.html"><a href="differential-abundance.html#ancom-bc"><i class="fa fa-check"></i><b>11.1.4</b> ANCOM-BC</a></li>
<li class="chapter" data-level="11.1.5" data-path="differential-abundance.html"><a href="differential-abundance.html#maaslin2"><i class="fa fa-check"></i><b>11.1.5</b> MaAsLin2</a></li>
<li class="chapter" data-level="11.1.6" data-path="differential-abundance.html"><a href="differential-abundance.html#linda"><i class="fa fa-check"></i><b>11.1.6</b> LinDA</a></li>
<li class="chapter" data-level="11.1.7" data-path="differential-abundance.html"><a href="differential-abundance.html#comparison-of-the-methods"><i class="fa fa-check"></i><b>11.1.7</b> Comparison of the methods</a></li>
<li class="chapter" data-level="11.1.8" data-path="differential-abundance.html"><a href="differential-abundance.html#confounding-variables"><i class="fa fa-check"></i><b>11.1.8</b> Confounding variables</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="differential-abundance.html"><a href="differential-abundance.html#tree-based-methods-1"><i class="fa fa-check"></i><b>11.2</b> Tree-based methods</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="differential-abundance.html"><a href="differential-abundance.html#group-wise-associations-testing-based-on-balances"><i class="fa fa-check"></i><b>11.2.1</b> Group-wise associations testing based on balances</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="differential-abundance.html"><a href="differential-abundance.html#session-info-7"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="machine_learning.html"><a href="machine_learning.html"><i class="fa fa-check"></i><b>12</b> Machine learning</a>
<ul>
<li class="chapter" data-level="12.1" data-path="machine_learning.html"><a href="machine_learning.html#supervised-machine-learning"><i class="fa fa-check"></i><b>12.1</b> Supervised machine learning</a></li>
<li class="chapter" data-level="12.2" data-path="machine_learning.html"><a href="machine_learning.html#unsupervised-machine-learning"><i class="fa fa-check"></i><b>12.2</b> Unsupervised machine learning</a></li>
<li class="chapter" data-level="" data-path="machine_learning.html"><a href="machine_learning.html#session-info-8"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="multi-assay-analyses.html"><a href="multi-assay-analyses.html"><i class="fa fa-check"></i><b>13</b> Multi-assay analyses</a>
<ul>
<li class="chapter" data-level="13.1" data-path="multi-assay-analyses.html"><a href="multi-assay-analyses.html#cross-correlation-analysis"><i class="fa fa-check"></i><b>13.1</b> Cross-correlation Analysis</a></li>
<li class="chapter" data-level="13.2" data-path="multi-assay-analyses.html"><a href="multi-assay-analyses.html#multi-omics-factor-analysis"><i class="fa fa-check"></i><b>13.2</b> Multi-Omics Factor Analysis</a></li>
<li class="chapter" data-level="" data-path="multi-assay-analyses.html"><a href="multi-assay-analyses.html#session-info-9"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="viz-chapter.html"><a href="viz-chapter.html"><i class="fa fa-check"></i><b>14</b> Visualization</a>
<ul>
<li class="chapter" data-level="14.1" data-path="viz-chapter.html"><a href="viz-chapter.html#pre-analysis-exploration"><i class="fa fa-check"></i><b>14.1</b> Pre-analysis exploration</a>
<ul>
<li class="chapter" data-level="14.1.1" data-path="viz-chapter.html"><a href="viz-chapter.html#accessing-row-and-column-data"><i class="fa fa-check"></i><b>14.1.1</b> Accessing row and column data</a></li>
<li class="chapter" data-level="14.1.2" data-path="viz-chapter.html"><a href="viz-chapter.html#viewing-abundance-and-prevalence-patterns"><i class="fa fa-check"></i><b>14.1.2</b> Viewing abundance and prevalence patterns</a></li>
</ul></li>
<li class="chapter" data-level="14.2" data-path="viz-chapter.html"><a href="viz-chapter.html#diversity-estimation"><i class="fa fa-check"></i><b>14.2</b> Diversity estimation</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="viz-chapter.html"><a href="viz-chapter.html#alpha-diversity-with-scatter-violin-and-box-plots"><i class="fa fa-check"></i><b>14.2.1</b> Alpha diversity with scatter, violin and box plots</a></li>
<li class="chapter" data-level="14.2.2" data-path="viz-chapter.html"><a href="viz-chapter.html#beta-diversity-with-shepard-and-coordination-plots"><i class="fa fa-check"></i><b>14.2.2</b> Beta diversity with Shepard and coordination plots</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="viz-chapter.html"><a href="viz-chapter.html#statistical-analysis"><i class="fa fa-check"></i><b>14.3</b> Statistical analysis</a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="viz-chapter.html"><a href="viz-chapter.html#heatmaps"><i class="fa fa-check"></i><b>14.3.1</b> Heatmaps</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="viz-chapter.html"><a href="viz-chapter.html#session-info-10"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="part"><span><b>III Appendix</b></span></li>
<li class="chapter" data-level="15" data-path="training.html"><a href="training.html"><i class="fa fa-check"></i><b>15</b> Training</a>
<ul>
<li class="chapter" data-level="15.1" data-path="training.html"><a href="training.html#coc"><i class="fa fa-check"></i><b>15.1</b> Code of Conduct</a></li>
<li class="chapter" data-level="15.2" data-path="training.html"><a href="training.html#checklist"><i class="fa fa-check"></i><b>15.2</b> Checklist</a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="training.html"><a href="training.html#setting-up-your-own-computer"><i class="fa fa-check"></i><b>15.2.1</b> Setting up Your own computer</a></li>
<li class="chapter" data-level="15.2.2" data-path="training.html"><a href="training.html#support-and-resources"><i class="fa fa-check"></i><b>15.2.2</b> Support and resources</a></li>
<li class="chapter" data-level="15.2.3" data-path="training.html"><a href="training.html#material"><i class="fa fa-check"></i><b>15.2.3</b> Study material</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="packages.html"><a href="packages.html#packages"><i class="fa fa-check"></i><b>15.3</b> Installing and loading the required R packages</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="resources.html"><a href="resources.html"><i class="fa fa-check"></i><b>16</b> Resources</a>
<ul>
<li class="chapter" data-level="16.1" data-path="resources.html"><a href="resources.html#data-containers-1"><i class="fa fa-check"></i><b>16.1</b> Data containers</a>
<ul>
<li class="chapter" data-level="16.1.1" data-path="resources.html"><a href="resources.html#resources-for-treesummarizedexperiment"><i class="fa fa-check"></i><b>16.1.1</b> Resources for TreeSummarizedExperiment</a></li>
<li class="chapter" data-level="16.1.2" data-path="resources.html"><a href="resources.html#other-relevant-containers"><i class="fa fa-check"></i><b>16.1.2</b> Other relevant containers</a></li>
<li class="chapter" data-level="16.1.3" data-path="resources.html"><a href="resources.html#alternative-containers-for-microbiome-data"><i class="fa fa-check"></i><b>16.1.3</b> Alternative containers for microbiome data</a></li>
<li class="chapter" data-level="16.1.4" data-path="resources.html"><a href="resources.html#resources-for-phyloseq"><i class="fa fa-check"></i><b>16.1.4</b> Resources for phyloseq</a></li>
</ul></li>
<li class="chapter" data-level="16.2" data-path="resources.html"><a href="resources.html#r-programming-resources"><i class="fa fa-check"></i><b>16.2</b> R programming resources</a>
<ul>
<li class="chapter" data-level="16.2.1" data-path="resources.html"><a href="resources.html#bioc_intro"><i class="fa fa-check"></i><b>16.2.1</b> Bioconductor Classes</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="resources.html"><a href="resources.html#reproducible-reporting-with-quarto"><i class="fa fa-check"></i><b>16.3</b> Reproducible reporting with Quarto</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="extras.html"><a href="extras.html"><i class="fa fa-check"></i><b>17</b> Extra material</a>
<ul>
<li class="chapter" data-level="17.1" data-path="extras.html"><a href="extras.html#interactive-3d-plots"><i class="fa fa-check"></i><b>17.1</b> Interactive 3D Plots</a></li>
<li class="chapter" data-level="17.2" data-path="extras.html"><a href="extras.html#permanova-comparison"><i class="fa fa-check"></i><b>17.2</b> PERMANOVA comparison</a></li>
<li class="chapter" data-level="17.3" data-path="extras.html"><a href="extras.html#bayesian-multinomial-logistic-normal-models"><i class="fa fa-check"></i><b>17.3</b> Bayesian Multinomial Logistic-Normal Models</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="acknowledgments.html"><a href="acknowledgments.html"><i class="fa fa-check"></i><b>18</b> Acknowledgments</a></li>
<li class="chapter" data-level="19" data-path="exercises.html"><a href="exercises.html"><i class="fa fa-check"></i><b>19</b> Exercises</a>
<ul>
<li class="chapter" data-level="19.1" data-path="exercises.html"><a href="exercises.html#workflows"><i class="fa fa-check"></i><b>19.1</b> Workflows</a>
<ul>
<li class="chapter" data-level="19.1.1" data-path="exercises.html"><a href="exercises.html#reproducible-reporting"><i class="fa fa-check"></i><b>19.1.1</b> Reproducible reporting</a></li>
</ul></li>
<li class="chapter" data-level="19.2" data-path="exercises.html"><a href="exercises.html#treesummarizedexperiment-basic-components"><i class="fa fa-check"></i><b>19.2</b> TreeSummarizedExperiment: basic components</a>
<ul>
<li class="chapter" data-level="19.2.1" data-path="exercises.html"><a href="exercises.html#basic-summaries"><i class="fa fa-check"></i><b>19.2.1</b> Basic summaries</a></li>
<li class="chapter" data-level="19.2.2" data-path="exercises.html"><a href="exercises.html#taxonomic-abundance-table-assay"><i class="fa fa-check"></i><b>19.2.2</b> Taxonomic abundance table (assay)</a></li>
<li class="chapter" data-level="19.2.3" data-path="exercises.html"><a href="exercises.html#sample-side-information"><i class="fa fa-check"></i><b>19.2.3</b> Sample side information</a></li>
<li class="chapter" data-level="19.2.4" data-path="exercises.html"><a href="exercises.html#feature-side-information"><i class="fa fa-check"></i><b>19.2.4</b> Feature side information</a></li>
<li class="chapter" data-level="19.2.5" data-path="exercises.html"><a href="exercises.html#other-components"><i class="fa fa-check"></i><b>19.2.5</b> Other components</a></li>
</ul></li>
<li class="chapter" data-level="19.3" data-path="exercises.html"><a href="exercises.html#treesummarizedexperiment-data-import"><i class="fa fa-check"></i><b>19.3</b> TreeSummarizedExperiment: data import</a></li>
<li class="chapter" data-level="19.4" data-path="exercises.html"><a href="exercises.html#data-manipulation"><i class="fa fa-check"></i><b>19.4</b> Data manipulation</a>
<ul>
<li class="chapter" data-level="19.4.1" data-path="exercises.html"><a href="exercises.html#subsetting-1"><i class="fa fa-check"></i><b>19.4.1</b> Subsetting</a></li>
<li class="chapter" data-level="19.4.2" data-path="exercises.html"><a href="exercises.html#library-sizes"><i class="fa fa-check"></i><b>19.4.2</b> Library sizes</a></li>
<li class="chapter" data-level="19.4.3" data-path="exercises.html"><a href="exercises.html#prevalent-and-core-taxonomic-features"><i class="fa fa-check"></i><b>19.4.3</b> Prevalent and core taxonomic features</a></li>
<li class="chapter" data-level="19.4.4" data-path="exercises.html"><a href="exercises.html#explore-the-data"><i class="fa fa-check"></i><b>19.4.4</b> Explore the data</a></li>
<li class="chapter" data-level="19.4.5" data-path="exercises.html"><a href="exercises.html#other-functions"><i class="fa fa-check"></i><b>19.4.5</b> Other functions</a></li>
</ul></li>
<li class="chapter" data-level="19.5" data-path="exercises.html"><a href="exercises.html#transformations"><i class="fa fa-check"></i><b>19.5</b> Transformations</a>
<ul>
<li class="chapter" data-level="19.5.1" data-path="exercises.html"><a href="exercises.html#transformations-1"><i class="fa fa-check"></i><b>19.5.1</b> Transformations</a></li>
</ul></li>
<li class="chapter" data-level="19.6" data-path="exercises.html"><a href="exercises.html#taxonomic-levels"><i class="fa fa-check"></i><b>19.6</b> Taxonomic levels</a>
<ul>
<li class="chapter" data-level="19.6.1" data-path="exercises.html"><a href="exercises.html#taxonomic-levels-1"><i class="fa fa-check"></i><b>19.6.1</b> Taxonomic levels</a></li>
<li class="chapter" data-level="19.6.2" data-path="exercises.html"><a href="exercises.html#alternative-experiments-altexp"><i class="fa fa-check"></i><b>19.6.2</b> Alternative experiments (altExp)</a></li>
</ul></li>
<li class="chapter" data-level="19.7" data-path="exercises.html"><a href="exercises.html#alpha-diversity"><i class="fa fa-check"></i><b>19.7</b> Alpha diversity</a>
<ul>
<li class="chapter" data-level="19.7.1" data-path="exercises.html"><a href="exercises.html#alpha-diversity-basics"><i class="fa fa-check"></i><b>19.7.1</b> Alpha diversity basics</a></li>
<li class="chapter" data-level="19.7.2" data-path="exercises.html"><a href="exercises.html#alpha-diversity-extra"><i class="fa fa-check"></i><b>19.7.2</b> Alpha diversity extra</a></li>
</ul></li>
<li class="chapter" data-level="19.8" data-path="exercises.html"><a href="exercises.html#community-composition"><i class="fa fa-check"></i><b>19.8</b> Community composition</a>
<ul>
<li class="chapter" data-level="19.8.1" data-path="exercises.html"><a href="exercises.html#beta-diversity-basics"><i class="fa fa-check"></i><b>19.8.1</b> Beta diversity basics</a></li>
<li class="chapter" data-level="19.8.2" data-path="exercises.html"><a href="exercises.html#beta-diversity-extra"><i class="fa fa-check"></i><b>19.8.2</b> Beta diversity extra</a></li>
</ul></li>
<li class="chapter" data-level="19.9" data-path="exercises.html"><a href="exercises.html#visualization-1"><i class="fa fa-check"></i><b>19.9</b> Visualization</a>
<ul>
<li class="chapter" data-level="19.9.1" data-path="exercises.html"><a href="exercises.html#multivariate-ordination"><i class="fa fa-check"></i><b>19.9.1</b> Multivariate ordination</a></li>
<li class="chapter" data-level="19.9.2" data-path="exercises.html"><a href="exercises.html#heatmap-visualization"><i class="fa fa-check"></i><b>19.9.2</b> Heatmap visualization</a></li>
</ul></li>
<li class="chapter" data-level="19.10" data-path="exercises.html"><a href="exercises.html#differential-abundance-1"><i class="fa fa-check"></i><b>19.10</b> Differential abundance</a>
<ul>
<li class="chapter" data-level="19.10.1" data-path="exercises.html"><a href="exercises.html#univariate-analyses"><i class="fa fa-check"></i><b>19.10.1</b> Univariate analyses</a></li>
<li class="chapter" data-level="19.10.2" data-path="exercises.html"><a href="exercises.html#differential-abundance-analysis-1"><i class="fa fa-check"></i><b>19.10.2</b> Differential abundance analysis</a></li>
</ul></li>
<li class="chapter" data-level="19.11" data-path="exercises.html"><a href="exercises.html#multiomics"><i class="fa fa-check"></i><b>19.11</b> Multiomics</a>
<ul>
<li class="chapter" data-level="19.11.1" data-path="exercises.html"><a href="exercises.html#introduction-to-multiassayexperiment-mae"><i class="fa fa-check"></i><b>19.11.1</b> Introduction to MultiAssayExperiment (MAE)</a></li>
<li class="chapter" data-level="19.11.2" data-path="exercises.html"><a href="exercises.html#introduction-to-multiomics"><i class="fa fa-check"></i><b>19.11.2</b> Introduction to multiomics</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="bibliography.html"><a href="bibliography.html"><i class="fa fa-check"></i>Bibliography</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Orchestrating Microbiome Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="clustering" class="section level1 hasAnchor" number="10">
<h1><span class="header-section-number">Chapter 10</span> Community typing (clustering)<a href="clustering.html#clustering" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<script>
document.addEventListener("click", function (event) {
    if (event.target.classList.contains("rebook-collapse")) {
        event.target.classList.toggle("active");
        var content = event.target.nextElementSibling;
        if (content.style.display === "block") {
            content.style.display = "none";
        } else {
            content.style.display = "block";
        }
    }
})
</script>
<style>
.rebook-collapse {
  background-color: #eee;
  color: #444;
  cursor: pointer;
  padding: 18px;
  width: 100%;
  border: none;
  text-align: left;
  outline: none;
  font-size: 15px;
}

.rebook-content {
  padding: 0 18px;
  display: none;
  overflow: hidden;
  background-color: #f1f1f1;
}
</style>
<div class="sourceCode" id="cb263"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb263-1"><a href="clustering.html#cb263-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mia)</span>
<span id="cb263-2"><a href="clustering.html#cb263-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;GlobalPatterns&quot;</span>, <span class="at">package=</span><span class="st">&quot;mia&quot;</span>)</span>
<span id="cb263-3"><a href="clustering.html#cb263-3" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> GlobalPatterns</span></code></pre></div>
<p>Clustering is an unsupervised machine learning technique. The idea of
it is to find clusters from the data. A cluster is a group of
features/samples that share pattern. For example, with clustering, we
can find group of samples that share similar community
composition. There are multiple clustering algorithms available.</p>
<div id="hiearchical-clustering" class="section level2 hasAnchor" number="10.1">
<h2><span class="header-section-number">10.1</span> Hiearchical clustering<a href="clustering.html#hiearchical-clustering" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Hiearchical clustering is a clustering method that aims to find
hiearchy between samples/features. There are to approaches:
agglomerative (“bottom-up”) and divisive (“top-down”).</p>
<p>In agglomerative approach, each observation is first unique cluster.
Algorithm continues by agglomerating similar clusters. Divisive
approach starts with one cluster that contains all the
observations. Clusters are splitted recursively to clusters that
differ the most. Clustering ends when each cluster contains only one
observation.</p>
<p>Hiearchical clustering can be visualized with dendrogram tree. In each
splitting point, the three is divided into two clusters leading to
hierarchy.</p>
<p>Let’s load data from mia package.</p>
<div class="sourceCode" id="cb264"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb264-1"><a href="clustering.html#cb264-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mia)</span>
<span id="cb264-2"><a href="clustering.html#cb264-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vegan)</span>
<span id="cb264-3"><a href="clustering.html#cb264-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb264-4"><a href="clustering.html#cb264-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Load experimental data</span></span>
<span id="cb264-5"><a href="clustering.html#cb264-5" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(peerj13075)</span>
<span id="cb264-6"><a href="clustering.html#cb264-6" aria-hidden="true" tabindex="-1"></a>(tse <span class="ot">&lt;-</span> peerj13075)</span></code></pre></div>
<pre><code>## class: TreeSummarizedExperiment 
## dim: 674 58 
## metadata(0):
## assays(1): counts
## rownames(674): OTU1 OTU2 ... OTU2567 OTU2569
## rowData names(6): kingdom phylum ... family genus
## colnames(58): ID1 ID2 ... ID57 ID58
## colData names(5): Sample Geographical_location Gender Age Diet
## reducedDimNames(0):
## mainExpName: NULL
## altExpNames(0):
## rowLinks: NULL
## rowTree: NULL
## colLinks: NULL
## colTree: NULL</code></pre>
<p>Hierarchical clustering requires 2 steps. In the fist step, dissimilarities are
calculated. In prior to that, data transformation is applied if needed. Since
sequencing data is compositional, relative transformation is applied.
In the second step, clustering is performed based on dissimilarities.</p>
<div class="sourceCode" id="cb266"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb266-1"><a href="clustering.html#cb266-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>( <span class="sc">!</span><span class="fu">require</span>(NbClust) ){<span class="fu">install.packages</span>(<span class="st">&quot;NbClust&quot;</span>); <span class="fu">library</span>(NbClust)}</span>
<span id="cb266-2"><a href="clustering.html#cb266-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>( <span class="sc">!</span><span class="fu">require</span>(cobiclust) ){<span class="fu">install.packages</span>(<span class="st">&quot;cobiclust&quot;</span>); <span class="fu">library</span>(cobiclust)}</span>
<span id="cb266-3"><a href="clustering.html#cb266-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb266-4"><a href="clustering.html#cb266-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply transformation</span></span>
<span id="cb266-5"><a href="clustering.html#cb266-5" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">transformCounts</span>(tse, <span class="at">method =</span> <span class="st">&quot;relabundance&quot;</span>)</span>
<span id="cb266-6"><a href="clustering.html#cb266-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the assay</span></span>
<span id="cb266-7"><a href="clustering.html#cb266-7" aria-hidden="true" tabindex="-1"></a>assay <span class="ot">&lt;-</span> <span class="fu">assay</span>(tse, <span class="st">&quot;relabundance&quot;</span>)</span>
<span id="cb266-8"><a href="clustering.html#cb266-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Transpose assay --&gt; samples are now in rows --&gt; we are clustering samples</span></span>
<span id="cb266-9"><a href="clustering.html#cb266-9" aria-hidden="true" tabindex="-1"></a>assay <span class="ot">&lt;-</span> <span class="fu">t</span>(assay)</span>
<span id="cb266-10"><a href="clustering.html#cb266-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb266-11"><a href="clustering.html#cb266-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate distances</span></span>
<span id="cb266-12"><a href="clustering.html#cb266-12" aria-hidden="true" tabindex="-1"></a>diss <span class="ot">&lt;-</span> <span class="fu">vegdist</span>(assay, <span class="at">method =</span> <span class="st">&quot;bray&quot;</span>)</span>
<span id="cb266-13"><a href="clustering.html#cb266-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb266-14"><a href="clustering.html#cb266-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform hierarchical clustering</span></span>
<span id="cb266-15"><a href="clustering.html#cb266-15" aria-hidden="true" tabindex="-1"></a>hc <span class="ot">&lt;-</span> <span class="fu">hclust</span>(diss, <span class="at">method =</span> <span class="st">&quot;complete&quot;</span>)</span>
<span id="cb266-16"><a href="clustering.html#cb266-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb266-17"><a href="clustering.html#cb266-17" aria-hidden="true" tabindex="-1"></a><span class="co"># To visualize, convert hclust object into dendrogram object</span></span>
<span id="cb266-18"><a href="clustering.html#cb266-18" aria-hidden="true" tabindex="-1"></a>dendro <span class="ot">&lt;-</span> <span class="fu">as.dendrogram</span>(hc)</span>
<span id="cb266-19"><a href="clustering.html#cb266-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb266-20"><a href="clustering.html#cb266-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot dendrogram</span></span>
<span id="cb266-21"><a href="clustering.html#cb266-21" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(dendro)</span></code></pre></div>
<p><img src="24_clustering_files/figure-html/hclust2-1.png" width="672" /></p>
<p>We can use dendrogram to determine the number of clusters. Usually the
tree is splitted where the branch length is the largest. However, as
we can see from the dendrogram, clusters are not clear. Algorithms are
available to identify the optimal number of clusters.</p>
<div class="sourceCode" id="cb267"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb267-1"><a href="clustering.html#cb267-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Determine the optimal number of clusters</span></span>
<span id="cb267-2"><a href="clustering.html#cb267-2" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">NbClust</span>(<span class="at">diss =</span> diss, <span class="at">distance =</span> <span class="cn">NULL</span>, <span class="at">method =</span> <span class="st">&quot;ward.D2&quot;</span>,</span>
<span id="cb267-3"><a href="clustering.html#cb267-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">index =</span> <span class="st">&quot;silhouette&quot;</span>)</span></code></pre></div>
<pre><code>## 
##  Only frey, mcclain, cindex, sihouette and dunn can be computed. To compute the other indices, data matrix is needed</code></pre>
<div class="sourceCode" id="cb269"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb269-1"><a href="clustering.html#cb269-1" aria-hidden="true" tabindex="-1"></a>res<span class="sc">$</span>Best.nc</span></code></pre></div>
<pre><code>## Number_clusters     Value_Index 
##         15.0000          0.4543</code></pre>
<p>Based on the result, let’s divide observations into 15 clusters.</p>
<div class="sourceCode" id="cb271"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb271-1"><a href="clustering.html#cb271-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>( <span class="sc">!</span><span class="fu">require</span>(dendextend) ){</span>
<span id="cb271-2"><a href="clustering.html#cb271-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;dendextend&quot;</span>)</span>
<span id="cb271-3"><a href="clustering.html#cb271-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(dendextend)</span>
<span id="cb271-4"><a href="clustering.html#cb271-4" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb271-5"><a href="clustering.html#cb271-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb271-6"><a href="clustering.html#cb271-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Find clusters</span></span>
<span id="cb271-7"><a href="clustering.html#cb271-7" aria-hidden="true" tabindex="-1"></a><span class="fu">cutree</span>(hc, <span class="at">k =</span> <span class="dv">15</span>) </span></code></pre></div>
<pre><code>##  ID1  ID2  ID3  ID4  ID5  ID6  ID7  ID8  ID9 ID10 ID11 ID12 ID13 ID14 ID15 ID16 
##    1    2    3    4    5    5    2    2    2    6    5    5    2    5    2    5 
## ID17 ID18 ID19 ID20 ID21 ID22 ID23 ID24 ID25 ID26 ID27 ID28 ID29 ID30 ID31 ID32 
##    5    4    5    5    7    8    9    5    3    1    8    8    8    3    1    2 
## ID33 ID34 ID35 ID36 ID37 ID38 ID39 ID40 ID41 ID42 ID43 ID44 ID45 ID46 ID47 ID48 
##    2    2    2    2    2    9   10   11    6    5    4    2    9    1   12    2 
## ID49 ID50 ID51 ID52 ID53 ID54 ID55 ID56 ID57 ID58 
##    4    4    2    8    7   13   14    3    8   15</code></pre>
<div class="sourceCode" id="cb273"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb273-1"><a href="clustering.html#cb273-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Making colors for 6 clusters</span></span>
<span id="cb273-2"><a href="clustering.html#cb273-2" aria-hidden="true" tabindex="-1"></a>col_val_map <span class="ot">&lt;-</span> randomcoloR<span class="sc">::</span><span class="fu">distinctColorPalette</span>(<span class="dv">15</span>) <span class="sc">%&gt;%</span></span>
<span id="cb273-3"><a href="clustering.html#cb273-3" aria-hidden="true" tabindex="-1"></a>     <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> <span class="fu">setNames</span>(<span class="fu">paste0</span>(<span class="st">&quot;clust_&quot;</span>,<span class="fu">seq</span>(<span class="dv">15</span>)))</span>
<span id="cb273-4"><a href="clustering.html#cb273-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb273-5"><a href="clustering.html#cb273-5" aria-hidden="true" tabindex="-1"></a>dend <span class="ot">&lt;-</span> <span class="fu">color_branches</span>(dendro, <span class="at">k=</span><span class="dv">15</span>, <span class="at">col=</span><span class="fu">unlist</span>(col_val_map))</span>
<span id="cb273-6"><a href="clustering.html#cb273-6" aria-hidden="true" tabindex="-1"></a><span class="fu">labels</span>(dend) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb273-7"><a href="clustering.html#cb273-7" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(dend)</span></code></pre></div>
<p><img src="24_clustering_files/figure-html/hclust4-1.png" width="672" /></p>
</div>
<div id="k-means-clustering" class="section level2 hasAnchor" number="10.2">
<h2><span class="header-section-number">10.2</span> K-means clustering<a href="clustering.html#k-means-clustering" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Hierarchical clustering did not yield clusters. Let’s try k-means
clustering instead. Here observations are divided into clusters so
that the distances between observations and cluster centers are
minimized; an observation belongs to cluster whose center is the
nearest.</p>
<p>The algorithm starts by dividing observation to random clusters whose
number is defined by user. The centroids of clusters are then
calculated. After that, observations’ allocation to clusters are
updated so that the means are minimized. Again, centroid are
calculated, and algorithm continues iteratively until the assignments
do not change.</p>
<p>The number of clusters can be determined based on algorithm. Here we
utilize silhouette analysis.</p>
<div class="sourceCode" id="cb274"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb274-1"><a href="clustering.html#cb274-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>( <span class="sc">!</span><span class="fu">require</span>(factoextra) ){</span>
<span id="cb274-2"><a href="clustering.html#cb274-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;factoextra&quot;</span>)</span>
<span id="cb274-3"><a href="clustering.html#cb274-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(factoextra)</span>
<span id="cb274-4"><a href="clustering.html#cb274-4" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb274-5"><a href="clustering.html#cb274-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb274-6"><a href="clustering.html#cb274-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert dist object into matrix</span></span>
<span id="cb274-7"><a href="clustering.html#cb274-7" aria-hidden="true" tabindex="-1"></a>diss <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(diss)</span>
<span id="cb274-8"><a href="clustering.html#cb274-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform silhouette analysis and plot the result</span></span>
<span id="cb274-9"><a href="clustering.html#cb274-9" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_nbclust</span>(diss, kmeans, <span class="at">method =</span> <span class="st">&quot;silhouette&quot;</span>)</span></code></pre></div>
<p><img src="24_clustering_files/figure-html/kmeans1-1.png" width="672" /></p>
<p>Based on the result of silhouette analysis, we choose 3 to be the number of clusters
in k-means clustering.</p>
<div class="sourceCode" id="cb275"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb275-1"><a href="clustering.html#cb275-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scater)</span>
<span id="cb275-2"><a href="clustering.html#cb275-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb275-3"><a href="clustering.html#cb275-3" aria-hidden="true" tabindex="-1"></a><span class="co"># The first step is random, add seed for reproducibility</span></span>
<span id="cb275-4"><a href="clustering.html#cb275-4" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">15463</span>)</span>
<span id="cb275-5"><a href="clustering.html#cb275-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform k-means clustering with 3 clusters</span></span>
<span id="cb275-6"><a href="clustering.html#cb275-6" aria-hidden="true" tabindex="-1"></a>km <span class="ot">&lt;-</span> <span class="fu">kmeans</span>(diss, <span class="dv">3</span>, <span class="at">nstart =</span> <span class="dv">25</span>)</span>
<span id="cb275-7"><a href="clustering.html#cb275-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Add the result to colData</span></span>
<span id="cb275-8"><a href="clustering.html#cb275-8" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(tse)<span class="sc">$</span>clusters <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(km<span class="sc">$</span>cluster)</span>
<span id="cb275-9"><a href="clustering.html#cb275-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb275-10"><a href="clustering.html#cb275-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform PCoA so that we can visualize clusters</span></span>
<span id="cb275-11"><a href="clustering.html#cb275-11" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">runMDS</span>(tse, <span class="at">exprs_values =</span> <span class="st">&quot;relabundance&quot;</span>, <span class="at">FUN =</span> vegan<span class="sc">::</span>vegdist, <span class="at">method =</span> <span class="st">&quot;bray&quot;</span>)</span>
<span id="cb275-12"><a href="clustering.html#cb275-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb275-13"><a href="clustering.html#cb275-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot PCoA and color clusters</span></span>
<span id="cb275-14"><a href="clustering.html#cb275-14" aria-hidden="true" tabindex="-1"></a><span class="fu">plotReducedDim</span>(tse, <span class="st">&quot;MDS&quot;</span>, <span class="at">colour_by =</span> <span class="st">&quot;clusters&quot;</span>)</span></code></pre></div>
<p><img src="24_clustering_files/figure-html/kmeans2-1.png" width="672" /></p>
</div>
<div id="dirichlet-multinomial-mixtures-dmm" class="section level2 hasAnchor" number="10.3">
<h2><span class="header-section-number">10.3</span> Dirichlet Multinomial Mixtures (DMM)<a href="clustering.html#dirichlet-multinomial-mixtures-dmm" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This section focus on DMM analysis.</p>
<p>One technique that allows to search for groups of samples that are
similar to each other is the <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0030126">Dirichlet-Multinomial Mixture
Model</a>. In
DMM, we first determine the number of clusters (k) that best fit the
data (model evidence) using Laplace approximation. After fitting the
model with k clusters, we obtain for each sample k probabilities that
reflect the probability that a sample belongs to the given cluster.</p>
<p>Let’s cluster the data with DMM clustering.</p>
<div class="sourceCode" id="cb276"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb276-1"><a href="clustering.html#cb276-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Runs model and calculates the most likely number of clusters from 1 to 7.</span></span>
<span id="cb276-2"><a href="clustering.html#cb276-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Since this is a large dataset it takes long computational time.</span></span>
<span id="cb276-3"><a href="clustering.html#cb276-3" aria-hidden="true" tabindex="-1"></a><span class="co"># For this reason we use only a subset of the data; agglomerated by Phylum as a rank.</span></span>
<span id="cb276-4"><a href="clustering.html#cb276-4" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> GlobalPatterns</span>
<span id="cb276-5"><a href="clustering.html#cb276-5" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">agglomerateByRank</span>(tse, <span class="at">rank =</span> <span class="st">&quot;Phylum&quot;</span>, <span class="at">agglomerateTree=</span><span class="cn">TRUE</span>)</span></code></pre></div>
<div class="sourceCode" id="cb277"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb277-1"><a href="clustering.html#cb277-1" aria-hidden="true" tabindex="-1"></a>tse_dmn <span class="ot">&lt;-</span> mia<span class="sc">::</span><span class="fu">runDMN</span>(tse, <span class="at">name =</span> <span class="st">&quot;DMN&quot;</span>, <span class="at">k =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">7</span>)</span></code></pre></div>
<div class="sourceCode" id="cb278"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb278-1"><a href="clustering.html#cb278-1" aria-hidden="true" tabindex="-1"></a><span class="co"># It is stored in metadata</span></span>
<span id="cb278-2"><a href="clustering.html#cb278-2" aria-hidden="true" tabindex="-1"></a>tse_dmn</span></code></pre></div>
<pre><code>## class: TreeSummarizedExperiment 
## dim: 67 26 
## metadata(2): agglomerated_by_rank DMN
## assays(1): counts
## rownames(67): Phylum:Crenarchaeota Phylum:Euryarchaeota ...
##   Phylum:Synergistetes Phylum:SR1
## rowData names(7): Kingdom Phylum ... Genus Species
## colnames(26): CL3 CC1 ... Even2 Even3
## colData names(7): X.SampleID Primer ... SampleType Description
## reducedDimNames(0):
## mainExpName: NULL
## altExpNames(0):
## rowLinks: a LinkDataFrame (67 rows)
## rowTree: 1 phylo tree(s) (66 leaves)
## colLinks: NULL
## colTree: NULL</code></pre>
<p>Return information on metadata that the object contains.</p>
<div class="sourceCode" id="cb280"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb280-1"><a href="clustering.html#cb280-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(<span class="fu">metadata</span>(tse_dmn))</span></code></pre></div>
<pre><code>## [1] &quot;agglomerated_by_rank&quot; &quot;DMN&quot;</code></pre>
<p>This returns a list of DMN objects for a closer investigation.</p>
<div class="sourceCode" id="cb282"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb282-1"><a href="clustering.html#cb282-1" aria-hidden="true" tabindex="-1"></a><span class="fu">getDMN</span>(tse_dmn)</span></code></pre></div>
<pre><code>## [[1]]
## class: DMN 
## k: 1 
## samples x taxa: 26 x 67 
## Laplace: 7715 BIC: 7802 AIC: 7760 
## 
## [[2]]
## class: DMN 
## k: 2 
## samples x taxa: 26 x 67 
## Laplace: 7673 BIC: 7927 AIC: 7842 
## 
## [[3]]
## class: DMN 
## k: 3 
## samples x taxa: 26 x 67 
## Laplace: 7690 BIC: 8076 AIC: 7948 
## 
## [[4]]
## class: DMN 
## k: 4 
## samples x taxa: 26 x 67 
## Laplace: 7752 BIC: 8274 AIC: 8103 
## 
## [[5]]
## class: DMN 
## k: 5 
## samples x taxa: 26 x 67 
## Laplace: 7854 BIC: 8553 AIC: 8340 
## 
## [[6]]
## class: DMN 
## k: 6 
## samples x taxa: 26 x 67 
## Laplace: 7926 BIC: 8796 AIC: 8540 
## 
## [[7]]
## class: DMN 
## k: 7 
## samples x taxa: 26 x 67 
## Laplace: 8003 BIC: 9051 AIC: 8752</code></pre>
<p>Show Laplace approximation (model evidence) for each model of the k models.</p>
<div class="sourceCode" id="cb284"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb284-1"><a href="clustering.html#cb284-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(miaViz)</span>
<span id="cb284-2"><a href="clustering.html#cb284-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDMNFit</span>(tse_dmn, <span class="at">type =</span> <span class="st">&quot;laplace&quot;</span>)</span></code></pre></div>
<p><img src="24_clustering_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<p>Return the model that has the best fit.</p>
<div class="sourceCode" id="cb285"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb285-1"><a href="clustering.html#cb285-1" aria-hidden="true" tabindex="-1"></a><span class="fu">getBestDMNFit</span>(tse_dmn, <span class="at">type =</span> <span class="st">&quot;laplace&quot;</span>)</span></code></pre></div>
<pre><code>## class: DMN 
## k: 2 
## samples x taxa: 26 x 67 
## Laplace: 7673 BIC: 7927 AIC: 7842</code></pre>
<div id="pcoa-for-asv-level-data-with-bray-curtis-with-dmm-clusters-shown-with-colors" class="section level3 hasAnchor" number="10.3.1">
<h3><span class="header-section-number">10.3.1</span> PCoA for ASV-level data with Bray-Curtis; with DMM clusters shown with colors<a href="clustering.html#pcoa-for-asv-level-data-with-bray-curtis-with-dmm-clusters-shown-with-colors" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Group samples and return DMNGroup object that contains a summary.
Patient status is used for grouping.</p>
<div class="sourceCode" id="cb287"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb287-1"><a href="clustering.html#cb287-1" aria-hidden="true" tabindex="-1"></a>dmn_group <span class="ot">&lt;-</span> <span class="fu">calculateDMNgroup</span>(tse_dmn, <span class="at">variable =</span> <span class="st">&quot;SampleType&quot;</span>,  <span class="at">exprs_values =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb287-2"><a href="clustering.html#cb287-2" aria-hidden="true" tabindex="-1"></a>                               <span class="at">k =</span> <span class="dv">2</span>, <span class="at">seed=</span>.Machine<span class="sc">$</span>integer.max)</span>
<span id="cb287-3"><a href="clustering.html#cb287-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb287-4"><a href="clustering.html#cb287-4" aria-hidden="true" tabindex="-1"></a>dmn_group</span></code></pre></div>
<pre><code>## class: DMNGroup 
## summary:
##                    k samples taxa    NLE  LogDet Laplace    BIC  AIC
## Feces              2       4   67 1078.3 -106.22   901.1 1171.9 1213
## Freshwater         2       2   67  889.6  -97.21   716.9  936.4 1025
## Freshwater (creek) 2       3   67 1600.3  860.38  1906.4 1674.5 1735
## Mock               2       3   67  980.2  110.61   911.4 1054.4 1115
## Ocean              2       3   67 1096.7  -56.93   944.2 1170.9 1232
## Sediment (estuary) 2       3   67 1195.5   18.63  1080.8 1269.7 1331
## Skin               2       3   67  992.6  -84.93   826.1 1066.8 1128
## Soil               2       3   67 1380.3   11.20  1261.8 1454.5 1515
## Tongue             2       2   67  783.0 -107.78   605.1  829.8  918</code></pre>
<p>Mixture weights (rough measure of the cluster size).</p>
<div class="sourceCode" id="cb289"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb289-1"><a href="clustering.html#cb289-1" aria-hidden="true" tabindex="-1"></a>DirichletMultinomial<span class="sc">::</span><span class="fu">mixturewt</span>(<span class="fu">getBestDMNFit</span>(tse_dmn))</span></code></pre></div>
<pre><code>##       pi theta
## 1 0.5385 20.60
## 2 0.4615 15.32</code></pre>
<p>Samples-cluster assignment probabilities / how probable it is that sample belongs
to each cluster</p>
<div class="sourceCode" id="cb291"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb291-1"><a href="clustering.html#cb291-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DirichletMultinomial<span class="sc">::</span><span class="fu">mixture</span>(<span class="fu">getBestDMNFit</span>(tse_dmn)))</span></code></pre></div>
<pre><code>##              [,1]      [,2]
## CL3     1.000e+00 4.562e-17
## CC1     1.000e+00 3.449e-22
## SV1     1.000e+00 1.794e-12
## M31Fcsw 6.909e-26 1.000e+00
## M11Fcsw 1.028e-16 1.000e+00
## M31Plmr 1.024e-13 1.000e+00</code></pre>
<p>Contribution of each taxa to each component</p>
<div class="sourceCode" id="cb293"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb293-1"><a href="clustering.html#cb293-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DirichletMultinomial<span class="sc">::</span><span class="fu">fitted</span>(<span class="fu">getBestDMNFit</span>(tse_dmn)))</span></code></pre></div>
<pre><code>##                         [,1]      [,2]
## Phylum:Crenarchaeota  0.3043 0.1354082
## Phylum:Euryarchaeota  0.2314 0.1468945
## Phylum:Actinobacteria 1.2107 1.0581104
## Phylum:Spirochaetes   0.2141 0.1318102
## Phylum:MVP-15         0.0299 0.0007698
## Phylum:Proteobacteria 6.8408 1.8113562</code></pre>
<p>Get the assignment probabilities</p>
<div class="sourceCode" id="cb295"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb295-1"><a href="clustering.html#cb295-1" aria-hidden="true" tabindex="-1"></a>prob <span class="ot">&lt;-</span> DirichletMultinomial<span class="sc">::</span><span class="fu">mixture</span>(<span class="fu">getBestDMNFit</span>(tse_dmn))</span>
<span id="cb295-2"><a href="clustering.html#cb295-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Add column names</span></span>
<span id="cb295-3"><a href="clustering.html#cb295-3" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(prob) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;comp1&quot;</span>, <span class="st">&quot;comp2&quot;</span>)</span>
<span id="cb295-4"><a href="clustering.html#cb295-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb295-5"><a href="clustering.html#cb295-5" aria-hidden="true" tabindex="-1"></a><span class="co"># For each row, finds column that has the highest value. Then extract the column </span></span>
<span id="cb295-6"><a href="clustering.html#cb295-6" aria-hidden="true" tabindex="-1"></a><span class="co"># names of highest values.</span></span>
<span id="cb295-7"><a href="clustering.html#cb295-7" aria-hidden="true" tabindex="-1"></a>vec <span class="ot">&lt;-</span> <span class="fu">colnames</span>(prob)[<span class="fu">max.col</span>(prob,<span class="at">ties.method =</span> <span class="st">&quot;first&quot;</span>)]</span></code></pre></div>
<p>Computing the euclidean PCoA and storing it as a data frame</p>
<div class="sourceCode" id="cb296"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb296-1"><a href="clustering.html#cb296-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Does clr transformation. Pseudocount is added, because data contains zeros.</span></span>
<span id="cb296-2"><a href="clustering.html#cb296-2" aria-hidden="true" tabindex="-1"></a><span class="fu">assay</span>(tse, <span class="st">&quot;pseudo&quot;</span>) <span class="ot">&lt;-</span> <span class="fu">assay</span>(tse, <span class="st">&quot;counts&quot;</span>) <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb296-3"><a href="clustering.html#cb296-3" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">transformCounts</span>(tse, <span class="at">assay_name =</span> <span class="st">&quot;pseudo&quot;</span>, <span class="at">method =</span> <span class="st">&quot;relabundance&quot;</span>)</span>
<span id="cb296-4"><a href="clustering.html#cb296-4" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">transformCounts</span>(tse, <span class="st">&quot;relabundance&quot;</span>, <span class="at">method =</span> <span class="st">&quot;clr&quot;</span>)</span>
<span id="cb296-5"><a href="clustering.html#cb296-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb296-6"><a href="clustering.html#cb296-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scater)</span>
<span id="cb296-7"><a href="clustering.html#cb296-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb296-8"><a href="clustering.html#cb296-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Does principal coordinate analysis</span></span>
<span id="cb296-9"><a href="clustering.html#cb296-9" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">calculateMDS</span>(tse, <span class="at">exprs_values =</span> <span class="st">&quot;clr&quot;</span>, <span class="at">method =</span> <span class="st">&quot;euclidean&quot;</span>)</span>
<span id="cb296-10"><a href="clustering.html#cb296-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb296-11"><a href="clustering.html#cb296-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates a data frame from principal coordinates</span></span>
<span id="cb296-12"><a href="clustering.html#cb296-12" aria-hidden="true" tabindex="-1"></a>euclidean_pcoa_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">pcoa1 =</span> df[,<span class="dv">1</span>], </span>
<span id="cb296-13"><a href="clustering.html#cb296-13" aria-hidden="true" tabindex="-1"></a>                                <span class="at">pcoa2 =</span> df[,<span class="dv">2</span>])</span></code></pre></div>
<div class="sourceCode" id="cb297"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb297-1"><a href="clustering.html#cb297-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates a data frame that contains principal coordinates and DMM information</span></span>
<span id="cb297-2"><a href="clustering.html#cb297-2" aria-hidden="true" tabindex="-1"></a>euclidean_dmm_pcoa_df <span class="ot">&lt;-</span> <span class="fu">cbind</span>(euclidean_pcoa_df,</span>
<span id="cb297-3"><a href="clustering.html#cb297-3" aria-hidden="true" tabindex="-1"></a>                               <span class="at">dmm_component =</span> vec)</span>
<span id="cb297-4"><a href="clustering.html#cb297-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates a plot</span></span>
<span id="cb297-5"><a href="clustering.html#cb297-5" aria-hidden="true" tabindex="-1"></a>euclidean_dmm_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> euclidean_dmm_pcoa_df, </span>
<span id="cb297-6"><a href="clustering.html#cb297-6" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">aes</span>(<span class="at">x=</span>pcoa1, <span class="at">y=</span>pcoa2,</span>
<span id="cb297-7"><a href="clustering.html#cb297-7" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">color =</span> dmm_component)) <span class="sc">+</span></span>
<span id="cb297-8"><a href="clustering.html#cb297-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb297-9"><a href="clustering.html#cb297-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Coordinate 1&quot;</span>,</span>
<span id="cb297-10"><a href="clustering.html#cb297-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">&quot;Coordinate 2&quot;</span>,</span>
<span id="cb297-11"><a href="clustering.html#cb297-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">&quot;PCoA with Aitchison distances&quot;</span>) <span class="sc">+</span>  </span>
<span id="cb297-12"><a href="clustering.html#cb297-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>)) <span class="co"># makes titles smaller</span></span>
<span id="cb297-13"><a href="clustering.html#cb297-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb297-14"><a href="clustering.html#cb297-14" aria-hidden="true" tabindex="-1"></a>euclidean_dmm_plot</span></code></pre></div>
<p><img src="24_clustering_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
</div>
</div>
<div id="community-detection" class="section level2 hasAnchor" number="10.4">
<h2><span class="header-section-number">10.4</span> Community Detection<a href="clustering.html#community-detection" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Another approach for discovering communities within the samples of the
data, is to run community detection algorithms after building a
graph. The following demonstration builds a graph based on the k
nearest-neighbors and performs the community detection on the fly.</p>
<p><em><code>bluster</code></em> <span class="citation">(<a href="#ref-R-bluster" role="doc-biblioref">Lun 2021</a>)</span> package offers several clustering methods,
among which graph-based are present, enabling the community detection
task.</p>
<p>Installing package:</p>
<div class="sourceCode" id="cb298"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb298-1"><a href="clustering.html#cb298-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(bluster)){</span>
<span id="cb298-2"><a href="clustering.html#cb298-2" aria-hidden="true" tabindex="-1"></a>  BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;bluster&quot;</span>)</span>
<span id="cb298-3"><a href="clustering.html#cb298-3" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p>The algorithm used is “short random walks” <span class="citation">(<a href="#ref-Pons2006" role="doc-biblioref">Pons and Latapy 2006</a>)</span>. Graph is
constructed using different k values (the number of nearest neighbors
to consider during graph construction) using the robust centered log
ratio (rclr) assay data. Then plotting the communities using UMAP
<span class="citation">(<a href="#ref-McInnes2018" role="doc-biblioref">McInnes, Healy, and Melville 2018</a>)</span> ordination as a visual exploration aid. In the
following demonstration we use the <code>enterotype</code> dataset from the
<span class="citation">(<a href="#ref-R-mia" role="doc-biblioref">Ernst, Shetty, and Lahti 2020</a>)</span> package.</p>
<div class="sourceCode" id="cb299"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb299-1"><a href="clustering.html#cb299-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(bluster)</span>
<span id="cb299-2"><a href="clustering.html#cb299-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork) <span class="co"># For arranging several plots as a grid</span></span>
<span id="cb299-3"><a href="clustering.html#cb299-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scater)</span>
<span id="cb299-4"><a href="clustering.html#cb299-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb299-5"><a href="clustering.html#cb299-5" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;enterotype&quot;</span>, <span class="at">package=</span><span class="st">&quot;mia&quot;</span>)</span>
<span id="cb299-6"><a href="clustering.html#cb299-6" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> enterotype</span>
<span id="cb299-7"><a href="clustering.html#cb299-7" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">transformCounts</span>(tse, <span class="at">method =</span> <span class="st">&quot;rclr&quot;</span>)</span>
<span id="cb299-8"><a href="clustering.html#cb299-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb299-9"><a href="clustering.html#cb299-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Performing and storing UMAP</span></span>
<span id="cb299-10"><a href="clustering.html#cb299-10" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">runUMAP</span>(tse, <span class="at">name=</span><span class="st">&quot;UMAP&quot;</span>, <span class="at">exprs_values=</span><span class="st">&quot;rclr&quot;</span>)</span>
<span id="cb299-11"><a href="clustering.html#cb299-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb299-12"><a href="clustering.html#cb299-12" aria-hidden="true" tabindex="-1"></a>k <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">5</span>,<span class="dv">10</span>)</span>
<span id="cb299-13"><a href="clustering.html#cb299-13" aria-hidden="true" tabindex="-1"></a>ClustAndPlot <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb299-14"><a href="clustering.html#cb299-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Creating the graph and running the short random walks algorithm  </span></span>
<span id="cb299-15"><a href="clustering.html#cb299-15" aria-hidden="true" tabindex="-1"></a>  graph_clusters <span class="ot">&lt;-</span> <span class="fu">clusterRows</span>(<span class="fu">t</span>(<span class="fu">assays</span>(tse)<span class="sc">$</span>rclr), <span class="fu">NNGraphParam</span>(<span class="at">k=</span>x))</span>
<span id="cb299-16"><a href="clustering.html#cb299-16" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb299-17"><a href="clustering.html#cb299-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Results of the clustering as a color for each sample</span></span>
<span id="cb299-18"><a href="clustering.html#cb299-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plotUMAP</span>(tse, <span class="at">colour_by =</span> <span class="fu">I</span>(graph_clusters)) <span class="sc">+</span></span>
<span id="cb299-19"><a href="clustering.html#cb299-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste0</span>(<span class="st">&quot;k = &quot;</span>, x))</span>
<span id="cb299-20"><a href="clustering.html#cb299-20" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb299-21"><a href="clustering.html#cb299-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb299-22"><a href="clustering.html#cb299-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Applying the function for different k values</span></span>
<span id="cb299-23"><a href="clustering.html#cb299-23" aria-hidden="true" tabindex="-1"></a>plots <span class="ot">&lt;-</span> <span class="fu">lapply</span>(k,ClustAndPlot)</span>
<span id="cb299-24"><a href="clustering.html#cb299-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb299-25"><a href="clustering.html#cb299-25" aria-hidden="true" tabindex="-1"></a><span class="co"># Displaying plots in a grid</span></span>
<span id="cb299-26"><a href="clustering.html#cb299-26" aria-hidden="true" tabindex="-1"></a>(plots[[<span class="dv">1</span>]] <span class="sc">+</span> plots[[<span class="dv">2</span>]]) <span class="sc">/</span> (plots[[<span class="dv">3</span>]] <span class="sc">+</span> plots[[<span class="dv">4</span>]])</span></code></pre></div>
<p><img src="24_clustering_files/figure-html/unnamed-chunk-14-1.png" width="672" /></p>
<p>Similarly, the <em><code>bluster</code></em> <span class="citation">(<a href="#ref-R-bluster" role="doc-biblioref">Lun 2021</a>)</span> package offers clustering
diagnostics that can be used for judging the clustering quality (see
<a href="http://bioconductor.org/packages/release/bioc/vignettes/bluster/inst/doc/diagnostics.html">Assorted clustering
diagnostics</a>).
In the following, Silhouette width as a diagnostic tool is computed
and results are visualized for each case presented earlier. For more
about Silhouettes read <span class="citation">(<a href="#ref-Rousseeuw1987" role="doc-biblioref">Rousseeuw 1987</a>)</span>.</p>
<div class="sourceCode" id="cb300"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb300-1"><a href="clustering.html#cb300-1" aria-hidden="true" tabindex="-1"></a>ClustDiagPlot <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb300-2"><a href="clustering.html#cb300-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Getting the clustering results</span></span>
<span id="cb300-3"><a href="clustering.html#cb300-3" aria-hidden="true" tabindex="-1"></a>  graph_clusters <span class="ot">&lt;-</span> <span class="fu">clusterRows</span>(<span class="fu">t</span>(<span class="fu">assays</span>(tse)<span class="sc">$</span>rclr), <span class="fu">NNGraphParam</span>(<span class="at">k=</span>x))</span>
<span id="cb300-4"><a href="clustering.html#cb300-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb300-5"><a href="clustering.html#cb300-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Computing the diagnostic info</span></span>
<span id="cb300-6"><a href="clustering.html#cb300-6" aria-hidden="true" tabindex="-1"></a>  sil <span class="ot">&lt;-</span> <span class="fu">approxSilhouette</span>(<span class="fu">t</span>(<span class="fu">assays</span>(tse)<span class="sc">$</span>rclr), graph_clusters)</span>
<span id="cb300-7"><a href="clustering.html#cb300-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb300-8"><a href="clustering.html#cb300-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plotting as a boxlpot to observe cluster separation</span></span>
<span id="cb300-9"><a href="clustering.html#cb300-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">boxplot</span>(<span class="fu">split</span>(sil<span class="sc">$</span>width, graph_clusters), <span class="at">main=</span><span class="fu">paste0</span>(<span class="st">&quot;k = &quot;</span>, x))</span>
<span id="cb300-10"><a href="clustering.html#cb300-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb300-11"><a href="clustering.html#cb300-11" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb300-12"><a href="clustering.html#cb300-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Applying the function for different k values</span></span>
<span id="cb300-13"><a href="clustering.html#cb300-13" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">lapply</span>(k,ClustDiagPlot)</span></code></pre></div>
<p><img src="24_clustering_files/figure-html/unnamed-chunk-15-1.png" width="672" /><img src="24_clustering_files/figure-html/unnamed-chunk-15-2.png" width="672" /><img src="24_clustering_files/figure-html/unnamed-chunk-15-3.png" width="672" /><img src="24_clustering_files/figure-html/unnamed-chunk-15-4.png" width="672" /></p>
</div>
<div id="biclustering" class="section level2 hasAnchor" number="10.5">
<h2><span class="header-section-number">10.5</span> Biclustering<a href="clustering.html#biclustering" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Biclustering methods cluster rows and columns simultaneously in order
to find subsets of correlated features/samples.</p>
<p>Here, we use following packages:</p>
<ul>
<li><a href="https://cran.r-project.org/web/packages/biclust/index.html">biclust</a></li>
<li><a href="https://besjournals.onlinelibrary.wiley.com/doi/abs/10.1111/2041-210X.13582">cobiclust</a></li>
</ul>
<p><em>cobiclust</em> is especially developed for microbiome data whereas <em>biclust</em> is more
general method. In this section, we show three different cases and example
solutions to apply biclustering to them.</p>
<ol style="list-style-type: decimal">
<li>Taxa vs samples</li>
<li>Taxa vs biomolecule/biomarker</li>
<li>Taxa vs taxa</li>
</ol>
<p>Biclusters can be visualized using heatmap or boxplot, for
instance. For checking purposes, also scatter plot might be valid
choice.</p>
<p>Check more ideas for heatmaps from chapters <a href="viz-chapter.html#viz-chapter">14</a> and
@ref(microbiome-community.</p>
<div id="taxa-vs-samples" class="section level3 hasAnchor" number="10.5.1">
<h3><span class="header-section-number">10.5.1</span> Taxa vs samples<a href="clustering.html#taxa-vs-samples" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>When you have microbial abundance matrices, we suggest to use
<em>cobiclust</em> which is designed for microbial data.</p>
<p>Load example data</p>
<div class="sourceCode" id="cb301"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb301-1"><a href="clustering.html#cb301-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mia)</span>
<span id="cb301-2"><a href="clustering.html#cb301-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;HintikkaXOData&quot;</span>)</span>
<span id="cb301-3"><a href="clustering.html#cb301-3" aria-hidden="true" tabindex="-1"></a>mae <span class="ot">&lt;-</span> HintikkaXOData</span></code></pre></div>
<p>Only the most prevalent taxa are included in analysis.</p>
<div class="sourceCode" id="cb302"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb302-1"><a href="clustering.html#cb302-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset data in the first experiment</span></span>
<span id="cb302-2"><a href="clustering.html#cb302-2" aria-hidden="true" tabindex="-1"></a>mae[[<span class="dv">1</span>]] <span class="ot">&lt;-</span> <span class="fu">subsetByPrevalentTaxa</span>(mae[[<span class="dv">1</span>]], <span class="at">rank =</span> <span class="st">&quot;Genus&quot;</span>, <span class="at">prevalence =</span> <span class="fl">0.2</span>, <span class="at">detection =</span> <span class="fl">0.001</span>)</span>
<span id="cb302-3"><a href="clustering.html#cb302-3" aria-hidden="true" tabindex="-1"></a><span class="co"># clr-transform in the first experiment</span></span>
<span id="cb302-4"><a href="clustering.html#cb302-4" aria-hidden="true" tabindex="-1"></a>mae[[<span class="dv">1</span>]] <span class="ot">&lt;-</span> <span class="fu">transformCounts</span>(mae[[<span class="dv">1</span>]], <span class="at">method =</span> <span class="st">&quot;relabundance&quot;</span>)</span>
<span id="cb302-5"><a href="clustering.html#cb302-5" aria-hidden="true" tabindex="-1"></a>mae[[<span class="dv">1</span>]] <span class="ot">&lt;-</span> <span class="fu">transformCounts</span>(mae[[<span class="dv">1</span>]], <span class="st">&quot;relabundance&quot;</span>, <span class="at">method =</span> <span class="st">&quot;rclr&quot;</span>)</span></code></pre></div>
<p><em>cobiclust</em> takes counts table as an input and gives <em>cobiclust</em> object as an output.
It includes clusters for taxa and samples.</p>
<div class="sourceCode" id="cb303"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb303-1"><a href="clustering.html#cb303-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Do clustering; use counts table´</span></span>
<span id="cb303-2"><a href="clustering.html#cb303-2" aria-hidden="true" tabindex="-1"></a>clusters <span class="ot">&lt;-</span> <span class="fu">cobiclust</span>(<span class="fu">assay</span>(mae[[<span class="dv">1</span>]], <span class="st">&quot;counts&quot;</span>))</span>
<span id="cb303-3"><a href="clustering.html#cb303-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb303-4"><a href="clustering.html#cb303-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Get clusters</span></span>
<span id="cb303-5"><a href="clustering.html#cb303-5" aria-hidden="true" tabindex="-1"></a>row_clusters <span class="ot">&lt;-</span> clusters<span class="sc">$</span>classification<span class="sc">$</span>rowclass</span>
<span id="cb303-6"><a href="clustering.html#cb303-6" aria-hidden="true" tabindex="-1"></a>col_clusters <span class="ot">&lt;-</span> clusters<span class="sc">$</span>classification<span class="sc">$</span>colclass</span>
<span id="cb303-7"><a href="clustering.html#cb303-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb303-8"><a href="clustering.html#cb303-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Add clusters to rowdata and coldata</span></span>
<span id="cb303-9"><a href="clustering.html#cb303-9" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(mae[[<span class="dv">1</span>]])<span class="sc">$</span>clusters <span class="ot">&lt;-</span> <span class="fu">factor</span>(row_clusters)</span>
<span id="cb303-10"><a href="clustering.html#cb303-10" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(mae[[<span class="dv">1</span>]])<span class="sc">$</span>clusters <span class="ot">&lt;-</span> <span class="fu">factor</span>(col_clusters)</span>
<span id="cb303-11"><a href="clustering.html#cb303-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb303-12"><a href="clustering.html#cb303-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Order data based on clusters</span></span>
<span id="cb303-13"><a href="clustering.html#cb303-13" aria-hidden="true" tabindex="-1"></a>mae[[<span class="dv">1</span>]] <span class="ot">&lt;-</span> mae[[<span class="dv">1</span>]][<span class="fu">order</span>(<span class="fu">rowData</span>(mae[[<span class="dv">1</span>]])<span class="sc">$</span>clusters), <span class="fu">order</span>(<span class="fu">colData</span>(mae[[<span class="dv">1</span>]])<span class="sc">$</span>clusters)]</span>
<span id="cb303-14"><a href="clustering.html#cb303-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb303-15"><a href="clustering.html#cb303-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Print clusters</span></span>
<span id="cb303-16"><a href="clustering.html#cb303-16" aria-hidden="true" tabindex="-1"></a>clusters<span class="sc">$</span>classification</span></code></pre></div>
<pre><code>## $rowclass
##  [1] 1 1 1 1 2 2 1 1 1 1 1 1 2 2 2 2 1 2 1 1 2 1 2 2 1 1 2 1 1 1 1 1 2 1 1 2 1 1
## [39] 1 1 1 1 1 1 1 1 1 2 1 2 1 1 1 2 1 1 1
## 
## $colclass
##  C1  C2  C3  C4  C5  C6  C7  C8  C9 C10 C11 C12 C13 C14 C15 C16 C17 C18 C19 C20 
##   1   2   2   2   2   2   2   2   2   2   2   2   2   2   2   2   2   2   2   2 
## C21 C22 C23 C24 C25 C26 C27 C28 C29 C30 C31 C32 C33 C34 C35 C36 C37 C38 C39 C40 
##   2   3   3   3   3   3   3   3   3   3   3   3   3   3   3   3   3   3   3   1</code></pre>
<p>Next we can plot clusters. Annotated heatmap is a common choice.</p>
<div class="sourceCode" id="cb305"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb305-1"><a href="clustering.html#cb305-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(pheatmap)){</span>
<span id="cb305-2"><a href="clustering.html#cb305-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;pheatmap&quot;</span>)</span>
<span id="cb305-3"><a href="clustering.html#cb305-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(pheatmap)</span>
<span id="cb305-4"><a href="clustering.html#cb305-4" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb305-5"><a href="clustering.html#cb305-5" aria-hidden="true" tabindex="-1"></a><span class="co"># z-transform for heatmap</span></span>
<span id="cb305-6"><a href="clustering.html#cb305-6" aria-hidden="true" tabindex="-1"></a>mae[[<span class="dv">1</span>]] <span class="ot">&lt;-</span> <span class="fu">transformCounts</span>(mae[[<span class="dv">1</span>]], <span class="at">assay_name =</span> <span class="st">&quot;rclr&quot;</span>,</span>
<span id="cb305-7"><a href="clustering.html#cb305-7" aria-hidden="true" tabindex="-1"></a>                            <span class="at">MARGIN =</span> <span class="st">&quot;features&quot;</span>,</span>
<span id="cb305-8"><a href="clustering.html#cb305-8" aria-hidden="true" tabindex="-1"></a>                            <span class="at">method =</span> <span class="st">&quot;z&quot;</span>, <span class="at">name =</span> <span class="st">&quot;clr_z&quot;</span>)</span>
<span id="cb305-9"><a href="clustering.html#cb305-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb305-10"><a href="clustering.html#cb305-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Create annotations. When column names are equal, they should share levels. </span></span>
<span id="cb305-11"><a href="clustering.html#cb305-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Here samples include 3 clusters, and taxa 2. That is why we have to make </span></span>
<span id="cb305-12"><a href="clustering.html#cb305-12" aria-hidden="true" tabindex="-1"></a><span class="co"># column names unique. </span></span>
<span id="cb305-13"><a href="clustering.html#cb305-13" aria-hidden="true" tabindex="-1"></a>annotation_col <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">colData</span>(mae[[<span class="dv">1</span>]])[, <span class="st">&quot;clusters&quot;</span>, <span class="at">drop =</span> F])</span>
<span id="cb305-14"><a href="clustering.html#cb305-14" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(annotation_col) <span class="ot">&lt;-</span> <span class="st">&quot;col_clusters&quot;</span></span>
<span id="cb305-15"><a href="clustering.html#cb305-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb305-16"><a href="clustering.html#cb305-16" aria-hidden="true" tabindex="-1"></a>annotation_row <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">rowData</span>(mae[[<span class="dv">1</span>]])[, <span class="st">&quot;clusters&quot;</span>, <span class="at">drop =</span> F])</span>
<span id="cb305-17"><a href="clustering.html#cb305-17" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(annotation_row) <span class="ot">&lt;-</span> <span class="st">&quot;row_clusters&quot;</span></span></code></pre></div>
<p>Plot the heatmap.</p>
<div class="sourceCode" id="cb306"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb306-1"><a href="clustering.html#cb306-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(<span class="fu">assay</span>(mae[[<span class="dv">1</span>]], <span class="st">&quot;clr_z&quot;</span>), <span class="at">cluster_rows =</span> F, <span class="at">cluster_cols =</span> F, </span>
<span id="cb306-2"><a href="clustering.html#cb306-2" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_col =</span> annotation_col,</span>
<span id="cb306-3"><a href="clustering.html#cb306-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_row =</span> annotation_row)</span></code></pre></div>
<p><img src="24_clustering_files/figure-html/cobiclust_3b-1.png" width="1344" /></p>
<p>Boxplot is commonly used to summarize the results:</p>
<div class="sourceCode" id="cb307"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb307-1"><a href="clustering.html#cb307-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(ggplot2)){</span>
<span id="cb307-2"><a href="clustering.html#cb307-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;ggplot2&quot;</span>)</span>
<span id="cb307-3"><a href="clustering.html#cb307-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(ggplot2)</span>
<span id="cb307-4"><a href="clustering.html#cb307-4" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb307-5"><a href="clustering.html#cb307-5" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(patchwork)){</span>
<span id="cb307-6"><a href="clustering.html#cb307-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;patchwork&quot;</span>)</span>
<span id="cb307-7"><a href="clustering.html#cb307-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(patchwork)</span>
<span id="cb307-8"><a href="clustering.html#cb307-8" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb307-9"><a href="clustering.html#cb307-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb307-10"><a href="clustering.html#cb307-10" aria-hidden="true" tabindex="-1"></a><span class="co"># ggplot requires data in melted format</span></span>
<span id="cb307-11"><a href="clustering.html#cb307-11" aria-hidden="true" tabindex="-1"></a>melt_assay <span class="ot">&lt;-</span> <span class="fu">meltAssay</span>(mae[[<span class="dv">1</span>]], <span class="at">assay_name =</span> <span class="st">&quot;rclr&quot;</span>, <span class="at">add_col_data =</span> T, <span class="at">add_row_data =</span> T)</span>
<span id="cb307-12"><a href="clustering.html#cb307-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb307-13"><a href="clustering.html#cb307-13" aria-hidden="true" tabindex="-1"></a><span class="co"># patchwork two plots side-by-side</span></span>
<span id="cb307-14"><a href="clustering.html#cb307-14" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(melt_assay) <span class="sc">+</span></span>
<span id="cb307-15"><a href="clustering.html#cb307-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> clusters.x, <span class="at">y =</span> rclr)) <span class="sc">+</span></span>
<span id="cb307-16"><a href="clustering.html#cb307-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Taxa clusters&quot;</span>)</span>
<span id="cb307-17"><a href="clustering.html#cb307-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb307-18"><a href="clustering.html#cb307-18" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(melt_assay) <span class="sc">+</span></span>
<span id="cb307-19"><a href="clustering.html#cb307-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> clusters.y, <span class="at">y =</span> rclr)) <span class="sc">+</span></span>
<span id="cb307-20"><a href="clustering.html#cb307-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Sample clusters&quot;</span>)</span>
<span id="cb307-21"><a href="clustering.html#cb307-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb307-22"><a href="clustering.html#cb307-22" aria-hidden="true" tabindex="-1"></a>p1 <span class="sc">+</span> p2</span></code></pre></div>
<p><img src="24_clustering_files/figure-html/cobiclust_4-1.png" width="672" /></p>
</div>
<div id="taxa-vs-biomolecules" class="section level3 hasAnchor" number="10.5.2">
<h3><span class="header-section-number">10.5.2</span> Taxa vs biomolecules<a href="clustering.html#taxa-vs-biomolecules" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Here, we analyze cross-correlation between taxa and metabolites. This
is a case, where we use <em>biclust</em> method which is suitable for numeric
matrices in general.</p>
<div class="sourceCode" id="cb308"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb308-1"><a href="clustering.html#cb308-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Samples must be in equal order </span></span>
<span id="cb308-2"><a href="clustering.html#cb308-2" aria-hidden="true" tabindex="-1"></a><span class="co"># (Only 1st experiment  was ordered in cobiclust step leading to unequal order)</span></span>
<span id="cb308-3"><a href="clustering.html#cb308-3" aria-hidden="true" tabindex="-1"></a>mae[[<span class="dv">1</span>]] <span class="ot">&lt;-</span> mae[[<span class="dv">1</span>]][ , <span class="fu">colnames</span>(mae[[<span class="dv">2</span>]]) ]</span>
<span id="cb308-4"><a href="clustering.html#cb308-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb308-5"><a href="clustering.html#cb308-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Make rownames unique since it is require by other steps</span></span>
<span id="cb308-6"><a href="clustering.html#cb308-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(mae[[<span class="dv">1</span>]]) <span class="ot">&lt;-</span> <span class="fu">make.unique</span>(<span class="fu">rownames</span>(mae[[<span class="dv">1</span>]]))</span>
<span id="cb308-7"><a href="clustering.html#cb308-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate correlations</span></span>
<span id="cb308-8"><a href="clustering.html#cb308-8" aria-hidden="true" tabindex="-1"></a>corr <span class="ot">&lt;-</span> <span class="fu">getExperimentCrossCorrelation</span>(mae, <span class="dv">1</span>, <span class="dv">2</span>, </span>
<span id="cb308-9"><a href="clustering.html#cb308-9" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">assay_name1 =</span> <span class="st">&quot;rclr&quot;</span>, </span>
<span id="cb308-10"><a href="clustering.html#cb308-10" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">assay_name2 =</span> <span class="st">&quot;nmr&quot;</span>, </span>
<span id="cb308-11"><a href="clustering.html#cb308-11" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">mode =</span> <span class="st">&quot;matrix&quot;</span>, </span>
<span id="cb308-12"><a href="clustering.html#cb308-12" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">cor_threshold =</span> <span class="fl">0.2</span>)</span></code></pre></div>
<p><em>biclust</em> takes matrix as an input and returns <em>biclust</em> object.</p>
<div class="sourceCode" id="cb309"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb309-1"><a href="clustering.html#cb309-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load package</span></span>
<span id="cb309-2"><a href="clustering.html#cb309-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(biclust)){</span>
<span id="cb309-3"><a href="clustering.html#cb309-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;biclust&quot;</span>)</span>
<span id="cb309-4"><a href="clustering.html#cb309-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(biclust)</span>
<span id="cb309-5"><a href="clustering.html#cb309-5" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb309-6"><a href="clustering.html#cb309-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb309-7"><a href="clustering.html#cb309-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Set seed for reproducibility</span></span>
<span id="cb309-8"><a href="clustering.html#cb309-8" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">3973</span>)</span>
<span id="cb309-9"><a href="clustering.html#cb309-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb309-10"><a href="clustering.html#cb309-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Find biclusters</span></span>
<span id="cb309-11"><a href="clustering.html#cb309-11" aria-hidden="true" tabindex="-1"></a>bc <span class="ot">&lt;-</span> <span class="fu">biclust</span>(corr, <span class="at">method=</span><span class="fu">BCPlaid</span>(), <span class="at">fit.model =</span> y <span class="sc">~</span> m,</span>
<span id="cb309-12"><a href="clustering.html#cb309-12" aria-hidden="true" tabindex="-1"></a>              <span class="at">background =</span> <span class="cn">TRUE</span>, <span class="at">shuffle =</span> <span class="dv">100</span>, <span class="at">back.fit =</span> <span class="dv">0</span>, <span class="at">max.layers =</span> <span class="dv">10</span>,</span>
<span id="cb309-13"><a href="clustering.html#cb309-13" aria-hidden="true" tabindex="-1"></a>              <span class="at">iter.startup =</span> <span class="dv">10</span>, <span class="at">iter.layer =</span> <span class="dv">100</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb309-14"><a href="clustering.html#cb309-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb309-15"><a href="clustering.html#cb309-15" aria-hidden="true" tabindex="-1"></a>bc</span></code></pre></div>
<pre><code>## 
## An object of class Biclust 
## 
## call:
##  biclust(x = corr, method = BCPlaid(), fit.model = y ~ m, background = TRUE, 
##      shuffle = 100, back.fit = 0, max.layers = 10, iter.startup = 10, 
##      iter.layer = 100, verbose = FALSE)
## 
## There was no cluster found</code></pre>
<p>The object includes cluster information. However compared to
<em>cobiclust</em>, <em>biclust</em> object includes only information about clusters
that were found, not general cluster.</p>
<p>Meaning that if one cluster size of 5 features was found out of 20 features,
those 15 features do not belong to any cluster. That is why we have to create an
additional cluster for features/samples that are not assigned into any cluster.</p>
<div class="sourceCode" id="cb311"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb311-1"><a href="clustering.html#cb311-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Functions for obtaining biclust information</span></span>
<span id="cb311-2"><a href="clustering.html#cb311-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb311-3"><a href="clustering.html#cb311-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Get clusters for rows and columns</span></span>
<span id="cb311-4"><a href="clustering.html#cb311-4" aria-hidden="true" tabindex="-1"></a>.get_biclusters_from_biclust <span class="ot">&lt;-</span> <span class="cf">function</span>(bc, assay){</span>
<span id="cb311-5"><a href="clustering.html#cb311-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get cluster information for columns and rows</span></span>
<span id="cb311-6"><a href="clustering.html#cb311-6" aria-hidden="true" tabindex="-1"></a>  bc_columns <span class="ot">&lt;-</span> <span class="fu">t</span>(bc<span class="sc">@</span>NumberxCol)</span>
<span id="cb311-7"><a href="clustering.html#cb311-7" aria-hidden="true" tabindex="-1"></a>  bc_columns <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(bc_columns)</span>
<span id="cb311-8"><a href="clustering.html#cb311-8" aria-hidden="true" tabindex="-1"></a>  bc_rows <span class="ot">&lt;-</span> bc<span class="sc">@</span>RowxNumber</span>
<span id="cb311-9"><a href="clustering.html#cb311-9" aria-hidden="true" tabindex="-1"></a>  bc_rows <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(bc_rows)</span>
<span id="cb311-10"><a href="clustering.html#cb311-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb311-11"><a href="clustering.html#cb311-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get data into right format</span></span>
<span id="cb311-12"><a href="clustering.html#cb311-12" aria-hidden="true" tabindex="-1"></a>  bc_columns <span class="ot">&lt;-</span> <span class="fu">.manipulate_bc_data</span>(bc_columns, assay, <span class="st">&quot;col&quot;</span>)</span>
<span id="cb311-13"><a href="clustering.html#cb311-13" aria-hidden="true" tabindex="-1"></a>  bc_rows <span class="ot">&lt;-</span> <span class="fu">.manipulate_bc_data</span>(bc_rows, assay, <span class="st">&quot;row&quot;</span>)</span>
<span id="cb311-14"><a href="clustering.html#cb311-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb311-15"><a href="clustering.html#cb311-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">list</span>(<span class="at">bc_columns =</span> bc_columns, <span class="at">bc_rows =</span> bc_rows))</span>
<span id="cb311-16"><a href="clustering.html#cb311-16" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb311-17"><a href="clustering.html#cb311-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb311-18"><a href="clustering.html#cb311-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Input clusters, and how many observations there should be, i.e.,</span></span>
<span id="cb311-19"><a href="clustering.html#cb311-19" aria-hidden="true" tabindex="-1"></a><span class="co"># the number of samples or features</span></span>
<span id="cb311-20"><a href="clustering.html#cb311-20" aria-hidden="true" tabindex="-1"></a>.manipulate_bc_data <span class="ot">&lt;-</span> <span class="cf">function</span>(bc_clusters, assay, row_col){</span>
<span id="cb311-21"><a href="clustering.html#cb311-21" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get right dimension</span></span>
<span id="cb311-22"><a href="clustering.html#cb311-22" aria-hidden="true" tabindex="-1"></a>  dim <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(row_col <span class="sc">==</span> <span class="st">&quot;col&quot;</span>, <span class="fu">ncol</span>(assay), <span class="fu">nrow</span>(assay))</span>
<span id="cb311-23"><a href="clustering.html#cb311-23" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get column/row names</span></span>
<span id="cb311-24"><a href="clustering.html#cb311-24" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>( row_col <span class="sc">==</span> <span class="st">&quot;col&quot;</span> ){</span>
<span id="cb311-25"><a href="clustering.html#cb311-25" aria-hidden="true" tabindex="-1"></a>    names <span class="ot">&lt;-</span> <span class="fu">colnames</span>(assay)</span>
<span id="cb311-26"><a href="clustering.html#cb311-26" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span>{</span>
<span id="cb311-27"><a href="clustering.html#cb311-27" aria-hidden="true" tabindex="-1"></a>    names <span class="ot">&lt;-</span> <span class="fu">rownames</span>(assay)</span>
<span id="cb311-28"><a href="clustering.html#cb311-28" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb311-29"><a href="clustering.html#cb311-29" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb311-30"><a href="clustering.html#cb311-30" aria-hidden="true" tabindex="-1"></a>  <span class="co"># If no clusters were found, create one. Otherwise create additional</span></span>
<span id="cb311-31"><a href="clustering.html#cb311-31" aria-hidden="true" tabindex="-1"></a>  <span class="co"># cluster which</span></span>
<span id="cb311-32"><a href="clustering.html#cb311-32" aria-hidden="true" tabindex="-1"></a>  <span class="co"># contain those samples that are not included in clusters that were found.</span></span>
<span id="cb311-33"><a href="clustering.html#cb311-33" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>( <span class="fu">nrow</span>(bc_clusters) <span class="sc">!=</span> dim ){</span>
<span id="cb311-34"><a href="clustering.html#cb311-34" aria-hidden="true" tabindex="-1"></a>      bc_clusters <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">cluster =</span> <span class="fu">rep</span>(<span class="cn">TRUE</span>, dim))</span>
<span id="cb311-35"><a href="clustering.html#cb311-35" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb311-36"><a href="clustering.html#cb311-36" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Create additional cluster that includes those samples/features that</span></span>
<span id="cb311-37"><a href="clustering.html#cb311-37" aria-hidden="true" tabindex="-1"></a>      <span class="co"># are not included in other clusters.</span></span>
<span id="cb311-38"><a href="clustering.html#cb311-38" aria-hidden="true" tabindex="-1"></a>      vec <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">rowSums</span>(bc_clusters) <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="cn">FALSE</span>, <span class="cn">TRUE</span>)</span>
<span id="cb311-39"><a href="clustering.html#cb311-39" aria-hidden="true" tabindex="-1"></a>      <span class="co"># If additional cluster contains samples, then add it</span></span>
<span id="cb311-40"><a href="clustering.html#cb311-40" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> ( <span class="fu">any</span>(vec) ){</span>
<span id="cb311-41"><a href="clustering.html#cb311-41" aria-hidden="true" tabindex="-1"></a>          bc_clusters <span class="ot">&lt;-</span> <span class="fu">cbind</span>(bc_clusters, vec)</span>
<span id="cb311-42"><a href="clustering.html#cb311-42" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb311-43"><a href="clustering.html#cb311-43" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb311-44"><a href="clustering.html#cb311-44" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Adjust row and column names</span></span>
<span id="cb311-45"><a href="clustering.html#cb311-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(bc_clusters) <span class="ot">&lt;-</span> names</span>
<span id="cb311-46"><a href="clustering.html#cb311-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(bc_clusters) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;cluster_&quot;</span>, <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(bc_clusters))</span>
<span id="cb311-47"><a href="clustering.html#cb311-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(bc_clusters)</span>
<span id="cb311-48"><a href="clustering.html#cb311-48" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb312"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb312-1"><a href="clustering.html#cb312-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get biclusters</span></span>
<span id="cb312-2"><a href="clustering.html#cb312-2" aria-hidden="true" tabindex="-1"></a>bcs <span class="ot">&lt;-</span> <span class="fu">.get_biclusters_from_biclust</span>(bc, corr)</span>
<span id="cb312-3"><a href="clustering.html#cb312-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb312-4"><a href="clustering.html#cb312-4" aria-hidden="true" tabindex="-1"></a>bicluster_rows <span class="ot">&lt;-</span> bcs<span class="sc">$</span>bc_rows</span>
<span id="cb312-5"><a href="clustering.html#cb312-5" aria-hidden="true" tabindex="-1"></a>bicluster_columns <span class="ot">&lt;-</span> bcs<span class="sc">$</span>bc_columns</span>
<span id="cb312-6"><a href="clustering.html#cb312-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb312-7"><a href="clustering.html#cb312-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Print biclusters for rows</span></span>
<span id="cb312-8"><a href="clustering.html#cb312-8" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(bicluster_rows)</span></code></pre></div>
<pre><code>##                                    cluster_1
## D_5__Ruminiclostridium 5                TRUE
## D_5__Lachnoclostridium                  TRUE
## D_5__Holdemania                         TRUE
## D_5__Anaerostipes                       TRUE
## D_5__uncultured_3                       TRUE
## D_5__Ruminococcaceae NK4A214 group      TRUE</code></pre>
<p>Let’s collect information for the scatter plot.</p>
<div class="sourceCode" id="cb314"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb314-1"><a href="clustering.html#cb314-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Function for obtaining sample-wise sum, mean, median, and mean variance</span></span>
<span id="cb314-2"><a href="clustering.html#cb314-2" aria-hidden="true" tabindex="-1"></a><span class="co"># for each cluster</span></span>
<span id="cb314-3"><a href="clustering.html#cb314-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb314-4"><a href="clustering.html#cb314-4" aria-hidden="true" tabindex="-1"></a>.sum_mean_median_var <span class="ot">&lt;-</span> <span class="cf">function</span>(tse1, tse2, assay_name1, assay_name2, clusters1, clusters2){</span>
<span id="cb314-5"><a href="clustering.html#cb314-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb314-6"><a href="clustering.html#cb314-6" aria-hidden="true" tabindex="-1"></a>  list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb314-7"><a href="clustering.html#cb314-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create a data frame that includes all the information</span></span>
<span id="cb314-8"><a href="clustering.html#cb314-8" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(clusters1) ){</span>
<span id="cb314-9"><a href="clustering.html#cb314-9" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Subset data based on cluster</span></span>
<span id="cb314-10"><a href="clustering.html#cb314-10" aria-hidden="true" tabindex="-1"></a>    tse_subset1 <span class="ot">&lt;-</span> tse1[clusters1[,i], ]</span>
<span id="cb314-11"><a href="clustering.html#cb314-11" aria-hidden="true" tabindex="-1"></a>    tse_subset2 <span class="ot">&lt;-</span> tse2[clusters2[,i], ]</span>
<span id="cb314-12"><a href="clustering.html#cb314-12" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Get assay</span></span>
<span id="cb314-13"><a href="clustering.html#cb314-13" aria-hidden="true" tabindex="-1"></a>    assay1 <span class="ot">&lt;-</span> <span class="fu">assay</span>(tse_subset1, assay_name1)</span>
<span id="cb314-14"><a href="clustering.html#cb314-14" aria-hidden="true" tabindex="-1"></a>    assay2 <span class="ot">&lt;-</span> <span class="fu">assay</span>(tse_subset2, assay_name2)</span>
<span id="cb314-15"><a href="clustering.html#cb314-15" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Calculate sum, mean, median, and mean variance</span></span>
<span id="cb314-16"><a href="clustering.html#cb314-16" aria-hidden="true" tabindex="-1"></a>    sum1 <span class="ot">&lt;-</span> <span class="fu">colSums2</span>(assay1, <span class="at">na.rm =</span> T)</span>
<span id="cb314-17"><a href="clustering.html#cb314-17" aria-hidden="true" tabindex="-1"></a>    mean1 <span class="ot">&lt;-</span> <span class="fu">colMeans2</span>(assay1, <span class="at">na.rm =</span> T)</span>
<span id="cb314-18"><a href="clustering.html#cb314-18" aria-hidden="true" tabindex="-1"></a>    median1 <span class="ot">&lt;-</span> <span class="fu">colMedians</span>(assay1, <span class="at">na.rm =</span> T)</span>
<span id="cb314-19"><a href="clustering.html#cb314-19" aria-hidden="true" tabindex="-1"></a>    var1 <span class="ot">&lt;-</span> <span class="fu">colVars</span>(assay1, <span class="at">na.rm =</span> T)</span>
<span id="cb314-20"><a href="clustering.html#cb314-20" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb314-21"><a href="clustering.html#cb314-21" aria-hidden="true" tabindex="-1"></a>    sum2 <span class="ot">&lt;-</span> <span class="fu">colSums2</span>(assay2, <span class="at">na.rm =</span> T)</span>
<span id="cb314-22"><a href="clustering.html#cb314-22" aria-hidden="true" tabindex="-1"></a>    mean2 <span class="ot">&lt;-</span> <span class="fu">colMeans2</span>(assay2, <span class="at">na.rm =</span> T)</span>
<span id="cb314-23"><a href="clustering.html#cb314-23" aria-hidden="true" tabindex="-1"></a>    median2 <span class="ot">&lt;-</span> <span class="fu">colMedians</span>(assay2, <span class="at">na.rm =</span> T)</span>
<span id="cb314-24"><a href="clustering.html#cb314-24" aria-hidden="true" tabindex="-1"></a>    var2 <span class="ot">&lt;-</span> <span class="fu">colVars</span>(assay2, <span class="at">na.rm =</span> T)</span>
<span id="cb314-25"><a href="clustering.html#cb314-25" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb314-26"><a href="clustering.html#cb314-26" aria-hidden="true" tabindex="-1"></a>    list[[i]] <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">sample =</span> <span class="fu">colnames</span>(tse1), sum1, sum2, mean1, mean2, </span>
<span id="cb314-27"><a href="clustering.html#cb314-27" aria-hidden="true" tabindex="-1"></a>                     median1, median2, var1, var2)</span>
<span id="cb314-28"><a href="clustering.html#cb314-28" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb314-29"><a href="clustering.html#cb314-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb314-30"><a href="clustering.html#cb314-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(list)</span>
<span id="cb314-31"><a href="clustering.html#cb314-31" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb314-32"><a href="clustering.html#cb314-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb314-33"><a href="clustering.html#cb314-33" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate info</span></span>
<span id="cb314-34"><a href="clustering.html#cb314-34" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">.sum_mean_median_var</span>(mae[[<span class="dv">1</span>]], mae[[<span class="dv">2</span>]], <span class="st">&quot;rclr&quot;</span>, <span class="st">&quot;nmr&quot;</span>, bicluster_rows, bicluster_columns)</span></code></pre></div>
<p>Now we can create a scatter plot. X-axis includes median clr abundance
of microbiome and y-axis median absolute concentration of each
metabolite. Each data point represents a single sample.</p>
<p>From the plots, we can see that there is low negative correlation in
both cluster 1 and 3. This means that when abundance of bacteria
belonging to cluster 1 or 3 is higher, the concentration of
metabolites of cluster 1 or 3 is lower, and vice versa.</p>
<div class="sourceCode" id="cb315"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb315-1"><a href="clustering.html#cb315-1" aria-hidden="true" tabindex="-1"></a>pics <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb315-2"><a href="clustering.html#cb315-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq_along</span>(df)){</span>
<span id="cb315-3"><a href="clustering.html#cb315-3" aria-hidden="true" tabindex="-1"></a>  pics[[i]] <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(df[[i]])  <span class="sc">+</span></span>
<span id="cb315-4"><a href="clustering.html#cb315-4" aria-hidden="true" tabindex="-1"></a>      <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x =</span> median1, <span class="at">y =</span> median2)) <span class="sc">+</span> </span>
<span id="cb315-5"><a href="clustering.html#cb315-5" aria-hidden="true" tabindex="-1"></a>      <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste0</span>(<span class="st">&quot;Cluster &quot;</span>, i),</span>
<span id="cb315-6"><a href="clustering.html#cb315-6" aria-hidden="true" tabindex="-1"></a>           <span class="at">x =</span> <span class="st">&quot;Taxa (rclr median)&quot;</span>,</span>
<span id="cb315-7"><a href="clustering.html#cb315-7" aria-hidden="true" tabindex="-1"></a>           <span class="at">y =</span> <span class="st">&quot;Metabolites (abs. median)&quot;</span>)</span>
<span id="cb315-8"><a href="clustering.html#cb315-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(pics[[i]])</span>
<span id="cb315-9"><a href="clustering.html#cb315-9" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p><img src="24_clustering_files/figure-html/biclust_6-1.png" width="33%" /></p>
<div class="sourceCode" id="cb316"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb316-1"><a href="clustering.html#cb316-1" aria-hidden="true" tabindex="-1"></a><span class="co"># pics[[1]] + pics[[2]] + pics[[3]]</span></span></code></pre></div>
<p><em>pheatmap</em> does not allow boolean values, so they must be converted into factors.</p>
<div class="sourceCode" id="cb317"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb317-1"><a href="clustering.html#cb317-1" aria-hidden="true" tabindex="-1"></a>bicluster_columns <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">apply</span>(bicluster_columns, <span class="dv">2</span>, as.factor))</span>
<span id="cb317-2"><a href="clustering.html#cb317-2" aria-hidden="true" tabindex="-1"></a>bicluster_rows <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">apply</span>(bicluster_rows, <span class="dv">2</span>, as.factor))</span></code></pre></div>
<p>Again, we can plot clusters with heatmap.</p>
<div class="sourceCode" id="cb318"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb318-1"><a href="clustering.html#cb318-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Adjust colors for all clusters</span></span>
<span id="cb318-2"><a href="clustering.html#cb318-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>( <span class="fu">ncol</span>(bicluster_rows) <span class="sc">&gt;</span> <span class="fu">ncol</span>(bicluster_columns) ){</span>
<span id="cb318-3"><a href="clustering.html#cb318-3" aria-hidden="true" tabindex="-1"></a>  cluster_names <span class="ot">&lt;-</span> <span class="fu">colnames</span>(bicluster_rows)</span>
<span id="cb318-4"><a href="clustering.html#cb318-4" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb318-5"><a href="clustering.html#cb318-5" aria-hidden="true" tabindex="-1"></a>  cluster_names <span class="ot">&lt;-</span> <span class="fu">colnames</span>(bicluster_columns)</span>
<span id="cb318-6"><a href="clustering.html#cb318-6" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb318-7"><a href="clustering.html#cb318-7" aria-hidden="true" tabindex="-1"></a>annotation_colors <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb318-8"><a href="clustering.html#cb318-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(name <span class="cf">in</span> cluster_names){</span>
<span id="cb318-9"><a href="clustering.html#cb318-9" aria-hidden="true" tabindex="-1"></a>  annotation_colors[[name]] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;TRUE&quot;</span> <span class="ot">=</span> <span class="st">&quot;red&quot;</span>, <span class="st">&quot;FALSE&quot;</span> <span class="ot">=</span> <span class="st">&quot;white&quot;</span>)</span>
<span id="cb318-10"><a href="clustering.html#cb318-10" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb318-11"><a href="clustering.html#cb318-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb318-12"><a href="clustering.html#cb318-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a heatmap</span></span>
<span id="cb318-13"><a href="clustering.html#cb318-13" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(corr, <span class="at">cluster_cols =</span> F, <span class="at">cluster_rows =</span> F,</span>
<span id="cb318-14"><a href="clustering.html#cb318-14" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_col =</span> bicluster_columns, </span>
<span id="cb318-15"><a href="clustering.html#cb318-15" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_row =</span> bicluster_rows,</span>
<span id="cb318-16"><a href="clustering.html#cb318-16" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_colors =</span> annotation_colors)</span></code></pre></div>
<p><img src="24_clustering_files/figure-html/biclust_8-1.png" width="960" /></p>
</div>
<div id="taxa-vs-taxa" class="section level3 hasAnchor" number="10.5.3">
<h3><span class="header-section-number">10.5.3</span> Taxa vs taxa<a href="clustering.html#taxa-vs-taxa" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Third and final example deals with situation where we want to analyze
correlation between taxa. <em>biclust</em> is suitable for this.</p>
<div class="sourceCode" id="cb319"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb319-1"><a href="clustering.html#cb319-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate cross-correlation</span></span>
<span id="cb319-2"><a href="clustering.html#cb319-2" aria-hidden="true" tabindex="-1"></a>corr <span class="ot">&lt;-</span> <span class="fu">getExperimentCrossCorrelation</span>(mae, <span class="dv">1</span>, <span class="dv">1</span>, </span>
<span id="cb319-3"><a href="clustering.html#cb319-3" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">assay_name1 =</span> <span class="st">&quot;rclr&quot;</span>, <span class="at">assay_name2 =</span> <span class="st">&quot;rclr&quot;</span>, </span>
<span id="cb319-4"><a href="clustering.html#cb319-4" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">mode =</span> <span class="st">&quot;matrix&quot;</span>,</span>
<span id="cb319-5"><a href="clustering.html#cb319-5" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">cor_threshold =</span> <span class="fl">0.2</span>, <span class="at">verbose =</span> F, <span class="at">show_warning =</span> F)</span>
<span id="cb319-6"><a href="clustering.html#cb319-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb319-7"><a href="clustering.html#cb319-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Find biclusters</span></span>
<span id="cb319-8"><a href="clustering.html#cb319-8" aria-hidden="true" tabindex="-1"></a>bc <span class="ot">&lt;-</span> <span class="fu">biclust</span>(corr, <span class="at">method=</span><span class="fu">BCPlaid</span>(), <span class="at">fit.model =</span> y <span class="sc">~</span> m,</span>
<span id="cb319-9"><a href="clustering.html#cb319-9" aria-hidden="true" tabindex="-1"></a>              <span class="at">background =</span> <span class="cn">TRUE</span>, <span class="at">shuffle =</span> <span class="dv">100</span>, <span class="at">back.fit =</span> <span class="dv">0</span>, <span class="at">max.layers =</span> <span class="dv">10</span>,</span>
<span id="cb319-10"><a href="clustering.html#cb319-10" aria-hidden="true" tabindex="-1"></a>              <span class="at">iter.startup =</span> <span class="dv">10</span>, <span class="at">iter.layer =</span> <span class="dv">100</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<div class="sourceCode" id="cb320"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb320-1"><a href="clustering.html#cb320-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get biclusters</span></span>
<span id="cb320-2"><a href="clustering.html#cb320-2" aria-hidden="true" tabindex="-1"></a>bcs <span class="ot">&lt;-</span> <span class="fu">.get_biclusters_from_biclust</span>(bc, corr)</span>
<span id="cb320-3"><a href="clustering.html#cb320-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb320-4"><a href="clustering.html#cb320-4" aria-hidden="true" tabindex="-1"></a>bicluster_rows <span class="ot">&lt;-</span> bcs<span class="sc">$</span>bc_rows</span>
<span id="cb320-5"><a href="clustering.html#cb320-5" aria-hidden="true" tabindex="-1"></a>bicluster_columns <span class="ot">&lt;-</span> bcs<span class="sc">$</span>bc_columns</span></code></pre></div>
<div class="sourceCode" id="cb321"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb321-1"><a href="clustering.html#cb321-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a column that combines information</span></span>
<span id="cb321-2"><a href="clustering.html#cb321-2" aria-hidden="true" tabindex="-1"></a><span class="co"># If row/column includes in multiple clusters, cluster numbers are separated with &quot;_&amp;_&quot;</span></span>
<span id="cb321-3"><a href="clustering.html#cb321-3" aria-hidden="true" tabindex="-1"></a>bicluster_columns<span class="sc">$</span>clusters <span class="ot">&lt;-</span> <span class="fu">apply</span>(bicluster_columns, <span class="dv">1</span>, </span>
<span id="cb321-4"><a href="clustering.html#cb321-4" aria-hidden="true" tabindex="-1"></a>                                    <span class="cf">function</span>(x){<span class="fu">paste</span>(<span class="fu">paste</span>(<span class="fu">which</span>(x)), <span class="at">collapse =</span> <span class="st">&quot;_&amp;_&quot;</span>) })</span>
<span id="cb321-5"><a href="clustering.html#cb321-5" aria-hidden="true" tabindex="-1"></a>bicluster_columns <span class="ot">&lt;-</span> bicluster_columns[, <span class="st">&quot;clusters&quot;</span>, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb321-6"><a href="clustering.html#cb321-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb321-7"><a href="clustering.html#cb321-7" aria-hidden="true" tabindex="-1"></a>bicluster_rows<span class="sc">$</span>clusters <span class="ot">&lt;-</span> <span class="fu">apply</span>(bicluster_rows, <span class="dv">1</span>, </span>
<span id="cb321-8"><a href="clustering.html#cb321-8" aria-hidden="true" tabindex="-1"></a>                                 <span class="cf">function</span>(x){<span class="fu">paste</span>(<span class="fu">paste</span>(<span class="fu">which</span>(x)), <span class="at">collapse =</span> <span class="st">&quot;_&amp;_&quot;</span>) })</span>
<span id="cb321-9"><a href="clustering.html#cb321-9" aria-hidden="true" tabindex="-1"></a>bicluster_rows <span class="ot">&lt;-</span> bicluster_rows[, <span class="st">&quot;clusters&quot;</span>, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span></code></pre></div>
<div class="sourceCode" id="cb322"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb322-1"><a href="clustering.html#cb322-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert boolean values into factor</span></span>
<span id="cb322-2"><a href="clustering.html#cb322-2" aria-hidden="true" tabindex="-1"></a>bicluster_columns <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">apply</span>(bicluster_columns, <span class="dv">2</span>, as.factor))</span>
<span id="cb322-3"><a href="clustering.html#cb322-3" aria-hidden="true" tabindex="-1"></a>bicluster_rows <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">apply</span>(bicluster_rows, <span class="dv">2</span>, as.factor))</span>
<span id="cb322-4"><a href="clustering.html#cb322-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb322-5"><a href="clustering.html#cb322-5" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(corr, <span class="at">cluster_cols =</span> F, <span class="at">cluster_rows =</span> F,</span>
<span id="cb322-6"><a href="clustering.html#cb322-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_col =</span> bicluster_columns, </span>
<span id="cb322-7"><a href="clustering.html#cb322-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_row =</span> bicluster_rows)</span></code></pre></div>
<p><img src="24_clustering_files/figure-html/biclust_12-1.png" width="1344" /></p>
</div>
</div>
<div id="additional-community-typing" class="section level2 hasAnchor" number="10.6">
<h2><span class="header-section-number">10.6</span> Additional Community Typing<a href="clustering.html#additional-community-typing" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For more community typing techniques applied to the ‘SprockettTHData’
data set, see the attached .Rmd file.</p>
<p>Link:</p>
<ul>
<li><a href="add-comm-typing.Rmd">Rmd</a></li>
</ul>

</div>
</div>
<h3>Bibliography<a href="bibliography.html#bibliography" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-R-mia" class="csl-entry">
Ernst, Felix G. M., Sudarshan Shetty, and Leo Lahti. 2020. <em>Mia: Microbiome Analysis</em>.
</div>
<div id="ref-R-bluster" class="csl-entry">
Lun, Aaron. 2021. <em><span class="nocase">bluster: Clustering Algorithms for Bioconductor</span></em>. <a href="https://bioconductor.org/packages/3.14/bioc/html/bluster.html">https://bioconductor.org/packages/3.14/bioc/html/bluster.html</a>.
</div>
<div id="ref-McInnes2018" class="csl-entry">
McInnes, L, J Healy, and J Melville. 2018. <span>“<span class="nocase">UMAP: Uniform Manifold Approximation and Projection for Dimension Reduction</span>.”</span> <em>arXiv e-Prints</em>, arXiv:1802.03426. <a href="https://arxiv.org/abs/1802.03426">https://arxiv.org/abs/1802.03426</a>.
</div>
<div id="ref-Pons2006" class="csl-entry">
Pons, P, and M Latapy. 2006. <span>“Computing Communities in Large Networks Using Random Walks.”</span> <em>Journal of Graph Algorithms and Applications</em> 10 (January): 191–218. <a href="https://doi.org/10.7155/jgaa.00124">https://doi.org/10.7155/jgaa.00124</a>.
</div>
<div id="ref-Rousseeuw1987" class="csl-entry">
Rousseeuw, Peter J. 1987. <span>“Silhouettes: A Graphical Aid to the Interpretation and Validation of Cluster Analysis.”</span> <em>Journal of Computational and Applied Mathematics</em> 20: 53–65. https://doi.org/<a href="https://doi.org/10.1016/0377-0427(87)90125-7">https://doi.org/10.1016/0377-0427(87)90125-7</a>.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="microbiome-community.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="differential-abundance.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["OMA.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
