<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 10 Community typing | Orchestrating Microbiome Analysis</title>
  <meta name="description" content="Chapter 10 Community typing | Orchestrating Microbiome Analysis" />
  <meta name="generator" content="bookdown 0.25 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 10 Community typing | Orchestrating Microbiome Analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="microbiome/OMA" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 10 Community typing | Orchestrating Microbiome Analysis" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="microbiome-community.html"/>
<link rel="next" href="biclustering.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<script src="libs/plotly-binding-4.10.0/plotly.js"></script>
<script src="libs/typedarray-0.1/typedarray.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="libs/plotly-htmlwidgets-css-2.5.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="libs/plotly-main-2.5.1/plotly-latest.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Orchestrating Microbiome Analysis</a></li>

<li class="divider"></li>
<li><a href="index.html#welcome">Welcome<span></span></a></li>
<li class="part"><span><b>I Introduction<span></span></b></span></li>
<li class="chapter" data-level="1" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>1</b> Introduction<span></span></a></li>
<li class="chapter" data-level="2" data-path="containers.html"><a href="containers.html"><i class="fa fa-check"></i><b>2</b> Microbiome Data<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="containers.html"><a href="containers.html#data-science-framework"><i class="fa fa-check"></i><b>2.1</b> Data science framework<span></span></a></li>
<li class="chapter" data-level="2.2" data-path="containers.html"><a href="containers.html#data-containers"><i class="fa fa-check"></i><b>2.2</b> Data containers<span></span></a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="containers.html"><a href="containers.html#assay-data"><i class="fa fa-check"></i><b>2.2.1</b> Assay data<span></span></a></li>
<li class="chapter" data-level="2.2.2" data-path="containers.html"><a href="containers.html#coldata"><i class="fa fa-check"></i><b>2.2.2</b> colData<span></span></a></li>
<li class="chapter" data-level="2.2.3" data-path="containers.html"><a href="containers.html#rowdata"><i class="fa fa-check"></i><b>2.2.3</b> rowData<span></span></a></li>
<li class="chapter" data-level="2.2.4" data-path="containers.html"><a href="containers.html#rowtree"><i class="fa fa-check"></i><b>2.2.4</b> rowTree<span></span></a></li>
<li class="chapter" data-level="2.2.5" data-path="containers.html"><a href="containers.html#alternative-experiments"><i class="fa fa-check"></i><b>2.2.5</b> Alternative experiments<span></span></a></li>
<li class="chapter" data-level="2.2.6" data-path="containers.html"><a href="containers.html#multiassayexperiments"><i class="fa fa-check"></i><b>2.2.6</b> MultiAssayExperiments<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="containers.html"><a href="containers.html#loading-experimental-microbiome-data"><i class="fa fa-check"></i><b>2.3</b> Loading experimental microbiome data<span></span></a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="containers.html"><a href="containers.html#import-from-external-files"><i class="fa fa-check"></i><b>2.3.1</b> Import from external files<span></span></a></li>
<li class="chapter" data-level="2.3.2" data-path="containers.html"><a href="containers.html#conversions-between-data-formats-in-r"><i class="fa fa-check"></i><b>2.3.2</b> Conversions between data formats in R<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="containers.html"><a href="containers.html#example-data"><i class="fa fa-check"></i><b>2.4</b> Demonstration data<span></span></a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="containers.html"><a href="containers.html#package-data"><i class="fa fa-check"></i><b>2.4.1</b> Package data<span></span></a></li>
<li class="chapter" data-level="2.4.2" data-path="containers.html"><a href="containers.html#experimenthub-data"><i class="fa fa-check"></i><b>2.4.2</b> ExperimentHub data<span></span></a></li>
<li class="chapter" data-level="2.4.3" data-path="containers.html"><a href="containers.html#other-data-sources"><i class="fa fa-check"></i><b>2.4.3</b> Other data sources<span></span></a></li>
</ul></li>
<li><a href="containers.html#session-info">Session Info<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="packages.html"><a href="packages.html"><i class="fa fa-check"></i><b>3</b> Packages<span></span></a>
<ul>
<li class="chapter" data-level="3.1" data-path="packages.html"><a href="packages.html#package-installation"><i class="fa fa-check"></i><b>3.1</b> Package installation<span></span></a></li>
<li class="chapter" data-level="3.2" data-path="packages.html"><a href="packages.html#some-available-packages"><i class="fa fa-check"></i><b>3.2</b> Some available packages<span></span></a></li>
<li><a href="packages.html#session-info-1">Session Info<span></span></a></li>
</ul></li>
<li class="part"><span><b>II Focus Topics<span></span></b></span></li>
<li class="chapter" data-level="4" data-path="datamanipulation.html"><a href="datamanipulation.html"><i class="fa fa-check"></i><b>4</b> Data Manipulation<span></span></a>
<ul>
<li class="chapter" data-level="4.1" data-path="datamanipulation.html"><a href="datamanipulation.html#tidying-and-subsetting"><i class="fa fa-check"></i><b>4.1</b> Tidying and subsetting<span></span></a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="datamanipulation.html"><a href="datamanipulation.html#tidy-data"><i class="fa fa-check"></i><b>4.1.1</b> Tidy data<span></span></a></li>
<li class="chapter" data-level="4.1.2" data-path="datamanipulation.html"><a href="datamanipulation.html#subsetting"><i class="fa fa-check"></i><b>4.1.2</b> Subsetting<span></span></a></li>
<li class="chapter" data-level="4.1.3" data-path="datamanipulation.html"><a href="datamanipulation.html#additional-functions"><i class="fa fa-check"></i><b>4.1.3</b> Additional functions<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>5</b> Exploration and quality Control<span></span></a>
<ul>
<li class="chapter" data-level="5.1" data-path="quality-control.html"><a href="quality-control.html#abundance"><i class="fa fa-check"></i><b>5.1</b> Abundance<span></span></a></li>
<li class="chapter" data-level="5.2" data-path="quality-control.html"><a href="quality-control.html#prevalence"><i class="fa fa-check"></i><b>5.2</b> Prevalence<span></span></a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="quality-control.html"><a href="quality-control.html#prevalent-microbiota-analysis"><i class="fa fa-check"></i><b>5.2.1</b> Prevalent microbiota analysis<span></span></a></li>
<li class="chapter" data-level="5.2.2" data-path="quality-control.html"><a href="quality-control.html#plotting-prevalence"><i class="fa fa-check"></i><b>5.2.2</b> Plotting prevalence<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="quality-control.html"><a href="quality-control.html#quality-control-1"><i class="fa fa-check"></i><b>5.3</b> Quality control<span></span></a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="quality-control.html"><a href="quality-control.html#top-taxa"><i class="fa fa-check"></i><b>5.3.1</b> Top taxa<span></span></a></li>
<li class="chapter" data-level="5.3.2" data-path="quality-control.html"><a href="quality-control.html#library-size"><i class="fa fa-check"></i><b>5.3.2</b> Library size<span></span></a></li>
<li class="chapter" data-level="5.3.3" data-path="quality-control.html"><a href="quality-control.html#contaminant-sequences"><i class="fa fa-check"></i><b>5.3.3</b> Contaminant sequences<span></span></a></li>
</ul></li>
<li><a href="quality-control.html#session-info-2">Session Info<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="taxonomic-information.html"><a href="taxonomic-information.html"><i class="fa fa-check"></i><b>6</b> Taxonomic Information<span></span></a>
<ul>
<li class="chapter" data-level="6.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#assigning-taxonomic-information."><i class="fa fa-check"></i><b>6.1</b> Assigning taxonomic information.<span></span></a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#dada2"><i class="fa fa-check"></i><b>6.1.1</b> dada2<span></span></a></li>
<li class="chapter" data-level="6.1.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#decipher"><i class="fa fa-check"></i><b>6.1.2</b> DECIPHER<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#functions-to-access-taxonomic-information"><i class="fa fa-check"></i><b>6.2</b> Functions to access taxonomic information<span></span></a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#generate-a-taxonomic-tree-on-the-fly"><i class="fa fa-check"></i><b>6.2.1</b> Generate a taxonomic tree on the fly<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="taxonomic-information.html"><a href="taxonomic-information.html#data-agglomeration"><i class="fa fa-check"></i><b>6.3</b> Data agglomeration<span></span></a></li>
<li class="chapter" data-level="6.4" data-path="taxonomic-information.html"><a href="taxonomic-information.html#data-transformation"><i class="fa fa-check"></i><b>6.4</b> Data transformation<span></span></a></li>
<li class="chapter" data-level="6.5" data-path="taxonomic-information.html"><a href="taxonomic-information.html#pick-specific"><i class="fa fa-check"></i><b>6.5</b> Pick specific<span></span></a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#abundances-of-all-taxa-in-specific-sample"><i class="fa fa-check"></i><b>6.5.1</b> Abundances of all taxa in specific sample<span></span></a></li>
<li class="chapter" data-level="6.5.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#abundances-of-specific-taxa-in-all-samples"><i class="fa fa-check"></i><b>6.5.2</b> Abundances of specific taxa in all samples<span></span></a></li>
</ul></li>
<li><a href="taxonomic-information.html#session-info-3">Session Info<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="community-diversity.html"><a href="community-diversity.html"><i class="fa fa-check"></i><b>7</b> Community diversity<span></span></a>
<ul>
<li class="chapter" data-level="7.1" data-path="community-diversity.html"><a href="community-diversity.html#estimation"><i class="fa fa-check"></i><b>7.1</b> Estimation<span></span></a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="community-diversity.html"><a href="community-diversity.html#richness"><i class="fa fa-check"></i><b>7.1.1</b> Richness<span></span></a></li>
<li class="chapter" data-level="7.1.2" data-path="community-diversity.html"><a href="community-diversity.html#diversity"><i class="fa fa-check"></i><b>7.1.2</b> Diversity<span></span></a></li>
<li class="chapter" data-level="7.1.3" data-path="community-diversity.html"><a href="community-diversity.html#faith-phylogenetic-diversity"><i class="fa fa-check"></i><b>7.1.3</b> Faith phylogenetic diversity<span></span></a></li>
<li class="chapter" data-level="7.1.4" data-path="community-diversity.html"><a href="community-diversity.html#evenness"><i class="fa fa-check"></i><b>7.1.4</b> Evenness<span></span></a></li>
<li class="chapter" data-level="7.1.5" data-path="community-diversity.html"><a href="community-diversity.html#dominance"><i class="fa fa-check"></i><b>7.1.5</b> Dominance<span></span></a></li>
<li class="chapter" data-level="7.1.6" data-path="community-diversity.html"><a href="community-diversity.html#rarity"><i class="fa fa-check"></i><b>7.1.6</b> Rarity<span></span></a></li>
<li class="chapter" data-level="7.1.7" data-path="community-diversity.html"><a href="community-diversity.html#divergence"><i class="fa fa-check"></i><b>7.1.7</b> Divergence<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="community-diversity.html"><a href="community-diversity.html#visualization"><i class="fa fa-check"></i><b>7.2</b> Visualization<span></span></a></li>
<li><a href="community-diversity.html#session-info-4">Session Info<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="community-similarity.html"><a href="community-similarity.html"><i class="fa fa-check"></i><b>8</b> Community similarity<span></span></a>
<ul>
<li class="chapter" data-level="8.1" data-path="community-similarity.html"><a href="community-similarity.html#explained-variance"><i class="fa fa-check"></i><b>8.1</b> Explained variance<span></span></a></li>
<li class="chapter" data-level="8.2" data-path="community-similarity.html"><a href="community-similarity.html#community-comparisons-by-beta-diversity-analysis"><i class="fa fa-check"></i><b>8.2</b> Community comparisons by beta diversity analysis<span></span></a></li>
<li class="chapter" data-level="8.3" data-path="community-similarity.html"><a href="community-similarity.html#other-ordination-methods"><i class="fa fa-check"></i><b>8.3</b> Other ordination methods<span></span></a></li>
<li class="chapter" data-level="8.4" data-path="community-similarity.html"><a href="community-similarity.html#visualizing-the-most-dominant-genus-on-pcoa"><i class="fa fa-check"></i><b>8.4</b> Visualizing the most dominant genus on PCoA<span></span></a>
<ul>
<li class="chapter" data-level="8.4.1" data-path="community-similarity.html"><a href="community-similarity.html#testing-differences-in-community-composition-between-sample-groups"><i class="fa fa-check"></i><b>8.4.1</b> Testing differences in community composition between sample groups<span></span></a></li>
<li class="chapter" data-level="8.4.2" data-path="community-similarity.html"><a href="community-similarity.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>8.4.2</b> Checking the homogeneity condition<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="community-similarity.html"><a href="community-similarity.html#further-reading"><i class="fa fa-check"></i><b>8.5</b> Further reading<span></span></a></li>
<li><a href="community-similarity.html#session-info-5">Session Info<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="microbiome-community.html"><a href="microbiome-community.html"><i class="fa fa-check"></i><b>9</b> Community composition<span></span></a>
<ul>
<li class="chapter" data-level="9.1" data-path="microbiome-community.html"><a href="microbiome-community.html#visualizing-taxonomic-composition"><i class="fa fa-check"></i><b>9.1</b> Visualizing taxonomic composition<span></span></a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="microbiome-community.html"><a href="microbiome-community.html#composition-barplot"><i class="fa fa-check"></i><b>9.1.1</b> Composition barplot<span></span></a></li>
<li class="chapter" data-level="9.1.2" data-path="microbiome-community.html"><a href="microbiome-community.html#composition-heatmap"><i class="fa fa-check"></i><b>9.1.2</b> Composition heatmap<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="community-typing.html"><a href="community-typing.html"><i class="fa fa-check"></i><b>10</b> Community typing<span></span></a>
<ul>
<li class="chapter" data-level="10.1" data-path="community-typing.html"><a href="community-typing.html#dirichlet-multinomial-mixtures-dmm"><i class="fa fa-check"></i><b>10.1</b> Dirichlet Multinomial Mixtures (DMM)<span></span></a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="community-typing.html"><a href="community-typing.html#pcoa-for-asv-level-data-with-bray-curtis-with-dmm-clusters-shown-with-colors"><i class="fa fa-check"></i><b>10.1.1</b> PCoA for ASV-level data with Bray-Curtis; with DMM clusters shown with colors<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="community-typing.html"><a href="community-typing.html#community-detection"><i class="fa fa-check"></i><b>10.2</b> Community Detection<span></span></a></li>
<li class="chapter" data-level="10.3" data-path="community-typing.html"><a href="community-typing.html#additional-community-typing"><i class="fa fa-check"></i><b>10.3</b> Additional Community Typing<span></span></a></li>
<li><a href="community-typing.html#session-info-6">Session Info<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="biclustering.html"><a href="biclustering.html"><i class="fa fa-check"></i><b>11</b> Biclustering<span></span></a>
<ul>
<li class="chapter" data-level="11.1" data-path="biclustering.html"><a href="biclustering.html#taxa-vs-samples"><i class="fa fa-check"></i><b>11.1</b> Taxa vs samples<span></span></a></li>
<li class="chapter" data-level="11.2" data-path="biclustering.html"><a href="biclustering.html#taxa-vs-biomolecules"><i class="fa fa-check"></i><b>11.2</b> Taxa vs biomolecules<span></span></a></li>
<li class="chapter" data-level="11.3" data-path="biclustering.html"><a href="biclustering.html#taxa-vs-taxa"><i class="fa fa-check"></i><b>11.3</b> Taxa vs taxa<span></span></a></li>
<li><a href="biclustering.html#session-info-7">Session Info<span></span></a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="differential-abundance.html"><a href="differential-abundance.html"><i class="fa fa-check"></i><b>12</b> Differential abundance<span></span></a>
<ul>
<li class="chapter" data-level="12.1" data-path="differential-abundance.html"><a href="differential-abundance.html#differential-abundance-analysis"><i class="fa fa-check"></i><b>12.1</b> Differential abundance analysis<span></span></a>
<ul>
<li class="chapter" data-level="12.1.1" data-path="differential-abundance.html"><a href="differential-abundance.html#examples-and-tools"><i class="fa fa-check"></i><b>12.1.1</b> Examples and tools<span></span></a></li>
<li class="chapter" data-level="12.1.2" data-path="differential-abundance.html"><a href="differential-abundance.html#prevalence-filtering"><i class="fa fa-check"></i><b>12.1.2</b> Prevalence Filtering<span></span></a></li>
<li class="chapter" data-level="12.1.3" data-path="differential-abundance.html"><a href="differential-abundance.html#aldex2"><i class="fa fa-check"></i><b>12.1.3</b> ALDEx2<span></span></a></li>
<li class="chapter" data-level="12.1.4" data-path="differential-abundance.html"><a href="differential-abundance.html#ancom-bc"><i class="fa fa-check"></i><b>12.1.4</b> ANCOM-BC<span></span></a></li>
<li class="chapter" data-level="12.1.5" data-path="differential-abundance.html"><a href="differential-abundance.html#maaslin2"><i class="fa fa-check"></i><b>12.1.5</b> MaAsLin2<span></span></a></li>
<li class="chapter" data-level="12.1.6" data-path="differential-abundance.html"><a href="differential-abundance.html#comparison-of-the-methods"><i class="fa fa-check"></i><b>12.1.6</b> Comparison of the methods<span></span></a></li>
<li class="chapter" data-level="12.1.7" data-path="differential-abundance.html"><a href="differential-abundance.html#confounding-variables"><i class="fa fa-check"></i><b>12.1.7</b> Confounding variables<span></span></a></li>
</ul></li>
<li class="chapter" data-level="12.2" data-path="differential-abundance.html"><a href="differential-abundance.html#tree-based-methods"><i class="fa fa-check"></i><b>12.2</b> Tree-based methods<span></span></a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="differential-abundance.html"><a href="differential-abundance.html#group-wise-associations-testing-based-on-balances-with-fido"><i class="fa fa-check"></i><b>12.2.1</b> Group-wise associations testing based on balances with fido<span></span></a></li>
</ul></li>
<li><a href="differential-abundance.html#session-info-8">Session Info<span></span></a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="multi-assay_analyses.html"><a href="multi-assay_analyses.html"><i class="fa fa-check"></i><b>13</b> Multi-assay analyses<span></span></a>
<ul>
<li class="chapter" data-level="13.1" data-path="multi-assay_analyses.html"><a href="multi-assay_analyses.html#cross-correlation-analysis"><i class="fa fa-check"></i><b>13.1</b> Cross-correlation Analysis<span></span></a></li>
<li class="chapter" data-level="13.2" data-path="multi-assay_analyses.html"><a href="multi-assay_analyses.html#multi-omics-factor-analysis"><i class="fa fa-check"></i><b>13.2</b> Multi-Omics Factor Analysis<span></span></a></li>
<li><a href="multi-assay_analyses.html#session-info-9">Session Info<span></span></a></li>
</ul></li>
<li class="part"><span><b>III Appendix<span></span></b></span></li>
<li class="chapter" data-level="14" data-path="resources.html"><a href="resources.html"><i class="fa fa-check"></i><b>14</b> Resources<span></span></a>
<ul>
<li class="chapter" data-level="14.1" data-path="resources.html"><a href="resources.html#data-containers-1"><i class="fa fa-check"></i><b>14.1</b> Data containers<span></span></a>
<ul>
<li class="chapter" data-level="14.1.1" data-path="resources.html"><a href="resources.html#resources-for-treesummarizedexperiment"><i class="fa fa-check"></i><b>14.1.1</b> Resources for TreeSummarizedExperiment<span></span></a></li>
<li class="chapter" data-level="14.1.2" data-path="resources.html"><a href="resources.html#other-relevant-containers"><i class="fa fa-check"></i><b>14.1.2</b> Other relevant containers<span></span></a></li>
<li class="chapter" data-level="14.1.3" data-path="resources.html"><a href="resources.html#alternative-containers-for-microbiome-data"><i class="fa fa-check"></i><b>14.1.3</b> Alternative containers for microbiome data<span></span></a></li>
<li class="chapter" data-level="14.1.4" data-path="resources.html"><a href="resources.html#resources-for-phyloseq"><i class="fa fa-check"></i><b>14.1.4</b> Resources for phyloseq<span></span></a></li>
</ul></li>
<li class="chapter" data-level="14.2" data-path="resources.html"><a href="resources.html#r-programming-resources"><i class="fa fa-check"></i><b>14.2</b> R programming resources<span></span></a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="resources.html"><a href="resources.html#bioc_intro"><i class="fa fa-check"></i><b>14.2.1</b> Bioconductor Classes<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="extras.html"><a href="extras.html"><i class="fa fa-check"></i><b>15</b> Extra material<span></span></a>
<ul>
<li class="chapter" data-level="15.1" data-path="extras.html"><a href="extras.html#interactive-3d-plots"><i class="fa fa-check"></i><b>15.1</b> Interactive 3D Plots<span></span></a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="acknowledgments.html"><a href="acknowledgments.html"><i class="fa fa-check"></i><b>16</b> Acknowledgments<span></span></a></li>
<li><a href="bibliography.html#bibliography">Bibliography<span></span></a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Orchestrating Microbiome Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="community-typing" class="section level1 hasAnchor" number="10">
<h1><span class="header-section-number">Chapter 10</span> Community typing<a href="community-typing.html#community-typing" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="dirichlet-multinomial-mixtures-dmm" class="section level2 hasAnchor" number="10.1">
<h2><span class="header-section-number">10.1</span> Dirichlet Multinomial Mixtures (DMM)<a href="community-typing.html#dirichlet-multinomial-mixtures-dmm" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This section focus on DMM analysis.</p>
<p>One technique that allows to search for groups of samples that are
similar to each other is the <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0030126">Dirichlet-Multinomial Mixture
Model</a>. In
DMM, we first determine the number of clusters (k) that best fit the
data (model evidence) using Laplace approximation. After fitting the
model with k clusters, we obtain for each sample k probabilities that
reflect the probability that a sample belongs to the given cluster.</p>
<p>Let’s cluster the data with DMM clustering.</p>
<div class="sourceCode" id="cb248"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb248-1"><a href="community-typing.html#cb248-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Runs model and calculates the most likely number of clusters from 1 to 7.</span></span>
<span id="cb248-2"><a href="community-typing.html#cb248-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Since this is a large dataset it takes long computational time.</span></span>
<span id="cb248-3"><a href="community-typing.html#cb248-3" aria-hidden="true" tabindex="-1"></a><span class="co"># For this reason we use only a subset of the data; agglomerated by Phylum as a rank.</span></span>
<span id="cb248-4"><a href="community-typing.html#cb248-4" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> GlobalPatterns</span>
<span id="cb248-5"><a href="community-typing.html#cb248-5" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">agglomerateByRank</span>(tse, <span class="at">rank =</span> <span class="st">&quot;Phylum&quot;</span>, <span class="at">agglomerateTree=</span><span class="cn">TRUE</span>)</span></code></pre></div>
<div class="sourceCode" id="cb249"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb249-1"><a href="community-typing.html#cb249-1" aria-hidden="true" tabindex="-1"></a>tse_dmn <span class="ot">&lt;-</span> mia<span class="sc">::</span><span class="fu">runDMN</span>(tse, <span class="at">name =</span> <span class="st">&quot;DMN&quot;</span>, <span class="at">k =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">7</span>)</span></code></pre></div>
<div class="sourceCode" id="cb250"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb250-1"><a href="community-typing.html#cb250-1" aria-hidden="true" tabindex="-1"></a><span class="co"># It is stored in metadata</span></span>
<span id="cb250-2"><a href="community-typing.html#cb250-2" aria-hidden="true" tabindex="-1"></a>tse_dmn</span></code></pre></div>
<pre><code>## class: TreeSummarizedExperiment 
## dim: 67 26 
## metadata(2): agglomerated_by_rank DMN
## assays(1): counts
## rownames(67): Phylum:Crenarchaeota Phylum:Euryarchaeota ...
##   Phylum:Synergistetes Phylum:SR1
## rowData names(7): Kingdom Phylum ... Genus Species
## colnames(26): CL3 CC1 ... Even2 Even3
## colData names(7): X.SampleID Primer ... SampleType Description
## reducedDimNames(0):
## mainExpName: NULL
## altExpNames(0):
## rowLinks: a LinkDataFrame (67 rows)
## rowTree: 1 phylo tree(s) (66 leaves)
## colLinks: NULL
## colTree: NULL</code></pre>
<p>Return information on metadata that the object contains.</p>
<div class="sourceCode" id="cb252"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb252-1"><a href="community-typing.html#cb252-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(<span class="fu">metadata</span>(tse_dmn))</span></code></pre></div>
<pre><code>## [1] &quot;agglomerated_by_rank&quot; &quot;DMN&quot;</code></pre>
<p>This returns a list of DMN objects for a closer investigation.</p>
<div class="sourceCode" id="cb254"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb254-1"><a href="community-typing.html#cb254-1" aria-hidden="true" tabindex="-1"></a><span class="fu">getDMN</span>(tse_dmn)</span></code></pre></div>
<pre><code>## [[1]]
## class: DMN 
## k: 1 
## samples x taxa: 26 x 67 
## Laplace: 7715 BIC: 7802 AIC: 7760 
## 
## [[2]]
## class: DMN 
## k: 2 
## samples x taxa: 26 x 67 
## Laplace: 7673 BIC: 7927 AIC: 7842 
## 
## [[3]]
## class: DMN 
## k: 3 
## samples x taxa: 26 x 67 
## Laplace: 7690 BIC: 8076 AIC: 7948 
## 
## [[4]]
## class: DMN 
## k: 4 
## samples x taxa: 26 x 67 
## Laplace: 7752 BIC: 8274 AIC: 8103 
## 
## [[5]]
## class: DMN 
## k: 5 
## samples x taxa: 26 x 67 
## Laplace: 7854 BIC: 8553 AIC: 8340 
## 
## [[6]]
## class: DMN 
## k: 6 
## samples x taxa: 26 x 67 
## Laplace: 7971 BIC: 8850 AIC: 8594 
## 
## [[7]]
## class: DMN 
## k: 7 
## samples x taxa: 26 x 67 
## Laplace: 8003 BIC: 9051 AIC: 8752</code></pre>
<p>Show Laplace approximation (model evidence) for each model of the k models.</p>
<div class="sourceCode" id="cb256"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb256-1"><a href="community-typing.html#cb256-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(miaViz)</span>
<span id="cb256-2"><a href="community-typing.html#cb256-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDMNFit</span>(tse_dmn, <span class="at">type =</span> <span class="st">&quot;laplace&quot;</span>)</span></code></pre></div>
<p><img src="21_microbiome_community_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
<p>Return the model that has the best fit.</p>
<div class="sourceCode" id="cb257"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb257-1"><a href="community-typing.html#cb257-1" aria-hidden="true" tabindex="-1"></a><span class="fu">getBestDMNFit</span>(tse_dmn, <span class="at">type =</span> <span class="st">&quot;laplace&quot;</span>)</span></code></pre></div>
<pre><code>## class: DMN 
## k: 2 
## samples x taxa: 26 x 67 
## Laplace: 7673 BIC: 7927 AIC: 7842</code></pre>
<div id="pcoa-for-asv-level-data-with-bray-curtis-with-dmm-clusters-shown-with-colors" class="section level3 hasAnchor" number="10.1.1">
<h3><span class="header-section-number">10.1.1</span> PCoA for ASV-level data with Bray-Curtis; with DMM clusters shown with colors<a href="community-typing.html#pcoa-for-asv-level-data-with-bray-curtis-with-dmm-clusters-shown-with-colors" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Group samples and return DMNGroup object that contains a summary.
Patient status is used for grouping.</p>
<div class="sourceCode" id="cb259"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb259-1"><a href="community-typing.html#cb259-1" aria-hidden="true" tabindex="-1"></a>dmn_group <span class="ot">&lt;-</span> <span class="fu">calculateDMNgroup</span>(tse_dmn, <span class="at">variable =</span> <span class="st">&quot;SampleType&quot;</span>,  <span class="at">exprs_values =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb259-2"><a href="community-typing.html#cb259-2" aria-hidden="true" tabindex="-1"></a>                               <span class="at">k =</span> <span class="dv">2</span>, <span class="at">seed=</span>.Machine<span class="sc">$</span>integer.max)</span>
<span id="cb259-3"><a href="community-typing.html#cb259-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb259-4"><a href="community-typing.html#cb259-4" aria-hidden="true" tabindex="-1"></a>dmn_group</span></code></pre></div>
<pre><code>## class: DMNGroup 
## summary:
##                    k samples taxa    NLE  LogDet Laplace    BIC  AIC
## Feces              2       4   67 1078.3 -106.19   901.1 1171.9 1213
## Freshwater         2       2   67  889.6  -97.28   716.9  936.4 1025
## Freshwater (creek) 2       3   67 1600.3  860.08  1906.3 1674.5 1735
## Mock               2       3   67 1008.4  -55.37   856.6 1082.5 1143
## Ocean              2       3   67 1100.0  -44.82   953.6 1174.2 1235
## Sediment (estuary) 2       3   67 1195.5   18.63  1080.8 1269.7 1331
## Skin               2       3   67  992.6  -84.81   826.2 1066.8 1128
## Soil               2       3   67 1380.3   11.21  1261.8 1454.5 1515
## Tongue             2       2   67  783.0 -107.74   605.1  829.8  918</code></pre>
<p>Mixture weights (rough measure of the cluster size).</p>
<div class="sourceCode" id="cb261"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb261-1"><a href="community-typing.html#cb261-1" aria-hidden="true" tabindex="-1"></a>DirichletMultinomial<span class="sc">::</span><span class="fu">mixturewt</span>(<span class="fu">getBestDMNFit</span>(tse_dmn))</span></code></pre></div>
<pre><code>##       pi theta
## 1 0.5385 20.60
## 2 0.4615 15.28</code></pre>
<p>Samples-cluster assignment probabilities / how probable it is that sample belongs
to each cluster</p>
<div class="sourceCode" id="cb263"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb263-1"><a href="community-typing.html#cb263-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DirichletMultinomial<span class="sc">::</span><span class="fu">mixture</span>(<span class="fu">getBestDMNFit</span>(tse_dmn)))</span></code></pre></div>
<pre><code>##              [,1]      [,2]
## CL3     1.000e+00 5.036e-17
## CC1     1.000e+00 3.835e-22
## SV1     1.000e+00 2.027e-12
## M31Fcsw 7.331e-26 1.000e+00
## M11Fcsw 1.065e-16 1.000e+00
## M31Plmr 9.973e-14 1.000e+00</code></pre>
<p>Contribution of each taxa to each component</p>
<div class="sourceCode" id="cb265"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb265-1"><a href="community-typing.html#cb265-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DirichletMultinomial<span class="sc">::</span><span class="fu">fitted</span>(<span class="fu">getBestDMNFit</span>(tse_dmn)))</span></code></pre></div>
<pre><code>##                         [,1]      [,2]
## Phylum:Crenarchaeota  0.3043 0.1354649
## Phylum:Euryarchaeota  0.2314 0.1468628
## Phylum:Actinobacteria 1.2107 1.0600329
## Phylum:Spirochaetes   0.2141 0.1318411
## Phylum:MVP-15         0.0299 0.0007659
## Phylum:Proteobacteria 6.8413 1.8151938</code></pre>
<p>Get the assignment probabilities</p>
<div class="sourceCode" id="cb267"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb267-1"><a href="community-typing.html#cb267-1" aria-hidden="true" tabindex="-1"></a>prob <span class="ot">&lt;-</span> DirichletMultinomial<span class="sc">::</span><span class="fu">mixture</span>(<span class="fu">getBestDMNFit</span>(tse_dmn))</span>
<span id="cb267-2"><a href="community-typing.html#cb267-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Add column names</span></span>
<span id="cb267-3"><a href="community-typing.html#cb267-3" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(prob) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;comp1&quot;</span>, <span class="st">&quot;comp2&quot;</span>)</span>
<span id="cb267-4"><a href="community-typing.html#cb267-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb267-5"><a href="community-typing.html#cb267-5" aria-hidden="true" tabindex="-1"></a><span class="co"># For each row, finds column that has the highest value. Then extract the column </span></span>
<span id="cb267-6"><a href="community-typing.html#cb267-6" aria-hidden="true" tabindex="-1"></a><span class="co"># names of highest values.</span></span>
<span id="cb267-7"><a href="community-typing.html#cb267-7" aria-hidden="true" tabindex="-1"></a>vec <span class="ot">&lt;-</span> <span class="fu">colnames</span>(prob)[<span class="fu">max.col</span>(prob,<span class="at">ties.method =</span> <span class="st">&quot;first&quot;</span>)]</span></code></pre></div>
<p>Computing the euclidean PCoA and storing it as a data frame</p>
<div class="sourceCode" id="cb268"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb268-1"><a href="community-typing.html#cb268-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Does clr transformation. Pseudocount is added, because data contains zeros.</span></span>
<span id="cb268-2"><a href="community-typing.html#cb268-2" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">transformCounts</span>(tse, <span class="at">method =</span> <span class="st">&quot;clr&quot;</span>, <span class="at">pseudocount =</span> <span class="dv">1</span>)</span>
<span id="cb268-3"><a href="community-typing.html#cb268-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb268-4"><a href="community-typing.html#cb268-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Gets clr table</span></span>
<span id="cb268-5"><a href="community-typing.html#cb268-5" aria-hidden="true" tabindex="-1"></a>clr_assay <span class="ot">&lt;-</span> <span class="fu">assays</span>(tse)<span class="sc">$</span>clr</span>
<span id="cb268-6"><a href="community-typing.html#cb268-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb268-7"><a href="community-typing.html#cb268-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Transposes it to get taxa to columns</span></span>
<span id="cb268-8"><a href="community-typing.html#cb268-8" aria-hidden="true" tabindex="-1"></a>clr_assay <span class="ot">&lt;-</span> <span class="fu">t</span>(clr_assay)</span>
<span id="cb268-9"><a href="community-typing.html#cb268-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb268-10"><a href="community-typing.html#cb268-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculates Euclidean distances between samples. Because taxa is in columns,</span></span>
<span id="cb268-11"><a href="community-typing.html#cb268-11" aria-hidden="true" tabindex="-1"></a><span class="co"># it is used to compare different samples.</span></span>
<span id="cb268-12"><a href="community-typing.html#cb268-12" aria-hidden="true" tabindex="-1"></a>euclidean_dist <span class="ot">&lt;-</span> vegan<span class="sc">::</span><span class="fu">vegdist</span>(clr_assay, <span class="at">method =</span> <span class="st">&quot;euclidean&quot;</span>)</span>
<span id="cb268-13"><a href="community-typing.html#cb268-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb268-14"><a href="community-typing.html#cb268-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Does principal coordinate analysis</span></span>
<span id="cb268-15"><a href="community-typing.html#cb268-15" aria-hidden="true" tabindex="-1"></a>euclidean_pcoa <span class="ot">&lt;-</span> ecodist<span class="sc">::</span><span class="fu">pco</span>(euclidean_dist)</span>
<span id="cb268-16"><a href="community-typing.html#cb268-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb268-17"><a href="community-typing.html#cb268-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates a data frame from principal coordinates</span></span>
<span id="cb268-18"><a href="community-typing.html#cb268-18" aria-hidden="true" tabindex="-1"></a>euclidean_pcoa_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">pcoa1 =</span> euclidean_pcoa<span class="sc">$</span>vectors[,<span class="dv">1</span>], </span>
<span id="cb268-19"><a href="community-typing.html#cb268-19" aria-hidden="true" tabindex="-1"></a>                                <span class="at">pcoa2 =</span> euclidean_pcoa<span class="sc">$</span>vectors[,<span class="dv">2</span>])</span></code></pre></div>
<div class="sourceCode" id="cb269"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb269-1"><a href="community-typing.html#cb269-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates a data frame that contains principal coordinates and DMM information</span></span>
<span id="cb269-2"><a href="community-typing.html#cb269-2" aria-hidden="true" tabindex="-1"></a>euclidean_dmm_pcoa_df <span class="ot">&lt;-</span> <span class="fu">cbind</span>(euclidean_pcoa_df,</span>
<span id="cb269-3"><a href="community-typing.html#cb269-3" aria-hidden="true" tabindex="-1"></a>                               <span class="at">dmm_component =</span> vec)</span>
<span id="cb269-4"><a href="community-typing.html#cb269-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates a plot</span></span>
<span id="cb269-5"><a href="community-typing.html#cb269-5" aria-hidden="true" tabindex="-1"></a>euclidean_dmm_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> euclidean_dmm_pcoa_df, </span>
<span id="cb269-6"><a href="community-typing.html#cb269-6" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">aes</span>(<span class="at">x=</span>pcoa1, <span class="at">y=</span>pcoa2,</span>
<span id="cb269-7"><a href="community-typing.html#cb269-7" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">color =</span> dmm_component)) <span class="sc">+</span></span>
<span id="cb269-8"><a href="community-typing.html#cb269-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb269-9"><a href="community-typing.html#cb269-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Coordinate 1&quot;</span>,</span>
<span id="cb269-10"><a href="community-typing.html#cb269-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">&quot;Coordinate 2&quot;</span>,</span>
<span id="cb269-11"><a href="community-typing.html#cb269-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">&quot;PCoA with Aitchison distances&quot;</span>) <span class="sc">+</span>  </span>
<span id="cb269-12"><a href="community-typing.html#cb269-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>)) <span class="co"># makes titles smaller</span></span>
<span id="cb269-13"><a href="community-typing.html#cb269-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb269-14"><a href="community-typing.html#cb269-14" aria-hidden="true" tabindex="-1"></a>euclidean_dmm_plot</span></code></pre></div>
<p><img src="21_microbiome_community_files/figure-html/unnamed-chunk-13-1.png" width="672" /></p>
</div>
</div>
<div id="community-detection" class="section level2 hasAnchor" number="10.2">
<h2><span class="header-section-number">10.2</span> Community Detection<a href="community-typing.html#community-detection" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Another approach for discovering communities within the samples of the
data, is to run community detection algorithms after building a
graph. The following demonstration builds a graph based on the k
nearest-neighbors and performs the community detection on the fly.</p>
<p><em><code>bluster</code></em> <span class="citation">(<a href="#ref-R-bluster" role="doc-biblioref">Lun 2021</a>)</span> package offers several clustering methods,
among which graph-based are present, enabling the community detection
task.</p>
<p>Installing package:</p>
<div class="sourceCode" id="cb270"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb270-1"><a href="community-typing.html#cb270-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(bluster)){</span>
<span id="cb270-2"><a href="community-typing.html#cb270-2" aria-hidden="true" tabindex="-1"></a>  BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;bluster&quot;</span>)</span>
<span id="cb270-3"><a href="community-typing.html#cb270-3" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p>The algorithm used is “short random walks” <span class="citation">(<a href="#ref-Pons2006" role="doc-biblioref">Pons and Latapy 2006</a>)</span>. Graph is
constructed using different k values (the number of nearest neighbors
to consider during graph construction) using the robust centered log
ratio (rclr) assay data. Then plotting the communities using UMAP
<span class="citation">(<a href="#ref-McInnes2018" role="doc-biblioref">McInnes, Healy, and Melville 2018</a>)</span> ordination as a visual exploration aid. In the
following demonstration we use the <code>enterotype</code> dataset from the
<span class="citation">(<a href="#ref-R-mia" role="doc-biblioref">Ernst, Shetty, and Lahti 2020</a>)</span> package.</p>
<div class="sourceCode" id="cb271"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb271-1"><a href="community-typing.html#cb271-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(bluster)</span>
<span id="cb271-2"><a href="community-typing.html#cb271-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork) <span class="co"># For arranging several plots as a grid</span></span>
<span id="cb271-3"><a href="community-typing.html#cb271-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scater)</span>
<span id="cb271-4"><a href="community-typing.html#cb271-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb271-5"><a href="community-typing.html#cb271-5" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;enterotype&quot;</span>, <span class="at">package=</span><span class="st">&quot;mia&quot;</span>)</span>
<span id="cb271-6"><a href="community-typing.html#cb271-6" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> enterotype</span>
<span id="cb271-7"><a href="community-typing.html#cb271-7" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">transformCounts</span>(tse, <span class="at">method =</span> <span class="st">&quot;rclr&quot;</span>)</span>
<span id="cb271-8"><a href="community-typing.html#cb271-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb271-9"><a href="community-typing.html#cb271-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Performing and storing UMAP</span></span>
<span id="cb271-10"><a href="community-typing.html#cb271-10" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">runUMAP</span>(tse, <span class="at">name=</span><span class="st">&quot;UMAP&quot;</span>, <span class="at">exprs_values=</span><span class="st">&quot;rclr&quot;</span>)</span>
<span id="cb271-11"><a href="community-typing.html#cb271-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb271-12"><a href="community-typing.html#cb271-12" aria-hidden="true" tabindex="-1"></a>k <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">5</span>,<span class="dv">10</span>)</span>
<span id="cb271-13"><a href="community-typing.html#cb271-13" aria-hidden="true" tabindex="-1"></a>ClustAndPlot <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb271-14"><a href="community-typing.html#cb271-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Creating the graph and running the short random walks algorithm  </span></span>
<span id="cb271-15"><a href="community-typing.html#cb271-15" aria-hidden="true" tabindex="-1"></a>  graph_clusters <span class="ot">&lt;-</span> <span class="fu">clusterRows</span>(<span class="fu">t</span>(<span class="fu">assays</span>(tse)<span class="sc">$</span>rclr), <span class="fu">NNGraphParam</span>(<span class="at">k=</span>x))</span>
<span id="cb271-16"><a href="community-typing.html#cb271-16" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb271-17"><a href="community-typing.html#cb271-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Results of the clustering as a color for each sample</span></span>
<span id="cb271-18"><a href="community-typing.html#cb271-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plotUMAP</span>(tse, <span class="at">colour_by =</span> <span class="fu">I</span>(graph_clusters)) <span class="sc">+</span></span>
<span id="cb271-19"><a href="community-typing.html#cb271-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste0</span>(<span class="st">&quot;k = &quot;</span>, x))</span>
<span id="cb271-20"><a href="community-typing.html#cb271-20" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb271-21"><a href="community-typing.html#cb271-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb271-22"><a href="community-typing.html#cb271-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Applying the function for different k values</span></span>
<span id="cb271-23"><a href="community-typing.html#cb271-23" aria-hidden="true" tabindex="-1"></a>plots <span class="ot">&lt;-</span> <span class="fu">lapply</span>(k,ClustAndPlot)</span>
<span id="cb271-24"><a href="community-typing.html#cb271-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb271-25"><a href="community-typing.html#cb271-25" aria-hidden="true" tabindex="-1"></a><span class="co"># Displaying plots in a grid</span></span>
<span id="cb271-26"><a href="community-typing.html#cb271-26" aria-hidden="true" tabindex="-1"></a>(plots[[<span class="dv">1</span>]] <span class="sc">+</span> plots[[<span class="dv">2</span>]]) <span class="sc">/</span> (plots[[<span class="dv">3</span>]] <span class="sc">+</span> plots[[<span class="dv">4</span>]])</span></code></pre></div>
<p><img src="21_microbiome_community_files/figure-html/unnamed-chunk-15-1.png" width="672" /></p>
<p>Similarly, the <em><code>bluster</code></em> <span class="citation">(<a href="#ref-R-bluster" role="doc-biblioref">Lun 2021</a>)</span> package offers clustering
diagnostics that can be used for judging the clustering quality (see
<a href="http://bioconductor.org/packages/release/bioc/vignettes/bluster/inst/doc/diagnostics.html">Assorted clustering
diagnostics</a>).
In the following, Silhouette width as a diagnostic tool is computed
and results are visualized for each case presented earlier. For more
about Silhouettes read <span class="citation">(<a href="#ref-Rousseeuw1987" role="doc-biblioref">Rousseeuw 1987</a>)</span>.</p>
<div class="sourceCode" id="cb272"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb272-1"><a href="community-typing.html#cb272-1" aria-hidden="true" tabindex="-1"></a>ClustDiagPlot <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb272-2"><a href="community-typing.html#cb272-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Getting the clustering results</span></span>
<span id="cb272-3"><a href="community-typing.html#cb272-3" aria-hidden="true" tabindex="-1"></a>  graph_clusters <span class="ot">&lt;-</span> <span class="fu">clusterRows</span>(<span class="fu">t</span>(<span class="fu">assays</span>(tse)<span class="sc">$</span>rclr), <span class="fu">NNGraphParam</span>(<span class="at">k=</span>x))</span>
<span id="cb272-4"><a href="community-typing.html#cb272-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb272-5"><a href="community-typing.html#cb272-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Computing the diagnostic info</span></span>
<span id="cb272-6"><a href="community-typing.html#cb272-6" aria-hidden="true" tabindex="-1"></a>  sil <span class="ot">&lt;-</span> <span class="fu">approxSilhouette</span>(<span class="fu">t</span>(<span class="fu">assays</span>(tse)<span class="sc">$</span>rclr), graph_clusters)</span>
<span id="cb272-7"><a href="community-typing.html#cb272-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb272-8"><a href="community-typing.html#cb272-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plotting as a boxlpot to observe cluster separation</span></span>
<span id="cb272-9"><a href="community-typing.html#cb272-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">boxplot</span>(<span class="fu">split</span>(sil<span class="sc">$</span>width, graph_clusters), <span class="at">main=</span><span class="fu">paste0</span>(<span class="st">&quot;k = &quot;</span>, x))</span>
<span id="cb272-10"><a href="community-typing.html#cb272-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb272-11"><a href="community-typing.html#cb272-11" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb272-12"><a href="community-typing.html#cb272-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Applying the function for different k values</span></span>
<span id="cb272-13"><a href="community-typing.html#cb272-13" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">lapply</span>(k,ClustDiagPlot)</span></code></pre></div>
<p><img src="21_microbiome_community_files/figure-html/unnamed-chunk-16-1.png" width="672" /><img src="21_microbiome_community_files/figure-html/unnamed-chunk-16-2.png" width="672" /><img src="21_microbiome_community_files/figure-html/unnamed-chunk-16-3.png" width="672" /><img src="21_microbiome_community_files/figure-html/unnamed-chunk-16-4.png" width="672" /></p>
</div>
<div id="additional-community-typing" class="section level2 hasAnchor" number="10.3">
<h2><span class="header-section-number">10.3</span> Additional Community Typing<a href="community-typing.html#additional-community-typing" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For more community typing techniques applied to the ‘SprockettTHData’ data set, see the attached .Rmd file.</p>
<p>Link:</p>
<ul>
<li><a href="add-comm-typing.Rmd">Rmd</a></li>
</ul>
</div>
<div id="session-info-6" class="section level2 unnumbered hasAnchor">
<h2>Session Info<a href="community-typing.html#session-info-6" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<button class="rebook-collapse">
View session info
</button>
<div class="rebook-content">
<pre><code>R version 4.1.2 (2021-11-01)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.3 LTS

Matrix products: default
BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] scater_1.22.0                  scuttle_1.4.0                 
 [3] bluster_1.4.0                  patchwork_1.1.1               
 [5] reshape2_1.4.4                 sechm_1.2.0                   
 [7] ggtree_3.2.1                   ape_5.6-2                     
 [9] pheatmap_1.0.12                miaViz_1.3.3                  
[11] ggraph_2.0.5                   ggplot2_3.3.5                 
[13] mia_1.3.19                     MultiAssayExperiment_1.20.0   
[15] TreeSummarizedExperiment_2.1.4 Biostrings_2.62.0             
[17] XVector_0.34.0                 SingleCellExperiment_1.16.0   
[19] SummarizedExperiment_1.24.0    Biobase_2.54.0                
[21] GenomicRanges_1.46.1           GenomeInfoDb_1.30.1           
[23] IRanges_2.28.0                 S4Vectors_0.32.3              
[25] BiocGenerics_0.40.0            MatrixGenerics_1.6.0          
[27] matrixStats_0.61.0-9001        ecodist_2.0.7                 
[29] BiocStyle_2.22.0               rebook_1.4.0                  

loaded via a namespace (and not attached):
  [1] utf8_1.2.2                  tidyselect_1.1.2           
  [3] RSQLite_2.2.10              grid_4.1.2                 
  [5] TSP_1.2-0                   BiocParallel_1.28.3        
  [7] Rtsne_0.15                  munsell_0.5.0              
  [9] ScaledMatrix_1.2.0          codetools_0.2-18           
 [11] withr_2.5.0                 colorspace_2.0-3           
 [13] filelock_1.0.2              highr_0.9                  
 [15] knitr_1.37                  labeling_0.4.2             
 [17] GenomeInfoDbData_1.2.7      polyclip_1.10-0            
 [19] bit64_4.0.5                 farver_2.1.0               
 [21] vctrs_0.3.8                 treeio_1.18.1              
 [23] generics_0.1.2              xfun_0.30                  
 [25] R6_2.5.1                    doParallel_1.0.17          
 [27] ggbeeswarm_0.6.0            clue_0.3-60                
 [29] graphlayouts_0.8.0          rsvd_1.0.5                 
 [31] seriation_1.3.3             bitops_1.0-7               
 [33] cachem_1.0.6                gridGraphics_0.5-1         
 [35] DelayedArray_0.20.0         assertthat_0.2.1           
 [37] scales_1.1.1                beeswarm_0.4.0             
 [39] gtable_0.3.0                beachmat_2.10.0            
 [41] tidygraph_1.2.0             rlang_1.0.2                
 [43] GlobalOptions_0.1.2         splines_4.1.2              
 [45] lazyeval_0.2.2              BiocManager_1.30.16        
 [47] yaml_2.3.5                  tools_4.1.2                
 [49] bookdown_0.25               ggplotify_0.1.0            
 [51] ellipsis_0.3.2              decontam_1.14.0            
 [53] jquerylib_0.1.4             RColorBrewer_1.1-2         
 [55] Rcpp_1.0.8.3                plyr_1.8.6                 
 [57] sparseMatrixStats_1.6.0     zlibbioc_1.40.0            
 [59] purrr_0.3.4                 RCurl_1.98-1.6             
 [61] GetoptLong_1.0.5            viridis_0.6.2              
 [63] cowplot_1.1.1               ggrepel_0.9.1              
 [65] cluster_2.1.2               DECIPHER_2.22.0            
 [67] magrittr_2.0.2              circlize_0.4.14            
 [69] ggnewscale_0.4.6            randomcoloR_1.1.0.1        
 [71] evaluate_0.15               XML_3.99-0.9               
 [73] gridExtra_2.3               shape_1.4.6                
 [75] compiler_4.1.2              tibble_3.1.6               
 [77] V8_4.1.0                    crayon_1.5.0               
 [79] htmltools_0.5.2             ggfun_0.0.5                
 [81] mgcv_1.8-39                 tidyr_1.2.0                
 [83] aplot_0.1.2                 DBI_1.1.2                  
 [85] tweenr_1.0.2                ComplexHeatmap_2.10.0      
 [87] MASS_7.3-55                 Matrix_1.4-0               
 [89] permute_0.9-7               cli_3.2.0                  
 [91] parallel_4.1.2              igraph_1.2.11              
 [93] pkgconfig_2.0.3             dir.expiry_1.2.0           
 [95] registry_0.5-1              foreach_1.5.2              
 [97] vipor_0.4.5                 bslib_0.3.1                
 [99] DirichletMultinomial_1.36.0 yulab.utils_0.0.4          
[101] stringr_1.4.0               digest_0.6.29              
[103] vegan_2.5-7                 graph_1.72.0               
[105] rmarkdown_2.13              tidytree_0.3.9             
[107] uwot_0.1.11                 DelayedMatrixStats_1.16.0  
[109] curl_4.3.2                  rjson_0.2.21               
[111] lifecycle_1.0.1             nlme_3.1-155               
[113] jsonlite_1.8.0              BiocNeighbors_1.12.0       
[115] CodeDepends_0.6.5           viridisLite_0.4.0          
[117] fansi_1.0.2                 pillar_1.7.0               
[119] lattice_0.20-45             fastmap_1.1.0              
[121] glue_1.6.2                  FNN_1.1.3                  
[123] png_0.1-7                   iterators_1.0.14           
[125] bit_4.0.4                   ggforce_0.3.3              
[127] stringi_1.7.6               sass_0.4.0                 
[129] blob_1.2.2                  BiocSingular_1.10.0        
[131] memoise_2.0.1               dplyr_1.0.8                
[133] irlba_2.3.5                </code></pre>
</div>

</div>
</div>
<h3>Bibliography<a href="bibliography.html#bibliography" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-R-mia" class="csl-entry">
Ernst, Felix G. M., Sudarshan Shetty, and Leo Lahti. 2020. <em>Mia: Microbiome Analysis</em>.
</div>
<div id="ref-R-bluster" class="csl-entry">
Lun, Aaron. 2021. <em><span class="nocase">bluster: Clustering Algorithms for Bioconductor</span></em>. <a href="https://bioconductor.org/packages/3.14/bioc/html/bluster.html">https://bioconductor.org/packages/3.14/bioc/html/bluster.html</a>.
</div>
<div id="ref-McInnes2018" class="csl-entry">
McInnes, Leland, John Healy, and James Melville. 2018. <span>“<span class="nocase">UMAP: Uniform Manifold Approximation and Projection for Dimension Reduction</span>.”</span> <em>arXiv e-Prints</em>, arXiv:1802.03426. <a href="https://arxiv.org/abs/1802.03426">https://arxiv.org/abs/1802.03426</a>.
</div>
<div id="ref-Pons2006" class="csl-entry">
Pons, Pascal, and Matthieu Latapy. 2006. <span>“Computing Communities in Large Networks Using Random Walks.”</span> <em>Journal of Graph Algorithms and Applications</em> 10 (January): 191–218. <a href="https://doi.org/10.7155/jgaa.00124">https://doi.org/10.7155/jgaa.00124</a>.
</div>
<div id="ref-Rousseeuw1987" class="csl-entry">
Rousseeuw, Peter J. 1987. <span>“Silhouettes: A Graphical Aid to the Interpretation and Validation of Cluster Analysis.”</span> <em>Journal of Computational and Applied Mathematics</em> 20: 53–65. https://doi.org/<a href="https://doi.org/10.1016/0377-0427(87)90125-7">https://doi.org/10.1016/0377-0427(87)90125-7</a>.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="microbiome-community.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="biclustering.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["OMA.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
