<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Microbiome Data | Orchestrating Microbiome Analysis</title>
  <meta name="description" content="Chapter 2 Microbiome Data | Orchestrating Microbiome Analysis" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Microbiome Data | Orchestrating Microbiome Analysis" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://github.com/microbiome/OMA" />
  
  
  <meta name="github-repo" content="microbiome/OMA" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Microbiome Data | Orchestrating Microbiome Analysis" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="intro.html"/>
<link rel="next" href="packages.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<script src="libs/plotly-binding-4.10.0/plotly.js"></script>
<script src="libs/typedarray-0.1/typedarray.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="libs/plotly-htmlwidgets-css-2.5.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="libs/plotly-main-2.5.1/plotly-latest.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Orchestrating Microbiome Analysis</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="part"><span><b>I Introduction</b></span></li>
<li class="chapter" data-level="1" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="containers.html"><a href="containers.html"><i class="fa fa-check"></i><b>2</b> Microbiome Data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="containers.html"><a href="containers.html#data-science-framework"><i class="fa fa-check"></i><b>2.1</b> Data science framework</a></li>
<li class="chapter" data-level="2.2" data-path="containers.html"><a href="containers.html#data-containers"><i class="fa fa-check"></i><b>2.2</b> Data containers</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="containers.html"><a href="containers.html#assay-data"><i class="fa fa-check"></i><b>2.2.1</b> Assay data</a></li>
<li class="chapter" data-level="2.2.2" data-path="containers.html"><a href="containers.html#coldata"><i class="fa fa-check"></i><b>2.2.2</b> colData</a></li>
<li class="chapter" data-level="2.2.3" data-path="containers.html"><a href="containers.html#rowdata"><i class="fa fa-check"></i><b>2.2.3</b> rowData</a></li>
<li class="chapter" data-level="2.2.4" data-path="containers.html"><a href="containers.html#rowtree"><i class="fa fa-check"></i><b>2.2.4</b> rowTree</a></li>
<li class="chapter" data-level="2.2.5" data-path="containers.html"><a href="containers.html#alternative-experiments"><i class="fa fa-check"></i><b>2.2.5</b> Alternative experiments</a></li>
<li class="chapter" data-level="2.2.6" data-path="containers.html"><a href="containers.html#multiassayexperiments"><i class="fa fa-check"></i><b>2.2.6</b> MultiAssayExperiments</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="containers.html"><a href="containers.html#loading-experimental-microbiome-data"><i class="fa fa-check"></i><b>2.3</b> Loading experimental microbiome data</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="containers.html"><a href="containers.html#import-from-external-files"><i class="fa fa-check"></i><b>2.3.1</b> Import from external files</a></li>
<li class="chapter" data-level="2.3.2" data-path="containers.html"><a href="containers.html#conversions-between-data-formats-in-r"><i class="fa fa-check"></i><b>2.3.2</b> Conversions between data formats in R</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="containers.html"><a href="containers.html#example-data"><i class="fa fa-check"></i><b>2.4</b> Demonstration data</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="containers.html"><a href="containers.html#package-data"><i class="fa fa-check"></i><b>2.4.1</b> Package data</a></li>
<li class="chapter" data-level="2.4.2" data-path="containers.html"><a href="containers.html#experimenthub-data"><i class="fa fa-check"></i><b>2.4.2</b> ExperimentHub data</a></li>
<li class="chapter" data-level="2.4.3" data-path="containers.html"><a href="containers.html#other-data-sources"><i class="fa fa-check"></i><b>2.4.3</b> Other data sources</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="containers.html"><a href="containers.html#session-info"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="packages.html"><a href="packages.html"><i class="fa fa-check"></i><b>3</b> Packages</a>
<ul>
<li class="chapter" data-level="3.1" data-path="packages.html"><a href="packages.html#package-installation"><i class="fa fa-check"></i><b>3.1</b> Package installation</a></li>
<li class="chapter" data-level="3.2" data-path="packages.html"><a href="packages.html#some-available-packages"><i class="fa fa-check"></i><b>3.2</b> Some available packages</a></li>
<li class="chapter" data-level="" data-path="packages.html"><a href="packages.html#session-info-1"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="part"><span><b>II Focus Topics</b></span></li>
<li class="chapter" data-level="4" data-path="datamanipulation.html"><a href="datamanipulation.html"><i class="fa fa-check"></i><b>4</b> Data Manipulation</a>
<ul>
<li class="chapter" data-level="4.1" data-path="datamanipulation.html"><a href="datamanipulation.html#tidying-and-subsetting"><i class="fa fa-check"></i><b>4.1</b> Tidying and subsetting</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="datamanipulation.html"><a href="datamanipulation.html#tidy-data"><i class="fa fa-check"></i><b>4.1.1</b> Tidy data</a></li>
<li class="chapter" data-level="4.1.2" data-path="datamanipulation.html"><a href="datamanipulation.html#subsetting"><i class="fa fa-check"></i><b>4.1.2</b> Subsetting</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>5</b> Exploration and quality Control</a>
<ul>
<li class="chapter" data-level="5.1" data-path="quality-control.html"><a href="quality-control.html#abundance"><i class="fa fa-check"></i><b>5.1</b> Abundance</a></li>
<li class="chapter" data-level="5.2" data-path="quality-control.html"><a href="quality-control.html#prevalence"><i class="fa fa-check"></i><b>5.2</b> Prevalence</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="quality-control.html"><a href="quality-control.html#prevalent-microbiota-analysis"><i class="fa fa-check"></i><b>5.2.1</b> Prevalent microbiota analysis</a></li>
<li class="chapter" data-level="5.2.2" data-path="quality-control.html"><a href="quality-control.html#plotting-prevalence"><i class="fa fa-check"></i><b>5.2.2</b> Plotting prevalence</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="quality-control.html"><a href="quality-control.html#quality-control-1"><i class="fa fa-check"></i><b>5.3</b> Quality control</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="quality-control.html"><a href="quality-control.html#top-taxa"><i class="fa fa-check"></i><b>5.3.1</b> Top taxa</a></li>
<li class="chapter" data-level="5.3.2" data-path="quality-control.html"><a href="quality-control.html#library-size"><i class="fa fa-check"></i><b>5.3.2</b> Library size</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="quality-control.html"><a href="quality-control.html#session-info-2"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="taxonomic-information.html"><a href="taxonomic-information.html"><i class="fa fa-check"></i><b>6</b> Taxonomic Information</a>
<ul>
<li class="chapter" data-level="6.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#assigning-taxonomic-information."><i class="fa fa-check"></i><b>6.1</b> Assigning taxonomic information.</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#dada2"><i class="fa fa-check"></i><b>6.1.1</b> dada2</a></li>
<li class="chapter" data-level="6.1.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#decipher"><i class="fa fa-check"></i><b>6.1.2</b> DECIPHER</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#functions-to-access-taxonomic-information"><i class="fa fa-check"></i><b>6.2</b> Functions to access taxonomic information</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#generate-a-taxonomic-tree-on-the-fly"><i class="fa fa-check"></i><b>6.2.1</b> Generate a taxonomic tree on the fly</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="taxonomic-information.html"><a href="taxonomic-information.html#data-agglomeration"><i class="fa fa-check"></i><b>6.3</b> Data agglomeration</a></li>
<li class="chapter" data-level="6.4" data-path="taxonomic-information.html"><a href="taxonomic-information.html#data-transformation"><i class="fa fa-check"></i><b>6.4</b> Data transformation</a></li>
<li class="chapter" data-level="6.5" data-path="taxonomic-information.html"><a href="taxonomic-information.html#pick-specific"><i class="fa fa-check"></i><b>6.5</b> Pick specific</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#abundances-of-all-taxa-in-specific-sample"><i class="fa fa-check"></i><b>6.5.1</b> Abundances of all taxa in specific sample</a></li>
<li class="chapter" data-level="6.5.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#abundances-of-specific-taxa-in-all-samples"><i class="fa fa-check"></i><b>6.5.2</b> Abundances of specific taxa in all samples</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="taxonomic-information.html"><a href="taxonomic-information.html#session-info-3"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="community-diversity.html"><a href="community-diversity.html"><i class="fa fa-check"></i><b>7</b> Community diversity</a>
<ul>
<li class="chapter" data-level="7.1" data-path="community-diversity.html"><a href="community-diversity.html#estimation"><i class="fa fa-check"></i><b>7.1</b> Estimation</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="community-diversity.html"><a href="community-diversity.html#richness"><i class="fa fa-check"></i><b>7.1.1</b> Richness</a></li>
<li class="chapter" data-level="7.1.2" data-path="community-diversity.html"><a href="community-diversity.html#diversity"><i class="fa fa-check"></i><b>7.1.2</b> Diversity</a></li>
<li class="chapter" data-level="7.1.3" data-path="community-diversity.html"><a href="community-diversity.html#evenness"><i class="fa fa-check"></i><b>7.1.3</b> Evenness</a></li>
<li class="chapter" data-level="7.1.4" data-path="community-diversity.html"><a href="community-diversity.html#dominance"><i class="fa fa-check"></i><b>7.1.4</b> Dominance</a></li>
<li class="chapter" data-level="7.1.5" data-path="community-diversity.html"><a href="community-diversity.html#rarity"><i class="fa fa-check"></i><b>7.1.5</b> Rarity</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="community-diversity.html"><a href="community-diversity.html#visualization"><i class="fa fa-check"></i><b>7.2</b> Visualization</a></li>
<li class="chapter" data-level="" data-path="community-diversity.html"><a href="community-diversity.html#session-info-4"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="community-similarity.html"><a href="community-similarity.html"><i class="fa fa-check"></i><b>8</b> Community similarity</a>
<ul>
<li class="chapter" data-level="8.1" data-path="community-similarity.html"><a href="community-similarity.html#explained-variance"><i class="fa fa-check"></i><b>8.1</b> Explained variance</a></li>
<li class="chapter" data-level="8.2" data-path="community-similarity.html"><a href="community-similarity.html#community-comparisons-by-beta-diversity-analysis"><i class="fa fa-check"></i><b>8.2</b> Community comparisons by beta diversity analysis</a></li>
<li class="chapter" data-level="8.3" data-path="community-similarity.html"><a href="community-similarity.html#other-ordination-methods"><i class="fa fa-check"></i><b>8.3</b> Other ordination methods</a></li>
<li class="chapter" data-level="8.4" data-path="community-similarity.html"><a href="community-similarity.html#visualizing-the-most-dominant-genus-on-pcoa"><i class="fa fa-check"></i><b>8.4</b> Visualizing the most dominant genus on PCoA</a>
<ul>
<li class="chapter" data-level="8.4.1" data-path="community-similarity.html"><a href="community-similarity.html#testing-differences-in-community-composition-between-sample-groups"><i class="fa fa-check"></i><b>8.4.1</b> Testing differences in community composition between sample groups</a></li>
<li class="chapter" data-level="8.4.2" data-path="community-similarity.html"><a href="community-similarity.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>8.4.2</b> Checking the homogeneity condition</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="community-similarity.html"><a href="community-similarity.html#further-reading"><i class="fa fa-check"></i><b>8.5</b> Further reading</a></li>
<li class="chapter" data-level="" data-path="community-similarity.html"><a href="community-similarity.html#session-info-5"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="microbiome-community.html"><a href="microbiome-community.html"><i class="fa fa-check"></i><b>9</b> Community composition</a>
<ul>
<li class="chapter" data-level="9.1" data-path="microbiome-community.html"><a href="microbiome-community.html#visualizing-taxonomic-composition"><i class="fa fa-check"></i><b>9.1</b> Visualizing taxonomic composition</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="microbiome-community.html"><a href="microbiome-community.html#composition-barplot"><i class="fa fa-check"></i><b>9.1.1</b> Composition barplot</a></li>
<li class="chapter" data-level="9.1.2" data-path="microbiome-community.html"><a href="microbiome-community.html#composition-heatmap"><i class="fa fa-check"></i><b>9.1.2</b> Composition heatmap</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="community-typing.html"><a href="community-typing.html"><i class="fa fa-check"></i><b>10</b> Community typing</a>
<ul>
<li class="chapter" data-level="10.0.1" data-path="community-typing.html"><a href="community-typing.html#dirichlet-multinomial-mixtures-dmm"><i class="fa fa-check"></i><b>10.0.1</b> Dirichlet Multinomial Mixtures (DMM)</a></li>
<li class="chapter" data-level="10.0.2" data-path="community-typing.html"><a href="community-typing.html#pcoa-for-asv-level-data-with-bray-curtis-with-dmm-clusters-shown-with-colors"><i class="fa fa-check"></i><b>10.0.2</b> PCoA for ASV-level data with Bray-Curtis; with DMM clusters shown with colors</a></li>
<li class="chapter" data-level="10.1" data-path="community-typing.html"><a href="community-typing.html#community-detection"><i class="fa fa-check"></i><b>10.1</b> Community Detection</a></li>
<li class="chapter" data-level="10.2" data-path="community-typing.html"><a href="community-typing.html#additional-community-typing"><i class="fa fa-check"></i><b>10.2</b> Additional Community Typing</a></li>
<li class="chapter" data-level="" data-path="community-typing.html"><a href="community-typing.html#session-info-6"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="differential-abundance.html"><a href="differential-abundance.html"><i class="fa fa-check"></i><b>11</b> Differential abundance</a>
<ul>
<li class="chapter" data-level="11.1" data-path="differential-abundance.html"><a href="differential-abundance.html#differential-abundance-analysis"><i class="fa fa-check"></i><b>11.1</b> Differential abundance analysis</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="differential-abundance.html"><a href="differential-abundance.html#examples-and-tools"><i class="fa fa-check"></i><b>11.1.1</b> Examples and tools</a></li>
<li class="chapter" data-level="11.1.2" data-path="differential-abundance.html"><a href="differential-abundance.html#prevalence-filtering"><i class="fa fa-check"></i><b>11.1.2</b> Prevalence Filtering</a></li>
<li class="chapter" data-level="11.1.3" data-path="differential-abundance.html"><a href="differential-abundance.html#aldex2"><i class="fa fa-check"></i><b>11.1.3</b> ALDEx2</a></li>
<li class="chapter" data-level="11.1.4" data-path="differential-abundance.html"><a href="differential-abundance.html#ancom-bc"><i class="fa fa-check"></i><b>11.1.4</b> ANCOM-BC</a></li>
<li class="chapter" data-level="11.1.5" data-path="differential-abundance.html"><a href="differential-abundance.html#maaslin2"><i class="fa fa-check"></i><b>11.1.5</b> MaAsLin2</a></li>
<li class="chapter" data-level="11.1.6" data-path="differential-abundance.html"><a href="differential-abundance.html#comparison-of-the-methods"><i class="fa fa-check"></i><b>11.1.6</b> Comparison of the methods</a></li>
<li class="chapter" data-level="11.1.7" data-path="differential-abundance.html"><a href="differential-abundance.html#confounding-variables"><i class="fa fa-check"></i><b>11.1.7</b> Confounding variables</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="differential-abundance.html"><a href="differential-abundance.html#tree-based-methods"><i class="fa fa-check"></i><b>11.2</b> Tree-based methods</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="differential-abundance.html"><a href="differential-abundance.html#group-wise-associations-testing-based-on-balances-with-fido"><i class="fa fa-check"></i><b>11.2.1</b> Group-wise associations testing based on balances with fido</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="differential-abundance.html"><a href="differential-abundance.html#session-info-7"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="multi-assay_analyses.html"><a href="multi-assay_analyses.html"><i class="fa fa-check"></i><b>12</b> Multi-assay analyses</a>
<ul>
<li class="chapter" data-level="" data-path="multi-assay_analyses.html"><a href="multi-assay_analyses.html#session-info-8"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="part"><span><b>III Appendix</b></span></li>
<li class="chapter" data-level="13" data-path="resources.html"><a href="resources.html"><i class="fa fa-check"></i><b>13</b> Resources</a>
<ul>
<li class="chapter" data-level="13.1" data-path="resources.html"><a href="resources.html#data-containers-1"><i class="fa fa-check"></i><b>13.1</b> Data containers</a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="resources.html"><a href="resources.html#resources-for-treesummarizedexperiment"><i class="fa fa-check"></i><b>13.1.1</b> Resources for TreeSummarizedExperiment</a></li>
<li class="chapter" data-level="13.1.2" data-path="resources.html"><a href="resources.html#other-relevant-containers"><i class="fa fa-check"></i><b>13.1.2</b> Other relevant containers</a></li>
<li class="chapter" data-level="13.1.3" data-path="resources.html"><a href="resources.html#alternative-containers-for-microbiome-data"><i class="fa fa-check"></i><b>13.1.3</b> Alternative containers for microbiome data</a></li>
<li class="chapter" data-level="13.1.4" data-path="resources.html"><a href="resources.html#resources-for-phyloseq"><i class="fa fa-check"></i><b>13.1.4</b> Resources for phyloseq</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="resources.html"><a href="resources.html#r-programming-resources"><i class="fa fa-check"></i><b>13.2</b> R programming resources</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="resources.html"><a href="resources.html#bioc_intro"><i class="fa fa-check"></i><b>13.2.1</b> Bioconductor Classes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="extras.html"><a href="extras.html"><i class="fa fa-check"></i><b>14</b> Extra material</a>
<ul>
<li class="chapter" data-level="14.1" data-path="extras.html"><a href="extras.html#interactive-3d-plots"><i class="fa fa-check"></i><b>14.1</b> Interactive 3D Plots</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="acknowledgments.html"><a href="acknowledgments.html"><i class="fa fa-check"></i><b>15</b> Acknowledgments</a></li>
<li class="chapter" data-level="" data-path="bibliography.html"><a href="bibliography.html"><i class="fa fa-check"></i>Bibliography</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Orchestrating Microbiome Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="containers" class="section level1" number="2">
<h1><span class="header-section-number">Chapter 2</span> Microbiome Data</h1>
<script>
document.addEventListener("click", function (event) {
    if (event.target.classList.contains("rebook-collapse")) {
        event.target.classList.toggle("active");
        var content = event.target.nextElementSibling;
        if (content.style.display === "block") {
            content.style.display = "none";
        } else {
            content.style.display = "block";
        }
    }
})
</script>
<style>
.rebook-collapse {
  background-color: #eee;
  color: #444;
  cursor: pointer;
  padding: 18px;
  width: 100%;
  border: none;
  text-align: left;
  outline: none;
  font-size: 15px;
}

.rebook-content {
  padding: 0 18px;
  display: none;
  overflow: hidden;
  background-color: #f1f1f1;
}
</style>
<div id="data-science-framework" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Data science framework</h2>
<p>The building blocks of the framework are <strong>data container</strong>
(SummarizedExperiment and its derivatives), <strong>packages</strong> from various
developers using the TreeSE container, open <strong>demonstration data
sets</strong>, in a separate chapter <a href="containers.html#example-data">2.4</a>, and <strong>online
tutorials</strong> including this online book as well as the various package
vignettes and other material.</p>
<p><img src="general/figures/FigureOverviewV2.png" width="672" /></p>
</div>
<div id="data-containers" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> Data containers</h2>
<p><a href="https://bioconductor.org/packages/release/bioc/html/SummarizedExperiment.html"><code>SummarizedExperiment</code></a>
(<code>SE</code>) is a generic and highly optimized container for complex data
structures. It has become a common choice for analysing various types
of biomedical profiling data, such as RNAseq, ChIp-Seq, microarrays,
flow cytometry, proteomics, and single-cell
sequencing.</p>
<p><a href="https://www.bioconductor.org/packages/release/bioc/html/TreeSummarizedExperiment.html"><code>TreeSummarizedExperiment</code></a>
(<code>TreeSE</code>) was developed as an extension to incorporate hierarchical
information (such as phylogenetic trees and sample hierarchies) and
reference sequences.</p>
<p><a href="https://www.bioconductor.org/packages/release/bioc/html/MultiAssayExperiment.html"><code>MultiAssayExperiment</code></a>
(<code>MAE</code>) provides an organized way to bind several different data
structures together in a single object. For example, we can bind
microbiome data (in <code>TreeSE</code> format) with metabolomic profiling data
(in <code>SE</code>) format, with shared sample metadata. This is convenient and
robust for instance in subsetting and other data manipulation
tasks. Microbiome data can be part of multiomics experiments and
analysis strategies and we want to outline the understanding in which
we think the packages explained and used in this book relate to these
experiment layouts using the <code>TreeSummarizedExperiment</code> and classes
beyond.</p>
<p>This section provides an introductions to these data containers. In
microbiome data science, these containers link taxonomic abundance
tables with rich side information on the features and
samples. Taxonomic abundance data can be obtained by 16S rRNA amplicon
or metagenomic sequencing, phylogenetic microarrays, or by other
means. Many microbiome experiments include multiple versions and types
of data generated independently or derived from each other through
transformation or agglomeration. We start by providing recommendations
on how to represent different varieties of multi-table data within the
<code>TreeSummarizedExperiment</code> class.</p>
<p>The options and recommendations are summarized in Table <a href="containers.html#tab:options">2.1</a>.</p>
<div id="assay-data" class="section level3" number="2.2.1">
<h3><span class="header-section-number">2.2.1</span> Assay data</h3>
<p>The original count-based taxonomic abundance tables may have different
transformations, such as logarithmic, Centered Log-Ratio (CLR), or relative
abundance. These are typically stored in <em><strong>assays</strong></em>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="containers.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mia)</span>
<span id="cb1-2"><a href="containers.html#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(GlobalPatterns, <span class="at">package=</span><span class="st">&quot;mia&quot;</span>)</span>
<span id="cb1-3"><a href="containers.html#cb1-3" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> GlobalPatterns</span>
<span id="cb1-4"><a href="containers.html#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">assays</span>(tse)</span></code></pre></div>
<pre><code>## List of length 1
## names(1): counts</code></pre>
<p>The <code>assays</code> slot contains the experimental data as count matrices. Multiple
matrices can be stored the result of <code>assays</code> is actually a list of matrices.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="containers.html#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">assays</span>(tse)</span></code></pre></div>
<pre><code>## List of length 1
## names(1): counts</code></pre>
<p>Individual assays can be accessed via <code>assay</code></p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="containers.html#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">assay</span>(tse, <span class="st">&quot;counts&quot;</span>)[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">7</span>]</span></code></pre></div>
<pre><code>##        CL3 CC1 SV1 M31Fcsw M11Fcsw M31Plmr M11Plmr
## 549322   0   0   0       0       0       0       0
## 522457   0   0   0       0       0       0       0
## 951      0   0   0       0       0       0       1
## 244423   0   0   0       0       0       0       0
## 586076   0   0   0       0       0       0       0</code></pre>
<p>To illustrate the use of multiple assays, the relative abundance data can be
calcualted and stored along the original count data using <code>relAbundanceCounts</code>.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="containers.html#cb7-1" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">relAbundanceCounts</span>(tse)</span>
<span id="cb7-2"><a href="containers.html#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="fu">assays</span>(tse)</span></code></pre></div>
<pre><code>## List of length 2
## names(2): counts relabundance</code></pre>
<p>Now there are two assays available in the <code>tse</code> object, <code>counts</code> and
<code>relabundance</code>.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="containers.html#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">assay</span>(tse, <span class="st">&quot;relabundance&quot;</span>)[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">7</span>]</span></code></pre></div>
<pre><code>##        CL3 CC1 SV1 M31Fcsw M11Fcsw M31Plmr   M11Plmr
## 549322   0   0   0       0       0       0 0.000e+00
## 522457   0   0   0       0       0       0 0.000e+00
## 951      0   0   0       0       0       0 2.305e-06
## 244423   0   0   0       0       0       0 0.000e+00
## 586076   0   0   0       0       0       0 0.000e+00</code></pre>
<p>Here the dimension of the count data remains unchanged. This is in
fact a requirement for any <code>SummarizedExperiment</code> object.</p>
</div>
<div id="coldata" class="section level3" number="2.2.2">
<h3><span class="header-section-number">2.2.2</span> colData</h3>
<p><code>colData</code> contains data on the samples.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="containers.html#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(tse)</span></code></pre></div>
<pre><code>## DataFrame with 26 rows and 7 columns
##         X.SampleID   Primer Final_Barcode Barcode_truncated_plus_T
##           &lt;factor&gt; &lt;factor&gt;      &lt;factor&gt;                 &lt;factor&gt;
## CL3        CL3      ILBC_01        AACGCA                   TGCGTT
## CC1        CC1      ILBC_02        AACTCG                   CGAGTT
## SV1        SV1      ILBC_03        AACTGT                   ACAGTT
## M31Fcsw    M31Fcsw  ILBC_04        AAGAGA                   TCTCTT
## M11Fcsw    M11Fcsw  ILBC_05        AAGCTG                   CAGCTT
## ...            ...      ...           ...                      ...
## TS28         TS28   ILBC_25        ACCAGA                   TCTGGT
## TS29         TS29   ILBC_26        ACCAGC                   GCTGGT
## Even1        Even1  ILBC_27        ACCGCA                   TGCGGT
## Even2        Even2  ILBC_28        ACCTCG                   CGAGGT
## Even3        Even3  ILBC_29        ACCTGT                   ACAGGT
##         Barcode_full_length SampleType
##                    &lt;factor&gt;   &lt;factor&gt;
## CL3             CTAGCGTGCGT      Soil 
## CC1             CATCGACGAGT      Soil 
## SV1             GTACGCACAGT      Soil 
## M31Fcsw         TCGACATCTCT      Feces
## M11Fcsw         CGACTGCAGCT      Feces
## ...                     ...        ...
## TS28            GCATCGTCTGG      Feces
## TS29            CTAGTCGCTGG      Feces
## Even1           TGACTCTGCGG      Mock 
## Even2           TCTGATCGAGG      Mock 
## Even3           AGAGAGACAGG      Mock 
##                                        Description
##                                           &lt;factor&gt;
## CL3     Calhoun South Carolina Pine soil, pH 4.9  
## CC1     Cedar Creek Minnesota, grassland, pH 6.1  
## SV1     Sevilleta new Mexico, desert scrub, pH 8.3
## M31Fcsw M3, Day 1, fecal swab, whole body study   
## M11Fcsw M1, Day 1, fecal swab, whole body study   
## ...                                            ...
## TS28                                       Twin #1
## TS29                                       Twin #2
## Even1                                      Even1  
## Even2                                      Even2  
## Even3                                      Even3</code></pre>
</div>
<div id="rowdata" class="section level3" number="2.2.3">
<h3><span class="header-section-number">2.2.3</span> rowData</h3>
<p><code>rowData</code> contains data on the features of the analyzed samples. Of particular
interest for the microbiome field this is used to store taxonomic information.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="containers.html#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(tse)</span></code></pre></div>
<pre><code>## DataFrame with 19216 rows and 7 columns
##            Kingdom        Phylum        Class        Order        Family
##        &lt;character&gt;   &lt;character&gt;  &lt;character&gt;  &lt;character&gt;   &lt;character&gt;
## 549322     Archaea Crenarchaeota Thermoprotei           NA            NA
## 522457     Archaea Crenarchaeota Thermoprotei           NA            NA
## 951        Archaea Crenarchaeota Thermoprotei Sulfolobales Sulfolobaceae
## 244423     Archaea Crenarchaeota        Sd-NA           NA            NA
## 586076     Archaea Crenarchaeota        Sd-NA           NA            NA
## ...            ...           ...          ...          ...           ...
## 278222    Bacteria           SR1           NA           NA            NA
## 463590    Bacteria           SR1           NA           NA            NA
## 535321    Bacteria           SR1           NA           NA            NA
## 200359    Bacteria           SR1           NA           NA            NA
## 271582    Bacteria           SR1           NA           NA            NA
##              Genus                Species
##        &lt;character&gt;            &lt;character&gt;
## 549322          NA                     NA
## 522457          NA                     NA
## 951     Sulfolobus Sulfolobusacidocalda..
## 244423          NA                     NA
## 586076          NA                     NA
## ...            ...                    ...
## 278222          NA                     NA
## 463590          NA                     NA
## 535321          NA                     NA
## 200359          NA                     NA
## 271582          NA                     NA</code></pre>
</div>
<div id="rowtree" class="section level3" number="2.2.4">
<h3><span class="header-section-number">2.2.4</span> rowTree</h3>
<p>Phylogenetic trees also play an important role for the microbiome field. The
<code>TreeSummarizedExperiment</code> class is able to keep track of feature and node
relations via two functions, <code>rowTree</code> and <code>rowLinks</code>.</p>
<p>A tree can be accessed via <code>rowTree</code> as <code>phylo</code> object.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="containers.html#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rowTree</span>(tse)</span></code></pre></div>
<pre><code>## 
## Phylogenetic tree with 19216 tips and 19215 internal nodes.
## 
## Tip labels:
##   549322, 522457, 951, 244423, 586076, 246140, ...
## Node labels:
##   , 0.858.4, 1.000.154, 0.764.3, 0.995.2, 1.000.2, ...
## 
## Rooted; includes branch lengths.</code></pre>
<p>The links to the individual features are available through <code>rowLinks</code>.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="containers.html#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rowLinks</span>(tse)</span></code></pre></div>
<pre><code>## LinkDataFrame with 19216 rows and 5 columns
##           nodeLab   nodeNum nodeLab_alias    isLeaf   whichTree
##       &lt;character&gt; &lt;integer&gt;   &lt;character&gt; &lt;logical&gt; &lt;character&gt;
## 1          549322         1       alias_1      TRUE       phylo
## 2          522457         2       alias_2      TRUE       phylo
## 3             951         3       alias_3      TRUE       phylo
## 4          244423         4       alias_4      TRUE       phylo
## 5          586076         5       alias_5      TRUE       phylo
## ...           ...       ...           ...       ...         ...
## 19212      278222     19212   alias_19212      TRUE       phylo
## 19213      463590     19213   alias_19213      TRUE       phylo
## 19214      535321     19214   alias_19214      TRUE       phylo
## 19215      200359     19215   alias_19215      TRUE       phylo
## 19216      271582     19216   alias_19216      TRUE       phylo</code></pre>
<p>Please note that there can be a 1:1 relationship between tree nodes and
features, but this is not a must have. This means there can be features, which
are not linked to nodes, and nodes, which are not linked to features. To change
the links in an existing object, the <code>changeTree</code> function is available.</p>
</div>
<div id="alternative-experiments" class="section level3" number="2.2.5">
<h3><span class="header-section-number">2.2.5</span> Alternative experiments</h3>
<p><em><strong>Alternative experiments</strong></em> differ from transformations as they can
contain complementary data, which is no longer tied to the same
dimensions as the assay data. However, the number of samples (columns)
must be the same.</p>
<p>This can come into play for instance when one has taxonomic abundance
profiles quantified with different measurement technologies, such as
phylogenetic microarrays, amplicon sequencing, or metagenomic
sequencing. Such alternative experiments that concern the same samples
can be stored as</p>
<ol style="list-style-type: decimal">
<li>Separate <em>assays</em> assuming that the taxonomic information can be mapped
between feature directly 1:1; or</li>
<li>data in the <em>altExp</em> slot of the <code>TreeSummarizedExperiment</code>, if the feature
dimensions differ. Each element of the <em>altExp</em> slot is a <code>SummarizedExperiment</code>
or an object from a derived class with independent feature data.</li>
</ol>
<p>As an example, we show how to store taxonomic abundance tables
agglomerated at different taxonomic levels. However, the data could as
well originate from entirely different measurement sources as long as
the samples are matched.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="containers.html#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Agglomerate the data to Phylym level</span></span>
<span id="cb19-2"><a href="containers.html#cb19-2" aria-hidden="true" tabindex="-1"></a>tse_phylum <span class="ot">&lt;-</span> <span class="fu">agglomerateByRank</span>(tse, <span class="st">&quot;Phylum&quot;</span>)</span>
<span id="cb19-3"><a href="containers.html#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="co"># both have the same number of columns (samples)</span></span>
<span id="cb19-4"><a href="containers.html#cb19-4" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(tse)</span></code></pre></div>
<pre><code>## [1] 19216    26</code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="containers.html#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(tse_phylum)</span></code></pre></div>
<pre><code>## [1] 67 26</code></pre>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="containers.html#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Add the new table as an alternative experiment</span></span>
<span id="cb23-2"><a href="containers.html#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="fu">altExp</span>(tse, <span class="st">&quot;Phylum&quot;</span>) <span class="ot">&lt;-</span> tse_phylum</span>
<span id="cb23-3"><a href="containers.html#cb23-3" aria-hidden="true" tabindex="-1"></a><span class="fu">altExpNames</span>(tse)</span></code></pre></div>
<pre><code>## [1] &quot;Phylum&quot;</code></pre>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="containers.html#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Pick a sample subset: this acts on both altExp and assay data</span></span>
<span id="cb25-2"><a href="containers.html#cb25-2" aria-hidden="true" tabindex="-1"></a>tse[,<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>]</span></code></pre></div>
<pre><code>## class: TreeSummarizedExperiment 
## dim: 19216 10 
## metadata(0):
## assays(2): counts relabundance
## rownames(19216): 549322 522457 ... 200359 271582
## rowData names(7): Kingdom Phylum ... Genus Species
## colnames(10): CL3 CC1 ... M31Tong M11Tong
## colData names(7): X.SampleID Primer ... SampleType Description
## reducedDimNames(0):
## mainExpName: NULL
## altExpNames(1): Phylum
## rowLinks: a LinkDataFrame (19216 rows)
## rowTree: 1 phylo tree(s) (19216 leaves)
## colLinks: NULL
## colTree: NULL</code></pre>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="containers.html#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(<span class="fu">altExp</span>(tse[,<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>],<span class="st">&quot;Phylum&quot;</span>))</span></code></pre></div>
<pre><code>## [1] 67 10</code></pre>
<p>For more details of altExp have a look at the <a href="https://bioconductor.org/packages/release/bioc/vignettes/SingleCellExperiment/inst/doc/intro.html">Intro vignette</a> of the
<code>SingleCellExperiment</code> package <span class="citation">(<a href="#ref-R-SingleCellExperiment" role="doc-biblioref">Lun and Risso 2020</a>)</span>.</p>
</div>
<div id="multiassayexperiments" class="section level3" number="2.2.6">
<h3><span class="header-section-number">2.2.6</span> MultiAssayExperiments</h3>
<p><em><strong>Multiple experiments</strong></em> relate to complementary measurement types,
such as transcriptomic or metabolomic profiling of the microbiome or
the host. Multiple experiments can be represented using the same
options as alternative experiments, or by using the
<code>MultiAssayExperiment</code> class <span class="citation">(<a href="#ref-R-MultiAssayExperiment" role="doc-biblioref">Ramos and Waldron 2020</a>)</span>. Depending on how the
datasets relate to each other the data can be stored as:</p>
<ol style="list-style-type: decimal">
<li>Separate <em>altExp</em> if the samples can be matched directly 1:1; or</li>
<li>As <code>MultiAssayExperiment</code> objects, in which the connections between
samples are defined through a <code>sampleMap</code>. Each element on the
<code>experimentsList</code> of an <code>MultiAssayExperiment</code> is <code>matrix</code> or
<code>matrix</code>-like object including <code>SummarizedExperiment</code> objects, and the
number of samples can differ between the elements.</li>
</ol>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="containers.html#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co">#TODO: Find the right dataset to explain a non 1:1 sample relationship</span></span></code></pre></div>
<p>For information have a look at the <a href="https://bioconductor.org/packages/release/bioc/vignettes/MultiAssayExperiment/inst/doc/MultiAssayExperiment.html">intro vignette</a> of the <code>MultiAssayExperiment</code> package.</p>
<table>
<caption><span id="tab:options">Table 2.1: </span> <strong>Recommended options for storing multiple data tables in microbiome studies</strong> The <em>assays</em> are best suited for data transformations (one-to-one match between samples and columns across the assays). The <em>alternative experiments</em> are particularly suitable for alternative versions of the data that are of same type but may have a different number of features (e.g. taxonomic groups); this is for instance the case with taxonomic abundance tables agglomerated at different levels (e.g. genus vs. phyla) or alternative profiling technologies (e.g. amplicon sequencing vs. shallow shotgun metagenomics). For alternative experiments one-to-one match between samples (cols) is required but the alternative experiment tables can have different numbers of features (rows). Finally, elements of the <em>MultiAssayExperiment</em> provide the most flexible way to incorporate multi-omic data tables with flexible numbers of samples and features. We recommend these conventions as the basis for methods development and application in microbiome studies.</caption>
<thead>
<tr class="header">
<th align="right">Option</th>
<th>Rows (features)</th>
<th align="right">Cols (samples)</th>
<th align="right">Recommended</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">assays</td>
<td>match</td>
<td align="right">match</td>
<td align="right">Data transformations</td>
</tr>
<tr class="even">
<td align="right">altExp</td>
<td>free</td>
<td align="right">match</td>
<td align="right">Alternative experiments</td>
</tr>
<tr class="odd">
<td align="right">MultiAssay</td>
<td>free</td>
<td align="right">free (mapping)</td>
<td align="right">Multi-omic experiments</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="loading-experimental-microbiome-data" class="section level2" number="2.3">
<h2><span class="header-section-number">2.3</span> Loading experimental microbiome data</h2>
<div id="import-from-external-files" class="section level3" number="2.3.1">
<h3><span class="header-section-number">2.3.1</span> Import from external files</h3>
<p>Microbiome (taxonomic) profiling data is commonly distributed in
various file formats. You can import such external data files as a
(Tree)SummarizedExperiment object but the details depend on the file
format. Here, we provide examples for common formats.</p>
<p><strong>CSV data tables</strong> can be imported with the standard R functions,
then converted to the desired format. For detailed examples, you can
check the <a href="https://bioconductor.org/help/course-materials/2019/BSS2019/04_Practical_CoreApproachesInBioconductor.html">Bioconductor course
material</a>
by Martin Morgan. The following example reads abundance tables,
taxonomic mapping tables, and sample metadata, assuming that the
input data files are properly prepared with appropriate row and
column names.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="containers.html#cb30-1" aria-hidden="true" tabindex="-1"></a>counts  <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(count_file)   <span class="co"># Abundance table (e.g. ASV data; to assay data)</span></span>
<span id="cb30-2"><a href="containers.html#cb30-2" aria-hidden="true" tabindex="-1"></a>tax     <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(tax_file)     <span class="co"># Taxonomy table (to rowData)</span></span>
<span id="cb30-3"><a href="containers.html#cb30-3" aria-hidden="true" tabindex="-1"></a>samples <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(sample_file)  <span class="co"># Sample data (to colData)</span></span>
<span id="cb30-4"><a href="containers.html#cb30-4" aria-hidden="true" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">SummarizedExperiment</span>(<span class="at">assays =</span> <span class="fu">list</span>(<span class="at">counts =</span> counts),</span>
<span id="cb30-5"><a href="containers.html#cb30-5" aria-hidden="true" tabindex="-1"></a>                           <span class="at">colData =</span> samples,</span>
<span id="cb30-6"><a href="containers.html#cb30-6" aria-hidden="true" tabindex="-1"></a>                           <span class="at">rowData =</span> tax)</span></code></pre></div>
<p>Specific import functions are provided for:</p>
<ul>
<li>Biom files (see <code>help(mia::loadFromBiom)</code>)</li>
<li>QIIME2 files (see <code>help(mia::loadFromQIIME2)</code>)</li>
<li>Mothur files (see <code>help(mia::loadFromMothur)</code>)</li>
</ul>
<div id="biom-example" class="section level4" number="2.3.1.1">
<h4><span class="header-section-number">2.3.1.1</span> Biom example</h4>
<p>This example shows how Biom files are imported into a
<code>TreeSummarizedExperiment</code> object.</p>
<p>The data is from following publication:
Tengeler AC <em>et al.</em> (2020) <a href="https://doi.org/10.1186/s40168-020-00816-x"><strong>Gut microbiota from persons with
attention-deficit/hyperactivity disorder affects the brain in
mice</strong></a>.</p>
<p>The data set consists of 3 files:</p>
<ul>
<li>biom file: abundance table and taxonomy information</li>
<li>csv file: sample metadata</li>
<li>tree file: phylogenetic tree</li>
</ul>
<p>Store the data in your desired local directory (for instance, <em>data/</em> under the
working directory), and define source file paths</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="containers.html#cb31-1" aria-hidden="true" tabindex="-1"></a>biom_file_path <span class="ot">&lt;-</span> <span class="st">&quot;data/Aggregated_humanization2.biom&quot;</span></span>
<span id="cb31-2"><a href="containers.html#cb31-2" aria-hidden="true" tabindex="-1"></a>sample_meta_file_path <span class="ot">&lt;-</span> <span class="st">&quot;data/Mapping_file_ADHD_aggregated.csv&quot;</span></span>
<span id="cb31-3"><a href="containers.html#cb31-3" aria-hidden="true" tabindex="-1"></a>tree_file_path <span class="ot">&lt;-</span> <span class="st">&quot;data/Data_humanization_phylo_aggregation.tre&quot;</span></span></code></pre></div>
<p>Now we can load the biom data into a SummarizedExperiment (SE) object.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="containers.html#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mia)</span>
<span id="cb32-2"><a href="containers.html#cb32-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-3"><a href="containers.html#cb32-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Imports the data</span></span>
<span id="cb32-4"><a href="containers.html#cb32-4" aria-hidden="true" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">loadFromBiom</span>(biom_file_path)</span>
<span id="cb32-5"><a href="containers.html#cb32-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-6"><a href="containers.html#cb32-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Check</span></span>
<span id="cb32-7"><a href="containers.html#cb32-7" aria-hidden="true" tabindex="-1"></a>se</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 151 27 
## metadata(0):
## assays(1): counts
## rownames(151): 1726470 1726471 ... 17264756 17264757
## rowData names(6): taxonomy1 taxonomy2 ... taxonomy5 taxonomy6
## colnames(27): A110 A111 ... A38 A39
## colData names(0):</code></pre>
<p>The <code>assays</code> slot includes a list of abundance tables. The imported
abundance table is named as “counts.” Let us inspect only the first
cols and rows.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="containers.html#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="fu">assays</span>(se)<span class="sc">$</span>counts[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span></code></pre></div>
<pre><code>##           A110  A111  A12
## 1726470  17722 11630    0
## 1726471  12052     0 2679
## 17264731     0   970    0</code></pre>
<p>The <code>rowdata</code> includes taxonomic information from the biom file. The <code>head()</code> command
shows just the beginning of the data table for an overview.</p>
<p><code>knitr::kable()</code> is for printing the information more nicely.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="containers.html#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">rowData</span>(se))</span></code></pre></div>
<pre><code>## DataFrame with 6 rows and 6 columns
##             taxonomy1          taxonomy2           taxonomy3
##           &lt;character&gt;        &lt;character&gt;         &lt;character&gt;
## 1726470  &quot;k__Bacteria   p__Bacteroidetes      c__Bacteroidia
## 1726471  &quot;k__Bacteria   p__Bacteroidetes      c__Bacteroidia
## 17264731 &quot;k__Bacteria   p__Bacteroidetes      c__Bacteroidia
## 17264726 &quot;k__Bacteria   p__Bacteroidetes      c__Bacteroidia
## 1726472  &quot;k__Bacteria p__Verrucomicrobia c__Verrucomicrobiae
## 17264724 &quot;k__Bacteria   p__Bacteroidetes      c__Bacteroidia
##                      taxonomy4              taxonomy5           taxonomy6
##                    &lt;character&gt;            &lt;character&gt;         &lt;character&gt;
## 1726470       o__Bacteroidales      f__Bacteroidaceae     g__Bacteroides&quot;
## 1726471       o__Bacteroidales      f__Bacteroidaceae     g__Bacteroides&quot;
## 17264731      o__Bacteroidales  f__Porphyromonadaceae g__Parabacteroides&quot;
## 17264726      o__Bacteroidales      f__Bacteroidaceae     g__Bacteroides&quot;
## 1726472  o__Verrucomicrobiales f__Verrucomicrobiaceae     g__Akkermansia&quot;
## 17264724      o__Bacteroidales      f__Bacteroidaceae     g__Bacteroides&quot;</code></pre>
<p>These taxonomic rank names (column names) are not real rank
names. Let’s replace them with real rank names.</p>
<p>In addition to that, the taxa names include, e.g., ’“k__’ before the name, so let’s
make them cleaner by removing them.</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="containers.html#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(<span class="fu">rowData</span>(se)) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Kingdom&quot;</span>, <span class="st">&quot;Phylum&quot;</span>, <span class="st">&quot;Class&quot;</span>, <span class="st">&quot;Order&quot;</span>, </span>
<span id="cb38-2"><a href="containers.html#cb38-2" aria-hidden="true" tabindex="-1"></a>                        <span class="st">&quot;Family&quot;</span>, <span class="st">&quot;Genus&quot;</span>)</span>
<span id="cb38-3"><a href="containers.html#cb38-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-4"><a href="containers.html#cb38-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Goes through the whole DataFrame. Removes &#39;.*[kpcofg]__&#39; from strings, where [kpcofg] </span></span>
<span id="cb38-5"><a href="containers.html#cb38-5" aria-hidden="true" tabindex="-1"></a><span class="co"># is any character from listed ones, and .* any character.</span></span>
<span id="cb38-6"><a href="containers.html#cb38-6" aria-hidden="true" tabindex="-1"></a>rowdata_modified <span class="ot">&lt;-</span> BiocParallel<span class="sc">::</span><span class="fu">bplapply</span>(<span class="fu">rowData</span>(se), </span>
<span id="cb38-7"><a href="containers.html#cb38-7" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">FUN =</span> stringr<span class="sc">::</span>str_remove, </span>
<span id="cb38-8"><a href="containers.html#cb38-8" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">pattern =</span> <span class="st">&#39;.*[kpcofg]__&#39;</span>)</span>
<span id="cb38-9"><a href="containers.html#cb38-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-10"><a href="containers.html#cb38-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Genus level has additional &#39;\&quot;&#39;, so let&#39;s delete that also</span></span>
<span id="cb38-11"><a href="containers.html#cb38-11" aria-hidden="true" tabindex="-1"></a>rowdata_modified <span class="ot">&lt;-</span> BiocParallel<span class="sc">::</span><span class="fu">bplapply</span>(rowdata_modified, </span>
<span id="cb38-12"><a href="containers.html#cb38-12" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">FUN =</span> stringr<span class="sc">::</span>str_remove, </span>
<span id="cb38-13"><a href="containers.html#cb38-13" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">pattern =</span> <span class="st">&#39;</span><span class="sc">\&quot;</span><span class="st">&#39;</span>)</span>
<span id="cb38-14"><a href="containers.html#cb38-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-15"><a href="containers.html#cb38-15" aria-hidden="true" tabindex="-1"></a><span class="co"># rowdata_modified is a list, so it is converted back to DataFrame format. </span></span>
<span id="cb38-16"><a href="containers.html#cb38-16" aria-hidden="true" tabindex="-1"></a>rowdata_modified <span class="ot">&lt;-</span> <span class="fu">DataFrame</span>(rowdata_modified)</span>
<span id="cb38-17"><a href="containers.html#cb38-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-18"><a href="containers.html#cb38-18" aria-hidden="true" tabindex="-1"></a><span class="co"># And then assigned back to the SE object</span></span>
<span id="cb38-19"><a href="containers.html#cb38-19" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(se) <span class="ot">&lt;-</span> rowdata_modified</span>
<span id="cb38-20"><a href="containers.html#cb38-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-21"><a href="containers.html#cb38-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Now we have a nicer table</span></span>
<span id="cb38-22"><a href="containers.html#cb38-22" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">rowData</span>(se))</span></code></pre></div>
<pre><code>## DataFrame with 6 rows and 6 columns
##              Kingdom          Phylum            Class              Order
##          &lt;character&gt;     &lt;character&gt;      &lt;character&gt;        &lt;character&gt;
## 1726470     Bacteria   Bacteroidetes      Bacteroidia      Bacteroidales
## 1726471     Bacteria   Bacteroidetes      Bacteroidia      Bacteroidales
## 17264731    Bacteria   Bacteroidetes      Bacteroidia      Bacteroidales
## 17264726    Bacteria   Bacteroidetes      Bacteroidia      Bacteroidales
## 1726472     Bacteria Verrucomicrobia Verrucomicrobiae Verrucomicrobiales
## 17264724    Bacteria   Bacteroidetes      Bacteroidia      Bacteroidales
##                       Family           Genus
##                  &lt;character&gt;     &lt;character&gt;
## 1726470       Bacteroidaceae     Bacteroides
## 1726471       Bacteroidaceae     Bacteroides
## 17264731  Porphyromonadaceae Parabacteroides
## 17264726      Bacteroidaceae     Bacteroides
## 1726472  Verrucomicrobiaceae     Akkermansia
## 17264724      Bacteroidaceae     Bacteroides</code></pre>
<p>We notice that the imported biom file did not contain the sample meta data
yet, so it includes an empty data frame.</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="containers.html#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">colData</span>(se))</span></code></pre></div>
<pre><code>## DataFrame with 6 rows and 0 columns</code></pre>
<p>Let us add a sample metadata file.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="containers.html#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="co"># We use this to check what type of data it is</span></span>
<span id="cb42-2"><a href="containers.html#cb42-2" aria-hidden="true" tabindex="-1"></a><span class="co"># read.table(sample_meta_file_path)</span></span>
<span id="cb42-3"><a href="containers.html#cb42-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-4"><a href="containers.html#cb42-4" aria-hidden="true" tabindex="-1"></a><span class="co"># It seems like a comma separated file and it does not include headers</span></span>
<span id="cb42-5"><a href="containers.html#cb42-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Let us read it and then convert from data.frame to DataFrame</span></span>
<span id="cb42-6"><a href="containers.html#cb42-6" aria-hidden="true" tabindex="-1"></a><span class="co"># (required for our purposes)</span></span>
<span id="cb42-7"><a href="containers.html#cb42-7" aria-hidden="true" tabindex="-1"></a>sample_meta <span class="ot">&lt;-</span> <span class="fu">DataFrame</span>(<span class="fu">read.table</span>(sample_meta_file_path, <span class="at">sep =</span> <span class="st">&quot;,&quot;</span>, <span class="at">header =</span> <span class="cn">FALSE</span>))</span>
<span id="cb42-8"><a href="containers.html#cb42-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-9"><a href="containers.html#cb42-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Add sample names to rownames</span></span>
<span id="cb42-10"><a href="containers.html#cb42-10" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sample_meta) <span class="ot">&lt;-</span> sample_meta[,<span class="dv">1</span>]</span>
<span id="cb42-11"><a href="containers.html#cb42-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-12"><a href="containers.html#cb42-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Delete column that included sample names</span></span>
<span id="cb42-13"><a href="containers.html#cb42-13" aria-hidden="true" tabindex="-1"></a>sample_meta[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb42-14"><a href="containers.html#cb42-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-15"><a href="containers.html#cb42-15" aria-hidden="true" tabindex="-1"></a><span class="co"># We can add headers</span></span>
<span id="cb42-16"><a href="containers.html#cb42-16" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(sample_meta) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;patient_status&quot;</span>, <span class="st">&quot;cohort&quot;</span>, <span class="st">&quot;patient_status_vs_cohort&quot;</span>, <span class="st">&quot;sample_name&quot;</span>)</span>
<span id="cb42-17"><a href="containers.html#cb42-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-18"><a href="containers.html#cb42-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Then it can be added to colData</span></span>
<span id="cb42-19"><a href="containers.html#cb42-19" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(se) <span class="ot">&lt;-</span> sample_meta</span></code></pre></div>
<p>Now <code>colData</code> includes the sample metadata.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="containers.html#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">colData</span>(se))</span></code></pre></div>
<pre><code>## DataFrame with 6 rows and 4 columns
##      patient_status      cohort patient_status_vs_cohort sample_name
##         &lt;character&gt; &lt;character&gt;              &lt;character&gt; &lt;character&gt;
## A110           ADHD    Cohort_1            ADHD_Cohort_1        A110
## A12            ADHD    Cohort_1            ADHD_Cohort_1         A12
## A15            ADHD    Cohort_1            ADHD_Cohort_1         A15
## A19            ADHD    Cohort_1            ADHD_Cohort_1         A19
## A21            ADHD    Cohort_2            ADHD_Cohort_2         A21
## A23            ADHD    Cohort_2            ADHD_Cohort_2         A23</code></pre>
<p>Now, let’s add a phylogenetic tree.</p>
<p>The current data object, se, is a SummarizedExperiment object. This
does not include a slot for adding a phylogenetic tree. In order to do
this, we can convert the SE object to an extended TreeSummarizedExperiment
object which includes also a <code>rowTree</code> slot.</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="containers.html#cb45-1" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">as</span>(se, <span class="st">&quot;TreeSummarizedExperiment&quot;</span>)</span>
<span id="cb45-2"><a href="containers.html#cb45-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-3"><a href="containers.html#cb45-3" aria-hidden="true" tabindex="-1"></a><span class="co"># tse includes same data as se</span></span>
<span id="cb45-4"><a href="containers.html#cb45-4" aria-hidden="true" tabindex="-1"></a>tse</span></code></pre></div>
<pre><code>## class: TreeSummarizedExperiment 
## dim: 151 27 
## metadata(0):
## assays(1): counts
## rownames(151): 1726470 1726471 ... 17264756 17264757
## rowData names(6): Kingdom Phylum ... Family Genus
## colnames(27): A110 A12 ... A35 A38
## colData names(4): patient_status cohort patient_status_vs_cohort
##   sample_name
## reducedDimNames(0):
## mainExpName: NULL
## altExpNames(0):
## rowLinks: NULL
## rowTree: NULL
## colLinks: NULL
## colTree: NULL</code></pre>
<p>Next, let us read the tree data file and add it to the R data object (tse).</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="containers.html#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Reads the tree file</span></span>
<span id="cb47-2"><a href="containers.html#cb47-2" aria-hidden="true" tabindex="-1"></a>tree <span class="ot">&lt;-</span> ape<span class="sc">::</span><span class="fu">read.tree</span>(tree_file_path)</span>
<span id="cb47-3"><a href="containers.html#cb47-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-4"><a href="containers.html#cb47-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Add tree to rowTree</span></span>
<span id="cb47-5"><a href="containers.html#cb47-5" aria-hidden="true" tabindex="-1"></a><span class="fu">rowTree</span>(tse) <span class="ot">&lt;-</span> tree</span>
<span id="cb47-6"><a href="containers.html#cb47-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-7"><a href="containers.html#cb47-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Check</span></span>
<span id="cb47-8"><a href="containers.html#cb47-8" aria-hidden="true" tabindex="-1"></a>tse</span></code></pre></div>
<pre><code>## class: TreeSummarizedExperiment 
## dim: 151 27 
## metadata(0):
## assays(1): counts
## rownames(151): 1726470 1726471 ... 17264756 17264757
## rowData names(6): Kingdom Phylum ... Family Genus
## colnames(27): A110 A12 ... A35 A38
## colData names(4): patient_status cohort patient_status_vs_cohort
##   sample_name
## reducedDimNames(0):
## mainExpName: NULL
## altExpNames(0):
## rowLinks: a LinkDataFrame (151 rows)
## rowTree: 1 phylo tree(s) (151 leaves)
## colLinks: NULL
## colTree: NULL</code></pre>
<p>Now <code>rowTree</code> includes a phylogenetic tree:</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="containers.html#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">rowTree</span>(tse))</span></code></pre></div>
</div>
</div>
<div id="conversions-between-data-formats-in-r" class="section level3" number="2.3.2">
<h3><span class="header-section-number">2.3.2</span> Conversions between data formats in R</h3>
<p>If the data has already been imported in R in another format, it
can be readily converted into <code>TreeSummarizedExperiment</code>, as shown in our next
example. Note that similar conversion functions to
<code>TreeSummarizedExperiment</code> are available for multiple data formats via
the <code>mia</code> package (see makeTreeSummarizedExperimentFrom* for phyloseq,
Biom, and DADA2).</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="containers.html#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mia)</span>
<span id="cb50-2"><a href="containers.html#cb50-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-3"><a href="containers.html#cb50-3" aria-hidden="true" tabindex="-1"></a><span class="co"># phyloseq example data</span></span>
<span id="cb50-4"><a href="containers.html#cb50-4" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(GlobalPatterns, <span class="at">package=</span><span class="st">&quot;phyloseq&quot;</span>) </span>
<span id="cb50-5"><a href="containers.html#cb50-5" aria-hidden="true" tabindex="-1"></a>GlobalPatterns_phyloseq <span class="ot">&lt;-</span> GlobalPatterns</span>
<span id="cb50-6"><a href="containers.html#cb50-6" aria-hidden="true" tabindex="-1"></a>GlobalPatterns_phyloseq</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 19216 taxa and 26 samples ]
## sample_data() Sample Data:       [ 26 samples by 7 sample variables ]
## tax_table()   Taxonomy Table:    [ 19216 taxa by 7 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 19216 tips and 19215 internal nodes ]</code></pre>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="containers.html#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="co"># convert phyloseq to TSE</span></span>
<span id="cb52-2"><a href="containers.html#cb52-2" aria-hidden="true" tabindex="-1"></a>GlobalPatterns_TSE <span class="ot">&lt;-</span> <span class="fu">makeTreeSummarizedExperimentFromPhyloseq</span>(GlobalPatterns_phyloseq) </span>
<span id="cb52-3"><a href="containers.html#cb52-3" aria-hidden="true" tabindex="-1"></a>GlobalPatterns_TSE</span></code></pre></div>
<pre><code>## class: TreeSummarizedExperiment 
## dim: 19216 26 
## metadata(0):
## assays(1): counts
## rownames(19216): 549322 522457 ... 200359 271582
## rowData names(7): Kingdom Phylum ... Genus Species
## colnames(26): CL3 CC1 ... Even2 Even3
## colData names(7): X.SampleID Primer ... SampleType Description
## reducedDimNames(0):
## mainExpName: NULL
## altExpNames(0):
## rowLinks: a LinkDataFrame (19216 rows)
## rowTree: 1 phylo tree(s) (19216 leaves)
## colLinks: NULL
## colTree: NULL</code></pre>
<p>We can also convert <code>TreeSummarizedExperiment</code> objects into <code>phyloseq</code>
with respect to the shared components that are supported by both
formats (i.e. taxonomic abundance table, sample metadata, taxonomic
table, phylogenetic tree, sequence information). This is useful for
instance when additional methods are available for <code>phyloseq</code>.</p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="containers.html#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="co"># convert TSE to phyloseq</span></span>
<span id="cb54-2"><a href="containers.html#cb54-2" aria-hidden="true" tabindex="-1"></a>GlobalPatterns_phyloseq2 <span class="ot">&lt;-</span> <span class="fu">makePhyloseqFromTreeSummarizedExperiment</span>(GlobalPatterns_TSE) </span>
<span id="cb54-3"><a href="containers.html#cb54-3" aria-hidden="true" tabindex="-1"></a>GlobalPatterns_phyloseq2</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 19216 taxa and 26 samples ]
## sample_data() Sample Data:       [ 26 samples by 7 sample variables ]
## tax_table()   Taxonomy Table:    [ 19216 taxa by 7 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 19216 tips and 19215 internal nodes ]</code></pre>
</div>
</div>
<div id="example-data" class="section level2" number="2.4">
<h2><span class="header-section-number">2.4</span> Demonstration data</h2>
<p>Open demonstration data for testing and benchmarking purposes is
available from multiple locations. This chapter introduces some
options. The other chapters of this book provide ample examples about
the use of the data.</p>
<div id="package-data" class="section level3" number="2.4.1">
<h3><span class="header-section-number">2.4.1</span> Package data</h3>
<p>The <code>mia</code> R package contains example data sets that are direct
conversions from the alternative <code>phyloseq</code> container to the
<code>TreeSummarizedExperiment</code> container.</p>
<p>List the <a href="https://microbiome.github.io/mia/reference/index.html">available
datasets</a> in
the <code>mia</code> package:</p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="containers.html#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mia)</span>
<span id="cb56-2"><a href="containers.html#cb56-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="at">package=</span><span class="st">&quot;mia&quot;</span>)</span></code></pre></div>
<p>Load the <code>GlobalPatterns</code> data from the <code>mia</code> package:</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="containers.html#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;GlobalPatterns&quot;</span>, <span class="at">package=</span><span class="st">&quot;mia&quot;</span>)</span>
<span id="cb57-2"><a href="containers.html#cb57-2" aria-hidden="true" tabindex="-1"></a>GlobalPatterns</span></code></pre></div>
<pre><code>## class: TreeSummarizedExperiment 
## dim: 19216 26 
## metadata(0):
## assays(1): counts
## rownames(19216): 549322 522457 ... 200359 271582
## rowData names(7): Kingdom Phylum ... Genus Species
## colnames(26): CL3 CC1 ... Even2 Even3
## colData names(7): X.SampleID Primer ... SampleType Description
## reducedDimNames(0):
## mainExpName: NULL
## altExpNames(0):
## rowLinks: a LinkDataFrame (19216 rows)
## rowTree: 1 phylo tree(s) (19216 leaves)
## colLinks: NULL
## colTree: NULL</code></pre>
<p>Check the documentation for this data set:</p>
<pre><code>## Help on topic &#39;GlobalPatterns&#39; was found in the following packages:
## 
##   Package               Library
##   phyloseq              /__w/_temp/Library
##   mia                   /__w/_temp/Library
## 
## 
## Using the first match ...</code></pre>
</div>
<div id="experimenthub-data" class="section level3" number="2.4.2">
<h3><span class="header-section-number">2.4.2</span> ExperimentHub data</h3>
<p><a href="https://bioconductor.org/packages/release/bioc/vignettes/ExperimentHub/inst/doc/ExperimentHub.html">ExperimentHub</a>
provides a variety of data resources, including the
<a href="https://bioconductor.org/packages/devel/data/experiment/html/microbiomeDataSets.html">microbiomeDataSets</a>
package.</p>
<p>A table of the available data sets is available through the <code>availableDataSets</code>
function.</p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="containers.html#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(microbiomeDataSets)</span>
<span id="cb60-2"><a href="containers.html#cb60-2" aria-hidden="true" tabindex="-1"></a><span class="fu">availableDataSets</span>()</span></code></pre></div>
<pre><code>##             Dataset
## 1  GrieneisenTSData
## 2    HintikkaXOData
## 3       LahtiMLData
## 4        LahtiMData
## 5       LahtiWAData
## 6      OKeefeDSData
## 7 SilvermanAGutData
## 8        SongQAData
## 9   SprockettTHData</code></pre>
<p>All data are downloaded from ExperimentHub and cached for local
re-use. Check the <a href="https://microbiome.github.io/microbiomeDataSets/reference/index.html">man pages of each
function</a>
for a detailed documentation of the data contents and references. Let
us retrieve a <em><a href="https://bioconductor.org/packages/3.14/MultiAssayExperiment">MultiAssayExperiment</a></em> data set:</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="containers.html#cb62-1" aria-hidden="true" tabindex="-1"></a>mae <span class="ot">&lt;-</span> <span class="fu">HintikkaXOData</span>()</span></code></pre></div>
<p>Data is available in <em><a href="https://bioconductor.org/packages/3.14/SummarizedExperiment">SummarizedExperiment</a></em>, <code>r Biocpkg("TreeSummarizedExperiment")</code>, and <code>r Biocpkg("MultiAssayExperiment")</code> data containers; see the separate
page on <a href="https://microbiome.github.io/OMA/multitable.html">alternative
containers</a> for more
details.</p>
</div>
<div id="other-data-sources" class="section level3" number="2.4.3">
<h3><span class="header-section-number">2.4.3</span> Other data sources</h3>
<p>The
<a href="https://waldronlab.io/curatedMetagenomicData">curatedMetagenomicData</a>
is an independent source that provides various example data sets as
<code>(Tree)SummarizedExperiment</code> objects. This resource provides curated
human microbiome data including gene families, marker abundance,
marker presence, pathway abundance, pathway coverage, and relative
abundance for samples from different body sites. See the package
homepage for more details on data availability and access.</p>
<p>As one example, let us retrieve the Vatanen (2016) <span class="citation">(<a href="#ref-Vatanen2016" role="doc-biblioref">Vatanen et al. 2016</a>)</span> data
set. This is a larger collection with a bit longer download time.</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="containers.html#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(curatedMetagenomicData)</span>
<span id="cb63-2"><a href="containers.html#cb63-2" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">curatedMetagenomicData</span>(<span class="st">&quot;Vatanen*&quot;</span>, <span class="at">dryrun =</span> <span class="cn">FALSE</span>, <span class="at">counts =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
</div>
<div id="session-info" class="section level2 unnumbered">
<h2>Session Info</h2>
<button class="rebook-collapse">
View session info
</button>
<div class="rebook-content">
<pre><code>R version 4.1.2 (2021-11-01)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.3 LTS

Matrix products: default
BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] microbiomeDataSets_1.1.5       phyloseq_1.38.0               
 [3] mia_1.3.8                      MultiAssayExperiment_1.20.0   
 [5] TreeSummarizedExperiment_2.1.4 Biostrings_2.62.0             
 [7] XVector_0.34.0                 SingleCellExperiment_1.16.0   
 [9] SummarizedExperiment_1.24.0    Biobase_2.54.0                
[11] GenomicRanges_1.46.1           GenomeInfoDb_1.30.0           
[13] IRanges_2.28.0                 S4Vectors_0.32.3              
[15] BiocGenerics_0.40.0            MatrixGenerics_1.6.0          
[17] matrixStats_0.61.0-9001        BiocStyle_2.22.0              
[19] rebook_1.4.0                  

loaded via a namespace (and not attached):
  [1] AnnotationHub_3.2.0           BiocFileCache_2.2.0          
  [3] plyr_1.8.6                    igraph_1.2.9                 
  [5] lazyeval_0.2.2                splines_4.1.2                
  [7] BiocParallel_1.28.1           ggplot2_3.3.5                
  [9] scater_1.22.0                 digest_0.6.28                
 [11] foreach_1.5.1                 yulab.utils_0.0.4            
 [13] htmltools_0.5.2               viridis_0.6.2                
 [15] fansi_0.5.0                   magrittr_2.0.1               
 [17] memoise_2.0.0                 ScaledMatrix_1.2.0           
 [19] cluster_2.1.2                 DECIPHER_2.22.0              
 [21] colorspace_2.0-2              rappdirs_0.3.3               
 [23] blob_1.2.2                    ggrepel_0.9.1                
 [25] xfun_0.28                     dplyr_1.0.7                  
 [27] crayon_1.4.2                  RCurl_1.98-1.5               
 [29] jsonlite_1.7.2                graph_1.72.0                 
 [31] survival_3.2-13               iterators_1.0.13             
 [33] ape_5.5                       glue_1.5.0                   
 [35] gtable_0.3.0                  zlibbioc_1.40.0              
 [37] DelayedArray_0.20.0           BiocSingular_1.10.0          
 [39] Rhdf5lib_1.16.0               scales_1.1.1                 
 [41] DBI_1.1.1                     Rcpp_1.0.7                   
 [43] xtable_1.8-4                  viridisLite_0.4.0            
 [45] decontam_1.14.0               tidytree_0.3.6               
 [47] bit_4.0.4                     rsvd_1.0.5                   
 [49] httr_1.4.2                    dir.expiry_1.2.0             
 [51] ellipsis_0.3.2                pkgconfig_2.0.3              
 [53] XML_3.99-0.8                  scuttle_1.4.0                
 [55] CodeDepends_0.6.5             sass_0.4.0                   
 [57] dbplyr_2.1.1                  utf8_1.2.2                   
 [59] AnnotationDbi_1.56.2          later_1.3.0                  
 [61] tidyselect_1.1.1              rlang_0.4.12                 
 [63] reshape2_1.4.4                munsell_0.5.0                
 [65] BiocVersion_3.14.0            tools_4.1.2                  
 [67] cachem_1.0.6                  DirichletMultinomial_1.36.0  
 [69] generics_0.1.1                RSQLite_2.2.8                
 [71] ExperimentHub_2.2.0           ade4_1.7-18                  
 [73] evaluate_0.14                 biomformat_1.22.0            
 [75] stringr_1.4.0                 fastmap_1.1.0                
 [77] yaml_2.2.1                    knitr_1.36                   
 [79] bit64_4.0.5                   purrr_0.3.4                  
 [81] KEGGREST_1.34.0               nlme_3.1-153                 
 [83] sparseMatrixStats_1.6.0       mime_0.12                    
 [85] compiler_4.1.2                interactiveDisplayBase_1.32.0
 [87] curl_4.3.2                    beeswarm_0.4.0               
 [89] filelock_1.0.2                png_0.1-7                    
 [91] treeio_1.18.1                 tibble_3.1.6                 
 [93] bslib_0.3.1                   stringi_1.7.5                
 [95] highr_0.9                     lattice_0.20-45              
 [97] Matrix_1.3-4                  vegan_2.5-7                  
 [99] permute_0.9-5                 multtest_2.50.0              
[101] vctrs_0.3.8                   pillar_1.6.4                 
[103] lifecycle_1.0.1               rhdf5filters_1.6.0           
[105] BiocManager_1.30.16           jquerylib_0.1.4              
[107] BiocNeighbors_1.12.0          data.table_1.14.2            
[109] bitops_1.0-7                  irlba_2.3.3                  
[111] httpuv_1.6.3                  R6_2.5.1                     
[113] promises_1.2.0.1              bookdown_0.24                
[115] gridExtra_2.3                 vipor_0.4.5                  
[117] codetools_0.2-18              MASS_7.3-54                  
[119] assertthat_0.2.1              rhdf5_2.38.0                 
[121] GenomeInfoDbData_1.2.7        mgcv_1.8-38                  
[123] parallel_4.1.2                grid_4.1.2                   
[125] beachmat_2.10.0               tidyr_1.1.4                  
[127] rmarkdown_2.11                DelayedMatrixStats_1.16.0    
[129] shiny_1.7.1                   ggbeeswarm_0.6.0             </code></pre>
</div>

</div>
</div>
<h3>Bibliography</h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-R-SingleCellExperiment" class="csl-entry">
Lun, Aaron, and Davide Risso. 2020. <em>SingleCellExperiment: S4 Classes for Single Cell Data</em>.
</div>
<div id="ref-R-MultiAssayExperiment" class="csl-entry">
Ramos, Marcel, and Levi Waldron. 2020. <em>MultiAssayExperiment: Software for the Integration of Multi-Omics Experiments in Bioconductor</em>. <a href="http://waldronlab.io/MultiAssayExperiment/">http://waldronlab.io/MultiAssayExperiment/</a>.
</div>
<div id="ref-Vatanen2016" class="csl-entry">
Vatanen, Tommi, Aleksandar D. Kostic, Eva d’Hennezel, Heli Siljander, Eric A. Franzosa, Moran Yassour, Raivo Kolde, et al. 2016. <span>“Variation in Microbiome LPS Immunogenicity Contributes to Autoimmunity in Humans.”</span> <em>Cell</em> 165 (May): 842–53. <a href="https://doi.org/10.1016/j.cell.2016.04.007">https://doi.org/10.1016/j.cell.2016.04.007</a>.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="intro.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="packages.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["OMA.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
