<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 11 Differential abundance | Orchestrating Microbiome Analysis</title>
  <meta name="description" content="Chapter 11 Differential abundance | Orchestrating Microbiome Analysis" />
  <meta name="generator" content="bookdown 0.32 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 11 Differential abundance | Orchestrating Microbiome Analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="microbiome/OMA" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 11 Differential abundance | Orchestrating Microbiome Analysis" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="clustering.html"/>
<link rel="next" href="machine_learning.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.1/htmlwidgets.js"></script>
<script src="libs/plotly-binding-4.10.1/plotly.js"></script>
<script src="libs/typedarray-0.1/typedarray.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="libs/plotly-htmlwidgets-css-2.11.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="libs/plotly-main-2.11.1/plotly-latest.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Orchestrating Microbiome Analysis</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="part"><span><b>I Introduction</b></span></li>
<li class="chapter" data-level="1" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="containers.html"><a href="containers.html"><i class="fa fa-check"></i><b>2</b> Microbiome Data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="containers.html"><a href="containers.html#data-science-framework"><i class="fa fa-check"></i><b>2.1</b> Data science framework</a></li>
<li class="chapter" data-level="2.2" data-path="containers.html"><a href="containers.html#data-containers"><i class="fa fa-check"></i><b>2.2</b> Data containers</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="containers.html"><a href="containers.html#assay-data"><i class="fa fa-check"></i><b>2.2.1</b> Assay data</a></li>
<li class="chapter" data-level="2.2.2" data-path="containers.html"><a href="containers.html#coldata"><i class="fa fa-check"></i><b>2.2.2</b> colData</a></li>
<li class="chapter" data-level="2.2.3" data-path="containers.html"><a href="containers.html#rowdata"><i class="fa fa-check"></i><b>2.2.3</b> rowData</a></li>
<li class="chapter" data-level="2.2.4" data-path="containers.html"><a href="containers.html#rowtree"><i class="fa fa-check"></i><b>2.2.4</b> rowTree</a></li>
<li class="chapter" data-level="2.2.5" data-path="containers.html"><a href="containers.html#alt-exp"><i class="fa fa-check"></i><b>2.2.5</b> Alternative experiments</a></li>
<li class="chapter" data-level="2.2.6" data-path="containers.html"><a href="containers.html#mae"><i class="fa fa-check"></i><b>2.2.6</b> MultiAssayExperiments</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="containers.html"><a href="containers.html#loading-experimental-microbiome-data"><i class="fa fa-check"></i><b>2.3</b> Loading experimental microbiome data</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="containers.html"><a href="containers.html#s-workflow"><i class="fa fa-check"></i><b>2.3.1</b> 16S workflow</a></li>
<li class="chapter" data-level="2.3.2" data-path="containers.html"><a href="containers.html#import-from-external-files"><i class="fa fa-check"></i><b>2.3.2</b> Import from external files</a></li>
<li class="chapter" data-level="2.3.3" data-path="containers.html"><a href="containers.html#conversions-between-data-formats-in-r"><i class="fa fa-check"></i><b>2.3.3</b> Conversions between data formats in R</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="containers.html"><a href="containers.html#example-data"><i class="fa fa-check"></i><b>2.4</b> Demonstration data</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="containers.html"><a href="containers.html#package-data"><i class="fa fa-check"></i><b>2.4.1</b> Package data</a></li>
<li class="chapter" data-level="2.4.2" data-path="containers.html"><a href="containers.html#experimenthub-data"><i class="fa fa-check"></i><b>2.4.2</b> ExperimentHub data</a></li>
<li class="chapter" data-level="2.4.3" data-path="containers.html"><a href="containers.html#other-data-sources"><i class="fa fa-check"></i><b>2.4.3</b> Other data sources</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="containers.html"><a href="containers.html#session-info"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="packages.html"><a href="packages.html"><i class="fa fa-check"></i><b>3</b> Packages</a>
<ul>
<li class="chapter" data-level="3.1" data-path="packages.html"><a href="packages.html#package-installation"><i class="fa fa-check"></i><b>3.1</b> Package installation</a></li>
<li class="chapter" data-level="3.2" data-path="packages.html"><a href="packages.html#some-available-packages"><i class="fa fa-check"></i><b>3.2</b> Some available packages</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="packages.html"><a href="packages.html#mia-family-of-methods"><i class="fa fa-check"></i><b>3.2.1</b> mia family of methods</a></li>
<li class="chapter" data-level="3.2.2" data-path="packages.html"><a href="packages.html#tree-based-methods-sub-tree-methods"><i class="fa fa-check"></i><b>3.2.2</b> Tree-based methods {sub-tree-methods}</a></li>
<li class="chapter" data-level="3.2.3" data-path="packages.html"><a href="packages.html#differential-abundance-sub-diff-abund"><i class="fa fa-check"></i><b>3.2.3</b> Differential abundance {sub-diff-abund}</a></li>
<li class="chapter" data-level="3.2.4" data-path="packages.html"><a href="packages.html#manipulation-sub-manipulation"><i class="fa fa-check"></i><b>3.2.4</b> Manipulation {sub-manipulation}</a></li>
<li class="chapter" data-level="3.2.5" data-path="packages.html"><a href="packages.html#data"><i class="fa fa-check"></i><b>3.2.5</b> Data</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="packages.html"><a href="packages.html#session-info-1"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="part"><span><b>II Focus Topics</b></span></li>
<li class="chapter" data-level="4" data-path="datamanipulation.html"><a href="datamanipulation.html"><i class="fa fa-check"></i><b>4</b> Data Manipulation</a>
<ul>
<li class="chapter" data-level="4.1" data-path="datamanipulation.html"><a href="datamanipulation.html#tidying-and-subsetting"><i class="fa fa-check"></i><b>4.1</b> Tidying and subsetting</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="datamanipulation.html"><a href="datamanipulation.html#tidy-data"><i class="fa fa-check"></i><b>4.1.1</b> Tidy data</a></li>
<li class="chapter" data-level="4.1.2" data-path="datamanipulation.html"><a href="datamanipulation.html#subsetting"><i class="fa fa-check"></i><b>4.1.2</b> Subsetting</a></li>
<li class="chapter" data-level="4.1.3" data-path="datamanipulation.html"><a href="datamanipulation.html#splitting"><i class="fa fa-check"></i><b>4.1.3</b> Splitting</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="datamanipulation.html"><a href="datamanipulation.html#merge-data"><i class="fa fa-check"></i><b>4.2</b> Merge data</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="datamanipulation.html"><a href="datamanipulation.html#additional-functions"><i class="fa fa-check"></i><b>4.2.1</b> Additional functions</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>5</b> Exploration and quality Control</a>
<ul>
<li class="chapter" data-level="5.1" data-path="quality-control.html"><a href="quality-control.html#abundance"><i class="fa fa-check"></i><b>5.1</b> Abundance</a></li>
<li class="chapter" data-level="5.2" data-path="quality-control.html"><a href="quality-control.html#prevalence"><i class="fa fa-check"></i><b>5.2</b> Prevalence</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="quality-control.html"><a href="quality-control.html#prevalence-analysis"><i class="fa fa-check"></i><b>5.2.1</b> Prevalence analysis</a></li>
<li class="chapter" data-level="5.2.2" data-path="quality-control.html"><a href="quality-control.html#rare-taxa"><i class="fa fa-check"></i><b>5.2.2</b> Rare taxa</a></li>
<li class="chapter" data-level="5.2.3" data-path="quality-control.html"><a href="quality-control.html#plotting-prevalence"><i class="fa fa-check"></i><b>5.2.3</b> Plotting prevalence</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="quality-control.html"><a href="quality-control.html#qc"><i class="fa fa-check"></i><b>5.3</b> Quality control</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="quality-control.html"><a href="quality-control.html#top-taxa"><i class="fa fa-check"></i><b>5.3.1</b> Top taxa</a></li>
<li class="chapter" data-level="5.3.2" data-path="quality-control.html"><a href="quality-control.html#library-size-read-count"><i class="fa fa-check"></i><b>5.3.2</b> Library size / read count</a></li>
<li class="chapter" data-level="5.3.3" data-path="quality-control.html"><a href="quality-control.html#contaminant-sequences"><i class="fa fa-check"></i><b>5.3.3</b> Contaminant sequences</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="quality-control.html"><a href="quality-control.html#session-info-2"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="taxonomic-information.html"><a href="taxonomic-information.html"><i class="fa fa-check"></i><b>6</b> Taxonomic Information</a>
<ul>
<li class="chapter" data-level="6.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#assigning-taxonomic-information."><i class="fa fa-check"></i><b>6.1</b> Assigning taxonomic information.</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#dada2"><i class="fa fa-check"></i><b>6.1.1</b> dada2</a></li>
<li class="chapter" data-level="6.1.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#decipher"><i class="fa fa-check"></i><b>6.1.2</b> DECIPHER</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#functions-to-access-taxonomic-information"><i class="fa fa-check"></i><b>6.2</b> Functions to access taxonomic information</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#generate-a-taxonomic-tree-on-the-fly"><i class="fa fa-check"></i><b>6.2.1</b> Generate a taxonomic tree on the fly</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="taxonomic-information.html"><a href="taxonomic-information.html#data-agglomeration"><i class="fa fa-check"></i><b>6.3</b> Data agglomeration</a></li>
<li class="chapter" data-level="6.4" data-path="taxonomic-information.html"><a href="taxonomic-information.html#data-transformation"><i class="fa fa-check"></i><b>6.4</b> Data transformation</a></li>
<li class="chapter" data-level="6.5" data-path="taxonomic-information.html"><a href="taxonomic-information.html#pick-specific"><i class="fa fa-check"></i><b>6.5</b> Pick specific</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#abundances-of-all-taxa-in-specific-sample"><i class="fa fa-check"></i><b>6.5.1</b> Abundances of all taxa in specific sample</a></li>
<li class="chapter" data-level="6.5.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#abundances-of-specific-taxa-in-all-samples"><i class="fa fa-check"></i><b>6.5.2</b> Abundances of specific taxa in all samples</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="taxonomic-information.html"><a href="taxonomic-information.html#session-info-3"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="community-diversity.html"><a href="community-diversity.html"><i class="fa fa-check"></i><b>7</b> Community diversity</a>
<ul>
<li class="chapter" data-level="7.1" data-path="community-diversity.html"><a href="community-diversity.html#estimation"><i class="fa fa-check"></i><b>7.1</b> Estimation</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="community-diversity.html"><a href="community-diversity.html#richness"><i class="fa fa-check"></i><b>7.1.1</b> Richness</a></li>
<li class="chapter" data-level="7.1.2" data-path="community-diversity.html"><a href="community-diversity.html#estimate-diversity"><i class="fa fa-check"></i><b>7.1.2</b> Diversity</a></li>
<li class="chapter" data-level="7.1.3" data-path="community-diversity.html"><a href="community-diversity.html#faith-diversity"><i class="fa fa-check"></i><b>7.1.3</b> Faith phylogenetic diversity</a></li>
<li class="chapter" data-level="7.1.4" data-path="community-diversity.html"><a href="community-diversity.html#evenness"><i class="fa fa-check"></i><b>7.1.4</b> Evenness</a></li>
<li class="chapter" data-level="7.1.5" data-path="community-diversity.html"><a href="community-diversity.html#dominance"><i class="fa fa-check"></i><b>7.1.5</b> Dominance</a></li>
<li class="chapter" data-level="7.1.6" data-path="community-diversity.html"><a href="community-diversity.html#rarity"><i class="fa fa-check"></i><b>7.1.6</b> Rarity</a></li>
<li class="chapter" data-level="7.1.7" data-path="community-diversity.html"><a href="community-diversity.html#divergence"><i class="fa fa-check"></i><b>7.1.7</b> Divergence</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="community-diversity.html"><a href="community-diversity.html#visualization"><i class="fa fa-check"></i><b>7.2</b> Visualization</a></li>
<li class="chapter" data-level="" data-path="community-diversity.html"><a href="community-diversity.html#session-info-4"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="community-similarity.html"><a href="community-similarity.html"><i class="fa fa-check"></i><b>8</b> Community similarity</a>
<ul>
<li class="chapter" data-level="8.1" data-path="community-similarity.html"><a href="community-similarity.html#explained-variance"><i class="fa fa-check"></i><b>8.1</b> Explained variance</a></li>
<li class="chapter" data-level="8.2" data-path="community-similarity.html"><a href="community-similarity.html#community-comparisons-by-beta-diversity-analysis"><i class="fa fa-check"></i><b>8.2</b> Community comparisons by beta diversity analysis</a></li>
<li class="chapter" data-level="8.3" data-path="community-similarity.html"><a href="community-similarity.html#other-ordination-methods"><i class="fa fa-check"></i><b>8.3</b> Other ordination methods</a></li>
<li class="chapter" data-level="8.4" data-path="community-similarity.html"><a href="community-similarity.html#pcoa-genus"><i class="fa fa-check"></i><b>8.4</b> Visualizing the most dominant genus on PCoA</a>
<ul>
<li class="chapter" data-level="8.4.1" data-path="community-similarity.html"><a href="community-similarity.html#testing-differences-in-community-composition-between-sample-groups"><i class="fa fa-check"></i><b>8.4.1</b> Testing differences in community composition between sample groups</a></li>
<li class="chapter" data-level="8.4.2" data-path="community-similarity.html"><a href="community-similarity.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>8.4.2</b> Checking the homogeneity condition</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="community-similarity.html"><a href="community-similarity.html#further-reading"><i class="fa fa-check"></i><b>8.5</b> Further reading</a></li>
<li class="chapter" data-level="" data-path="community-similarity.html"><a href="community-similarity.html#session-info-5"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="microbiome-community.html"><a href="microbiome-community.html"><i class="fa fa-check"></i><b>9</b> Community composition</a>
<ul>
<li class="chapter" data-level="9.1" data-path="microbiome-community.html"><a href="microbiome-community.html#visual-composition"><i class="fa fa-check"></i><b>9.1</b> Visualizing taxonomic composition</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="microbiome-community.html"><a href="microbiome-community.html#composition-barplot"><i class="fa fa-check"></i><b>9.1.1</b> Composition barplot</a></li>
<li class="chapter" data-level="9.1.2" data-path="microbiome-community.html"><a href="microbiome-community.html#composition-heatmap"><i class="fa fa-check"></i><b>9.1.2</b> Composition heatmap</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="microbiome-community.html"><a href="microbiome-community.html#session-info-6"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="clustering.html"><a href="clustering.html"><i class="fa fa-check"></i><b>10</b> Community typing (clustering)</a>
<ul>
<li class="chapter" data-level="10.1" data-path="clustering.html"><a href="clustering.html#hiearchical-clustering"><i class="fa fa-check"></i><b>10.1</b> Hiearchical clustering</a></li>
<li class="chapter" data-level="10.2" data-path="clustering.html"><a href="clustering.html#k-means-clustering"><i class="fa fa-check"></i><b>10.2</b> K-means clustering</a></li>
<li class="chapter" data-level="10.3" data-path="clustering.html"><a href="clustering.html#dirichlet-multinomial-mixtures-dmm"><i class="fa fa-check"></i><b>10.3</b> Dirichlet Multinomial Mixtures (DMM)</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="clustering.html"><a href="clustering.html#pcoa-for-asv-level-data-with-bray-curtis-with-dmm-clusters-shown-with-colors"><i class="fa fa-check"></i><b>10.3.1</b> PCoA for ASV-level data with Bray-Curtis; with DMM clusters shown with colors</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="clustering.html"><a href="clustering.html#community-detection"><i class="fa fa-check"></i><b>10.4</b> Community Detection</a></li>
<li class="chapter" data-level="10.5" data-path="clustering.html"><a href="clustering.html#biclustering"><i class="fa fa-check"></i><b>10.5</b> Biclustering</a>
<ul>
<li class="chapter" data-level="10.5.1" data-path="clustering.html"><a href="clustering.html#taxa-vs-samples"><i class="fa fa-check"></i><b>10.5.1</b> Taxa vs samples</a></li>
<li class="chapter" data-level="10.5.2" data-path="clustering.html"><a href="clustering.html#taxa-vs-biomolecules"><i class="fa fa-check"></i><b>10.5.2</b> Taxa vs biomolecules</a></li>
<li class="chapter" data-level="10.5.3" data-path="clustering.html"><a href="clustering.html#taxa-vs-taxa"><i class="fa fa-check"></i><b>10.5.3</b> Taxa vs taxa</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="clustering.html"><a href="clustering.html#additional-community-typing"><i class="fa fa-check"></i><b>10.6</b> Additional Community Typing</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="differential-abundance.html"><a href="differential-abundance.html"><i class="fa fa-check"></i><b>11</b> Differential abundance</a>
<ul>
<li class="chapter" data-level="11.1" data-path="differential-abundance.html"><a href="differential-abundance.html#differential-abundance-analysis"><i class="fa fa-check"></i><b>11.1</b> Differential abundance analysis</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="differential-abundance.html"><a href="differential-abundance.html#examples-and-tools"><i class="fa fa-check"></i><b>11.1.1</b> Examples and tools</a></li>
<li class="chapter" data-level="11.1.2" data-path="differential-abundance.html"><a href="differential-abundance.html#prevalence-filtering"><i class="fa fa-check"></i><b>11.1.2</b> Prevalence Filtering</a></li>
<li class="chapter" data-level="11.1.3" data-path="differential-abundance.html"><a href="differential-abundance.html#aldex2"><i class="fa fa-check"></i><b>11.1.3</b> ALDEx2</a></li>
<li class="chapter" data-level="11.1.4" data-path="differential-abundance.html"><a href="differential-abundance.html#ancom-bc"><i class="fa fa-check"></i><b>11.1.4</b> ANCOM-BC</a></li>
<li class="chapter" data-level="11.1.5" data-path="differential-abundance.html"><a href="differential-abundance.html#maaslin2"><i class="fa fa-check"></i><b>11.1.5</b> MaAsLin2</a></li>
<li class="chapter" data-level="11.1.6" data-path="differential-abundance.html"><a href="differential-abundance.html#linda"><i class="fa fa-check"></i><b>11.1.6</b> LinDA</a></li>
<li class="chapter" data-level="11.1.7" data-path="differential-abundance.html"><a href="differential-abundance.html#comparison-of-the-methods"><i class="fa fa-check"></i><b>11.1.7</b> Comparison of the methods</a></li>
<li class="chapter" data-level="11.1.8" data-path="differential-abundance.html"><a href="differential-abundance.html#confounding-variables"><i class="fa fa-check"></i><b>11.1.8</b> Confounding variables</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="differential-abundance.html"><a href="differential-abundance.html#tree-based-methods"><i class="fa fa-check"></i><b>11.2</b> Tree-based methods</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="differential-abundance.html"><a href="differential-abundance.html#group-wise-associations-testing-based-on-balances"><i class="fa fa-check"></i><b>11.2.1</b> Group-wise associations testing based on balances</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="differential-abundance.html"><a href="differential-abundance.html#session-info-7"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="machine_learning.html"><a href="machine_learning.html"><i class="fa fa-check"></i><b>12</b> Machine learning</a>
<ul>
<li class="chapter" data-level="12.1" data-path="machine_learning.html"><a href="machine_learning.html#supervised-machine-learning"><i class="fa fa-check"></i><b>12.1</b> Supervised machine learning</a></li>
<li class="chapter" data-level="12.2" data-path="machine_learning.html"><a href="machine_learning.html#unsupervised-machine-learning"><i class="fa fa-check"></i><b>12.2</b> Unsupervised machine learning</a></li>
<li class="chapter" data-level="" data-path="machine_learning.html"><a href="machine_learning.html#session-info-8"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="multi-assay-analyses.html"><a href="multi-assay-analyses.html"><i class="fa fa-check"></i><b>13</b> Multi-assay analyses</a>
<ul>
<li class="chapter" data-level="13.1" data-path="multi-assay-analyses.html"><a href="multi-assay-analyses.html#cross-correlation-analysis"><i class="fa fa-check"></i><b>13.1</b> Cross-correlation Analysis</a></li>
<li class="chapter" data-level="13.2" data-path="multi-assay-analyses.html"><a href="multi-assay-analyses.html#multi-omics-factor-analysis"><i class="fa fa-check"></i><b>13.2</b> Multi-Omics Factor Analysis</a></li>
<li class="chapter" data-level="" data-path="multi-assay-analyses.html"><a href="multi-assay-analyses.html#session-info-9"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="viz-chapter.html"><a href="viz-chapter.html"><i class="fa fa-check"></i><b>14</b> Visualization</a>
<ul>
<li class="chapter" data-level="14.1" data-path="viz-chapter.html"><a href="viz-chapter.html#pre-analysis-exploration"><i class="fa fa-check"></i><b>14.1</b> Pre-analysis exploration</a>
<ul>
<li class="chapter" data-level="14.1.1" data-path="viz-chapter.html"><a href="viz-chapter.html#accessing-row-and-column-data"><i class="fa fa-check"></i><b>14.1.1</b> Accessing row and column data</a></li>
<li class="chapter" data-level="14.1.2" data-path="viz-chapter.html"><a href="viz-chapter.html#viewing-abundance-and-prevalence-patterns"><i class="fa fa-check"></i><b>14.1.2</b> Viewing abundance and prevalence patterns</a></li>
</ul></li>
<li class="chapter" data-level="14.2" data-path="viz-chapter.html"><a href="viz-chapter.html#diversity-estimation"><i class="fa fa-check"></i><b>14.2</b> Diversity estimation</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="viz-chapter.html"><a href="viz-chapter.html#alpha-diversity-with-scatter-violin-and-box-plots"><i class="fa fa-check"></i><b>14.2.1</b> Alpha diversity with scatter, violin and box plots</a></li>
<li class="chapter" data-level="14.2.2" data-path="viz-chapter.html"><a href="viz-chapter.html#beta-diversity-with-shepard-and-coordination-plots"><i class="fa fa-check"></i><b>14.2.2</b> Beta diversity with Shepard and coordination plots</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="viz-chapter.html"><a href="viz-chapter.html#statistical-analysis"><i class="fa fa-check"></i><b>14.3</b> Statistical analysis</a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="viz-chapter.html"><a href="viz-chapter.html#heatmaps"><i class="fa fa-check"></i><b>14.3.1</b> Heatmaps</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="viz-chapter.html"><a href="viz-chapter.html#session-info-10"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="part"><span><b>III Appendix</b></span></li>
<li class="chapter" data-level="15" data-path="training.html"><a href="training.html"><i class="fa fa-check"></i><b>15</b> Training</a>
<ul>
<li class="chapter" data-level="15.1" data-path="training.html"><a href="training.html#coc"><i class="fa fa-check"></i><b>15.1</b> Code of Conduct</a></li>
<li class="chapter" data-level="15.2" data-path="training.html"><a href="training.html#checklist"><i class="fa fa-check"></i><b>15.2</b> Checklist</a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="training.html"><a href="training.html#setting-up-your-own-computer"><i class="fa fa-check"></i><b>15.2.1</b> Setting up Your own computer</a></li>
<li class="chapter" data-level="15.2.2" data-path="training.html"><a href="training.html#support-and-resources"><i class="fa fa-check"></i><b>15.2.2</b> Support and resources</a></li>
<li class="chapter" data-level="15.2.3" data-path="training.html"><a href="training.html#material"><i class="fa fa-check"></i><b>15.2.3</b> Study material</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="packages.html"><a href="packages.html#packages"><i class="fa fa-check"></i><b>15.3</b> Installing and loading the required R packages</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="resources.html"><a href="resources.html"><i class="fa fa-check"></i><b>16</b> Resources</a>
<ul>
<li class="chapter" data-level="16.1" data-path="resources.html"><a href="resources.html#data-containers-1"><i class="fa fa-check"></i><b>16.1</b> Data containers</a>
<ul>
<li class="chapter" data-level="16.1.1" data-path="resources.html"><a href="resources.html#resources-for-treesummarizedexperiment"><i class="fa fa-check"></i><b>16.1.1</b> Resources for TreeSummarizedExperiment</a></li>
<li class="chapter" data-level="16.1.2" data-path="resources.html"><a href="resources.html#other-relevant-containers"><i class="fa fa-check"></i><b>16.1.2</b> Other relevant containers</a></li>
<li class="chapter" data-level="16.1.3" data-path="resources.html"><a href="resources.html#alternative-containers-for-microbiome-data"><i class="fa fa-check"></i><b>16.1.3</b> Alternative containers for microbiome data</a></li>
<li class="chapter" data-level="16.1.4" data-path="resources.html"><a href="resources.html#resources-for-phyloseq"><i class="fa fa-check"></i><b>16.1.4</b> Resources for phyloseq</a></li>
</ul></li>
<li class="chapter" data-level="16.2" data-path="resources.html"><a href="resources.html#r-programming-resources"><i class="fa fa-check"></i><b>16.2</b> R programming resources</a>
<ul>
<li class="chapter" data-level="16.2.1" data-path="resources.html"><a href="resources.html#bioc_intro"><i class="fa fa-check"></i><b>16.2.1</b> Bioconductor Classes</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="resources.html"><a href="resources.html#reproducible-reporting-with-quarto"><i class="fa fa-check"></i><b>16.3</b> Reproducible reporting with Quarto</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="extras.html"><a href="extras.html"><i class="fa fa-check"></i><b>17</b> Extra material</a>
<ul>
<li class="chapter" data-level="17.1" data-path="extras.html"><a href="extras.html#interactive-3d-plots"><i class="fa fa-check"></i><b>17.1</b> Interactive 3D Plots</a></li>
<li class="chapter" data-level="17.2" data-path="extras.html"><a href="extras.html#permanova-comparison"><i class="fa fa-check"></i><b>17.2</b> PERMANOVA comparison</a></li>
<li class="chapter" data-level="17.3" data-path="extras.html"><a href="extras.html#bayesian-multinomial-logistic-normal-models"><i class="fa fa-check"></i><b>17.3</b> Bayesian Multinomial Logistic-Normal Models</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="acknowledgments.html"><a href="acknowledgments.html"><i class="fa fa-check"></i><b>18</b> Acknowledgments</a></li>
<li class="chapter" data-level="19" data-path="exercises.html"><a href="exercises.html"><i class="fa fa-check"></i><b>19</b> Exercises</a>
<ul>
<li class="chapter" data-level="19.1" data-path="exercises.html"><a href="exercises.html#workflows"><i class="fa fa-check"></i><b>19.1</b> Workflows</a>
<ul>
<li class="chapter" data-level="19.1.1" data-path="exercises.html"><a href="exercises.html#reproducible-reporting"><i class="fa fa-check"></i><b>19.1.1</b> Reproducible reporting</a></li>
</ul></li>
<li class="chapter" data-level="19.2" data-path="exercises.html"><a href="exercises.html#treesummarizedexperiment-basic-components"><i class="fa fa-check"></i><b>19.2</b> TreeSummarizedExperiment: basic components</a>
<ul>
<li class="chapter" data-level="19.2.1" data-path="exercises.html"><a href="exercises.html#basic-summaries"><i class="fa fa-check"></i><b>19.2.1</b> Basic summaries</a></li>
<li class="chapter" data-level="19.2.2" data-path="exercises.html"><a href="exercises.html#taxonomic-abundance-table-assay"><i class="fa fa-check"></i><b>19.2.2</b> Taxonomic abundance table (assay)</a></li>
<li class="chapter" data-level="19.2.3" data-path="exercises.html"><a href="exercises.html#sample-side-information"><i class="fa fa-check"></i><b>19.2.3</b> Sample side information</a></li>
<li class="chapter" data-level="19.2.4" data-path="exercises.html"><a href="exercises.html#feature-side-information"><i class="fa fa-check"></i><b>19.2.4</b> Feature side information</a></li>
<li class="chapter" data-level="19.2.5" data-path="exercises.html"><a href="exercises.html#other-components"><i class="fa fa-check"></i><b>19.2.5</b> Other components</a></li>
</ul></li>
<li class="chapter" data-level="19.3" data-path="exercises.html"><a href="exercises.html#treesummarizedexperiment-data-import"><i class="fa fa-check"></i><b>19.3</b> TreeSummarizedExperiment: data import</a></li>
<li class="chapter" data-level="19.4" data-path="exercises.html"><a href="exercises.html#data-manipulation"><i class="fa fa-check"></i><b>19.4</b> Data manipulation</a>
<ul>
<li class="chapter" data-level="19.4.1" data-path="exercises.html"><a href="exercises.html#subsetting-1"><i class="fa fa-check"></i><b>19.4.1</b> Subsetting</a></li>
<li class="chapter" data-level="19.4.2" data-path="exercises.html"><a href="exercises.html#library-sizes"><i class="fa fa-check"></i><b>19.4.2</b> Library sizes</a></li>
<li class="chapter" data-level="19.4.3" data-path="exercises.html"><a href="exercises.html#prevalent-and-core-taxonomic-features"><i class="fa fa-check"></i><b>19.4.3</b> Prevalent and core taxonomic features</a></li>
<li class="chapter" data-level="19.4.4" data-path="exercises.html"><a href="exercises.html#explore-the-data"><i class="fa fa-check"></i><b>19.4.4</b> Explore the data</a></li>
<li class="chapter" data-level="19.4.5" data-path="exercises.html"><a href="exercises.html#other-functions"><i class="fa fa-check"></i><b>19.4.5</b> Other functions</a></li>
</ul></li>
<li class="chapter" data-level="19.5" data-path="exercises.html"><a href="exercises.html#transformations"><i class="fa fa-check"></i><b>19.5</b> Transformations</a>
<ul>
<li class="chapter" data-level="19.5.1" data-path="exercises.html"><a href="exercises.html#transformations-1"><i class="fa fa-check"></i><b>19.5.1</b> Transformations</a></li>
</ul></li>
<li class="chapter" data-level="19.6" data-path="exercises.html"><a href="exercises.html#taxonomic-levels"><i class="fa fa-check"></i><b>19.6</b> Taxonomic levels</a>
<ul>
<li class="chapter" data-level="19.6.1" data-path="exercises.html"><a href="exercises.html#taxonomic-levels-1"><i class="fa fa-check"></i><b>19.6.1</b> Taxonomic levels</a></li>
<li class="chapter" data-level="19.6.2" data-path="exercises.html"><a href="exercises.html#alternative-experiments-altexp"><i class="fa fa-check"></i><b>19.6.2</b> Alternative experiments (altExp)</a></li>
</ul></li>
<li class="chapter" data-level="19.7" data-path="exercises.html"><a href="exercises.html#alpha-diversity"><i class="fa fa-check"></i><b>19.7</b> Alpha diversity</a>
<ul>
<li class="chapter" data-level="19.7.1" data-path="exercises.html"><a href="exercises.html#alpha-diversity-basics"><i class="fa fa-check"></i><b>19.7.1</b> Alpha diversity basics</a></li>
<li class="chapter" data-level="19.7.2" data-path="exercises.html"><a href="exercises.html#alpha-diversity-extra"><i class="fa fa-check"></i><b>19.7.2</b> Alpha diversity extra</a></li>
</ul></li>
<li class="chapter" data-level="19.8" data-path="exercises.html"><a href="exercises.html#community-composition"><i class="fa fa-check"></i><b>19.8</b> Community composition</a>
<ul>
<li class="chapter" data-level="19.8.1" data-path="exercises.html"><a href="exercises.html#beta-diversity-basics"><i class="fa fa-check"></i><b>19.8.1</b> Beta diversity basics</a></li>
<li class="chapter" data-level="19.8.2" data-path="exercises.html"><a href="exercises.html#beta-diversity-extra"><i class="fa fa-check"></i><b>19.8.2</b> Beta diversity extra</a></li>
</ul></li>
<li class="chapter" data-level="19.9" data-path="exercises.html"><a href="exercises.html#visualization-1"><i class="fa fa-check"></i><b>19.9</b> Visualization</a>
<ul>
<li class="chapter" data-level="19.9.1" data-path="exercises.html"><a href="exercises.html#multivariate-ordination"><i class="fa fa-check"></i><b>19.9.1</b> Multivariate ordination</a></li>
<li class="chapter" data-level="19.9.2" data-path="exercises.html"><a href="exercises.html#heatmap-visualization"><i class="fa fa-check"></i><b>19.9.2</b> Heatmap visualization</a></li>
</ul></li>
<li class="chapter" data-level="19.10" data-path="exercises.html"><a href="exercises.html#differential-abundance-1"><i class="fa fa-check"></i><b>19.10</b> Differential abundance</a>
<ul>
<li class="chapter" data-level="19.10.1" data-path="exercises.html"><a href="exercises.html#univariate-analyses"><i class="fa fa-check"></i><b>19.10.1</b> Univariate analyses</a></li>
<li class="chapter" data-level="19.10.2" data-path="exercises.html"><a href="exercises.html#differential-abundance-analysis-1"><i class="fa fa-check"></i><b>19.10.2</b> Differential abundance analysis</a></li>
</ul></li>
<li class="chapter" data-level="19.11" data-path="exercises.html"><a href="exercises.html#multiomics"><i class="fa fa-check"></i><b>19.11</b> Multiomics</a>
<ul>
<li class="chapter" data-level="19.11.1" data-path="exercises.html"><a href="exercises.html#introduction-to-multiassayexperiment-mae"><i class="fa fa-check"></i><b>19.11.1</b> Introduction to MultiAssayExperiment (MAE)</a></li>
<li class="chapter" data-level="19.11.2" data-path="exercises.html"><a href="exercises.html#introduction-to-multiomics"><i class="fa fa-check"></i><b>19.11.2</b> Introduction to multiomics</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="bibliography.html"><a href="bibliography.html"><i class="fa fa-check"></i>Bibliography</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Orchestrating Microbiome Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="differential-abundance" class="section level1 hasAnchor" number="11">
<h1><span class="header-section-number">Chapter 11</span> Differential abundance<a href="differential-abundance.html#differential-abundance" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<script>
document.addEventListener("click", function (event) {
    if (event.target.classList.contains("rebook-collapse")) {
        event.target.classList.toggle("active");
        var content = event.target.nextElementSibling;
        if (content.style.display === "block") {
            content.style.display = "none";
        } else {
            content.style.display = "block";
        }
    }
})
</script>
<style>
.rebook-collapse {
  background-color: #eee;
  color: #444;
  cursor: pointer;
  padding: 18px;
  width: 100%;
  border: none;
  text-align: left;
  outline: none;
  font-size: 15px;
}

.rebook-content {
  padding: 0 18px;
  display: none;
  overflow: hidden;
  background-color: #f1f1f1;
}
</style>
<div id="differential-abundance-analysis" class="section level2 hasAnchor" number="11.1">
<h2><span class="header-section-number">11.1</span> Differential abundance analysis<a href="differential-abundance.html#differential-abundance-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This section provides an overview and examples of <em>differential
abundance analysis (DAA)</em> based on one of the <a href="https://microbiome.github.io/mia/reference/mia-datasets.html">openly available
datasets</a>
in mia to illustrate how to perform differential abundance analysis
(DAA). DAA identifies differences in the abundances of individual
taxonomic groups between two or more groups (e.g. treatment vs
control). This can be performed at any phylogenetic level.</p>
<p>We perform DAA to identify biomarkers and/or gain understanding of a
complex system by looking at its isolated components. For example,
identifying that a bacterial taxon is different between a patient
group with disease <em>X</em> vs a healthy control group might lead to
important insights into the pathophysiology. Changes in the microbiota
might be cause or a consequence of a disease. Either way, it can
help to understand the system as a whole. Be aware that this approach
has also been criticized recently <span class="citation">(<a href="#ref-Quinn2021" role="doc-biblioref">Quinn, Gordon-Rodriguez, and Erb 2021</a>)</span>.</p>
<div id="examples-and-tools" class="section level3 hasAnchor" number="11.1.1">
<h3><span class="header-section-number">11.1.1</span> Examples and tools<a href="differential-abundance.html#examples-and-tools" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>There are many tools to perform DAA. The most popular tools, without going into
evaluating whether or not they perform well for this task, are:</p>
<ul>
<li>ALDEx2 <span class="citation">(<a href="#ref-Gloor2016" role="doc-biblioref">Gloor, Macklaim, and Fernandes 2016</a>)</span></li>
<li>ANCOMBC <span class="citation">(<a href="#ref-ancombc2020" role="doc-biblioref">Lin and Peddada 2020a</a>)</span></li>
<li>corncob <span class="citation">(<a href="#ref-Martin2021" role="doc-biblioref">Martin, Witten, and Willis 2021</a>)</span></li>
<li>DESeq2 <span class="citation">(<a href="#ref-Love2014" role="doc-biblioref">Love, Huber, and Anders 2014</a>)</span></li>
<li>edgeR <span class="citation">(<a href="#ref-Chen2016" role="doc-biblioref">Chen, Lun, and Smyth 2016</a>)</span></li>
<li>lefser <span class="citation">(<a href="#ref-Khlebrodova2021" role="doc-biblioref">Khleborodova 2021</a>)</span></li>
<li>Maaslin2 <span class="citation">(<a href="#ref-Mallick2020" role="doc-biblioref">Mallick, Rahnavard, and McIver 2020</a>)</span></li>
<li>metagenomeSeq <span class="citation">(<a href="#ref-Paulson2017" role="doc-biblioref">Paulson, Talukder, and Bravo 2017</a>)</span></li>
<li>limma <span class="citation">(<a href="#ref-Ritchie2015" role="doc-biblioref">Ritchie et al. 2015</a>)</span></li>
<li>LinDA <span class="citation">(<a href="#ref-Zhou2022" role="doc-biblioref">Zhou et al. 2022</a>)</span></li>
<li><a href="https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/t.test">t-test</a><br />
</li>
<li><a href="https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/wilcox.test">Wilcoxon test</a></li>
</ul>
<p>We recommend to have a look at <span class="citation">Nearing et al. (<a href="#ref-Nearing2022" role="doc-biblioref">2022</a>)</span>
who compared all these listed methods across 38
different datasets. Because different methods use different approaches
(parametric vs non-parametric, different normalization techiniques, assumptions
etc.), results can differ between methods.
Unfortunately, as <span class="citation">Nearing et al. (<a href="#ref-Nearing2022" role="doc-biblioref">2022</a>)</span> point out, they
can differ substantially. Therefore, it is highly recommended to pick several
methods to get an idea about how robust and potentially reproducible your
findings are depending on the method. In this section we demonstrate 4 methods
that can be recommended based on recent literature (ANCOM-BC, ALDEx2, Maaslin2
and LinDA) and we will compare the results between them.
Note that the purpose of this section is to show how to perform DAA in R, not
how to correctly do causal inference. Depending on your experimental setup
and your theory, you must determine how to specify any model exactly.
E.g., there might be confounding factors that might drive (the absence of)
differences between the shown groups that we ignore here for simplicity.
However, we will show how you could include covariates in those models.
Furthermore, we picked a dataset that merely has microbial abundances in a TSE
object as well as a grouping variable in the sample data. We simplify the
analysis by only including 2 of the 3 groups.</p>
<div class="sourceCode" id="cb323"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb323-1"><a href="differential-abundance.html#cb323-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mia)</span>
<span id="cb323-2"><a href="differential-abundance.html#cb323-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb323-3"><a href="differential-abundance.html#cb323-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidySummarizedExperiment)</span>
<span id="cb323-4"><a href="differential-abundance.html#cb323-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ALDEx2)</span>
<span id="cb323-5"><a href="differential-abundance.html#cb323-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Maaslin2)</span>
<span id="cb323-6"><a href="differential-abundance.html#cb323-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(MicrobiomeStat)</span>
<span id="cb323-7"><a href="differential-abundance.html#cb323-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(knitr)</span>
<span id="cb323-8"><a href="differential-abundance.html#cb323-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb323-9"><a href="differential-abundance.html#cb323-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb323-10"><a href="differential-abundance.html#cb323-10" aria-hidden="true" tabindex="-1"></a><span class="co"># set random seed because some tools can randomly vary and then produce </span></span>
<span id="cb323-11"><a href="differential-abundance.html#cb323-11" aria-hidden="true" tabindex="-1"></a><span class="co"># different results:</span></span>
<span id="cb323-12"><a href="differential-abundance.html#cb323-12" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">13253</span>)</span>
<span id="cb323-13"><a href="differential-abundance.html#cb323-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb323-14"><a href="differential-abundance.html#cb323-14" aria-hidden="true" tabindex="-1"></a><span class="co"># For ANCOMBC we need development version</span></span>
<span id="cb323-15"><a href="differential-abundance.html#cb323-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Can be installed with:</span></span>
<span id="cb323-16"><a href="differential-abundance.html#cb323-16" aria-hidden="true" tabindex="-1"></a><span class="co"># remotes::install_github(&quot;FrederickHuangLin/ANCOMBC&quot;)</span></span>
<span id="cb323-17"><a href="differential-abundance.html#cb323-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ANCOMBC)</span>
<span id="cb323-18"><a href="differential-abundance.html#cb323-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb323-19"><a href="differential-abundance.html#cb323-19" aria-hidden="true" tabindex="-1"></a><span class="co"># we use a demo dataset and restrict it to two geo locations</span></span>
<span id="cb323-20"><a href="differential-abundance.html#cb323-20" aria-hidden="true" tabindex="-1"></a><span class="co"># for easy illustration</span></span>
<span id="cb323-21"><a href="differential-abundance.html#cb323-21" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(peerj13075)</span>
<span id="cb323-22"><a href="differential-abundance.html#cb323-22" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> peerj13075</span>
<span id="cb323-23"><a href="differential-abundance.html#cb323-23" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> tse[ ,tse<span class="sc">$</span>Geographical_location <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Pune&quot;</span>, <span class="st">&quot;Nashik&quot;</span>)]</span>
<span id="cb323-24"><a href="differential-abundance.html#cb323-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Let us make this a factor</span></span>
<span id="cb323-25"><a href="differential-abundance.html#cb323-25" aria-hidden="true" tabindex="-1"></a>tse<span class="sc">$</span>Geographical_location <span class="ot">&lt;-</span> <span class="fu">factor</span>(tse<span class="sc">$</span>Geographical_location)</span>
<span id="cb323-26"><a href="differential-abundance.html#cb323-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb323-27"><a href="differential-abundance.html#cb323-27" aria-hidden="true" tabindex="-1"></a><span class="co"># how many observations do we have per group?</span></span>
<span id="cb323-28"><a href="differential-abundance.html#cb323-28" aria-hidden="true" tabindex="-1"></a><span class="fu">count</span>(<span class="fu">as.data.frame</span>(<span class="fu">colData</span>(tse)), Geographical_location) <span class="sc">%&gt;%</span> <span class="fu">kable</span>()</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">Geographical_location</th>
<th align="right">n</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Nashik</td>
<td align="right">11</td>
</tr>
<tr class="even">
<td align="left">Pune</td>
<td align="right">36</td>
</tr>
</tbody>
</table>
</div>
<div id="prevalence-filtering" class="section level3 hasAnchor" number="11.1.2">
<h3><span class="header-section-number">11.1.2</span> Prevalence Filtering<a href="differential-abundance.html#prevalence-filtering" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Before we jump to our analyses, we may want to perform prevalence filtering.
<span class="citation">Nearing et al. (<a href="#ref-Nearing2022" role="doc-biblioref">2022</a>)</span> found that applying a 10% threshold
for the prevalence of the taxa generally resulted in more robust results.
Some tools have builtin arguments for that. By applying the threshold to our
input data, we can make sure it is applied for all tools. Below we show how to
do this in <code>mia</code>:</p>
<div class="sourceCode" id="cb324"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb324-1"><a href="differential-abundance.html#cb324-1" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">subsetByPrevalentTaxa</span>(tse, <span class="at">detection =</span> <span class="dv">0</span>, <span class="at">prevalence =</span> <span class="fl">0.1</span>)</span></code></pre></div>
</div>
<div id="aldex2" class="section level3 hasAnchor" number="11.1.3">
<h3><span class="header-section-number">11.1.3</span> ALDEx2<a href="differential-abundance.html#aldex2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>In this section, we will show how to perform a simple ALDEx2 analysis.
If you wanted to pick a single method, this method could be recommended to use.
According to the developers experience, it tends to identify the common
features identified by other methods. This statement is in line with a recent
independent evaluation by <span class="citation">Nearing et al. (<a href="#ref-Nearing2022" role="doc-biblioref">2022</a>)</span>.<br />
Please also have a look at the more extensive
<a href="https://bioconductor.org/packages/release/bioc/vignettes/ALDEx2/inst/doc/ALDEx2_vignette.html">vignette</a>
that covers this flexible tool in more depth. ALDEx2 estimates technical
variation within each sample per taxon by utilizing the Dirichlet distribution.
It furthermore applies the centered-log-ratio transformation (or closely
related log-ratio transforms). Depending on the experimental setup, it will
perform a two sample Welch’s T-test and Wilcoxon-test or a one-way ANOVA and
Kruskal-Wallis-test. For more complex study designs, there is a possibility to
utilize the <code>glm</code> functionality within ALDEx2. The Benjamini-Hochberg procedure
is applied in any case to correct for multiple testing. Below we show a simple
example that illustrates the workflow.</p>
<div class="sourceCode" id="cb325"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb325-1"><a href="differential-abundance.html#cb325-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate Monte Carlo samples of the Dirichlet distribution for each sample.</span></span>
<span id="cb325-2"><a href="differential-abundance.html#cb325-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert each instance using the centered log-ratio transform.</span></span>
<span id="cb325-3"><a href="differential-abundance.html#cb325-3" aria-hidden="true" tabindex="-1"></a><span class="co"># This is the input for all further analyses.</span></span>
<span id="cb325-4"><a href="differential-abundance.html#cb325-4" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">254</span>)</span>
<span id="cb325-5"><a href="differential-abundance.html#cb325-5" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">aldex.clr</span>(<span class="fu">assay</span>(tse), tse<span class="sc">$</span>Geographical_location)     </span></code></pre></div>
<p>The t-test:</p>
<div class="sourceCode" id="cb326"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb326-1"><a href="differential-abundance.html#cb326-1" aria-hidden="true" tabindex="-1"></a><span class="co"># calculates expected values of the Welch&#39;s t-test and Wilcoxon rank</span></span>
<span id="cb326-2"><a href="differential-abundance.html#cb326-2" aria-hidden="true" tabindex="-1"></a><span class="co"># test on the data returned by aldex.clr</span></span>
<span id="cb326-3"><a href="differential-abundance.html#cb326-3" aria-hidden="true" tabindex="-1"></a>x_tt <span class="ot">&lt;-</span> <span class="fu">aldex.ttest</span>(x, <span class="at">paired.test =</span> <span class="cn">FALSE</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p>Effect sizes:</p>
<div class="sourceCode" id="cb327"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb327-1"><a href="differential-abundance.html#cb327-1" aria-hidden="true" tabindex="-1"></a><span class="co"># determines the median clr abundance of the feature in all samples and in</span></span>
<span id="cb327-2"><a href="differential-abundance.html#cb327-2" aria-hidden="true" tabindex="-1"></a><span class="co"># groups, the median difference between the two groups, the median variation</span></span>
<span id="cb327-3"><a href="differential-abundance.html#cb327-3" aria-hidden="true" tabindex="-1"></a><span class="co"># within each group and the effect size, which is the median of the ratio</span></span>
<span id="cb327-4"><a href="differential-abundance.html#cb327-4" aria-hidden="true" tabindex="-1"></a><span class="co"># of the between group difference and the larger of the variance within groups</span></span>
<span id="cb327-5"><a href="differential-abundance.html#cb327-5" aria-hidden="true" tabindex="-1"></a>x_effect <span class="ot">&lt;-</span> <span class="fu">aldex.effect</span>(x, <span class="at">CI =</span> <span class="cn">TRUE</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb327-6"><a href="differential-abundance.html#cb327-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb327-7"><a href="differential-abundance.html#cb327-7" aria-hidden="true" tabindex="-1"></a><span class="co"># combine all outputs </span></span>
<span id="cb327-8"><a href="differential-abundance.html#cb327-8" aria-hidden="true" tabindex="-1"></a>aldex_out <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(x_tt, x_effect)</span></code></pre></div>
<p>Now, we can create a so called Bland-Altman or MA plot (left). It shows the
association between the relative abundance and the magnitude of the difference
per sample. Next to that, we can also create a plot that shows the dispersion
on the x-axis instead of log-ratio abundance. Red dots represent genera that are
differentially abundant (<span class="math inline">\(q \leq 0.1\)</span>) between the 2 groups. Black points are
rare taxa and grey ones are abundant taxa. The dashed line represent an effect
size of 1. See <span class="citation">Gloor, Macklaim, and Fernandes (<a href="#ref-Gloor2016" role="doc-biblioref">2016</a>)</span> to learn more about these plots.</p>
<div class="sourceCode" id="cb328"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb328-1"><a href="differential-abundance.html#cb328-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb328-2"><a href="differential-abundance.html#cb328-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aldex.plot</span>(</span>
<span id="cb328-3"><a href="differential-abundance.html#cb328-3" aria-hidden="true" tabindex="-1"></a>    aldex_out, </span>
<span id="cb328-4"><a href="differential-abundance.html#cb328-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">&quot;MA&quot;</span>, </span>
<span id="cb328-5"><a href="differential-abundance.html#cb328-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">test =</span> <span class="st">&quot;welch&quot;</span>, </span>
<span id="cb328-6"><a href="differential-abundance.html#cb328-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">&quot;Log-ratio abundance&quot;</span>,</span>
<span id="cb328-7"><a href="differential-abundance.html#cb328-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylab =</span> <span class="st">&quot;Difference&quot;</span>,</span>
<span id="cb328-8"><a href="differential-abundance.html#cb328-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">cutoff =</span> <span class="fl">0.05</span></span>
<span id="cb328-9"><a href="differential-abundance.html#cb328-9" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb328-10"><a href="differential-abundance.html#cb328-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aldex.plot</span>(</span>
<span id="cb328-11"><a href="differential-abundance.html#cb328-11" aria-hidden="true" tabindex="-1"></a>    aldex_out, </span>
<span id="cb328-12"><a href="differential-abundance.html#cb328-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">&quot;MW&quot;</span>, </span>
<span id="cb328-13"><a href="differential-abundance.html#cb328-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">test =</span> <span class="st">&quot;welch&quot;</span>,</span>
<span id="cb328-14"><a href="differential-abundance.html#cb328-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">&quot;Dispersion&quot;</span>,</span>
<span id="cb328-15"><a href="differential-abundance.html#cb328-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylab =</span> <span class="st">&quot;Difference&quot;</span>,</span>
<span id="cb328-16"><a href="differential-abundance.html#cb328-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">cutoff =</span> <span class="fl">0.05</span></span>
<span id="cb328-17"><a href="differential-abundance.html#cb328-17" aria-hidden="true" tabindex="-1"></a>  )</span></code></pre></div>
<p>The evaluation as differential abundant in above plots is based on the
corrected pvalue. According to the ALDEx2 developers, the safest
approach is to identify those features where the 95% CI of the effect
size does not cross 0. As we can see in below table, this is not the
case for any of the identified genera (see overlap column, which
indicates the proportion of overlap). Also, the authors recommend to
focus on effect sizes and CIs rather than interpreting the pvalue. To
keep the comparison simple, we will here use the pvalue as decision
criterion. But please be aware that the effect size together with the
CI is a better answer to the question we are typically interested in
(see also <a href="https://www.nature.com/articles/d41586-019-00857-9">this
article</a>).</p>
<div class="sourceCode" id="cb329"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb329-1"><a href="differential-abundance.html#cb329-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames_to_column</span>(aldex_out, <span class="st">&quot;genus&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb329-2"><a href="differential-abundance.html#cb329-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(wi.eBH <span class="sc">&lt;=</span> <span class="fl">0.05</span>)  <span class="sc">%&gt;%</span> <span class="co"># here we chose the wilcoxon output rather than tt</span></span>
<span id="cb329-3"><a href="differential-abundance.html#cb329-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(genus, we.eBH, wi.eBH, effect, overlap) <span class="sc">%&gt;%</span></span>
<span id="cb329-4"><a href="differential-abundance.html#cb329-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>()</span></code></pre></div>
</div>
<div id="ancom-bc" class="section level3 hasAnchor" number="11.1.4">
<h3><span class="header-section-number">11.1.4</span> ANCOM-BC<a href="differential-abundance.html#ancom-bc" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The analysis of composition of microbiomes with bias correction
(ANCOM-BC) <span class="citation">(<a href="#ref-Das2020" role="doc-biblioref">Lin and Peddada 2020b</a>)</span> is a recently developed method for differential
abundance testing. It is based on an earlier published approach
<span class="citation">(<a href="#ref-Mandal2015" role="doc-biblioref">Mandal et al. 2015</a>)</span>. The previous version of ANCOM was among the methods
that produced the most consistent results and is probably a
conservative approach <span class="citation">(<a href="#ref-Nearing2022" role="doc-biblioref">Nearing et al. 2022</a>)</span>. However, the new ANCOM-BC
method operates quite differently compared to the former ANCOM method.</p>
<p>As the only method, ANCOM-BC incorporates the so called <em>sampling
fraction</em> into the model. The latter term could be empirically
estimated by the ratio of the library size to the microbial
load. According to the authors, variations in this sampling fraction
would bias differential abundance analyses if ignored. Furthermore,
this method provides p-values and confidence intervals for each
taxon. It also controls the FDR and it is computationally simple to
implement.</p>
<p>As we will see below, to obtain results, all that is needed is to pass
a tse object to the <code>ancombc()</code> function. Below, we first specify a
formula. In this formula, other covariates could potentially be
included to adjust for confounding. We show this further below.
Please check the <a href="https://rdrr.io/github/FrederickHuangLin/ANCOMBC/man/ancombc.html">function documentation</a>
to learn about the additional arguments that we specify below.</p>
<p>We recommend to use or higher in real case studies.</p>
<div class="sourceCode" id="cb330"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb330-1"><a href="differential-abundance.html#cb330-1" aria-hidden="true" tabindex="-1"></a><span class="co"># perform the analysis </span></span>
<span id="cb330-2"><a href="differential-abundance.html#cb330-2" aria-hidden="true" tabindex="-1"></a>out <span class="ot">&lt;-</span> <span class="fu">ancombc2</span>(</span>
<span id="cb330-3"><a href="differential-abundance.html#cb330-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> tse,</span>
<span id="cb330-4"><a href="differential-abundance.html#cb330-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">tax_level=</span><span class="st">&quot;family&quot;</span>,</span>
<span id="cb330-5"><a href="differential-abundance.html#cb330-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">fix_formula =</span> <span class="st">&quot;Geographical_location&quot;</span>, </span>
<span id="cb330-6"><a href="differential-abundance.html#cb330-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">p_adj_method =</span> <span class="st">&quot;fdr&quot;</span>, </span>
<span id="cb330-7"><a href="differential-abundance.html#cb330-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">prv_cut =</span> <span class="dv">0</span>, <span class="co"># prev filtering has been done above already</span></span>
<span id="cb330-8"><a href="differential-abundance.html#cb330-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">lib_cut =</span> <span class="dv">0</span>, </span>
<span id="cb330-9"><a href="differential-abundance.html#cb330-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">group =</span> <span class="st">&quot;Geographical_location&quot;</span>, </span>
<span id="cb330-10"><a href="differential-abundance.html#cb330-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">struc_zero =</span> <span class="cn">TRUE</span>, </span>
<span id="cb330-11"><a href="differential-abundance.html#cb330-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">neg_lb =</span> <span class="cn">TRUE</span>,</span>
<span id="cb330-12"><a href="differential-abundance.html#cb330-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">iter_control =</span> <span class="fu">list</span>(<span class="at">tol =</span> <span class="fl">1e-5</span>, <span class="at">max_iter =</span> <span class="dv">20</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>),</span>
<span id="cb330-13"><a href="differential-abundance.html#cb330-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">em_control =</span> <span class="fu">list</span>(<span class="at">tol =</span> <span class="fl">1e-5</span>, <span class="at">max_iter =</span> <span class="dv">20</span>),  </span>
<span id="cb330-14"><a href="differential-abundance.html#cb330-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">alpha =</span> <span class="fl">0.05</span>, </span>
<span id="cb330-15"><a href="differential-abundance.html#cb330-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">global =</span> <span class="cn">TRUE</span> <span class="co"># multi group comparison will be deactivated automatically </span></span>
<span id="cb330-16"><a href="differential-abundance.html#cb330-16" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb330-17"><a href="differential-abundance.html#cb330-17" aria-hidden="true" tabindex="-1"></a><span class="co"># store the results in res </span></span>
<span id="cb330-18"><a href="differential-abundance.html#cb330-18" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> out<span class="sc">$</span>res</span></code></pre></div>
<p>The object <code>out</code> contains all model output. Again, see the
<a href="https://rdrr.io/github/FrederickHuangLin/ANCOMBC/man/ancombc.html">documentation of the function</a>
under <strong>Value</strong> for an explanation of all the output objects. Our question
whether taxa are differentially abundant can be answered by looking at the
<code>res</code> object, which now contains dataframes with the coefficients,
standard errors, p-values and q-values. Conveniently, there is a dataframe
<code>diff_abn</code>. Here, for each taxon it is indicated whether it is differentially
abundant between the groups (again, keep in mind that the answer is not
black-white). Below we show the first 6 entries of this dataframe:</p>
<div class="sourceCode" id="cb331"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb331-1"><a href="differential-abundance.html#cb331-1" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(res))</span></code></pre></div>
</div>
<div id="maaslin2" class="section level3 hasAnchor" number="11.1.5">
<h3><span class="header-section-number">11.1.5</span> MaAsLin2<a href="differential-abundance.html#maaslin2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Next, we will illustrate how to use MaAsLin2, which is the next generation of
MaAsLin. As it is based on generalized linear models, it is flexible for different study designs and covariate
structures. The official package tutorial can be found <a href="https://github.com/biobakery/biobakery/wiki/maaslin2">here</a>.</p>
<div class="sourceCode" id="cb332"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb332-1"><a href="differential-abundance.html#cb332-1" aria-hidden="true" tabindex="-1"></a><span class="co"># maaslin expects features as columns and samples as rows </span></span>
<span id="cb332-2"><a href="differential-abundance.html#cb332-2" aria-hidden="true" tabindex="-1"></a><span class="co"># for both the asv/otu table as well as meta data </span></span>
<span id="cb332-3"><a href="differential-abundance.html#cb332-3" aria-hidden="true" tabindex="-1"></a>asv <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">assay</span>(tse))</span>
<span id="cb332-4"><a href="differential-abundance.html#cb332-4" aria-hidden="true" tabindex="-1"></a>meta_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">colData</span>(tse))</span>
<span id="cb332-5"><a href="differential-abundance.html#cb332-5" aria-hidden="true" tabindex="-1"></a><span class="co"># you can specifiy different GLMs/normalizations/transforms. We used similar</span></span>
<span id="cb332-6"><a href="differential-abundance.html#cb332-6" aria-hidden="true" tabindex="-1"></a><span class="co"># settings as in Nearing et al. (2021) here:</span></span>
<span id="cb332-7"><a href="differential-abundance.html#cb332-7" aria-hidden="true" tabindex="-1"></a>fit_data <span class="ot">&lt;-</span> <span class="fu">Maaslin2</span>(</span>
<span id="cb332-8"><a href="differential-abundance.html#cb332-8" aria-hidden="true" tabindex="-1"></a>  asv,</span>
<span id="cb332-9"><a href="differential-abundance.html#cb332-9" aria-hidden="true" tabindex="-1"></a>  meta_data,</span>
<span id="cb332-10"><a href="differential-abundance.html#cb332-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">output =</span> <span class="st">&quot;DAA example&quot;</span>,</span>
<span id="cb332-11"><a href="differential-abundance.html#cb332-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">transform =</span> <span class="st">&quot;AST&quot;</span>,</span>
<span id="cb332-12"><a href="differential-abundance.html#cb332-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">fixed_effects =</span> <span class="st">&quot;Geographical_location&quot;</span>,</span>
<span id="cb332-13"><a href="differential-abundance.html#cb332-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># random_effects = c(...), # you can also fit MLM by specifying random effects</span></span>
<span id="cb332-14"><a href="differential-abundance.html#cb332-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># specifying a ref is especially important if you have more than 2 levels</span></span>
<span id="cb332-15"><a href="differential-abundance.html#cb332-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">reference =</span> <span class="st">&quot;Geographical_location,Pune&quot;</span>,  </span>
<span id="cb332-16"><a href="differential-abundance.html#cb332-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">normalization =</span> <span class="st">&quot;TSS&quot;</span>,</span>
<span id="cb332-17"><a href="differential-abundance.html#cb332-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">standardize =</span> <span class="cn">FALSE</span>,</span>
<span id="cb332-18"><a href="differential-abundance.html#cb332-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">min_prevalence =</span> <span class="dv">0</span> <span class="co"># prev filterin already done</span></span>
<span id="cb332-19"><a href="differential-abundance.html#cb332-19" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p>Which genera are identified as differentially abundant? (leave out “head” to see all).</p>
<div class="sourceCode" id="cb333"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb333-1"><a href="differential-abundance.html#cb333-1" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(<span class="fu">filter</span>(fit_data<span class="sc">$</span>results, qval <span class="sc">&lt;=</span> <span class="fl">0.05</span>)))</span></code></pre></div>
<p>A folder will be created that is called like the above specified
output. It contains also figures to visualize the difference between
genera for the significant ones.</p>
</div>
<div id="linda" class="section level3 hasAnchor" number="11.1.6">
<h3><span class="header-section-number">11.1.6</span> LinDA<a href="differential-abundance.html#linda" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Lastly, we cover linear models for differential abundance analysis of
microbiome compositional data (<span class="citation">Zhou et al. (<a href="#ref-Zhou2022" role="doc-biblioref">2022</a>)</span>). This tool is very similar
to ANCOMBC with few differences: 1) LinDA correct for the
compositional bias differently using the mode of all regression
coefficients. 2) The authors claim that it runs 100-1000x faster than
ANCOMBC and 3) it support hierarchical models. The latter could be
ignored as ANCOMBC will be supporting hierarchical models with the
next release. Nevertheless, LinDA seems a promising tool that achieves
the best power/fdr trade-off together with ANCOMBC according to the
authors. The speed might make it the choice for bigger datasets or
datasets with a very high number of features.</p>
<div class="sourceCode" id="cb334"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb334-1"><a href="differential-abundance.html#cb334-1" aria-hidden="true" tabindex="-1"></a>otu.tab <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">assay</span>(tse))</span>
<span id="cb334-2"><a href="differential-abundance.html#cb334-2" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(tse)) <span class="sc">%&gt;%</span> <span class="fu">select</span>(Geographical_location)</span>
<span id="cb334-3"><a href="differential-abundance.html#cb334-3" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">linda</span>(</span>
<span id="cb334-4"><a href="differential-abundance.html#cb334-4" aria-hidden="true" tabindex="-1"></a>  otu.tab, </span>
<span id="cb334-5"><a href="differential-abundance.html#cb334-5" aria-hidden="true" tabindex="-1"></a>  meta, </span>
<span id="cb334-6"><a href="differential-abundance.html#cb334-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">formula =</span> <span class="st">&#39;~Geographical_location&#39;</span>, </span>
<span id="cb334-7"><a href="differential-abundance.html#cb334-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">alpha =</span> <span class="fl">0.05</span>, </span>
<span id="cb334-8"><a href="differential-abundance.html#cb334-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">prev.filter =</span> <span class="dv">0</span>, </span>
<span id="cb334-9"><a href="differential-abundance.html#cb334-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">mean.abund.filter =</span> <span class="dv">0</span>)</span>
<span id="cb334-10"><a href="differential-abundance.html#cb334-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb334-11"><a href="differential-abundance.html#cb334-11" aria-hidden="true" tabindex="-1"></a><span class="co"># to scan the table for genera where H0 could be rejected:</span></span>
<span id="cb334-12"><a href="differential-abundance.html#cb334-12" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(<span class="fu">filter</span>(<span class="fu">as.data.frame</span>(res<span class="sc">$</span>output), Geographical_locationPune.reject)))</span></code></pre></div>
</div>
<div id="comparison-of-the-methods" class="section level3 hasAnchor" number="11.1.7">
<h3><span class="header-section-number">11.1.7</span> Comparison of the methods<a href="differential-abundance.html#comparison-of-the-methods" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>When we compare the methods in the context of a research question, we could
look at e.g. at whether they agree based on the applied decision criterion
(e.g. adjusted p value &lt; 0.05). That is what we illustrate here. First we will
look at how many taxa were identified by each method to begin with. In the next
step we will look at the intersection of identified taxa. To achieve that, we
first create a dataframe that summarises the decision criterion for each method
and shows a score from 0 to 3 indicating how many methods agreed on a particular
taxon.</p>
<div class="sourceCode" id="cb335"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb335-1"><a href="differential-abundance.html#cb335-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Rename &quot;taxon&quot; column from ancombc results  so that it match with others</span></span>
<span id="cb335-2"><a href="differential-abundance.html#cb335-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(out<span class="sc">$</span>res)[<span class="fu">colnames</span>(out<span class="sc">$</span>res) <span class="sc">==</span> <span class="st">&quot;taxon&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;genus&quot;</span></span>
<span id="cb335-3"><a href="differential-abundance.html#cb335-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb335-4"><a href="differential-abundance.html#cb335-4" aria-hidden="true" tabindex="-1"></a>summ <span class="ot">&lt;-</span> <span class="fu">full_join</span>(</span>
<span id="cb335-5"><a href="differential-abundance.html#cb335-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames_to_column</span>(aldex_out, <span class="st">&quot;genus&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb335-6"><a href="differential-abundance.html#cb335-6" aria-hidden="true" tabindex="-1"></a>      <span class="fu">select</span>(genus, <span class="at">aldex2 =</span> wi.eBH),</span>
<span id="cb335-7"><a href="differential-abundance.html#cb335-7" aria-hidden="true" tabindex="-1"></a>    out<span class="sc">$</span>res <span class="sc">%&gt;%</span></span>
<span id="cb335-8"><a href="differential-abundance.html#cb335-8" aria-hidden="true" tabindex="-1"></a>      <span class="fu">select</span>(genus, <span class="at">ancombc =</span> diff_Geographical_locationPune),</span>
<span id="cb335-9"><a href="differential-abundance.html#cb335-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">by =</span> <span class="st">&quot;genus&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb335-10"><a href="differential-abundance.html#cb335-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">full_join</span>(</span>
<span id="cb335-11"><a href="differential-abundance.html#cb335-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(fit_data<span class="sc">$</span>results, <span class="at">genus =</span> feature, <span class="at">maaslin2 =</span> qval), </span>
<span id="cb335-12"><a href="differential-abundance.html#cb335-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">by =</span> <span class="st">&quot;genus&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb335-13"><a href="differential-abundance.html#cb335-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">full_join</span>(</span>
<span id="cb335-14"><a href="differential-abundance.html#cb335-14" aria-hidden="true" tabindex="-1"></a>      <span class="fu">rownames_to_column</span>(<span class="fu">as.data.frame</span>(res<span class="sc">$</span>output), <span class="st">&quot;genus&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb335-15"><a href="differential-abundance.html#cb335-15" aria-hidden="true" tabindex="-1"></a>        <span class="fu">select</span>(genus, <span class="at">LinDA =</span> Geographical_locationPune.reject), </span>
<span id="cb335-16"><a href="differential-abundance.html#cb335-16" aria-hidden="true" tabindex="-1"></a>      <span class="at">by =</span> <span class="st">&quot;genus&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb335-17"><a href="differential-abundance.html#cb335-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb335-18"><a href="differential-abundance.html#cb335-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">across</span>(<span class="fu">c</span>(aldex2, maaslin2), <span class="sc">~</span> .x <span class="sc">&lt;=</span> <span class="fl">0.05</span>),</span>
<span id="cb335-19"><a href="differential-abundance.html#cb335-19" aria-hidden="true" tabindex="-1"></a>    <span class="co"># the following line would be necessary without prevalence filtering </span></span>
<span id="cb335-20"><a href="differential-abundance.html#cb335-20" aria-hidden="true" tabindex="-1"></a>    <span class="co"># as some methods output NA</span></span>
<span id="cb335-21"><a href="differential-abundance.html#cb335-21" aria-hidden="true" tabindex="-1"></a>    <span class="co">#across(-genus, function(x) ifelse(is.na(x), FALSE, x)),</span></span>
<span id="cb335-22"><a href="differential-abundance.html#cb335-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">score =</span> <span class="fu">rowSums</span>(<span class="fu">across</span>(<span class="fu">c</span>(aldex2, ancombc, maaslin2, LinDA)))</span>
<span id="cb335-23"><a href="differential-abundance.html#cb335-23" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb335-24"><a href="differential-abundance.html#cb335-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb335-25"><a href="differential-abundance.html#cb335-25" aria-hidden="true" tabindex="-1"></a><span class="co"># This is how it looks like:</span></span>
<span id="cb335-26"><a href="differential-abundance.html#cb335-26" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(summ))</span></code></pre></div>
<p>Now we can answer our questions:</p>
<div class="sourceCode" id="cb336"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb336-1"><a href="differential-abundance.html#cb336-1" aria-hidden="true" tabindex="-1"></a><span class="co"># how many genera were identified by each method?</span></span>
<span id="cb336-2"><a href="differential-abundance.html#cb336-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summarise</span>(summ, <span class="fu">across</span>(<span class="fu">where</span>(is.logical), sum)) <span class="sc">%&gt;%</span></span>
<span id="cb336-3"><a href="differential-abundance.html#cb336-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>()</span>
<span id="cb336-4"><a href="differential-abundance.html#cb336-4" aria-hidden="true" tabindex="-1"></a><span class="co"># which genera are identified by all methods?</span></span>
<span id="cb336-5"><a href="differential-abundance.html#cb336-5" aria-hidden="true" tabindex="-1"></a><span class="fu">filter</span>(summ, score <span class="sc">==</span> <span class="dv">4</span>) <span class="sc">%&gt;%</span> <span class="fu">kable</span>()</span></code></pre></div>
<p>We see that each method identified at least 9 genera as differentially
abundant. Eight of those that were identified by ALDEx2,
were also identified by the other methods. We could plot the data for
any method or for those taxa that were identified by all methods:</p>
<div class="sourceCode" id="cb337"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb337-1"><a href="differential-abundance.html#cb337-1" aria-hidden="true" tabindex="-1"></a>plot_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(<span class="fu">assay</span>(tse)))</span>
<span id="cb337-2"><a href="differential-abundance.html#cb337-2" aria-hidden="true" tabindex="-1"></a>plot_data<span class="sc">$</span>Geographical_location <span class="ot">&lt;-</span> tse<span class="sc">$</span>Geographical_location</span>
<span id="cb337-3"><a href="differential-abundance.html#cb337-3" aria-hidden="true" tabindex="-1"></a><span class="co"># create a plot for each genus where the score is indicated in the title</span></span>
<span id="cb337-4"><a href="differential-abundance.html#cb337-4" aria-hidden="true" tabindex="-1"></a>plots <span class="ot">&lt;-</span> <span class="fu">pmap</span>(<span class="fu">select</span>(summ, genus, score), <span class="cf">function</span>(genus, score) {</span>
<span id="cb337-5"><a href="differential-abundance.html#cb337-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(plot_data, <span class="fu">aes</span>(<span class="st">&quot;Geographical_location&quot;</span>, genus)) <span class="sc">+</span></span>
<span id="cb337-6"><a href="differential-abundance.html#cb337-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">fill =</span> Geographical_location), <span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb337-7"><a href="differential-abundance.html#cb337-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.2</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb337-8"><a href="differential-abundance.html#cb337-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">&quot;Score {score}&quot;</span>)) <span class="sc">+</span></span>
<span id="cb337-9"><a href="differential-abundance.html#cb337-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb337-10"><a href="differential-abundance.html#cb337-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>)</span>
<span id="cb337-11"><a href="differential-abundance.html#cb337-11" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb337-12"><a href="differential-abundance.html#cb337-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb337-13"><a href="differential-abundance.html#cb337-13" aria-hidden="true" tabindex="-1"></a><span class="co"># now we can show only those genera that have at least score 3 (or 2 or 1)</span></span>
<span id="cb337-14"><a href="differential-abundance.html#cb337-14" aria-hidden="true" tabindex="-1"></a>robust_plots <span class="ot">&lt;-</span> plots[summ<span class="sc">$</span>score <span class="sc">==</span> <span class="dv">4</span> <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(summ<span class="sc">$</span>score)] </span>
<span id="cb337-15"><a href="differential-abundance.html#cb337-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb337-16"><a href="differential-abundance.html#cb337-16" aria-hidden="true" tabindex="-1"></a><span class="co"># to display this nicely in the book we use patchwork here:</span></span>
<span id="cb337-17"><a href="differential-abundance.html#cb337-17" aria-hidden="true" tabindex="-1"></a><span class="co"># (we show first ones)</span></span>
<span id="cb337-18"><a href="differential-abundance.html#cb337-18" aria-hidden="true" tabindex="-1"></a>robust_plots[[<span class="dv">1</span>]] <span class="sc">+</span> </span>
<span id="cb337-19"><a href="differential-abundance.html#cb337-19" aria-hidden="true" tabindex="-1"></a>  robust_plots[[<span class="dv">2</span>]] <span class="sc">+</span> </span>
<span id="cb337-20"><a href="differential-abundance.html#cb337-20" aria-hidden="true" tabindex="-1"></a>  robust_plots[[<span class="dv">3</span>]] <span class="sc">+</span> </span>
<span id="cb337-21"><a href="differential-abundance.html#cb337-21" aria-hidden="true" tabindex="-1"></a>  robust_plots[[<span class="dv">4</span>]] <span class="sc">+</span></span>
<span id="cb337-22"><a href="differential-abundance.html#cb337-22" aria-hidden="true" tabindex="-1"></a>  robust_plots[[<span class="dv">5</span>]] <span class="sc">+</span></span>
<span id="cb337-23"><a href="differential-abundance.html#cb337-23" aria-hidden="true" tabindex="-1"></a>  robust_plots[[<span class="dv">6</span>]] <span class="sc">+</span></span>
<span id="cb337-24"><a href="differential-abundance.html#cb337-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_layout</span>(<span class="at">nrow =</span> <span class="dv">2</span>)</span>
<span id="cb337-25"><a href="differential-abundance.html#cb337-25" aria-hidden="true" tabindex="-1"></a><span class="co"># or if we have most trust in any specific method we can show genera that </span></span>
<span id="cb337-26"><a href="differential-abundance.html#cb337-26" aria-hidden="true" tabindex="-1"></a><span class="co"># are differentially abundant according to that method and then look in the</span></span>
<span id="cb337-27"><a href="differential-abundance.html#cb337-27" aria-hidden="true" tabindex="-1"></a><span class="co"># title how many methods also identified it (we only show first 6 here):</span></span>
<span id="cb337-28"><a href="differential-abundance.html#cb337-28" aria-hidden="true" tabindex="-1"></a>ancombc_plots <span class="ot">&lt;-</span> plots[summ<span class="sc">$</span>ancombc <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(summ<span class="sc">$</span>score)] </span>
<span id="cb337-29"><a href="differential-abundance.html#cb337-29" aria-hidden="true" tabindex="-1"></a>ancombc_plots[[<span class="dv">1</span>]] <span class="sc">+</span> </span>
<span id="cb337-30"><a href="differential-abundance.html#cb337-30" aria-hidden="true" tabindex="-1"></a>  ancombc_plots[[<span class="dv">2</span>]] <span class="sc">+</span> </span>
<span id="cb337-31"><a href="differential-abundance.html#cb337-31" aria-hidden="true" tabindex="-1"></a>  ancombc_plots[[<span class="dv">3</span>]] <span class="sc">+</span> </span>
<span id="cb337-32"><a href="differential-abundance.html#cb337-32" aria-hidden="true" tabindex="-1"></a>  ancombc_plots[[<span class="dv">4</span>]] <span class="sc">+</span></span>
<span id="cb337-33"><a href="differential-abundance.html#cb337-33" aria-hidden="true" tabindex="-1"></a>  ancombc_plots[[<span class="dv">5</span>]] <span class="sc">+</span></span>
<span id="cb337-34"><a href="differential-abundance.html#cb337-34" aria-hidden="true" tabindex="-1"></a>  ancombc_plots[[<span class="dv">6</span>]] </span></code></pre></div>
</div>
<div id="confounding-variables" class="section level3 hasAnchor" number="11.1.8">
<h3><span class="header-section-number">11.1.8</span> Confounding variables<a href="differential-abundance.html#confounding-variables" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To perform causal inference, it is crucial that the method is able to include
covariates in the model. This is not possible with e.g. the Wilcoxon test.
Other methods such as both ANCOM methods, ALDEx2, LinDA, MaAsLin2 and others
allow this. Below we show how to include a covariate in ANCOM-BC.
It is very similar for all the methods that allow this. Since in this dataset
there are no covariates, I first simulate a new variable and add it to the TSE
object.</p>
<div class="sourceCode" id="cb338"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb338-1"><a href="differential-abundance.html#cb338-1" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">FIXME</span><span class="co">: switch to a faster example / method</span></span>
<span id="cb338-2"><a href="differential-abundance.html#cb338-2" aria-hidden="true" tabindex="-1"></a>out_cov <span class="ot">=</span> <span class="fu">ancombc2</span>(</span>
<span id="cb338-3"><a href="differential-abundance.html#cb338-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> tse, </span>
<span id="cb338-4"><a href="differential-abundance.html#cb338-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">fix_formula =</span> <span class="st">&quot;Geographical_location + Age&quot;</span>, <span class="co"># here we add Age to the model</span></span>
<span id="cb338-5"><a href="differential-abundance.html#cb338-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">p_adj_method =</span> <span class="st">&quot;fdr&quot;</span>, </span>
<span id="cb338-6"><a href="differential-abundance.html#cb338-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">prv_cut =</span> <span class="dv">0</span>,  <span class="co"># we did that already</span></span>
<span id="cb338-7"><a href="differential-abundance.html#cb338-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">lib_cut =</span> <span class="dv">0</span>, </span>
<span id="cb338-8"><a href="differential-abundance.html#cb338-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">group =</span> <span class="st">&quot;Geographical_location&quot;</span>,</span>
<span id="cb338-9"><a href="differential-abundance.html#cb338-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">struc_zero =</span> <span class="cn">TRUE</span>, </span>
<span id="cb338-10"><a href="differential-abundance.html#cb338-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">neg_lb =</span> <span class="cn">TRUE</span>, </span>
<span id="cb338-11"><a href="differential-abundance.html#cb338-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">iter_control =</span> <span class="fu">list</span>(<span class="at">tol =</span> <span class="fl">1e-5</span>, <span class="at">max_iter =</span> <span class="dv">20</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>),</span>
<span id="cb338-12"><a href="differential-abundance.html#cb338-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">em_control =</span> <span class="fu">list</span>(<span class="at">tol =</span> <span class="fl">1e-5</span>, <span class="at">max_iter =</span> <span class="dv">20</span>),</span>
<span id="cb338-13"><a href="differential-abundance.html#cb338-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">alpha =</span> <span class="fl">0.05</span>, </span>
<span id="cb338-14"><a href="differential-abundance.html#cb338-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">global =</span> <span class="cn">TRUE</span> <span class="co"># multi group comparison will be deactivated automatically </span></span>
<span id="cb338-15"><a href="differential-abundance.html#cb338-15" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb338-16"><a href="differential-abundance.html#cb338-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb338-17"><a href="differential-abundance.html#cb338-17" aria-hidden="true" tabindex="-1"></a><span class="co"># now the model answers the question: holding Age constant, are </span></span>
<span id="cb338-18"><a href="differential-abundance.html#cb338-18" aria-hidden="true" tabindex="-1"></a><span class="co"># bacterial taxa differentially abundant? Or, if that is of interest,</span></span>
<span id="cb338-19"><a href="differential-abundance.html#cb338-19" aria-hidden="true" tabindex="-1"></a><span class="co"># holding phenotype constant, is Age associated with bacterial abundance?</span></span>
<span id="cb338-20"><a href="differential-abundance.html#cb338-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Again we only show the first 6 entries.</span></span>
<span id="cb338-21"><a href="differential-abundance.html#cb338-21" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(out_cov<span class="sc">$</span>res))</span></code></pre></div>
<p>In the next section of this book chapter we cover methods that can also take
into account the phylogenetic information of bacterial taxa to perform
group-wise associations.</p>
</div>
</div>
<div id="tree-based-methods" class="section level2 hasAnchor" number="11.2">
<h2><span class="header-section-number">11.2</span> Tree-based methods<a href="differential-abundance.html#tree-based-methods" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="group-wise-associations-testing-based-on-balances" class="section level3 hasAnchor" number="11.2.1">
<h3><span class="header-section-number">11.2.1</span> Group-wise associations testing based on balances<a href="differential-abundance.html#group-wise-associations-testing-based-on-balances" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For testing associations based on balances, check the philr R/Bioconductor package.</p>
</div>
</div>
<div id="session-info-7" class="section level2 unnumbered hasAnchor">
<h2>Session Info<a href="differential-abundance.html#session-info-7" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<button class="rebook-collapse">
View session info
</button>
<div class="rebook-content">
<pre><code>R version 4.2.1 (2022-06-23)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.4 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ANCOMBC_2.1.2                  lubridate_1.9.2               
 [3] forcats_1.0.0                  stringr_1.5.0                 
 [5] dplyr_1.1.0                    purrr_1.0.1                   
 [7] readr_2.1.4                    tidyr_1.3.0                   
 [9] tibble_3.1.8                   ggplot2_3.4.1                 
[11] tidyverse_2.0.0                knitr_1.42                    
[13] MicrobiomeStat_1.1             Maaslin2_1.10.0               
[15] ALDEx2_1.28.1                  zCompositions_1.4.0-1         
[17] truncnorm_1.0-8                NADA_1.6-1.1                  
[19] survival_3.5-3                 MASS_7.3-58.2                 
[21] tidySummarizedExperiment_1.6.1 patchwork_1.1.2               
[23] mia_1.7.7                      MultiAssayExperiment_1.24.0   
[25] TreeSummarizedExperiment_2.1.4 Biostrings_2.66.0             
[27] XVector_0.38.0                 SingleCellExperiment_1.20.0   
[29] SummarizedExperiment_1.28.0    Biobase_2.58.0                
[31] GenomicRanges_1.50.2           GenomeInfoDb_1.34.9           
[33] IRanges_2.32.0                 S4Vectors_0.36.1              
[35] BiocGenerics_0.44.0            MatrixGenerics_1.10.0         
[37] matrixStats_0.63.0-9003        BiocStyle_2.24.0              
[39] rebook_1.6.0                  

loaded via a namespace (and not attached):
  [1] estimability_1.4.1          coda_0.19-4                
  [3] bit64_4.0.5                 multcomp_1.4-22            
  [5] irlba_2.3.5.1               DelayedArray_0.24.0        
  [7] data.table_1.14.8           rpart_4.1.19               
  [9] doParallel_1.0.17           RCurl_1.98-1.10            
 [11] generics_0.1.3              ScaledMatrix_1.6.0         
 [13] TH.data_1.1-1               timeSeries_4021.105        
 [15] RSQLite_2.3.0               proxy_0.4-27               
 [17] bit_4.0.5                   tzdb_0.3.0                 
 [19] DirichletMultinomial_1.40.0 viridis_0.6.2              
 [21] xfun_0.37                   fBasics_4021.93            
 [23] hms_1.1.2                   jquerylib_0.1.4            
 [25] evaluate_0.20               DEoptimR_1.0-11            
 [27] fansi_1.0.4                 readxl_1.4.2               
 [29] igraph_1.4.1                DBI_1.1.3                  
 [31] htmlwidgets_1.6.1           Rmpfr_0.9-1                
 [33] CVXR_1.0-11                 ellipsis_0.3.2             
 [35] energy_1.7-11               backports_1.4.1            
 [37] bookdown_0.32               permute_0.9-7              
 [39] deldir_1.0-6                sparseMatrixStats_1.10.0   
 [41] vctrs_0.5.2                 cachem_1.0.7               
 [43] withr_2.5.0                 robustbase_0.95-0          
 [45] emmeans_1.8.4-1             checkmate_2.1.0            
 [47] vegan_2.6-4                 treeio_1.22.0              
 [49] getopt_1.20.3               cluster_2.1.4              
 [51] gsl_2.1-8                   ape_5.7                    
 [53] dir.expiry_1.4.0            lazyeval_0.2.2             
 [55] crayon_1.5.2                pkgconfig_2.0.3            
 [57] nlme_3.1-162                vipor_0.4.5                
 [59] nnet_7.3-18                 rlang_1.0.6                
 [61] spatial_7.3-16              lifecycle_1.0.3            
 [63] sandwich_3.0-2              filelock_1.0.2             
 [65] phyloseq_1.40.0             rsvd_1.0.5                 
 [67] cellranger_1.1.0            rngtools_1.5.2             
 [69] graph_1.74.0                Matrix_1.5-3               
 [71] lpsymphony_1.24.0           zoo_1.8-11                 
 [73] Rhdf5lib_1.18.2             boot_1.3-28.1              
 [75] base64enc_0.1-3             beeswarm_0.4.0             
 [77] png_0.1-8                   viridisLite_0.4.1          
 [79] stabledist_0.7-1            rootSolve_1.8.2.3          
 [81] bitops_1.0-7                rhdf5filters_1.8.0         
 [83] blob_1.2.3                  DelayedMatrixStats_1.20.0  
 [85] doRNG_1.8.6                 decontam_1.18.0            
 [87] jpeg_0.1-10                 DECIPHER_2.26.0            
 [89] beachmat_2.14.0             scales_1.2.1               
 [91] memoise_2.0.1               magrittr_2.0.3             
 [93] plyr_1.8.8                  zlibbioc_1.44.0            
 [95] compiler_4.2.1              RColorBrewer_1.1-3         
 [97] clue_0.3-64                 lme4_1.1-31                
 [99] cli_3.6.0                   ade4_1.7-22                
[101] lmerTest_3.1-3              htmlTable_2.4.1            
[103] Formula_1.2-5               mgcv_1.8-41                
[105] tidyselect_1.2.0            stringi_1.7.12             
[107] yaml_2.3.7                  BiocSingular_1.14.0        
[109] latticeExtra_0.6-30         ggrepel_0.9.3              
[111] grid_4.2.1                  sass_0.4.5                 
[113] tools_4.2.1                 lmom_2.9                   
[115] timechange_0.2.0            parallel_4.2.1             
[117] rstudioapi_0.14             foreign_0.8-84             
[119] foreach_1.5.2               statip_0.2.3               
[121] optparse_1.7.3              gridExtra_2.3              
[123] gld_2.6.6                   stable_1.1.6               
[125] RcppZiggurat_0.1.6          digest_0.6.31              
[127] BiocManager_1.30.20         Rcpp_1.0.10                
[129] scuttle_1.8.4               httr_1.4.5                 
[131] Rdpack_2.4                  colorspace_2.1-0           
[133] XML_3.99-0.13               modeest_2.4.0              
[135] splines_4.2.1               yulab.utils_0.0.6          
[137] rmutil_1.1.10               statmod_1.5.0              
[139] tidytree_0.4.2              expm_0.999-7               
[141] scater_1.26.1               multtest_2.52.0            
[143] Exact_3.2                   plotly_4.10.1              
[145] xtable_1.8-4                gmp_0.7-1                  
[147] jsonlite_1.8.4              nloptr_2.0.3               
[149] CodeDepends_0.6.5           timeDate_4022.108          
[151] Rfast_2.0.7                 R6_2.5.1                   
[153] Hmisc_4.8-0                 pillar_1.8.1               
[155] htmltools_0.5.4             glue_1.6.2                 
[157] fastmap_1.1.1               minqa_1.2.5                
[159] BiocParallel_1.32.5         BiocNeighbors_1.16.0       
[161] class_7.3-21                codetools_0.2-19           
[163] pcaPP_2.0-3                 mvtnorm_1.1-3              
[165] utf8_1.2.3                  lattice_0.20-45            
[167] bslib_0.4.2                 numDeriv_2016.8-1.1        
[169] ggbeeswarm_0.7.1            DescTools_0.99.48          
[171] interp_1.1-3                biglm_0.9-2.1              
[173] rmarkdown_2.20              biomformat_1.24.0          
[175] munsell_0.5.0               e1071_1.7-13               
[177] rhdf5_2.40.0                GenomeInfoDbData_1.2.9     
[179] iterators_1.0.14            reshape2_1.4.4             
[181] gtable_0.3.1                rbibutils_2.2.13           </code></pre>
</div>

</div>
</div>
<h3>Bibliography<a href="bibliography.html#bibliography" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Chen2016" class="csl-entry">
Chen, Yunshun, Aaron A T Lun, and Gordon K Smyth. 2016. <span>“From Reads to Genes to Pathways: Differential Expression Analysis of RNA-Seq Experiments Using Rsubread and the edgeR Quasi-Likelihood Pipeline.”</span> <em>F1000Research</em> 5: 1438. <a href="https://doi.org/10.12688/f1000research.8987.2">https://doi.org/10.12688/f1000research.8987.2</a>.
</div>
<div id="ref-Gloor2016" class="csl-entry">
Gloor, Gregory B., Jean M. Macklaim, and Andrew D. Fernandes. 2016. <span>“Displaying <span>Variation</span> in <span>Large Datasets</span>: <span>Plotting</span> a <span>Visual Summary</span> of <span>Effect Sizes</span>.”</span> <em>Journal of Computational and Graphical Statistics</em> 25 (3): 971–79. <a href="https://doi.org/10.1080/10618600.2015.1131161">https://doi.org/10.1080/10618600.2015.1131161</a>.
</div>
<div id="ref-Khlebrodova2021" class="csl-entry">
Khleborodova, Asya. 2021. <em>Lefser: R Implementation of the LEfSE Method for Microbiome Biomarker Discovery</em>. <a href="https://github.com/waldronlab/lefser">https://github.com/waldronlab/lefser</a>.
</div>
<div id="ref-ancombc2020" class="csl-entry">
Lin, Huang, and Shyamal Das Peddada. 2020a. <span>“Analysis of Compositions of Microbiomes with Bias Correction.”</span> <em>Nature Communications</em> 11 (1): 1–11. https://doi.org/<a href="https://doi.org/10.1038/s41467-020-17041-7">https://doi.org/10.1038/s41467-020-17041-7</a>.
</div>
<div id="ref-Das2020" class="csl-entry">
———. 2020b. <span>“Analysis of Compositions of Microbiomes with Bias Correction.”</span> <em>Nat Commun</em> 11 (1): 3514. <a href="https://doi.org/10.1038/s41467-020-17041-7">https://doi.org/10.1038/s41467-020-17041-7</a>.
</div>
<div id="ref-Love2014" class="csl-entry">
Love, Michael I., Wolfgang Huber, and Simon Anders. 2014. <span>“Moderated Estimation of Fold Change and Dispersion for RNA-Seq Data with DESeq2.”</span> <em>Genome Biology</em> 15: 550. <a href="https://doi.org/10.1186/s13059-014-0550-8">https://doi.org/10.1186/s13059-014-0550-8</a>.
</div>
<div id="ref-Mallick2020" class="csl-entry">
Mallick, Himel, Ali Rahnavard, and Lauren J. McIver. 2020. <em>MaAsLin 2: Multivariable Association in Population-Scale Meta-Omics Studies.</em> <a href="http://huttenhower.sph.harvard.edu/maaslin2">http://huttenhower.sph.harvard.edu/maaslin2</a>.
</div>
<div id="ref-Mandal2015" class="csl-entry">
Mandal, Siddhartha, Will Van Treuren, Richard A. White, Merete Eggesbø, Rob Knight, and Shyamal D. Peddada. 2015. <span>“Analysis of Composition of Microbiomes: A Novel Method for Studying Microbial Composition.”</span> <em>Microbial Ecology in Health &amp; Disease</em> 26 (0). <a href="https://doi.org/10.3402/mehd.v26.27663">https://doi.org/10.3402/mehd.v26.27663</a>.
</div>
<div id="ref-Martin2021" class="csl-entry">
Martin, Bryan D, Daniela Witten, and Amy D Willis. 2021. <em>Corncob: Count Regression for Correlated Observations with the Beta-Binomial</em>. <a href="https://CRAN.R-project.org/package=corncob">https://CRAN.R-project.org/package=corncob</a>.
</div>
<div id="ref-Nearing2022" class="csl-entry">
Nearing, Jacob T., Gavin M. Douglas, Molly G. Hayes, Jocelyn MacDonald, Dhwani K. Desai, Nicole Allward, Casey M. A. Jones, et al. 2022. <span>“Microbiome Differential Abundance Methods Produce Different Results Across 38 Datasets.”</span> <em>Nature Communications</em> 13 (1): 342. <a href="https://doi.org/10.1038/s41467-022-28034-z">https://doi.org/10.1038/s41467-022-28034-z</a>.
</div>
<div id="ref-Paulson2017" class="csl-entry">
Paulson, JN, H Talukder, and HC Bravo. 2017. <span>“Longitudinal Differential Abundance Analysis of Marker-Gene Surveys Using Smoothing Splines.”</span> <em>Biorxiv</em>. <a href="https://doi.org/10.1101/099457">https://doi.org/10.1101/099457</a>.
</div>
<div id="ref-Quinn2021" class="csl-entry">
Quinn, Thomas P., Elliott Gordon-Rodriguez, and Ionas Erb. 2021. <span>“A <span>Critique</span> of <span>Differential Abundance Analysis</span>, and <span>Advocacy</span> for an <span>Alternative</span>.”</span> <em>arXiv:2104.07266 [q-Bio, Stat]</em>, April. <a href="https://arxiv.org/abs/2104.07266">https://arxiv.org/abs/2104.07266</a>.
</div>
<div id="ref-Ritchie2015" class="csl-entry">
Ritchie, Matthew E, Belinda Phipson, Di Wu, Yifang Hu, Charity W Law, Wei Shi, and Gordon K Smyth. 2015. <span>“<span class="nocase">limma</span> Powers Differential Expression Analyses for <span>RNA</span>-Sequencing and Microarray Studies.”</span> <em>Nucleic Acids Research</em> 43 (7): e47. <a href="https://doi.org/10.1093/nar/gkv007">https://doi.org/10.1093/nar/gkv007</a>.
</div>
<div id="ref-Zhou2022" class="csl-entry">
Zhou, Huijuan, Kejun He, Jun Chen, and Xianyang Zhang. 2022. <span>“<span>LinDA</span>: Linear Models for Differential Abundance Analysis of Microbiome Compositional Data.”</span> <em>Genome Biology</em> 23 (1): 95. <a href="https://doi.org/10.1186/s13059-022-02655-5">https://doi.org/10.1186/s13059-022-02655-5</a>.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="clustering.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="machine_learning.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["OMA.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
