<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 11 Differential Abundance | Orchestrating Microbiome Analysis</title>
  <meta name="description" content="Chapter 11 Differential Abundance | Orchestrating Microbiome Analysis" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 11 Differential Abundance | Orchestrating Microbiome Analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="microbiome/OMA" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 11 Differential Abundance | Orchestrating Microbiome Analysis" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="clustering.html"/>
<link rel="next" href="machine_learning.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Orchestrating Microbiome Analysis with Bioconductor</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="part"><span><b>I Introduction</b></span></li>
<li class="chapter" data-level="1" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="packages.html"><a href="packages.html"><i class="fa fa-check"></i><b>2</b> Packages</a>
<ul>
<li class="chapter" data-level="2.1" data-path="packages.html"><a href="packages.html#package-installation"><i class="fa fa-check"></i><b>2.1</b> Package installation</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="packages.html"><a href="packages.html#packages_specific"><i class="fa fa-check"></i><b>2.1.1</b> Installing specific packages</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="packages.html"><a href="packages.html#ecosystem"><i class="fa fa-check"></i><b>2.2</b> Package ecosystem</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="packages.html"><a href="packages.html#mia-family-of-methods"><i class="fa fa-check"></i><b>2.2.1</b> mia family of methods</a></li>
<li class="chapter" data-level="2.2.2" data-path="packages.html"><a href="packages.html#sub-tree-methods"><i class="fa fa-check"></i><b>2.2.2</b> Tree-based methods</a></li>
<li class="chapter" data-level="2.2.3" data-path="packages.html"><a href="packages.html#sub-diff-abund"><i class="fa fa-check"></i><b>2.2.3</b> Differential abundance</a></li>
<li class="chapter" data-level="2.2.4" data-path="packages.html"><a href="packages.html#sub-manipulation"><i class="fa fa-check"></i><b>2.2.4</b> Manipulation</a></li>
<li class="chapter" data-level="2.2.5" data-path="packages.html"><a href="packages.html#further-options"><i class="fa fa-check"></i><b>2.2.5</b> Further options</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="containers.html"><a href="containers.html"><i class="fa fa-check"></i><b>3</b> Microbiome Data</a>
<ul>
<li class="chapter" data-level="3.1" data-path="containers.html"><a href="containers.html#data-science-framework"><i class="fa fa-check"></i><b>3.1</b> Data science framework</a></li>
<li class="chapter" data-level="3.2" data-path="containers.html"><a href="containers.html#data-containers"><i class="fa fa-check"></i><b>3.2</b> Data containers</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="containers.html"><a href="containers.html#assay-slot"><i class="fa fa-check"></i><b>3.2.1</b> Assay data</a></li>
<li class="chapter" data-level="3.2.2" data-path="containers.html"><a href="containers.html#coldata"><i class="fa fa-check"></i><b>3.2.2</b> colData</a></li>
<li class="chapter" data-level="3.2.3" data-path="containers.html"><a href="containers.html#rowdata"><i class="fa fa-check"></i><b>3.2.3</b> rowData</a></li>
<li class="chapter" data-level="3.2.4" data-path="containers.html"><a href="containers.html#rowtree"><i class="fa fa-check"></i><b>3.2.4</b> rowTree</a></li>
<li class="chapter" data-level="3.2.5" data-path="containers.html"><a href="containers.html#alt-exp"><i class="fa fa-check"></i><b>3.2.5</b> Alternative experiments</a></li>
<li class="chapter" data-level="3.2.6" data-path="containers.html"><a href="containers.html#mae"><i class="fa fa-check"></i><b>3.2.6</b> MultiAssayExperiments</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="containers.html"><a href="containers.html#example-data"><i class="fa fa-check"></i><b>3.3</b> Demonstration data</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="containers.html"><a href="containers.html#package-data"><i class="fa fa-check"></i><b>3.3.1</b> Package data</a></li>
<li class="chapter" data-level="3.3.2" data-path="containers.html"><a href="containers.html#experimenthub-data"><i class="fa fa-check"></i><b>3.3.2</b> ExperimentHub data</a></li>
<li class="chapter" data-level="3.3.3" data-path="containers.html"><a href="containers.html#curated-metagenomic-data"><i class="fa fa-check"></i><b>3.3.3</b> Curated metagenomic data</a></li>
<li class="chapter" data-level="3.3.4" data-path="containers.html"><a href="containers.html#other-data-sources"><i class="fa fa-check"></i><b>3.3.4</b> Other data sources</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="containers.html"><a href="containers.html#loading-experimental-microbiome-data"><i class="fa fa-check"></i><b>3.4</b> Loading experimental microbiome data</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="containers.html"><a href="containers.html#s-workflow"><i class="fa fa-check"></i><b>3.4.1</b> 16S workflow</a></li>
<li class="chapter" data-level="3.4.2" data-path="containers.html"><a href="containers.html#import-from-file"><i class="fa fa-check"></i><b>3.4.2</b> Import from external files</a></li>
<li class="chapter" data-level="3.4.3" data-path="containers.html"><a href="containers.html#constructing-treesummarizedexperiment"><i class="fa fa-check"></i><b>3.4.3</b> Constructing TreeSummarizedExperiment</a></li>
<li class="chapter" data-level="3.4.4" data-path="containers.html"><a href="containers.html#constructing-multiassayexperiment"><i class="fa fa-check"></i><b>3.4.4</b> Constructing MultiAssayExperiment</a></li>
<li class="chapter" data-level="3.4.5" data-path="containers.html"><a href="containers.html#import-functions-for-standard-formats"><i class="fa fa-check"></i><b>3.4.5</b> Import functions for standard formats</a></li>
<li class="chapter" data-level="3.4.6" data-path="containers.html"><a href="containers.html#conversions-between-data-formats-in-r"><i class="fa fa-check"></i><b>3.4.6</b> Conversions between data formats in R</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>II Focus Topics</b></span></li>
<li class="chapter" data-level="4" data-path="datamanipulation.html"><a href="datamanipulation.html"><i class="fa fa-check"></i><b>4</b> Data Manipulation</a>
<ul>
<li class="chapter" data-level="4.1" data-path="datamanipulation.html"><a href="datamanipulation.html#tidying-and-subsetting"><i class="fa fa-check"></i><b>4.1</b> Tidying and subsetting</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="datamanipulation.html"><a href="datamanipulation.html#tidy-data"><i class="fa fa-check"></i><b>4.1.1</b> Tidy data</a></li>
<li class="chapter" data-level="4.1.2" data-path="datamanipulation.html"><a href="datamanipulation.html#subsetting"><i class="fa fa-check"></i><b>4.1.2</b> Subsetting</a></li>
<li class="chapter" data-level="4.1.3" data-path="datamanipulation.html"><a href="datamanipulation.html#splitting"><i class="fa fa-check"></i><b>4.1.3</b> Splitting</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="datamanipulation.html"><a href="datamanipulation.html#add-or-modify-data"><i class="fa fa-check"></i><b>4.2</b> Add or modify data</a></li>
<li class="chapter" data-level="4.3" data-path="datamanipulation.html"><a href="datamanipulation.html#merge-data"><i class="fa fa-check"></i><b>4.3</b> Merge data</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="datamanipulation.html"><a href="datamanipulation.html#additional-functions"><i class="fa fa-check"></i><b>4.3.1</b> Additional functions</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>5</b> Exploration and Quality Control</a>
<ul>
<li class="chapter" data-level="5.1" data-path="quality-control.html"><a href="quality-control.html#abundance"><i class="fa fa-check"></i><b>5.1</b> Abundance</a></li>
<li class="chapter" data-level="5.2" data-path="quality-control.html"><a href="quality-control.html#prevalence"><i class="fa fa-check"></i><b>5.2</b> Prevalence</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="quality-control.html"><a href="quality-control.html#prevalence-analysis"><i class="fa fa-check"></i><b>5.2.1</b> Prevalence analysis</a></li>
<li class="chapter" data-level="5.2.2" data-path="quality-control.html"><a href="quality-control.html#rare-taxa"><i class="fa fa-check"></i><b>5.2.2</b> Rare taxa</a></li>
<li class="chapter" data-level="5.2.3" data-path="quality-control.html"><a href="quality-control.html#plotting-prevalence"><i class="fa fa-check"></i><b>5.2.3</b> Plotting prevalence</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="quality-control.html"><a href="quality-control.html#qc"><i class="fa fa-check"></i><b>5.3</b> Quality control</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="quality-control.html"><a href="quality-control.html#top-taxa"><i class="fa fa-check"></i><b>5.3.1</b> Top taxa</a></li>
<li class="chapter" data-level="5.3.2" data-path="quality-control.html"><a href="quality-control.html#library-size-read-count"><i class="fa fa-check"></i><b>5.3.2</b> Library size / read count</a></li>
<li class="chapter" data-level="5.3.3" data-path="quality-control.html"><a href="quality-control.html#contaminant-sequences"><i class="fa fa-check"></i><b>5.3.3</b> Contaminant sequences</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="taxonomic-information.html"><a href="taxonomic-information.html"><i class="fa fa-check"></i><b>6</b> Taxonomic Information</a>
<ul>
<li class="chapter" data-level="6.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#assigning-taxonomic-information."><i class="fa fa-check"></i><b>6.1</b> Assigning taxonomic information.</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#dada2"><i class="fa fa-check"></i><b>6.1.1</b> dada2</a></li>
<li class="chapter" data-level="6.1.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#decipher"><i class="fa fa-check"></i><b>6.1.2</b> DECIPHER</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#functions-to-access-taxonomic-information"><i class="fa fa-check"></i><b>6.2</b> Functions to access taxonomic information</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#fly-tree"><i class="fa fa-check"></i><b>6.2.1</b> Generate a taxonomic tree on the fly</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="taxonomic-information.html"><a href="taxonomic-information.html#data-agglomeration"><i class="fa fa-check"></i><b>6.3</b> Data agglomeration</a></li>
<li class="chapter" data-level="6.4" data-path="taxonomic-information.html"><a href="taxonomic-information.html#assay-transform"><i class="fa fa-check"></i><b>6.4</b> Data transformation</a></li>
<li class="chapter" data-level="6.5" data-path="taxonomic-information.html"><a href="taxonomic-information.html#pick-specific"><i class="fa fa-check"></i><b>6.5</b> Pick specific</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#abundances-of-all-taxa-in-specific-sample"><i class="fa fa-check"></i><b>6.5.1</b> Abundances of all taxa in specific sample</a></li>
<li class="chapter" data-level="6.5.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#abundances-of-specific-taxa-in-all-samples"><i class="fa fa-check"></i><b>6.5.2</b> Abundances of specific taxa in all samples</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="community-diversity.html"><a href="community-diversity.html"><i class="fa fa-check"></i><b>7</b> Community Diversity</a>
<ul>
<li class="chapter" data-level="7.1" data-path="community-diversity.html"><a href="community-diversity.html#estimation"><i class="fa fa-check"></i><b>7.1</b> Estimation</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="community-diversity.html"><a href="community-diversity.html#richness"><i class="fa fa-check"></i><b>7.1.1</b> Richness</a></li>
<li class="chapter" data-level="7.1.2" data-path="community-diversity.html"><a href="community-diversity.html#estimate-diversity"><i class="fa fa-check"></i><b>7.1.2</b> Diversity</a></li>
<li class="chapter" data-level="7.1.3" data-path="community-diversity.html"><a href="community-diversity.html#faith-diversity"><i class="fa fa-check"></i><b>7.1.3</b> Faith phylogenetic diversity</a></li>
<li class="chapter" data-level="7.1.4" data-path="community-diversity.html"><a href="community-diversity.html#evenness"><i class="fa fa-check"></i><b>7.1.4</b> Evenness</a></li>
<li class="chapter" data-level="7.1.5" data-path="community-diversity.html"><a href="community-diversity.html#dominance"><i class="fa fa-check"></i><b>7.1.5</b> Dominance</a></li>
<li class="chapter" data-level="7.1.6" data-path="community-diversity.html"><a href="community-diversity.html#rarity"><i class="fa fa-check"></i><b>7.1.6</b> Rarity</a></li>
<li class="chapter" data-level="7.1.7" data-path="community-diversity.html"><a href="community-diversity.html#divergence"><i class="fa fa-check"></i><b>7.1.7</b> Divergence</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="community-diversity.html"><a href="community-diversity.html#visualization"><i class="fa fa-check"></i><b>7.2</b> Visualization</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="community-similarity.html"><a href="community-similarity.html"><i class="fa fa-check"></i><b>8</b> Community Similarity</a>
<ul>
<li class="chapter" data-level="8.1" data-path="community-similarity.html"><a href="community-similarity.html#unsupervized-ordination"><i class="fa fa-check"></i><b>8.1</b> Unsupervized ordination</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="community-similarity.html"><a href="community-similarity.html#comparing-communities-by-beta-diversity-analysis"><i class="fa fa-check"></i><b>8.1.1</b> Comparing communities by beta diversity analysis</a></li>
<li class="chapter" data-level="8.1.2" data-path="community-similarity.html"><a href="community-similarity.html#other-ord-methods"><i class="fa fa-check"></i><b>8.1.2</b> Other ordination methods</a></li>
<li class="chapter" data-level="8.1.3" data-path="community-similarity.html"><a href="community-similarity.html#explained-variance"><i class="fa fa-check"></i><b>8.1.3</b> Explained variance</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="community-similarity.html"><a href="community-similarity.html#supervized-ordination"><i class="fa fa-check"></i><b>8.2</b> Supervized ordination</a></li>
<li class="chapter" data-level="8.3" data-path="community-similarity.html"><a href="community-similarity.html#case-studies"><i class="fa fa-check"></i><b>8.3</b> Case studies</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="community-similarity.html"><a href="community-similarity.html#dbrda-workflow"><i class="fa fa-check"></i><b>8.3.1</b> Testing differences in community composition between sample groups</a></li>
<li class="chapter" data-level="8.3.2" data-path="community-similarity.html"><a href="community-similarity.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>8.3.2</b> Checking the homogeneity condition</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="community-similarity.html"><a href="community-similarity.html#summary"><i class="fa fa-check"></i><b>8.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="microbiome-community.html"><a href="microbiome-community.html"><i class="fa fa-check"></i><b>9</b> Community Composition</a>
<ul>
<li class="chapter" data-level="9.1" data-path="microbiome-community.html"><a href="microbiome-community.html#visual-composition"><i class="fa fa-check"></i><b>9.1</b> Visualizing taxonomic composition</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="microbiome-community.html"><a href="microbiome-community.html#composition-barplot"><i class="fa fa-check"></i><b>9.1.1</b> Composition barplot</a></li>
<li class="chapter" data-level="9.1.2" data-path="microbiome-community.html"><a href="microbiome-community.html#composition-heatmap"><i class="fa fa-check"></i><b>9.1.2</b> Composition heatmap</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="clustering.html"><a href="clustering.html"><i class="fa fa-check"></i><b>10</b> Community Typing (Clustering)</a>
<ul>
<li class="chapter" data-level="10.1" data-path="clustering.html"><a href="clustering.html#introduction-to-bluster-package"><i class="fa fa-check"></i><b>10.1</b> Introduction to bluster package</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="clustering.html"><a href="clustering.html#sample-clustering"><i class="fa fa-check"></i><b>10.1.1</b> Sample clustering</a></li>
<li class="chapter" data-level="10.1.2" data-path="clustering.html"><a href="clustering.html#taxa-clustering"><i class="fa fa-check"></i><b>10.1.2</b> Taxa clustering</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="clustering.html"><a href="clustering.html#hierarchical-clustering"><i class="fa fa-check"></i><b>10.2</b> Hierarchical clustering</a></li>
<li class="chapter" data-level="10.3" data-path="clustering.html"><a href="clustering.html#k-means-clustering"><i class="fa fa-check"></i><b>10.3</b> K-means clustering</a></li>
<li class="chapter" data-level="10.4" data-path="clustering.html"><a href="clustering.html#dirichlet-multinomial-mixtures-dmm"><i class="fa fa-check"></i><b>10.4</b> Dirichlet Multinomial Mixtures (DMM)</a>
<ul>
<li class="chapter" data-level="10.4.1" data-path="clustering.html"><a href="clustering.html#pcoa-for-asv-level-data-with-bray-curtis-with-dmm-clusters-shown-with-colors"><i class="fa fa-check"></i><b>10.4.1</b> PCoA for ASV-level data with Bray-Curtis; with DMM clusters shown with colors</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="clustering.html"><a href="clustering.html#community-detection"><i class="fa fa-check"></i><b>10.5</b> Community Detection</a></li>
<li class="chapter" data-level="10.6" data-path="clustering.html"><a href="clustering.html#biclustering"><i class="fa fa-check"></i><b>10.6</b> Biclustering</a>
<ul>
<li class="chapter" data-level="10.6.1" data-path="clustering.html"><a href="clustering.html#taxa-vs-samples"><i class="fa fa-check"></i><b>10.6.1</b> Taxa vs samples</a></li>
<li class="chapter" data-level="10.6.2" data-path="clustering.html"><a href="clustering.html#taxa-vs-biomolecules"><i class="fa fa-check"></i><b>10.6.2</b> Taxa vs biomolecules</a></li>
<li class="chapter" data-level="10.6.3" data-path="clustering.html"><a href="clustering.html#taxa-vs-taxa"><i class="fa fa-check"></i><b>10.6.3</b> Taxa vs taxa</a></li>
</ul></li>
<li class="chapter" data-level="10.7" data-path="clustering.html"><a href="clustering.html#additional-community-typing"><i class="fa fa-check"></i><b>10.7</b> Additional Community Typing</a>
<ul>
<li class="chapter" data-level="10.7.1" data-path="clustering.html"><a href="clustering.html#hierarchical-clustering-demo"><i class="fa fa-check"></i><b>10.7.1</b> Hierarchical clustering demo</a></li>
<li class="chapter" data-level="10.7.2" data-path="clustering.html"><a href="clustering.html#k-means-clustering-demo"><i class="fa fa-check"></i><b>10.7.2</b> K-means clustering Demo</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="differential-abundance.html"><a href="differential-abundance.html"><i class="fa fa-check"></i><b>11</b> Differential Abundance</a>
<ul>
<li class="chapter" data-level="11.1" data-path="differential-abundance.html"><a href="differential-abundance.html#differential-abundance-analysis"><i class="fa fa-check"></i><b>11.1</b> Differential abundance analysis</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="differential-abundance.html"><a href="differential-abundance.html#examples-and-tools"><i class="fa fa-check"></i><b>11.1.1</b> Examples and tools</a></li>
<li class="chapter" data-level="11.1.2" data-path="differential-abundance.html"><a href="differential-abundance.html#prevalence-filtering"><i class="fa fa-check"></i><b>11.1.2</b> Prevalence Filtering</a></li>
<li class="chapter" data-level="11.1.3" data-path="differential-abundance.html"><a href="differential-abundance.html#aldex2"><i class="fa fa-check"></i><b>11.1.3</b> ALDEx2</a></li>
<li class="chapter" data-level="11.1.4" data-path="differential-abundance.html"><a href="differential-abundance.html#ancom-bc"><i class="fa fa-check"></i><b>11.1.4</b> ANCOM-BC</a></li>
<li class="chapter" data-level="11.1.5" data-path="differential-abundance.html"><a href="differential-abundance.html#maaslin2"><i class="fa fa-check"></i><b>11.1.5</b> MaAsLin2</a></li>
<li class="chapter" data-level="11.1.6" data-path="differential-abundance.html"><a href="differential-abundance.html#linda"><i class="fa fa-check"></i><b>11.1.6</b> LinDA</a></li>
<li class="chapter" data-level="11.1.7" data-path="differential-abundance.html"><a href="differential-abundance.html#zicoseq"><i class="fa fa-check"></i><b>11.1.7</b> ZicoSeq</a></li>
<li class="chapter" data-level="11.1.8" data-path="differential-abundance.html"><a href="differential-abundance.html#comparison-of-methods"><i class="fa fa-check"></i><b>11.1.8</b> Comparison of methods</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="differential-abundance.html"><a href="differential-abundance.html#confounding-variables"><i class="fa fa-check"></i><b>11.2</b> Confounding variables</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="differential-abundance.html"><a href="differential-abundance.html#ancombc"><i class="fa fa-check"></i><b>11.2.1</b> ANCOMBC</a></li>
<li class="chapter" data-level="11.2.2" data-path="differential-abundance.html"><a href="differential-abundance.html#linda-1"><i class="fa fa-check"></i><b>11.2.2</b> LinDA</a></li>
<li class="chapter" data-level="11.2.3" data-path="differential-abundance.html"><a href="differential-abundance.html#zicoseq-1"><i class="fa fa-check"></i><b>11.2.3</b> ZicoSeq</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="differential-abundance.html"><a href="differential-abundance.html#tree-based-methods"><i class="fa fa-check"></i><b>11.3</b> Tree-based methods</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="differential-abundance.html"><a href="differential-abundance.html#group-wise-associations-testing-based-on-balances"><i class="fa fa-check"></i><b>11.3.1</b> Group-wise associations testing based on balances</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="machine_learning.html"><a href="machine_learning.html"><i class="fa fa-check"></i><b>12</b> Machine learning</a>
<ul>
<li class="chapter" data-level="12.1" data-path="machine_learning.html"><a href="machine_learning.html#supervised-machine-learning"><i class="fa fa-check"></i><b>12.1</b> Supervised machine learning</a></li>
<li class="chapter" data-level="12.2" data-path="machine_learning.html"><a href="machine_learning.html#unsupervised-machine-learning"><i class="fa fa-check"></i><b>12.2</b> Unsupervised machine learning</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="multi-assay-analyses.html"><a href="multi-assay-analyses.html"><i class="fa fa-check"></i><b>13</b> Multi-assay analyses</a>
<ul>
<li class="chapter" data-level="13.1" data-path="multi-assay-analyses.html"><a href="multi-assay-analyses.html#cross-correlation"><i class="fa fa-check"></i><b>13.1</b> Cross-correlation Analysis</a></li>
<li class="chapter" data-level="13.2" data-path="multi-assay-analyses.html"><a href="multi-assay-analyses.html#mofa"><i class="fa fa-check"></i><b>13.2</b> Multi-Omics Factor Analysis</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="viz-chapter.html"><a href="viz-chapter.html"><i class="fa fa-check"></i><b>14</b> Visualization</a>
<ul>
<li class="chapter" data-level="14.1" data-path="viz-chapter.html"><a href="viz-chapter.html#pre-analysis-exploration"><i class="fa fa-check"></i><b>14.1</b> Pre-analysis exploration</a>
<ul>
<li class="chapter" data-level="14.1.1" data-path="viz-chapter.html"><a href="viz-chapter.html#accessing-row-and-column-data"><i class="fa fa-check"></i><b>14.1.1</b> Accessing row and column data</a></li>
<li class="chapter" data-level="14.1.2" data-path="viz-chapter.html"><a href="viz-chapter.html#viewing-abundance-and-prevalence-patterns"><i class="fa fa-check"></i><b>14.1.2</b> Viewing abundance and prevalence patterns</a></li>
</ul></li>
<li class="chapter" data-level="14.2" data-path="viz-chapter.html"><a href="viz-chapter.html#diversity-estimation"><i class="fa fa-check"></i><b>14.2</b> Diversity estimation</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="viz-chapter.html"><a href="viz-chapter.html#alpha-diversity-with-scatter-violin-and-box-plots"><i class="fa fa-check"></i><b>14.2.1</b> Alpha diversity with scatter, violin and box plots</a></li>
<li class="chapter" data-level="14.2.2" data-path="viz-chapter.html"><a href="viz-chapter.html#beta-diversity-with-shepard-and-coordination-plots"><i class="fa fa-check"></i><b>14.2.2</b> Beta diversity with Shepard and coordination plots</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="viz-chapter.html"><a href="viz-chapter.html#statistical-analysis"><i class="fa fa-check"></i><b>14.3</b> Statistical analysis</a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="viz-chapter.html"><a href="viz-chapter.html#heatmaps"><i class="fa fa-check"></i><b>14.3.1</b> Heatmaps</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>III Training</b></span></li>
<li class="chapter" data-level="15" data-path="training.html"><a href="training.html"><i class="fa fa-check"></i><b>15</b> Training</a>
<ul>
<li class="chapter" data-level="15.1" data-path="training.html"><a href="training.html#checklist"><i class="fa fa-check"></i><b>15.1</b> Checklist</a></li>
<li class="chapter" data-level="15.2" data-path="training.html"><a href="training.html#software"><i class="fa fa-check"></i><b>15.2</b> Recommended software</a></li>
<li class="chapter" data-level="15.3" data-path="training.html"><a href="training.html#material"><i class="fa fa-check"></i><b>15.3</b> Study material</a></li>
<li class="chapter" data-level="15.4" data-path="training.html"><a href="training.html#support-and-resources"><i class="fa fa-check"></i><b>15.4</b> Support and resources</a></li>
<li class="chapter" data-level="15.5" data-path="training.html"><a href="training.html#coc"><i class="fa fa-check"></i><b>15.5</b> Code of Conduct</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="resources.html"><a href="resources.html"><i class="fa fa-check"></i><b>16</b> Resources</a>
<ul>
<li class="chapter" data-level="16.1" data-path="resources.html"><a href="resources.html#data-containers-1"><i class="fa fa-check"></i><b>16.1</b> Data containers</a>
<ul>
<li class="chapter" data-level="16.1.1" data-path="resources.html"><a href="resources.html#data-container-documentation"><i class="fa fa-check"></i><b>16.1.1</b> Data container documentation</a></li>
<li class="chapter" data-level="16.1.2" data-path="resources.html"><a href="resources.html#other-relevant-containers"><i class="fa fa-check"></i><b>16.1.2</b> Other relevant containers</a></li>
<li class="chapter" data-level="16.1.3" data-path="resources.html"><a href="resources.html#alternative-containers-for-microbiome-data"><i class="fa fa-check"></i><b>16.1.3</b> Alternative containers for microbiome data</a></li>
<li class="chapter" data-level="16.1.4" data-path="resources.html"><a href="resources.html#resources-for-phyloseq"><i class="fa fa-check"></i><b>16.1.4</b> Resources for phyloseq</a></li>
</ul></li>
<li class="chapter" data-level="16.2" data-path="resources.html"><a href="resources.html#r-programming-resources"><i class="fa fa-check"></i><b>16.2</b> R programming resources</a>
<ul>
<li class="chapter" data-level="16.2.1" data-path="resources.html"><a href="resources.html#base-r-and-rstudio"><i class="fa fa-check"></i><b>16.2.1</b> Base R and RStudio</a></li>
<li class="chapter" data-level="16.2.2" data-path="resources.html"><a href="resources.html#bioc_intro"><i class="fa fa-check"></i><b>16.2.2</b> Bioconductor Classes</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="resources.html"><a href="resources.html#quarto"><i class="fa fa-check"></i><b>16.3</b> Reproducible reporting with Quarto</a>
<ul>
<li class="chapter" data-level="16.3.1" data-path="resources.html"><a href="resources.html#learn-quarto"><i class="fa fa-check"></i><b>16.3.1</b> Learn Quarto</a></li>
<li class="chapter" data-level="16.3.2" data-path="resources.html"><a href="resources.html#additional-material-on-rmarkdown"><i class="fa fa-check"></i><b>16.3.2</b> Additional material on Rmarkdown</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="exercises.html"><a href="exercises.html"><i class="fa fa-check"></i><b>17</b> Exercises</a>
<ul>
<li class="chapter" data-level="17.1" data-path="exercises.html"><a href="exercises.html#workflows"><i class="fa fa-check"></i><b>17.1</b> Workflows</a>
<ul>
<li class="chapter" data-level="17.1.1" data-path="exercises.html"><a href="exercises.html#reproducible-reporting-with-quarto"><i class="fa fa-check"></i><b>17.1.1</b> Reproducible reporting with Quarto</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="exercises.html"><a href="exercises.html#data-containers-treese"><i class="fa fa-check"></i><b>17.2</b> Data containers: TreeSE</a>
<ul>
<li class="chapter" data-level="17.2.1" data-path="exercises.html"><a href="exercises.html#construct-TreeSE"><i class="fa fa-check"></i><b>17.2.1</b> Constructing a data object</a></li>
<li class="chapter" data-level="17.2.2" data-path="exercises.html"><a href="exercises.html#importing-data"><i class="fa fa-check"></i><b>17.2.2</b> Importing data</a></li>
<li class="chapter" data-level="17.2.3" data-path="exercises.html"><a href="exercises.html#preliminary-exploration"><i class="fa fa-check"></i><b>17.2.3</b> Preliminary exploration</a></li>
<li class="chapter" data-level="17.2.4" data-path="exercises.html"><a href="exercises.html#assay-retrieval"><i class="fa fa-check"></i><b>17.2.4</b> Assay retrieval</a></li>
<li class="chapter" data-level="17.2.5" data-path="exercises.html"><a href="exercises.html#sample-information"><i class="fa fa-check"></i><b>17.2.5</b> Sample information</a></li>
<li class="chapter" data-level="17.2.6" data-path="exercises.html"><a href="exercises.html#feature-information"><i class="fa fa-check"></i><b>17.2.6</b> Feature information</a></li>
<li class="chapter" data-level="17.2.7" data-path="exercises.html"><a href="exercises.html#other-elements"><i class="fa fa-check"></i><b>17.2.7</b> Other elements</a></li>
</ul></li>
<li class="chapter" data-level="17.3" data-path="exercises.html"><a href="exercises.html#data-manipulation"><i class="fa fa-check"></i><b>17.3</b> Data manipulation</a>
<ul>
<li class="chapter" data-level="17.3.1" data-path="exercises.html"><a href="exercises.html#subsetting-1"><i class="fa fa-check"></i><b>17.3.1</b> Subsetting</a></li>
<li class="chapter" data-level="17.3.2" data-path="exercises.html"><a href="exercises.html#library-sizes"><i class="fa fa-check"></i><b>17.3.2</b> Library sizes</a></li>
<li class="chapter" data-level="17.3.3" data-path="exercises.html"><a href="exercises.html#prevalent-and-core-taxonomic-features"><i class="fa fa-check"></i><b>17.3.3</b> Prevalent and core taxonomic features</a></li>
<li class="chapter" data-level="17.3.4" data-path="exercises.html"><a href="exercises.html#data-exploration"><i class="fa fa-check"></i><b>17.3.4</b> Data exploration</a></li>
<li class="chapter" data-level="17.3.5" data-path="exercises.html"><a href="exercises.html#other-functions"><i class="fa fa-check"></i><b>17.3.5</b> Other functions</a></li>
<li class="chapter" data-level="17.3.6" data-path="exercises.html"><a href="exercises.html#assay-transformation"><i class="fa fa-check"></i><b>17.3.6</b> Assay transformation</a></li>
</ul></li>
<li class="chapter" data-level="17.4" data-path="exercises.html"><a href="exercises.html#abundance-tables"><i class="fa fa-check"></i><b>17.4</b> Abundance tables</a>
<ul>
<li class="chapter" data-level="17.4.1" data-path="exercises.html"><a href="exercises.html#taxonomic-levels"><i class="fa fa-check"></i><b>17.4.1</b> Taxonomic levels</a></li>
<li class="chapter" data-level="17.4.2" data-path="exercises.html"><a href="exercises.html#alternative-experiments"><i class="fa fa-check"></i><b>17.4.2</b> Alternative experiments</a></li>
</ul></li>
<li class="chapter" data-level="17.5" data-path="exercises.html"><a href="exercises.html#community-alpha-diversity"><i class="fa fa-check"></i><b>17.5</b> Community (alpha) diversity</a>
<ul>
<li class="chapter" data-level="17.5.1" data-path="exercises.html"><a href="exercises.html#estimation-1"><i class="fa fa-check"></i><b>17.5.1</b> Estimation</a></li>
<li class="chapter" data-level="17.5.2" data-path="exercises.html"><a href="exercises.html#visualization-1"><i class="fa fa-check"></i><b>17.5.2</b> Visualization</a></li>
<li class="chapter" data-level="17.5.3" data-path="exercises.html"><a href="exercises.html#correlation"><i class="fa fa-check"></i><b>17.5.3</b> Correlation</a></li>
<li class="chapter" data-level="17.5.4" data-path="exercises.html"><a href="exercises.html#differences-between-groups"><i class="fa fa-check"></i><b>17.5.4</b> Differences between groups</a></li>
</ul></li>
<li class="chapter" data-level="17.6" data-path="exercises.html"><a href="exercises.html#community-similarity-1"><i class="fa fa-check"></i><b>17.6</b> Community similarity</a>
<ul>
<li class="chapter" data-level="17.6.1" data-path="exercises.html"><a href="exercises.html#reduced-dimensions-retrieval"><i class="fa fa-check"></i><b>17.6.1</b> Reduced dimensions retrieval</a></li>
<li class="chapter" data-level="17.6.2" data-path="exercises.html"><a href="exercises.html#visualization-basics-with-pca"><i class="fa fa-check"></i><b>17.6.2</b> Visualization basics with PCA</a></li>
<li class="chapter" data-level="17.6.3" data-path="exercises.html"><a href="exercises.html#principal-coordinate-analysis-pcoa"><i class="fa fa-check"></i><b>17.6.3</b> Principal Coordinate Analysis (PCoA)</a></li>
<li class="chapter" data-level="17.6.4" data-path="exercises.html"><a href="exercises.html#permanova-analysis"><i class="fa fa-check"></i><b>17.6.4</b> PERMANOVA analysis</a></li>
<li class="chapter" data-level="17.6.5" data-path="exercises.html"><a href="exercises.html#redundance-analysis-rda"><i class="fa fa-check"></i><b>17.6.5</b> Redundance analysis (RDA)</a></li>
<li class="chapter" data-level="17.6.6" data-path="exercises.html"><a href="exercises.html#beta-diversity-analysis"><i class="fa fa-check"></i><b>17.6.6</b> Beta diversity analysis</a></li>
</ul></li>
<li class="chapter" data-level="17.7" data-path="exercises.html"><a href="exercises.html#differential-abundance-1"><i class="fa fa-check"></i><b>17.7</b> Differential abundance</a>
<ul>
<li class="chapter" data-level="17.7.1" data-path="exercises.html"><a href="exercises.html#univariate-analyses"><i class="fa fa-check"></i><b>17.7.1</b> Univariate analyses</a></li>
<li class="chapter" data-level="17.7.2" data-path="exercises.html"><a href="exercises.html#differential-abundance-analysis-1"><i class="fa fa-check"></i><b>17.7.2</b> Differential abundance analysis</a></li>
</ul></li>
<li class="chapter" data-level="17.8" data-path="exercises.html"><a href="exercises.html#visualization-2"><i class="fa fa-check"></i><b>17.8</b> Visualization</a>
<ul>
<li class="chapter" data-level="17.8.1" data-path="exercises.html"><a href="exercises.html#multivariate-ordination"><i class="fa fa-check"></i><b>17.8.1</b> Multivariate ordination</a></li>
<li class="chapter" data-level="17.8.2" data-path="exercises.html"><a href="exercises.html#heatmap-visualization"><i class="fa fa-check"></i><b>17.8.2</b> Heatmap visualization</a></li>
</ul></li>
<li class="chapter" data-level="17.9" data-path="exercises.html"><a href="exercises.html#multiomics"><i class="fa fa-check"></i><b>17.9</b> Multiomics</a>
<ul>
<li class="chapter" data-level="17.9.1" data-path="exercises.html"><a href="exercises.html#basic-exploration"><i class="fa fa-check"></i><b>17.9.1</b> Basic exploration</a></li>
<li class="chapter" data-level="17.9.2" data-path="exercises.html"><a href="exercises.html#experiment-agglomeration"><i class="fa fa-check"></i><b>17.9.2</b> Experiment agglomeration</a></li>
<li class="chapter" data-level="17.9.3" data-path="exercises.html"><a href="exercises.html#experiment-transformation"><i class="fa fa-check"></i><b>17.9.3</b> Experiment transformation</a></li>
<li class="chapter" data-level="17.9.4" data-path="exercises.html"><a href="exercises.html#assay-extraction"><i class="fa fa-check"></i><b>17.9.4</b> Assay extraction</a></li>
<li class="chapter" data-level="17.9.5" data-path="exercises.html"><a href="exercises.html#mae-reconstruction"><i class="fa fa-check"></i><b>17.9.5</b> MAE reconstruction</a></li>
<li class="chapter" data-level="17.9.6" data-path="exercises.html"><a href="exercises.html#cross-correlation-analysis"><i class="fa fa-check"></i><b>17.9.6</b> Cross-correlation analysis</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>IV Appendix</b></span></li>
<li class="chapter" data-level="18" data-path="extras.html"><a href="extras.html"><i class="fa fa-check"></i><b>18</b> Extra material</a>
<ul>
<li class="chapter" data-level="18.1" data-path="extras.html"><a href="extras.html#compare-permanova"><i class="fa fa-check"></i><b>18.1</b> PERMANOVA comparison</a></li>
<li class="chapter" data-level="18.2" data-path="extras.html"><a href="extras.html#bayesian-multinomial-logistic-normal-models"><i class="fa fa-check"></i><b>18.2</b> Bayesian Multinomial Logistic-Normal Models</a></li>
<li class="chapter" data-level="18.3" data-path="extras.html"><a href="extras.html#interactive-3d-plots"><i class="fa fa-check"></i><b>18.3</b> Interactive 3D Plots</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="authors-and-contributors.html"><a href="authors-and-contributors.html"><i class="fa fa-check"></i>Authors and contributors</a>
<ul>
<li class="chapter" data-level="" data-path="authors-and-contributors.html"><a href="authors-and-contributors.html#leo-lahti-dsc"><i class="fa fa-check"></i><em>Leo Lahti, DSc</em></a></li>
<li class="chapter" data-level="" data-path="authors-and-contributors.html"><a href="authors-and-contributors.html#tuomas-borman"><i class="fa fa-check"></i><em>Tuomas Borman</em></a></li>
<li class="chapter" data-level="" data-path="authors-and-contributors.html"><a href="authors-and-contributors.html#giulio-benedetti"><i class="fa fa-check"></i><em>Giulio Benedetti</em></a></li>
<li class="chapter" data-level="" data-path="authors-and-contributors.html"><a href="authors-and-contributors.html#felix-ernst-phd"><i class="fa fa-check"></i><em>Felix Ernst, PhD</em></a></li>
<li class="chapter" data-level="" data-path="authors-and-contributors.html"><a href="authors-and-contributors.html#contributors"><i class="fa fa-check"></i>Contributors</a></li>
<li class="chapter" data-level="" data-path="authors-and-contributors.html"><a href="authors-and-contributors.html#acknowledgments"><i class="fa fa-check"></i>Acknowledgments</a></li>
<li class="chapter" data-level="" data-path="authors-and-contributors.html"><a href="authors-and-contributors.html#support"><i class="fa fa-check"></i>Support</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="sessioninfo.html"><a href="sessioninfo.html"><i class="fa fa-check"></i>Sessioninfo</a></li>
<li class="chapter" data-level="" data-path="bibliography.html"><a href="bibliography.html"><i class="fa fa-check"></i>Bibliography</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Orchestrating Microbiome Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="differential-abundance" class="section level1 hasAnchor" number="11">
<h1><span class="header-section-number">Chapter 11</span> Differential Abundance<a href="differential-abundance.html#differential-abundance" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<script>
document.addEventListener("click", function (event) {
    if (event.target.classList.contains("rebook-collapse")) {
        event.target.classList.toggle("active");
        var content = event.target.nextElementSibling;
        if (content.style.display === "block") {
            content.style.display = "none";
        } else {
            content.style.display = "block";
        }
    }
})
</script>
<style>
.rebook-collapse {
  background-color: #eee;
  color: #444;
  cursor: pointer;
  padding: 18px;
  width: 100%;
  border: none;
  text-align: left;
  outline: none;
  font-size: 15px;
}

.rebook-content {
  padding: 0 18px;
  display: none;
  overflow: hidden;
  background-color: #f1f1f1;
}
</style>
<div id="differential-abundance-analysis" class="section level2 hasAnchor" number="11.1">
<h2><span class="header-section-number">11.1</span> Differential abundance analysis<a href="differential-abundance.html#differential-abundance-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This section provides an introduction to Differential Abundance Analysis (DAA),
which is used to identify differences in the abundances of individual taxa (at
any taxonomic level) between two or more groups, such as treatment and control.
Here, we demonstrate its implementation on peerj13075, one of the <a href="containers.html#package-data">mia datasets</a>.</p>
<p>The goal of DAA is to identify biomarkers of a certain phenotype or condition,
and gain understanding of a complex system by looking at its isolated components.
For example, the identification of a bacterial taxon that is differentially
abundant between healthy patients and diseased patients can lead to important
insights into the pathophysiology of the disease. In other words, differentially
abundant taxa can be involved in the dynamics of the disease, which in turn helps
understand the system as a whole. Despite its relevance in current research,
the DAA approach has also been subject to debate <span class="citation">(Quinn, Gordon-Rodriguez, and Erb 2021)</span>.</p>
<div id="examples-and-tools" class="section level3 hasAnchor" number="11.1.1">
<h3><span class="header-section-number">11.1.1</span> Examples and tools<a href="differential-abundance.html#examples-and-tools" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Due to the size and complexity of microbiome sequencing data, in this field
DAA still faces several statistical challenges <span class="citation">(Yang and Chen 2022)</span>. These include:</p>
<ul>
<li><p>High variability. The abundance of a specific taxon could range over
several orders of magnitude.</p></li>
<li><p>Zero-inflation. In a typical microbiome dataset, more than 70% of the
values are zeros. Zeros could be due to either physical absence
(structural zeros) or insufficient sampling effort (sampling zeros).</p></li>
<li><p>Compositionality. Increase or decrease in the (absolute) abundance of
one taxon at the sampling site will lead to apparent changes in the
relative abundances of other taxa in the sample.</p></li>
</ul>
<p>As a consequence, the following approaches have been developed:</p>
<ul>
<li><p>Over-dispersed count models has been proposed to address zero
inflation, such as the negative binomial model used by edgeR
<span class="citation">(Y. Chen, Lun, and Smyth 2016)</span> and DESeq2 <span class="citation">(Love, Huber, and Anders 2014)</span>, the beta-binomial model used by
corncorb <span class="citation">(Martin, Witten, and Willis 2021)</span>.</p></li>
<li><p>Zero-inflated mixture models has aslo been proposed to address zero
inflation, such as zero-inflated log-normal/normal mixture model
used by metagenomeSeq <span class="citation">(Paulson, Talukder, and Bravo 2017)</span> and RAIDA <span class="citation">(Sohn, Du, and An 2015)</span>,
zero-inflated beta-binomial model used by ZIBB <span class="citation">(T. Hu, Gallins, and Zhou 2018)</span>, and
zero-inflated negative binomial model used by Omnibus
<span class="citation">(J. Chen et al. 2018)</span>.</p></li>
<li><p>Bayesian methods have been used to impute the zeros for methods
working on proportion data, accounting for sampling variability and
sequencing depth variation. Examples include ALDEx2 <span class="citation">(Gloor, Macklaim, and Fernandes 2016)</span> and
eBay <span class="citation">(Liu, Zhao, and Wang 2020)</span>.</p></li>
<li><p>Other methods use the pseudo-count approach to impute the zeros,
such as MaAsLin2 <span class="citation">(Mallick, Rahnavard, and McIver 2020)</span> and ANCOMBC <span class="citation">(Lin and Peddada 2020)</span>.</p></li>
<li><p>Different strategies have been used to address compositional
effects, including:</p>
<ul>
<li><p>Robust normalization. For example, trimmed mean of M-values (TMM)
normalization used by edgeR, relative log expression (RLE)
normalization used by DESeq2 <span class="citation">(Love, Huber, and Anders 2014)</span>, cumulative sum scaling
(CSS) normalization used by metagenomeSeq, centered log-ratio
transformation (CLR) normalization used by ALDEx2 <span class="citation">(Gloor, Macklaim, and Fernandes 2016)</span> and
geometric mean of pairwise ratios (GMPR) normalization used by
Omnibus <span class="citation">(J. Chen et al. 2018)</span>. Wrench normalization <span class="citation">(Kumar et al. 2018)</span> corrects
the compositional bias by an empirical Bayes approach, which has
been recommended in metagenomeSeq <span class="citation">(Paulson, Talukder, and Bravo 2017)</span> tutorial.</p></li>
<li><p>Reference taxa approach used by DACOMP <span class="citation">(Brill, Amnon, and Ruth 2022)</span> and RAIDA <span class="citation">(Sohn, Du, and An 2015)</span>.</p></li>
<li><p>Analyzing the pattern of pairwise log ratios, such as ANCOM <span class="citation">(Mandal et al. 2015)</span>.</p></li>
<li><p>Bias-correction used by ANCOMBC <span class="citation">(Lin and Peddada 2020)</span>.</p></li>
</ul></li>
</ul>
<p>Some of the popular tools for differential abundance analysis include:</p>
<ul>
<li>ALDEx2 <span class="citation">(Gloor, Macklaim, and Fernandes 2016)</span></li>
<li>ANCOMBC <span class="citation">(Lin and Peddada 2020)</span></li>
<li>corncob <span class="citation">(Martin, Witten, and Willis 2021)</span></li>
<li>DESeq2 <span class="citation">(Love, Huber, and Anders 2014)</span></li>
<li>edgeR <span class="citation">(Y. Chen, Lun, and Smyth 2016)</span></li>
<li>lefser <span class="citation">(Khleborodova 2021)</span></li>
<li>MaAsLin2 <span class="citation">(Mallick, Rahnavard, and McIver 2020)</span></li>
<li>metagenomeSeq <span class="citation">(Paulson, Talukder, and Bravo 2017)</span></li>
<li>limma <span class="citation">(Ritchie et al. 2015)</span></li>
<li>LinDA <span class="citation">(H. Zhou et al. 2022)</span></li>
<li>ZicoSeq <span class="citation">(Yang and Chen 2022)</span></li>
<li>LDM <span class="citation">(Y. Hu and Satten 2020)</span></li>
<li>RAIDA <span class="citation">(Sohn, Du, and An 2015)</span></li>
<li>DACOMP <span class="citation">(Brill, Amnon, and Ruth 2022)</span></li>
<li>Omnibus <span class="citation">(J. Chen et al. 2018)</span></li>
<li>eBay <span class="citation">(Liu, Zhao, and Wang 2020)</span></li>
<li>ZINQ <span class="citation">(Ling et al. 2021)</span></li>
<li>ANCOM <span class="citation">(Mandal et al. 2015)</span></li>
<li>fastANCOM <span class="citation">(C. Zhou et al. 2022)</span></li>
<li><a href="https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/t.test">t-test</a><br />
</li>
<li><a href="https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/wilcox.test">Wilcoxon test</a></li>
</ul>
<p>We recommend to have a look at <span class="citation">Nearing et al. (2022)</span> who compared all these
methods across 38 different datasets. Because different methods use
different approaches (parametric vs non-parametric, different
normalization techiniques, assumptions etc.), the results may differ
between methods, sometimes substantially as <span class="citation">Nearing et al. (2022)</span> pointed
out. More recently <span class="citation">Yang and Chen (2022)</span> comprehensively evaluated these methods
via a semi-parametric framework and 106 real datasets. <span class="citation">Yang and Chen (2022)</span> also
concluded that different DA methods can sometimes produce discordant
results, opening to the possibility for cherry-picking tools in favor
of one’s own hypothesis. Therefore it is highly recommended to pick
several methods to assess how robust and potentially reproducible your
findings are with different methods.</p>
<p>In this section we demonstrate the use of four methods that can be
recommended based on recent literature (ANCOM-BC <span class="citation">(Lin and Peddada 2020)</span>,
<em>ALDEx2</em> <span class="citation">(Gloor, Macklaim, and Fernandes 2016)</span>, <em>Maaslin2</em> <span class="citation">(Mallick, Rahnavard, and McIver 2020)</span>, <em>LinDA</em> <span class="citation">(H. Zhou et al. 2022)</span>
and <em>ZicoSeq</em> <span class="citation">(Yang and Chen 2022)</span>).</p>
<p>The purpose of this section is to show how to perform DAA in R, not
how to correctly do causal inference. Depending on your experimental
setup and your theory, you must determine how to specify any model
exactly. E.g., there might be confounding factors that might drive
(the absence of) differences between the shown groups that we ignore
here for simplicity. Or your dataset is repeated sampling design,
matched-pair design or the general longitudianl design. We will
demonstrate how to include covariates in those models. We picked a
dataset that merely has microbial abundances in a TSE object as well
as a grouping variable in the sample data. We simplify the examples by
only including two of the three groups.</p>
<div class="sourceCode" id="cb338"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb338-1"><a href="differential-abundance.html#cb338-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mia)</span>
<span id="cb338-2"><a href="differential-abundance.html#cb338-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb338-3"><a href="differential-abundance.html#cb338-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidySummarizedExperiment)</span>
<span id="cb338-4"><a href="differential-abundance.html#cb338-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(knitr)</span>
<span id="cb338-5"><a href="differential-abundance.html#cb338-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb338-6"><a href="differential-abundance.html#cb338-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ALDEx2)</span>
<span id="cb338-7"><a href="differential-abundance.html#cb338-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Maaslin2)</span>
<span id="cb338-8"><a href="differential-abundance.html#cb338-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(MicrobiomeStat)</span>
<span id="cb338-9"><a href="differential-abundance.html#cb338-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ANCOMBC)</span>
<span id="cb338-10"><a href="differential-abundance.html#cb338-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GUniFrac)</span>
<span id="cb338-11"><a href="differential-abundance.html#cb338-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb338-12"><a href="differential-abundance.html#cb338-12" aria-hidden="true" tabindex="-1"></a><span class="co"># set random seed because some tools can randomly vary and then produce </span></span>
<span id="cb338-13"><a href="differential-abundance.html#cb338-13" aria-hidden="true" tabindex="-1"></a><span class="co"># different results:</span></span>
<span id="cb338-14"><a href="differential-abundance.html#cb338-14" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">13253</span>)</span>
<span id="cb338-15"><a href="differential-abundance.html#cb338-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb338-16"><a href="differential-abundance.html#cb338-16" aria-hidden="true" tabindex="-1"></a><span class="co"># we use a demo dataset and restrict it to two geo locations</span></span>
<span id="cb338-17"><a href="differential-abundance.html#cb338-17" aria-hidden="true" tabindex="-1"></a><span class="co"># for easy illustration</span></span>
<span id="cb338-18"><a href="differential-abundance.html#cb338-18" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(peerj13075)</span>
<span id="cb338-19"><a href="differential-abundance.html#cb338-19" aria-hidden="true" tabindex="-1"></a>tse0 <span class="ot">&lt;-</span> peerj13075</span>
<span id="cb338-20"><a href="differential-abundance.html#cb338-20" aria-hidden="true" tabindex="-1"></a>tse0 <span class="ot">&lt;-</span> tse0[ ,tse0<span class="sc">$</span>Geographical_location <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Pune&quot;</span>, <span class="st">&quot;Nashik&quot;</span>)]</span>
<span id="cb338-21"><a href="differential-abundance.html#cb338-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Let us make this a factor</span></span>
<span id="cb338-22"><a href="differential-abundance.html#cb338-22" aria-hidden="true" tabindex="-1"></a>tse0<span class="sc">$</span>Geographical_location <span class="ot">&lt;-</span> <span class="fu">factor</span>(tse0<span class="sc">$</span>Geographical_location)</span>
<span id="cb338-23"><a href="differential-abundance.html#cb338-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb338-24"><a href="differential-abundance.html#cb338-24" aria-hidden="true" tabindex="-1"></a><span class="co"># how many observations do we have per group?</span></span>
<span id="cb338-25"><a href="differential-abundance.html#cb338-25" aria-hidden="true" tabindex="-1"></a><span class="fu">as.data.frame</span>(<span class="fu">colData</span>(tse0)) <span class="sc">%&gt;%</span> </span>
<span id="cb338-26"><a href="differential-abundance.html#cb338-26" aria-hidden="true" tabindex="-1"></a><span class="fu">count</span>(Geographical_location) <span class="sc">%&gt;%</span></span>
<span id="cb338-27"><a href="differential-abundance.html#cb338-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>()</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">Geographical_location</th>
<th align="right">n</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Nashik</td>
<td align="right">11</td>
</tr>
<tr class="even">
<td align="left">Pune</td>
<td align="right">36</td>
</tr>
</tbody>
</table>
</div>
<div id="prevalence-filtering" class="section level3 hasAnchor" number="11.1.2">
<h3><span class="header-section-number">11.1.2</span> Prevalence Filtering<a href="differential-abundance.html#prevalence-filtering" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Before we jump to our analyses, we may want to perform some data
manipulation.</p>
<p>Let us here do aggregation to genus level, add relative abundance
assay, and perform prevalence filtering.</p>
<div class="sourceCode" id="cb339"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb339-1"><a href="differential-abundance.html#cb339-1" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">agglomerateByRank</span>(tse0, <span class="at">rank =</span> <span class="st">&quot;genus&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb339-2"><a href="differential-abundance.html#cb339-2" aria-hidden="true" tabindex="-1"></a>       <span class="fu">transformCounts</span>(<span class="at">assay.type =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb339-3"><a href="differential-abundance.html#cb339-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">method =</span> <span class="st">&quot;relabundance&quot;</span>,</span>
<span id="cb339-4"><a href="differential-abundance.html#cb339-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">MARGIN =</span> <span class="st">&quot;samples&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb339-5"><a href="differential-abundance.html#cb339-5" aria-hidden="true" tabindex="-1"></a>       <span class="co"># subset based on the relative abundance assay              </span></span>
<span id="cb339-6"><a href="differential-abundance.html#cb339-6" aria-hidden="true" tabindex="-1"></a>       <span class="fu">subsetByPrevalentTaxa</span>(<span class="at">detection =</span> <span class="dv">0</span>,</span>
<span id="cb339-7"><a href="differential-abundance.html#cb339-7" aria-hidden="true" tabindex="-1"></a>                             <span class="at">prevalence =</span> <span class="dv">10</span><span class="sc">/</span><span class="dv">100</span>,</span>
<span id="cb339-8"><a href="differential-abundance.html#cb339-8" aria-hidden="true" tabindex="-1"></a>                 <span class="at">assay.type =</span> <span class="st">&quot;relabundance&quot;</span>)</span>
<span id="cb339-9"><a href="differential-abundance.html#cb339-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb339-10"><a href="differential-abundance.html#cb339-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Add also clr abundances</span></span>
<span id="cb339-11"><a href="differential-abundance.html#cb339-11" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">transformCounts</span>(tse, <span class="at">method=</span><span class="st">&quot;clr&quot;</span>, <span class="at">pseudocount=</span><span class="dv">1</span>) <span class="co"># not bale to run</span></span></code></pre></div>
<p>Regarding prevalence filtering, <span class="citation">Nearing et al. (2022)</span> found that applying a 10%
threshold for the prevalence of the taxa generally resulted in more
robust results. Some tools have builtin arguments for that. By
applying the threshold to our input data, we can make sure it is
applied for all tools.</p>
</div>
<div id="aldex2" class="section level3 hasAnchor" number="11.1.3">
<h3><span class="header-section-number">11.1.3</span> ALDEx2<a href="differential-abundance.html#aldex2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>In this section, we will show how to perform a simple ALDEx2 analysis.
If you wanted to pick a single method, this method could be recommended to use.
According to the developers experience, it tends to identify the common
features identified by other methods. This statement is in line with a recent
independent evaluation by <span class="citation">Nearing et al. (2022)</span>.<br />
Please also have a look at the more extensive
<a href="https://bioconductor.org/packages/release/bioc/vignettes/ALDEx2/inst/doc/ALDEx2_vignette.html">vignette</a>
that covers this flexible tool in more depth. ALDEx2 estimates technical
variation within each sample per taxon by utilizing the Dirichlet distribution.
It furthermore applies the centered-log-ratio transformation (or closely
related log-ratio transforms). Depending on the experimental setup, it will
perform a two sample Welch’s T-test and Wilcoxon-test or a one-way ANOVA and
Kruskal-Wallis-test. For more complex study designs, there is a possibility to
utilize the <code>glm</code> functionality within ALDEx2.</p>
<p>The Benjamini-Hochberg procedure is applied by default to correct for
multiple testing. Below we show a simple example that illustrates the
workflow.</p>
<div class="sourceCode" id="cb340"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb340-1"><a href="differential-abundance.html#cb340-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate Monte Carlo samples of the Dirichlet distribution for each sample.</span></span>
<span id="cb340-2"><a href="differential-abundance.html#cb340-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert each instance using the centered log-ratio transform.</span></span>
<span id="cb340-3"><a href="differential-abundance.html#cb340-3" aria-hidden="true" tabindex="-1"></a><span class="co"># This is the input for all further analyses.</span></span>
<span id="cb340-4"><a href="differential-abundance.html#cb340-4" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">254</span>)</span>
<span id="cb340-5"><a href="differential-abundance.html#cb340-5" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">aldex.clr</span>(<span class="fu">assay</span>(tse), tse<span class="sc">$</span>Geographical_location)     </span></code></pre></div>
<p>The t-test:</p>
<div class="sourceCode" id="cb341"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb341-1"><a href="differential-abundance.html#cb341-1" aria-hidden="true" tabindex="-1"></a><span class="co"># calculates expected values of the Welch&#39;s t-test and Wilcoxon rank</span></span>
<span id="cb341-2"><a href="differential-abundance.html#cb341-2" aria-hidden="true" tabindex="-1"></a><span class="co"># test on the data returned by aldex.clr</span></span>
<span id="cb341-3"><a href="differential-abundance.html#cb341-3" aria-hidden="true" tabindex="-1"></a>x_tt <span class="ot">&lt;-</span> <span class="fu">aldex.ttest</span>(x, <span class="at">paired.test =</span> <span class="cn">FALSE</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p>Effect sizes:</p>
<div class="sourceCode" id="cb342"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb342-1"><a href="differential-abundance.html#cb342-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Determines the median clr abundance of the feature in all samples and in</span></span>
<span id="cb342-2"><a href="differential-abundance.html#cb342-2" aria-hidden="true" tabindex="-1"></a><span class="co"># groups, the median difference between the two groups, the median variation</span></span>
<span id="cb342-3"><a href="differential-abundance.html#cb342-3" aria-hidden="true" tabindex="-1"></a><span class="co"># within each group and the effect size, which is the median of the ratio</span></span>
<span id="cb342-4"><a href="differential-abundance.html#cb342-4" aria-hidden="true" tabindex="-1"></a><span class="co"># of the between group difference and the larger of the variance within groups</span></span>
<span id="cb342-5"><a href="differential-abundance.html#cb342-5" aria-hidden="true" tabindex="-1"></a>x_effect <span class="ot">&lt;-</span> <span class="fu">aldex.effect</span>(x, <span class="at">CI =</span> <span class="cn">TRUE</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb342-6"><a href="differential-abundance.html#cb342-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb342-7"><a href="differential-abundance.html#cb342-7" aria-hidden="true" tabindex="-1"></a><span class="co"># combine all outputs </span></span>
<span id="cb342-8"><a href="differential-abundance.html#cb342-8" aria-hidden="true" tabindex="-1"></a>aldex_out <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(x_tt, x_effect)</span></code></pre></div>
<p>Now, we can create a so called Bland-Altman or MA plot (left). It shows the
association between the relative abundance and the magnitude of the difference
per sample. Next to that, we can also create a plot that shows the dispersion
on the x-axis instead of log-ratio abundance. Red dots represent genera that are
differentially abundant (<span class="math inline">\(q \leq 0.1\)</span>) between the 2 groups. Black points are
rare taxa and grey ones are abundant taxa. The dashed line represent an effect
size of 1. See <span class="citation">Gloor, Macklaim, and Fernandes (2016)</span> to learn more about these plots.</p>
<div class="sourceCode" id="cb343"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb343-1"><a href="differential-abundance.html#cb343-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb343-2"><a href="differential-abundance.html#cb343-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aldex.plot</span>(</span>
<span id="cb343-3"><a href="differential-abundance.html#cb343-3" aria-hidden="true" tabindex="-1"></a>    aldex_out, </span>
<span id="cb343-4"><a href="differential-abundance.html#cb343-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">&quot;MA&quot;</span>, </span>
<span id="cb343-5"><a href="differential-abundance.html#cb343-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">test =</span> <span class="st">&quot;welch&quot;</span>, </span>
<span id="cb343-6"><a href="differential-abundance.html#cb343-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">&quot;Log-ratio abundance&quot;</span>,</span>
<span id="cb343-7"><a href="differential-abundance.html#cb343-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylab =</span> <span class="st">&quot;Difference&quot;</span>,</span>
<span id="cb343-8"><a href="differential-abundance.html#cb343-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">cutoff =</span> <span class="fl">0.05</span></span>
<span id="cb343-9"><a href="differential-abundance.html#cb343-9" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb343-10"><a href="differential-abundance.html#cb343-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aldex.plot</span>(</span>
<span id="cb343-11"><a href="differential-abundance.html#cb343-11" aria-hidden="true" tabindex="-1"></a>    aldex_out, </span>
<span id="cb343-12"><a href="differential-abundance.html#cb343-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">&quot;MW&quot;</span>, </span>
<span id="cb343-13"><a href="differential-abundance.html#cb343-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">test =</span> <span class="st">&quot;welch&quot;</span>,</span>
<span id="cb343-14"><a href="differential-abundance.html#cb343-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">&quot;Dispersion&quot;</span>,</span>
<span id="cb343-15"><a href="differential-abundance.html#cb343-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylab =</span> <span class="st">&quot;Difference&quot;</span>,</span>
<span id="cb343-16"><a href="differential-abundance.html#cb343-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">cutoff =</span> <span class="fl">0.05</span></span>
<span id="cb343-17"><a href="differential-abundance.html#cb343-17" aria-hidden="true" tabindex="-1"></a>  )</span></code></pre></div>
<p><img src="30_differential_abundance_files/figure-html/unnamed-chunk-1-1.png" width="672" /></p>
<p>The evaluation as differential abundant in above plots is based on the
corrected p-value. According to the ALDEx2 developers, the safest
approach is to identify those features where the 95% CI of the effect
size does not cross 0. As we can see in below table, this is not the
case for any of the identified genera (see overlap column, which
indicates the proportion of overlap). Also, the authors recommend to
focus on effect sizes and CIs rather than interpreting the p-value. To
keep the comparison simple, we will here use the p-value as decision
criterion. But please be aware that the effect size together with the
CI is a better answer to the question we are typically interested in
(see also <a href="https://www.nature.com/articles/d41586-019-00857-9">this
article</a>).</p>
<div class="sourceCode" id="cb344"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb344-1"><a href="differential-abundance.html#cb344-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames_to_column</span>(aldex_out, <span class="st">&quot;genus&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb344-2"><a href="differential-abundance.html#cb344-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(wi.eBH <span class="sc">&lt;=</span> <span class="fl">0.05</span>)  <span class="sc">%&gt;%</span> <span class="co"># here we chose the wilcoxon output rather than tt</span></span>
<span id="cb344-3"><a href="differential-abundance.html#cb344-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(genus, we.eBH, wi.eBH, effect, overlap) <span class="sc">%&gt;%</span></span>
<span id="cb344-4"><a href="differential-abundance.html#cb344-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>()</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">genus</th>
<th align="right">we.eBH</th>
<th align="right">wi.eBH</th>
<th align="right">effect</th>
<th align="right">overlap</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Anaerococcus</td>
<td align="right">0.0540</td>
<td align="right">0.0150</td>
<td align="right">0.9595</td>
<td align="right">0.1546</td>
</tr>
<tr class="even">
<td align="left">Calditerricola</td>
<td align="right">0.0769</td>
<td align="right">0.0299</td>
<td align="right">-0.7162</td>
<td align="right">0.1702</td>
</tr>
<tr class="odd">
<td align="left">Chitinivibrio</td>
<td align="right">0.1216</td>
<td align="right">0.0484</td>
<td align="right">-0.7700</td>
<td align="right">0.1776</td>
</tr>
<tr class="even">
<td align="left">Corynebacterium</td>
<td align="right">0.0280</td>
<td align="right">0.0035</td>
<td align="right">1.1857</td>
<td align="right">0.1037</td>
</tr>
<tr class="odd">
<td align="left">Desulfosporomusa</td>
<td align="right">0.0851</td>
<td align="right">0.0359</td>
<td align="right">-0.8604</td>
<td align="right">0.1733</td>
</tr>
<tr class="even">
<td align="left">Geobacillus</td>
<td align="right">0.0370</td>
<td align="right">0.0081</td>
<td align="right">-1.0962</td>
<td align="right">0.1293</td>
</tr>
<tr class="odd">
<td align="left">Jeotgalicoccus</td>
<td align="right">0.0276</td>
<td align="right">0.0251</td>
<td align="right">-0.9052</td>
<td align="right">0.1676</td>
</tr>
<tr class="even">
<td align="left">Paenibacillus</td>
<td align="right">0.0837</td>
<td align="right">0.0345</td>
<td align="right">-0.9380</td>
<td align="right">0.1932</td>
</tr>
<tr class="odd">
<td align="left">Virgibacillus</td>
<td align="right">0.1103</td>
<td align="right">0.0443</td>
<td align="right">-0.8750</td>
<td align="right">0.1960</td>
</tr>
</tbody>
</table>
</div>
<div id="ancom-bc" class="section level3 hasAnchor" number="11.1.4">
<h3><span class="header-section-number">11.1.4</span> ANCOM-BC<a href="differential-abundance.html#ancom-bc" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The analysis of composition of microbiomes with bias correction
(ANCOM-BC) <span class="citation">(Lin and Peddada 2020)</span> is a recently developed method for differential
abundance testing. It is based on an earlier published approach
<span class="citation">(Mandal et al. 2015)</span>. The previous version of ANCOM was among the methods
that produced the most consistent results and is probably a
conservative approach <span class="citation">(Nearing et al. 2022)</span>. However, the new ANCOM-BC
method operates quite differently compared to the former ANCOM method.</p>
<p>As the only method, ANCOM-BC incorporates the so called <em>sampling
fraction</em> into the model. The latter term could be empirically
estimated by the ratio of the library size to the microbial
load. According to the authors, variations in this sampling fraction
would bias differential abundance analyses if ignored. Furthermore,
this method provides p-values and confidence intervals for each
taxon. It also controls the FDR and it is computationally simple to
implement.</p>
<p>Note that the original method was implemented in the <code>ancombc()</code> function (see
<a href="https://www.bioconductor.org/packages/release/bioc/vignettes/ANCOMBC/inst/doc/ANCOMBC.html">extended tutorial</a>).
The method has since then been updated and new features have been added to enable
multi-group comparisons and repeated measurements among other improvements.
We do not cover the more advanced features of ANCOMBC in this tutorial
as these features are documented in detail in this
<a href="https://www.bioconductor.org/packages/release/bioc/vignettes/ANCOMBC/inst/doc/ANCOMBC2.html">tutorial</a>.</p>
<p>We now proceed with a simple example. First, we specify a formula. In this
formula, other covariates could potentially be included to adjust for
confounding. We show this further below. Again, please make sure to check the
<a href="https://rdrr.io/github/FrederickHuangLin/ANCOMBC/man/ancombc.html">function documentation</a>
as well as the linked tutorials to learn about the additional arguments
that we specify.</p>
<div class="sourceCode" id="cb345"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb345-1"><a href="differential-abundance.html#cb345-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Agglomerate data to genus level and add this new abundance table to the altExp slot</span></span>
<span id="cb345-2"><a href="differential-abundance.html#cb345-2" aria-hidden="true" tabindex="-1"></a><span class="fu">altExp</span>(tse, <span class="st">&quot;genus&quot;</span>) <span class="ot">&lt;-</span> <span class="fu">agglomerateByRank</span>(tse, <span class="st">&quot;genus&quot;</span>)</span>
<span id="cb345-3"><a href="differential-abundance.html#cb345-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb345-4"><a href="differential-abundance.html#cb345-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify prevalent genera</span></span>
<span id="cb345-5"><a href="differential-abundance.html#cb345-5" aria-hidden="true" tabindex="-1"></a>prevalent.genera <span class="ot">&lt;-</span> <span class="fu">getPrevalentFeatures</span>(<span class="fu">altExp</span>(tse, <span class="st">&quot;genus&quot;</span>), <span class="at">detection =</span> <span class="dv">0</span>, <span class="at">prevalence =</span> <span class="dv">30</span><span class="sc">/</span><span class="dv">100</span>)</span>
<span id="cb345-6"><a href="differential-abundance.html#cb345-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb345-7"><a href="differential-abundance.html#cb345-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Run ANCOM-BC at the genus level and only including the prevalent genera</span></span>
<span id="cb345-8"><a href="differential-abundance.html#cb345-8" aria-hidden="true" tabindex="-1"></a>out <span class="ot">&lt;-</span> <span class="fu">ancombc2</span>(</span>
<span id="cb345-9"><a href="differential-abundance.html#cb345-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> <span class="fu">altExp</span>(tse, <span class="st">&quot;genus&quot;</span>)[prevalent.genera, ],</span>
<span id="cb345-10"><a href="differential-abundance.html#cb345-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">assay_name =</span> <span class="st">&quot;counts&quot;</span>, </span>
<span id="cb345-11"><a href="differential-abundance.html#cb345-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">fix_formula =</span> <span class="st">&quot;Geographical_location&quot;</span>, </span>
<span id="cb345-12"><a href="differential-abundance.html#cb345-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">p_adj_method =</span> <span class="st">&quot;fdr&quot;</span>, </span>
<span id="cb345-13"><a href="differential-abundance.html#cb345-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">prv_cut =</span> <span class="dv">0</span>,</span>
<span id="cb345-14"><a href="differential-abundance.html#cb345-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">group =</span> <span class="st">&quot;Geographical_location&quot;</span>, </span>
<span id="cb345-15"><a href="differential-abundance.html#cb345-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">struc_zero =</span> <span class="cn">TRUE</span>, </span>
<span id="cb345-16"><a href="differential-abundance.html#cb345-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">neg_lb =</span> <span class="cn">TRUE</span>,</span>
<span id="cb345-17"><a href="differential-abundance.html#cb345-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">global =</span> <span class="cn">TRUE</span> <span class="co"># multi group comparison will be deactivated automatically </span></span>
<span id="cb345-18"><a href="differential-abundance.html#cb345-18" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<div class="sourceCode" id="cb346"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb346-1"><a href="differential-abundance.html#cb346-1" aria-hidden="true" tabindex="-1"></a><span class="co"># store the FDR adjusted results [test on v2.0.3] </span></span>
<span id="cb346-2"><a href="differential-abundance.html#cb346-2" aria-hidden="true" tabindex="-1"></a>ancombc_result <span class="ot">&lt;-</span> <span class="fu">cbind.data.frame</span>(<span class="at">taxid =</span> out<span class="sc">$</span>res<span class="sc">$</span>taxon,</span>
<span id="cb346-3"><a href="differential-abundance.html#cb346-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">ancombc =</span> <span class="fu">as.vector</span>(out<span class="sc">$</span>res<span class="sc">$</span>q_Geographical_locationPune))</span></code></pre></div>
<div class="sourceCode" id="cb347"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb347-1"><a href="differential-abundance.html#cb347-1" aria-hidden="true" tabindex="-1"></a><span class="co"># store the FDR adjusted results [test on v1.2.2]</span></span>
<span id="cb347-2"><a href="differential-abundance.html#cb347-2" aria-hidden="true" tabindex="-1"></a>ancombc_result <span class="ot">&lt;-</span> out<span class="sc">$</span>res <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">starts_with</span>(<span class="fu">c</span>(<span class="st">&quot;taxon&quot;</span>, <span class="st">&quot;lfc&quot;</span>, <span class="st">&quot;q&quot;</span>)))</span></code></pre></div>
<p>The object <code>out</code> contains all model output. Again, see the
<a href="https://rdrr.io/github/FrederickHuangLin/ANCOMBC/man/ancombc.html">documentation of the
function</a>
under <strong>Value</strong> for details. Our question whether taxa are
differentially abundant can be answered by looking at the <code>res</code>
object, which contains dataframes with the coefficients, standard
errors, p-values and q-values. Below we show the first entries of this
dataframe.</p>
<div class="sourceCode" id="cb348"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb348-1"><a href="differential-abundance.html#cb348-1" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(ancombc_result))</span></code></pre></div>
<table>
<colgroup>
<col width="15%" />
<col width="15%" />
<col width="28%" />
<col width="13%" />
<col width="26%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">taxon</th>
<th align="right">lfc_(Intercept)</th>
<th align="right">lfc_Geographical_locationPune</th>
<th align="right">q_(Intercept)</th>
<th align="right">q_Geographical_locationPune</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Abyssicoccus</td>
<td align="right">0.0904</td>
<td align="right">0.1619</td>
<td align="right">0.7617</td>
<td align="right">0.6060</td>
</tr>
<tr class="even">
<td align="left">Acidaminococcus</td>
<td align="right">0.2093</td>
<td align="right">-0.2665</td>
<td align="right">0.7148</td>
<td align="right">0.6066</td>
</tr>
<tr class="odd">
<td align="left">Acinetobacter</td>
<td align="right">1.5838</td>
<td align="right">-1.6285</td>
<td align="right">0.0861</td>
<td align="right">0.0725</td>
</tr>
<tr class="even">
<td align="left">Actinomyces</td>
<td align="right">0.2896</td>
<td align="right">-0.3084</td>
<td align="right">0.6344</td>
<td align="right">0.5269</td>
</tr>
<tr class="odd">
<td align="left">Aerococcus</td>
<td align="right">-0.2045</td>
<td align="right">0.6671</td>
<td align="right">0.5381</td>
<td align="right">0.0366</td>
</tr>
<tr class="even">
<td align="left">Aeromonas</td>
<td align="right">-0.3480</td>
<td align="right">0.6383</td>
<td align="right">0.5110</td>
<td align="right">0.1952</td>
</tr>
</tbody>
</table>
</div>
<div id="maaslin2" class="section level3 hasAnchor" number="11.1.5">
<h3><span class="header-section-number">11.1.5</span> MaAsLin2<a href="differential-abundance.html#maaslin2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Let us next illustrate MaAsLin2 <span class="citation">(Mallick, Rahnavard, and McIver 2020)</span>. This method is based on
generalized linear models and flexible for different study designs
and covariate structures. For details, check their
<a href="https://github.com/biobakery/biobakery/wiki/maaslin2">Biobakery tutorial</a>.</p>
<div class="sourceCode" id="cb349"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb349-1"><a href="differential-abundance.html#cb349-1" aria-hidden="true" tabindex="-1"></a><span class="co"># maaslin expects features as columns and samples as rows </span></span>
<span id="cb349-2"><a href="differential-abundance.html#cb349-2" aria-hidden="true" tabindex="-1"></a><span class="co"># for both the abundance table as well as metadata </span></span>
<span id="cb349-3"><a href="differential-abundance.html#cb349-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb349-4"><a href="differential-abundance.html#cb349-4" aria-hidden="true" tabindex="-1"></a><span class="co"># We can specify different GLMs/normalizations/transforms.</span></span>
<span id="cb349-5"><a href="differential-abundance.html#cb349-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Let us use similar settings as in Nearing et al. (2021):</span></span>
<span id="cb349-6"><a href="differential-abundance.html#cb349-6" aria-hidden="true" tabindex="-1"></a>maaslin2_out <span class="ot">&lt;-</span> <span class="fu">Maaslin2</span>(</span>
<span id="cb349-7"><a href="differential-abundance.html#cb349-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">t</span>(<span class="fu">assay</span>(tse)),</span>
<span id="cb349-8"><a href="differential-abundance.html#cb349-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="fu">colData</span>(tse)),</span>
<span id="cb349-9"><a href="differential-abundance.html#cb349-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">output =</span> <span class="st">&quot;DAA example&quot;</span>,</span>
<span id="cb349-10"><a href="differential-abundance.html#cb349-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">transform =</span> <span class="st">&quot;AST&quot;</span>,</span>
<span id="cb349-11"><a href="differential-abundance.html#cb349-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">fixed_effects =</span> <span class="st">&quot;Geographical_location&quot;</span>,</span>
<span id="cb349-12"><a href="differential-abundance.html#cb349-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># random_effects = c(...), # you can also fit MLM by specifying random effects</span></span>
<span id="cb349-13"><a href="differential-abundance.html#cb349-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># specifying a ref is especially important if you have more than 2 levels</span></span>
<span id="cb349-14"><a href="differential-abundance.html#cb349-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">reference =</span> <span class="st">&quot;Geographical_location,Pune&quot;</span>,  </span>
<span id="cb349-15"><a href="differential-abundance.html#cb349-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">normalization =</span> <span class="st">&quot;TSS&quot;</span>,</span>
<span id="cb349-16"><a href="differential-abundance.html#cb349-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">standardize =</span> <span class="cn">FALSE</span>,</span>
<span id="cb349-17"><a href="differential-abundance.html#cb349-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">min_prevalence =</span> <span class="dv">0</span> <span class="co"># prev filterin already done</span></span>
<span id="cb349-18"><a href="differential-abundance.html#cb349-18" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p>Which genera are identified as differentially abundant? (leave out “head” to see all).</p>
<div class="sourceCode" id="cb350"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb350-1"><a href="differential-abundance.html#cb350-1" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(<span class="fu">filter</span>(maaslin2_out<span class="sc">$</span>results, qval <span class="sc">&lt;=</span> <span class="fl">0.05</span>)))</span></code></pre></div>
<table style="width:100%;">
<colgroup>
<col width="16%" />
<col width="19%" />
<col width="6%" />
<col width="6%" />
<col width="6%" />
<col width="4%" />
<col width="24%" />
<col width="4%" />
<col width="2%" />
<col width="9%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">feature</th>
<th align="left">metadata</th>
<th align="left">value</th>
<th align="right">coef</th>
<th align="right">stderr</th>
<th align="right">pval</th>
<th align="left">name</th>
<th align="right">qval</th>
<th align="right">N</th>
<th align="right">N.not.zero</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Fructobacillus</td>
<td align="left">Geographical_location</td>
<td align="left">Nashik</td>
<td align="right">0.0080</td>
<td align="right">0.0011</td>
<td align="right">0</td>
<td align="left">Geographical_locationNashik</td>
<td align="right">0</td>
<td align="right">47</td>
<td align="right">9</td>
</tr>
<tr class="even">
<td align="left">Desulfosporomusa</td>
<td align="left">Geographical_location</td>
<td align="left">Nashik</td>
<td align="right">0.0373</td>
<td align="right">0.0059</td>
<td align="right">0</td>
<td align="left">Geographical_locationNashik</td>
<td align="right">0</td>
<td align="right">47</td>
<td align="right">13</td>
</tr>
<tr class="odd">
<td align="left">Geobacillus</td>
<td align="left">Geographical_location</td>
<td align="left">Nashik</td>
<td align="right">0.1294</td>
<td align="right">0.0207</td>
<td align="right">0</td>
<td align="left">Geographical_locationNashik</td>
<td align="right">0</td>
<td align="right">47</td>
<td align="right">27</td>
</tr>
<tr class="even">
<td align="left">Pullulanibacillus</td>
<td align="left">Geographical_location</td>
<td align="left">Nashik</td>
<td align="right">0.0395</td>
<td align="right">0.0062</td>
<td align="right">0</td>
<td align="left">Geographical_locationNashik</td>
<td align="right">0</td>
<td align="right">47</td>
<td align="right">9</td>
</tr>
<tr class="odd">
<td align="left">Chitinivibrio</td>
<td align="left">Geographical_location</td>
<td align="left">Nashik</td>
<td align="right">0.0274</td>
<td align="right">0.0045</td>
<td align="right">0</td>
<td align="left">Geographical_locationNashik</td>
<td align="right">0</td>
<td align="right">47</td>
<td align="right">10</td>
</tr>
<tr class="even">
<td align="left">Thermoanaerobacter</td>
<td align="left">Geographical_location</td>
<td align="left">Nashik</td>
<td align="right">0.0089</td>
<td align="right">0.0015</td>
<td align="right">0</td>
<td align="left">Geographical_locationNashik</td>
<td align="right">0</td>
<td align="right">47</td>
<td align="right">10</td>
</tr>
</tbody>
</table>
<p>This will create a folder that is called like in the output specified
above. It contains also figures to visualize difference between
significant taxa.</p>
</div>
<div id="linda" class="section level3 hasAnchor" number="11.1.6">
<h3><span class="header-section-number">11.1.6</span> LinDA<a href="differential-abundance.html#linda" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Lastly, we cover linear models for differential abundance analysis of
microbiome compositional data (<span class="citation">H. Zhou et al. (2022)</span>). This is very similar to
ANCOMBC with few differences: 1) LinDA correct for the compositional
bias differently using the mode of all regression coefficients. 2) it
is faster (100x-1000x than ANCOMBC and according to the authors); 3)
it supports hierarchical models. The latest ANCOMBC versions are also
supporting hierarchical models. Nevertheless, LinDA seems a promising
tool that achieves a very good power/fdr trade-off together with
ANCOMBC according to the review. The speed improvements might make it
critical especially for datasets that have higher sample or feature
set sizes.</p>
<div class="sourceCode" id="cb351"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb351-1"><a href="differential-abundance.html#cb351-1" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(tse)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(Geographical_location)</span>
<span id="cb351-2"><a href="differential-abundance.html#cb351-2" aria-hidden="true" tabindex="-1"></a>linda.res <span class="ot">&lt;-</span> <span class="fu">linda</span>(</span>
<span id="cb351-3"><a href="differential-abundance.html#cb351-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>(<span class="fu">assay</span>(tse)), </span>
<span id="cb351-4"><a href="differential-abundance.html#cb351-4" aria-hidden="true" tabindex="-1"></a>  meta, </span>
<span id="cb351-5"><a href="differential-abundance.html#cb351-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">formula =</span> <span class="st">&#39;~Geographical_location&#39;</span>, </span>
<span id="cb351-6"><a href="differential-abundance.html#cb351-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">alpha =</span> <span class="fl">0.05</span>, </span>
<span id="cb351-7"><a href="differential-abundance.html#cb351-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">prev.filter =</span> <span class="dv">0</span>, </span>
<span id="cb351-8"><a href="differential-abundance.html#cb351-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">mean.abund.filter =</span> <span class="dv">0</span>)</span></code></pre></div>
<pre><code>## 0  features are filtered!
## The filtered data has  47  samples and  262  features will be tested!
## Pseudo-count approach is used.
## Fit linear models ...
## Completed.</code></pre>
<div class="sourceCode" id="cb353"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb353-1"><a href="differential-abundance.html#cb353-1" aria-hidden="true" tabindex="-1"></a>linda_out <span class="ot">&lt;-</span> linda.res<span class="sc">$</span>output<span class="sc">$</span>Geographical_locationPune</span></code></pre></div>
<div class="sourceCode" id="cb354"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb354-1"><a href="differential-abundance.html#cb354-1" aria-hidden="true" tabindex="-1"></a><span class="co"># to scan the table for genera where H0 could be rejected:</span></span>
<span id="cb354-2"><a href="differential-abundance.html#cb354-2" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(<span class="fu">filter</span>(<span class="fu">as.data.frame</span>(linda_out), reject)))</span></code></pre></div>
<table>
<colgroup>
<col width="20%" />
<col width="11%" />
<col width="19%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="3%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">baseMean</th>
<th align="right">log2FoldChange</th>
<th align="right">lfcSE</th>
<th align="right">stat</th>
<th align="right">pvalue</th>
<th align="right">padj</th>
<th align="left">reject</th>
<th align="right">df</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Acidaminococcus</td>
<td align="right">1194.7</td>
<td align="right">-1.9084</td>
<td align="right">0.3579</td>
<td align="right">-5.332</td>
<td align="right">0.0000</td>
<td align="right">0.0000</td>
<td align="left">TRUE</td>
<td align="right">45</td>
</tr>
<tr class="even">
<td align="left">Aciditerrimonas</td>
<td align="right">393.5</td>
<td align="right">-0.6655</td>
<td align="right">0.2184</td>
<td align="right">-3.048</td>
<td align="right">0.0039</td>
<td align="right">0.0165</td>
<td align="left">TRUE</td>
<td align="right">45</td>
</tr>
<tr class="odd">
<td align="left">Actinomadura</td>
<td align="right">836.9</td>
<td align="right">-1.7985</td>
<td align="right">0.3596</td>
<td align="right">-5.001</td>
<td align="right">0.0000</td>
<td align="right">0.0001</td>
<td align="left">TRUE</td>
<td align="right">45</td>
</tr>
<tr class="even">
<td align="left">Agromyces</td>
<td align="right">938.1</td>
<td align="right">-1.7530</td>
<td align="right">0.3910</td>
<td align="right">-4.483</td>
<td align="right">0.0001</td>
<td align="right">0.0004</td>
<td align="left">TRUE</td>
<td align="right">45</td>
</tr>
<tr class="odd">
<td align="left">Aminivibrio</td>
<td align="right">416.9</td>
<td align="right">-0.7489</td>
<td align="right">0.2352</td>
<td align="right">-3.185</td>
<td align="right">0.0026</td>
<td align="right">0.0128</td>
<td align="left">TRUE</td>
<td align="right">45</td>
</tr>
<tr class="even">
<td align="left">Amycolatopsis</td>
<td align="right">556.1</td>
<td align="right">-0.9299</td>
<td align="right">0.3320</td>
<td align="right">-2.801</td>
<td align="right">0.0075</td>
<td align="right">0.0302</td>
<td align="left">TRUE</td>
<td align="right">45</td>
</tr>
</tbody>
</table>
</div>
<div id="zicoseq" class="section level3 hasAnchor" number="11.1.7">
<h3><span class="header-section-number">11.1.7</span> ZicoSeq<a href="differential-abundance.html#zicoseq" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Subsequently, we add a linear model and permutation-based method, see
details at <a href="https://cran.r-project.org/web/packages/GUniFrac/vignettes/ZicoSeq.html">tutorial</a>.</p>
<p>This approach has been assessed to exhibit high power and a low false
discovery rate, which has the following components:</p>
<ol style="list-style-type: decimal">
<li><p>Winsorization to decrease the influence of outliers;</p></li>
<li><p>Posterior sampling based on a beta mixture prior to address
sampling variability and zero inflation;</p></li>
<li><p>Reference-based multiple-stage normalization to address
compositional effects;</p></li>
</ol>
<div class="sourceCode" id="cb355"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb355-1"><a href="differential-abundance.html#cb355-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb355-2"><a href="differential-abundance.html#cb355-2" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(tse))</span>
<span id="cb355-3"><a href="differential-abundance.html#cb355-3" aria-hidden="true" tabindex="-1"></a>zicoseq.obj <span class="ot">&lt;-</span> GUniFrac<span class="sc">::</span><span class="fu">ZicoSeq</span>(<span class="at">meta.dat =</span> meta, </span>
<span id="cb355-4"><a href="differential-abundance.html#cb355-4" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">feature.dat =</span> <span class="fu">as.matrix</span>(<span class="fu">assay</span>(tse)),</span>
<span id="cb355-5"><a href="differential-abundance.html#cb355-5" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">grp.name =</span> <span class="st">&#39;Geographical_location&#39;</span>,</span>
<span id="cb355-6"><a href="differential-abundance.html#cb355-6" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">adj.name =</span> <span class="cn">NULL</span>, </span>
<span id="cb355-7"><a href="differential-abundance.html#cb355-7" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">feature.dat.type =</span> <span class="st">&#39;count&#39;</span>,</span>
<span id="cb355-8"><a href="differential-abundance.html#cb355-8" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">prev.filter =</span> <span class="dv">0</span>,</span>
<span id="cb355-9"><a href="differential-abundance.html#cb355-9" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">perm.no =</span> <span class="dv">999</span>,</span>
<span id="cb355-10"><a href="differential-abundance.html#cb355-10" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">mean.abund.filter =</span> <span class="dv">0</span>,</span>
<span id="cb355-11"><a href="differential-abundance.html#cb355-11" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">max.abund.filter =</span> <span class="dv">0</span>,</span>
<span id="cb355-12"><a href="differential-abundance.html#cb355-12" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">return.feature.dat =</span> T)</span></code></pre></div>
<pre><code>## 0  features are filtered!
## The data has  47  samples and  262  features will be tested!
## On average,  1  outlier counts will be replaced for each feature!
## Fitting beta mixture ...
## Finding the references ...
## Permutation testing ...
## ...................................................................................................
## ...................................................................................................
## ...................................................................................................
## ...................................................................................................
## ...................................................................................................
## ...................................................................................................
## Completed!</code></pre>
<div class="sourceCode" id="cb357"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb357-1"><a href="differential-abundance.html#cb357-1" aria-hidden="true" tabindex="-1"></a>zicoseq_out <span class="ot">&lt;-</span> <span class="fu">cbind.data.frame</span>(<span class="at">p.raw=</span>zicoseq.obj<span class="sc">$</span>p.raw, <span class="at">p.adj.fdr=</span>zicoseq.obj<span class="sc">$</span>p.adj.fdr) </span></code></pre></div>
<div class="sourceCode" id="cb358"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb358-1"><a href="differential-abundance.html#cb358-1" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(<span class="fu">filter</span>(zicoseq_out, p.adj.fdr<span class="sc">&lt;</span><span class="fl">0.05</span>)))</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">p.raw</th>
<th align="right">p.adj.fdr</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Actinomadura</td>
<td align="right">0.001</td>
<td align="right">0.0001</td>
</tr>
<tr class="even">
<td align="left">Agromyces</td>
<td align="right">0.001</td>
<td align="right">0.0038</td>
</tr>
<tr class="odd">
<td align="left">Aneurinibacillus</td>
<td align="right">0.001</td>
<td align="right">0.0071</td>
</tr>
<tr class="even">
<td align="left">Anoxybacillus</td>
<td align="right">0.001</td>
<td align="right">0.0001</td>
</tr>
<tr class="odd">
<td align="left">Chitinispirillum</td>
<td align="right">0.001</td>
<td align="right">0.0100</td>
</tr>
<tr class="even">
<td align="left">Chitinivibrio</td>
<td align="right">0.001</td>
<td align="right">0.0001</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb359"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb359-1"><a href="differential-abundance.html#cb359-1" aria-hidden="true" tabindex="-1"></a><span class="do">## x-axis is the effect size: R2 * direction of coefficient</span></span>
<span id="cb359-2"><a href="differential-abundance.html#cb359-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ZicoSeq.plot</span>(<span class="at">ZicoSeq.obj =</span> zicoseq.obj,</span>
<span id="cb359-3"><a href="differential-abundance.html#cb359-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">meta.dat =</span> meta,</span>
<span id="cb359-4"><a href="differential-abundance.html#cb359-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">pvalue.type =</span><span class="st">&#39;p.adj.fdr&#39;</span>)</span></code></pre></div>
<p><img src="30_differential_abundance_files/figure-html/ZicoSeqplot-1.png" width="672" /></p>
</div>
<div id="comparison-of-methods" class="section level3 hasAnchor" number="11.1.8">
<h3><span class="header-section-number">11.1.8</span> Comparison of methods<a href="differential-abundance.html#comparison-of-methods" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The different methods yield somewhat different results but they could
be also expected to overlap to a substantial degree. As an exercirse,
you can compare the outcomes between the different methods in terms of
effect sizes, significances, or other aspects that are comparable
between the methods.</p>
</div>
</div>
<div id="confounding-variables" class="section level2 hasAnchor" number="11.2">
<h2><span class="header-section-number">11.2</span> Confounding variables<a href="differential-abundance.html#confounding-variables" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Confounders are common in experimental research. In general, these can be
classified into 3 types:</p>
<ul>
<li><p>Biological confounder, such as age, sex, etc.</p></li>
<li><p>Technical confounder that caused by data collection, storage, DNA
extraction, sequencing process, etc.</p></li>
<li><p>Confounder caused by experimental models, such as cage effect,
sample background, etc.</p></li>
</ul>
<p>Adjusting confounder is necessary and important to reach a valid
conclusion. To perform causal inference, it is crucial that the method
is able to include covariates in the model. This is not possible with
e.g. the Wilcoxon test. Other methods such as DESeq2, edgeR, ANCOMBC,
LDM, Aldex2, Corncob, MaAsLin2, ZicoSeq, fastANCOM and ZINQ allow
this. Below we show how to include a confounder/covariate in ANCOMBC,
LinDA and ZicoSeq.</p>
<div id="ancombc" class="section level3 hasAnchor" number="11.2.1">
<h3><span class="header-section-number">11.2.1</span> ANCOMBC<a href="differential-abundance.html#ancombc" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb360"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb360-1"><a href="differential-abundance.html#cb360-1" aria-hidden="true" tabindex="-1"></a><span class="co"># perform the analysis </span></span>
<span id="cb360-2"><a href="differential-abundance.html#cb360-2" aria-hidden="true" tabindex="-1"></a>ancombc_cov <span class="ot">&lt;-</span> <span class="fu">ancombc2</span>(</span>
<span id="cb360-3"><a href="differential-abundance.html#cb360-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> tse,</span>
<span id="cb360-4"><a href="differential-abundance.html#cb360-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">assay_name =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb360-5"><a href="differential-abundance.html#cb360-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">tax_level =</span> <span class="st">&quot;genus&quot;</span>,</span>
<span id="cb360-6"><a href="differential-abundance.html#cb360-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">fix_formula =</span> <span class="st">&quot;Geographical_location + Age&quot;</span>, </span>
<span id="cb360-7"><a href="differential-abundance.html#cb360-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">p_adj_method =</span> <span class="st">&quot;fdr&quot;</span>, </span>
<span id="cb360-8"><a href="differential-abundance.html#cb360-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">lib_cut =</span> <span class="dv">0</span>, </span>
<span id="cb360-9"><a href="differential-abundance.html#cb360-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">group =</span> <span class="st">&quot;Geographical_location&quot;</span>, </span>
<span id="cb360-10"><a href="differential-abundance.html#cb360-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">struc_zero =</span> <span class="cn">TRUE</span>, </span>
<span id="cb360-11"><a href="differential-abundance.html#cb360-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">neg_lb =</span> <span class="cn">TRUE</span>,</span>
<span id="cb360-12"><a href="differential-abundance.html#cb360-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">alpha =</span> <span class="fl">0.05</span>, </span>
<span id="cb360-13"><a href="differential-abundance.html#cb360-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">global =</span> <span class="cn">TRUE</span> <span class="co"># multi group comparison will be deactivated automatically </span></span>
<span id="cb360-14"><a href="differential-abundance.html#cb360-14" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb360-15"><a href="differential-abundance.html#cb360-15" aria-hidden="true" tabindex="-1"></a><span class="co"># now the model answers the question: holding Age constant, are </span></span>
<span id="cb360-16"><a href="differential-abundance.html#cb360-16" aria-hidden="true" tabindex="-1"></a><span class="co"># bacterial taxa differentially abundant? Or, if that is of interest,</span></span>
<span id="cb360-17"><a href="differential-abundance.html#cb360-17" aria-hidden="true" tabindex="-1"></a><span class="co"># holding phenotype constant, is Age associated with bacterial abundance?</span></span>
<span id="cb360-18"><a href="differential-abundance.html#cb360-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Again we only show the first 6 entries.</span></span></code></pre></div>
<div class="sourceCode" id="cb361"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb361-1"><a href="differential-abundance.html#cb361-1" aria-hidden="true" tabindex="-1"></a>tab <span class="ot">&lt;-</span> ancombc_cov<span class="sc">$</span>res <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">starts_with</span>(<span class="fu">c</span>(<span class="st">&quot;taxon&quot;</span>, <span class="st">&quot;lfc&quot;</span>, <span class="st">&quot;q&quot;</span>)))</span>
<span id="cb361-2"><a href="differential-abundance.html#cb361-2" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(tab))</span></code></pre></div>
<table style="width:100%;">
<colgroup>
<col width="9%" />
<col width="9%" />
<col width="18%" />
<col width="9%" />
<col width="10%" />
<col width="8%" />
<col width="16%" />
<col width="7%" />
<col width="9%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">taxon</th>
<th align="right">lfc_(Intercept)</th>
<th align="right">lfc_Geographical_locationPune</th>
<th align="right">lfc_AgeElderly</th>
<th align="right">lfc_AgeMiddle_age</th>
<th align="right">q_(Intercept)</th>
<th align="right">q_Geographical_locationPune</th>
<th align="right">q_AgeElderly</th>
<th align="right">q_AgeMiddle_age</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Abyssicoccus</td>
<td align="right">0.0723</td>
<td align="right">0.0319</td>
<td align="right">0.3264</td>
<td align="right">0.2208</td>
<td align="right">0.7456</td>
<td align="right">0.9763</td>
<td align="right">0.7328</td>
<td align="right">1.0000</td>
</tr>
<tr class="even">
<td align="left">Acidaminococcus</td>
<td align="right">0.2364</td>
<td align="right">0.0584</td>
<td align="right">-0.6752</td>
<td align="right">-0.3702</td>
<td align="right">0.6251</td>
<td align="right">0.9721</td>
<td align="right">0.4517</td>
<td align="right">0.9279</td>
</tr>
<tr class="odd">
<td align="left">Acinetobacter</td>
<td align="right">1.5023</td>
<td align="right">-3.2362</td>
<td align="right">2.2397</td>
<td align="right">2.1993</td>
<td align="right">0.0833</td>
<td align="right">0.0013</td>
<td align="right">0.1015</td>
<td align="right">0.0639</td>
</tr>
<tr class="even">
<td align="left">Actinomyces</td>
<td align="right">0.3740</td>
<td align="right">0.0866</td>
<td align="right">-0.7335</td>
<td align="right">-0.6522</td>
<td align="right">0.3437</td>
<td align="right">0.9439</td>
<td align="right">0.4352</td>
<td align="right">0.7294</td>
</tr>
<tr class="odd">
<td align="left">Actinoplanes</td>
<td align="right">0.9135</td>
<td align="right">-1.1217</td>
<td align="right">-0.1497</td>
<td align="right">0.0000</td>
<td align="right">0.0020</td>
<td align="right">0.0461</td>
<td align="right">0.8662</td>
<td align="right">1.0000</td>
</tr>
<tr class="even">
<td align="left">Aerococcus</td>
<td align="right">-0.2178</td>
<td align="right">0.6432</td>
<td align="right">0.0922</td>
<td align="right">0.0864</td>
<td align="right">0.3378</td>
<td align="right">0.1375</td>
<td align="right">0.8662</td>
<td align="right">1.0000</td>
</tr>
</tbody>
</table>
</div>
<div id="linda-1" class="section level3 hasAnchor" number="11.2.2">
<h3><span class="header-section-number">11.2.2</span> LinDA<a href="differential-abundance.html#linda-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb362"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb362-1"><a href="differential-abundance.html#cb362-1" aria-hidden="true" tabindex="-1"></a>linda_cov <span class="ot">&lt;-</span> <span class="fu">linda</span>(</span>
<span id="cb362-2"><a href="differential-abundance.html#cb362-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>(<span class="fu">assay</span>(tse, <span class="st">&quot;counts&quot;</span>)), </span>
<span id="cb362-3"><a href="differential-abundance.html#cb362-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(tse)), </span>
<span id="cb362-4"><a href="differential-abundance.html#cb362-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">formula =</span> <span class="st">&#39;~ Geographical_location + Age&#39;</span>, </span>
<span id="cb362-5"><a href="differential-abundance.html#cb362-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">alpha =</span> <span class="fl">0.05</span>, </span>
<span id="cb362-6"><a href="differential-abundance.html#cb362-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">prev.filter =</span> <span class="dv">0</span>, </span>
<span id="cb362-7"><a href="differential-abundance.html#cb362-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">mean.abund.filter =</span> <span class="dv">0</span>)</span></code></pre></div>
<pre><code>## 0  features are filtered!
## The filtered data has  47  samples and  262  features will be tested!
## Pseudo-count approach is used.
## Fit linear models ...
## Completed.</code></pre>
<div class="sourceCode" id="cb364"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb364-1"><a href="differential-abundance.html#cb364-1" aria-hidden="true" tabindex="-1"></a>linda.res <span class="ot">&lt;-</span> linda_cov<span class="sc">$</span>output<span class="sc">$</span>Geographical_locationPune</span></code></pre></div>
<div class="sourceCode" id="cb365"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb365-1"><a href="differential-abundance.html#cb365-1" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(<span class="fu">filter</span>(linda.res, reject<span class="sc">==</span>T)))</span></code></pre></div>
<table>
<colgroup>
<col width="21%" />
<col width="11%" />
<col width="18%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="3%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">baseMean</th>
<th align="right">log2FoldChange</th>
<th align="right">lfcSE</th>
<th align="right">stat</th>
<th align="right">pvalue</th>
<th align="right">padj</th>
<th align="left">reject</th>
<th align="right">df</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Acidaminococcus</td>
<td align="right">1140.5</td>
<td align="right">-1.438</td>
<td align="right">0.4390</td>
<td align="right">-3.276</td>
<td align="right">0.0021</td>
<td align="right">0.0182</td>
<td align="left">TRUE</td>
<td align="right">43</td>
</tr>
<tr class="even">
<td align="left">Actinomadura</td>
<td align="right">804.7</td>
<td align="right">-1.520</td>
<td align="right">0.4477</td>
<td align="right">-3.394</td>
<td align="right">0.0015</td>
<td align="right">0.0139</td>
<td align="left">TRUE</td>
<td align="right">43</td>
</tr>
<tr class="odd">
<td align="left">Anaerococcus</td>
<td align="right">939.0</td>
<td align="right">6.795</td>
<td align="right">1.3978</td>
<td align="right">4.861</td>
<td align="right">0.0000</td>
<td align="right">0.0004</td>
<td align="left">TRUE</td>
<td align="right">43</td>
</tr>
<tr class="even">
<td align="left">Aneurinibacillus</td>
<td align="right">988.0</td>
<td align="right">-1.687</td>
<td align="right">0.5426</td>
<td align="right">-3.110</td>
<td align="right">0.0033</td>
<td align="right">0.0256</td>
<td align="left">TRUE</td>
<td align="right">43</td>
</tr>
<tr class="odd">
<td align="left">Anoxybacillus</td>
<td align="right">1712.8</td>
<td align="right">-2.727</td>
<td align="right">0.5574</td>
<td align="right">-4.893</td>
<td align="right">0.0000</td>
<td align="right">0.0004</td>
<td align="left">TRUE</td>
<td align="right">43</td>
</tr>
<tr class="even">
<td align="left">Brachybacterium</td>
<td align="right">403.0</td>
<td align="right">2.060</td>
<td align="right">0.7303</td>
<td align="right">2.821</td>
<td align="right">0.0072</td>
<td align="right">0.0450</td>
<td align="left">TRUE</td>
<td align="right">43</td>
</tr>
</tbody>
</table>
</div>
<div id="zicoseq-1" class="section level3 hasAnchor" number="11.2.3">
<h3><span class="header-section-number">11.2.3</span> ZicoSeq<a href="differential-abundance.html#zicoseq-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb366"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb366-1"><a href="differential-abundance.html#cb366-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb366-2"><a href="differential-abundance.html#cb366-2" aria-hidden="true" tabindex="-1"></a>zicoseq.obj <span class="ot">&lt;-</span> GUniFrac<span class="sc">::</span><span class="fu">ZicoSeq</span>(<span class="at">meta.dat =</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(tse)) , </span>
<span id="cb366-3"><a href="differential-abundance.html#cb366-3" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">feature.dat =</span> <span class="fu">as.matrix</span>(<span class="fu">assay</span>(tse)),</span>
<span id="cb366-4"><a href="differential-abundance.html#cb366-4" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">grp.name =</span> <span class="st">&#39;Geographical_location&#39;</span>,</span>
<span id="cb366-5"><a href="differential-abundance.html#cb366-5" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">adj.name =</span> <span class="st">&#39;Gender&#39;</span>, </span>
<span id="cb366-6"><a href="differential-abundance.html#cb366-6" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">feature.dat.type =</span> <span class="st">&#39;count&#39;</span>,</span>
<span id="cb366-7"><a href="differential-abundance.html#cb366-7" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">prev.filter =</span> <span class="dv">0</span>,</span>
<span id="cb366-8"><a href="differential-abundance.html#cb366-8" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">perm.no =</span> <span class="dv">999</span>,</span>
<span id="cb366-9"><a href="differential-abundance.html#cb366-9" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">mean.abund.filter =</span> <span class="dv">0</span>,</span>
<span id="cb366-10"><a href="differential-abundance.html#cb366-10" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">max.abund.filter =</span> <span class="dv">0</span>,</span>
<span id="cb366-11"><a href="differential-abundance.html#cb366-11" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">return.feature.dat =</span> T)</span></code></pre></div>
<pre><code>## 0  features are filtered!
## The data has  47  samples and  262  features will be tested!
## On average,  1  outlier counts will be replaced for each feature!
## Fitting beta mixture ...
## Finding the references ...
## Permutation testing ...
## ...................................................................................................
## ...................................................................................................
## ...................................................................................................
## ...................................................................................................
## ...................................................................................................
## ...................................................................................................
## Completed!</code></pre>
<div class="sourceCode" id="cb368"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb368-1"><a href="differential-abundance.html#cb368-1" aria-hidden="true" tabindex="-1"></a>zicoseq_out <span class="ot">&lt;-</span> <span class="fu">cbind.data.frame</span>(<span class="at">p.raw=</span>zicoseq.obj<span class="sc">$</span>p.raw,</span>
<span id="cb368-2"><a href="differential-abundance.html#cb368-2" aria-hidden="true" tabindex="-1"></a>                                <span class="at">p.adj.fdr=</span>zicoseq.obj<span class="sc">$</span>p.adj.fdr) </span></code></pre></div>
<div class="sourceCode" id="cb369"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb369-1"><a href="differential-abundance.html#cb369-1" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(<span class="fu">filter</span>(zicoseq_out, p.adj.fdr<span class="sc">&lt;</span><span class="fl">0.05</span>)))</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">p.raw</th>
<th align="right">p.adj.fdr</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Actinomadura</td>
<td align="right">0.001</td>
<td align="right">0.0004</td>
</tr>
<tr class="even">
<td align="left">Agromyces</td>
<td align="right">0.001</td>
<td align="right">0.0030</td>
</tr>
<tr class="odd">
<td align="left">Aneurinibacillus</td>
<td align="right">0.001</td>
<td align="right">0.0036</td>
</tr>
<tr class="even">
<td align="left">Anoxybacillus</td>
<td align="right">0.001</td>
<td align="right">0.0001</td>
</tr>
<tr class="odd">
<td align="left">Chitinispirillum</td>
<td align="right">0.001</td>
<td align="right">0.0089</td>
</tr>
<tr class="even">
<td align="left">Chitinivibrio</td>
<td align="right">0.001</td>
<td align="right">0.0001</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="tree-based-methods" class="section level2 hasAnchor" number="11.3">
<h2><span class="header-section-number">11.3</span> Tree-based methods<a href="differential-abundance.html#tree-based-methods" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Let us next cover phylogeny-aware methods to perform group-wise
associations.</p>
<div id="group-wise-associations-testing-based-on-balances" class="section level3 hasAnchor" number="11.3.1">
<h3><span class="header-section-number">11.3.1</span> Group-wise associations testing based on balances<a href="differential-abundance.html#group-wise-associations-testing-based-on-balances" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For testing associations based on balances, check the philr
R/Bioconductor package.</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="clustering.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="machine_learning.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/microbiome/OMA/blob/master/30_differential_abundance.Rmd",
"text": "Edit"
},
"history": {
"link": "https://github.com/microbiome/OMA/commits/master/30_differential_abundance.Rmd",
"text": null
},
"view": {
"link": "https://github.com/microbiome/OMA/blob/master/30_differential_abundance.Rmd",
"text": null
},
"download": ["OMA.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
