<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 Differential Abundance | Orchestrating Microbiome Analysis</title>
  <meta name="description" content="Chapter 9 Differential Abundance | Orchestrating Microbiome Analysis" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 Differential Abundance | Orchestrating Microbiome Analysis" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://github.com/microbiome/OMA" />
  
  
  <meta name="github-repo" content="microbiome/OMA" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 Differential Abundance | Orchestrating Microbiome Analysis" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="microbiome-community.html"/>
<link rel="next" href="microbiome-timeseries.html"/>
<script src="libs/header-attrs-2.9/header-attrs.js"></script>
<script src="libs/jquery-3.5.1/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<script src="libs/plotly-binding-4.9.4.1/plotly.js"></script>
<script src="libs/typedarray-0.1/typedarray.min.js"></script>
<link href="libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
<link href="libs/plotly-htmlwidgets-css-1.57.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="libs/plotly-main-1.57.1/plotly-latest.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Orchestrating Microbiome Analysis</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="" data-path="introduction-to-bioconductor-classes.html"><a href="introduction-to-bioconductor-classes.html"><i class="fa fa-check"></i>Introduction to Bioconductor Classes</a></li>
<li class="part"><span><b>I Introduction</b></span></li>
<li class="chapter" data-level="1" data-path="data-introduction.html"><a href="data-introduction.html"><i class="fa fa-check"></i><b>1</b> Data Infrastructure</a>
<ul>
<li class="chapter" data-level="1.1" data-path="data-introduction.html"><a href="data-introduction.html#installation"><i class="fa fa-check"></i><b>1.1</b> Installation</a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="data-introduction.html"><a href="data-introduction.html#packages"><i class="fa fa-check"></i><b>1.1.1</b> Packages</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="data-introduction.html"><a href="data-introduction.html#background"><i class="fa fa-check"></i><b>1.2</b> Background</a></li>
<li class="chapter" data-level="1.3" data-path="data-introduction.html"><a href="data-introduction.html#loading-experimental-microbiome-data"><i class="fa fa-check"></i><b>1.3</b> Loading experimental microbiome data</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="data-introduction.html"><a href="data-introduction.html#importing-data-from-external-files"><i class="fa fa-check"></i><b>1.3.1</b> Importing data from external files</a></li>
<li class="chapter" data-level="1.3.2" data-path="data-introduction.html"><a href="data-introduction.html#conversions-between-data-formats-in-r"><i class="fa fa-check"></i><b>1.3.2</b> Conversions between data formats in R</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="data-introduction.html"><a href="data-introduction.html#metadata"><i class="fa fa-check"></i><b>1.4</b> Metadata</a></li>
<li class="chapter" data-level="1.5" data-path="data-introduction.html"><a href="data-introduction.html#microbiome-and-tree-data-specific-aspects"><i class="fa fa-check"></i><b>1.5</b> Microbiome and tree data specific aspects</a>
<ul>
<li class="chapter" data-level="1.5.1" data-path="data-introduction.html"><a href="data-introduction.html#assays"><i class="fa fa-check"></i><b>1.5.1</b> Assays</a></li>
<li class="chapter" data-level="1.5.2" data-path="data-introduction.html"><a href="data-introduction.html#coldata"><i class="fa fa-check"></i><b>1.5.2</b> colData</a></li>
<li class="chapter" data-level="1.5.3" data-path="data-introduction.html"><a href="data-introduction.html#rowdata"><i class="fa fa-check"></i><b>1.5.3</b> rowData</a></li>
<li class="chapter" data-level="1.5.4" data-path="data-introduction.html"><a href="data-introduction.html#rowtree"><i class="fa fa-check"></i><b>1.5.4</b> rowTree</a></li>
</ul></li>
<li class="chapter" data-level="1.6" data-path="data-introduction.html"><a href="data-introduction.html#data-conversion"><i class="fa fa-check"></i><b>1.6</b> Data conversion</a>
<ul>
<li class="chapter" data-level="1.6.1" data-path="data-introduction.html"><a href="data-introduction.html#tidy-data"><i class="fa fa-check"></i><b>1.6.1</b> Tidy data</a></li>
</ul></li>
<li class="chapter" data-level="1.7" data-path="data-introduction.html"><a href="data-introduction.html#conclusion"><i class="fa fa-check"></i><b>1.7</b> Conclusion</a></li>
<li class="chapter" data-level="1.8" data-path="data-introduction.html"><a href="data-introduction.html#additional-reading"><i class="fa fa-check"></i><b>1.8</b> Additional Reading</a>
<ul>
<li class="chapter" data-level="1.8.1" data-path="data-introduction.html"><a href="data-introduction.html#lecture-slides"><i class="fa fa-check"></i><b>1.8.1</b> Lecture slides</a></li>
<li class="chapter" data-level="1.8.2" data-path="data-introduction.html"><a href="data-introduction.html#r-programming-resources"><i class="fa fa-check"></i><b>1.8.2</b> R programming resources</a></li>
<li class="chapter" data-level="1.8.3" data-path="data-introduction.html"><a href="data-introduction.html#resources-for-treesummarizedexperiment"><i class="fa fa-check"></i><b>1.8.3</b> Resources for TreeSummarizedExperiment</a></li>
<li class="chapter" data-level="1.8.4" data-path="data-introduction.html"><a href="data-introduction.html#resources-for-phyloseq"><i class="fa fa-check"></i><b>1.8.4</b> Resources for phyloseq</a></li>
<li class="chapter" data-level="1.8.5" data-path="data-introduction.html"><a href="data-introduction.html#further-reading"><i class="fa fa-check"></i><b>1.8.5</b> Further reading</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-introduction.html"><a href="data-introduction.html#session-info"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="part"><span><b>II Focus Topics</b></span></li>
<li class="chapter" data-level="2" data-path="overview.html"><a href="overview.html"><i class="fa fa-check"></i><b>2</b> Overview</a>
<ul>
<li class="chapter" data-level="2.1" data-path="overview.html"><a href="overview.html#introduction"><i class="fa fa-check"></i><b>2.1</b> Introduction</a></li>
<li class="chapter" data-level="2.2" data-path="overview.html"><a href="overview.html#microbiome-experiment-types"><i class="fa fa-check"></i><b>2.2</b> Microbiome experiment types</a></li>
<li class="chapter" data-level="2.3" data-path="overview.html"><a href="overview.html#processing-steps"><i class="fa fa-check"></i><b>2.3</b> Processing steps</a></li>
<li class="chapter" data-level="2.4" data-path="overview.html"><a href="overview.html#quick-start"><i class="fa fa-check"></i><b>2.4</b> Quick Start</a></li>
<li class="chapter" data-level="" data-path="overview.html"><a href="overview.html#session-info-1"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="taxonomic-information.html"><a href="taxonomic-information.html"><i class="fa fa-check"></i><b>3</b> Taxonomic Information</a>
<ul>
<li class="chapter" data-level="3.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#assigning-taxonomic-information."><i class="fa fa-check"></i><b>3.1</b> Assigning taxonomic information.</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#dada2"><i class="fa fa-check"></i><b>3.1.1</b> dada2</a></li>
<li class="chapter" data-level="3.1.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#decipher"><i class="fa fa-check"></i><b>3.1.2</b> DECIPHER</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#functions-to-access-taxonomic-information"><i class="fa fa-check"></i><b>3.2</b> Functions to access taxonomic information</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#generate-a-taxonomic-tree-on-the-fly"><i class="fa fa-check"></i><b>3.2.1</b> Generate a taxonomic tree on the fly</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="taxonomic-information.html"><a href="taxonomic-information.html#data-agglomeration"><i class="fa fa-check"></i><b>3.3</b> Data agglomeration</a></li>
<li class="chapter" data-level="3.4" data-path="taxonomic-information.html"><a href="taxonomic-information.html#data-transformation"><i class="fa fa-check"></i><b>3.4</b> Data transformation</a></li>
<li class="chapter" data-level="3.5" data-path="taxonomic-information.html"><a href="taxonomic-information.html#pick-specific"><i class="fa fa-check"></i><b>3.5</b> Pick specific</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#abundances-of-all-taxa-in-specific-sample"><i class="fa fa-check"></i><b>3.5.1</b> Abundances of all taxa in specific sample</a></li>
<li class="chapter" data-level="3.5.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#abundances-of-specific-taxa-in-all-samples"><i class="fa fa-check"></i><b>3.5.2</b> Abundances of specific taxa in all samples</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="taxonomic-information.html"><a href="taxonomic-information.html#session-info-2"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>4</b> Quality Control</a>
<ul>
<li class="chapter" data-level="4.1" data-path="quality-control.html"><a href="quality-control.html#get-top-taxa-and-taxonomy"><i class="fa fa-check"></i><b>4.1</b> Get top taxa and taxonomy</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="quality-control.html"><a href="quality-control.html#features"><i class="fa fa-check"></i><b>4.1.1</b> Features</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="quality-control.html"><a href="quality-control.html#library-size"><i class="fa fa-check"></i><b>4.2</b> Library size</a></li>
<li class="chapter" data-level="" data-path="quality-control.html"><a href="quality-control.html#session-info-3"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html"><i class="fa fa-check"></i><b>5</b> Microbiome Exploration</a>
<ul>
<li class="chapter" data-level="5.1" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#prevalence"><i class="fa fa-check"></i><b>5.1</b> Prevalence</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#prevalent-microbiota-analysis"><i class="fa fa-check"></i><b>5.1.1</b> Prevalent microbiota analysis</a></li>
<li class="chapter" data-level="5.1.2" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#plotting-prevalence"><i class="fa fa-check"></i><b>5.1.2</b> Plotting prevalence</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#session-info-4"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="alpha-diversity.html"><a href="alpha-diversity.html"><i class="fa fa-check"></i><b>6</b> Alpha Diversity</a>
<ul>
<li class="chapter" data-level="6.1" data-path="alpha-diversity.html"><a href="alpha-diversity.html#estimating-alpha-diversity"><i class="fa fa-check"></i><b>6.1</b> Estimating alpha diversity</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="alpha-diversity.html"><a href="alpha-diversity.html#richness"><i class="fa fa-check"></i><b>6.1.1</b> Richness</a></li>
<li class="chapter" data-level="6.1.2" data-path="alpha-diversity.html"><a href="alpha-diversity.html#diversity"><i class="fa fa-check"></i><b>6.1.2</b> Diversity</a></li>
<li class="chapter" data-level="6.1.3" data-path="alpha-diversity.html"><a href="alpha-diversity.html#evenness"><i class="fa fa-check"></i><b>6.1.3</b> Evenness</a></li>
<li class="chapter" data-level="6.1.4" data-path="alpha-diversity.html"><a href="alpha-diversity.html#dominance"><i class="fa fa-check"></i><b>6.1.4</b> Dominance</a></li>
<li class="chapter" data-level="6.1.5" data-path="alpha-diversity.html"><a href="alpha-diversity.html#rarity"><i class="fa fa-check"></i><b>6.1.5</b> Rarity</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="alpha-diversity.html"><a href="alpha-diversity.html#visualize-alpha-diversities"><i class="fa fa-check"></i><b>6.2</b> Visualize alpha diversities</a></li>
<li class="chapter" data-level="" data-path="alpha-diversity.html"><a href="alpha-diversity.html#session-info-5"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="beta-diversity.html"><a href="beta-diversity.html"><i class="fa fa-check"></i><b>7</b> Beta Diversity</a>
<ul>
<li class="chapter" data-level="7.1" data-path="beta-diversity.html"><a href="beta-diversity.html#explained-variance"><i class="fa fa-check"></i><b>7.1</b> Explained variance</a></li>
<li class="chapter" data-level="7.2" data-path="beta-diversity.html"><a href="beta-diversity.html#estimating-beta-diversity"><i class="fa fa-check"></i><b>7.2</b> Estimating beta diversity</a></li>
<li class="chapter" data-level="7.3" data-path="beta-diversity.html"><a href="beta-diversity.html#other-ordination-methods"><i class="fa fa-check"></i><b>7.3</b> Other ordination methods</a></li>
<li class="chapter" data-level="7.4" data-path="beta-diversity.html"><a href="beta-diversity.html#visualizing-the-most-dominant-genus-on-pcoa"><i class="fa fa-check"></i><b>7.4</b> Visualizing the most dominant genus on PCoA</a></li>
<li class="chapter" data-level="7.5" data-path="beta-diversity.html"><a href="beta-diversity.html#community-comparisons-todo-combine-with-the-material-above-for-simplicity"><i class="fa fa-check"></i><b>7.5</b> Community comparisons [TODO combine with the material above for simplicity?]</a>
<ul>
<li class="chapter" data-level="7.5.1" data-path="beta-diversity.html"><a href="beta-diversity.html#testing-differences-in-community-composition-between-sample-groups"><i class="fa fa-check"></i><b>7.5.1</b> Testing differences in community composition between sample groups</a></li>
<li class="chapter" data-level="7.5.2" data-path="beta-diversity.html"><a href="beta-diversity.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>7.5.2</b> Checking the homogeneity condition</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="beta-diversity.html"><a href="beta-diversity.html#further-reading-1"><i class="fa fa-check"></i><b>7.6</b> Further reading</a></li>
<li class="chapter" data-level="" data-path="beta-diversity.html"><a href="beta-diversity.html#session-info-6"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="microbiome-community.html"><a href="microbiome-community.html"><i class="fa fa-check"></i><b>8</b> Microbiome Community</a>
<ul>
<li class="chapter" data-level="8.1" data-path="microbiome-community.html"><a href="microbiome-community.html#community-composition"><i class="fa fa-check"></i><b>8.1</b> Community composition</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="microbiome-community.html"><a href="microbiome-community.html#composition-barplot"><i class="fa fa-check"></i><b>8.1.1</b> Composition barplot</a></li>
<li class="chapter" data-level="8.1.2" data-path="microbiome-community.html"><a href="microbiome-community.html#composition-heatmap"><i class="fa fa-check"></i><b>8.1.2</b> Composition heatmap</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="microbiome-community.html"><a href="microbiome-community.html#community-typing"><i class="fa fa-check"></i><b>8.2</b> Community typing</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="microbiome-community.html"><a href="microbiome-community.html#dirichlet-multinomial-mixtures-dmm"><i class="fa fa-check"></i><b>8.2.1</b> Dirichlet Multinomial Mixtures (DMM)</a></li>
<li class="chapter" data-level="8.2.2" data-path="microbiome-community.html"><a href="microbiome-community.html#pcoa-for-asv-level-data-with-bray-curtis-with-dmm-clusters-shown-with-colors"><i class="fa fa-check"></i><b>8.2.2</b> PCoA for ASV-level data with Bray-Curtis; with DMM clusters shown with colors</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="microbiome-community.html"><a href="microbiome-community.html#session-info-7"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="differential-abundance.html"><a href="differential-abundance.html"><i class="fa fa-check"></i><b>9</b> Differential Abundance</a>
<ul>
<li class="chapter" data-level="9.1" data-path="differential-abundance.html"><a href="differential-abundance.html#tree-based-methods"><i class="fa fa-check"></i><b>9.1</b> Tree-based methods</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="differential-abundance.html"><a href="differential-abundance.html#group-wise-associations-testing-based-on-balances"><i class="fa fa-check"></i><b>9.1.1</b> Group-wise associations testing based on balances</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="differential-abundance.html"><a href="differential-abundance.html#session-info-8"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="microbiome-timeseries.html"><a href="microbiome-timeseries.html"><i class="fa fa-check"></i><b>10</b> Microbiome Time Series</a>
<ul>
<li class="chapter" data-level="10.1" data-path="microbiome-timeseries.html"><a href="microbiome-timeseries.html#stability"><i class="fa fa-check"></i><b>10.1</b> Stability</a></li>
<li class="chapter" data-level="10.2" data-path="microbiome-timeseries.html"><a href="microbiome-timeseries.html#tipping-elements"><i class="fa fa-check"></i><b>10.2</b> Tipping elements</a></li>
<li class="chapter" data-level="" data-path="microbiome-timeseries.html"><a href="microbiome-timeseries.html#session-info-9"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="multitable.html"><a href="multitable.html"><i class="fa fa-check"></i><b>11</b> Representation of Multiple Data Tables</a>
<ul>
<li class="chapter" data-level="11.1" data-path="multitable.html"><a href="multitable.html#assay-data"><i class="fa fa-check"></i><b>11.1</b> Assay data</a></li>
<li class="chapter" data-level="11.2" data-path="multitable.html"><a href="multitable.html#alternative-experiments"><i class="fa fa-check"></i><b>11.2</b> Alternative experiments</a></li>
<li class="chapter" data-level="11.3" data-path="multitable.html"><a href="multitable.html#multiassayexperiments"><i class="fa fa-check"></i><b>11.3</b> MultiAssayExperiments</a></li>
<li class="chapter" data-level="" data-path="multitable.html"><a href="multitable.html#session-info-10"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="part"><span><b>III Appendix</b></span></li>
<li class="chapter" data-level="12" data-path="example-data.html"><a href="example-data.html"><i class="fa fa-check"></i><b>12</b> Example Data</a>
<ul>
<li class="chapter" data-level="12.1" data-path="example-data.html"><a href="example-data.html#package-data"><i class="fa fa-check"></i><b>12.1</b> Package data</a>
<ul>
<li class="chapter" data-level="12.1.1" data-path="example-data.html"><a href="example-data.html#globalpatterns"><i class="fa fa-check"></i><b>12.1.1</b> GlobalPatterns</a></li>
<li class="chapter" data-level="12.1.2" data-path="example-data.html"><a href="example-data.html#enterotype"><i class="fa fa-check"></i><b>12.1.2</b> Enterotype</a></li>
<li class="chapter" data-level="12.1.3" data-path="example-data.html"><a href="example-data.html#esophagus"><i class="fa fa-check"></i><b>12.1.3</b> Esophagus</a></li>
<li class="chapter" data-level="12.1.4" data-path="example-data.html"><a href="example-data.html#soilrep"><i class="fa fa-check"></i><b>12.1.4</b> Soilrep</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="example-data.html"><a href="example-data.html#session-info-11"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="part"><span><b>IV Appendix</b></span></li>
<li class="chapter" data-level="13" data-path="function-ref.html"><a href="function-ref.html"><i class="fa fa-check"></i><b>13</b> Function Reference</a>
<ul>
<li class="chapter" data-level="13.1" data-path="function-ref.html"><a href="function-ref.html#objects"><i class="fa fa-check"></i><b>13.1</b> Objects</a></li>
<li class="chapter" data-level="13.2" data-path="function-ref.html"><a href="function-ref.html#reduced-dimension-pca-t-sne-umap-mds-etc"><i class="fa fa-check"></i><b>13.2</b> Reduced dimension (PCA, t-SNE, UMAP, MDS, etc)</a></li>
<li class="chapter" data-level="13.3" data-path="function-ref.html"><a href="function-ref.html#plotting"><i class="fa fa-check"></i><b>13.3</b> Plotting</a></li>
</ul></li>
<li class="part"><span><b>V Appendix</b></span></li>
<li class="chapter" data-level="14" data-path="extras.html"><a href="extras.html"><i class="fa fa-check"></i><b>14</b> Extras</a>
<ul>
<li class="chapter" data-level="14.1" data-path="extras.html"><a href="extras.html#interactive-3d-plots"><i class="fa fa-check"></i><b>14.1</b> Interactive 3D Plots</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="bibliography.html"><a href="bibliography.html"><i class="fa fa-check"></i>Bibliography</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Orchestrating Microbiome Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="differential-abundance" class="section level1" number="9">
<h1><span class="header-section-number">Chapter 9</span> Differential Abundance</h1>
<script>
document.addEventListener("click", function (event) {
    if (event.target.classList.contains("rebook-collapse")) {
        event.target.classList.toggle("active");
        var content = event.target.nextElementSibling;
        if (content.style.display === "block") {
            content.style.display = "none";
        } else {
            content.style.display = "block";
        }
    }
})
</script>
<style>
.rebook-collapse {
  background-color: #eee;
  color: #444;
  cursor: pointer;
  padding: 18px;
  width: 100%;
  border: none;
  text-align: left;
  outline: none;
  font-size: 15px;
}

.rebook-content {
  padding: 0 18px;
  display: none;
  overflow: hidden;
  background-color: #f1f1f1;
}
</style>
<p>A number of methods for differential abundance analysis are available,
and reviewed elsewhere.</p>
<div id="tree-based-methods" class="section level2" number="9.1">
<h2><span class="header-section-number">9.1</span> Tree-based methods</h2>
<div id="group-wise-associations-testing-based-on-balances" class="section level3" number="9.1.1">
<h3><span class="header-section-number">9.1.1</span> Group-wise associations testing based on balances</h3>
<p><a href="https://bioconductor.org/packages/release/bioc/html/TreeSummarizedExperiment.html">TreeSummarizedExperiment</a>
frequently includes a Phylogenetic tree along with associated data about the experiment (at <code>colData</code>), that holds covariates which can be used for analyzing group-wise associations.</p>
<p>Such an analysis could be performed with the function <code>pibble</code> from the <code>fido</code> package, that offers a Multinomial Logistic-Normal Linear Regression model; see <a href="https://jsilve24.github.io/fido/articles/introduction-to-fido.html">vignette</a> of package.</p>
<p>The following presents such an exemplary analysis based on the <a href="https://doi.org/10.1038/s41467-020-17541-6">Sprockett et al. (2020)</a> available through <code>microbiomeDataSets</code> package.</p>
<div class="sourceCode" id="cb199"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb199-1"><a href="differential-abundance.html#cb199-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(fido)){</span>
<span id="cb199-2"><a href="differential-abundance.html#cb199-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># installing the fido package</span></span>
<span id="cb199-3"><a href="differential-abundance.html#cb199-3" aria-hidden="true" tabindex="-1"></a>  devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;jsilve24/fido&quot;</span>)</span>
<span id="cb199-4"><a href="differential-abundance.html#cb199-4" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<pre><code>## driver       (NA -&gt; 16e449946...) [GitHub]
## distribut... (NA -&gt; 0.2.2       ) [CRAN]
## tensorA      (NA -&gt; 0.36.2      ) [CRAN]
## checkmate    (NA -&gt; 2.0.0       ) [CRAN]
## svUnit       (NA -&gt; 1.0.6       ) [CRAN]
## HDInterval   (NA -&gt; 0.2.2       ) [CRAN]
## RcppGSL      (NA -&gt; 0.3.9       ) [CRAN]
## posterior    (NA -&gt; 1.0.1       ) [CRAN]
## arrayhelpers (NA -&gt; 1.1-0       ) [CRAN]
## coda         (NA -&gt; 0.19-4      ) [CRAN]
## ggdist       (NA -&gt; 3.0.0       ) [CRAN]
## RcppZiggurat (NA -&gt; 0.1.6       ) [CRAN]
## RcppNumer... (NA -&gt; 0.4-0       ) [CRAN]
## tidybayes    (NA -&gt; 3.0.0       ) [CRAN]
## 
##   
   checking for file ‘/tmp/RtmpiALWgt/remotes7c329de8600/jsilve24-driver-16e4499/DESCRIPTION’ ...
  
✔  checking for file ‘/tmp/RtmpiALWgt/remotes7c329de8600/jsilve24-driver-16e4499/DESCRIPTION’
## 
  
─  preparing ‘driver’:
## 
  
✔  checking DESCRIPTION meta-information
## 
  
─  checking for LF line-endings in source and make files and shell scripts
## 
  
─  checking for empty or unneeded directories
## 
  
   Omitted ‘LazyData’ from DESCRIPTION
## 
  
─  building ‘driver_0.1.1.tar.gz’
## 
  
   
## 
  
   checking for file ‘/tmp/RtmpiALWgt/remotes7c332b730a4/jsilve24-fido-c692141/DESCRIPTION’ ...
  
✔  checking for file ‘/tmp/RtmpiALWgt/remotes7c332b730a4/jsilve24-fido-c692141/DESCRIPTION’
## 
  
─  preparing ‘fido’:
## 
  
   checking DESCRIPTION meta-information ...
  
✔  checking DESCRIPTION meta-information
## 
  
─  cleaning src
## ─  running ‘cleanup’
## 
  
   Warning: /tmp/RtmpdNtyvB/Rbuild95fe7f9cc4/fido/man/loglikPibbleCollapsed.Rd:28: unknown macro &#39;\item&#39;
##    Warning: /tmp/RtmpdNtyvB/Rbuild95fe7f9cc4/fido/man/loglikPibbleCollapsed.Rd:30: unknown macro &#39;\item&#39;
##    Warning: /tmp/RtmpdNtyvB/Rbuild95fe7f9cc4/fido/man/loglikPibbleCollapsed.Rd:33: unexpected section header &#39;\value&#39;
## 
  
   Warning: /tmp/RtmpdNtyvB/Rbuild95fe7f9cc4/fido/man/loglikPibbleCollapsed.Rd:41: unexpected section header &#39;\description&#39;
##    Warning: /tmp/RtmpdNtyvB/Rbuild95fe7f9cc4/fido/man/loglikPibbleCollapsed.Rd:51: unexpected section header &#39;\examples&#39;
##    Warning: /tmp/RtmpdNtyvB/Rbuild95fe7f9cc4/fido/man/loglikPibbleCollapsed.Rd:83: unexpected END_OF_INPUT &#39;
##    &#39;
## 
  
─  checking for LF line-endings in source and make files and shell scripts
## 
  
─  checking for empty or unneeded directories
## 
  
─  building ‘fido_0.1.13.tar.gz’
## 
  
   
## </code></pre>
<p>Loading the libraries and importing data:</p>
<div class="sourceCode" id="cb201"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb201-1"><a href="differential-abundance.html#cb201-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(fido)</span>
<span id="cb201-2"><a href="differential-abundance.html#cb201-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mia)</span>
<span id="cb201-3"><a href="differential-abundance.html#cb201-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(microbiomeDataSets)</span>
<span id="cb201-4"><a href="differential-abundance.html#cb201-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb201-5"><a href="differential-abundance.html#cb201-5" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">SprockettTHData</span>()</span>
<span id="cb201-6"><a href="differential-abundance.html#cb201-6" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(<span class="fu">colData</span>(tse))</span></code></pre></div>
<pre><code>##  [1] &quot;Sample_ID&quot;                &quot;Sample_Type&quot;             
##  [3] &quot;Subject_ID&quot;               &quot;Sex&quot;                     
##  [5] &quot;Country&quot;                  &quot;Age_Days&quot;                
##  [7] &quot;Age_Months&quot;               &quot;Age_Years&quot;               
##  [9] &quot;Days_After_Birth&quot;         &quot;Months_After_Birth&quot;      
## [11] &quot;Age_Group&quot;                &quot;Age_Class&quot;               
## [13] &quot;Feeding_Status_Corrected&quot; &quot;Study_Code&quot;              
## [15] &quot;AgeGroup_By_SampleType&quot;   &quot;Dyad&quot;                    
## [17] &quot;Delivery_Mode&quot;            &quot;Community&quot;               
## [19] &quot;Community_Type&quot;           &quot;Community_Subtype&quot;       
## [21] &quot;Sample_Label&quot;             &quot;Cohort&quot;                  
## [23] &quot;Sample_Collection_Date&quot;   &quot;Community_Alias&quot;</code></pre>
<p>We pick three covariates (“Sex,”“Age_Years,”“Delivery_Mode”) during this analysis as an example, and beforehand we check for missing data:</p>
<div class="sourceCode" id="cb203"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb203-1"><a href="differential-abundance.html#cb203-1" aria-hidden="true" tabindex="-1"></a>cov_names <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Sex&quot;</span>,<span class="st">&quot;Age_Years&quot;</span>,<span class="st">&quot;Delivery_Mode&quot;</span>)</span>
<span id="cb203-2"><a href="differential-abundance.html#cb203-2" aria-hidden="true" tabindex="-1"></a>na_counts <span class="ot">&lt;-</span> <span class="fu">apply</span>(<span class="fu">is.na</span>(<span class="fu">colData</span>(tse)[,cov_names]), <span class="dv">2</span>, sum)</span>
<span id="cb203-3"><a href="differential-abundance.html#cb203-3" aria-hidden="true" tabindex="-1"></a>na_summary<span class="ot">&lt;-</span><span class="fu">as.data.frame</span>(na_counts,<span class="at">row.names=</span>cov_names)</span>
<span id="cb203-4"><a href="differential-abundance.html#cb203-4" aria-hidden="true" tabindex="-1"></a>na_summary</span></code></pre></div>
<pre><code>##               na_counts
## Sex                   0
## Age_Years          1603
## Delivery_Mode      1100</code></pre>
<p>We drop samples with na values at the covariates (features) under analysis:</p>
<div class="sourceCode" id="cb205"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb205-1"><a href="differential-abundance.html#cb205-1" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> tse[ , <span class="sc">!</span><span class="fu">is.na</span>(<span class="fu">colData</span>(tse)<span class="sc">$</span>Delivery_Mode) ]</span>
<span id="cb205-2"><a href="differential-abundance.html#cb205-2" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> tse[ , <span class="sc">!</span><span class="fu">is.na</span>(<span class="fu">colData</span>(tse)<span class="sc">$</span>Age_Years) ]</span>
<span id="cb205-3"><a href="differential-abundance.html#cb205-3" aria-hidden="true" tabindex="-1"></a>tse</span></code></pre></div>
<pre><code>## class: TreeSummarizedExperiment 
## dim: 2319 483 
## metadata(0):
## assays(1): counts
## rownames(2319): ASV_4_Candidatus_Methanogranum ASV_9_Methanobrevibacter
##   ... ASV_6837_Proteus_cibarius/hauseri+4 ASV_6838_Proteus
## rowData names(8): Kingdom Phylum ... Species lowest_rank
## colnames(483): 100.BF1 101.BF9A ... 98.MF35A 99.MF134A
## colData names(24): Sample_ID Sample_Type ... Sample_Collection_Date
##   Community_Alias
## reducedDimNames(0):
## mainExpName: NULL
## altExpNames(0):
## rowLinks: a LinkDataFrame (2319 rows)
## rowTree: 1 phylo tree(s) (2319 leaves)
## colLinks: NULL
## colTree: NULL
## referenceSeq: a DNAStringSet (2319 sequences)</code></pre>
<p>We agglomerate the data at a Phylum rank.
Note: Large assay data (along with the covariates/features data) could prevent the analysis later,
since the computation will construct matrices that would not always fit memory.</p>
<div class="sourceCode" id="cb207"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb207-1"><a href="differential-abundance.html#cb207-1" aria-hidden="true" tabindex="-1"></a>tse_phylum <span class="ot">&lt;-</span> <span class="fu">agglomerateByRank</span>(tse, <span class="st">&quot;Phylum&quot;</span>)</span>
<span id="cb207-2"><a href="differential-abundance.html#cb207-2" aria-hidden="true" tabindex="-1"></a>tse_phylum</span></code></pre></div>
<pre><code>## class: TreeSummarizedExperiment 
## dim: 20 483 
## metadata(0):
## assays(1): counts
## rownames(20): Phylum:Euryarchaeota Phylum:Firmicutes ...
##   Phylum:Fusobacteria Phylum:Epsilonbacteraeota
## rowData names(8): Kingdom Phylum ... Species lowest_rank
## colnames(483): 100.BF1 101.BF9A ... 98.MF35A 99.MF134A
## colData names(24): Sample_ID Sample_Type ... Sample_Collection_Date
##   Community_Alias
## reducedDimNames(0):
## mainExpName: NULL
## altExpNames(0):
## rowLinks: a LinkDataFrame (20 rows)
## rowTree: 1 phylo tree(s) (2319 leaves)
## colLinks: NULL
## colTree: NULL
## referenceSeq: a DNAStringSet (20 sequences)</code></pre>
<p>We extract the counts assay and feature data to build the model matrix having an extra row of ones presenting the intercept for the regression task later:</p>
<div class="sourceCode" id="cb209"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb209-1"><a href="differential-abundance.html#cb209-1" aria-hidden="true" tabindex="-1"></a>Y <span class="ot">&lt;-</span> <span class="fu">assays</span>(tse_phylum)<span class="sc">$</span>counts</span>
<span id="cb209-2"><a href="differential-abundance.html#cb209-2" aria-hidden="true" tabindex="-1"></a><span class="co"># design matrix</span></span>
<span id="cb209-3"><a href="differential-abundance.html#cb209-3" aria-hidden="true" tabindex="-1"></a><span class="co"># taking 3 covariates</span></span>
<span id="cb209-4"><a href="differential-abundance.html#cb209-4" aria-hidden="true" tabindex="-1"></a>sample_data<span class="ot">&lt;-</span><span class="fu">as.data.frame</span>(<span class="fu">colData</span>(tse_phylum)[,cov_names])</span>
<span id="cb209-5"><a href="differential-abundance.html#cb209-5" aria-hidden="true" tabindex="-1"></a>X <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">model.matrix</span>(<span class="sc">~</span>Sex<span class="sc">+</span>Age_Years<span class="sc">+</span>Delivery_Mode,<span class="at">data=</span>sample_data))</span>
<span id="cb209-6"><a href="differential-abundance.html#cb209-6" aria-hidden="true" tabindex="-1"></a>X[,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>##                      100.BF1 101.BF9A 102.BF36A 103.BF53A 104.BF83A
## (Intercept)             1.00     1.00      1.00      1.00      1.00
## SexMale                 0.00     0.00      0.00      0.00      0.00
## Age_Years               0.88     0.89      0.94      0.99      1.05
## Delivery_ModeVaginal    1.00     1.00      1.00      1.00      1.00</code></pre>
<div class="sourceCode" id="cb211"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb211-1"><a href="differential-abundance.html#cb211-1" aria-hidden="true" tabindex="-1"></a>Y[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>##                       100.BF1 101.BF9A 102.BF36A 103.BF53A 104.BF83A
## Phylum:Euryarchaeota        0        0         0         0         0
## Phylum:Firmicutes       10724    12129     15453     14446      4160
## Phylum:Tenericutes          0      173         0         3         2
## Phylum:Actinobacteria   18127    19122      1395      9666     19633
## Kingdom:Eukaryota           0        0         0         0         0</code></pre>
<p>Building the parameters for the <code>pibble</code> call to build the model; see more at <a href="https://jsilve24.github.io/fido/articles/introduction-to-fido.html">vignette</a>:</p>
<div class="sourceCode" id="cb213"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb213-1"><a href="differential-abundance.html#cb213-1" aria-hidden="true" tabindex="-1"></a>n_taxa<span class="ot">&lt;-</span><span class="fu">nrow</span>(Y)</span>
<span id="cb213-2"><a href="differential-abundance.html#cb213-2" aria-hidden="true" tabindex="-1"></a>upsilon <span class="ot">&lt;-</span> n_taxa<span class="sc">+</span><span class="dv">3</span></span>
<span id="cb213-3"><a href="differential-abundance.html#cb213-3" aria-hidden="true" tabindex="-1"></a>Omega <span class="ot">&lt;-</span> <span class="fu">diag</span>(n_taxa)</span>
<span id="cb213-4"><a href="differential-abundance.html#cb213-4" aria-hidden="true" tabindex="-1"></a>G <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">diag</span>(n_taxa<span class="dv">-1</span>), <span class="sc">-</span><span class="dv">1</span>)</span>
<span id="cb213-5"><a href="differential-abundance.html#cb213-5" aria-hidden="true" tabindex="-1"></a>Xi <span class="ot">&lt;-</span> (upsilon<span class="sc">-</span>n_taxa)<span class="sc">*</span>G<span class="sc">%*%</span>Omega<span class="sc">%*%</span><span class="fu">t</span>(G)</span>
<span id="cb213-6"><a href="differential-abundance.html#cb213-6" aria-hidden="true" tabindex="-1"></a>Theta <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="dv">0</span>, n_taxa<span class="dv">-1</span>, <span class="fu">nrow</span>(X))</span>
<span id="cb213-7"><a href="differential-abundance.html#cb213-7" aria-hidden="true" tabindex="-1"></a>Gamma <span class="ot">&lt;-</span> <span class="fu">diag</span>(<span class="fu">nrow</span>(X))</span></code></pre></div>
<p>Automatically initializing the priors and visualizing their distributions:</p>
<div class="sourceCode" id="cb214"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb214-1"><a href="differential-abundance.html#cb214-1" aria-hidden="true" tabindex="-1"></a>priors <span class="ot">&lt;-</span> <span class="fu">pibble</span>(<span class="cn">NULL</span>, X, upsilon, Theta, Gamma, Xi)</span>
<span id="cb214-2"><a href="differential-abundance.html#cb214-2" aria-hidden="true" tabindex="-1"></a><span class="fu">names_covariates</span>(priors) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(X)</span>
<span id="cb214-3"><a href="differential-abundance.html#cb214-3" aria-hidden="true" tabindex="-1"></a>fido<span class="sc">::</span><span class="fu">plot</span>(priors, <span class="at">pars=</span><span class="st">&quot;Lambda&quot;</span>) <span class="sc">+</span> ggplot2<span class="sc">::</span><span class="fu">xlim</span>(<span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">5</span>))</span></code></pre></div>
<p><img src="17_differential_abundance_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
<p>Estimating the posterior by including the data at <code>Y</code>.
Note: Some computational failures could occur (see <a href="https://github-wiki-see.page/m/jsilve24/fido/wiki/Frequently-Asked-Questions">discussion</a>)
the arguments <code>multDirichletBoot</code> <code>calcGradHess</code> could be passed in such case.</p>
<div class="sourceCode" id="cb215"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb215-1"><a href="differential-abundance.html#cb215-1" aria-hidden="true" tabindex="-1"></a>priors<span class="sc">$</span>Y <span class="ot">&lt;-</span> Y </span>
<span id="cb215-2"><a href="differential-abundance.html#cb215-2" aria-hidden="true" tabindex="-1"></a>posterior <span class="ot">&lt;-</span> <span class="fu">refit</span>(priors, <span class="at">optim_method=</span><span class="st">&quot;adam&quot;</span>, <span class="at">multDirichletBoot=</span><span class="fl">0.5</span>) <span class="co"># ,, calcGradHess=FALSE</span></span></code></pre></div>
<p>Printing a summary about the posterior predictive distribution:</p>
<div class="sourceCode" id="cb216"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb216-1"><a href="differential-abundance.html#cb216-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ppc_summary</span>(posterior)</span></code></pre></div>
<pre><code>## Proportions of Observations within 95% Credible Interval: 0.9981</code></pre>
<p>Plotting the summary of the posterior distributions of the regression parameters:</p>
<div class="sourceCode" id="cb218"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb218-1"><a href="differential-abundance.html#cb218-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names_categories</span>(posterior) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(Y)</span>
<span id="cb218-2"><a href="differential-abundance.html#cb218-2" aria-hidden="true" tabindex="-1"></a>fido<span class="sc">::</span><span class="fu">plot</span>(posterior,<span class="at">par=</span><span class="st">&quot;Lambda&quot;</span>,<span class="at">focus.cov=</span><span class="fu">rownames</span>(X)[<span class="dv">2</span><span class="sc">:</span><span class="dv">4</span>])</span></code></pre></div>
<p><img src="17_differential_abundance_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
<p>Seemingly the covariate “Age_Years” does not have effect on the model as “Delivery_Mode” would,
and “Sex” to some extent. Let’s take a closer look at the two latter ones:</p>
<div class="sourceCode" id="cb219"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb219-1"><a href="differential-abundance.html#cb219-1" aria-hidden="true" tabindex="-1"></a>fido<span class="sc">::</span><span class="fu">plot</span>(posterior, <span class="at">par=</span><span class="st">&quot;Lambda&quot;</span>, <span class="at">focus.cov =</span> <span class="fu">rownames</span>(X)[<span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">4</span>)])</span></code></pre></div>
<p><img src="17_differential_abundance_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
</div>
</div>
<div id="session-info-8" class="section level2 unnumbered">
<h2>Session Info</h2>
<button class="rebook-collapse">
View session info
</button>
<div class="rebook-content">
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.2 LTS

Matrix products: default
BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] microbiomeDataSets_1.1.1       MultiAssayExperiment_1.19.5   
 [3] mia_1.1.8                      TreeSummarizedExperiment_2.1.3
 [5] Biostrings_2.61.2              XVector_0.33.0                
 [7] SingleCellExperiment_1.15.1    SummarizedExperiment_1.23.1   
 [9] Biobase_2.53.0                 GenomicRanges_1.45.0          
[11] GenomeInfoDb_1.29.3            IRanges_2.27.0                
[13] S4Vectors_0.31.0               BiocGenerics_0.39.1           
[15] MatrixGenerics_1.5.2           matrixStats_0.60.0            
[17] fido_0.1.13                    BiocStyle_2.21.3              
[19] rebook_1.3.0                  

loaded via a namespace (and not attached):
  [1] utf8_1.2.2                    tidyselect_1.1.1             
  [3] RSQLite_2.2.7                 AnnotationDbi_1.55.1         
  [5] grid_4.1.0                    BiocParallel_1.27.2          
  [7] devtools_2.4.2                munsell_0.5.0                
  [9] ScaledMatrix_1.1.0            codetools_0.2-18             
 [11] withr_2.4.2                   colorspace_2.0-2             
 [13] filelock_1.0.2                highr_0.9                    
 [15] knitr_1.33                    labeling_0.4.2               
 [17] GenomeInfoDbData_1.2.6        bit64_4.0.5                  
 [19] farver_2.1.0                  driver_0.1.1                 
 [21] rprojroot_2.0.2               coda_0.19-4                  
 [23] vctrs_0.3.8                   treeio_1.17.2                
 [25] generics_0.1.0                xfun_0.24                    
 [27] BiocFileCache_2.1.1           R6_2.5.0                     
 [29] ggbeeswarm_0.6.0              rsvd_1.0.5                   
 [31] bitops_1.0-7                  cachem_1.0.5                 
 [33] DelayedArray_0.19.1           assertthat_0.2.1             
 [35] promises_1.2.0.1              scales_1.1.1                 
 [37] beeswarm_0.4.0                gtable_0.3.0                 
 [39] beachmat_2.9.0                processx_3.5.2               
 [41] rlang_0.4.11                  splines_4.1.0                
 [43] lazyeval_0.2.2                checkmate_2.0.0              
 [45] BiocManager_1.30.16           yaml_2.2.1                   
 [47] reshape2_1.4.4                abind_1.4-5                  
 [49] backports_1.2.1               httpuv_1.6.1                 
 [51] tensorA_0.36.2                tools_4.1.0                  
 [53] usethis_2.0.1                 bookdown_0.22                
 [55] ggplot2_3.3.5                 ellipsis_0.3.2               
 [57] decontam_1.13.0               RColorBrewer_1.1-2           
 [59] jquerylib_0.1.4               posterior_1.0.1              
 [61] sessioninfo_1.1.1             Rcpp_1.0.7                   
 [63] plyr_1.8.6                    sparseMatrixStats_1.5.0      
 [65] zlibbioc_1.39.0               purrr_0.3.4                  
 [67] RCurl_1.98-1.3                ps_1.6.0                     
 [69] prettyunits_1.1.1             viridis_0.6.1                
 [71] ggrepel_0.9.1                 cluster_2.1.2                
 [73] fs_1.5.0                      DECIPHER_2.21.0              
 [75] magrittr_2.0.1                ggdist_3.0.0                 
 [77] tidybayes_3.0.0               pkgload_1.2.1                
 [79] mime_0.11                     evaluate_0.14                
 [81] arrayhelpers_1.1-0            xtable_1.8-4                 
 [83] XML_3.99-0.6                  gridExtra_2.3                
 [85] testthat_3.0.4                compiler_4.1.0               
 [87] scater_1.21.3                 tibble_3.1.3                 
 [89] crayon_1.4.1                  htmltools_0.5.1.1            
 [91] mgcv_1.8-36                   later_1.2.0                  
 [93] tidyr_1.1.3                   DBI_1.1.1                    
 [95] ExperimentHub_2.1.4           dbplyr_2.1.1                 
 [97] MASS_7.3-54                   rappdirs_0.3.3               
 [99] Matrix_1.3-4                  permute_0.9-5                
[101] cli_3.0.1                     parallel_4.1.0               
[103] pkgconfig_2.0.3               dir.expiry_1.1.0             
[105] scuttle_1.3.0                 svUnit_1.0.6                 
[107] vipor_0.4.5                   bslib_0.2.5.1                
[109] DirichletMultinomial_1.35.0   stringr_1.4.0                
[111] distributional_0.2.2          callr_3.7.0                  
[113] digest_0.6.27                 vegan_2.5-7                  
[115] graph_1.71.2                  rmarkdown_2.9                
[117] tidytree_0.3.4                DelayedMatrixStats_1.15.0    
[119] curl_4.3.2                    shiny_1.6.0                  
[121] lifecycle_1.0.0               nlme_3.1-152                 
[123] jsonlite_1.7.2                BiocNeighbors_1.11.0         
[125] desc_1.3.0                    CodeDepends_0.6.5            
[127] viridisLite_0.4.0             fansi_0.5.0                  
[129] pillar_1.6.2                  lattice_0.20-44              
[131] KEGGREST_1.33.0               fastmap_1.1.0                
[133] httr_1.4.2                    pkgbuild_1.2.0               
[135] interactiveDisplayBase_1.31.2 glue_1.4.2                   
[137] remotes_2.4.0                 png_0.1-7                    
[139] BiocVersion_3.14.0            bit_4.0.4                    
[141] stringi_1.7.3                 sass_0.4.0                   
[143] blob_1.2.2                    BiocSingular_1.9.1           
[145] AnnotationHub_3.1.4           memoise_2.0.0                
[147] dplyr_1.0.7                   irlba_2.3.3                  
[149] ape_5.5                      </code></pre>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="microbiome-community.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="microbiome-timeseries.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["OMA.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
