<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 17 Exercises | Orchestrating Microbiome Analysis</title>
  <meta name="description" content="Chapter 17 Exercises | Orchestrating Microbiome Analysis" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 17 Exercises | Orchestrating Microbiome Analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="microbiome/OMA" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 17 Exercises | Orchestrating Microbiome Analysis" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="resources.html"/>
<link rel="next" href="extras.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Orchestrating Microbiome Analysis</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="part"><span><b>I Introduction</b></span></li>
<li class="chapter" data-level="1" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="packages.html"><a href="packages.html"><i class="fa fa-check"></i><b>2</b> Packages</a>
<ul>
<li class="chapter" data-level="2.1" data-path="packages.html"><a href="packages.html#package-installation"><i class="fa fa-check"></i><b>2.1</b> Package installation</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="packages.html"><a href="packages.html#packages_specific"><i class="fa fa-check"></i><b>2.1.1</b> Installing specific packages</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="packages.html"><a href="packages.html#package-ecosystem"><i class="fa fa-check"></i><b>2.2</b> Package ecosystem</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="packages.html"><a href="packages.html#mia-family-of-methods"><i class="fa fa-check"></i><b>2.2.1</b> mia family of methods</a></li>
<li class="chapter" data-level="2.2.2" data-path="packages.html"><a href="packages.html#sub-tree-methods"><i class="fa fa-check"></i><b>2.2.2</b> Tree-based methods</a></li>
<li class="chapter" data-level="2.2.3" data-path="packages.html"><a href="packages.html#sub-diff-abund"><i class="fa fa-check"></i><b>2.2.3</b> Differential abundance</a></li>
<li class="chapter" data-level="2.2.4" data-path="packages.html"><a href="packages.html#sub-manipulation"><i class="fa fa-check"></i><b>2.2.4</b> Manipulation</a></li>
<li class="chapter" data-level="2.2.5" data-path="packages.html"><a href="packages.html#further-options"><i class="fa fa-check"></i><b>2.2.5</b> Further options</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="packages.html"><a href="packages.html#session-info"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="containers.html"><a href="containers.html"><i class="fa fa-check"></i><b>3</b> Microbiome Data</a>
<ul>
<li class="chapter" data-level="3.1" data-path="containers.html"><a href="containers.html#data-science-framework"><i class="fa fa-check"></i><b>3.1</b> Data science framework</a></li>
<li class="chapter" data-level="3.2" data-path="containers.html"><a href="containers.html#data-containers"><i class="fa fa-check"></i><b>3.2</b> Data containers</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="containers.html"><a href="containers.html#assay-data"><i class="fa fa-check"></i><b>3.2.1</b> Assay data</a></li>
<li class="chapter" data-level="3.2.2" data-path="containers.html"><a href="containers.html#coldata"><i class="fa fa-check"></i><b>3.2.2</b> colData</a></li>
<li class="chapter" data-level="3.2.3" data-path="containers.html"><a href="containers.html#rowdata"><i class="fa fa-check"></i><b>3.2.3</b> rowData</a></li>
<li class="chapter" data-level="3.2.4" data-path="containers.html"><a href="containers.html#rowtree"><i class="fa fa-check"></i><b>3.2.4</b> rowTree</a></li>
<li class="chapter" data-level="3.2.5" data-path="containers.html"><a href="containers.html#alt-exp"><i class="fa fa-check"></i><b>3.2.5</b> Alternative experiments</a></li>
<li class="chapter" data-level="3.2.6" data-path="containers.html"><a href="containers.html#mae"><i class="fa fa-check"></i><b>3.2.6</b> MultiAssayExperiments</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="containers.html"><a href="containers.html#example-data"><i class="fa fa-check"></i><b>3.3</b> Demonstration data</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="containers.html"><a href="containers.html#package-data"><i class="fa fa-check"></i><b>3.3.1</b> Package data</a></li>
<li class="chapter" data-level="3.3.2" data-path="containers.html"><a href="containers.html#experimenthub-data"><i class="fa fa-check"></i><b>3.3.2</b> ExperimentHub data</a></li>
<li class="chapter" data-level="3.3.3" data-path="containers.html"><a href="containers.html#curated-metagenomic-data"><i class="fa fa-check"></i><b>3.3.3</b> Curated metagenomic data</a></li>
<li class="chapter" data-level="3.3.4" data-path="containers.html"><a href="containers.html#other-data-sources"><i class="fa fa-check"></i><b>3.3.4</b> Other data sources</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="containers.html"><a href="containers.html#loading-experimental-microbiome-data"><i class="fa fa-check"></i><b>3.4</b> Loading experimental microbiome data</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="containers.html"><a href="containers.html#s-workflow"><i class="fa fa-check"></i><b>3.4.1</b> 16S workflow</a></li>
<li class="chapter" data-level="3.4.2" data-path="containers.html"><a href="containers.html#import-from-external-files"><i class="fa fa-check"></i><b>3.4.2</b> Import from external files</a></li>
<li class="chapter" data-level="3.4.3" data-path="containers.html"><a href="containers.html#constructing-treesummarizedexperiment"><i class="fa fa-check"></i><b>3.4.3</b> Constructing TreeSummarizedExperiment</a></li>
<li class="chapter" data-level="3.4.4" data-path="containers.html"><a href="containers.html#constructing-multiassayexperiment"><i class="fa fa-check"></i><b>3.4.4</b> Constructing MultiAssayExperiment</a></li>
<li class="chapter" data-level="3.4.5" data-path="containers.html"><a href="containers.html#import-functions-for-standard-formats"><i class="fa fa-check"></i><b>3.4.5</b> Import functions for standard formats</a></li>
<li class="chapter" data-level="3.4.6" data-path="containers.html"><a href="containers.html#conversions-between-data-formats-in-r"><i class="fa fa-check"></i><b>3.4.6</b> Conversions between data formats in R</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="containers.html"><a href="containers.html#session-info-1"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="part"><span><b>II Focus Topics</b></span></li>
<li class="chapter" data-level="4" data-path="datamanipulation.html"><a href="datamanipulation.html"><i class="fa fa-check"></i><b>4</b> Data Manipulation</a>
<ul>
<li class="chapter" data-level="4.1" data-path="datamanipulation.html"><a href="datamanipulation.html#tidying-and-subsetting"><i class="fa fa-check"></i><b>4.1</b> Tidying and subsetting</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="datamanipulation.html"><a href="datamanipulation.html#tidy-data"><i class="fa fa-check"></i><b>4.1.1</b> Tidy data</a></li>
<li class="chapter" data-level="4.1.2" data-path="datamanipulation.html"><a href="datamanipulation.html#subsetting"><i class="fa fa-check"></i><b>4.1.2</b> Subsetting</a></li>
<li class="chapter" data-level="4.1.3" data-path="datamanipulation.html"><a href="datamanipulation.html#splitting"><i class="fa fa-check"></i><b>4.1.3</b> Splitting</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="datamanipulation.html"><a href="datamanipulation.html#add-or-modify-data"><i class="fa fa-check"></i><b>4.2</b> Add or modify data</a></li>
<li class="chapter" data-level="4.3" data-path="datamanipulation.html"><a href="datamanipulation.html#merge-data"><i class="fa fa-check"></i><b>4.3</b> Merge data</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="datamanipulation.html"><a href="datamanipulation.html#additional-functions"><i class="fa fa-check"></i><b>4.3.1</b> Additional functions</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>5</b> Exploration and quality Control</a>
<ul>
<li class="chapter" data-level="5.1" data-path="quality-control.html"><a href="quality-control.html#abundance"><i class="fa fa-check"></i><b>5.1</b> Abundance</a></li>
<li class="chapter" data-level="5.2" data-path="quality-control.html"><a href="quality-control.html#prevalence"><i class="fa fa-check"></i><b>5.2</b> Prevalence</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="quality-control.html"><a href="quality-control.html#prevalence-analysis"><i class="fa fa-check"></i><b>5.2.1</b> Prevalence analysis</a></li>
<li class="chapter" data-level="5.2.2" data-path="quality-control.html"><a href="quality-control.html#rare-taxa"><i class="fa fa-check"></i><b>5.2.2</b> Rare taxa</a></li>
<li class="chapter" data-level="5.2.3" data-path="quality-control.html"><a href="quality-control.html#plotting-prevalence"><i class="fa fa-check"></i><b>5.2.3</b> Plotting prevalence</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="quality-control.html"><a href="quality-control.html#qc"><i class="fa fa-check"></i><b>5.3</b> Quality control</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="quality-control.html"><a href="quality-control.html#top-taxa"><i class="fa fa-check"></i><b>5.3.1</b> Top taxa</a></li>
<li class="chapter" data-level="5.3.2" data-path="quality-control.html"><a href="quality-control.html#library-size-read-count"><i class="fa fa-check"></i><b>5.3.2</b> Library size / read count</a></li>
<li class="chapter" data-level="5.3.3" data-path="quality-control.html"><a href="quality-control.html#contaminant-sequences"><i class="fa fa-check"></i><b>5.3.3</b> Contaminant sequences</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="quality-control.html"><a href="quality-control.html#session-info-2"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="taxonomic-information.html"><a href="taxonomic-information.html"><i class="fa fa-check"></i><b>6</b> Taxonomic Information</a>
<ul>
<li class="chapter" data-level="6.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#assigning-taxonomic-information."><i class="fa fa-check"></i><b>6.1</b> Assigning taxonomic information.</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#dada2"><i class="fa fa-check"></i><b>6.1.1</b> dada2</a></li>
<li class="chapter" data-level="6.1.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#decipher"><i class="fa fa-check"></i><b>6.1.2</b> DECIPHER</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#functions-to-access-taxonomic-information"><i class="fa fa-check"></i><b>6.2</b> Functions to access taxonomic information</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#generate-a-taxonomic-tree-on-the-fly"><i class="fa fa-check"></i><b>6.2.1</b> Generate a taxonomic tree on the fly</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="taxonomic-information.html"><a href="taxonomic-information.html#data-agglomeration"><i class="fa fa-check"></i><b>6.3</b> Data agglomeration</a></li>
<li class="chapter" data-level="6.4" data-path="taxonomic-information.html"><a href="taxonomic-information.html#data-transformation"><i class="fa fa-check"></i><b>6.4</b> Data transformation</a></li>
<li class="chapter" data-level="6.5" data-path="taxonomic-information.html"><a href="taxonomic-information.html#pick-specific"><i class="fa fa-check"></i><b>6.5</b> Pick specific</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#abundances-of-all-taxa-in-specific-sample"><i class="fa fa-check"></i><b>6.5.1</b> Abundances of all taxa in specific sample</a></li>
<li class="chapter" data-level="6.5.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#abundances-of-specific-taxa-in-all-samples"><i class="fa fa-check"></i><b>6.5.2</b> Abundances of specific taxa in all samples</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="taxonomic-information.html"><a href="taxonomic-information.html#session-info-3"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="community-diversity.html"><a href="community-diversity.html"><i class="fa fa-check"></i><b>7</b> Community diversity</a>
<ul>
<li class="chapter" data-level="7.1" data-path="community-diversity.html"><a href="community-diversity.html#estimation"><i class="fa fa-check"></i><b>7.1</b> Estimation</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="community-diversity.html"><a href="community-diversity.html#richness"><i class="fa fa-check"></i><b>7.1.1</b> Richness</a></li>
<li class="chapter" data-level="7.1.2" data-path="community-diversity.html"><a href="community-diversity.html#estimate-diversity"><i class="fa fa-check"></i><b>7.1.2</b> Diversity</a></li>
<li class="chapter" data-level="7.1.3" data-path="community-diversity.html"><a href="community-diversity.html#faith-diversity"><i class="fa fa-check"></i><b>7.1.3</b> Faith phylogenetic diversity</a></li>
<li class="chapter" data-level="7.1.4" data-path="community-diversity.html"><a href="community-diversity.html#evenness"><i class="fa fa-check"></i><b>7.1.4</b> Evenness</a></li>
<li class="chapter" data-level="7.1.5" data-path="community-diversity.html"><a href="community-diversity.html#dominance"><i class="fa fa-check"></i><b>7.1.5</b> Dominance</a></li>
<li class="chapter" data-level="7.1.6" data-path="community-diversity.html"><a href="community-diversity.html#rarity"><i class="fa fa-check"></i><b>7.1.6</b> Rarity</a></li>
<li class="chapter" data-level="7.1.7" data-path="community-diversity.html"><a href="community-diversity.html#divergence"><i class="fa fa-check"></i><b>7.1.7</b> Divergence</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="community-diversity.html"><a href="community-diversity.html#visualization"><i class="fa fa-check"></i><b>7.2</b> Visualization</a></li>
<li class="chapter" data-level="" data-path="community-diversity.html"><a href="community-diversity.html#session-info-4"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="community-similarity.html"><a href="community-similarity.html"><i class="fa fa-check"></i><b>8</b> Community similarity</a>
<ul>
<li class="chapter" data-level="8.1" data-path="community-similarity.html"><a href="community-similarity.html#explained-variance"><i class="fa fa-check"></i><b>8.1</b> Explained variance</a></li>
<li class="chapter" data-level="8.2" data-path="community-similarity.html"><a href="community-similarity.html#community-comparisons-by-beta-diversity-analysis"><i class="fa fa-check"></i><b>8.2</b> Community comparisons by beta diversity analysis</a></li>
<li class="chapter" data-level="8.3" data-path="community-similarity.html"><a href="community-similarity.html#other-ordination-methods"><i class="fa fa-check"></i><b>8.3</b> Other ordination methods</a></li>
<li class="chapter" data-level="8.4" data-path="community-similarity.html"><a href="community-similarity.html#pcoa-genus"><i class="fa fa-check"></i><b>8.4</b> Visualizing the most dominant genus on PCoA</a>
<ul>
<li class="chapter" data-level="8.4.1" data-path="community-similarity.html"><a href="community-similarity.html#testing-differences-in-community-composition-between-sample-groups"><i class="fa fa-check"></i><b>8.4.1</b> Testing differences in community composition between sample groups</a></li>
<li class="chapter" data-level="8.4.2" data-path="community-similarity.html"><a href="community-similarity.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>8.4.2</b> Checking the homogeneity condition</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="community-similarity.html"><a href="community-similarity.html#further-reading"><i class="fa fa-check"></i><b>8.5</b> Further reading</a></li>
<li class="chapter" data-level="" data-path="community-similarity.html"><a href="community-similarity.html#session-info-5"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="microbiome-community.html"><a href="microbiome-community.html"><i class="fa fa-check"></i><b>9</b> Community composition</a>
<ul>
<li class="chapter" data-level="9.1" data-path="microbiome-community.html"><a href="microbiome-community.html#visual-composition"><i class="fa fa-check"></i><b>9.1</b> Visualizing taxonomic composition</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="microbiome-community.html"><a href="microbiome-community.html#composition-barplot"><i class="fa fa-check"></i><b>9.1.1</b> Composition barplot</a></li>
<li class="chapter" data-level="9.1.2" data-path="microbiome-community.html"><a href="microbiome-community.html#composition-heatmap"><i class="fa fa-check"></i><b>9.1.2</b> Composition heatmap</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="microbiome-community.html"><a href="microbiome-community.html#session-info-6"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="clustering.html"><a href="clustering.html"><i class="fa fa-check"></i><b>10</b> Community typing (clustering)</a>
<ul>
<li class="chapter" data-level="10.1" data-path="clustering.html"><a href="clustering.html#introduction-to-bluster-package"><i class="fa fa-check"></i><b>10.1</b> Introduction to bluster package</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="clustering.html"><a href="clustering.html#sample-clustering"><i class="fa fa-check"></i><b>10.1.1</b> Sample clustering</a></li>
<li class="chapter" data-level="10.1.2" data-path="clustering.html"><a href="clustering.html#taxa-clustering"><i class="fa fa-check"></i><b>10.1.2</b> Taxa clustering</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="clustering.html"><a href="clustering.html#hierarchical-clustering"><i class="fa fa-check"></i><b>10.2</b> Hierarchical clustering</a></li>
<li class="chapter" data-level="10.3" data-path="clustering.html"><a href="clustering.html#k-means-clustering"><i class="fa fa-check"></i><b>10.3</b> K-means clustering</a></li>
<li class="chapter" data-level="10.4" data-path="clustering.html"><a href="clustering.html#dirichlet-multinomial-mixtures-dmm"><i class="fa fa-check"></i><b>10.4</b> Dirichlet Multinomial Mixtures (DMM)</a>
<ul>
<li class="chapter" data-level="10.4.1" data-path="clustering.html"><a href="clustering.html#pcoa-for-asv-level-data-with-bray-curtis-with-dmm-clusters-shown-with-colors"><i class="fa fa-check"></i><b>10.4.1</b> PCoA for ASV-level data with Bray-Curtis; with DMM clusters shown with colors</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="clustering.html"><a href="clustering.html#community-detection"><i class="fa fa-check"></i><b>10.5</b> Community Detection</a></li>
<li class="chapter" data-level="10.6" data-path="clustering.html"><a href="clustering.html#biclustering"><i class="fa fa-check"></i><b>10.6</b> Biclustering</a>
<ul>
<li class="chapter" data-level="10.6.1" data-path="clustering.html"><a href="clustering.html#taxa-vs-samples"><i class="fa fa-check"></i><b>10.6.1</b> Taxa vs samples</a></li>
<li class="chapter" data-level="10.6.2" data-path="clustering.html"><a href="clustering.html#taxa-vs-biomolecules"><i class="fa fa-check"></i><b>10.6.2</b> Taxa vs biomolecules</a></li>
<li class="chapter" data-level="10.6.3" data-path="clustering.html"><a href="clustering.html#taxa-vs-taxa"><i class="fa fa-check"></i><b>10.6.3</b> Taxa vs taxa</a></li>
</ul></li>
<li class="chapter" data-level="10.7" data-path="clustering.html"><a href="clustering.html#additional-community-typing"><i class="fa fa-check"></i><b>10.7</b> Additional Community Typing</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="differential-abundance.html"><a href="differential-abundance.html"><i class="fa fa-check"></i><b>11</b> Differential abundance</a>
<ul>
<li class="chapter" data-level="11.1" data-path="differential-abundance.html"><a href="differential-abundance.html#differential-abundance-analysis"><i class="fa fa-check"></i><b>11.1</b> Differential abundance analysis</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="differential-abundance.html"><a href="differential-abundance.html#examples-and-tools"><i class="fa fa-check"></i><b>11.1.1</b> Examples and tools</a></li>
<li class="chapter" data-level="11.1.2" data-path="differential-abundance.html"><a href="differential-abundance.html#prevalence-filtering"><i class="fa fa-check"></i><b>11.1.2</b> Prevalence Filtering</a></li>
<li class="chapter" data-level="11.1.3" data-path="differential-abundance.html"><a href="differential-abundance.html#aldex2"><i class="fa fa-check"></i><b>11.1.3</b> ALDEx2</a></li>
<li class="chapter" data-level="11.1.4" data-path="differential-abundance.html"><a href="differential-abundance.html#ancom-bc"><i class="fa fa-check"></i><b>11.1.4</b> ANCOM-BC</a></li>
<li class="chapter" data-level="11.1.5" data-path="differential-abundance.html"><a href="differential-abundance.html#maaslin2"><i class="fa fa-check"></i><b>11.1.5</b> MaAsLin2</a></li>
<li class="chapter" data-level="11.1.6" data-path="differential-abundance.html"><a href="differential-abundance.html#linda"><i class="fa fa-check"></i><b>11.1.6</b> LinDA</a></li>
<li class="chapter" data-level="11.1.7" data-path="differential-abundance.html"><a href="differential-abundance.html#zicoseq"><i class="fa fa-check"></i><b>11.1.7</b> ZicoSeq</a></li>
<li class="chapter" data-level="11.1.8" data-path="differential-abundance.html"><a href="differential-abundance.html#comparison-of-methods"><i class="fa fa-check"></i><b>11.1.8</b> Comparison of methods</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="differential-abundance.html"><a href="differential-abundance.html#confounding-variables"><i class="fa fa-check"></i><b>11.2</b> Confounding variables</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="differential-abundance.html"><a href="differential-abundance.html#ancombc"><i class="fa fa-check"></i><b>11.2.1</b> ANCOMBC</a></li>
<li class="chapter" data-level="11.2.2" data-path="differential-abundance.html"><a href="differential-abundance.html#linda-1"><i class="fa fa-check"></i><b>11.2.2</b> LinDA</a></li>
<li class="chapter" data-level="11.2.3" data-path="differential-abundance.html"><a href="differential-abundance.html#zicoseq-1"><i class="fa fa-check"></i><b>11.2.3</b> ZicoSeq</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="differential-abundance.html"><a href="differential-abundance.html#tree-based-methods"><i class="fa fa-check"></i><b>11.3</b> Tree-based methods</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="differential-abundance.html"><a href="differential-abundance.html#group-wise-associations-testing-based-on-balances"><i class="fa fa-check"></i><b>11.3.1</b> Group-wise associations testing based on balances</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="differential-abundance.html"><a href="differential-abundance.html#session-info-7"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="machine_learning.html"><a href="machine_learning.html"><i class="fa fa-check"></i><b>12</b> Machine learning</a>
<ul>
<li class="chapter" data-level="12.1" data-path="machine_learning.html"><a href="machine_learning.html#supervised-machine-learning"><i class="fa fa-check"></i><b>12.1</b> Supervised machine learning</a></li>
<li class="chapter" data-level="12.2" data-path="machine_learning.html"><a href="machine_learning.html#unsupervised-machine-learning"><i class="fa fa-check"></i><b>12.2</b> Unsupervised machine learning</a></li>
<li class="chapter" data-level="" data-path="machine_learning.html"><a href="machine_learning.html#session-info-8"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="multi-assay-analyses.html"><a href="multi-assay-analyses.html"><i class="fa fa-check"></i><b>13</b> Multi-assay analyses</a>
<ul>
<li class="chapter" data-level="13.1" data-path="multi-assay-analyses.html"><a href="multi-assay-analyses.html#cross-correlation-analysis"><i class="fa fa-check"></i><b>13.1</b> Cross-correlation Analysis</a></li>
<li class="chapter" data-level="13.2" data-path="multi-assay-analyses.html"><a href="multi-assay-analyses.html#multi-omics-factor-analysis"><i class="fa fa-check"></i><b>13.2</b> Multi-Omics Factor Analysis</a></li>
<li class="chapter" data-level="" data-path="multi-assay-analyses.html"><a href="multi-assay-analyses.html#session-info-9"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="viz-chapter.html"><a href="viz-chapter.html"><i class="fa fa-check"></i><b>14</b> Visualization</a>
<ul>
<li class="chapter" data-level="14.1" data-path="viz-chapter.html"><a href="viz-chapter.html#pre-analysis-exploration"><i class="fa fa-check"></i><b>14.1</b> Pre-analysis exploration</a>
<ul>
<li class="chapter" data-level="14.1.1" data-path="viz-chapter.html"><a href="viz-chapter.html#accessing-row-and-column-data"><i class="fa fa-check"></i><b>14.1.1</b> Accessing row and column data</a></li>
<li class="chapter" data-level="14.1.2" data-path="viz-chapter.html"><a href="viz-chapter.html#viewing-abundance-and-prevalence-patterns"><i class="fa fa-check"></i><b>14.1.2</b> Viewing abundance and prevalence patterns</a></li>
</ul></li>
<li class="chapter" data-level="14.2" data-path="viz-chapter.html"><a href="viz-chapter.html#diversity-estimation"><i class="fa fa-check"></i><b>14.2</b> Diversity estimation</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="viz-chapter.html"><a href="viz-chapter.html#alpha-diversity-with-scatter-violin-and-box-plots"><i class="fa fa-check"></i><b>14.2.1</b> Alpha diversity with scatter, violin and box plots</a></li>
<li class="chapter" data-level="14.2.2" data-path="viz-chapter.html"><a href="viz-chapter.html#beta-diversity-with-shepard-and-coordination-plots"><i class="fa fa-check"></i><b>14.2.2</b> Beta diversity with Shepard and coordination plots</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="viz-chapter.html"><a href="viz-chapter.html#statistical-analysis"><i class="fa fa-check"></i><b>14.3</b> Statistical analysis</a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="viz-chapter.html"><a href="viz-chapter.html#heatmaps"><i class="fa fa-check"></i><b>14.3.1</b> Heatmaps</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="viz-chapter.html"><a href="viz-chapter.html#session-info-10"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="part"><span><b>III Training</b></span></li>
<li class="chapter" data-level="15" data-path="training.html"><a href="training.html"><i class="fa fa-check"></i><b>15</b> Training</a>
<ul>
<li class="chapter" data-level="15.1" data-path="training.html"><a href="training.html#checklist"><i class="fa fa-check"></i><b>15.1</b> Checklist</a></li>
<li class="chapter" data-level="15.2" data-path="training.html"><a href="training.html#software"><i class="fa fa-check"></i><b>15.2</b> Recommended software</a></li>
<li class="chapter" data-level="15.3" data-path="training.html"><a href="training.html#material"><i class="fa fa-check"></i><b>15.3</b> Study material</a></li>
<li class="chapter" data-level="15.4" data-path="training.html"><a href="training.html#support-and-resources"><i class="fa fa-check"></i><b>15.4</b> Support and resources</a></li>
<li class="chapter" data-level="15.5" data-path="training.html"><a href="training.html#coc"><i class="fa fa-check"></i><b>15.5</b> Code of Conduct</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="resources.html"><a href="resources.html"><i class="fa fa-check"></i><b>16</b> Resources</a>
<ul>
<li class="chapter" data-level="16.1" data-path="resources.html"><a href="resources.html#data-containers-1"><i class="fa fa-check"></i><b>16.1</b> Data containers</a>
<ul>
<li class="chapter" data-level="16.1.1" data-path="resources.html"><a href="resources.html#resources-for-treesummarizedexperiment"><i class="fa fa-check"></i><b>16.1.1</b> Resources for TreeSummarizedExperiment</a></li>
<li class="chapter" data-level="16.1.2" data-path="resources.html"><a href="resources.html#other-relevant-containers"><i class="fa fa-check"></i><b>16.1.2</b> Other relevant containers</a></li>
<li class="chapter" data-level="16.1.3" data-path="resources.html"><a href="resources.html#alternative-containers-for-microbiome-data"><i class="fa fa-check"></i><b>16.1.3</b> Alternative containers for microbiome data</a></li>
<li class="chapter" data-level="16.1.4" data-path="resources.html"><a href="resources.html#resources-for-phyloseq"><i class="fa fa-check"></i><b>16.1.4</b> Resources for phyloseq</a></li>
</ul></li>
<li class="chapter" data-level="16.2" data-path="resources.html"><a href="resources.html#r-programming-resources"><i class="fa fa-check"></i><b>16.2</b> R programming resources</a>
<ul>
<li class="chapter" data-level="16.2.1" data-path="resources.html"><a href="resources.html#bioc_intro"><i class="fa fa-check"></i><b>16.2.1</b> Bioconductor Classes</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="resources.html"><a href="resources.html#quarto"><i class="fa fa-check"></i><b>16.3</b> Reproducible reporting with Quarto</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="exercises.html"><a href="exercises.html"><i class="fa fa-check"></i><b>17</b> Exercises</a>
<ul>
<li class="chapter" data-level="17.1" data-path="exercises.html"><a href="exercises.html#workflows"><i class="fa fa-check"></i><b>17.1</b> Workflows</a>
<ul>
<li class="chapter" data-level="17.1.1" data-path="exercises.html"><a href="exercises.html#reproducible-reporting-with-quarto"><i class="fa fa-check"></i><b>17.1.1</b> Reproducible reporting with Quarto</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="exercises.html"><a href="exercises.html#data-containers-treese"><i class="fa fa-check"></i><b>17.2</b> Data containers: TreeSE</a>
<ul>
<li class="chapter" data-level="17.2.1" data-path="exercises.html"><a href="exercises.html#constructing-a-data-object"><i class="fa fa-check"></i><b>17.2.1</b> Constructing a data object</a></li>
<li class="chapter" data-level="17.2.2" data-path="exercises.html"><a href="exercises.html#importing-data"><i class="fa fa-check"></i><b>17.2.2</b> Importing data</a></li>
<li class="chapter" data-level="17.2.3" data-path="exercises.html"><a href="exercises.html#basic-summaries"><i class="fa fa-check"></i><b>17.2.3</b> Basic summaries</a></li>
<li class="chapter" data-level="17.2.4" data-path="exercises.html"><a href="exercises.html#taxonomic-abundance-table-assay"><i class="fa fa-check"></i><b>17.2.4</b> Taxonomic abundance table (assay)</a></li>
<li class="chapter" data-level="17.2.5" data-path="exercises.html"><a href="exercises.html#sample-side-information"><i class="fa fa-check"></i><b>17.2.5</b> Sample side information</a></li>
<li class="chapter" data-level="17.2.6" data-path="exercises.html"><a href="exercises.html#feature-side-information"><i class="fa fa-check"></i><b>17.2.6</b> Feature side information</a></li>
<li class="chapter" data-level="17.2.7" data-path="exercises.html"><a href="exercises.html#other-elements"><i class="fa fa-check"></i><b>17.2.7</b> Other elements</a></li>
</ul></li>
<li class="chapter" data-level="17.3" data-path="exercises.html"><a href="exercises.html#data-manipulation"><i class="fa fa-check"></i><b>17.3</b> Data manipulation</a>
<ul>
<li class="chapter" data-level="17.3.1" data-path="exercises.html"><a href="exercises.html#subsetting-1"><i class="fa fa-check"></i><b>17.3.1</b> Subsetting</a></li>
<li class="chapter" data-level="17.3.2" data-path="exercises.html"><a href="exercises.html#library-sizes"><i class="fa fa-check"></i><b>17.3.2</b> Library sizes</a></li>
<li class="chapter" data-level="17.3.3" data-path="exercises.html"><a href="exercises.html#prevalent-and-core-taxonomic-features"><i class="fa fa-check"></i><b>17.3.3</b> Prevalent and core taxonomic features</a></li>
<li class="chapter" data-level="17.3.4" data-path="exercises.html"><a href="exercises.html#data-exploration"><i class="fa fa-check"></i><b>17.3.4</b> Data exploration</a></li>
<li class="chapter" data-level="17.3.5" data-path="exercises.html"><a href="exercises.html#other-functions"><i class="fa fa-check"></i><b>17.3.5</b> Other functions</a></li>
<li class="chapter" data-level="17.3.6" data-path="exercises.html"><a href="exercises.html#transformations"><i class="fa fa-check"></i><b>17.3.6</b> Transformations</a></li>
</ul></li>
<li class="chapter" data-level="17.4" data-path="exercises.html"><a href="exercises.html#abundance-tables"><i class="fa fa-check"></i><b>17.4</b> Abundance tables</a>
<ul>
<li class="chapter" data-level="17.4.1" data-path="exercises.html"><a href="exercises.html#taxonomic-levels"><i class="fa fa-check"></i><b>17.4.1</b> Taxonomic levels</a></li>
<li class="chapter" data-level="17.4.2" data-path="exercises.html"><a href="exercises.html#alternative-experiments-altexp"><i class="fa fa-check"></i><b>17.4.2</b> Alternative experiments (altExp)</a></li>
</ul></li>
<li class="chapter" data-level="17.5" data-path="exercises.html"><a href="exercises.html#community-diversity-alpha-diversity"><i class="fa fa-check"></i><b>17.5</b> Community diversity (alpha diversity)</a>
<ul>
<li class="chapter" data-level="17.5.1" data-path="exercises.html"><a href="exercises.html#alpha-diversity-basics"><i class="fa fa-check"></i><b>17.5.1</b> Alpha diversity basics</a></li>
<li class="chapter" data-level="17.5.2" data-path="exercises.html"><a href="exercises.html#alpha-diversity-extra"><i class="fa fa-check"></i><b>17.5.2</b> Alpha diversity extra</a></li>
</ul></li>
<li class="chapter" data-level="17.6" data-path="exercises.html"><a href="exercises.html#community-composition-beta-diversity"><i class="fa fa-check"></i><b>17.6</b> Community composition (beta diversity)</a>
<ul>
<li class="chapter" data-level="17.6.1" data-path="exercises.html"><a href="exercises.html#beta-diversity-basics"><i class="fa fa-check"></i><b>17.6.1</b> Beta diversity basics</a></li>
<li class="chapter" data-level="17.6.2" data-path="exercises.html"><a href="exercises.html#beta-diversity-extra"><i class="fa fa-check"></i><b>17.6.2</b> Beta diversity extra</a></li>
</ul></li>
<li class="chapter" data-level="17.7" data-path="exercises.html"><a href="exercises.html#differential-abundance-1"><i class="fa fa-check"></i><b>17.7</b> Differential abundance</a>
<ul>
<li class="chapter" data-level="17.7.1" data-path="exercises.html"><a href="exercises.html#univariate-analyses"><i class="fa fa-check"></i><b>17.7.1</b> Univariate analyses</a></li>
<li class="chapter" data-level="17.7.2" data-path="exercises.html"><a href="exercises.html#differential-abundance-analysis-1"><i class="fa fa-check"></i><b>17.7.2</b> Differential abundance analysis</a></li>
</ul></li>
<li class="chapter" data-level="17.8" data-path="exercises.html"><a href="exercises.html#visualization-1"><i class="fa fa-check"></i><b>17.8</b> Visualization</a>
<ul>
<li class="chapter" data-level="17.8.1" data-path="exercises.html"><a href="exercises.html#multivariate-ordination"><i class="fa fa-check"></i><b>17.8.1</b> Multivariate ordination</a></li>
<li class="chapter" data-level="17.8.2" data-path="exercises.html"><a href="exercises.html#heatmap-visualization"><i class="fa fa-check"></i><b>17.8.2</b> Heatmap visualization</a></li>
</ul></li>
<li class="chapter" data-level="17.9" data-path="exercises.html"><a href="exercises.html#multiomics"><i class="fa fa-check"></i><b>17.9</b> Multiomics</a>
<ul>
<li class="chapter" data-level="17.9.1" data-path="exercises.html"><a href="exercises.html#multiassayexperiment-mae-data-container"><i class="fa fa-check"></i><b>17.9.1</b> MultiAssayExperiment (MAE) data container</a></li>
<li class="chapter" data-level="17.9.2" data-path="exercises.html"><a href="exercises.html#multi-omic-data-exploration"><i class="fa fa-check"></i><b>17.9.2</b> Multi-omic data exploration</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>IV Appendix</b></span></li>
<li class="chapter" data-level="18" data-path="extras.html"><a href="extras.html"><i class="fa fa-check"></i><b>18</b> Extra material</a>
<ul>
<li class="chapter" data-level="18.1" data-path="extras.html"><a href="extras.html#permanova-comparison"><i class="fa fa-check"></i><b>18.1</b> PERMANOVA comparison</a></li>
<li class="chapter" data-level="18.2" data-path="extras.html"><a href="extras.html#bayesian-multinomial-logistic-normal-models"><i class="fa fa-check"></i><b>18.2</b> Bayesian Multinomial Logistic-Normal Models</a></li>
<li class="chapter" data-level="18.3" data-path="extras.html"><a href="extras.html#interactive-3d-plots"><i class="fa fa-check"></i><b>18.3</b> Interactive 3D Plots</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="acknowledgments.html"><a href="acknowledgments.html"><i class="fa fa-check"></i><b>19</b> Acknowledgments</a></li>
<li class="chapter" data-level="" data-path="bibliography.html"><a href="bibliography.html"><i class="fa fa-check"></i>Bibliography</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Orchestrating Microbiome Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="exercises" class="section level1 hasAnchor" number="17">
<h1><span class="header-section-number">Chapter 17</span> Exercises<a href="exercises.html#exercises" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Here you can find assignments on different topics.</p>
<p><strong>Tips for exercises:</strong></p>
<ul>
<li>Add comments that explain what each line or lines of code do. This helps you and others to understand your code and find bugs. Furthermore, it is easier for you to reuse the code, and it promotes transparency.</li>
<li>Interpret results by comparing them to literature. List main findings, so that results can easily be understood by others without advanced data analytics knowledge.</li>
<li>Avoid hard-coding. Use variables which get values in the beginning of the pipeline. That way it is easier for you to modify parameters and reuse the code.</li>
</ul>
<div id="workflows" class="section level2 hasAnchor" number="17.1">
<h2><span class="header-section-number">17.1</span> Workflows<a href="exercises.html#workflows" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="reproducible-reporting-with-quarto" class="section level3 hasAnchor" number="17.1.1">
<h3><span class="header-section-number">17.1.1</span> Reproducible reporting with Quarto<a href="exercises.html#reproducible-reporting-with-quarto" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>Create a new Quarto file</li>
</ol>
<ul>
<li>Rstudio has ready-made templates for this</li>
<li><a href="https://quarto.org/docs/tools/rstudio.html#creating-documents">Creating Quarto Documents</a></li>
</ul>
<ol start="2" style="list-style-type: decimal">
<li>Add a code chunk and name it.</li>
<li>Render (or knit) the file into pdf or html format (Hint: <a href="https://quarto.org/docs/tools/rstudio.html#render-and-preview">Quarto Rstudio rendering</a>)</li>
<li>Import e.g., iris dataset, and add a dotplot with a title (Hint: <a href="https://ggplot2.tidyverse.org/reference/geom_dotplot.html">geom_dotplot</a>)</li>
<li>Create another code chunk and plot.</li>
<li>Adjust figure size and hide the code chunk from output report.</li>
</ol>
<ul>
<li><a href="https://quarto.org/docs/authoring/diagrams.html#sizing">Sizing</a></li>
<li><a href="https://quarto.org/docs/computations/r.html#chunk-options">chunk-options</a></li>
</ul>
<ol start="7" style="list-style-type: decimal">
<li>Add some text.</li>
<li>Add R commands within the text (Hint: <a href="https://quarto.org/docs/visual-editor/technical.html#code-chunks">code-chunks</a>)</li>
<li>Update HTML file from the qmd file (Hint: <a href="https://quarto.org/docs/tools/rstudio.html#render-and-preview">Quarto Rstudio rendering</a>)</li>
</ol>
<p>For tips on Quarto, see <a href="https://quarto.org/docs/authoring/markdown-basics.html">Quarto tutorial</a></p>
</div>
</div>
<div id="data-containers-treese" class="section level2 hasAnchor" number="17.2">
<h2><span class="header-section-number">17.2</span> Data containers: TreeSE<a href="exercises.html#data-containers-treese" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="constructing-a-data-object" class="section level3 hasAnchor" number="17.2.1">
<h3><span class="header-section-number">17.2.1</span> Constructing a data object<a href="exercises.html#constructing-a-data-object" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Import data from CSV files to TreeSE (see shared data folder for example data sets).</p>
<ol style="list-style-type: decimal">
<li>Import the data files in R</li>
<li>Construct a TreeSE data object (see <a href="https://microbiome.github.io/OMA/containers.html#loading-experimental-microbiome-data">Ch. 2</a>)</li>
<li>Check that importing is done correctly. E.g., choose random samples and features,
and check that their values equal between raw files and TreeSE.</li>
</ol>
<p>Useful functions: DataFrame, TreeSummarizedExperiment, matrix, rownames, colnames, SimpleList</p>
</div>
<div id="importing-data" class="section level3 hasAnchor" number="17.2.2">
<h3><span class="header-section-number">17.2.2</span> Importing data<a href="exercises.html#importing-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>You can also check the <a href="https://microbiome.github.io/mia/reference/index.html">function reference in the mia package</a></p>
<ol style="list-style-type: decimal">
<li>Import data from another format (functions: loadFromMetaphlan | loadFromMothur | loadFromQIIME2 | makeTreeSummarizedExperimentFromBiom | makeTreeSummarizedExperimentFromDADA2 …)</li>
<li>Try out conversions between TreeSE and phyloseq data containers (makeTreeSummarizedExperimentFromPhyloseq; makephyloseqFromTreeSummarizedExperiment)</li>
</ol>
</div>
<div id="basic-summaries" class="section level3 hasAnchor" number="17.2.3">
<h3><span class="header-section-number">17.2.3</span> Basic summaries<a href="exercises.html#basic-summaries" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>Load experimental dataset from mia (e.g. <code>peerj13075</code> with the <code>data()</code> command; see OMA <a href="https://microbiome.github.io/OMA/containers.html#example-data">section 2.4 Demonstration Data</a>; also see <a href="https://microbiome.github.io/OMA/containers.html#assay-data">loading experimental data</a>).</li>
<li>Check a summary about the TreeSE object loaded (Hint: <code>summary()</code>)</li>
<li>What are the dimensions? (How many samples there are, and how many taxa in each taxonomic rank?) (Hint: material in <a href="https://microbiome.github.io/OMA/containers.html#data-science-framework">Section 2</a> may help)</li>
<li>List sample and features names for the data (rownames, colnames..)</li>
</ol>
</div>
<div id="taxonomic-abundance-table-assay" class="section level3 hasAnchor" number="17.2.4">
<h3><span class="header-section-number">17.2.4</span> Taxonomic abundance table (assay)<a href="exercises.html#taxonomic-abundance-table-assay" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>(Load example data)</li>
<li>Fetch the list of available assays (Hints: <a href="https://microbiome.github.io/OMA/containers.html#assay-data">assayNames</a>)</li>
<li>Fetch the <code>counts</code> assay, and show part of it. (Hint: <a href="https://microbiome.github.io/OMA/containers.html#assay-data">assay-data</a>)</li>
</ol>
</div>
<div id="sample-side-information" class="section level3 hasAnchor" number="17.2.5">
<h3><span class="header-section-number">17.2.5</span> Sample side information<a href="exercises.html#sample-side-information" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>(Load example data)</li>
<li>Fetch and show data about samples (Hint: <a href="https://microbiome.github.io/OMA/containers.html#coldata">colData</a>)</li>
<li>Get abundance data for all taxa for a specific sample (sample names: function <code>colnames(tse)</code>)</li>
</ol>
<ul>
<li><a href="https://microbiome.github.io/OMA/taxonomic-information.html#abundances-of-all-taxa-in-specific-sample">example</a></li>
</ul>
</div>
<div id="feature-side-information" class="section level3 hasAnchor" number="17.2.6">
<h3><span class="header-section-number">17.2.6</span> Feature side information<a href="exercises.html#feature-side-information" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>(Load example data)</li>
<li>Fetch and show data on the (taxonomic) features of the analyzed samples (Hint: <a href="https://microbiome.github.io/OMA/containers.html#rowdata">rowData</a>)</li>
<li>Get abundance data for all samples given a specific features (Hint: <a href="https://microbiome.github.io/OMA/taxonomic-information.html#abundances-of-specific-taxa-in-all-samples">example</a>)</li>
</ol>
<p>Optional:</p>
<ol start="4" style="list-style-type: decimal">
<li>Create taxonomy tree based on the taxonomy mappings display its information:</li>
</ol>
<ul>
<li><a href="https://microbiome.github.io/OMA/taxonomic-information.html#generate-a-taxonomic-tree-on-the-fly">generate a taxonomic tree on the fly</a></li>
<li><a href="https://microbiome.github.io/OMA/containers.html#rowtree">rowtree</a></li>
</ul>
</div>
<div id="other-elements" class="section level3 hasAnchor" number="17.2.7">
<h3><span class="header-section-number">17.2.7</span> Other elements<a href="exercises.html#other-elements" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Try to extract some of the <a href="https://f1000research.com/articles/9-1246/v2">other TreeSE elements</a>. These are not always included:</p>
<ul>
<li>Experiment metadata</li>
<li>Sample tree (colTree)</li>
<li>Phylogenetic tree (rowTree)</li>
<li>Feature sequences information (DNA sequence slot)</li>
</ul>
</div>
</div>
<div id="data-manipulation" class="section level2 hasAnchor" number="17.3">
<h2><span class="header-section-number">17.3</span> Data manipulation<a href="exercises.html#data-manipulation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="subsetting-1" class="section level3 hasAnchor" number="17.3.1">
<h3><span class="header-section-number">17.3.1</span> Subsetting<a href="exercises.html#subsetting-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>Subset the TreeSE object to specific samples</li>
<li>Subset the TreeSE object to specific features</li>
<li>Subset the TreeSE object to specific samples and features</li>
</ol>
</div>
<div id="library-sizes" class="section level3 hasAnchor" number="17.3.2">
<h3><span class="header-section-number">17.3.2</span> Library sizes<a href="exercises.html#library-sizes" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>Calculate library sizes</li>
<li>Subsample / rarify the counts (see: subsampleCounts)</li>
</ol>
<p>Useful functions: nrow, ncol, dim, summary, table, quantile, unique, addPerCellQC, agglomerateByRank</p>
</div>
<div id="prevalent-and-core-taxonomic-features" class="section level3 hasAnchor" number="17.3.3">
<h3><span class="header-section-number">17.3.3</span> Prevalent and core taxonomic features<a href="exercises.html#prevalent-and-core-taxonomic-features" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>Estimate prevalence for your chosen feature (row, taxonomic group)</li>
<li>Identify all prevalent features and subset the data accordingly</li>
<li>Report the thresholds and the dimensionality of the data before and after subsetting</li>
<li>Visualize prevalence</li>
</ol>
<p>Useful functions: getPrevalence, getPrevalentTaxa, subsetByPrevalentTaxa</p>
</div>
<div id="data-exploration" class="section level3 hasAnchor" number="17.3.4">
<h3><span class="header-section-number">17.3.4</span> Data exploration<a href="exercises.html#data-exploration" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>Summarize sample metadata variables. (How many age groups, how they are distributed? 0%, 25%, 50%, 75%, and 100% quantiles of library size?)</li>
<li>Create two histograms. Another shows the distribution of absolute counts, another shows how CLR transformed values are distributed.</li>
<li>Visualize how relative abundances are distributed between taxa in samples.</li>
</ol>
<p>Useful functions: nrow, ncol, dim, summary, table, quantile, unique, transformCounts, ggplot, wilcox.test, agglomerateByRank, plotAbundance</p>
</div>
<div id="other-functions" class="section level3 hasAnchor" number="17.3.5">
<h3><span class="header-section-number">17.3.5</span> Other functions<a href="exercises.html#other-functions" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>Merge data objects (merge, mergeSEs)</li>
<li>Melt the data for visualization purposes (meltAssay)</li>
</ol>
</div>
<div id="transformations" class="section level3 hasAnchor" number="17.3.6">
<h3><span class="header-section-number">17.3.6</span> Transformations<a href="exercises.html#transformations" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>Transform abundance data with relative abundance and add a relative abundance assay (see <a href="https://microbiome.github.io/OMA/taxonomic-information.html#data-transformation">data-transformation</a>)</li>
<li>Transform abundance data with clr transformation and add a new assay</li>
<li>List the available assays by name</li>
<li>Pick one of the assays and show a subset of it</li>
<li>Subset the entire TreeSE data object, and check how this affects individual (transformed) assays</li>
</ol>
<p>Optional:</p>
<ol start="6" style="list-style-type: decimal">
<li>If the data has phylogenetic tree, perform the phILR transformation</li>
</ol>
</div>
</div>
<div id="abundance-tables" class="section level2 hasAnchor" number="17.4">
<h2><span class="header-section-number">17.4</span> Abundance tables<a href="exercises.html#abundance-tables" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="taxonomic-levels" class="section level3 hasAnchor" number="17.4.1">
<h3><span class="header-section-number">17.4.1</span> Taxonomic levels<a href="exercises.html#taxonomic-levels" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>List the available taxonomic ranks in the data</li>
<li>Merge the data to Phylum level</li>
<li>Report dimensionality before and after aggomeration</li>
</ol>
<p>Optional:</p>
<ol start="4" style="list-style-type: decimal">
<li>Perform CLR transformation on the data; does this affect agglomeration?</li>
<li>List full taxonomic information for some given taxa (Hint: <a href="https://microbiome.github.io/mia/reference/taxonomy-methods.html">mapTaxonomy</a>)</li>
</ol>
<p>Useful functions: <a href="https://microbiome.github.io/mia/reference/taxonomy-methods.html">taxonomyRanks</a>, agglomerateByRank, mergeRows</p>
</div>
<div id="alternative-experiments-altexp" class="section level3 hasAnchor" number="17.4.2">
<h3><span class="header-section-number">17.4.2</span> Alternative experiments (altExp)<a href="exercises.html#alternative-experiments-altexp" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>Create taxonomic abundance tables for all different levels (splitByRanks)</li>
<li>Check the available alternative experiment (altExp) names before and after splitByRanks</li>
<li>Pick specific “experiment” (taxonomic rank) from specific altExp; and a specific assay</li>
</ol>
<p>Optional:</p>
<ol start="4" style="list-style-type: decimal">
<li>Split the data based on other features (splitOn)</li>
</ol>
</div>
</div>
<div id="community-diversity-alpha-diversity" class="section level2 hasAnchor" number="17.5">
<h2><span class="header-section-number">17.5</span> Community diversity (alpha diversity)<a href="exercises.html#community-diversity-alpha-diversity" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="alpha-diversity-basics" class="section level3 hasAnchor" number="17.5.1">
<h3><span class="header-section-number">17.5.1</span> Alpha diversity basics<a href="exercises.html#alpha-diversity-basics" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>Calculate alpha diversity indices</li>
<li>Test if data agglomeration to higher taxonomic ranks affects the indices</li>
<li>Look for differences in alpha diversity between groups or correlation with a continuous variable</li>
</ol>
<p>Useful functions: estimateDiversity, colSums, agglomerateByRank, kruskal.test, cor</p>
</div>
<div id="alpha-diversity-extra" class="section level3 hasAnchor" number="17.5.2">
<h3><span class="header-section-number">17.5.2</span> Alpha diversity extra<a href="exercises.html#alpha-diversity-extra" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>Estimate Shannon diversity for the data</li>
<li>Try also another diversity index and compare the results with a scatterplot</li>
<li>Compare Shannon diversity between groups (boxplot)</li>
<li>Is diversity significantly different between vegan and mixed diet?</li>
<li>Calculate and visualize library size, compare with diversity</li>
</ol>
<p>Useful functions: estimateDiversity, colSums, geom_point, geom_boxplot</p>
</div>
</div>
<div id="community-composition-beta-diversity" class="section level2 hasAnchor" number="17.6">
<h2><span class="header-section-number">17.6</span> Community composition (beta diversity)<a href="exercises.html#community-composition-beta-diversity" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="beta-diversity-basics" class="section level3 hasAnchor" number="17.6.1">
<h3><span class="header-section-number">17.6.1</span> Beta diversity basics<a href="exercises.html#beta-diversity-basics" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>Visualize community variation with different methods (PCA, MDS, NMDS, etc.) with plotReduceDim and with different dissimilarities and transformations,plot also other than the first two axes.</li>
<li>Use PERMANOVA to test differences in beta diversity. You can also try including continuous and/or categorical covariates</li>
<li>If there are statistical differences in PERMANOVA, test PERMDISP2 (betadisper function)</li>
<li>Do clustering</li>
<li>Try RDA to test the variance explained by external variables</li>
</ol>
</div>
<div id="beta-diversity-extra" class="section level3 hasAnchor" number="17.6.2">
<h3><span class="header-section-number">17.6.2</span> Beta diversity extra<a href="exercises.html#beta-diversity-extra" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>Install the latest development version of mia from GitHub.</li>
<li>Load experimental dataset from mia.</li>
<li>Create a PCoA with Aitchison dissimilarities. How much coordinate 1 explains the differences? How about coordinate 2?</li>
<li>Create dbRDA with Bray-Curtis dissimilarities on relative abundances. Use PERMANOVA. Can differences between samples be explained with variables of sample meta data?</li>
<li>Analyze diets’ association on beta diversity. Calculate dbRDA and then PERMANOVA. Visualize coefficients. Which taxa’s abundances differ the most between samples?</li>
<li>Interpret your results. Is there association between community composition and location? What are those taxa that differ the most; find information from literature.</li>
</ol>
<p>Useful functions: runMDS, runRDA, anova.cca, transformCounts, agglomerateByRank, ggplot, plotReducedDim, vegan::adonis2</p>
</div>
</div>
<div id="differential-abundance-1" class="section level2 hasAnchor" number="17.7">
<h2><span class="header-section-number">17.7</span> Differential abundance<a href="exercises.html#differential-abundance-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="univariate-analyses" class="section level3 hasAnchor" number="17.7.1">
<h3><span class="header-section-number">17.7.1</span> Univariate analyses<a href="exercises.html#univariate-analyses" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>Get the abundances for an individual feature (taxonomic group / row)</li>
<li>Visualize the abundances per group with boxplot / jitterplot</li>
<li>Is the difference significant (Wilcoxon test)?</li>
<li>Is the difference significant (linear model with covariates)?</li>
<li>How do transformations affect the outcome (log10, clr..)?</li>
<li>Get p-values for all features (taxa), for instance with a for loop</li>
<li>Do multiple testing correction</li>
<li>Compare the results from different tests with a scatterplot</li>
</ol>
<p>Useful functions: [], ggplot2::geom_boxplot, ggplot2::geom_jitter, wilcox.test, lm.test, transformCounts, p.adjust</p>
</div>
<div id="differential-abundance-analysis-1" class="section level3 hasAnchor" number="17.7.2">
<h3><span class="header-section-number">17.7.2</span> Differential abundance analysis<a href="exercises.html#differential-abundance-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>install the latest development version of mia from GitHub.</li>
<li>Load experimental dataset from mia.</li>
<li>Compare abundances of each taxa between groups. First, use Wilcoxon or Kruskall-Wallis test. Then use some other method dedicated to microbiome data.</li>
<li>Summarize findings by plotting a taxa vs samples heatmap. Add column annotation that tells the group of each sample, and row annotation that tells whether the difference of certain taxa was statistically significant.</li>
<li>Choose statistically significant taxa and visualize their abundances with boxplot &amp; jitterplot.</li>
</ol>
<p>Useful functions: wilcox.test, kruskal.test, ggplot, pheatmap, ComplexHeatMap::Heatmap, ancombc, aldex2, maaslin2, agglomerateByRank, transformCounts, subsetByPrevalentTaxa</p>
</div>
</div>
<div id="visualization-1" class="section level2 hasAnchor" number="17.8">
<h2><span class="header-section-number">17.8</span> Visualization<a href="exercises.html#visualization-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="multivariate-ordination" class="section level3 hasAnchor" number="17.8.1">
<h3><span class="header-section-number">17.8.1</span> Multivariate ordination<a href="exercises.html#multivariate-ordination" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>Load experimental dataset from mia.</li>
<li>Create PCoA with Bray-Curtis dissimilarities</li>
<li>Create PCA with Aitchison dissimilarities</li>
<li>Visualize and compare both</li>
<li>Test other transformations, dissimilarities, and ordination methods</li>
</ol>
<p>Useful functions: runMDS, runNMDS, transformCounts, ggplot, plotReducedDim</p>
</div>
<div id="heatmap-visualization" class="section level3 hasAnchor" number="17.8.2">
<h3><span class="header-section-number">17.8.2</span> Heatmap visualization<a href="exercises.html#heatmap-visualization" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>Load experimental dataset from mia.</li>
<li>Visualize abundances with heatmap</li>
<li>Visualize abundances with heatmap after CLR + Z transformation</li>
</ol>
<p>See the OMA book for examples.</p>
</div>
</div>
<div id="multiomics" class="section level2 hasAnchor" number="17.9">
<h2><span class="header-section-number">17.9</span> Multiomics<a href="exercises.html#multiomics" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="multiassayexperiment-mae-data-container" class="section level3 hasAnchor" number="17.9.1">
<h3><span class="header-section-number">17.9.1</span> MultiAssayExperiment (MAE) data container<a href="exercises.html#multiassayexperiment-mae-data-container" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>Create TreeSE data containers from individual CSV files.</li>
<li>Combine TreeSE into MAE.</li>
<li>Check that each individual experiment of MAE equals corresponding TreeSE.</li>
<li>Take a subset of MAE (e.g., 10 first samples), and observe the subsetted MAE.</li>
</ol>
<p>Useful functions: DataFrame, TreeSummarizedExperiment, matrix, rownames, colnames, MultiAssayExperiment, ExperimentList, SimpleList</p>
</div>
<div id="multi-omic-data-exploration" class="section level3 hasAnchor" number="17.9.2">
<h3><span class="header-section-number">17.9.2</span> Multi-omic data exploration<a href="exercises.html#multi-omic-data-exploration" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>Load experimental dataset from microbiomeDataSets (e.g., HintikkaXOData).</li>
<li>Analyze correlations between experiments. (Taxa vs lipids, Taxa vs biomarkers, Lipids vs biomarkers)</li>
<li>Agglomerate taxa data.</li>
<li>Apply CLR to taxa data, apply log10 to lipids and biomarkers.</li>
<li>Perform cross-correlation analyses and visualize results with heatmaps. (Use Spearman coefficients)</li>
<li>Is there significant correlations? Interpret your results.</li>
</ol>
<p>Useful functions: pheatmap, ComplexHeatMap::Heatmap, ggplot, transformCounts, testExperimentCrossAssociation</p>

</div>
</div>
</div>



            </section>

          </div>
        </div>
      </div>
<a href="resources.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="extras.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["OMA.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
