<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 1 Introduction | Microbiome Analysis</title>
  <meta name="description" content="Chapter 1 Introduction | Microbiome Analysis" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 1 Introduction | Microbiome Analysis" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 1 Introduction | Microbiome Analysis" />
  
  
  

<meta name="author" content="" />


<meta name="date" content="2020-11-08" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="Data-operations.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Microbiome Analysis</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="intro.html"><a href="intro.html#installation"><i class="fa fa-check"></i><b>1.1</b> Installation</a></li>
<li class="chapter" data-level="1.2" data-path="intro.html"><a href="intro.html#package-data"><i class="fa fa-check"></i><b>1.2</b> Package data</a></li>
<li class="chapter" data-level="1.3" data-path="intro.html"><a href="intro.html#tools"><i class="fa fa-check"></i><b>1.3</b> Tools</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="Data-operations.html"><a href="Data-operations.html"><i class="fa fa-check"></i><b>2</b> Data operations</a><ul>
<li class="chapter" data-level="2.1" data-path="Data-operations.html"><a href="Data-operations.html#common-terminologies"><i class="fa fa-check"></i><b>2.1</b> Common terminologies</a></li>
<li class="chapter" data-level="2.2" data-path="Data-operations.html"><a href="Data-operations.html#retrieving-data-elements"><i class="fa fa-check"></i><b>2.2</b> Retrieving data elements</a></li>
<li class="chapter" data-level="2.3" data-path="Data-operations.html"><a href="Data-operations.html#extract-specific-elements"><i class="fa fa-check"></i><b>2.3</b> Extract specific elements</a><ul>
<li class="chapter" data-level="2.3.1" data-path="Data-operations.html"><a href="Data-operations.html#assays"><i class="fa fa-check"></i><b>2.3.1</b> Assays</a></li>
<li class="chapter" data-level="2.3.2" data-path="Data-operations.html"><a href="Data-operations.html#coldata"><i class="fa fa-check"></i><b>2.3.2</b> colData</a></li>
<li class="chapter" data-level="2.3.3" data-path="Data-operations.html"><a href="Data-operations.html#rowdata"><i class="fa fa-check"></i><b>2.3.3</b> rowData</a></li>
<li class="chapter" data-level="2.3.4" data-path="Data-operations.html"><a href="Data-operations.html#rowtree"><i class="fa fa-check"></i><b>2.3.4</b> rowTree</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="Data-operations.html"><a href="Data-operations.html#library-size"><i class="fa fa-check"></i><b>2.4</b> Library size</a></li>
<li class="chapter" data-level="2.5" data-path="Data-operations.html"><a href="Data-operations.html#agglomerate"><i class="fa fa-check"></i><b>2.5</b> Agglomerate</a></li>
<li class="chapter" data-level="2.6" data-path="Data-operations.html"><a href="Data-operations.html#get-unique"><i class="fa fa-check"></i><b>2.6</b> Get unique</a></li>
<li class="chapter" data-level="2.7" data-path="Data-operations.html"><a href="Data-operations.html#pick-specific"><i class="fa fa-check"></i><b>2.7</b> Pick specific</a><ul>
<li class="chapter" data-level="2.7.1" data-path="Data-operations.html"><a href="Data-operations.html#abundances-of-all-taxa-in-specific-sample"><i class="fa fa-check"></i><b>2.7.1</b> Abundances of all taxa in specific sample</a></li>
<li class="chapter" data-level="2.7.2" data-path="Data-operations.html"><a href="Data-operations.html#abundances-of-specific-taxa-in-all-samples"><i class="fa fa-check"></i><b>2.7.2</b> Abundances of specific taxa in all samples</a></li>
</ul></li>
<li class="chapter" data-level="2.8" data-path="Data-operations.html"><a href="Data-operations.html#join"><i class="fa fa-check"></i><b>2.8</b> Join</a><ul>
<li class="chapter" data-level="2.8.1" data-path="Data-operations.html"><a href="Data-operations.html#convert-to-long-data.frame"><i class="fa fa-check"></i><b>2.8.1</b> Convert to long data.frame</a></li>
</ul></li>
<li class="chapter" data-level="2.9" data-path="Data-operations.html"><a href="Data-operations.html#get-top-taxa-and-taxonomy"><i class="fa fa-check"></i><b>2.9</b> Get top taxa and taxonomy</a><ul>
<li class="chapter" data-level="2.9.1" data-path="Data-operations.html"><a href="Data-operations.html#features"><i class="fa fa-check"></i><b>2.9.1</b> Features</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="bibliography.html"><a href="bibliography.html"><i class="fa fa-check"></i>Bibliography</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Microbiome Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="intro" class="section level1">
<h1><span class="header-section-number">Chapter 1</span> Introduction</h1>
<div id="installation" class="section level2">
<h2><span class="header-section-number">1.1</span> Installation</h2>
<p>Install the development version from GitHub using <code>remotes</code> R package.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="intro.html#cb1-1"></a><span class="co"># install remotes </span></span>
<span id="cb1-2"><a href="intro.html#cb1-2"></a><span class="co">#install.packages(&quot;remotes&quot;)</span></span>
<span id="cb1-3"><a href="intro.html#cb1-3"></a>remotes<span class="op">::</span><span class="kw">install_github</span>(<span class="st">&quot;FelixErnst/mia&quot;</span>)</span></code></pre></div>
<pre><code>## Microbiom... (3012ee51f... -&gt; cd40180ab...) [GitHub]
## generics     (0.0.2        -&gt; 0.1.0       ) [CRAN]
## labeling     (0.3          -&gt; 0.4.2       ) [CRAN]
## cpp11        (0.2.2        -&gt; 0.2.4       ) [CRAN]
## 
##   
   checking for file ‘/tmp/Rtmpltr73W/remotes34b7ad6b18e/FelixErnst-MicrobiomeExperiment-cd40180/DESCRIPTION’ ...
  
✔  checking for file ‘/tmp/Rtmpltr73W/remotes34b7ad6b18e/FelixErnst-MicrobiomeExperiment-cd40180/DESCRIPTION’
## 
  
─  preparing ‘MicrobiomeExperiment’:
## 
  
   checking DESCRIPTION meta-information ...
  
✔  checking DESCRIPTION meta-information
## 
  
─  checking for LF line-endings in source and make files and shell scripts
## 
  
─  checking for empty or unneeded directories
## ─  looking to see if a ‘data/datalist’ file should be added
## 
  
─  building ‘MicrobiomeExperiment_0.99.0.9014.tar.gz’ (480ms)
## 
  
   
## 
  
   checking for file ‘/tmp/Rtmpltr73W/remotes34b79f1cc91/FelixErnst-mia-d5ba1c2/DESCRIPTION’ ...
  
✔  checking for file ‘/tmp/Rtmpltr73W/remotes34b79f1cc91/FelixErnst-mia-d5ba1c2/DESCRIPTION’
## 
  
─  preparing ‘mia’:
## 
  
   checking DESCRIPTION meta-information ...
  
✔  checking DESCRIPTION meta-information
## 
  
─  checking for LF line-endings in source and make files and shell scripts
## 
  
─  checking for empty or unneeded directories
## ─  looking to see if a ‘data/datalist’ file should be added
## 
  
─  building ‘mia_0.0.0.9002.tar.gz’ (451ms)
## 
  
   
## </code></pre>
</div>
<div id="package-data" class="section level2">
<h2><span class="header-section-number">1.2</span> Package data</h2>
<p>The datasets in <code>mia</code> are conversions of the <code>phyloseq</code> datasets <code>GlobalPatterns</code> <code>enterotype</code>, <code>esophagus</code> and <code>soilrep</code>.
Taxa contains example taxonomic data derived from the metagenomeFeatures package.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="intro.html#cb3-1"></a><span class="kw">library</span>(mia)</span></code></pre></div>
<pre><code>## Loading required package: SummarizedExperiment</code></pre>
<pre><code>## Loading required package: MatrixGenerics</code></pre>
<pre><code>## Loading required package: matrixStats</code></pre>
<pre><code>## 
## Attaching package: &#39;MatrixGenerics&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:matrixStats&#39;:
## 
##     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
##     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
##     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
##     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
##     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
##     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
##     colWeightedMeans, colWeightedMedians, colWeightedSds,
##     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
##     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
##     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
##     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
##     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
##     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
##     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
##     rowWeightedSds, rowWeightedVars</code></pre>
<pre><code>## Loading required package: GenomicRanges</code></pre>
<pre><code>## Loading required package: stats4</code></pre>
<pre><code>## Loading required package: BiocGenerics</code></pre>
<pre><code>## Loading required package: parallel</code></pre>
<pre><code>## 
## Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:parallel&#39;:
## 
##     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
##     clusterExport, clusterMap, parApply, parCapply, parLapply,
##     parLapplyLB, parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     IQR, mad, sd, var, xtabs</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     anyDuplicated, append, as.data.frame, basename, cbind, colnames,
##     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
##     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
##     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
##     union, unique, unsplit, which.max, which.min</code></pre>
<pre><code>## Loading required package: S4Vectors</code></pre>
<pre><code>## 
## Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:base&#39;:
## 
##     expand.grid</code></pre>
<pre><code>## Loading required package: IRanges</code></pre>
<pre><code>## Loading required package: GenomeInfoDb</code></pre>
<pre><code>## Loading required package: Biobase</code></pre>
<pre><code>## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     &#39;browseVignettes()&#39;. To cite Bioconductor, see
##     &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.</code></pre>
<pre><code>## 
## Attaching package: &#39;Biobase&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:MatrixGenerics&#39;:
## 
##     rowMedians</code></pre>
<pre><code>## The following objects are masked from &#39;package:matrixStats&#39;:
## 
##     anyMissing, rowMedians</code></pre>
<pre><code>## Loading required package: SingleCellExperiment</code></pre>
<pre><code>## Loading required package: TreeSummarizedExperiment</code></pre>
<pre><code>## Loading required package: MicrobiomeExperiment</code></pre>
<pre><code>## Loading required package: Biostrings</code></pre>
<pre><code>## Loading required package: XVector</code></pre>
<pre><code>## 
## Attaching package: &#39;Biostrings&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:base&#39;:
## 
##     strsplit</code></pre>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="intro.html#cb34-1"></a><span class="co"># Example how to load data</span></span>
<span id="cb34-2"><a href="intro.html#cb34-2"></a><span class="kw">data</span>(<span class="st">&quot;GlobalPatterns&quot;</span>)</span>
<span id="cb34-3"><a href="intro.html#cb34-3"></a></span>
<span id="cb34-4"><a href="intro.html#cb34-4"></a>GlobalPatterns</span></code></pre></div>
<pre><code>## class: TreeSummarizedExperiment 
## dim: 19216 26 
## metadata(0):
## assays(1): counts
## rownames(19216): 549322 522457 ... 200359 271582
## rowData names(7): Kingdom Phylum ... Genus Species
## colnames(26): CL3 CC1 ... Even2 Even3
## colData names(7): X.SampleID Primer ... SampleType Description
## reducedDimNames(0):
## altExpNames(0):
## rowLinks: a LinkDataFrame (19216 rows)
## rowTree: a phylo (19216 leaves)
## colLinks: NULL
## colTree: NULL</code></pre>
</div>
<div id="tools" class="section level2">
<h2><span class="header-section-number">1.3</span> Tools</h2>
<ol style="list-style-type: decimal">
<li><code>mia</code> : Microbiome analysis tools<br />
</li>
<li><code>miaViz</code> : Visualization support for <code>mia</code></li>
</ol>
<p><strong>See also:</strong></p>
<p><a href="https://www.bioconductor.org/packages/release/bioc/html/TreeSummarizedExperiment.html"><code>TreeSummarizedExperiment</code></a><br />
<a href="https://github.com/FelixErnst/MicrobiomeExperiment"><code>MicrobiomeExperiment</code></a> <a href="https://bioconductor.org/packages/release/bioc/html/SummarizedExperiment.html"><code>SummarizedExperiment</code></a><br />
<a href="https://bioconductor.org/packages/devel/bioc/html/microbiome.html"><code>microbiome</code></a></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="Data-operations.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["MiaBook.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
