<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 12 Machine learning | Orchestrating Microbiome Analysis</title>
  <meta name="description" content="Chapter 12 Machine learning | Orchestrating Microbiome Analysis" />
  <meta name="generator" content="bookdown 0.33 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 12 Machine learning | Orchestrating Microbiome Analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="microbiome/OMA" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 12 Machine learning | Orchestrating Microbiome Analysis" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="differential-abundance.html"/>
<link rel="next" href="multi-assay-analyses.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Orchestrating Microbiome Analysis</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="part"><span><b>I Introduction</b></span></li>
<li class="chapter" data-level="1" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="containers.html"><a href="containers.html"><i class="fa fa-check"></i><b>2</b> Microbiome Data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="containers.html"><a href="containers.html#data-science-framework"><i class="fa fa-check"></i><b>2.1</b> Data science framework</a></li>
<li class="chapter" data-level="2.2" data-path="containers.html"><a href="containers.html#data-containers"><i class="fa fa-check"></i><b>2.2</b> Data containers</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="containers.html"><a href="containers.html#assay-data"><i class="fa fa-check"></i><b>2.2.1</b> Assay data</a></li>
<li class="chapter" data-level="2.2.2" data-path="containers.html"><a href="containers.html#coldata"><i class="fa fa-check"></i><b>2.2.2</b> colData</a></li>
<li class="chapter" data-level="2.2.3" data-path="containers.html"><a href="containers.html#rowdata"><i class="fa fa-check"></i><b>2.2.3</b> rowData</a></li>
<li class="chapter" data-level="2.2.4" data-path="containers.html"><a href="containers.html#rowtree"><i class="fa fa-check"></i><b>2.2.4</b> rowTree</a></li>
<li class="chapter" data-level="2.2.5" data-path="containers.html"><a href="containers.html#alt-exp"><i class="fa fa-check"></i><b>2.2.5</b> Alternative experiments</a></li>
<li class="chapter" data-level="2.2.6" data-path="containers.html"><a href="containers.html#mae"><i class="fa fa-check"></i><b>2.2.6</b> MultiAssayExperiments</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="containers.html"><a href="containers.html#loading-experimental-microbiome-data"><i class="fa fa-check"></i><b>2.3</b> Loading experimental microbiome data</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="containers.html"><a href="containers.html#s-workflow"><i class="fa fa-check"></i><b>2.3.1</b> 16S workflow</a></li>
<li class="chapter" data-level="2.3.2" data-path="containers.html"><a href="containers.html#import-from-external-files"><i class="fa fa-check"></i><b>2.3.2</b> Import from external files</a></li>
<li class="chapter" data-level="2.3.3" data-path="containers.html"><a href="containers.html#conversions-between-data-formats-in-r"><i class="fa fa-check"></i><b>2.3.3</b> Conversions between data formats in R</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="containers.html"><a href="containers.html#example-data"><i class="fa fa-check"></i><b>2.4</b> Demonstration data</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="containers.html"><a href="containers.html#package-data"><i class="fa fa-check"></i><b>2.4.1</b> Package data</a></li>
<li class="chapter" data-level="2.4.2" data-path="containers.html"><a href="containers.html#experimenthub-data"><i class="fa fa-check"></i><b>2.4.2</b> ExperimentHub data</a></li>
<li class="chapter" data-level="2.4.3" data-path="containers.html"><a href="containers.html#other-data-sources"><i class="fa fa-check"></i><b>2.4.3</b> Other data sources</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="containers.html"><a href="containers.html#session-info"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="packages.html"><a href="packages.html"><i class="fa fa-check"></i><b>3</b> Packages</a>
<ul>
<li class="chapter" data-level="3.1" data-path="packages.html"><a href="packages.html#package-installation"><i class="fa fa-check"></i><b>3.1</b> Package installation</a></li>
<li class="chapter" data-level="3.2" data-path="packages.html"><a href="packages.html#some-available-packages"><i class="fa fa-check"></i><b>3.2</b> Some available packages</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="packages.html"><a href="packages.html#mia-family-of-methods"><i class="fa fa-check"></i><b>3.2.1</b> mia family of methods</a></li>
<li class="chapter" data-level="3.2.2" data-path="packages.html"><a href="packages.html#tree-based-methods-sub-tree-methods"><i class="fa fa-check"></i><b>3.2.2</b> Tree-based methods {sub-tree-methods}</a></li>
<li class="chapter" data-level="3.2.3" data-path="packages.html"><a href="packages.html#differential-abundance-sub-diff-abund"><i class="fa fa-check"></i><b>3.2.3</b> Differential abundance {sub-diff-abund}</a></li>
<li class="chapter" data-level="3.2.4" data-path="packages.html"><a href="packages.html#manipulation-sub-manipulation"><i class="fa fa-check"></i><b>3.2.4</b> Manipulation {sub-manipulation}</a></li>
<li class="chapter" data-level="3.2.5" data-path="packages.html"><a href="packages.html#data"><i class="fa fa-check"></i><b>3.2.5</b> Data</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="packages.html"><a href="packages.html#session-info-1"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="part"><span><b>II Focus Topics</b></span></li>
<li class="chapter" data-level="4" data-path="datamanipulation.html"><a href="datamanipulation.html"><i class="fa fa-check"></i><b>4</b> Data Manipulation</a>
<ul>
<li class="chapter" data-level="4.1" data-path="datamanipulation.html"><a href="datamanipulation.html#tidying-and-subsetting"><i class="fa fa-check"></i><b>4.1</b> Tidying and subsetting</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="datamanipulation.html"><a href="datamanipulation.html#tidy-data"><i class="fa fa-check"></i><b>4.1.1</b> Tidy data</a></li>
<li class="chapter" data-level="4.1.2" data-path="datamanipulation.html"><a href="datamanipulation.html#subsetting"><i class="fa fa-check"></i><b>4.1.2</b> Subsetting</a></li>
<li class="chapter" data-level="4.1.3" data-path="datamanipulation.html"><a href="datamanipulation.html#splitting"><i class="fa fa-check"></i><b>4.1.3</b> Splitting</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="datamanipulation.html"><a href="datamanipulation.html#merge-data"><i class="fa fa-check"></i><b>4.2</b> Merge data</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="datamanipulation.html"><a href="datamanipulation.html#additional-functions"><i class="fa fa-check"></i><b>4.2.1</b> Additional functions</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>5</b> Exploration and quality Control</a>
<ul>
<li class="chapter" data-level="5.1" data-path="quality-control.html"><a href="quality-control.html#abundance"><i class="fa fa-check"></i><b>5.1</b> Abundance</a></li>
<li class="chapter" data-level="5.2" data-path="quality-control.html"><a href="quality-control.html#prevalence"><i class="fa fa-check"></i><b>5.2</b> Prevalence</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="quality-control.html"><a href="quality-control.html#prevalence-analysis"><i class="fa fa-check"></i><b>5.2.1</b> Prevalence analysis</a></li>
<li class="chapter" data-level="5.2.2" data-path="quality-control.html"><a href="quality-control.html#rare-taxa"><i class="fa fa-check"></i><b>5.2.2</b> Rare taxa</a></li>
<li class="chapter" data-level="5.2.3" data-path="quality-control.html"><a href="quality-control.html#plotting-prevalence"><i class="fa fa-check"></i><b>5.2.3</b> Plotting prevalence</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="quality-control.html"><a href="quality-control.html#qc"><i class="fa fa-check"></i><b>5.3</b> Quality control</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="quality-control.html"><a href="quality-control.html#top-taxa"><i class="fa fa-check"></i><b>5.3.1</b> Top taxa</a></li>
<li class="chapter" data-level="5.3.2" data-path="quality-control.html"><a href="quality-control.html#library-size-read-count"><i class="fa fa-check"></i><b>5.3.2</b> Library size / read count</a></li>
<li class="chapter" data-level="5.3.3" data-path="quality-control.html"><a href="quality-control.html#contaminant-sequences"><i class="fa fa-check"></i><b>5.3.3</b> Contaminant sequences</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="quality-control.html"><a href="quality-control.html#session-info-2"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="taxonomic-information.html"><a href="taxonomic-information.html"><i class="fa fa-check"></i><b>6</b> Taxonomic Information</a>
<ul>
<li class="chapter" data-level="6.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#assigning-taxonomic-information."><i class="fa fa-check"></i><b>6.1</b> Assigning taxonomic information.</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#dada2"><i class="fa fa-check"></i><b>6.1.1</b> dada2</a></li>
<li class="chapter" data-level="6.1.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#decipher"><i class="fa fa-check"></i><b>6.1.2</b> DECIPHER</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#functions-to-access-taxonomic-information"><i class="fa fa-check"></i><b>6.2</b> Functions to access taxonomic information</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#generate-a-taxonomic-tree-on-the-fly"><i class="fa fa-check"></i><b>6.2.1</b> Generate a taxonomic tree on the fly</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="taxonomic-information.html"><a href="taxonomic-information.html#data-agglomeration"><i class="fa fa-check"></i><b>6.3</b> Data agglomeration</a></li>
<li class="chapter" data-level="6.4" data-path="taxonomic-information.html"><a href="taxonomic-information.html#data-transformation"><i class="fa fa-check"></i><b>6.4</b> Data transformation</a></li>
<li class="chapter" data-level="6.5" data-path="taxonomic-information.html"><a href="taxonomic-information.html#pick-specific"><i class="fa fa-check"></i><b>6.5</b> Pick specific</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#abundances-of-all-taxa-in-specific-sample"><i class="fa fa-check"></i><b>6.5.1</b> Abundances of all taxa in specific sample</a></li>
<li class="chapter" data-level="6.5.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#abundances-of-specific-taxa-in-all-samples"><i class="fa fa-check"></i><b>6.5.2</b> Abundances of specific taxa in all samples</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="taxonomic-information.html"><a href="taxonomic-information.html#session-info-3"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="community-diversity.html"><a href="community-diversity.html"><i class="fa fa-check"></i><b>7</b> Community diversity</a>
<ul>
<li class="chapter" data-level="7.1" data-path="community-diversity.html"><a href="community-diversity.html#estimation"><i class="fa fa-check"></i><b>7.1</b> Estimation</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="community-diversity.html"><a href="community-diversity.html#richness"><i class="fa fa-check"></i><b>7.1.1</b> Richness</a></li>
<li class="chapter" data-level="7.1.2" data-path="community-diversity.html"><a href="community-diversity.html#estimate-diversity"><i class="fa fa-check"></i><b>7.1.2</b> Diversity</a></li>
<li class="chapter" data-level="7.1.3" data-path="community-diversity.html"><a href="community-diversity.html#faith-diversity"><i class="fa fa-check"></i><b>7.1.3</b> Faith phylogenetic diversity</a></li>
<li class="chapter" data-level="7.1.4" data-path="community-diversity.html"><a href="community-diversity.html#evenness"><i class="fa fa-check"></i><b>7.1.4</b> Evenness</a></li>
<li class="chapter" data-level="7.1.5" data-path="community-diversity.html"><a href="community-diversity.html#dominance"><i class="fa fa-check"></i><b>7.1.5</b> Dominance</a></li>
<li class="chapter" data-level="7.1.6" data-path="community-diversity.html"><a href="community-diversity.html#rarity"><i class="fa fa-check"></i><b>7.1.6</b> Rarity</a></li>
<li class="chapter" data-level="7.1.7" data-path="community-diversity.html"><a href="community-diversity.html#divergence"><i class="fa fa-check"></i><b>7.1.7</b> Divergence</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="community-diversity.html"><a href="community-diversity.html#visualization"><i class="fa fa-check"></i><b>7.2</b> Visualization</a></li>
<li class="chapter" data-level="" data-path="community-diversity.html"><a href="community-diversity.html#session-info-4"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="community-similarity.html"><a href="community-similarity.html"><i class="fa fa-check"></i><b>8</b> Community similarity</a>
<ul>
<li class="chapter" data-level="8.1" data-path="community-similarity.html"><a href="community-similarity.html#explained-variance"><i class="fa fa-check"></i><b>8.1</b> Explained variance</a></li>
<li class="chapter" data-level="8.2" data-path="community-similarity.html"><a href="community-similarity.html#community-comparisons-by-beta-diversity-analysis"><i class="fa fa-check"></i><b>8.2</b> Community comparisons by beta diversity analysis</a></li>
<li class="chapter" data-level="8.3" data-path="community-similarity.html"><a href="community-similarity.html#other-ordination-methods"><i class="fa fa-check"></i><b>8.3</b> Other ordination methods</a></li>
<li class="chapter" data-level="8.4" data-path="community-similarity.html"><a href="community-similarity.html#pcoa-genus"><i class="fa fa-check"></i><b>8.4</b> Visualizing the most dominant genus on PCoA</a>
<ul>
<li class="chapter" data-level="8.4.1" data-path="community-similarity.html"><a href="community-similarity.html#testing-differences-in-community-composition-between-sample-groups"><i class="fa fa-check"></i><b>8.4.1</b> Testing differences in community composition between sample groups</a></li>
<li class="chapter" data-level="8.4.2" data-path="community-similarity.html"><a href="community-similarity.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>8.4.2</b> Checking the homogeneity condition</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="community-similarity.html"><a href="community-similarity.html#further-reading"><i class="fa fa-check"></i><b>8.5</b> Further reading</a></li>
<li class="chapter" data-level="" data-path="community-similarity.html"><a href="community-similarity.html#session-info-5"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="microbiome-community.html"><a href="microbiome-community.html"><i class="fa fa-check"></i><b>9</b> Community composition</a>
<ul>
<li class="chapter" data-level="9.1" data-path="microbiome-community.html"><a href="microbiome-community.html#visual-composition"><i class="fa fa-check"></i><b>9.1</b> Visualizing taxonomic composition</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="microbiome-community.html"><a href="microbiome-community.html#composition-barplot"><i class="fa fa-check"></i><b>9.1.1</b> Composition barplot</a></li>
<li class="chapter" data-level="9.1.2" data-path="microbiome-community.html"><a href="microbiome-community.html#composition-heatmap"><i class="fa fa-check"></i><b>9.1.2</b> Composition heatmap</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="microbiome-community.html"><a href="microbiome-community.html#session-info-6"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="clustering.html"><a href="clustering.html"><i class="fa fa-check"></i><b>10</b> Community typing (clustering)</a>
<ul>
<li class="chapter" data-level="10.1" data-path="clustering.html"><a href="clustering.html#hiearchical-clustering"><i class="fa fa-check"></i><b>10.1</b> Hiearchical clustering</a></li>
<li class="chapter" data-level="10.2" data-path="clustering.html"><a href="clustering.html#k-means-clustering"><i class="fa fa-check"></i><b>10.2</b> K-means clustering</a></li>
<li class="chapter" data-level="10.3" data-path="clustering.html"><a href="clustering.html#dirichlet-multinomial-mixtures-dmm"><i class="fa fa-check"></i><b>10.3</b> Dirichlet Multinomial Mixtures (DMM)</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="clustering.html"><a href="clustering.html#pcoa-for-asv-level-data-with-bray-curtis-with-dmm-clusters-shown-with-colors"><i class="fa fa-check"></i><b>10.3.1</b> PCoA for ASV-level data with Bray-Curtis; with DMM clusters shown with colors</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="clustering.html"><a href="clustering.html#community-detection"><i class="fa fa-check"></i><b>10.4</b> Community Detection</a></li>
<li class="chapter" data-level="10.5" data-path="clustering.html"><a href="clustering.html#biclustering"><i class="fa fa-check"></i><b>10.5</b> Biclustering</a>
<ul>
<li class="chapter" data-level="10.5.1" data-path="clustering.html"><a href="clustering.html#taxa-vs-samples"><i class="fa fa-check"></i><b>10.5.1</b> Taxa vs samples</a></li>
<li class="chapter" data-level="10.5.2" data-path="clustering.html"><a href="clustering.html#taxa-vs-biomolecules"><i class="fa fa-check"></i><b>10.5.2</b> Taxa vs biomolecules</a></li>
<li class="chapter" data-level="10.5.3" data-path="clustering.html"><a href="clustering.html#taxa-vs-taxa"><i class="fa fa-check"></i><b>10.5.3</b> Taxa vs taxa</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="clustering.html"><a href="clustering.html#additional-community-typing"><i class="fa fa-check"></i><b>10.6</b> Additional Community Typing</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="differential-abundance.html"><a href="differential-abundance.html"><i class="fa fa-check"></i><b>11</b> Differential abundance</a>
<ul>
<li class="chapter" data-level="11.1" data-path="differential-abundance.html"><a href="differential-abundance.html#differential-abundance-analysis"><i class="fa fa-check"></i><b>11.1</b> Differential abundance analysis</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="differential-abundance.html"><a href="differential-abundance.html#examples-and-tools"><i class="fa fa-check"></i><b>11.1.1</b> Examples and tools</a></li>
<li class="chapter" data-level="11.1.2" data-path="differential-abundance.html"><a href="differential-abundance.html#prevalence-filtering"><i class="fa fa-check"></i><b>11.1.2</b> Prevalence Filtering</a></li>
<li class="chapter" data-level="11.1.3" data-path="differential-abundance.html"><a href="differential-abundance.html#aldex2"><i class="fa fa-check"></i><b>11.1.3</b> ALDEx2</a></li>
<li class="chapter" data-level="11.1.4" data-path="differential-abundance.html"><a href="differential-abundance.html#ancom-bc"><i class="fa fa-check"></i><b>11.1.4</b> ANCOM-BC</a></li>
<li class="chapter" data-level="11.1.5" data-path="differential-abundance.html"><a href="differential-abundance.html#maaslin2"><i class="fa fa-check"></i><b>11.1.5</b> MaAsLin2</a></li>
<li class="chapter" data-level="11.1.6" data-path="differential-abundance.html"><a href="differential-abundance.html#linda"><i class="fa fa-check"></i><b>11.1.6</b> LinDA</a></li>
<li class="chapter" data-level="11.1.7" data-path="differential-abundance.html"><a href="differential-abundance.html#comparison-of-the-methods"><i class="fa fa-check"></i><b>11.1.7</b> Comparison of the methods</a></li>
<li class="chapter" data-level="11.1.8" data-path="differential-abundance.html"><a href="differential-abundance.html#confounding-variables"><i class="fa fa-check"></i><b>11.1.8</b> Confounding variables</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="differential-abundance.html"><a href="differential-abundance.html#tree-based-methods"><i class="fa fa-check"></i><b>11.2</b> Tree-based methods</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="differential-abundance.html"><a href="differential-abundance.html#group-wise-associations-testing-based-on-balances"><i class="fa fa-check"></i><b>11.2.1</b> Group-wise associations testing based on balances</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="differential-abundance.html"><a href="differential-abundance.html#session-info-7"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="machine_learning.html"><a href="machine_learning.html"><i class="fa fa-check"></i><b>12</b> Machine learning</a>
<ul>
<li class="chapter" data-level="12.1" data-path="machine_learning.html"><a href="machine_learning.html#supervised-machine-learning"><i class="fa fa-check"></i><b>12.1</b> Supervised machine learning</a></li>
<li class="chapter" data-level="12.2" data-path="machine_learning.html"><a href="machine_learning.html#unsupervised-machine-learning"><i class="fa fa-check"></i><b>12.2</b> Unsupervised machine learning</a></li>
<li class="chapter" data-level="" data-path="machine_learning.html"><a href="machine_learning.html#session-info-8"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="multi-assay-analyses.html"><a href="multi-assay-analyses.html"><i class="fa fa-check"></i><b>13</b> Multi-assay analyses</a>
<ul>
<li class="chapter" data-level="13.1" data-path="multi-assay-analyses.html"><a href="multi-assay-analyses.html#cross-correlation-analysis"><i class="fa fa-check"></i><b>13.1</b> Cross-correlation Analysis</a></li>
<li class="chapter" data-level="13.2" data-path="multi-assay-analyses.html"><a href="multi-assay-analyses.html#multi-omics-factor-analysis"><i class="fa fa-check"></i><b>13.2</b> Multi-Omics Factor Analysis</a></li>
<li class="chapter" data-level="" data-path="multi-assay-analyses.html"><a href="multi-assay-analyses.html#session-info-9"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="viz-chapter.html"><a href="viz-chapter.html"><i class="fa fa-check"></i><b>14</b> Visualization</a>
<ul>
<li class="chapter" data-level="14.1" data-path="viz-chapter.html"><a href="viz-chapter.html#pre-analysis-exploration"><i class="fa fa-check"></i><b>14.1</b> Pre-analysis exploration</a>
<ul>
<li class="chapter" data-level="14.1.1" data-path="viz-chapter.html"><a href="viz-chapter.html#accessing-row-and-column-data"><i class="fa fa-check"></i><b>14.1.1</b> Accessing row and column data</a></li>
<li class="chapter" data-level="14.1.2" data-path="viz-chapter.html"><a href="viz-chapter.html#viewing-abundance-and-prevalence-patterns"><i class="fa fa-check"></i><b>14.1.2</b> Viewing abundance and prevalence patterns</a></li>
</ul></li>
<li class="chapter" data-level="14.2" data-path="viz-chapter.html"><a href="viz-chapter.html#diversity-estimation"><i class="fa fa-check"></i><b>14.2</b> Diversity estimation</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="viz-chapter.html"><a href="viz-chapter.html#alpha-diversity-with-scatter-violin-and-box-plots"><i class="fa fa-check"></i><b>14.2.1</b> Alpha diversity with scatter, violin and box plots</a></li>
<li class="chapter" data-level="14.2.2" data-path="viz-chapter.html"><a href="viz-chapter.html#beta-diversity-with-shepard-and-coordination-plots"><i class="fa fa-check"></i><b>14.2.2</b> Beta diversity with Shepard and coordination plots</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="viz-chapter.html"><a href="viz-chapter.html#statistical-analysis"><i class="fa fa-check"></i><b>14.3</b> Statistical analysis</a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="viz-chapter.html"><a href="viz-chapter.html#heatmaps"><i class="fa fa-check"></i><b>14.3.1</b> Heatmaps</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="viz-chapter.html"><a href="viz-chapter.html#session-info-10"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="part"><span><b>III Appendix</b></span></li>
<li class="chapter" data-level="15" data-path="training.html"><a href="training.html"><i class="fa fa-check"></i><b>15</b> Training</a>
<ul>
<li class="chapter" data-level="15.1" data-path="training.html"><a href="training.html#coc"><i class="fa fa-check"></i><b>15.1</b> Code of Conduct</a></li>
<li class="chapter" data-level="15.2" data-path="training.html"><a href="training.html#checklist"><i class="fa fa-check"></i><b>15.2</b> Checklist</a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="training.html"><a href="training.html#setting-up-your-own-computer"><i class="fa fa-check"></i><b>15.2.1</b> Setting up Your own computer</a></li>
<li class="chapter" data-level="15.2.2" data-path="training.html"><a href="training.html#support-and-resources"><i class="fa fa-check"></i><b>15.2.2</b> Support and resources</a></li>
<li class="chapter" data-level="15.2.3" data-path="training.html"><a href="training.html#material"><i class="fa fa-check"></i><b>15.2.3</b> Study material</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="packages.html"><a href="packages.html#packages"><i class="fa fa-check"></i><b>15.3</b> Installing and loading the required R packages</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="resources.html"><a href="resources.html"><i class="fa fa-check"></i><b>16</b> Resources</a>
<ul>
<li class="chapter" data-level="16.1" data-path="resources.html"><a href="resources.html#data-containers-1"><i class="fa fa-check"></i><b>16.1</b> Data containers</a>
<ul>
<li class="chapter" data-level="16.1.1" data-path="resources.html"><a href="resources.html#resources-for-treesummarizedexperiment"><i class="fa fa-check"></i><b>16.1.1</b> Resources for TreeSummarizedExperiment</a></li>
<li class="chapter" data-level="16.1.2" data-path="resources.html"><a href="resources.html#other-relevant-containers"><i class="fa fa-check"></i><b>16.1.2</b> Other relevant containers</a></li>
<li class="chapter" data-level="16.1.3" data-path="resources.html"><a href="resources.html#alternative-containers-for-microbiome-data"><i class="fa fa-check"></i><b>16.1.3</b> Alternative containers for microbiome data</a></li>
<li class="chapter" data-level="16.1.4" data-path="resources.html"><a href="resources.html#resources-for-phyloseq"><i class="fa fa-check"></i><b>16.1.4</b> Resources for phyloseq</a></li>
</ul></li>
<li class="chapter" data-level="16.2" data-path="resources.html"><a href="resources.html#r-programming-resources"><i class="fa fa-check"></i><b>16.2</b> R programming resources</a>
<ul>
<li class="chapter" data-level="16.2.1" data-path="resources.html"><a href="resources.html#bioc_intro"><i class="fa fa-check"></i><b>16.2.1</b> Bioconductor Classes</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="resources.html"><a href="resources.html#reproducible-reporting-with-quarto"><i class="fa fa-check"></i><b>16.3</b> Reproducible reporting with Quarto</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="acknowledgments.html"><a href="acknowledgments.html"><i class="fa fa-check"></i><b>17</b> Acknowledgments</a></li>
<li class="chapter" data-level="18" data-path="exercises.html"><a href="exercises.html"><i class="fa fa-check"></i><b>18</b> Exercises</a>
<ul>
<li class="chapter" data-level="18.1" data-path="exercises.html"><a href="exercises.html#workflows"><i class="fa fa-check"></i><b>18.1</b> Workflows</a>
<ul>
<li class="chapter" data-level="18.1.1" data-path="exercises.html"><a href="exercises.html#reproducible-reporting"><i class="fa fa-check"></i><b>18.1.1</b> Reproducible reporting</a></li>
</ul></li>
<li class="chapter" data-level="18.2" data-path="exercises.html"><a href="exercises.html#treesummarizedexperiment-basic-components"><i class="fa fa-check"></i><b>18.2</b> TreeSummarizedExperiment: basic components</a>
<ul>
<li class="chapter" data-level="18.2.1" data-path="exercises.html"><a href="exercises.html#basic-summaries"><i class="fa fa-check"></i><b>18.2.1</b> Basic summaries</a></li>
<li class="chapter" data-level="18.2.2" data-path="exercises.html"><a href="exercises.html#taxonomic-abundance-table-assay"><i class="fa fa-check"></i><b>18.2.2</b> Taxonomic abundance table (assay)</a></li>
<li class="chapter" data-level="18.2.3" data-path="exercises.html"><a href="exercises.html#sample-side-information"><i class="fa fa-check"></i><b>18.2.3</b> Sample side information</a></li>
<li class="chapter" data-level="18.2.4" data-path="exercises.html"><a href="exercises.html#feature-side-information"><i class="fa fa-check"></i><b>18.2.4</b> Feature side information</a></li>
<li class="chapter" data-level="18.2.5" data-path="exercises.html"><a href="exercises.html#other-components"><i class="fa fa-check"></i><b>18.2.5</b> Other components</a></li>
</ul></li>
<li class="chapter" data-level="18.3" data-path="exercises.html"><a href="exercises.html#treesummarizedexperiment-data-import"><i class="fa fa-check"></i><b>18.3</b> TreeSummarizedExperiment: data import</a></li>
<li class="chapter" data-level="18.4" data-path="exercises.html"><a href="exercises.html#data-manipulation"><i class="fa fa-check"></i><b>18.4</b> Data manipulation</a>
<ul>
<li class="chapter" data-level="18.4.1" data-path="exercises.html"><a href="exercises.html#subsetting-1"><i class="fa fa-check"></i><b>18.4.1</b> Subsetting</a></li>
<li class="chapter" data-level="18.4.2" data-path="exercises.html"><a href="exercises.html#library-sizes"><i class="fa fa-check"></i><b>18.4.2</b> Library sizes</a></li>
<li class="chapter" data-level="18.4.3" data-path="exercises.html"><a href="exercises.html#prevalent-and-core-taxonomic-features"><i class="fa fa-check"></i><b>18.4.3</b> Prevalent and core taxonomic features</a></li>
<li class="chapter" data-level="18.4.4" data-path="exercises.html"><a href="exercises.html#explore-the-data"><i class="fa fa-check"></i><b>18.4.4</b> Explore the data</a></li>
<li class="chapter" data-level="18.4.5" data-path="exercises.html"><a href="exercises.html#other-functions"><i class="fa fa-check"></i><b>18.4.5</b> Other functions</a></li>
</ul></li>
<li class="chapter" data-level="18.5" data-path="exercises.html"><a href="exercises.html#transformations"><i class="fa fa-check"></i><b>18.5</b> Transformations</a>
<ul>
<li class="chapter" data-level="18.5.1" data-path="exercises.html"><a href="exercises.html#transformations-1"><i class="fa fa-check"></i><b>18.5.1</b> Transformations</a></li>
</ul></li>
<li class="chapter" data-level="18.6" data-path="exercises.html"><a href="exercises.html#taxonomic-levels"><i class="fa fa-check"></i><b>18.6</b> Taxonomic levels</a>
<ul>
<li class="chapter" data-level="18.6.1" data-path="exercises.html"><a href="exercises.html#taxonomic-levels-1"><i class="fa fa-check"></i><b>18.6.1</b> Taxonomic levels</a></li>
<li class="chapter" data-level="18.6.2" data-path="exercises.html"><a href="exercises.html#alternative-experiments-altexp"><i class="fa fa-check"></i><b>18.6.2</b> Alternative experiments (altExp)</a></li>
</ul></li>
<li class="chapter" data-level="18.7" data-path="exercises.html"><a href="exercises.html#alpha-diversity"><i class="fa fa-check"></i><b>18.7</b> Alpha diversity</a>
<ul>
<li class="chapter" data-level="18.7.1" data-path="exercises.html"><a href="exercises.html#alpha-diversity-basics"><i class="fa fa-check"></i><b>18.7.1</b> Alpha diversity basics</a></li>
<li class="chapter" data-level="18.7.2" data-path="exercises.html"><a href="exercises.html#alpha-diversity-extra"><i class="fa fa-check"></i><b>18.7.2</b> Alpha diversity extra</a></li>
</ul></li>
<li class="chapter" data-level="18.8" data-path="exercises.html"><a href="exercises.html#community-composition"><i class="fa fa-check"></i><b>18.8</b> Community composition</a>
<ul>
<li class="chapter" data-level="18.8.1" data-path="exercises.html"><a href="exercises.html#beta-diversity-basics"><i class="fa fa-check"></i><b>18.8.1</b> Beta diversity basics</a></li>
<li class="chapter" data-level="18.8.2" data-path="exercises.html"><a href="exercises.html#beta-diversity-extra"><i class="fa fa-check"></i><b>18.8.2</b> Beta diversity extra</a></li>
</ul></li>
<li class="chapter" data-level="18.9" data-path="exercises.html"><a href="exercises.html#visualization-1"><i class="fa fa-check"></i><b>18.9</b> Visualization</a>
<ul>
<li class="chapter" data-level="18.9.1" data-path="exercises.html"><a href="exercises.html#multivariate-ordination"><i class="fa fa-check"></i><b>18.9.1</b> Multivariate ordination</a></li>
<li class="chapter" data-level="18.9.2" data-path="exercises.html"><a href="exercises.html#heatmap-visualization"><i class="fa fa-check"></i><b>18.9.2</b> Heatmap visualization</a></li>
</ul></li>
<li class="chapter" data-level="18.10" data-path="exercises.html"><a href="exercises.html#differential-abundance-1"><i class="fa fa-check"></i><b>18.10</b> Differential abundance</a>
<ul>
<li class="chapter" data-level="18.10.1" data-path="exercises.html"><a href="exercises.html#univariate-analyses"><i class="fa fa-check"></i><b>18.10.1</b> Univariate analyses</a></li>
<li class="chapter" data-level="18.10.2" data-path="exercises.html"><a href="exercises.html#differential-abundance-analysis-1"><i class="fa fa-check"></i><b>18.10.2</b> Differential abundance analysis</a></li>
</ul></li>
<li class="chapter" data-level="18.11" data-path="exercises.html"><a href="exercises.html#multiomics"><i class="fa fa-check"></i><b>18.11</b> Multiomics</a>
<ul>
<li class="chapter" data-level="18.11.1" data-path="exercises.html"><a href="exercises.html#introduction-to-multiassayexperiment-mae"><i class="fa fa-check"></i><b>18.11.1</b> Introduction to MultiAssayExperiment (MAE)</a></li>
<li class="chapter" data-level="18.11.2" data-path="exercises.html"><a href="exercises.html#introduction-to-multiomics"><i class="fa fa-check"></i><b>18.11.2</b> Introduction to multiomics</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="bibliography.html"><a href="bibliography.html"><i class="fa fa-check"></i>Bibliography</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Orchestrating Microbiome Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="machine_learning" class="section level1 hasAnchor" number="12">
<h1><span class="header-section-number">Chapter 12</span> Machine learning<a href="machine_learning.html#machine_learning" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<script>
document.addEventListener("click", function (event) {
    if (event.target.classList.contains("rebook-collapse")) {
        event.target.classList.toggle("active");
        var content = event.target.nextElementSibling;
        if (content.style.display === "block") {
            content.style.display = "none";
        } else {
            content.style.display = "block";
        }
    }
})
</script>
<style>
.rebook-collapse {
  background-color: #eee;
  color: #444;
  cursor: pointer;
  padding: 18px;
  width: 100%;
  border: none;
  text-align: left;
  outline: none;
  font-size: 15px;
}

.rebook-content {
  padding: 0 18px;
  display: none;
  overflow: hidden;
  background-color: #f1f1f1;
}
</style>
<p>Machine learning (ML) is a part of artificial intelligence. There are multiple
definitions, but “machine” refers to computation and “learning” to improving
performance based on the data by finding patterns from it. Machine learning
includes wide variety of methods from simple statistical methods to more
complex methods such as neural-networks.</p>
<p>Machine learning can be divided into supervised and unsupervised machine learning.
Supervised ML is used to predict outcome based on the data. Unsupervised ML is used,
for example, to reduce dimensionality (e.g. PCA) and to find clusters from the
data (e.g., k-means clustering).</p>
<div id="supervised-machine-learning" class="section level2 hasAnchor" number="12.1">
<h2><span class="header-section-number">12.1</span> Supervised machine learning<a href="machine_learning.html#supervised-machine-learning" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>“Supervised” means that the training data is introduced before. The training data
contains labels (e.g., patient status), and the model is fitted based on the
training data. After fitting, the model is utilized to predict labels of data whose
labels are not known.</p>
<div class="sourceCode" id="cb344"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb344-1"><a href="machine_learning.html#cb344-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mia)</span>
<span id="cb344-2"><a href="machine_learning.html#cb344-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb344-3"><a href="machine_learning.html#cb344-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Load experimental data</span></span>
<span id="cb344-4"><a href="machine_learning.html#cb344-4" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(peerj13075)</span>
<span id="cb344-5"><a href="machine_learning.html#cb344-5" aria-hidden="true" tabindex="-1"></a>(tse <span class="ot">&lt;-</span> peerj13075)</span></code></pre></div>
<pre><code>## class: TreeSummarizedExperiment 
## dim: 674 58 
## metadata(0):
## assays(1): counts
## rownames(674): OTU1 OTU2 ... OTU2567 OTU2569
## rowData names(6): kingdom phylum ... family genus
## colnames(58): ID1 ID2 ... ID57 ID58
## colData names(5): Sample Geographical_location Gender Age Diet
## reducedDimNames(0):
## mainExpName: NULL
## altExpNames(0):
## rowLinks: NULL
## rowTree: NULL
## colLinks: NULL
## colTree: NULL</code></pre>
<p>Let’s first preprocess the data.</p>
<div class="sourceCode" id="cb346"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb346-1"><a href="machine_learning.html#cb346-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Agglomerate data</span></span>
<span id="cb346-2"><a href="machine_learning.html#cb346-2" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">agglomerateByRank</span>(tse, <span class="at">rank =</span> <span class="st">&quot;order&quot;</span>)</span>
<span id="cb346-3"><a href="machine_learning.html#cb346-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb346-4"><a href="machine_learning.html#cb346-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply CLR transform</span></span>
<span id="cb346-5"><a href="machine_learning.html#cb346-5" aria-hidden="true" tabindex="-1"></a><span class="fu">assay</span>(tse, <span class="st">&quot;pseudo&quot;</span>) <span class="ot">&lt;-</span> <span class="fu">assay</span>(tse, <span class="st">&quot;counts&quot;</span>) <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb346-6"><a href="machine_learning.html#cb346-6" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">transformCounts</span>(tse, <span class="at">assay_name =</span> <span class="st">&quot;pseudo&quot;</span>, <span class="at">method =</span> <span class="st">&quot;relabundance&quot;</span>)</span>
<span id="cb346-7"><a href="machine_learning.html#cb346-7" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">transformCounts</span>(tse, <span class="at">assay_name =</span> <span class="st">&quot;relabundance&quot;</span>, <span class="at">method =</span> <span class="st">&quot;clr&quot;</span>)</span>
<span id="cb346-8"><a href="machine_learning.html#cb346-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb346-9"><a href="machine_learning.html#cb346-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Get assay</span></span>
<span id="cb346-10"><a href="machine_learning.html#cb346-10" aria-hidden="true" tabindex="-1"></a>assay <span class="ot">&lt;-</span> <span class="fu">assay</span>(tse, <span class="st">&quot;clr&quot;</span>)</span>
<span id="cb346-11"><a href="machine_learning.html#cb346-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Transpose assay</span></span>
<span id="cb346-12"><a href="machine_learning.html#cb346-12" aria-hidden="true" tabindex="-1"></a>assay <span class="ot">&lt;-</span> <span class="fu">t</span>(assay)</span>
<span id="cb346-13"><a href="machine_learning.html#cb346-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb346-14"><a href="machine_learning.html#cb346-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert into data.frame</span></span>
<span id="cb346-15"><a href="machine_learning.html#cb346-15" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(assay)</span>
<span id="cb346-16"><a href="machine_learning.html#cb346-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb346-17"><a href="machine_learning.html#cb346-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Add labels to assay</span></span>
<span id="cb346-18"><a href="machine_learning.html#cb346-18" aria-hidden="true" tabindex="-1"></a>labels <span class="ot">&lt;-</span> <span class="fu">colData</span>(tse)<span class="sc">$</span>Diet</span>
<span id="cb346-19"><a href="machine_learning.html#cb346-19" aria-hidden="true" tabindex="-1"></a>labels <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(labels)</span>
<span id="cb346-20"><a href="machine_learning.html#cb346-20" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>diet <span class="ot">&lt;-</span> labels </span>
<span id="cb346-21"><a href="machine_learning.html#cb346-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb346-22"><a href="machine_learning.html#cb346-22" aria-hidden="true" tabindex="-1"></a>df[<span class="dv">5</span>, <span class="dv">5</span>]</span></code></pre></div>
<pre><code>## [1] -0.4612</code></pre>
<p>In the example below, we use <a href="https://journals.asm.org/doi/10.1128/mBio.00434-20">mikropml</a>
package. We try to predict the diet type based on the data.</p>
<div class="sourceCode" id="cb348"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb348-1"><a href="machine_learning.html#cb348-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>( <span class="sc">!</span><span class="fu">require</span>(<span class="st">&quot;mikropml&quot;</span>) ){</span>
<span id="cb348-2"><a href="machine_learning.html#cb348-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;mikropml&quot;</span>)</span>
<span id="cb348-3"><a href="machine_learning.html#cb348-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(mikropml)</span>
<span id="cb348-4"><a href="machine_learning.html#cb348-4" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb348-5"><a href="machine_learning.html#cb348-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb348-6"><a href="machine_learning.html#cb348-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Run random forest </span></span>
<span id="cb348-7"><a href="machine_learning.html#cb348-7" aria-hidden="true" tabindex="-1"></a>results <span class="ot">&lt;-</span> <span class="fu">run_ml</span>(df, <span class="st">&quot;rf&quot;</span>, <span class="at">outcome_colname =</span> <span class="st">&#39;diet&#39;</span>, </span>
<span id="cb348-8"><a href="machine_learning.html#cb348-8" aria-hidden="true" tabindex="-1"></a>                  <span class="at">kfold =</span> <span class="dv">2</span>, <span class="at">cv_times =</span> <span class="dv">5</span>, <span class="at">training_frac =</span> <span class="fl">0.8</span>)</span>
<span id="cb348-9"><a href="machine_learning.html#cb348-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb348-10"><a href="machine_learning.html#cb348-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Print result</span></span>
<span id="cb348-11"><a href="machine_learning.html#cb348-11" aria-hidden="true" tabindex="-1"></a><span class="fu">confusionMatrix</span>(<span class="at">data =</span> results<span class="sc">$</span>trained_model<span class="sc">$</span>finalModel<span class="sc">$</span>predicted, </span>
<span id="cb348-12"><a href="machine_learning.html#cb348-12" aria-hidden="true" tabindex="-1"></a>                <span class="at">reference =</span> results<span class="sc">$</span>trained_model<span class="sc">$</span>finalModel<span class="sc">$</span>y)</span></code></pre></div>
<pre><code>## Confusion Matrix and Statistics
## 
##           Reference
## Prediction Mixed Veg
##      Mixed    12   9
##      Veg      11  15
##                                         
##                Accuracy : 0.574         
##                  95% CI : (0.422, 0.717)
##     No Information Rate : 0.511         
##     P-Value [Acc &gt; NIR] : 0.233         
##                                         
##                   Kappa : 0.147         
##                                         
##  Mcnemar&#39;s Test P-Value : 0.823         
##                                         
##             Sensitivity : 0.522         
##             Specificity : 0.625         
##          Pos Pred Value : 0.571         
##          Neg Pred Value : 0.577         
##              Prevalence : 0.489         
##          Detection Rate : 0.255         
##    Detection Prevalence : 0.447         
##       Balanced Accuracy : 0.573         
##                                         
##        &#39;Positive&#39; Class : Mixed         
## </code></pre>
<p>mikropml offers easier interface to <a href="https://cran.r-project.org/web/packages/caret/index.html">caret</a>
package. However, we can also use it directly.</p>
<p>Let’s use xgboost model which is another commonly used algorithm in bioinformatics.</p>
<div class="sourceCode" id="cb350"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb350-1"><a href="machine_learning.html#cb350-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Set seed for reproducibility</span></span>
<span id="cb350-2"><a href="machine_learning.html#cb350-2" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">6358</span>)</span>
<span id="cb350-3"><a href="machine_learning.html#cb350-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb350-4"><a href="machine_learning.html#cb350-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Specify train control</span></span>
<span id="cb350-5"><a href="machine_learning.html#cb350-5" aria-hidden="true" tabindex="-1"></a>train_control <span class="ot">&lt;-</span> <span class="fu">trainControl</span>(<span class="at">method =</span> <span class="st">&quot;cv&quot;</span>, <span class="at">number =</span> <span class="dv">5</span>,</span>
<span id="cb350-6"><a href="machine_learning.html#cb350-6" aria-hidden="true" tabindex="-1"></a>                              <span class="at">classProbs =</span> <span class="cn">TRUE</span>, </span>
<span id="cb350-7"><a href="machine_learning.html#cb350-7" aria-hidden="true" tabindex="-1"></a>                              <span class="at">savePredictions =</span> <span class="st">&quot;final&quot;</span>,</span>
<span id="cb350-8"><a href="machine_learning.html#cb350-8" aria-hidden="true" tabindex="-1"></a>                              <span class="at">allowParallel =</span> <span class="cn">TRUE</span>)</span>
<span id="cb350-9"><a href="machine_learning.html#cb350-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb350-10"><a href="machine_learning.html#cb350-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Specify hyperparameter tuning grid</span></span>
<span id="cb350-11"><a href="machine_learning.html#cb350-11" aria-hidden="true" tabindex="-1"></a>tune_grid <span class="ot">&lt;-</span> <span class="fu">expand.grid</span>(<span class="at">nrounds =</span> <span class="fu">c</span>(<span class="dv">50</span>, <span class="dv">100</span>, <span class="dv">200</span>),</span>
<span id="cb350-12"><a href="machine_learning.html#cb350-12" aria-hidden="true" tabindex="-1"></a>                         <span class="at">max_depth =</span> <span class="fu">c</span>(<span class="dv">6</span>, <span class="dv">8</span>, <span class="dv">10</span>),</span>
<span id="cb350-13"><a href="machine_learning.html#cb350-13" aria-hidden="true" tabindex="-1"></a>                         <span class="at">colsample_bytree =</span> <span class="fu">c</span>(<span class="fl">0.6</span>, <span class="fl">0.8</span>, <span class="dv">1</span>),</span>
<span id="cb350-14"><a href="machine_learning.html#cb350-14" aria-hidden="true" tabindex="-1"></a>                         <span class="at">eta =</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="fl">0.3</span>),</span>
<span id="cb350-15"><a href="machine_learning.html#cb350-15" aria-hidden="true" tabindex="-1"></a>                         <span class="at">gamma =</span> <span class="dv">0</span>,</span>
<span id="cb350-16"><a href="machine_learning.html#cb350-16" aria-hidden="true" tabindex="-1"></a>                         <span class="at">min_child_weight =</span> <span class="fu">c</span>(<span class="dv">3</span>, <span class="dv">4</span>, <span class="dv">5</span>),</span>
<span id="cb350-17"><a href="machine_learning.html#cb350-17" aria-hidden="true" tabindex="-1"></a>                         <span class="at">subsample =</span> <span class="fu">c</span>(<span class="fl">0.6</span>, <span class="fl">0.8</span>)</span>
<span id="cb350-18"><a href="machine_learning.html#cb350-18" aria-hidden="true" tabindex="-1"></a>                         )</span>
<span id="cb350-19"><a href="machine_learning.html#cb350-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb350-20"><a href="machine_learning.html#cb350-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Train the model, use LOOCV to evaluate performance</span></span>
<span id="cb350-21"><a href="machine_learning.html#cb350-21" aria-hidden="true" tabindex="-1"></a>model <span class="ot">&lt;-</span> <span class="fu">train</span>(<span class="at">x =</span> assay, </span>
<span id="cb350-22"><a href="machine_learning.html#cb350-22" aria-hidden="true" tabindex="-1"></a>               <span class="at">y =</span> labels, </span>
<span id="cb350-23"><a href="machine_learning.html#cb350-23" aria-hidden="true" tabindex="-1"></a>               <span class="at">method =</span> <span class="st">&quot;xgbTree&quot;</span>,</span>
<span id="cb350-24"><a href="machine_learning.html#cb350-24" aria-hidden="true" tabindex="-1"></a>               <span class="at">objective =</span> <span class="st">&quot;binary:logistic&quot;</span>,</span>
<span id="cb350-25"><a href="machine_learning.html#cb350-25" aria-hidden="true" tabindex="-1"></a>               <span class="at">trControl =</span> train_control,</span>
<span id="cb350-26"><a href="machine_learning.html#cb350-26" aria-hidden="true" tabindex="-1"></a>               <span class="at">tuneGrid =</span> tune_grid,</span>
<span id="cb350-27"><a href="machine_learning.html#cb350-27" aria-hidden="true" tabindex="-1"></a>               <span class="at">metric =</span> <span class="st">&quot;AUC&quot;</span>,</span>
<span id="cb350-28"><a href="machine_learning.html#cb350-28" aria-hidden="true" tabindex="-1"></a>               <span class="at">verbosity =</span> <span class="dv">0</span></span>
<span id="cb350-29"><a href="machine_learning.html#cb350-29" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p>Let’s create ROC curve which is a commonly used method in binary classification.
For unbalanced data, you might want to plot precision-recall curve.</p>
<div class="sourceCode" id="cb351"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb351-1"><a href="machine_learning.html#cb351-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>( <span class="sc">!</span><span class="fu">require</span>(MLeval) ){</span>
<span id="cb351-2"><a href="machine_learning.html#cb351-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;MLeval&quot;</span>)</span>
<span id="cb351-3"><a href="machine_learning.html#cb351-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(MLeval)</span>
<span id="cb351-4"><a href="machine_learning.html#cb351-4" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb351-5"><a href="machine_learning.html#cb351-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate different evaluation metrics</span></span>
<span id="cb351-6"><a href="machine_learning.html#cb351-6" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">evalm</span>(model, <span class="at">showplots =</span> <span class="cn">FALSE</span>)</span>
<span id="cb351-7"><a href="machine_learning.html#cb351-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb351-8"><a href="machine_learning.html#cb351-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Use patchwork to plot ROC and precision-recall curve side-by-side</span></span>
<span id="cb351-9"><a href="machine_learning.html#cb351-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb351-10"><a href="machine_learning.html#cb351-10" aria-hidden="true" tabindex="-1"></a>res<span class="sc">$</span>roc <span class="sc">+</span> res<span class="sc">$</span>proc <span class="sc">+</span> </span>
<span id="cb351-11"><a href="machine_learning.html#cb351-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plot_layout</span>(<span class="at">guides =</span> <span class="st">&quot;collect&quot;</span>) <span class="sc">&amp;</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&#39;bottom&#39;</span>)</span></code></pre></div>
<p><img src="40_machine_learning_files/figure-html/super5-1.png" width="672" /></p>
</div>
<div id="unsupervised-machine-learning" class="section level2 hasAnchor" number="12.2">
<h2><span class="header-section-number">12.2</span> Unsupervised machine learning<a href="machine_learning.html#unsupervised-machine-learning" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>“Unsupervised” means that the labels (e.g., patient status is not known),
and patterns are learned based only the abundance table, for instance.
Unsupervised ML is also known as a data mining where patterns are extracted
from big datasets.</p>
<p>For unsupervised machine learning, please refer to chapters that are listed below:</p>
<ul>
<li>Chapter <a href="clustering.html#clustering">10</a></li>
<li>Chapter <a href="community-similarity.html#community-similarity">8</a></li>
</ul>
</div>
<div id="session-info-8" class="section level2 unnumbered hasAnchor">
<h2>Session Info<a href="machine_learning.html#session-info-8" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<button class="rebook-collapse">
View session info
</button>
<div class="rebook-content">
<pre><code>R version 4.2.1 (2022-06-23)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.4 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] patchwork_1.1.2                MLeval_0.3                    
 [3] caret_6.0-93                   lattice_0.20-45               
 [5] ggplot2_3.4.1                  mikropml_1.5.0                
 [7] mia_1.7.9                      MultiAssayExperiment_1.24.0   
 [9] TreeSummarizedExperiment_2.1.4 Biostrings_2.66.0             
[11] XVector_0.38.0                 SingleCellExperiment_1.20.0   
[13] SummarizedExperiment_1.28.0    Biobase_2.58.0                
[15] GenomicRanges_1.50.2           GenomeInfoDb_1.34.9           
[17] IRanges_2.32.0                 S4Vectors_0.36.1              
[19] BiocGenerics_0.44.0            MatrixGenerics_1.10.0         
[21] matrixStats_0.63.0-9003        BiocStyle_2.24.0              
[23] rebook_1.6.0                  

loaded via a namespace (and not attached):
  [1] plyr_1.8.8                  lazyeval_0.2.2             
  [3] splines_4.2.1               BiocParallel_1.32.5        
  [5] listenv_0.9.0               scater_1.26.1              
  [7] digest_0.6.31               foreach_1.5.2              
  [9] yulab.utils_0.0.6           htmltools_0.5.4            
 [11] viridis_0.6.2               fansi_1.0.4                
 [13] magrittr_2.0.3              memoise_2.0.1              
 [15] MLmetrics_1.1.1             ScaledMatrix_1.6.0         
 [17] cluster_2.1.4               ROCR_1.0-11                
 [19] DECIPHER_2.26.0             recipes_1.0.4              
 [21] globals_0.16.2              gower_1.0.1                
 [23] hardhat_1.2.0               timechange_0.2.0           
 [25] colorspace_2.1-0            blob_1.2.3                 
 [27] ggrepel_0.9.3               xfun_0.37                  
 [29] dplyr_1.1.0                 crayon_1.5.2               
 [31] RCurl_1.98-1.10             jsonlite_1.8.4             
 [33] graph_1.74.0                survival_3.5-3             
 [35] iterators_1.0.14            ape_5.7                    
 [37] glue_1.6.2                  gtable_0.3.1               
 [39] ipred_0.9-13                zlibbioc_1.44.0            
 [41] DelayedArray_0.24.0         kernlab_0.9-32             
 [43] BiocSingular_1.14.0         shape_1.4.6                
 [45] future.apply_1.10.0         scales_1.2.1               
 [47] DBI_1.1.3                   Rcpp_1.0.10                
 [49] viridisLite_0.4.1           decontam_1.18.0            
 [51] tidytree_0.4.2              proxy_0.4-27               
 [53] bit_4.0.5                   rsvd_1.0.5                 
 [55] lava_1.7.1                  prodlim_2019.11.13         
 [57] glmnet_4.1-6                dir.expiry_1.4.0           
 [59] farver_2.1.1                pkgconfig_2.0.3            
 [61] XML_3.99-0.13               scuttle_1.8.4              
 [63] nnet_7.3-18                 CodeDepends_0.6.5          
 [65] sass_0.4.5                  utf8_1.2.3                 
 [67] labeling_0.4.2              tidyselect_1.2.0           
 [69] rlang_1.0.6                 reshape2_1.4.4             
 [71] munsell_0.5.0               tools_4.2.1                
 [73] cachem_1.0.7                xgboost_1.7.3.1            
 [75] cli_3.6.0                   DirichletMultinomial_1.40.0
 [77] generics_0.1.3              RSQLite_2.3.0              
 [79] evaluate_0.20               stringr_1.5.0              
 [81] fastmap_1.1.1               yaml_2.3.7                 
 [83] ModelMetrics_1.2.2.2        knitr_1.42                 
 [85] bit64_4.0.5                 randomForest_4.7-1.1       
 [87] purrr_1.0.1                 future_1.32.0              
 [89] nlme_3.1-162                sparseMatrixStats_1.10.0   
 [91] compiler_4.2.1              beeswarm_0.4.0             
 [93] filelock_1.0.2              e1071_1.7-13               
 [95] treeio_1.22.0               tibble_3.2.0               
 [97] bslib_0.4.2                 stringi_1.7.12             
 [99] highr_0.10                  Matrix_1.5-3               
[101] vegan_2.6-4                 permute_0.9-7              
[103] vctrs_0.5.2                 pillar_1.8.1               
[105] lifecycle_1.0.3             BiocManager_1.30.20        
[107] jquerylib_0.1.4             BiocNeighbors_1.16.0       
[109] data.table_1.14.8           bitops_1.0-7               
[111] irlba_2.3.5.1               R6_2.5.1                   
[113] bookdown_0.33               gridExtra_2.3              
[115] vipor_0.4.5                 parallelly_1.34.0          
[117] codetools_0.2-19            MASS_7.3-58.3              
[119] withr_2.5.0                 GenomeInfoDbData_1.2.9     
[121] mgcv_1.8-42                 parallel_4.2.1             
[123] grid_4.2.1                  rpart_4.1.19               
[125] beachmat_2.14.0             timeDate_4022.108          
[127] tidyr_1.3.0                 class_7.3-21               
[129] rmarkdown_2.20              DelayedMatrixStats_1.20.0  
[131] pROC_1.18.0                 lubridate_1.9.2            
[133] ggbeeswarm_0.7.1           </code></pre>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="differential-abundance.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="multi-assay-analyses.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["OMA.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
