<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 Microbiome Diversity | Microbiome Analysis</title>
  <meta name="description" content="Chapter 6 Microbiome Diversity | Microbiome Analysis" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 Microbiome Diversity | Microbiome Analysis" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://github.com/microbiome/MiaBook" />
  
  
  <meta name="github-repo" content="microbiome/MiaBook" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 Microbiome Diversity | Microbiome Analysis" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="microbiome-exploration.html"/>
<link rel="next" href="microbiome-community.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Microbiome Analysis</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="part"><span><b>I Introduction</b></span></li>
<li class="chapter" data-level="1" data-path="data-introduction.html"><a href="data-introduction.html"><i class="fa fa-check"></i><b>1</b> Data Infrastructure</a><ul>
<li class="chapter" data-level="1.1" data-path="data-introduction.html"><a href="data-introduction.html#installation"><i class="fa fa-check"></i><b>1.1</b> Installation</a><ul>
<li class="chapter" data-level="1.1.1" data-path="data-introduction.html"><a href="data-introduction.html#packages"><i class="fa fa-check"></i><b>1.1.1</b> Packages</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="data-introduction.html"><a href="data-introduction.html#background"><i class="fa fa-check"></i><b>1.2</b> Background</a></li>
<li class="chapter" data-level="1.3" data-path="data-introduction.html"><a href="data-introduction.html#loading-experimental-microbiome-data"><i class="fa fa-check"></i><b>1.3</b> Loading experimental microbiome data</a></li>
<li class="chapter" data-level="1.4" data-path="data-introduction.html"><a href="data-introduction.html#metadata"><i class="fa fa-check"></i><b>1.4</b> Metadata</a></li>
<li class="chapter" data-level="1.5" data-path="data-introduction.html"><a href="data-introduction.html#microbiome-and-tree-data-specific-aspects"><i class="fa fa-check"></i><b>1.5</b> Microbiome and tree data specific aspects</a><ul>
<li class="chapter" data-level="1.5.1" data-path="data-introduction.html"><a href="data-introduction.html#assays"><i class="fa fa-check"></i><b>1.5.1</b> Assays</a></li>
<li class="chapter" data-level="1.5.2" data-path="data-introduction.html"><a href="data-introduction.html#coldata"><i class="fa fa-check"></i><b>1.5.2</b> colData</a></li>
<li class="chapter" data-level="1.5.3" data-path="data-introduction.html"><a href="data-introduction.html#rowdata"><i class="fa fa-check"></i><b>1.5.3</b> rowData</a></li>
<li class="chapter" data-level="1.5.4" data-path="data-introduction.html"><a href="data-introduction.html#rowtree"><i class="fa fa-check"></i><b>1.5.4</b> rowTree</a></li>
</ul></li>
<li class="chapter" data-level="1.6" data-path="data-introduction.html"><a href="data-introduction.html#data-conversion"><i class="fa fa-check"></i><b>1.6</b> Data conversion</a><ul>
<li class="chapter" data-level="1.6.1" data-path="data-introduction.html"><a href="data-introduction.html#tidy-data"><i class="fa fa-check"></i><b>1.6.1</b> Tidy data</a></li>
</ul></li>
<li class="chapter" data-level="1.7" data-path="data-introduction.html"><a href="data-introduction.html#conclusion"><i class="fa fa-check"></i><b>1.7</b> Conclusion</a></li>
<li class="chapter" data-level="" data-path="data-introduction.html"><a href="data-introduction.html#session-info"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="part"><span><b>II Focus Topics</b></span></li>
<li class="chapter" data-level="2" data-path="overview.html"><a href="overview.html"><i class="fa fa-check"></i><b>2</b> Overview</a><ul>
<li class="chapter" data-level="2.1" data-path="overview.html"><a href="overview.html#introduction"><i class="fa fa-check"></i><b>2.1</b> Introduction</a></li>
<li class="chapter" data-level="2.2" data-path="overview.html"><a href="overview.html#microbiome-experiment-types"><i class="fa fa-check"></i><b>2.2</b> Microbiome experiment types</a></li>
<li class="chapter" data-level="2.3" data-path="overview.html"><a href="overview.html#processing-steps"><i class="fa fa-check"></i><b>2.3</b> Processing steps</a></li>
<li class="chapter" data-level="2.4" data-path="overview.html"><a href="overview.html#quick-start"><i class="fa fa-check"></i><b>2.4</b> Quick Start</a></li>
<li class="chapter" data-level="" data-path="overview.html"><a href="overview.html#session-info-1"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="taxonomic-information.html"><a href="taxonomic-information.html"><i class="fa fa-check"></i><b>3</b> Taxonomic information</a><ul>
<li class="chapter" data-level="3.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#assigning-taxonomic-information."><i class="fa fa-check"></i><b>3.1</b> Assigning taxonomic information.</a><ul>
<li class="chapter" data-level="3.1.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#dada2"><i class="fa fa-check"></i><b>3.1.1</b> dada2</a></li>
<li class="chapter" data-level="3.1.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#decipher"><i class="fa fa-check"></i><b>3.1.2</b> DECIPHER</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#functions-to-access-taxonomic-information"><i class="fa fa-check"></i><b>3.2</b> Functions to access taxonomic information</a><ul>
<li class="chapter" data-level="3.2.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#generate-a-taxonomic-tree-on-the-fly"><i class="fa fa-check"></i><b>3.2.1</b> Generate a taxonomic tree on the fly</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="taxonomic-information.html"><a href="taxonomic-information.html#data-agglomeration"><i class="fa fa-check"></i><b>3.3</b> Data agglomeration</a></li>
<li class="chapter" data-level="3.4" data-path="taxonomic-information.html"><a href="taxonomic-information.html#pick-specific"><i class="fa fa-check"></i><b>3.4</b> Pick specific</a><ul>
<li class="chapter" data-level="3.4.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#abundances-of-all-taxa-in-specific-sample"><i class="fa fa-check"></i><b>3.4.1</b> Abundances of all taxa in specific sample</a></li>
<li class="chapter" data-level="3.4.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#abundances-of-specific-taxa-in-all-samples"><i class="fa fa-check"></i><b>3.4.2</b> Abundances of specific taxa in all samples</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="taxonomic-information.html"><a href="taxonomic-information.html#session-info-2"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>4</b> Quality Control</a><ul>
<li class="chapter" data-level="4.1" data-path="quality-control.html"><a href="quality-control.html#get-top-taxa-and-taxonomy"><i class="fa fa-check"></i><b>4.1</b> Get top taxa and taxonomy</a><ul>
<li class="chapter" data-level="4.1.1" data-path="quality-control.html"><a href="quality-control.html#features"><i class="fa fa-check"></i><b>4.1.1</b> Features</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="quality-control.html"><a href="quality-control.html#library-size"><i class="fa fa-check"></i><b>4.2</b> Library size</a></li>
<li class="chapter" data-level="" data-path="quality-control.html"><a href="quality-control.html#session-info-3"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html"><i class="fa fa-check"></i><b>5</b> Microbiome Exploration</a><ul>
<li class="chapter" data-level="5.1" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#prevalence"><i class="fa fa-check"></i><b>5.1</b> Prevalence</a><ul>
<li class="chapter" data-level="5.1.1" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#prevalent-microbiota-analysis"><i class="fa fa-check"></i><b>5.1.1</b> Prevalent microbiota analysis</a></li>
<li class="chapter" data-level="5.1.2" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#plotting-prevalence"><i class="fa fa-check"></i><b>5.1.2</b> Plotting prevalence</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#session-info-4"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="microbiome-diversity.html"><a href="microbiome-diversity.html"><i class="fa fa-check"></i><b>6</b> Microbiome Diversity</a><ul>
<li class="chapter" data-level="6.1" data-path="microbiome-diversity.html"><a href="microbiome-diversity.html#alpha-diversity-indices"><i class="fa fa-check"></i><b>6.1</b> Alpha diversity indices</a><ul>
<li class="chapter" data-level="6.1.1" data-path="microbiome-diversity.html"><a href="microbiome-diversity.html#estimating-alpha-diversity"><i class="fa fa-check"></i><b>6.1.1</b> Estimating alpha diversity</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="microbiome-diversity.html"><a href="microbiome-diversity.html#beta-diversity"><i class="fa fa-check"></i><b>6.2</b> Beta diversity</a><ul>
<li class="chapter" data-level="6.2.1" data-path="microbiome-diversity.html"><a href="microbiome-diversity.html#estimating-beta-diversity"><i class="fa fa-check"></i><b>6.2.1</b> Estimating beta diversity</a></li>
<li class="chapter" data-level="6.2.2" data-path="microbiome-diversity.html"><a href="microbiome-diversity.html#other-ordination-methods"><i class="fa fa-check"></i><b>6.2.2</b> Other ordination methods</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="microbiome-diversity.html"><a href="microbiome-diversity.html#community-comparisons-todo-combine-with-the-material-above-for-simplicity"><i class="fa fa-check"></i><b>6.3</b> Community comparisons [TODO combine with the material above for simplicity?]</a><ul>
<li class="chapter" data-level="6.3.1" data-path="microbiome-diversity.html"><a href="microbiome-diversity.html#testing-differences-in-community-composition-between-sample-groups"><i class="fa fa-check"></i><b>6.3.1</b> Testing differences in community composition between sample groups</a></li>
<li class="chapter" data-level="6.3.2" data-path="microbiome-diversity.html"><a href="microbiome-diversity.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>6.3.2</b> Checking the homogeneity condition</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="microbiome-diversity.html"><a href="microbiome-diversity.html#further-reading"><i class="fa fa-check"></i><b>6.4</b> Further reading</a></li>
<li class="chapter" data-level="" data-path="microbiome-diversity.html"><a href="microbiome-diversity.html#session-info-5"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="microbiome-community.html"><a href="microbiome-community.html"><i class="fa fa-check"></i><b>7</b> Microbiome Community</a><ul>
<li class="chapter" data-level="7.1" data-path="microbiome-community.html"><a href="microbiome-community.html#community-composition"><i class="fa fa-check"></i><b>7.1</b> Community composition</a></li>
<li class="chapter" data-level="7.2" data-path="microbiome-community.html"><a href="microbiome-community.html#community-typing"><i class="fa fa-check"></i><b>7.2</b> Community typing</a></li>
<li class="chapter" data-level="" data-path="microbiome-community.html"><a href="microbiome-community.html#session-info-6"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="differential-abundance.html"><a href="differential-abundance.html"><i class="fa fa-check"></i><b>8</b> Differential abundance</a><ul>
<li class="chapter" data-level="" data-path="differential-abundance.html"><a href="differential-abundance.html#session-info-7"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="microbiome-timeseries.html"><a href="microbiome-timeseries.html"><i class="fa fa-check"></i><b>9</b> Microbiome time series</a><ul>
<li class="chapter" data-level="9.1" data-path="microbiome-timeseries.html"><a href="microbiome-timeseries.html#stability"><i class="fa fa-check"></i><b>9.1</b> Stability</a></li>
<li class="chapter" data-level="9.2" data-path="microbiome-timeseries.html"><a href="microbiome-timeseries.html#tipping-elements"><i class="fa fa-check"></i><b>9.2</b> Tipping elements</a></li>
<li class="chapter" data-level="" data-path="microbiome-timeseries.html"><a href="microbiome-timeseries.html#session-info-8"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="multitable.html"><a href="multitable.html"><i class="fa fa-check"></i><b>10</b> Multi-table methods</a><ul>
<li class="chapter" data-level="" data-path="multitable.html"><a href="multitable.html#session-info-9"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="part"><span><b>III Appendix</b></span></li>
<li class="chapter" data-level="11" data-path="example-data.html"><a href="example-data.html"><i class="fa fa-check"></i><b>11</b> Example data</a><ul>
<li class="chapter" data-level="11.1" data-path="example-data.html"><a href="example-data.html#package-data"><i class="fa fa-check"></i><b>11.1</b> Package data</a><ul>
<li class="chapter" data-level="11.1.1" data-path="example-data.html"><a href="example-data.html#globalpatterns"><i class="fa fa-check"></i><b>11.1.1</b> GlobalPatterns</a></li>
<li class="chapter" data-level="11.1.2" data-path="example-data.html"><a href="example-data.html#enterotype"><i class="fa fa-check"></i><b>11.1.2</b> Enterotype</a></li>
<li class="chapter" data-level="11.1.3" data-path="example-data.html"><a href="example-data.html#esophagus"><i class="fa fa-check"></i><b>11.1.3</b> Esophagus</a></li>
<li class="chapter" data-level="11.1.4" data-path="example-data.html"><a href="example-data.html#soilrep"><i class="fa fa-check"></i><b>11.1.4</b> Soilrep</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="example-data.html"><a href="example-data.html#session-info-10"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="bibliography.html"><a href="bibliography.html"><i class="fa fa-check"></i>Bibliography</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Microbiome Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="microbiome-diversity" class="section level1">
<h1><span class="header-section-number">Chapter 6</span> Microbiome Diversity</h1>
<script>
document.addEventListener("click", function (event) {
    if (event.target.classList.contains("rebook-collapse")) {
        event.target.classList.toggle("active");
        var content = event.target.nextElementSibling;
        if (content.style.display === "block") {
            content.style.display = "none";
        } else {
            content.style.display = "block";
        }
    }
})
</script>
<style>
.rebook-collapse {
  background-color: #eee;
  color: #444;
  cursor: pointer;
  padding: 18px;
  width: 100%;
  border: none;
  text-align: left;
  outline: none;
  font-size: 15px;
}

.rebook-content {
  padding: 0 18px;
  display: none;
  overflow: hidden;
  background-color: #f1f1f1;
}
</style>
<p>Diversity estimates are a central topic in microbiome data analysis.</p>
<p>There are three commonly employed levels of diversity measurements,
which are trying to put a number on different aspects of the questions
associated with diversity <span class="citation">(<span class="citeproc-not-found" data-reference-id="Whittaker1690"><strong>???</strong></span>)</span>.</p>
<p>Many different ways for estimating such diversity measurements have been
described in the literature. Which measurement is best or applicable for your
samples, is not the aim of the following sections.</p>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="microbiome-diversity.html#cb90-1"></a><span class="kw">library</span>(mia)</span>
<span id="cb90-2"><a href="microbiome-diversity.html#cb90-2"></a><span class="kw">data</span>(<span class="st">&quot;GlobalPatterns&quot;</span>)</span>
<span id="cb90-3"><a href="microbiome-diversity.html#cb90-3"></a>se &lt;-<span class="st"> </span>GlobalPatterns</span></code></pre></div>
<div id="alpha-diversity-indices" class="section level2">
<h2><span class="header-section-number">6.1</span> Alpha diversity indices</h2>
<p><strong><em>Alpha diversity</em></strong>, also sometimes interchangeably used with the
term <strong><em>species diversity</em></strong>, summarizes the distribution of species
abundances in a given sample into a single number that depends on
species richness and evenness. Diversity indices measure the overall
community heterogeneity. A number of ecological diversity measures are
available. The Hill coefficient combines many standard indices into a
single equation that provides observed richness, inverse Simpson, and
Shannon diversity, and generalized diversity as special cases. In
general, diversity increases together with increasing richness and
evenness. Sometimes richness, evenness, and dominance are considered
to be variants of alpha diversity.</p>
<p><strong>Richness</strong> refers to the total number of species in a community
(sample). The simplest richness index is the number of observed
species (observed richness). Assuming limited sampling from the
community, however, this may underestimate the true species
richness. Several estimators are available, including for instance ACE
<span class="citation">(<span class="citeproc-not-found" data-reference-id="reference"><strong>???</strong></span>)</span> and Chao1 <span class="citation">(<span class="citeproc-not-found" data-reference-id="reference"><strong>???</strong></span>)</span>. Richness estimates are unaffected
by species abundances.</p>
<p><strong>Evenness</strong> focuses on species abundances, and can thus complement
the number of species. A typical evenness index is the Pielou’s
evenness, which is Shannon diversity normalized by the observed
richness.</p>
<p><strong>Dominance</strong> indices are in general negatively correlated with
diversity, and sometimes used in ecological literature. High
dominance is obtained when one or few species have a high share of
the total species abundance in the community.</p>
<div id="estimating-alpha-diversity" class="section level3">
<h3><span class="header-section-number">6.1.1</span> Estimating alpha diversity</h3>
<p>Alpha diversity can be estimated with wrapper functions that interact
with other packages implementing the calculation, such as <em><code>vegan</code></em>
<span class="citation">(Oksanen et al. <a href="#ref-R-vegan" role="doc-biblioref">2020</a>)</span>.</p>
<p>The main function, <code>estimateAlphaDiversity</code>, calculates the selected
diversity index based on the selected assay data and adds it to the
<code>colData</code> of the <code>SummarizedExperiment</code> under the given column <code>name</code>.</p>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="microbiome-diversity.html#cb91-1"></a>se &lt;-<span class="st"> </span>mia<span class="op">::</span><span class="kw">estimateAlphaDiversity</span>(se, <span class="dt">abund_values =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb91-2"><a href="microbiome-diversity.html#cb91-2"></a>                             <span class="dt">index =</span> <span class="st">&quot;shannon&quot;</span>, <span class="dt">name =</span> <span class="st">&quot;shannon&quot;</span>)</span>
<span id="cb91-3"><a href="microbiome-diversity.html#cb91-3"></a><span class="kw">head</span>(<span class="kw">colData</span>(se)<span class="op">$</span>shannon)</span></code></pre></div>
<pre><code>##     CL3     CC1     SV1 M31Fcsw M11Fcsw M31Plmr 
##   6.577   6.777   6.498   3.828   3.288   4.289</code></pre>
<p>This allows the values to analyzed directly from the <code>colData</code>, for example
by plotting them using <code>plotColData</code> from the <em><code>scater</code></em> package <span class="citation">(McCarthy et al. <a href="#ref-R-scater" role="doc-biblioref">2020</a>)</span>.</p>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="microbiome-diversity.html#cb93-1"></a><span class="kw">library</span>(scater)</span>
<span id="cb93-2"><a href="microbiome-diversity.html#cb93-2"></a><span class="kw">plotColData</span>(se, <span class="st">&quot;shannon&quot;</span>, <span class="st">&quot;SampleType&quot;</span>, <span class="dt">colour_by =</span> <span class="st">&quot;SampleType&quot;</span>) <span class="op">+</span></span>
<span id="cb93-3"><a href="microbiome-diversity.html#cb93-3"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle=</span><span class="dv">45</span>,<span class="dt">hjust=</span><span class="dv">1</span>))</span></code></pre></div>
<div class="figure"><span id="fig:plot-div-shannon"></span>
<img src="14-microbiome-diversity_files/figure-html/plot-div-shannon-1.png" alt="Shannon diversity estimates plotted grouped by sample type." width="672" />
<p class="caption">
Figure 6.1: Shannon diversity estimates plotted grouped by sample type.
</p>
</div>
<p>All available indices will be calculated by default…</p>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="microbiome-diversity.html#cb94-1"></a>se &lt;-<span class="st"> </span><span class="kw">estimateAlphaDiversity</span>(se)</span></code></pre></div>
<p>.. and a plot comparing them can then be constructed directly.</p>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="microbiome-diversity.html#cb95-1"></a>plots &lt;-<span class="st"> </span><span class="kw">lapply</span>(<span class="kw">c</span>(<span class="st">&quot;shannon&quot;</span>,<span class="st">&quot;simpson&quot;</span>,<span class="st">&quot;inv_simpson&quot;</span>, <span class="st">&quot;richness&quot;</span>, <span class="st">&quot;chao1&quot;</span>, <span class="st">&quot;ACE&quot;</span>),</span>
<span id="cb95-2"><a href="microbiome-diversity.html#cb95-2"></a>                plotColData,</span>
<span id="cb95-3"><a href="microbiome-diversity.html#cb95-3"></a>                <span class="dt">object =</span> se,</span>
<span id="cb95-4"><a href="microbiome-diversity.html#cb95-4"></a>                <span class="dt">x =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb95-5"><a href="microbiome-diversity.html#cb95-5"></a>                <span class="dt">colour_by =</span> <span class="st">&quot;SampleType&quot;</span>)</span>
<span id="cb95-6"><a href="microbiome-diversity.html#cb95-6"></a>plots &lt;-<span class="st"> </span><span class="kw">lapply</span>(plots,<span class="st">&quot;+&quot;</span>, <span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle=</span><span class="dv">45</span>,<span class="dt">hjust=</span><span class="dv">1</span>)))</span>
<span id="cb95-7"><a href="microbiome-diversity.html#cb95-7"></a>ggpubr<span class="op">::</span><span class="kw">ggarrange</span>(<span class="dt">plotlist =</span> plots, <span class="dt">nrow =</span> <span class="dv">2</span>, <span class="dt">ncol =</span> <span class="dv">3</span>, <span class="dt">common.legend =</span> <span class="ot">TRUE</span>, <span class="dt">legend =</span> <span class="st">&quot;right&quot;</span>)</span></code></pre></div>
<p><img src="14-microbiome-diversity_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
</div>
</div>
<div id="beta-diversity" class="section level2">
<h2><span class="header-section-number">6.2</span> Beta diversity</h2>
<p>Where alpha diversity focuses on community variation within a
community (sample), beta diversity quantifies (dis-)similarites
between communities (samples).</p>
<p>Technically, beta diversities are usually represented as <code>dist</code>
objects, which contain triangular data describing the distance between
each pair of samples. These distances can be further subjected to
ordination with methods such as multi-dimensional scaling (also known
as PCoA) to retrieve reduced dimensions for further evaluation or
visualization. [TODO we could here instead link to ordination chapter?]</p>
<div id="estimating-beta-diversity" class="section level3">
<h3><span class="header-section-number">6.2.1</span> Estimating beta diversity</h3>
<p>In the following examples distances are calculated by variable
functions supplied to the <code>FUN</code> argument. The function can defined by
the user. It must return a <code>dist</code> function, which can then be used to
calculate reduced dimension either via ordination methods (such as MDS
or NMDS), and the results can be stored in the <code>reducedDim</code>.</p>
<p>This entire process is wrapped in the <code>runMDS</code> and <code>runNMDS</code>
functions.</p>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="microbiome-diversity.html#cb96-1"></a>se &lt;-<span class="st"> </span><span class="kw">runMDS2</span>(se, <span class="dt">FUN =</span> vegan<span class="op">::</span>vegdist, <span class="dt">name =</span> <span class="st">&quot;MDS_BC&quot;</span>)</span></code></pre></div>
<p>Sample similarities can be visualized on a lower-dimensional display
(typically 2D) using the <code>plotReducedDim</code> function in the
<code>scater</code>package. This provides also further tools to incorporate
additional information using variations in colour, shape or size.</p>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="microbiome-diversity.html#cb97-1"></a><span class="kw">plotReducedDim</span>(se, <span class="st">&quot;MDS_BC&quot;</span>, <span class="dt">colour_by =</span> <span class="st">&quot;SampleType&quot;</span>)</span></code></pre></div>
<div class="figure"><span id="fig:plot-mds-bray-curtis"></span>
<img src="14-microbiome-diversity_files/figure-html/plot-mds-bray-curtis-1.png" alt="MDS plot based on the Bray-Curtis distances on the GlobalPattern dataset." width="672" />
<p class="caption">
Figure 6.2: MDS plot based on the Bray-Curtis distances on the GlobalPattern dataset.
</p>
</div>
<p>With additional tools from the <code>ggplot2</code> universe, comparisons can be
performed informing on the applicability to visualize sample similarities in a
meaningful way.</p>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="microbiome-diversity.html#cb98-1"></a>se &lt;-<span class="st"> </span><span class="kw">runMDS2</span>(se, <span class="dt">FUN =</span> vegan<span class="op">::</span>vegdist, <span class="dt">name =</span> <span class="st">&quot;MDS_euclidean&quot;</span>,</span>
<span id="cb98-2"><a href="microbiome-diversity.html#cb98-2"></a>             <span class="dt">method =</span> <span class="st">&quot;euclidean&quot;</span>)</span>
<span id="cb98-3"><a href="microbiome-diversity.html#cb98-3"></a>se &lt;-<span class="st"> </span><span class="kw">runNMDS</span>(se, <span class="dt">FUN =</span> vegan<span class="op">::</span>vegdist, <span class="dt">name =</span> <span class="st">&quot;NMDS_BC&quot;</span>)</span></code></pre></div>
<pre><code>## initial  value 47.733208 
## iter   5 value 33.853364
## iter  10 value 32.891200
## final  value 32.823570 
## converged</code></pre>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="microbiome-diversity.html#cb100-1"></a>se &lt;-<span class="st"> </span><span class="kw">runNMDS</span>(se, <span class="dt">FUN =</span> vegan<span class="op">::</span>vegdist, <span class="dt">name =</span> <span class="st">&quot;NMDS_euclidean&quot;</span>,</span>
<span id="cb100-2"><a href="microbiome-diversity.html#cb100-2"></a>              <span class="dt">method =</span> <span class="st">&quot;euclidean&quot;</span>)</span></code></pre></div>
<pre><code>## initial  value 31.882673 
## final  value 31.882673 
## converged</code></pre>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="microbiome-diversity.html#cb102-1"></a>plots &lt;-<span class="st"> </span><span class="kw">lapply</span>(<span class="kw">c</span>(<span class="st">&quot;MDS_BC&quot;</span>,<span class="st">&quot;MDS_euclidean&quot;</span>,<span class="st">&quot;NMDS_BC&quot;</span>,<span class="st">&quot;NMDS_euclidean&quot;</span>),</span>
<span id="cb102-2"><a href="microbiome-diversity.html#cb102-2"></a>                plotReducedDim, <span class="dt">object =</span> se, <span class="dt">colour_by =</span> <span class="st">&quot;SampleType&quot;</span>)</span>
<span id="cb102-3"><a href="microbiome-diversity.html#cb102-3"></a>ggpubr<span class="op">::</span><span class="kw">ggarrange</span>(<span class="dt">plotlist =</span> plots, <span class="dt">nrow =</span> <span class="dv">2</span>, <span class="dt">ncol =</span> <span class="dv">2</span>, <span class="dt">common.legend =</span> <span class="ot">TRUE</span>,</span>
<span id="cb102-4"><a href="microbiome-diversity.html#cb102-4"></a>                  <span class="dt">legend =</span> <span class="st">&quot;right&quot;</span>)</span></code></pre></div>
<div class="figure"><span id="fig:plot-mds-nmds-comparison"></span>
<img src="14-microbiome-diversity_files/figure-html/plot-mds-nmds-comparison-1.png" alt="Comparison of MDS and NMDS plots based on the Bray-Curtis or euclidean distances on the GlobalPattern dataset." width="672" />
<p class="caption">
Figure 6.3: Comparison of MDS and NMDS plots based on the Bray-Curtis or euclidean distances on the GlobalPattern dataset.
</p>
</div>
<p>The <em>UniFrac</em> method is a special case, as it requires data on the
relationship of features in form on a <code>phylo</code> tree. <code>calculateUniFrac</code>
performs the calculation to return a <code>dist</code> object, which can again be
used within <code>runMDS</code>.</p>
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb103-1"><a href="microbiome-diversity.html#cb103-1"></a>se &lt;-<span class="st"> </span><span class="kw">runMDS2</span>(se, <span class="dt">FUN =</span> calculateUniFrac, <span class="dt">name =</span> <span class="st">&quot;UniFrac&quot;</span>,</span>
<span id="cb103-2"><a href="microbiome-diversity.html#cb103-2"></a>              <span class="dt">tree =</span> <span class="kw">rowTree</span>(se),</span>
<span id="cb103-3"><a href="microbiome-diversity.html#cb103-3"></a>              <span class="dt">ntop =</span> <span class="kw">nrow</span>(se))</span></code></pre></div>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="microbiome-diversity.html#cb104-1"></a><span class="kw">plotReducedDim</span>(se, <span class="st">&quot;UniFrac&quot;</span>, <span class="dt">colour_by =</span> <span class="st">&quot;SampleType&quot;</span>)</span></code></pre></div>
<div class="figure"><span id="fig:plot-unifrac"></span>
<img src="14-microbiome-diversity_files/figure-html/plot-unifrac-1.png" alt="UniFrac distances scaled by MDS of the GlobalPattern dataset." width="672" />
<p class="caption">
Figure 6.4: UniFrac distances scaled by MDS of the GlobalPattern dataset.
</p>
</div>
</div>
<div id="other-ordination-methods" class="section level3">
<h3><span class="header-section-number">6.2.2</span> Other ordination methods</h3>
<p>Other dimension reduction methods, such as <code>PCA</code>, <code>t-SNE</code> and <code>UMAP</code> are
inherited directly from the <code>scater</code> package.</p>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="microbiome-diversity.html#cb105-1"></a>se &lt;-<span class="st"> </span><span class="kw">runPCA</span>(se, <span class="dt">name =</span> <span class="st">&quot;PCA&quot;</span>, <span class="dt">exprs_values =</span> <span class="st">&quot;counts&quot;</span>, <span class="dt">ncomponents =</span> <span class="dv">10</span>)</span></code></pre></div>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="microbiome-diversity.html#cb106-1"></a><span class="kw">plotReducedDim</span>(se, <span class="st">&quot;PCA&quot;</span>, <span class="dt">colour_by =</span> <span class="st">&quot;SampleType&quot;</span>)</span></code></pre></div>
<div class="figure"><span id="fig:plot-pca"></span>
<img src="14-microbiome-diversity_files/figure-html/plot-pca-1.png" alt="PCA plot on the GlobalPatterns data set containing sample from different sources." width="672" />
<p class="caption">
Figure 6.5: PCA plot on the GlobalPatterns data set containing sample from different sources.
</p>
</div>
<p>As mentioned before, applicability of the different methods depends on your
sample set.</p>
<p>FIXME: let us switch to UMAP for the examples?</p>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="microbiome-diversity.html#cb107-1"></a>se &lt;-<span class="st"> </span><span class="kw">runTSNE</span>(se, <span class="dt">name =</span> <span class="st">&quot;TSNE&quot;</span>, <span class="dt">exprs_values =</span> <span class="st">&quot;counts&quot;</span>, <span class="dt">ncomponents =</span> <span class="dv">3</span>)</span></code></pre></div>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="microbiome-diversity.html#cb108-1"></a><span class="kw">plotReducedDim</span>(se, <span class="st">&quot;TSNE&quot;</span>, <span class="dt">colour_by =</span> <span class="st">&quot;SampleType&quot;</span>, <span class="dt">ncomponents =</span> <span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>))</span></code></pre></div>
<div class="figure"><span id="fig:plot-tsne"></span>
<img src="14-microbiome-diversity_files/figure-html/plot-tsne-1.png" alt="t-SNE plot on the GlobalPatterns data set containing sample from different sources." width="672" />
<p class="caption">
Figure 6.6: t-SNE plot on the GlobalPatterns data set containing sample from different sources.
</p>
</div>
</div>
</div>
<div id="community-comparisons-todo-combine-with-the-material-above-for-simplicity" class="section level2">
<h2><span class="header-section-number">6.3</span> Community comparisons [TODO combine with the material above for simplicity?]</h2>
<p>A typical comparison of community composition starts with a visual
comparison of the groups on a 2D ordination.</p>
<p>Let us load an example data set:</p>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="microbiome-diversity.html#cb109-1"></a><span class="kw">library</span>(microbiomeDataSets)</span>
<span id="cb109-2"><a href="microbiome-diversity.html#cb109-2"></a>se.lahti &lt;-<span class="st"> </span><span class="kw">LahtiMData</span>()</span></code></pre></div>
<p>Then we estimate relative abundances and MDS ordination based on
Bray-Curtis (BC) dissimilarity between the groups, and visualize the
results.</p>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="microbiome-diversity.html#cb110-1"></a>se.lahti &lt;-<span class="st"> </span><span class="kw">relAbundanceCounts</span>(se.lahti)</span>
<span id="cb110-2"><a href="microbiome-diversity.html#cb110-2"></a>se.lahti &lt;-<span class="st"> </span><span class="kw">runNMDS</span>(se.lahti, <span class="dt">FUN =</span> vegan<span class="op">::</span>vegdist, <span class="dt">name =</span> <span class="st">&quot;BC&quot;</span>, <span class="dt">nmdsFUN =</span> <span class="st">&quot;monoMDS&quot;</span>,</span>
<span id="cb110-3"><a href="microbiome-diversity.html#cb110-3"></a>                    <span class="dt">exprs_values =</span> <span class="st">&quot;relabundance&quot;</span>,</span>
<span id="cb110-4"><a href="microbiome-diversity.html#cb110-4"></a>                    <span class="dt">keep_dist =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<div class="sourceCode" id="cb111"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb111-1"><a href="microbiome-diversity.html#cb111-1"></a><span class="kw">plotReducedDim</span>(se.lahti, <span class="st">&quot;BC&quot;</span>, <span class="dt">colour_by =</span> <span class="st">&quot;group&quot;</span>)</span></code></pre></div>
<p><img src="14-microbiome-diversity_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
<p>No clear difference between the groups can be visually observed.</p>
<div id="testing-differences-in-community-composition-between-sample-groups" class="section level3">
<h3><span class="header-section-number">6.3.1</span> Testing differences in community composition between sample groups</h3>
<p>The permutational analysis of variance (PERMANOVA) <span class="citation">(Anderson <a href="#ref-Anderson2001" role="doc-biblioref">2001</a>)</span> is
a widely used non-parametric multivariate method that can be used to
estimate the actual statistical significance of differences in the
observed community composition between two groups of
samples.</p>
<p>PERMANOVA evaluates the hypothesis that the centroids and dispersion
of the community are equivalent between the compared groups. A small
p-value indicates that the compared groups have, on average, a
different community composition.</p>
<p>This method is implemented in the <code>vegan</code> package (function <code>adonis</code>).</p>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="microbiome-diversity.html#cb112-1"></a><span class="kw">library</span>(vegan)</span>
<span id="cb112-2"><a href="microbiome-diversity.html#cb112-2"></a>permanova &lt;-<span class="st"> </span>vegan<span class="op">::</span><span class="kw">adonis</span>(<span class="kw">t</span>(<span class="kw">assay</span>(se.lahti,<span class="st">&quot;relabundance&quot;</span>)) <span class="op">~</span><span class="st"> </span>group,</span>
<span id="cb112-3"><a href="microbiome-diversity.html#cb112-3"></a>                           <span class="dt">data =</span> <span class="kw">colData</span>(se.lahti),</span>
<span id="cb112-4"><a href="microbiome-diversity.html#cb112-4"></a>                           <span class="dt">permutations =</span> <span class="dv">9999</span>)</span>
<span id="cb112-5"><a href="microbiome-diversity.html#cb112-5"></a></span>
<span id="cb112-6"><a href="microbiome-diversity.html#cb112-6"></a><span class="co"># P-value</span></span>
<span id="cb112-7"><a href="microbiome-diversity.html#cb112-7"></a><span class="kw">print</span>(<span class="kw">as.data.frame</span>(permanova<span class="op">$</span>aov.tab)[<span class="st">&quot;group&quot;</span>, <span class="st">&quot;Pr(&gt;F)&quot;</span>])</span></code></pre></div>
<pre><code>## [1] 0.2741</code></pre>
<p>In this case, the community composition is not significantly different
between the groups.</p>
<p>Let us visualize the model coefficients for species that exhibit the
largest differences between the groups. This gives some insights into
how the groups tend to differ from each other in terms of community
composition.</p>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="microbiome-diversity.html#cb114-1"></a>coef &lt;-<span class="st"> </span><span class="kw">coefficients</span>(permanova)[<span class="st">&quot;group1&quot;</span>,]</span>
<span id="cb114-2"><a href="microbiome-diversity.html#cb114-2"></a>top.coef &lt;-<span class="st"> </span><span class="kw">sort</span>(<span class="kw">head</span>(coef[<span class="kw">rev</span>(<span class="kw">order</span>(<span class="kw">abs</span>(coef)))],<span class="dv">20</span>))</span></code></pre></div>
<div class="sourceCode" id="cb115"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb115-1"><a href="microbiome-diversity.html#cb115-1"></a><span class="kw">ggplot</span>(<span class="kw">data.frame</span>(<span class="dt">x =</span> top.coef,</span>
<span id="cb115-2"><a href="microbiome-diversity.html#cb115-2"></a>                  <span class="dt">y =</span> <span class="kw">factor</span>(<span class="kw">names</span>(top.coef),</span>
<span id="cb115-3"><a href="microbiome-diversity.html#cb115-3"></a>                                      <span class="kw">unique</span>(<span class="kw">names</span>(top.coef)))),</span>
<span id="cb115-4"><a href="microbiome-diversity.html#cb115-4"></a>        <span class="kw">aes</span>(<span class="dt">x =</span> x, <span class="dt">y =</span> y)) <span class="op">+</span></span>
<span id="cb115-5"><a href="microbiome-diversity.html#cb115-5"></a><span class="st">    </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>) <span class="op">+</span></span>
<span id="cb115-6"><a href="microbiome-diversity.html#cb115-6"></a><span class="st">    </span><span class="kw">labs</span>(<span class="dt">x=</span><span class="st">&quot;&quot;</span>,<span class="dt">y=</span><span class="st">&quot;&quot;</span>,<span class="dt">title=</span><span class="st">&quot;Top Taxa&quot;</span>) <span class="op">+</span></span>
<span id="cb115-7"><a href="microbiome-diversity.html#cb115-7"></a><span class="st">    </span><span class="kw">theme_bw</span>()</span></code></pre></div>
<p><img src="14-microbiome-diversity_files/figure-html/plot-top-coef-anova-1.png" width="672" /></p>
<p>In the above example, the largest differences between the two groups
can be attributed to Bacteroides intestinalis (elevated in the first
group) and Faecalibacterium prausnitzii (elevated in the second
group), and many other co-varying species.</p>
</div>
<div id="checking-the-homogeneity-condition" class="section level3">
<h3><span class="header-section-number">6.3.2</span> Checking the homogeneity condition</h3>
<p>It is important to note that the application of PERMANOVA assumes
homogeneous group dispersions (variances). This can be tested with the
PERMDISP2 method <span class="citation">(Anderson <a href="#ref-Anderson2006" role="doc-biblioref">2006</a>)</span>.</p>
<div class="sourceCode" id="cb116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb116-1"><a href="microbiome-diversity.html#cb116-1"></a><span class="kw">anova</span>(vegan<span class="op">::</span><span class="kw">betadisper</span>(<span class="kw">attr</span>(<span class="kw">reducedDim</span>(se.lahti,<span class="st">&quot;BC&quot;</span>),<span class="st">&quot;dist&quot;</span>),</span>
<span id="cb116-2"><a href="microbiome-diversity.html#cb116-2"></a>                        <span class="kw">colData</span>(se.lahti)<span class="op">$</span>group))</span></code></pre></div>
<pre><code>## Analysis of Variance Table
## 
## Response: Distances
##           Df Sum Sq Mean Sq F value Pr(&gt;F)
## Groups     1  0.000 0.00002       0   0.95
## Residuals 42  0.158 0.00376</code></pre>
<p>In our example, the groups have similar dispersions, and PERMANOVA is
an appropriate choice for comparing community compositions.</p>
</div>
</div>
<div id="further-reading" class="section level2">
<h2><span class="header-section-number">6.4</span> Further reading</h2>
<p>In certain settings beta diversities might be used to group samples without
prior knowledge. For this we want to point to excellent resources on
<a href="http://bioconductor.org/books/release/OSCA/clustering.html">how to extract information from the clusters</a>.</p>
<p>See also <a href="15-microbiome-community.md">community typing</a>.</p>
</div>
<div id="session-info-5" class="section level2 unnumbered">
<h2>Session Info</h2>
<button class="rebook-collapse">
View session info
</button>
<div class="rebook-content">
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.1 LTS

Matrix products: default
BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] vegan_2.5-7                      lattice_0.20-41                 
 [3] permute_0.9-5                    microbiomeDataSets_0.98.4       
 [5] MultiAssayExperiment_1.16.0      scater_1.18.3                   
 [7] ggplot2_3.3.3                    mia_0.98.21                     
 [9] MicrobiomeExperiment_0.99.0.9014 Biostrings_2.58.0               
[11] XVector_0.30.0                   TreeSummarizedExperiment_1.6.2  
[13] SingleCellExperiment_1.12.0      SummarizedExperiment_1.20.0     
[15] Biobase_2.50.0                   GenomicRanges_1.42.0            
[17] GenomeInfoDb_1.26.2              IRanges_2.24.1                  
[19] S4Vectors_0.28.1                 BiocGenerics_0.36.0             
[21] MatrixGenerics_1.2.0             matrixStats_0.57.0              
[23] BiocStyle_2.18.1                 rebook_1.0.0                    
[25] BiocManager_1.30.10             

loaded via a namespace (and not attached):
  [1] readxl_1.3.1                  backports_1.2.1              
  [3] AnnotationHub_2.22.0          BiocFileCache_1.14.0         
  [5] splines_4.0.3                 BiocParallel_1.24.1          
  [7] digest_0.6.27                 htmltools_0.5.0              
  [9] viridis_0.5.1                 magrittr_2.0.1               
 [11] memoise_1.1.0                 cluster_2.1.0                
 [13] openxlsx_4.2.3                colorspace_2.0-0             
 [15] blob_1.2.1                    rappdirs_0.3.1               
 [17] haven_2.3.1                   xfun_0.19                    
 [19] dplyr_1.0.2                   callr_3.5.1                  
 [21] crayon_1.3.4                  RCurl_1.98-1.2               
 [23] graph_1.68.0                  ape_5.4-1                    
 [25] glue_1.4.2                    gtable_0.3.0                 
 [27] zlibbioc_1.36.0               DelayedArray_0.16.0          
 [29] car_3.0-10                    BiocSingular_1.6.0           
 [31] abind_1.4-5                   scales_1.1.1                 
 [33] DBI_1.1.0                     rstatix_0.6.0                
 [35] Rcpp_1.0.5                    xtable_1.8-4                 
 [37] viridisLite_0.3.0             foreign_0.8-81               
 [39] bit_4.0.4                     rsvd_1.0.3                   
 [41] httr_1.4.2                    ellipsis_0.3.1               
 [43] pkgconfig_2.0.3               XML_3.99-0.5                 
 [45] farver_2.0.3                  scuttle_1.0.4                
 [47] CodeDepends_0.6.5             dbplyr_2.0.0                 
 [49] AnnotationDbi_1.52.0          tidyselect_1.1.0             
 [51] labeling_0.4.2                rlang_0.4.10                 
 [53] later_1.1.0.1                 munsell_0.5.0                
 [55] BiocVersion_3.12.0            cellranger_1.1.0             
 [57] tools_4.0.3                   DirichletMultinomial_1.32.0  
 [59] generics_0.1.0                RSQLite_2.2.1                
 [61] ExperimentHub_1.16.0          broom_0.7.3                  
 [63] fastmap_1.0.1                 evaluate_0.14                
 [65] stringr_1.4.0                 yaml_2.2.1                   
 [67] processx_3.4.5                knitr_1.30                   
 [69] bit64_4.0.5                   zip_2.1.1                    
 [71] purrr_0.3.4                   nlme_3.1-151                 
 [73] sparseMatrixStats_1.2.0       mime_0.9                     
 [75] compiler_4.0.3                beeswarm_0.2.3               
 [77] curl_4.3                      interactiveDisplayBase_1.28.0
 [79] ggsignif_0.6.0                tibble_3.0.4                 
 [81] stringi_1.5.3                 highr_0.8                    
 [83] ps_1.5.0                      forcats_0.5.0                
 [85] Matrix_1.3-0                  vctrs_0.3.6                  
 [87] pillar_1.4.7                  lifecycle_0.2.0              
 [89] BiocNeighbors_1.8.2           data.table_1.13.6            
 [91] cowplot_1.1.1                 bitops_1.0-6                 
 [93] irlba_2.3.3                   httpuv_1.5.4                 
 [95] R6_2.5.0                      bookdown_0.21                
 [97] promises_1.1.1                gridExtra_2.3                
 [99] rio_0.5.16                    vipor_0.4.5                  
[101] codetools_0.2-18              MASS_7.3-53                  
[103] assertthat_0.2.1              withr_2.3.0                  
[105] GenomeInfoDbData_1.2.4        mgcv_1.8-33                  
[107] hms_0.5.3                     grid_4.0.3                   
[109] beachmat_2.6.4                tidyr_1.1.2                  
[111] rmarkdown_2.6                 DelayedMatrixStats_1.12.1    
[113] carData_3.0-4                 Rtsne_0.15                   
[115] ggpubr_0.4.0                  shiny_1.5.0                  
[117] ggbeeswarm_0.6.0             </code></pre>
</div>

</div>
</div>
<h3>Bibliography</h3>
<div id="refs" class="references">
<div id="ref-Anderson2001">
<p>Anderson, Marti J. 2001. “A New Method for Non-Parametric Multivariate Analysis of Variance.” <em>Austral Ecology</em> 26 (1): 32–46. <a href="https://doi.org/10.1111/j.1442-9993.2001.01070.pp.x">https://doi.org/10.1111/j.1442-9993.2001.01070.pp.x</a>.</p>
</div>
<div id="ref-Anderson2006">
<p>Anderson, Marti J. 2006. “Distance-Based Tests for Homogeneity of Multivariate Dispersions.” <em>Biometrics</em> 62: 245–53. <a href="https://doi.org/10.1111/j.1541-0420.2005.00440.x">https://doi.org/10.1111/j.1541-0420.2005.00440.x</a>.</p>
</div>
<div id="ref-R-scater">
<p>McCarthy, Davis, Kieran Campbell, Aaron Lun, and Quin Wills. 2020. <em>Scater: Single-Cell Analysis Toolkit for Gene Expression Data in R</em>. <a href="http://bioconductor.org/packages/scater/">http://bioconductor.org/packages/scater/</a>.</p>
</div>
<div id="ref-R-vegan">
<p>Oksanen, Jari, F. Guillaume Blanchet, Michael Friendly, Roeland Kindt, Pierre Legendre, Dan McGlinn, Peter R. Minchin, et al. 2020. <em>Vegan: Community Ecology Package</em>. <a href="https://CRAN.R-project.org/package=vegan">https://CRAN.R-project.org/package=vegan</a>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="microbiome-exploration.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="microbiome-community.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["MiaBook.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
