<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 Microbiome Community | Orchestrating Microbiome Analysis</title>
  <meta name="description" content="Chapter 9 Microbiome Community | Orchestrating Microbiome Analysis" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 Microbiome Community | Orchestrating Microbiome Analysis" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://github.com/microbiome/OMA" />
  
  
  <meta name="github-repo" content="microbiome/OMA" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 Microbiome Community | Orchestrating Microbiome Analysis" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="beta-diversity.html"/>
<link rel="next" href="community-typing.html"/>
<script src="libs/header-attrs-2.10/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<script src="libs/plotly-binding-4.9.4.1/plotly.js"></script>
<script src="libs/typedarray-0.1/typedarray.min.js"></script>
<link href="libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
<link href="libs/plotly-htmlwidgets-css-1.57.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="libs/plotly-main-1.57.1/plotly-latest.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Orchestrating Microbiome Analysis</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="chapter" data-level="" data-path="introduction-to-bioconductor-classes.html"><a href="introduction-to-bioconductor-classes.html"><i class="fa fa-check"></i>Introduction to Bioconductor Classes</a></li>
<li class="part"><span><b>I Introduction</b></span></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#community"><i class="fa fa-check"></i><b>1.1</b> Community</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#developers-and-contributors"><i class="fa fa-check"></i><b>1.2</b> Developers and contributors</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#acknowledgements"><i class="fa fa-check"></i><b>1.3</b> Acknowledgements</a></li>
<li class="chapter" data-level="1.4" data-path="introduction.html"><a href="introduction.html#package-ecosystem"><i class="fa fa-check"></i><b>1.4</b> Package ecosystem</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data-introduction.html"><a href="data-introduction.html"><i class="fa fa-check"></i><b>2</b> Data Infrastructure</a>
<ul>
<li class="chapter" data-level="2.1" data-path="data-introduction.html"><a href="data-introduction.html#installation"><i class="fa fa-check"></i><b>2.1</b> Installation</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="data-introduction.html"><a href="data-introduction.html#packages"><i class="fa fa-check"></i><b>2.1.1</b> Packages</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="data-introduction.html"><a href="data-introduction.html#background"><i class="fa fa-check"></i><b>2.2</b> Background</a></li>
<li class="chapter" data-level="2.3" data-path="data-introduction.html"><a href="data-introduction.html#loading-experimental-microbiome-data"><i class="fa fa-check"></i><b>2.3</b> Loading experimental microbiome data</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="data-introduction.html"><a href="data-introduction.html#importing-data-from-external-files"><i class="fa fa-check"></i><b>2.3.1</b> Importing data from external files</a></li>
<li class="chapter" data-level="2.3.2" data-path="data-introduction.html"><a href="data-introduction.html#conversions-between-data-formats-in-r"><i class="fa fa-check"></i><b>2.3.2</b> Conversions between data formats in R</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="data-introduction.html"><a href="data-introduction.html#metadata"><i class="fa fa-check"></i><b>2.4</b> Metadata</a></li>
<li class="chapter" data-level="2.5" data-path="data-introduction.html"><a href="data-introduction.html#microbiome-and-tree-data-specific-aspects"><i class="fa fa-check"></i><b>2.5</b> Microbiome and tree data specific aspects</a>
<ul>
<li class="chapter" data-level="2.5.1" data-path="data-introduction.html"><a href="data-introduction.html#assays"><i class="fa fa-check"></i><b>2.5.1</b> Assays</a></li>
<li class="chapter" data-level="2.5.2" data-path="data-introduction.html"><a href="data-introduction.html#coldata"><i class="fa fa-check"></i><b>2.5.2</b> colData</a></li>
<li class="chapter" data-level="2.5.3" data-path="data-introduction.html"><a href="data-introduction.html#rowdata"><i class="fa fa-check"></i><b>2.5.3</b> rowData</a></li>
<li class="chapter" data-level="2.5.4" data-path="data-introduction.html"><a href="data-introduction.html#rowtree"><i class="fa fa-check"></i><b>2.5.4</b> rowTree</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="data-introduction.html"><a href="data-introduction.html#data-manipulation"><i class="fa fa-check"></i><b>2.6</b> Data manipulation</a>
<ul>
<li class="chapter" data-level="2.6.1" data-path="data-introduction.html"><a href="data-introduction.html#tidy-data"><i class="fa fa-check"></i><b>2.6.1</b> Tidy data</a></li>
<li class="chapter" data-level="2.6.2" data-path="data-introduction.html"><a href="data-introduction.html#subsetting"><i class="fa fa-check"></i><b>2.6.2</b> Subsetting</a></li>
</ul></li>
<li class="chapter" data-level="2.7" data-path="data-introduction.html"><a href="data-introduction.html#conclusion"><i class="fa fa-check"></i><b>2.7</b> Conclusion</a></li>
<li class="chapter" data-level="2.8" data-path="data-introduction.html"><a href="data-introduction.html#additional-reading"><i class="fa fa-check"></i><b>2.8</b> Additional Reading</a>
<ul>
<li class="chapter" data-level="2.8.1" data-path="data-introduction.html"><a href="data-introduction.html#lecture-slides"><i class="fa fa-check"></i><b>2.8.1</b> Lecture slides</a></li>
<li class="chapter" data-level="2.8.2" data-path="data-introduction.html"><a href="data-introduction.html#r-programming-resources"><i class="fa fa-check"></i><b>2.8.2</b> R programming resources</a></li>
<li class="chapter" data-level="2.8.3" data-path="data-introduction.html"><a href="data-introduction.html#resources-for-treesummarizedexperiment"><i class="fa fa-check"></i><b>2.8.3</b> Resources for TreeSummarizedExperiment</a></li>
<li class="chapter" data-level="2.8.4" data-path="data-introduction.html"><a href="data-introduction.html#resources-for-phyloseq"><i class="fa fa-check"></i><b>2.8.4</b> Resources for phyloseq</a></li>
<li class="chapter" data-level="2.8.5" data-path="data-introduction.html"><a href="data-introduction.html#further-reading"><i class="fa fa-check"></i><b>2.8.5</b> Further reading</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-introduction.html"><a href="data-introduction.html#session-info"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="part"><span><b>II Focus Topics</b></span></li>
<li class="chapter" data-level="3" data-path="manipulation.html"><a href="manipulation.html"><i class="fa fa-check"></i><b>3</b> Data manipulation</a></li>
<li class="chapter" data-level="4" data-path="taxonomic-information.html"><a href="taxonomic-information.html"><i class="fa fa-check"></i><b>4</b> Taxonomic Information</a>
<ul>
<li class="chapter" data-level="4.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#assigning-taxonomic-information."><i class="fa fa-check"></i><b>4.1</b> Assigning taxonomic information.</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#dada2"><i class="fa fa-check"></i><b>4.1.1</b> dada2</a></li>
<li class="chapter" data-level="4.1.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#decipher"><i class="fa fa-check"></i><b>4.1.2</b> DECIPHER</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#functions-to-access-taxonomic-information"><i class="fa fa-check"></i><b>4.2</b> Functions to access taxonomic information</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#generate-a-taxonomic-tree-on-the-fly"><i class="fa fa-check"></i><b>4.2.1</b> Generate a taxonomic tree on the fly</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="taxonomic-information.html"><a href="taxonomic-information.html#data-agglomeration"><i class="fa fa-check"></i><b>4.3</b> Data agglomeration</a></li>
<li class="chapter" data-level="4.4" data-path="taxonomic-information.html"><a href="taxonomic-information.html#data-transformation"><i class="fa fa-check"></i><b>4.4</b> Data transformation</a></li>
<li class="chapter" data-level="4.5" data-path="taxonomic-information.html"><a href="taxonomic-information.html#pick-specific"><i class="fa fa-check"></i><b>4.5</b> Pick specific</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#abundances-of-all-taxa-in-specific-sample"><i class="fa fa-check"></i><b>4.5.1</b> Abundances of all taxa in specific sample</a></li>
<li class="chapter" data-level="4.5.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#abundances-of-specific-taxa-in-all-samples"><i class="fa fa-check"></i><b>4.5.2</b> Abundances of specific taxa in all samples</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="taxonomic-information.html"><a href="taxonomic-information.html#session-info-1"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>5</b> Quality Control</a>
<ul>
<li class="chapter" data-level="5.1" data-path="quality-control.html"><a href="quality-control.html#get-top-taxa-and-taxonomy"><i class="fa fa-check"></i><b>5.1</b> Get top taxa and taxonomy</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="quality-control.html"><a href="quality-control.html#features"><i class="fa fa-check"></i><b>5.1.1</b> Features</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="quality-control.html"><a href="quality-control.html#library-size"><i class="fa fa-check"></i><b>5.2</b> Library size</a></li>
<li class="chapter" data-level="" data-path="quality-control.html"><a href="quality-control.html#session-info-2"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html"><i class="fa fa-check"></i><b>6</b> Microbiome Exploration</a>
<ul>
<li class="chapter" data-level="6.1" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#prevalence"><i class="fa fa-check"></i><b>6.1</b> Prevalence</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#prevalent-microbiota-analysis"><i class="fa fa-check"></i><b>6.1.1</b> Prevalent microbiota analysis</a></li>
<li class="chapter" data-level="6.1.2" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#plotting-prevalence"><i class="fa fa-check"></i><b>6.1.2</b> Plotting prevalence</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#abundance"><i class="fa fa-check"></i><b>6.2</b> Abundance</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="alpha-diversity.html"><a href="alpha-diversity.html"><i class="fa fa-check"></i><b>7</b> Alpha Diversity</a>
<ul>
<li class="chapter" data-level="7.1" data-path="alpha-diversity.html"><a href="alpha-diversity.html#estimating-alpha-diversity"><i class="fa fa-check"></i><b>7.1</b> Estimating alpha diversity</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="alpha-diversity.html"><a href="alpha-diversity.html#richness"><i class="fa fa-check"></i><b>7.1.1</b> Richness</a></li>
<li class="chapter" data-level="7.1.2" data-path="alpha-diversity.html"><a href="alpha-diversity.html#diversity"><i class="fa fa-check"></i><b>7.1.2</b> Diversity</a></li>
<li class="chapter" data-level="7.1.3" data-path="alpha-diversity.html"><a href="alpha-diversity.html#evenness"><i class="fa fa-check"></i><b>7.1.3</b> Evenness</a></li>
<li class="chapter" data-level="7.1.4" data-path="alpha-diversity.html"><a href="alpha-diversity.html#dominance"><i class="fa fa-check"></i><b>7.1.4</b> Dominance</a></li>
<li class="chapter" data-level="7.1.5" data-path="alpha-diversity.html"><a href="alpha-diversity.html#rarity"><i class="fa fa-check"></i><b>7.1.5</b> Rarity</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="alpha-diversity.html"><a href="alpha-diversity.html#visualize-alpha-diversities"><i class="fa fa-check"></i><b>7.2</b> Visualize alpha diversities</a></li>
<li class="chapter" data-level="" data-path="alpha-diversity.html"><a href="alpha-diversity.html#session-info-3"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="beta-diversity.html"><a href="beta-diversity.html"><i class="fa fa-check"></i><b>8</b> Beta Diversity</a>
<ul>
<li class="chapter" data-level="8.1" data-path="beta-diversity.html"><a href="beta-diversity.html#explained-variance"><i class="fa fa-check"></i><b>8.1</b> Explained variance</a></li>
<li class="chapter" data-level="8.2" data-path="beta-diversity.html"><a href="beta-diversity.html#estimating-beta-diversity"><i class="fa fa-check"></i><b>8.2</b> Estimating beta diversity</a></li>
<li class="chapter" data-level="8.3" data-path="beta-diversity.html"><a href="beta-diversity.html#other-ordination-methods"><i class="fa fa-check"></i><b>8.3</b> Other ordination methods</a></li>
<li class="chapter" data-level="8.4" data-path="beta-diversity.html"><a href="beta-diversity.html#visualizing-the-most-dominant-genus-on-pcoa"><i class="fa fa-check"></i><b>8.4</b> Visualizing the most dominant genus on PCoA</a></li>
<li class="chapter" data-level="8.5" data-path="beta-diversity.html"><a href="beta-diversity.html#community-comparisons-todo-combine-with-the-material-above-for-simplicity"><i class="fa fa-check"></i><b>8.5</b> Community comparisons [TODO combine with the material above for simplicity?]</a>
<ul>
<li class="chapter" data-level="8.5.1" data-path="beta-diversity.html"><a href="beta-diversity.html#testing-differences-in-community-composition-between-sample-groups"><i class="fa fa-check"></i><b>8.5.1</b> Testing differences in community composition between sample groups</a></li>
<li class="chapter" data-level="8.5.2" data-path="beta-diversity.html"><a href="beta-diversity.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>8.5.2</b> Checking the homogeneity condition</a></li>
</ul></li>
<li class="chapter" data-level="8.6" data-path="beta-diversity.html"><a href="beta-diversity.html#further-reading-1"><i class="fa fa-check"></i><b>8.6</b> Further reading</a></li>
<li class="chapter" data-level="" data-path="beta-diversity.html"><a href="beta-diversity.html#session-info-4"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="microbiome-community.html"><a href="microbiome-community.html"><i class="fa fa-check"></i><b>9</b> Microbiome Community</a>
<ul>
<li class="chapter" data-level="9.1" data-path="microbiome-community.html"><a href="microbiome-community.html#community-composition"><i class="fa fa-check"></i><b>9.1</b> Community composition</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="microbiome-community.html"><a href="microbiome-community.html#composition-barplot"><i class="fa fa-check"></i><b>9.1.1</b> Composition barplot</a></li>
<li class="chapter" data-level="9.1.2" data-path="microbiome-community.html"><a href="microbiome-community.html#composition-heatmap"><i class="fa fa-check"></i><b>9.1.2</b> Composition heatmap</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="microbiome-community.html"><a href="microbiome-community.html#session-info-5"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="community-typing.html"><a href="community-typing.html"><i class="fa fa-check"></i><b>10</b> Community typing</a>
<ul>
<li class="chapter" data-level="10.1" data-path="community-typing.html"><a href="community-typing.html#community-typing-1"><i class="fa fa-check"></i><b>10.1</b> Community typing</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="community-typing.html"><a href="community-typing.html#dirichlet-multinomial-mixtures-dmm"><i class="fa fa-check"></i><b>10.1.1</b> Dirichlet Multinomial Mixtures (DMM)</a></li>
<li class="chapter" data-level="10.1.2" data-path="community-typing.html"><a href="community-typing.html#pcoa-for-asv-level-data-with-bray-curtis-with-dmm-clusters-shown-with-colors"><i class="fa fa-check"></i><b>10.1.2</b> PCoA for ASV-level data with Bray-Curtis; with DMM clusters shown with colors</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="community-typing.html"><a href="community-typing.html#community-detection"><i class="fa fa-check"></i><b>10.2</b> Community Detection</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="differential-abundance.html"><a href="differential-abundance.html"><i class="fa fa-check"></i><b>11</b> Differential Abundance</a>
<ul>
<li class="chapter" data-level="11.1" data-path="differential-abundance.html"><a href="differential-abundance.html#introduction-to-differential-abundance-analysis"><i class="fa fa-check"></i><b>11.1</b> Introduction to differential abundance analysis</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="differential-abundance.html"><a href="differential-abundance.html#prevalence-filtering"><i class="fa fa-check"></i><b>11.1.1</b> Prevalence Filtering</a></li>
<li class="chapter" data-level="11.1.2" data-path="differential-abundance.html"><a href="differential-abundance.html#aldex2"><i class="fa fa-check"></i><b>11.1.2</b> ALDEx2</a></li>
<li class="chapter" data-level="11.1.3" data-path="differential-abundance.html"><a href="differential-abundance.html#ancom-bc"><i class="fa fa-check"></i><b>11.1.3</b> ANCOM-BC</a></li>
<li class="chapter" data-level="11.1.4" data-path="differential-abundance.html"><a href="differential-abundance.html#maaslin2"><i class="fa fa-check"></i><b>11.1.4</b> MaAsLin2</a></li>
<li class="chapter" data-level="11.1.5" data-path="differential-abundance.html"><a href="differential-abundance.html#comparison-of-the-methods"><i class="fa fa-check"></i><b>11.1.5</b> Comparison of the methods</a></li>
<li class="chapter" data-level="11.1.6" data-path="differential-abundance.html"><a href="differential-abundance.html#confounding-variables"><i class="fa fa-check"></i><b>11.1.6</b> Confounding variables</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="differential-abundance.html"><a href="differential-abundance.html#tree-based-methods"><i class="fa fa-check"></i><b>11.2</b> Tree-based methods</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="differential-abundance.html"><a href="differential-abundance.html#group-wise-associations-testing-based-on-balances-with-fido"><i class="fa fa-check"></i><b>11.2.1</b> Group-wise associations testing based on balances with fido</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="differential-abundance.html"><a href="differential-abundance.html#session-info-6"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="multitable.html"><a href="multitable.html"><i class="fa fa-check"></i><b>12</b> Representation of Multiple Data Tables</a>
<ul>
<li class="chapter" data-level="12.1" data-path="multitable.html"><a href="multitable.html#assay-data"><i class="fa fa-check"></i><b>12.1</b> Assay data</a></li>
<li class="chapter" data-level="12.2" data-path="multitable.html"><a href="multitable.html#alternative-experiments"><i class="fa fa-check"></i><b>12.2</b> Alternative experiments</a></li>
<li class="chapter" data-level="12.3" data-path="multitable.html"><a href="multitable.html#multiassayexperiments"><i class="fa fa-check"></i><b>12.3</b> MultiAssayExperiments</a></li>
<li class="chapter" data-level="" data-path="multitable.html"><a href="multitable.html#session-info-7"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="part"><span><b>III Appendix</b></span></li>
<li class="chapter" data-level="13" data-path="example-data.html"><a href="example-data.html"><i class="fa fa-check"></i><b>13</b> Example Data</a>
<ul>
<li class="chapter" data-level="13.1" data-path="example-data.html"><a href="example-data.html#package-data"><i class="fa fa-check"></i><b>13.1</b> Package data</a></li>
<li class="chapter" data-level="13.2" data-path="example-data.html"><a href="example-data.html#experimenthub-data"><i class="fa fa-check"></i><b>13.2</b> ExperimentHub data</a></li>
<li class="chapter" data-level="13.3" data-path="example-data.html"><a href="example-data.html#other-data-sources"><i class="fa fa-check"></i><b>13.3</b> Other data sources</a></li>
<li class="chapter" data-level="" data-path="example-data.html"><a href="example-data.html#session-info-8"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="function-ref.html"><a href="function-ref.html"><i class="fa fa-check"></i><b>14</b> Function Reference</a>
<ul>
<li class="chapter" data-level="14.1" data-path="function-ref.html"><a href="function-ref.html#objects"><i class="fa fa-check"></i><b>14.1</b> Objects</a></li>
<li class="chapter" data-level="14.2" data-path="function-ref.html"><a href="function-ref.html#reduced-dimension-pca-t-sne-umap-mds-etc"><i class="fa fa-check"></i><b>14.2</b> Reduced dimension (PCA, t-SNE, UMAP, MDS, etc)</a></li>
<li class="chapter" data-level="14.3" data-path="function-ref.html"><a href="function-ref.html#plotting"><i class="fa fa-check"></i><b>14.3</b> Plotting</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="extras.html"><a href="extras.html"><i class="fa fa-check"></i><b>15</b> Extras</a>
<ul>
<li class="chapter" data-level="15.1" data-path="extras.html"><a href="extras.html#interactive-3d-plots"><i class="fa fa-check"></i><b>15.1</b> Interactive 3D Plots</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="bibliography.html"><a href="bibliography.html"><i class="fa fa-check"></i>Bibliography</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Orchestrating Microbiome Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="microbiome-community" class="section level1" number="9">
<h1><span class="header-section-number">Chapter 9</span> Microbiome Community</h1>
<script>
document.addEventListener("click", function (event) {
    if (event.target.classList.contains("rebook-collapse")) {
        event.target.classList.toggle("active");
        var content = event.target.nextElementSibling;
        if (content.style.display === "block") {
            content.style.display = "none";
        } else {
            content.style.display = "block";
        }
    }
})
</script>
<style>
.rebook-collapse {
  background-color: #eee;
  color: #444;
  cursor: pointer;
  padding: 18px;
  width: 100%;
  border: none;
  text-align: left;
  outline: none;
  font-size: 15px;
}

.rebook-content {
  padding: 0 18px;
  display: none;
  overflow: hidden;
  background-color: #f1f1f1;
}
</style>
<pre><code>## Loading required package: ecodist</code></pre>
<pre><code>## Warning in library(package, lib.loc = lib.loc, character.only = TRUE,
## logical.return = TRUE, : there is no package called &#39;ecodist&#39;</code></pre>
<pre><code>## Installing package into &#39;/__w/_temp/Library&#39;
## (as &#39;lib&#39; is unspecified)</code></pre>
<div class="sourceCode" id="cb195"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb195-1"><a href="microbiome-community.html#cb195-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mia)</span>
<span id="cb195-2"><a href="microbiome-community.html#cb195-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;GlobalPatterns&quot;</span>, <span class="at">package=</span><span class="st">&quot;mia&quot;</span>)</span>
<span id="cb195-3"><a href="microbiome-community.html#cb195-3" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> GlobalPatterns</span></code></pre></div>
<div id="community-composition" class="section level2" number="9.1">
<h2><span class="header-section-number">9.1</span> Community composition</h2>
<div id="composition-barplot" class="section level3" number="9.1.1">
<h3><span class="header-section-number">9.1.1</span> Composition barplot</h3>
<p>A typical way to visualize microbiome composition is by using
composition barplot. In the following, relative abundance is
calculated and top taxa are retrieved for the Phylum rank. Thereafter,
the barplot is visualized ordering rank by abundance values and
samples by “Bacteroidetes”:</p>
<div class="sourceCode" id="cb196"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb196-1"><a href="microbiome-community.html#cb196-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(miaViz)</span>
<span id="cb196-2"><a href="microbiome-community.html#cb196-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Computing relative abundance</span></span>
<span id="cb196-3"><a href="microbiome-community.html#cb196-3" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">relAbundanceCounts</span>(tse)</span>
<span id="cb196-4"><a href="microbiome-community.html#cb196-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb196-5"><a href="microbiome-community.html#cb196-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Getting top taxa on a Phylum level</span></span>
<span id="cb196-6"><a href="microbiome-community.html#cb196-6" aria-hidden="true" tabindex="-1"></a>tse_phylum <span class="ot">&lt;-</span> <span class="fu">agglomerateByRank</span>(tse, <span class="at">rank =</span><span class="st">&quot;Phylum&quot;</span>, <span class="at">onRankOnly=</span><span class="cn">TRUE</span>)</span>
<span id="cb196-7"><a href="microbiome-community.html#cb196-7" aria-hidden="true" tabindex="-1"></a>top_taxa <span class="ot">&lt;-</span> <span class="fu">getTopTaxa</span>(tse_phylum,<span class="at">top =</span> <span class="dv">5</span>, <span class="at">abund_values =</span> <span class="st">&quot;relabundance&quot;</span>)</span>
<span id="cb196-8"><a href="microbiome-community.html#cb196-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb196-9"><a href="microbiome-community.html#cb196-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Renaming the &quot;Phylum&quot; rank to keep only top taxa and the rest to &quot;Other&quot;</span></span>
<span id="cb196-10"><a href="microbiome-community.html#cb196-10" aria-hidden="true" tabindex="-1"></a>phylum_renamed <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">rowData</span>(tse)<span class="sc">$</span>Phylum,</span>
<span id="cb196-11"><a href="microbiome-community.html#cb196-11" aria-hidden="true" tabindex="-1"></a>                   <span class="cf">function</span>(x){<span class="cf">if</span> (x <span class="sc">%in%</span> top_taxa) {x} <span class="cf">else</span> {<span class="st">&quot;Other&quot;</span>}})</span>
<span id="cb196-12"><a href="microbiome-community.html#cb196-12" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(tse)<span class="sc">$</span>Phylum <span class="ot">&lt;-</span> <span class="fu">as.character</span>(phylum_renamed)</span>
<span id="cb196-13"><a href="microbiome-community.html#cb196-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb196-14"><a href="microbiome-community.html#cb196-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualizing the composition barplot, with samples order by &quot;Bacteroidetes&quot;</span></span>
<span id="cb196-15"><a href="microbiome-community.html#cb196-15" aria-hidden="true" tabindex="-1"></a><span class="fu">plotAbundance</span>(tse, <span class="at">abund_values=</span><span class="st">&quot;relabundance&quot;</span>, <span class="at">rank =</span> <span class="st">&quot;Phylum&quot;</span>,</span>
<span id="cb196-16"><a href="microbiome-community.html#cb196-16" aria-hidden="true" tabindex="-1"></a>              <span class="at">order_rank_by=</span><span class="st">&quot;abund&quot;</span>, <span class="at">order_sample_by =</span> <span class="st">&quot;Bacteroidetes&quot;</span>)</span></code></pre></div>
<p><img src="21_microbiome_community_files/figure-html/unnamed-chunk-1-1.png" width="672" /></p>
</div>
<div id="composition-heatmap" class="section level3" number="9.1.2">
<h3><span class="header-section-number">9.1.2</span> Composition heatmap</h3>
<p>Community composition can be visualized with heatmap, where the
horizontal axis represents samples and the vertical axis the
taxa. Color of each intersection point represents abundance of a taxon
in a specific sample.</p>
<p>Here, abundances are first CLR (centered log-ratio) transformed to
remove compositionality bias. Then Z transformation is applied to
CLR-transformed data. This shifts all taxa to zero mean and unit
variance, allowing visual comparison between taxa that have different
absolute abundance levels. After these rough visual exploration
techniques, we can visualize the abundances at Phylum level.</p>
<div class="sourceCode" id="cb197"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb197-1"><a href="microbiome-community.html#cb197-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb197-2"><a href="microbiome-community.html#cb197-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Add clr-transformation on samples</span></span>
<span id="cb197-3"><a href="microbiome-community.html#cb197-3" aria-hidden="true" tabindex="-1"></a>tse_phylum <span class="ot">&lt;-</span> <span class="fu">transformSamples</span>(tse_phylum, <span class="at">method =</span> <span class="st">&quot;clr&quot;</span>, <span class="at">pseudocount =</span> <span class="dv">1</span>)</span>
<span id="cb197-4"><a href="microbiome-community.html#cb197-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Add z-transformation on features (taxa)</span></span>
<span id="cb197-5"><a href="microbiome-community.html#cb197-5" aria-hidden="true" tabindex="-1"></a>tse_phylum <span class="ot">&lt;-</span> <span class="fu">transformFeatures</span>(tse_phylum, <span class="at">abund_values =</span> <span class="st">&quot;clr&quot;</span>, </span>
<span id="cb197-6"><a href="microbiome-community.html#cb197-6" aria-hidden="true" tabindex="-1"></a>                                <span class="at">method =</span> <span class="st">&quot;z&quot;</span>, <span class="at">name =</span> <span class="st">&quot;clr_z&quot;</span>)</span>
<span id="cb197-7"><a href="microbiome-community.html#cb197-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Melts the assay</span></span>
<span id="cb197-8"><a href="microbiome-community.html#cb197-8" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">meltAssay</span>(tse_phylum, <span class="at">abund_values =</span> <span class="st">&quot;clr_z&quot;</span>)</span>
<span id="cb197-9"><a href="microbiome-community.html#cb197-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb197-10"><a href="microbiome-community.html#cb197-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Determines the scaling of colours</span></span>
<span id="cb197-11"><a href="microbiome-community.html#cb197-11" aria-hidden="true" tabindex="-1"></a>maxval <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">max</span>(<span class="fu">abs</span>(df<span class="sc">$</span>clr_z)))</span>
<span id="cb197-12"><a href="microbiome-community.html#cb197-12" aria-hidden="true" tabindex="-1"></a>limits <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="sc">-</span>maxval, maxval)</span>
<span id="cb197-13"><a href="microbiome-community.html#cb197-13" aria-hidden="true" tabindex="-1"></a>breaks <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="at">from =</span> <span class="fu">min</span>(limits), <span class="at">to =</span> <span class="fu">max</span>(limits), <span class="at">by =</span> <span class="fl">0.5</span>)</span>
<span id="cb197-14"><a href="microbiome-community.html#cb197-14" aria-hidden="true" tabindex="-1"></a>colours <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;darkblue&quot;</span>, <span class="st">&quot;blue&quot;</span>, <span class="st">&quot;white&quot;</span>, <span class="st">&quot;red&quot;</span>, <span class="st">&quot;darkred&quot;</span>)</span>
<span id="cb197-15"><a href="microbiome-community.html#cb197-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb197-16"><a href="microbiome-community.html#cb197-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates a ggplot object</span></span>
<span id="cb197-17"><a href="microbiome-community.html#cb197-17" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x =</span> SampleID, <span class="at">y =</span> FeatureID, <span class="at">fill =</span> clr_z)) <span class="sc">+</span></span>
<span id="cb197-18"><a href="microbiome-community.html#cb197-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb197-19"><a href="microbiome-community.html#cb197-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradientn</span>(<span class="at">name =</span> <span class="st">&quot;CLR + Z transform&quot;</span>, </span>
<span id="cb197-20"><a href="microbiome-community.html#cb197-20" aria-hidden="true" tabindex="-1"></a>                       <span class="at">breaks =</span> breaks, <span class="at">limits =</span> limits, <span class="at">colours =</span> colours) <span class="sc">+</span> </span>
<span id="cb197-21"><a href="microbiome-community.html#cb197-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>),</span>
<span id="cb197-22"><a href="microbiome-community.html#cb197-22" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">45</span>, <span class="at">hjust=</span><span class="dv">1</span>),</span>
<span id="cb197-23"><a href="microbiome-community.html#cb197-23" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.key.size =</span> <span class="fu">unit</span>(<span class="dv">1</span>, <span class="st">&quot;cm&quot;</span>)) <span class="sc">+</span></span>
<span id="cb197-24"><a href="microbiome-community.html#cb197-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Samples&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Taxa&quot;</span>)</span></code></pre></div>
<p><img src="21_microbiome_community_files/figure-html/heatmap-1.png" width="672" /></p>
<p><em>pheatmap</em> is a package that provides methods to plot clustered heatmaps.</p>
<div class="sourceCode" id="cb198"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb198-1"><a href="microbiome-community.html#cb198-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(pheatmap)){</span>
<span id="cb198-2"><a href="microbiome-community.html#cb198-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;pheatmap&quot;</span>)</span>
<span id="cb198-3"><a href="microbiome-community.html#cb198-3" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb198-4"><a href="microbiome-community.html#cb198-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb198-5"><a href="microbiome-community.html#cb198-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb198-6"><a href="microbiome-community.html#cb198-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Takes subset: only samples from feces, skin, or tongue</span></span>
<span id="cb198-7"><a href="microbiome-community.html#cb198-7" aria-hidden="true" tabindex="-1"></a>tse_phylum_subset <span class="ot">&lt;-</span> tse_phylum[ , <span class="fu">colData</span>(tse_phylum)<span class="sc">$</span>SampleType <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Feces&quot;</span>, <span class="st">&quot;Skin&quot;</span>, <span class="st">&quot;Tongue&quot;</span>) ]</span>
<span id="cb198-8"><a href="microbiome-community.html#cb198-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb198-9"><a href="microbiome-community.html#cb198-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Does clr-transformation</span></span>
<span id="cb198-10"><a href="microbiome-community.html#cb198-10" aria-hidden="true" tabindex="-1"></a>tse_phylum_subset <span class="ot">&lt;-</span> <span class="fu">transformSamples</span>(tse_phylum_subset, <span class="at">method =</span> <span class="st">&quot;clr&quot;</span>, <span class="at">pseudocount =</span> <span class="dv">1</span>)</span>
<span id="cb198-11"><a href="microbiome-community.html#cb198-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Does z-transformation</span></span>
<span id="cb198-12"><a href="microbiome-community.html#cb198-12" aria-hidden="true" tabindex="-1"></a>tse_phylum_subset <span class="ot">&lt;-</span> <span class="fu">transformFeatures</span>(tse_phylum_subset, <span class="at">abund_values =</span> <span class="st">&quot;clr&quot;</span>, </span>
<span id="cb198-13"><a href="microbiome-community.html#cb198-13" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">method =</span> <span class="st">&quot;z&quot;</span>, <span class="at">name =</span> <span class="st">&quot;clr_z&quot;</span>)</span>
<span id="cb198-14"><a href="microbiome-community.html#cb198-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb198-15"><a href="microbiome-community.html#cb198-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Get n most abundant taxa, and subsets the data by them</span></span>
<span id="cb198-16"><a href="microbiome-community.html#cb198-16" aria-hidden="true" tabindex="-1"></a>top_taxa <span class="ot">&lt;-</span> <span class="fu">getTopTaxa</span>(tse_phylum_subset, <span class="at">top =</span> <span class="dv">20</span>)</span>
<span id="cb198-17"><a href="microbiome-community.html#cb198-17" aria-hidden="true" tabindex="-1"></a>tse_phylum_subset <span class="ot">&lt;-</span> tse_phylum_subset[top_taxa, ]</span>
<span id="cb198-18"><a href="microbiome-community.html#cb198-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb198-19"><a href="microbiome-community.html#cb198-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Gets the assay table</span></span>
<span id="cb198-20"><a href="microbiome-community.html#cb198-20" aria-hidden="true" tabindex="-1"></a>mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(tse_phylum_subset, <span class="st">&quot;clr_z&quot;</span>)</span>
<span id="cb198-21"><a href="microbiome-community.html#cb198-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb198-22"><a href="microbiome-community.html#cb198-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates the heatmap</span></span>
<span id="cb198-23"><a href="microbiome-community.html#cb198-23" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(mat)</span></code></pre></div>
<p><img src="21_microbiome_community_files/figure-html/pheatmap1-1.png" width="672" /></p>
<p>We can create clusters by hierarchical clustering and
visualize them with dendrogram.</p>
<div class="sourceCode" id="cb199"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb199-1"><a href="microbiome-community.html#cb199-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Package for creating dendrograms</span></span>
<span id="cb199-2"><a href="microbiome-community.html#cb199-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(dendextend)){</span>
<span id="cb199-3"><a href="microbiome-community.html#cb199-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;dendextend&quot;</span>)</span>
<span id="cb199-4"><a href="microbiome-community.html#cb199-4" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb199-5"><a href="microbiome-community.html#cb199-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dendextend)</span>
<span id="cb199-6"><a href="microbiome-community.html#cb199-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb199-7"><a href="microbiome-community.html#cb199-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Hierarchical clustering</span></span>
<span id="cb199-8"><a href="microbiome-community.html#cb199-8" aria-hidden="true" tabindex="-1"></a>taxa_clusters <span class="ot">&lt;-</span> <span class="fu">hclust</span>(<span class="fu">dist</span>(mat), <span class="at">method =</span> <span class="st">&quot;complete&quot;</span>)</span>
<span id="cb199-9"><a href="microbiome-community.html#cb199-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb199-10"><a href="microbiome-community.html#cb199-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates a dendrogram</span></span>
<span id="cb199-11"><a href="microbiome-community.html#cb199-11" aria-hidden="true" tabindex="-1"></a>taxa_dendrogram <span class="ot">&lt;-</span> <span class="fu">as.dendrogram</span>(taxa_clusters)</span>
<span id="cb199-12"><a href="microbiome-community.html#cb199-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb199-13"><a href="microbiome-community.html#cb199-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Plots it</span></span>
<span id="cb199-14"><a href="microbiome-community.html#cb199-14" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(taxa_dendrogram)</span></code></pre></div>
<p><img src="21_microbiome_community_files/figure-html/pheatmap2-1.png" width="672" /></p>
<p>Based on dendrogram, we decide to create three clusters.</p>
<div class="sourceCode" id="cb200"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb200-1"><a href="microbiome-community.html#cb200-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates clusters</span></span>
<span id="cb200-2"><a href="microbiome-community.html#cb200-2" aria-hidden="true" tabindex="-1"></a>taxa_clusters <span class="ot">&lt;-</span> <span class="fu">cutree</span>(<span class="at">tree =</span> taxa_dendrogram, <span class="at">k =</span> <span class="dv">3</span>)</span>
<span id="cb200-3"><a href="microbiome-community.html#cb200-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb200-4"><a href="microbiome-community.html#cb200-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Prints taxa and their clusters</span></span>
<span id="cb200-5"><a href="microbiome-community.html#cb200-5" aria-hidden="true" tabindex="-1"></a>taxa_clusters </span></code></pre></div>
<pre><code>##       Firmicutes    Bacteroidetes   Proteobacteria   Actinobacteria 
##                1                1                2                3 
##    Cyanobacteria     Fusobacteria      Tenericutes  Verrucomicrobia 
##                2                2                1                1 
##    Lentisphaerae    Euryarchaeota    Acidobacteria     Spirochaetes 
##                1                1                3                2 
##   Planctomycetes           Thermi      Chloroflexi              SR1 
##                3                3                3                2 
##    Synergistetes    Crenarchaeota Gemmatimonadetes           SAR406 
##                2                3                3                1</code></pre>
<div class="sourceCode" id="cb202"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb202-1"><a href="microbiome-community.html#cb202-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates clusters, and adds information to rowData</span></span>
<span id="cb202-2"><a href="microbiome-community.html#cb202-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(tse_phylum_subset)<span class="sc">$</span>clusters <span class="ot">&lt;-</span> <span class="fu">cutree</span>(<span class="at">tree =</span> taxa_dendrogram, <span class="at">k =</span> <span class="dv">3</span>)</span>
<span id="cb202-3"><a href="microbiome-community.html#cb202-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb202-4"><a href="microbiome-community.html#cb202-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Prints taxa and their clusters</span></span>
<span id="cb202-5"><a href="microbiome-community.html#cb202-5" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(tse_phylum_subset)<span class="sc">$</span>clusters</span></code></pre></div>
<pre><code>##       Firmicutes    Bacteroidetes   Proteobacteria   Actinobacteria 
##                1                1                2                3 
##    Cyanobacteria     Fusobacteria      Tenericutes  Verrucomicrobia 
##                2                2                1                1 
##    Lentisphaerae    Euryarchaeota    Acidobacteria     Spirochaetes 
##                1                1                3                2 
##   Planctomycetes           Thermi      Chloroflexi              SR1 
##                3                3                3                2 
##    Synergistetes    Crenarchaeota Gemmatimonadetes           SAR406 
##                2                3                3                1</code></pre>
<p>Now we can create heatmap with additional annotations.</p>
<div class="sourceCode" id="cb204"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb204-1"><a href="microbiome-community.html#cb204-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates data frame that includes cluster data</span></span>
<span id="cb204-2"><a href="microbiome-community.html#cb204-2" aria-hidden="true" tabindex="-1"></a>taxa_clusters <span class="ot">&lt;-</span> <span class="fu">rowData</span>(tse_phylum_subset)<span class="sc">$</span>clusters</span>
<span id="cb204-3"><a href="microbiome-community.html#cb204-3" aria-hidden="true" tabindex="-1"></a>taxa_clusters <span class="ot">&lt;-</span> <span class="fu">as.character</span>(taxa_clusters)</span>
<span id="cb204-4"><a href="microbiome-community.html#cb204-4" aria-hidden="true" tabindex="-1"></a>taxa_clusters <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">cluster =</span> taxa_clusters)</span>
<span id="cb204-5"><a href="microbiome-community.html#cb204-5" aria-hidden="true" tabindex="-1"></a><span class="fu">row.names</span>(taxa_clusters) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(tse_phylum_subset)</span>
<span id="cb204-6"><a href="microbiome-community.html#cb204-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb204-7"><a href="microbiome-community.html#cb204-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates data frame that includes data on sample type</span></span>
<span id="cb204-8"><a href="microbiome-community.html#cb204-8" aria-hidden="true" tabindex="-1"></a>sample_types <span class="ot">&lt;-</span> <span class="fu">unfactor</span>(<span class="fu">colData</span>(tse_phylum_subset)<span class="sc">$</span>SampleType)</span>
<span id="cb204-9"><a href="microbiome-community.html#cb204-9" aria-hidden="true" tabindex="-1"></a>sample_types <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">sample_types =</span> sample_types)</span>
<span id="cb204-10"><a href="microbiome-community.html#cb204-10" aria-hidden="true" tabindex="-1"></a><span class="fu">row.names</span>(sample_types) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(tse_phylum_subset)</span>
<span id="cb204-11"><a href="microbiome-community.html#cb204-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb204-12"><a href="microbiome-community.html#cb204-12" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(mat, <span class="at">annotation_row =</span> taxa_clusters, </span>
<span id="cb204-13"><a href="microbiome-community.html#cb204-13" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_col =</span> sample_types)</span></code></pre></div>
<p><img src="21_microbiome_community_files/figure-html/pheatmap5-1.png" width="672" /></p>
<p>In addition to <em>pheatmap</em> package, there are also other packages that provide
functions for more complex heatmaps. One example is <em>iheatmapr</em> package. Examples
of using it you can find from its
<a href="https://docs.ropensci.org/iheatmapr/articles/full_vignettes/iheatmapr.html">vignette</a>.</p>
</div>
</div>
<div id="session-info-5" class="section level2 unnumbered">
<h2>Session Info</h2>
<button class="rebook-collapse">
View session info
</button>
<div class="rebook-content">
<pre><code>R version 4.1.1 (2021-08-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.3 LTS

Matrix products: default
BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] dendextend_1.15.1              pheatmap_1.0.12               
 [3] miaViz_1.1.4                   ggraph_2.0.5                  
 [5] ggplot2_3.3.5                  mia_1.1.14                    
 [7] TreeSummarizedExperiment_2.1.4 Biostrings_2.61.2             
 [9] XVector_0.33.0                 SingleCellExperiment_1.15.2   
[11] SummarizedExperiment_1.23.4    Biobase_2.53.0                
[13] GenomicRanges_1.45.0           GenomeInfoDb_1.29.8           
[15] IRanges_2.27.2                 S4Vectors_0.31.3              
[17] BiocGenerics_0.39.2            MatrixGenerics_1.5.4          
[19] matrixStats_0.60.1-9001        ecodist_2.0.7                 
[21] BiocStyle_2.21.3               rebook_1.3.1                  

loaded via a namespace (and not attached):
  [1] plyr_1.8.6                  igraph_1.2.6               
  [3] lazyeval_0.2.2              splines_4.1.1              
  [5] BiocParallel_1.27.6         scater_1.21.3              
  [7] digest_0.6.27               yulab.utils_0.0.2          
  [9] htmltools_0.5.2             viridis_0.6.1              
 [11] fansi_0.5.0                 magrittr_2.0.1             
 [13] memoise_2.0.0               ScaledMatrix_1.1.0         
 [15] cluster_2.1.2               DECIPHER_2.21.0            
 [17] graphlayouts_0.7.1          colorspace_2.0-2           
 [19] blob_1.2.2                  ggrepel_0.9.1              
 [21] xfun_0.25                   dplyr_1.0.7                
 [23] crayon_1.4.1                RCurl_1.98-1.4             
 [25] jsonlite_1.7.2              graph_1.71.2               
 [27] ape_5.5                     glue_1.4.2                 
 [29] polyclip_1.10-0             gtable_0.3.0               
 [31] zlibbioc_1.39.0             DelayedArray_0.19.2        
 [33] BiocSingular_1.9.1          scales_1.1.1               
 [35] DBI_1.1.1                   Rcpp_1.0.7                 
 [37] viridisLite_0.4.0           decontam_1.13.0            
 [39] gridGraphics_0.5-1          tidytree_0.3.5             
 [41] bit_4.0.4                   rsvd_1.0.5                 
 [43] RColorBrewer_1.1-2          dir.expiry_1.1.0           
 [45] ellipsis_0.3.2              pkgconfig_2.0.3            
 [47] XML_3.99-0.7                farver_2.1.0               
 [49] scuttle_1.3.1               CodeDepends_0.6.5          
 [51] sass_0.4.0                  utf8_1.2.2                 
 [53] ggplotify_0.1.0             tidyselect_1.1.1           
 [55] labeling_0.4.2              rlang_0.4.11               
 [57] reshape2_1.4.4              munsell_0.5.0              
 [59] tools_4.1.1                 cachem_1.0.6               
 [61] DirichletMultinomial_1.35.0 generics_0.1.0             
 [63] RSQLite_2.2.8               evaluate_0.14              
 [65] stringr_1.4.0               fastmap_1.1.0              
 [67] yaml_2.2.1                  ggtree_3.1.4               
 [69] knitr_1.33                  bit64_4.0.5                
 [71] tidygraph_1.2.0             purrr_0.3.4                
 [73] nlme_3.1-153                sparseMatrixStats_1.5.3    
 [75] aplot_0.1.0                 compiler_4.1.1             
 [77] beeswarm_0.4.0              filelock_1.0.2             
 [79] treeio_1.17.2               tibble_3.1.4               
 [81] tweenr_1.0.2                bslib_0.3.0                
 [83] stringi_1.7.4               highr_0.9                  
 [85] lattice_0.20-44             Matrix_1.3-4               
 [87] vegan_2.5-7                 permute_0.9-5              
 [89] vctrs_0.3.8                 pillar_1.6.2               
 [91] lifecycle_1.0.0             BiocManager_1.30.16        
 [93] jquerylib_0.1.4             BiocNeighbors_1.11.0       
 [95] bitops_1.0-7                irlba_2.3.3                
 [97] patchwork_1.1.1             R6_2.5.1                   
 [99] bookdown_0.24               gridExtra_2.3              
[101] vipor_0.4.5                 codetools_0.2-18           
[103] MASS_7.3-54                 assertthat_0.2.1           
[105] withr_2.4.2                 GenomeInfoDbData_1.2.6     
[107] mgcv_1.8-36                 parallel_4.1.1             
[109] grid_4.1.1                  ggfun_0.0.3                
[111] beachmat_2.9.1              tidyr_1.1.3                
[113] rmarkdown_2.10              DelayedMatrixStats_1.15.4  
[115] ggnewscale_0.4.5            ggforce_0.3.3              
[117] ggbeeswarm_0.6.0           </code></pre>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="beta-diversity.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="community-typing.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["OMA.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
