<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 Microbiome Community | Interactive 3D plots</title>
  <meta name="description" content="Chapter 6 Microbiome Community | Interactive 3D plots" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 Microbiome Community | Interactive 3D plots" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://github.com/microbiome/OMA" />
  
  
  <meta name="github-repo" content="microbiome/OMA" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 Microbiome Community | Interactive 3D plots" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="microbiome-exploration.html"/>
<link rel="next" href="differential-abundance.html"/>
<script src="libs/header-attrs-2.9/header-attrs.js"></script>
<script src="libs/jquery-3.5.1/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<script src="libs/plotly-binding-4.9.4.1/plotly.js"></script>
<script src="libs/typedarray-0.1/typedarray.min.js"></script>
<link href="libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
<link href="libs/plotly-htmlwidgets-css-1.57.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="libs/plotly-main-1.57.1/plotly-latest.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Orchestrating Microbiome Analysis</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="part"><span><b>I Introduction</b></span></li>
<li class="chapter" data-level="1" data-path="data-introduction.html"><a href="data-introduction.html"><i class="fa fa-check"></i><b>1</b> Data Infrastructure</a>
<ul>
<li class="chapter" data-level="1.1" data-path="data-introduction.html"><a href="data-introduction.html#installation"><i class="fa fa-check"></i><b>1.1</b> Installation</a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="data-introduction.html"><a href="data-introduction.html#packages"><i class="fa fa-check"></i><b>1.1.1</b> Packages</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="data-introduction.html"><a href="data-introduction.html#background"><i class="fa fa-check"></i><b>1.2</b> Background</a></li>
<li class="chapter" data-level="1.3" data-path="data-introduction.html"><a href="data-introduction.html#loading-experimental-microbiome-data"><i class="fa fa-check"></i><b>1.3</b> Loading experimental microbiome data</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="data-introduction.html"><a href="data-introduction.html#importing-data-from-external-files"><i class="fa fa-check"></i><b>1.3.1</b> Importing data from external files</a></li>
<li class="chapter" data-level="1.3.2" data-path="data-introduction.html"><a href="data-introduction.html#conversions-between-data-formats-in-r"><i class="fa fa-check"></i><b>1.3.2</b> Conversions between data formats in R</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="data-introduction.html"><a href="data-introduction.html#metadata"><i class="fa fa-check"></i><b>1.4</b> Metadata</a></li>
<li class="chapter" data-level="1.5" data-path="data-introduction.html"><a href="data-introduction.html#microbiome-and-tree-data-specific-aspects"><i class="fa fa-check"></i><b>1.5</b> Microbiome and tree data specific aspects</a>
<ul>
<li class="chapter" data-level="1.5.1" data-path="data-introduction.html"><a href="data-introduction.html#assays"><i class="fa fa-check"></i><b>1.5.1</b> Assays</a></li>
<li class="chapter" data-level="1.5.2" data-path="data-introduction.html"><a href="data-introduction.html#coldata"><i class="fa fa-check"></i><b>1.5.2</b> colData</a></li>
<li class="chapter" data-level="1.5.3" data-path="data-introduction.html"><a href="data-introduction.html#rowdata"><i class="fa fa-check"></i><b>1.5.3</b> rowData</a></li>
<li class="chapter" data-level="1.5.4" data-path="data-introduction.html"><a href="data-introduction.html#rowtree"><i class="fa fa-check"></i><b>1.5.4</b> rowTree</a></li>
</ul></li>
<li class="chapter" data-level="1.6" data-path="data-introduction.html"><a href="data-introduction.html#data-conversion"><i class="fa fa-check"></i><b>1.6</b> Data conversion</a>
<ul>
<li class="chapter" data-level="1.6.1" data-path="data-introduction.html"><a href="data-introduction.html#tidy-data"><i class="fa fa-check"></i><b>1.6.1</b> Tidy data</a></li>
</ul></li>
<li class="chapter" data-level="1.7" data-path="data-introduction.html"><a href="data-introduction.html#conclusion"><i class="fa fa-check"></i><b>1.7</b> Conclusion</a></li>
<li class="chapter" data-level="1.8" data-path="data-introduction.html"><a href="data-introduction.html#additional-reading"><i class="fa fa-check"></i><b>1.8</b> Additional Reading</a>
<ul>
<li class="chapter" data-level="1.8.1" data-path="data-introduction.html"><a href="data-introduction.html#lecture-slides"><i class="fa fa-check"></i><b>1.8.1</b> Lecture slides</a></li>
<li class="chapter" data-level="1.8.2" data-path="data-introduction.html"><a href="data-introduction.html#r-programming-resources"><i class="fa fa-check"></i><b>1.8.2</b> R programming resources</a></li>
<li class="chapter" data-level="1.8.3" data-path="data-introduction.html"><a href="data-introduction.html#resources-for-treesummarizedexperiment"><i class="fa fa-check"></i><b>1.8.3</b> Resources for TreeSummarizedExperiment</a></li>
<li class="chapter" data-level="1.8.4" data-path="data-introduction.html"><a href="data-introduction.html#resources-for-phyloseq"><i class="fa fa-check"></i><b>1.8.4</b> Resources for phyloseq</a></li>
<li class="chapter" data-level="1.8.5" data-path="data-introduction.html"><a href="data-introduction.html#further-reading"><i class="fa fa-check"></i><b>1.8.5</b> Further reading</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-introduction.html"><a href="data-introduction.html#session-info"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="part"><span><b>II Focus Topics</b></span></li>
<li class="chapter" data-level="2" data-path="overview.html"><a href="overview.html"><i class="fa fa-check"></i><b>2</b> Overview</a>
<ul>
<li class="chapter" data-level="2.1" data-path="overview.html"><a href="overview.html#introduction"><i class="fa fa-check"></i><b>2.1</b> Introduction</a></li>
<li class="chapter" data-level="2.2" data-path="overview.html"><a href="overview.html#microbiome-experiment-types"><i class="fa fa-check"></i><b>2.2</b> Microbiome experiment types</a></li>
<li class="chapter" data-level="2.3" data-path="overview.html"><a href="overview.html#processing-steps"><i class="fa fa-check"></i><b>2.3</b> Processing steps</a></li>
<li class="chapter" data-level="2.4" data-path="overview.html"><a href="overview.html#quick-start"><i class="fa fa-check"></i><b>2.4</b> Quick Start</a></li>
<li class="chapter" data-level="" data-path="overview.html"><a href="overview.html#session-info-1"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="taxonomic-information.html"><a href="taxonomic-information.html"><i class="fa fa-check"></i><b>3</b> Taxonomic information</a>
<ul>
<li class="chapter" data-level="3.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#assigning-taxonomic-information."><i class="fa fa-check"></i><b>3.1</b> Assigning taxonomic information.</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#dada2"><i class="fa fa-check"></i><b>3.1.1</b> dada2</a></li>
<li class="chapter" data-level="3.1.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#decipher"><i class="fa fa-check"></i><b>3.1.2</b> DECIPHER</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#functions-to-access-taxonomic-information"><i class="fa fa-check"></i><b>3.2</b> Functions to access taxonomic information</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#generate-a-taxonomic-tree-on-the-fly"><i class="fa fa-check"></i><b>3.2.1</b> Generate a taxonomic tree on the fly</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="taxonomic-information.html"><a href="taxonomic-information.html#data-agglomeration"><i class="fa fa-check"></i><b>3.3</b> Data agglomeration</a></li>
<li class="chapter" data-level="3.4" data-path="taxonomic-information.html"><a href="taxonomic-information.html#pick-specific"><i class="fa fa-check"></i><b>3.4</b> Pick specific</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#abundances-of-all-taxa-in-specific-sample"><i class="fa fa-check"></i><b>3.4.1</b> Abundances of all taxa in specific sample</a></li>
<li class="chapter" data-level="3.4.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#abundances-of-specific-taxa-in-all-samples"><i class="fa fa-check"></i><b>3.4.2</b> Abundances of specific taxa in all samples</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="taxonomic-information.html"><a href="taxonomic-information.html#session-info-2"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>4</b> Quality Control</a>
<ul>
<li class="chapter" data-level="4.1" data-path="quality-control.html"><a href="quality-control.html#get-top-taxa-and-taxonomy"><i class="fa fa-check"></i><b>4.1</b> Get top taxa and taxonomy</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="quality-control.html"><a href="quality-control.html#features"><i class="fa fa-check"></i><b>4.1.1</b> Features</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="quality-control.html"><a href="quality-control.html#library-size"><i class="fa fa-check"></i><b>4.2</b> Library size</a></li>
<li class="chapter" data-level="" data-path="quality-control.html"><a href="quality-control.html#session-info-3"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html"><i class="fa fa-check"></i><b>5</b> Microbiome Exploration</a>
<ul>
<li class="chapter" data-level="5.1" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#prevalence"><i class="fa fa-check"></i><b>5.1</b> Prevalence</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#prevalent-microbiota-analysis"><i class="fa fa-check"></i><b>5.1.1</b> Prevalent microbiota analysis</a></li>
<li class="chapter" data-level="5.1.2" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#plotting-prevalence"><i class="fa fa-check"></i><b>5.1.2</b> Plotting prevalence</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#session-info-4"><i class="fa fa-check"></i>Session Info</a></li>
<li class="chapter" data-level="5.2" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#alpha-diversity"><i class="fa fa-check"></i><b>5.2</b> Alpha diversity</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#estimating-alpha-diversity"><i class="fa fa-check"></i><b>5.2.1</b> Estimating alpha diversity</a></li>
<li class="chapter" data-level="5.2.2" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#visualize-alpha-diversities"><i class="fa fa-check"></i><b>5.2.2</b> Visualize alpha diversities</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#beta-diversity"><i class="fa fa-check"></i><b>5.3</b> Beta diversity</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#explained-variance"><i class="fa fa-check"></i><b>5.3.1</b> Explained variance</a></li>
<li class="chapter" data-level="5.3.2" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#estimating-beta-diversity"><i class="fa fa-check"></i><b>5.3.2</b> Estimating beta diversity</a></li>
<li class="chapter" data-level="5.3.3" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#other-ordination-methods"><i class="fa fa-check"></i><b>5.3.3</b> Other ordination methods</a></li>
<li class="chapter" data-level="5.3.4" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#visualizing-the-most-dominant-genus-on-pcoa"><i class="fa fa-check"></i><b>5.3.4</b> Visualizing the most dominant genus on PCoA</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#community-comparisons-todo-combine-with-the-material-above-for-simplicity"><i class="fa fa-check"></i><b>5.4</b> Community comparisons [TODO combine with the material above for simplicity?]</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#testing-differences-in-community-composition-between-sample-groups"><i class="fa fa-check"></i><b>5.4.1</b> Testing differences in community composition between sample groups</a></li>
<li class="chapter" data-level="5.4.2" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>5.4.2</b> Checking the homogeneity condition</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#further-reading-1"><i class="fa fa-check"></i><b>5.5</b> Further reading</a></li>
<li class="chapter" data-level="" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#session-info-5"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="microbiome-community.html"><a href="microbiome-community.html"><i class="fa fa-check"></i><b>6</b> Microbiome Community</a>
<ul>
<li class="chapter" data-level="6.1" data-path="microbiome-community.html"><a href="microbiome-community.html#community-composition"><i class="fa fa-check"></i><b>6.1</b> Community composition</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="microbiome-community.html"><a href="microbiome-community.html#composition-barplot"><i class="fa fa-check"></i><b>6.1.1</b> Composition barplot</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="microbiome-community.html"><a href="microbiome-community.html#community-typing"><i class="fa fa-check"></i><b>6.2</b> Community typing</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="microbiome-community.html"><a href="microbiome-community.html#dirichlet-multinomial-mixtures-dmm"><i class="fa fa-check"></i><b>6.2.1</b> Dirichlet Multinomial Mixtures (DMM)</a></li>
<li class="chapter" data-level="6.2.2" data-path="microbiome-community.html"><a href="microbiome-community.html#pcoa-for-asv-level-data-with-bray-curtis-with-dmm-clusters-shown-with-colors"><i class="fa fa-check"></i><b>6.2.2</b> PCoA for ASV-level data with Bray-Curtis; with DMM clusters shown with colors</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="microbiome-community.html"><a href="microbiome-community.html#session-info-6"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="differential-abundance.html"><a href="differential-abundance.html"><i class="fa fa-check"></i><b>7</b> Differential abundance</a>
<ul>
<li class="chapter" data-level="" data-path="differential-abundance.html"><a href="differential-abundance.html#session-info-7"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="microbiome-timeseries.html"><a href="microbiome-timeseries.html"><i class="fa fa-check"></i><b>8</b> Microbiome time series</a>
<ul>
<li class="chapter" data-level="8.1" data-path="microbiome-timeseries.html"><a href="microbiome-timeseries.html#stability"><i class="fa fa-check"></i><b>8.1</b> Stability</a></li>
<li class="chapter" data-level="8.2" data-path="microbiome-timeseries.html"><a href="microbiome-timeseries.html#tipping-elements"><i class="fa fa-check"></i><b>8.2</b> Tipping elements</a></li>
<li class="chapter" data-level="" data-path="microbiome-timeseries.html"><a href="microbiome-timeseries.html#session-info-8"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="multitable.html"><a href="multitable.html"><i class="fa fa-check"></i><b>9</b> Representation of multiple data tables</a>
<ul>
<li class="chapter" data-level="9.1" data-path="multitable.html"><a href="multitable.html#assay-data"><i class="fa fa-check"></i><b>9.1</b> Assay data</a></li>
<li class="chapter" data-level="9.2" data-path="multitable.html"><a href="multitable.html#alternative-experiments"><i class="fa fa-check"></i><b>9.2</b> Alternative experiments</a></li>
<li class="chapter" data-level="9.3" data-path="multitable.html"><a href="multitable.html#multiassayexperiments"><i class="fa fa-check"></i><b>9.3</b> MultiAssayExperiments</a></li>
<li class="chapter" data-level="" data-path="multitable.html"><a href="multitable.html#session-info-9"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="part"><span><b>III Appendix</b></span></li>
<li class="chapter" data-level="10" data-path="example-data.html"><a href="example-data.html"><i class="fa fa-check"></i><b>10</b> Example data</a>
<ul>
<li class="chapter" data-level="10.1" data-path="example-data.html"><a href="example-data.html#package-data"><i class="fa fa-check"></i><b>10.1</b> Package data</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="example-data.html"><a href="example-data.html#globalpatterns"><i class="fa fa-check"></i><b>10.1.1</b> GlobalPatterns</a></li>
<li class="chapter" data-level="10.1.2" data-path="example-data.html"><a href="example-data.html#enterotype"><i class="fa fa-check"></i><b>10.1.2</b> Enterotype</a></li>
<li class="chapter" data-level="10.1.3" data-path="example-data.html"><a href="example-data.html#esophagus"><i class="fa fa-check"></i><b>10.1.3</b> Esophagus</a></li>
<li class="chapter" data-level="10.1.4" data-path="example-data.html"><a href="example-data.html#soilrep"><i class="fa fa-check"></i><b>10.1.4</b> Soilrep</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="example-data.html"><a href="example-data.html#session-info-10"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="part"><span><b>IV Appendix</b></span></li>
<li class="chapter" data-level="11" data-path="function-ref.html"><a href="function-ref.html"><i class="fa fa-check"></i><b>11</b> Function reference</a>
<ul>
<li class="chapter" data-level="11.1" data-path="function-ref.html"><a href="function-ref.html#objects"><i class="fa fa-check"></i><b>11.1</b> Objects</a></li>
<li class="chapter" data-level="11.2" data-path="function-ref.html"><a href="function-ref.html#reduced-dimension-pca-t-sne-umap-mds-etc"><i class="fa fa-check"></i><b>11.2</b> Reduced dimension (PCA, t-SNE, UMAP, MDS, etc)</a></li>
<li class="chapter" data-level="11.3" data-path="function-ref.html"><a href="function-ref.html#plotting"><i class="fa fa-check"></i><b>11.3</b> Plotting</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="bibliography.html"><a href="bibliography.html"><i class="fa fa-check"></i>Bibliography</a></li>
<li class="chapter" data-level="" data-path="interactive-3d-plots.html"><a href="interactive-3d-plots.html"><i class="fa fa-check"></i>Interactive 3D plots</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Interactive 3D plots</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="microbiome-community" class="section level1" number="6">
<h1><span class="header-section-number">Chapter 6</span> Microbiome Community</h1>
<script>
document.addEventListener("click", function (event) {
    if (event.target.classList.contains("rebook-collapse")) {
        event.target.classList.toggle("active");
        var content = event.target.nextElementSibling;
        if (content.style.display === "block") {
            content.style.display = "none";
        } else {
            content.style.display = "block";
        }
    }
})
</script>
<style>
.rebook-collapse {
  background-color: #eee;
  color: #444;
  cursor: pointer;
  padding: 18px;
  width: 100%;
  border: none;
  text-align: left;
  outline: none;
  font-size: 15px;
}

.rebook-content {
  padding: 0 18px;
  display: none;
  overflow: hidden;
  background-color: #f1f1f1;
}
</style>
<pre><code>## Loading required package: ecodist</code></pre>
<div class="sourceCode" id="cb160"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb160-1"><a href="microbiome-community.html#cb160-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mia)</span>
<span id="cb160-2"><a href="microbiome-community.html#cb160-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;GlobalPatterns&quot;</span>)</span>
<span id="cb160-3"><a href="microbiome-community.html#cb160-3" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> GlobalPatterns</span></code></pre></div>
<div id="community-composition" class="section level2" number="6.1">
<h2><span class="header-section-number">6.1</span> Community composition</h2>
<p>Heatmaps</p>
<div id="composition-barplot" class="section level3" number="6.1.1">
<h3><span class="header-section-number">6.1.1</span> Composition barplot</h3>
<p>A typical way to visualize microbiome composition is by using composition barplot.
In the following, relative abundance is calculated and top 5 taxa is retrieved for the
Phylum rank. Thereafter, the barplot is visualized ordering rank by abundance values
and samples by “Bacteroidetes”:</p>
<div class="sourceCode" id="cb161"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb161-1"><a href="microbiome-community.html#cb161-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(miaViz)</span>
<span id="cb161-2"><a href="microbiome-community.html#cb161-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Computing relative abundance</span></span>
<span id="cb161-3"><a href="microbiome-community.html#cb161-3" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">relAbundanceCounts</span>(tse)</span>
<span id="cb161-4"><a href="microbiome-community.html#cb161-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb161-5"><a href="microbiome-community.html#cb161-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Getting top taxa on a Phylum level</span></span>
<span id="cb161-6"><a href="microbiome-community.html#cb161-6" aria-hidden="true" tabindex="-1"></a>tse_phylum <span class="ot">&lt;-</span> <span class="fu">agglomerateByRank</span>(tse, <span class="at">rank =</span><span class="st">&quot;Phylum&quot;</span>, <span class="at">onRankOnly=</span><span class="cn">TRUE</span>)</span>
<span id="cb161-7"><a href="microbiome-community.html#cb161-7" aria-hidden="true" tabindex="-1"></a>top_taxa <span class="ot">&lt;-</span> <span class="fu">getTopTaxa</span>(tse_phylum,<span class="at">top =</span> <span class="dv">5</span>, <span class="at">abund_values =</span> <span class="st">&quot;relabundance&quot;</span>)</span>
<span id="cb161-8"><a href="microbiome-community.html#cb161-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb161-9"><a href="microbiome-community.html#cb161-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Renaming the &quot;Phylum&quot; rank to keep only top taxa and the rest to &quot;Other&quot;</span></span>
<span id="cb161-10"><a href="microbiome-community.html#cb161-10" aria-hidden="true" tabindex="-1"></a>phylum_renamed <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">rowData</span>(tse)<span class="sc">$</span>Phylum,</span>
<span id="cb161-11"><a href="microbiome-community.html#cb161-11" aria-hidden="true" tabindex="-1"></a>                   <span class="cf">function</span>(x){<span class="cf">if</span> (x <span class="sc">%in%</span> top_taxa) {x} <span class="cf">else</span> {<span class="st">&quot;Other&quot;</span>}})</span>
<span id="cb161-12"><a href="microbiome-community.html#cb161-12" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(tse)<span class="sc">$</span>Phylum <span class="ot">&lt;-</span> <span class="fu">as.character</span>(phylum_renamed)</span>
<span id="cb161-13"><a href="microbiome-community.html#cb161-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb161-14"><a href="microbiome-community.html#cb161-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualizing the composition barplot, with samples order by &quot;Bacteroidetes&quot;</span></span>
<span id="cb161-15"><a href="microbiome-community.html#cb161-15" aria-hidden="true" tabindex="-1"></a><span class="fu">plotAbundance</span>(tse, <span class="at">abund_values=</span><span class="st">&quot;relabundance&quot;</span>, <span class="at">rank =</span> <span class="st">&quot;Phylum&quot;</span>,</span>
<span id="cb161-16"><a href="microbiome-community.html#cb161-16" aria-hidden="true" tabindex="-1"></a>              <span class="at">order_rank_by=</span><span class="st">&quot;abund&quot;</span>, <span class="at">order_sample_by =</span> <span class="st">&quot;Bacteroidetes&quot;</span>)</span></code></pre></div>
<p><img src="15_microbiome_community_files/figure-html/unnamed-chunk-1-1.png" width="672" /></p>
</div>
</div>
<div id="community-typing" class="section level2" number="6.2">
<h2><span class="header-section-number">6.2</span> Community typing</h2>
<div id="dirichlet-multinomial-mixtures-dmm" class="section level3" number="6.2.1">
<h3><span class="header-section-number">6.2.1</span> Dirichlet Multinomial Mixtures (DMM)</h3>
<p>This section focus on DMM analysis.</p>
<p>One technique that allows to search for groups of samples that are
similar to each other is the <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0030126">Dirichlet-Multinomial Mixture
Model</a>. In
DMM, we first determine the number of clusters (k) that best fit the
data (model evidence) using Laplace approximation. After fitting the
model with k clusters, we obtain for each sample k probabilities that
reflect the probability that a sample belongs to the given cluster.</p>
<p>Let’s cluster the data with DMM clustering.</p>
<div class="sourceCode" id="cb162"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb162-1"><a href="microbiome-community.html#cb162-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Runs model and calculates the most likely number of clusters from 1 to 7.</span></span>
<span id="cb162-2"><a href="microbiome-community.html#cb162-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Since this is a large dataset it takes long computational time.</span></span>
<span id="cb162-3"><a href="microbiome-community.html#cb162-3" aria-hidden="true" tabindex="-1"></a><span class="co"># For this reason we use only a subset of the data; agglomerated by Phylum as a rank.</span></span>
<span id="cb162-4"><a href="microbiome-community.html#cb162-4" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> GlobalPatterns</span>
<span id="cb162-5"><a href="microbiome-community.html#cb162-5" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">agglomerateByRank</span>(tse, <span class="at">rank =</span> <span class="st">&quot;Phylum&quot;</span>, <span class="at">agglomerateTree=</span><span class="cn">TRUE</span>)</span>
<span id="cb162-6"><a href="microbiome-community.html#cb162-6" aria-hidden="true" tabindex="-1"></a>tse_dmn <span class="ot">&lt;-</span> <span class="fu">runDMN</span>(tse, <span class="at">name =</span> <span class="st">&quot;DMN&quot;</span>, <span class="at">k =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">7</span>)</span></code></pre></div>
<div class="sourceCode" id="cb163"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb163-1"><a href="microbiome-community.html#cb163-1" aria-hidden="true" tabindex="-1"></a><span class="co"># It is stored in metadata</span></span>
<span id="cb163-2"><a href="microbiome-community.html#cb163-2" aria-hidden="true" tabindex="-1"></a>tse_dmn</span></code></pre></div>
<pre><code>## class: TreeSummarizedExperiment 
## dim: 67 26 
## metadata(1): DMN
## assays(1): counts
## rownames(67): Phylum:Crenarchaeota Phylum:Euryarchaeota ...
##   Phylum:Synergistetes Phylum:SR1
## rowData names(7): Kingdom Phylum ... Genus Species
## colnames(26): CL3 CC1 ... Even2 Even3
## colData names(7): X.SampleID Primer ... SampleType Description
## reducedDimNames(0):
## mainExpName: NULL
## altExpNames(0):
## rowLinks: a LinkDataFrame (67 rows)
## rowTree: 1 phylo tree(s) (66 leaves)
## colLinks: NULL
## colTree: NULL</code></pre>
<p>Return information on metadata that the object contains.</p>
<div class="sourceCode" id="cb165"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb165-1"><a href="microbiome-community.html#cb165-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(<span class="fu">metadata</span>(tse_dmn))</span></code></pre></div>
<pre><code>## [1] &quot;DMN&quot;</code></pre>
<p>This returns a list of DMN objects for a closer investigation.</p>
<div class="sourceCode" id="cb167"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb167-1"><a href="microbiome-community.html#cb167-1" aria-hidden="true" tabindex="-1"></a><span class="fu">getDMN</span>(tse_dmn)</span></code></pre></div>
<pre><code>## [[1]]
## class: DMN 
## k: 1 
## samples x taxa: 26 x 67 
## Laplace: 7715 BIC: 7802 AIC: 7760 
## 
## [[2]]
## class: DMN 
## k: 2 
## samples x taxa: 26 x 67 
## Laplace: 7673 BIC: 7927 AIC: 7842 
## 
## [[3]]
## class: DMN 
## k: 3 
## samples x taxa: 26 x 67 
## Laplace: 7690 BIC: 8076 AIC: 7948 
## 
## [[4]]
## class: DMN 
## k: 4 
## samples x taxa: 26 x 67 
## Laplace: 7741 BIC: 8282 AIC: 8112 
## 
## [[5]]
## class: DMN 
## k: 5 
## samples x taxa: 26 x 67 
## Laplace: 7859 BIC: 8604 AIC: 8391 
## 
## [[6]]
## class: DMN 
## k: 6 
## samples x taxa: 26 x 67 
## Laplace: 7953 BIC: 8850 AIC: 8594 
## 
## [[7]]
## class: DMN 
## k: 7 
## samples x taxa: 26 x 67 
## Laplace: 8064 BIC: 9086 AIC: 8787</code></pre>
<p>Show Laplace approximation (model evidence) for each model of the k models.</p>
<div class="sourceCode" id="cb169"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb169-1"><a href="microbiome-community.html#cb169-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(miaViz)</span>
<span id="cb169-2"><a href="microbiome-community.html#cb169-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDMNFit</span>(tse_dmn, <span class="at">type =</span> <span class="st">&quot;laplace&quot;</span>)</span></code></pre></div>
<p><img src="15_microbiome_community_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
<p>Return the model that has the best fit.</p>
<div class="sourceCode" id="cb170"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb170-1"><a href="microbiome-community.html#cb170-1" aria-hidden="true" tabindex="-1"></a><span class="fu">getBestDMNFit</span>(tse_dmn, <span class="at">type =</span> <span class="st">&quot;laplace&quot;</span>)</span></code></pre></div>
<pre><code>## class: DMN 
## k: 2 
## samples x taxa: 26 x 67 
## Laplace: 7673 BIC: 7927 AIC: 7842</code></pre>
</div>
<div id="pcoa-for-asv-level-data-with-bray-curtis-with-dmm-clusters-shown-with-colors" class="section level3" number="6.2.2">
<h3><span class="header-section-number">6.2.2</span> PCoA for ASV-level data with Bray-Curtis; with DMM clusters shown with colors</h3>
<p>Group samples and return DMNGroup object that contains a summary.
Patient status is used for grouping.</p>
<div class="sourceCode" id="cb172"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb172-1"><a href="microbiome-community.html#cb172-1" aria-hidden="true" tabindex="-1"></a>dmn_group <span class="ot">&lt;-</span> <span class="fu">calculateDMNgroup</span>(tse_dmn, <span class="at">variable =</span> <span class="st">&quot;SampleType&quot;</span>,  <span class="at">exprs_values =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb172-2"><a href="microbiome-community.html#cb172-2" aria-hidden="true" tabindex="-1"></a>                               <span class="at">k =</span> <span class="dv">2</span>, <span class="at">seed=</span>.Machine<span class="sc">$</span>integer.max)</span>
<span id="cb172-3"><a href="microbiome-community.html#cb172-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb172-4"><a href="microbiome-community.html#cb172-4" aria-hidden="true" tabindex="-1"></a>dmn_group</span></code></pre></div>
<pre><code>## class: DMNGroup 
## summary:
##                    k samples taxa    NLE  LogDet Laplace    BIC  AIC
## Feces              2       4   67 1078.3 -106.19   901.1 1171.9 1213
## Freshwater         2       2   67  889.6  -97.28   716.9  936.4 1025
## Freshwater (creek) 2       3   67 1600.3  860.08  1906.3 1674.5 1735
## Mock               2       3   67 1008.4  -55.37   856.6 1082.5 1143
## Ocean              2       3   67 1096.7  -56.21   944.6 1170.9 1232
## Sediment (estuary) 2       3   67 1195.5   18.63  1080.8 1269.7 1331
## Skin               2       3   67  992.6  -84.81   826.2 1066.8 1128
## Soil               2       3   67 1380.3   11.21  1261.8 1454.5 1515
## Tongue             2       2   67  783.0 -107.74   605.1  829.8  918</code></pre>
<p>Mixture weights (rough measure of the cluster size).</p>
<div class="sourceCode" id="cb174"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb174-1"><a href="microbiome-community.html#cb174-1" aria-hidden="true" tabindex="-1"></a>DirichletMultinomial<span class="sc">::</span><span class="fu">mixturewt</span>(<span class="fu">getBestDMNFit</span>(tse_dmn))</span></code></pre></div>
<pre><code>##       pi theta
## 1 0.5385 20.58
## 2 0.4615 15.32</code></pre>
<p>Samples-cluster assignment probabilities / how probable it is that sample belongs
to each cluster</p>
<div class="sourceCode" id="cb176"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb176-1"><a href="microbiome-community.html#cb176-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DirichletMultinomial<span class="sc">::</span><span class="fu">mixture</span>(<span class="fu">getBestDMNFit</span>(tse_dmn)))</span></code></pre></div>
<pre><code>##              [,1]      [,2]
## CL3     1.000e+00 4.479e-17
## CC1     1.000e+00 3.380e-22
## SV1     1.000e+00 1.707e-12
## M31Fcsw 7.424e-26 1.000e+00
## M11Fcsw 1.092e-16 1.000e+00
## M31Plmr 1.150e-13 1.000e+00</code></pre>
<p>Contribution of each taxa to each component</p>
<div class="sourceCode" id="cb178"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb178-1"><a href="microbiome-community.html#cb178-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DirichletMultinomial<span class="sc">::</span><span class="fu">fitted</span>(<span class="fu">getBestDMNFit</span>(tse_dmn)))</span></code></pre></div>
<pre><code>##                          [,1]      [,2]
## Phylum:Crenarchaeota  0.30382 0.1354082
## Phylum:Euryarchaeota  0.23115 0.1468929
## Phylum:Actinobacteria 1.21356 1.0581486
## Phylum:Spirochaetes   0.21393 0.1318099
## Phylum:MVP-15         0.02982 0.0007646
## Phylum:Proteobacteria 6.84509 1.8115328</code></pre>
<p>Get the assignment probabilities</p>
<div class="sourceCode" id="cb180"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb180-1"><a href="microbiome-community.html#cb180-1" aria-hidden="true" tabindex="-1"></a>prob <span class="ot">&lt;-</span> DirichletMultinomial<span class="sc">::</span><span class="fu">mixture</span>(<span class="fu">getBestDMNFit</span>(tse_dmn))</span>
<span id="cb180-2"><a href="microbiome-community.html#cb180-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Add column names</span></span>
<span id="cb180-3"><a href="microbiome-community.html#cb180-3" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(prob) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;comp1&quot;</span>, <span class="st">&quot;comp2&quot;</span>)</span>
<span id="cb180-4"><a href="microbiome-community.html#cb180-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb180-5"><a href="microbiome-community.html#cb180-5" aria-hidden="true" tabindex="-1"></a><span class="co"># For each row, finds column that has the highest value. Then extract the column </span></span>
<span id="cb180-6"><a href="microbiome-community.html#cb180-6" aria-hidden="true" tabindex="-1"></a><span class="co"># names of highest values.</span></span>
<span id="cb180-7"><a href="microbiome-community.html#cb180-7" aria-hidden="true" tabindex="-1"></a>vec <span class="ot">&lt;-</span> <span class="fu">colnames</span>(prob)[<span class="fu">max.col</span>(prob,<span class="at">ties.method =</span> <span class="st">&quot;first&quot;</span>)]</span></code></pre></div>
<p>Computing the euclidean PCoA and storing it as a dataframe</p>
<div class="sourceCode" id="cb181"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb181-1"><a href="microbiome-community.html#cb181-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Does clr transformation. Pseudocount is added, because data contains zeros.</span></span>
<span id="cb181-2"><a href="microbiome-community.html#cb181-2" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">transformCounts</span>(tse, <span class="at">method =</span> <span class="st">&quot;clr&quot;</span>, <span class="at">pseudocount =</span> <span class="dv">1</span>)</span>
<span id="cb181-3"><a href="microbiome-community.html#cb181-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-4"><a href="microbiome-community.html#cb181-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Gets clr table</span></span>
<span id="cb181-5"><a href="microbiome-community.html#cb181-5" aria-hidden="true" tabindex="-1"></a>clr_assay <span class="ot">&lt;-</span> <span class="fu">assays</span>(tse)<span class="sc">$</span>clr</span>
<span id="cb181-6"><a href="microbiome-community.html#cb181-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-7"><a href="microbiome-community.html#cb181-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Transposes it to get taxa to columns</span></span>
<span id="cb181-8"><a href="microbiome-community.html#cb181-8" aria-hidden="true" tabindex="-1"></a>clr_assay <span class="ot">&lt;-</span> <span class="fu">t</span>(clr_assay)</span>
<span id="cb181-9"><a href="microbiome-community.html#cb181-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-10"><a href="microbiome-community.html#cb181-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculates Euclidean distances between samples. Because taxa is in columns,</span></span>
<span id="cb181-11"><a href="microbiome-community.html#cb181-11" aria-hidden="true" tabindex="-1"></a><span class="co"># it is used to compare different samples.</span></span>
<span id="cb181-12"><a href="microbiome-community.html#cb181-12" aria-hidden="true" tabindex="-1"></a>euclidean_dist <span class="ot">&lt;-</span> vegan<span class="sc">::</span><span class="fu">vegdist</span>(clr_assay, <span class="at">method =</span> <span class="st">&quot;euclidean&quot;</span>)</span>
<span id="cb181-13"><a href="microbiome-community.html#cb181-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-14"><a href="microbiome-community.html#cb181-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Does principal coordinate analysis</span></span>
<span id="cb181-15"><a href="microbiome-community.html#cb181-15" aria-hidden="true" tabindex="-1"></a>euclidean_pcoa <span class="ot">&lt;-</span> ecodist<span class="sc">::</span><span class="fu">pco</span>(euclidean_dist)</span>
<span id="cb181-16"><a href="microbiome-community.html#cb181-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-17"><a href="microbiome-community.html#cb181-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates a data frame from principal coordinates</span></span>
<span id="cb181-18"><a href="microbiome-community.html#cb181-18" aria-hidden="true" tabindex="-1"></a>euclidean_pcoa_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">pcoa1 =</span> euclidean_pcoa<span class="sc">$</span>vectors[,<span class="dv">1</span>], </span>
<span id="cb181-19"><a href="microbiome-community.html#cb181-19" aria-hidden="true" tabindex="-1"></a>                                <span class="at">pcoa2 =</span> euclidean_pcoa<span class="sc">$</span>vectors[,<span class="dv">2</span>])</span></code></pre></div>
<div class="sourceCode" id="cb182"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb182-1"><a href="microbiome-community.html#cb182-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates a data frame that contains principal coordinates and DMM information</span></span>
<span id="cb182-2"><a href="microbiome-community.html#cb182-2" aria-hidden="true" tabindex="-1"></a>euclidean_dmm_pcoa_df <span class="ot">&lt;-</span> <span class="fu">cbind</span>(euclidean_pcoa_df,</span>
<span id="cb182-3"><a href="microbiome-community.html#cb182-3" aria-hidden="true" tabindex="-1"></a>                               <span class="at">dmm_component =</span> vec)</span>
<span id="cb182-4"><a href="microbiome-community.html#cb182-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates a plot</span></span>
<span id="cb182-5"><a href="microbiome-community.html#cb182-5" aria-hidden="true" tabindex="-1"></a>euclidean_dmm_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> euclidean_dmm_pcoa_df, </span>
<span id="cb182-6"><a href="microbiome-community.html#cb182-6" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">aes</span>(<span class="at">x=</span>pcoa1, <span class="at">y=</span>pcoa2,</span>
<span id="cb182-7"><a href="microbiome-community.html#cb182-7" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">color =</span> dmm_component)) <span class="sc">+</span></span>
<span id="cb182-8"><a href="microbiome-community.html#cb182-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb182-9"><a href="microbiome-community.html#cb182-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Coordinate 1&quot;</span>,</span>
<span id="cb182-10"><a href="microbiome-community.html#cb182-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">&quot;Coordinate 2&quot;</span>,</span>
<span id="cb182-11"><a href="microbiome-community.html#cb182-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">&quot;PCoA with Aitchison distances&quot;</span>) <span class="sc">+</span>  </span>
<span id="cb182-12"><a href="microbiome-community.html#cb182-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>)) <span class="co"># makes titles smaller</span></span>
<span id="cb182-13"><a href="microbiome-community.html#cb182-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-14"><a href="microbiome-community.html#cb182-14" aria-hidden="true" tabindex="-1"></a>euclidean_dmm_plot</span></code></pre></div>
<p><img src="15_microbiome_community_files/figure-html/unnamed-chunk-13-1.png" width="672" /></p>
</div>
</div>
<div id="session-info-6" class="section level2 unnumbered">
<h2>Session Info</h2>
<button class="rebook-collapse">
View session info
</button>
<div class="rebook-content">
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.2 LTS

Matrix products: default
BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] miaViz_1.1.2                   ggraph_2.0.5                  
 [3] ggplot2_3.3.5                  mia_1.1.7                     
 [5] TreeSummarizedExperiment_2.1.3 Biostrings_2.61.1             
 [7] XVector_0.33.0                 SingleCellExperiment_1.15.1   
 [9] SummarizedExperiment_1.23.1    Biobase_2.53.0                
[11] GenomicRanges_1.45.0           GenomeInfoDb_1.29.3           
[13] IRanges_2.27.0                 S4Vectors_0.31.0              
[15] BiocGenerics_0.39.1            MatrixGenerics_1.5.1          
[17] matrixStats_0.59.0             ecodist_2.0.7                 
[19] BiocStyle_2.21.3               rebook_1.3.0                  

loaded via a namespace (and not attached):
  [1] ggtree_3.1.2                ggnewscale_0.4.5           
  [3] ggbeeswarm_0.6.0            colorspace_2.0-2           
  [5] ellipsis_0.3.2              scuttle_1.3.0              
  [7] BiocNeighbors_1.11.0        aplot_0.0.6                
  [9] farver_2.1.0                graphlayouts_0.7.1         
 [11] ggrepel_0.9.1               bit64_4.0.5                
 [13] fansi_0.5.0                 decontam_1.13.0            
 [15] splines_4.1.0               codetools_0.2-18           
 [17] sparseMatrixStats_1.5.0     cachem_1.0.5               
 [19] knitr_1.33                  scater_1.21.2              
 [21] polyclip_1.10-0             jsonlite_1.7.2             
 [23] cluster_2.1.2               graph_1.71.2               
 [25] ggforce_0.3.3               BiocManager_1.30.16        
 [27] compiler_4.1.0              rvcheck_0.1.8              
 [29] assertthat_0.2.1            Matrix_1.3-4               
 [31] fastmap_1.1.0               lazyeval_0.2.2             
 [33] tweenr_1.0.2                BiocSingular_1.9.1         
 [35] htmltools_0.5.1.1           tools_4.1.0                
 [37] igraph_1.2.6                rsvd_1.0.5                 
 [39] gtable_0.3.0                glue_1.4.2                 
 [41] GenomeInfoDbData_1.2.6      reshape2_1.4.4             
 [43] dplyr_1.0.7                 Rcpp_1.0.7                 
 [45] jquerylib_0.1.4             vctrs_0.3.8                
 [47] ape_5.5                     nlme_3.1-152               
 [49] DECIPHER_2.21.0             DelayedMatrixStats_1.15.0  
 [51] xfun_0.24                   stringr_1.4.0              
 [53] beachmat_2.9.0              lifecycle_1.0.0            
 [55] irlba_2.3.3                 XML_3.99-0.6               
 [57] zlibbioc_1.39.0             MASS_7.3-54                
 [59] scales_1.1.1                tidygraph_1.2.0            
 [61] parallel_4.1.0              yaml_2.2.1                 
 [63] memoise_2.0.0               gridExtra_2.3              
 [65] sass_0.4.0                  stringi_1.7.3              
 [67] RSQLite_2.2.7               highr_0.9                  
 [69] ScaledMatrix_1.1.0          permute_0.9-5              
 [71] tidytree_0.3.4              filelock_1.0.2             
 [73] BiocParallel_1.27.2         rlang_0.4.11               
 [75] pkgconfig_2.0.3             bitops_1.0-7               
 [77] evaluate_0.14               lattice_0.20-44            
 [79] purrr_0.3.4                 labeling_0.4.2             
 [81] patchwork_1.1.1             treeio_1.17.2              
 [83] CodeDepends_0.6.5           bit_4.0.4                  
 [85] tidyselect_1.1.1            plyr_1.8.6                 
 [87] magrittr_2.0.1              bookdown_0.22              
 [89] R6_2.5.0                    generics_0.1.0             
 [91] DelayedArray_0.19.1         DBI_1.1.1                  
 [93] withr_2.4.2                 mgcv_1.8-36                
 [95] pillar_1.6.1                RCurl_1.98-1.3             
 [97] tibble_3.1.2                dir.expiry_1.1.0           
 [99] crayon_1.4.1                utf8_1.2.1                 
[101] rmarkdown_2.9               viridis_0.6.1              
[103] grid_4.1.0                  blob_1.2.1                 
[105] vegan_2.5-7                 digest_0.6.27              
[107] tidyr_1.1.3                 munsell_0.5.0              
[109] DirichletMultinomial_1.35.0 beeswarm_0.4.0             
[111] viridisLite_0.4.0           vipor_0.4.5                
[113] bslib_0.2.5.1              </code></pre>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="microbiome-exploration.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="differential-abundance.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["OMA.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
