<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 Community composition | Orchestrating Microbiome Analysis</title>
  <meta name="description" content="Chapter 9 Community composition | Orchestrating Microbiome Analysis" />
  <meta name="generator" content="bookdown 0.33 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 Community composition | Orchestrating Microbiome Analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="microbiome/OMA" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 Community composition | Orchestrating Microbiome Analysis" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="community-similarity.html"/>
<link rel="next" href="clustering.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.1/htmlwidgets.js"></script>
<script src="libs/plotly-binding-4.10.1/plotly.js"></script>
<script src="libs/typedarray-0.1/typedarray.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="libs/plotly-htmlwidgets-css-2.11.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="libs/plotly-main-2.11.1/plotly-latest.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Orchestrating Microbiome Analysis</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="part"><span><b>I Introduction</b></span></li>
<li class="chapter" data-level="1" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="containers.html"><a href="containers.html"><i class="fa fa-check"></i><b>2</b> Microbiome Data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="containers.html"><a href="containers.html#data-science-framework"><i class="fa fa-check"></i><b>2.1</b> Data science framework</a></li>
<li class="chapter" data-level="2.2" data-path="containers.html"><a href="containers.html#data-containers"><i class="fa fa-check"></i><b>2.2</b> Data containers</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="containers.html"><a href="containers.html#assay-data"><i class="fa fa-check"></i><b>2.2.1</b> Assay data</a></li>
<li class="chapter" data-level="2.2.2" data-path="containers.html"><a href="containers.html#coldata"><i class="fa fa-check"></i><b>2.2.2</b> colData</a></li>
<li class="chapter" data-level="2.2.3" data-path="containers.html"><a href="containers.html#rowdata"><i class="fa fa-check"></i><b>2.2.3</b> rowData</a></li>
<li class="chapter" data-level="2.2.4" data-path="containers.html"><a href="containers.html#rowtree"><i class="fa fa-check"></i><b>2.2.4</b> rowTree</a></li>
<li class="chapter" data-level="2.2.5" data-path="containers.html"><a href="containers.html#alt-exp"><i class="fa fa-check"></i><b>2.2.5</b> Alternative experiments</a></li>
<li class="chapter" data-level="2.2.6" data-path="containers.html"><a href="containers.html#mae"><i class="fa fa-check"></i><b>2.2.6</b> MultiAssayExperiments</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="containers.html"><a href="containers.html#loading-experimental-microbiome-data"><i class="fa fa-check"></i><b>2.3</b> Loading experimental microbiome data</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="containers.html"><a href="containers.html#s-workflow"><i class="fa fa-check"></i><b>2.3.1</b> 16S workflow</a></li>
<li class="chapter" data-level="2.3.2" data-path="containers.html"><a href="containers.html#import-from-external-files"><i class="fa fa-check"></i><b>2.3.2</b> Import from external files</a></li>
<li class="chapter" data-level="2.3.3" data-path="containers.html"><a href="containers.html#conversions-between-data-formats-in-r"><i class="fa fa-check"></i><b>2.3.3</b> Conversions between data formats in R</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="containers.html"><a href="containers.html#example-data"><i class="fa fa-check"></i><b>2.4</b> Demonstration data</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="containers.html"><a href="containers.html#package-data"><i class="fa fa-check"></i><b>2.4.1</b> Package data</a></li>
<li class="chapter" data-level="2.4.2" data-path="containers.html"><a href="containers.html#experimenthub-data"><i class="fa fa-check"></i><b>2.4.2</b> ExperimentHub data</a></li>
<li class="chapter" data-level="2.4.3" data-path="containers.html"><a href="containers.html#other-data-sources"><i class="fa fa-check"></i><b>2.4.3</b> Other data sources</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="containers.html"><a href="containers.html#session-info"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="packages.html"><a href="packages.html"><i class="fa fa-check"></i><b>3</b> Packages</a>
<ul>
<li class="chapter" data-level="3.1" data-path="packages.html"><a href="packages.html#package-installation"><i class="fa fa-check"></i><b>3.1</b> Package installation</a></li>
<li class="chapter" data-level="3.2" data-path="packages.html"><a href="packages.html#some-available-packages"><i class="fa fa-check"></i><b>3.2</b> Some available packages</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="packages.html"><a href="packages.html#mia-family-of-methods"><i class="fa fa-check"></i><b>3.2.1</b> mia family of methods</a></li>
<li class="chapter" data-level="3.2.2" data-path="packages.html"><a href="packages.html#tree-based-methods-sub-tree-methods"><i class="fa fa-check"></i><b>3.2.2</b> Tree-based methods {sub-tree-methods}</a></li>
<li class="chapter" data-level="3.2.3" data-path="packages.html"><a href="packages.html#differential-abundance-sub-diff-abund"><i class="fa fa-check"></i><b>3.2.3</b> Differential abundance {sub-diff-abund}</a></li>
<li class="chapter" data-level="3.2.4" data-path="packages.html"><a href="packages.html#manipulation-sub-manipulation"><i class="fa fa-check"></i><b>3.2.4</b> Manipulation {sub-manipulation}</a></li>
<li class="chapter" data-level="3.2.5" data-path="packages.html"><a href="packages.html#data"><i class="fa fa-check"></i><b>3.2.5</b> Data</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="packages.html"><a href="packages.html#session-info-1"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="part"><span><b>II Focus Topics</b></span></li>
<li class="chapter" data-level="4" data-path="datamanipulation.html"><a href="datamanipulation.html"><i class="fa fa-check"></i><b>4</b> Data Manipulation</a>
<ul>
<li class="chapter" data-level="4.1" data-path="datamanipulation.html"><a href="datamanipulation.html#tidying-and-subsetting"><i class="fa fa-check"></i><b>4.1</b> Tidying and subsetting</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="datamanipulation.html"><a href="datamanipulation.html#tidy-data"><i class="fa fa-check"></i><b>4.1.1</b> Tidy data</a></li>
<li class="chapter" data-level="4.1.2" data-path="datamanipulation.html"><a href="datamanipulation.html#subsetting"><i class="fa fa-check"></i><b>4.1.2</b> Subsetting</a></li>
<li class="chapter" data-level="4.1.3" data-path="datamanipulation.html"><a href="datamanipulation.html#splitting"><i class="fa fa-check"></i><b>4.1.3</b> Splitting</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="datamanipulation.html"><a href="datamanipulation.html#merge-data"><i class="fa fa-check"></i><b>4.2</b> Merge data</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="datamanipulation.html"><a href="datamanipulation.html#additional-functions"><i class="fa fa-check"></i><b>4.2.1</b> Additional functions</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>5</b> Exploration and quality Control</a>
<ul>
<li class="chapter" data-level="5.1" data-path="quality-control.html"><a href="quality-control.html#abundance"><i class="fa fa-check"></i><b>5.1</b> Abundance</a></li>
<li class="chapter" data-level="5.2" data-path="quality-control.html"><a href="quality-control.html#prevalence"><i class="fa fa-check"></i><b>5.2</b> Prevalence</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="quality-control.html"><a href="quality-control.html#prevalence-analysis"><i class="fa fa-check"></i><b>5.2.1</b> Prevalence analysis</a></li>
<li class="chapter" data-level="5.2.2" data-path="quality-control.html"><a href="quality-control.html#rare-taxa"><i class="fa fa-check"></i><b>5.2.2</b> Rare taxa</a></li>
<li class="chapter" data-level="5.2.3" data-path="quality-control.html"><a href="quality-control.html#plotting-prevalence"><i class="fa fa-check"></i><b>5.2.3</b> Plotting prevalence</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="quality-control.html"><a href="quality-control.html#qc"><i class="fa fa-check"></i><b>5.3</b> Quality control</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="quality-control.html"><a href="quality-control.html#top-taxa"><i class="fa fa-check"></i><b>5.3.1</b> Top taxa</a></li>
<li class="chapter" data-level="5.3.2" data-path="quality-control.html"><a href="quality-control.html#library-size-read-count"><i class="fa fa-check"></i><b>5.3.2</b> Library size / read count</a></li>
<li class="chapter" data-level="5.3.3" data-path="quality-control.html"><a href="quality-control.html#contaminant-sequences"><i class="fa fa-check"></i><b>5.3.3</b> Contaminant sequences</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="quality-control.html"><a href="quality-control.html#session-info-2"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="taxonomic-information.html"><a href="taxonomic-information.html"><i class="fa fa-check"></i><b>6</b> Taxonomic Information</a>
<ul>
<li class="chapter" data-level="6.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#assigning-taxonomic-information."><i class="fa fa-check"></i><b>6.1</b> Assigning taxonomic information.</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#dada2"><i class="fa fa-check"></i><b>6.1.1</b> dada2</a></li>
<li class="chapter" data-level="6.1.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#decipher"><i class="fa fa-check"></i><b>6.1.2</b> DECIPHER</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#functions-to-access-taxonomic-information"><i class="fa fa-check"></i><b>6.2</b> Functions to access taxonomic information</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#generate-a-taxonomic-tree-on-the-fly"><i class="fa fa-check"></i><b>6.2.1</b> Generate a taxonomic tree on the fly</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="taxonomic-information.html"><a href="taxonomic-information.html#data-agglomeration"><i class="fa fa-check"></i><b>6.3</b> Data agglomeration</a></li>
<li class="chapter" data-level="6.4" data-path="taxonomic-information.html"><a href="taxonomic-information.html#data-transformation"><i class="fa fa-check"></i><b>6.4</b> Data transformation</a></li>
<li class="chapter" data-level="6.5" data-path="taxonomic-information.html"><a href="taxonomic-information.html#pick-specific"><i class="fa fa-check"></i><b>6.5</b> Pick specific</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#abundances-of-all-taxa-in-specific-sample"><i class="fa fa-check"></i><b>6.5.1</b> Abundances of all taxa in specific sample</a></li>
<li class="chapter" data-level="6.5.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#abundances-of-specific-taxa-in-all-samples"><i class="fa fa-check"></i><b>6.5.2</b> Abundances of specific taxa in all samples</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="taxonomic-information.html"><a href="taxonomic-information.html#session-info-3"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="community-diversity.html"><a href="community-diversity.html"><i class="fa fa-check"></i><b>7</b> Community diversity</a>
<ul>
<li class="chapter" data-level="7.1" data-path="community-diversity.html"><a href="community-diversity.html#estimation"><i class="fa fa-check"></i><b>7.1</b> Estimation</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="community-diversity.html"><a href="community-diversity.html#richness"><i class="fa fa-check"></i><b>7.1.1</b> Richness</a></li>
<li class="chapter" data-level="7.1.2" data-path="community-diversity.html"><a href="community-diversity.html#estimate-diversity"><i class="fa fa-check"></i><b>7.1.2</b> Diversity</a></li>
<li class="chapter" data-level="7.1.3" data-path="community-diversity.html"><a href="community-diversity.html#faith-diversity"><i class="fa fa-check"></i><b>7.1.3</b> Faith phylogenetic diversity</a></li>
<li class="chapter" data-level="7.1.4" data-path="community-diversity.html"><a href="community-diversity.html#evenness"><i class="fa fa-check"></i><b>7.1.4</b> Evenness</a></li>
<li class="chapter" data-level="7.1.5" data-path="community-diversity.html"><a href="community-diversity.html#dominance"><i class="fa fa-check"></i><b>7.1.5</b> Dominance</a></li>
<li class="chapter" data-level="7.1.6" data-path="community-diversity.html"><a href="community-diversity.html#rarity"><i class="fa fa-check"></i><b>7.1.6</b> Rarity</a></li>
<li class="chapter" data-level="7.1.7" data-path="community-diversity.html"><a href="community-diversity.html#divergence"><i class="fa fa-check"></i><b>7.1.7</b> Divergence</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="community-diversity.html"><a href="community-diversity.html#visualization"><i class="fa fa-check"></i><b>7.2</b> Visualization</a></li>
<li class="chapter" data-level="" data-path="community-diversity.html"><a href="community-diversity.html#session-info-4"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="community-similarity.html"><a href="community-similarity.html"><i class="fa fa-check"></i><b>8</b> Community similarity</a>
<ul>
<li class="chapter" data-level="8.1" data-path="community-similarity.html"><a href="community-similarity.html#explained-variance"><i class="fa fa-check"></i><b>8.1</b> Explained variance</a></li>
<li class="chapter" data-level="8.2" data-path="community-similarity.html"><a href="community-similarity.html#community-comparisons-by-beta-diversity-analysis"><i class="fa fa-check"></i><b>8.2</b> Community comparisons by beta diversity analysis</a></li>
<li class="chapter" data-level="8.3" data-path="community-similarity.html"><a href="community-similarity.html#other-ordination-methods"><i class="fa fa-check"></i><b>8.3</b> Other ordination methods</a></li>
<li class="chapter" data-level="8.4" data-path="community-similarity.html"><a href="community-similarity.html#pcoa-genus"><i class="fa fa-check"></i><b>8.4</b> Visualizing the most dominant genus on PCoA</a>
<ul>
<li class="chapter" data-level="8.4.1" data-path="community-similarity.html"><a href="community-similarity.html#testing-differences-in-community-composition-between-sample-groups"><i class="fa fa-check"></i><b>8.4.1</b> Testing differences in community composition between sample groups</a></li>
<li class="chapter" data-level="8.4.2" data-path="community-similarity.html"><a href="community-similarity.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>8.4.2</b> Checking the homogeneity condition</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="community-similarity.html"><a href="community-similarity.html#further-reading"><i class="fa fa-check"></i><b>8.5</b> Further reading</a></li>
<li class="chapter" data-level="" data-path="community-similarity.html"><a href="community-similarity.html#session-info-5"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="microbiome-community.html"><a href="microbiome-community.html"><i class="fa fa-check"></i><b>9</b> Community composition</a>
<ul>
<li class="chapter" data-level="9.1" data-path="microbiome-community.html"><a href="microbiome-community.html#visual-composition"><i class="fa fa-check"></i><b>9.1</b> Visualizing taxonomic composition</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="microbiome-community.html"><a href="microbiome-community.html#composition-barplot"><i class="fa fa-check"></i><b>9.1.1</b> Composition barplot</a></li>
<li class="chapter" data-level="9.1.2" data-path="microbiome-community.html"><a href="microbiome-community.html#composition-heatmap"><i class="fa fa-check"></i><b>9.1.2</b> Composition heatmap</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="microbiome-community.html"><a href="microbiome-community.html#session-info-6"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="clustering.html"><a href="clustering.html"><i class="fa fa-check"></i><b>10</b> Community typing (clustering)</a>
<ul>
<li class="chapter" data-level="10.1" data-path="clustering.html"><a href="clustering.html#hiearchical-clustering"><i class="fa fa-check"></i><b>10.1</b> Hiearchical clustering</a></li>
<li class="chapter" data-level="10.2" data-path="clustering.html"><a href="clustering.html#k-means-clustering"><i class="fa fa-check"></i><b>10.2</b> K-means clustering</a></li>
<li class="chapter" data-level="10.3" data-path="clustering.html"><a href="clustering.html#dirichlet-multinomial-mixtures-dmm"><i class="fa fa-check"></i><b>10.3</b> Dirichlet Multinomial Mixtures (DMM)</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="clustering.html"><a href="clustering.html#pcoa-for-asv-level-data-with-bray-curtis-with-dmm-clusters-shown-with-colors"><i class="fa fa-check"></i><b>10.3.1</b> PCoA for ASV-level data with Bray-Curtis; with DMM clusters shown with colors</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="clustering.html"><a href="clustering.html#community-detection"><i class="fa fa-check"></i><b>10.4</b> Community Detection</a></li>
<li class="chapter" data-level="10.5" data-path="clustering.html"><a href="clustering.html#biclustering"><i class="fa fa-check"></i><b>10.5</b> Biclustering</a>
<ul>
<li class="chapter" data-level="10.5.1" data-path="clustering.html"><a href="clustering.html#taxa-vs-samples"><i class="fa fa-check"></i><b>10.5.1</b> Taxa vs samples</a></li>
<li class="chapter" data-level="10.5.2" data-path="clustering.html"><a href="clustering.html#taxa-vs-biomolecules"><i class="fa fa-check"></i><b>10.5.2</b> Taxa vs biomolecules</a></li>
<li class="chapter" data-level="10.5.3" data-path="clustering.html"><a href="clustering.html#taxa-vs-taxa"><i class="fa fa-check"></i><b>10.5.3</b> Taxa vs taxa</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="clustering.html"><a href="clustering.html#additional-community-typing"><i class="fa fa-check"></i><b>10.6</b> Additional Community Typing</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="differential-abundance.html"><a href="differential-abundance.html"><i class="fa fa-check"></i><b>11</b> Differential abundance</a>
<ul>
<li class="chapter" data-level="11.1" data-path="differential-abundance.html"><a href="differential-abundance.html#differential-abundance-analysis"><i class="fa fa-check"></i><b>11.1</b> Differential abundance analysis</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="differential-abundance.html"><a href="differential-abundance.html#examples-and-tools"><i class="fa fa-check"></i><b>11.1.1</b> Examples and tools</a></li>
<li class="chapter" data-level="11.1.2" data-path="differential-abundance.html"><a href="differential-abundance.html#prevalence-filtering"><i class="fa fa-check"></i><b>11.1.2</b> Prevalence Filtering</a></li>
<li class="chapter" data-level="11.1.3" data-path="differential-abundance.html"><a href="differential-abundance.html#aldex2"><i class="fa fa-check"></i><b>11.1.3</b> ALDEx2</a></li>
<li class="chapter" data-level="11.1.4" data-path="differential-abundance.html"><a href="differential-abundance.html#ancom-bc"><i class="fa fa-check"></i><b>11.1.4</b> ANCOM-BC</a></li>
<li class="chapter" data-level="11.1.5" data-path="differential-abundance.html"><a href="differential-abundance.html#maaslin2"><i class="fa fa-check"></i><b>11.1.5</b> MaAsLin2</a></li>
<li class="chapter" data-level="11.1.6" data-path="differential-abundance.html"><a href="differential-abundance.html#linda"><i class="fa fa-check"></i><b>11.1.6</b> LinDA</a></li>
<li class="chapter" data-level="11.1.7" data-path="differential-abundance.html"><a href="differential-abundance.html#comparison-of-the-methods"><i class="fa fa-check"></i><b>11.1.7</b> Comparison of the methods</a></li>
<li class="chapter" data-level="11.1.8" data-path="differential-abundance.html"><a href="differential-abundance.html#confounding-variables"><i class="fa fa-check"></i><b>11.1.8</b> Confounding variables</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="differential-abundance.html"><a href="differential-abundance.html#tree-based-methods"><i class="fa fa-check"></i><b>11.2</b> Tree-based methods</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="differential-abundance.html"><a href="differential-abundance.html#group-wise-associations-testing-based-on-balances"><i class="fa fa-check"></i><b>11.2.1</b> Group-wise associations testing based on balances</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="differential-abundance.html"><a href="differential-abundance.html#session-info-7"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="machine_learning.html"><a href="machine_learning.html"><i class="fa fa-check"></i><b>12</b> Machine learning</a>
<ul>
<li class="chapter" data-level="12.1" data-path="machine_learning.html"><a href="machine_learning.html#supervised-machine-learning"><i class="fa fa-check"></i><b>12.1</b> Supervised machine learning</a></li>
<li class="chapter" data-level="12.2" data-path="machine_learning.html"><a href="machine_learning.html#unsupervised-machine-learning"><i class="fa fa-check"></i><b>12.2</b> Unsupervised machine learning</a></li>
<li class="chapter" data-level="" data-path="machine_learning.html"><a href="machine_learning.html#session-info-8"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="multi-assay-analyses.html"><a href="multi-assay-analyses.html"><i class="fa fa-check"></i><b>13</b> Multi-assay analyses</a>
<ul>
<li class="chapter" data-level="13.1" data-path="multi-assay-analyses.html"><a href="multi-assay-analyses.html#cross-correlation-analysis"><i class="fa fa-check"></i><b>13.1</b> Cross-correlation Analysis</a></li>
<li class="chapter" data-level="13.2" data-path="multi-assay-analyses.html"><a href="multi-assay-analyses.html#multi-omics-factor-analysis"><i class="fa fa-check"></i><b>13.2</b> Multi-Omics Factor Analysis</a></li>
<li class="chapter" data-level="" data-path="multi-assay-analyses.html"><a href="multi-assay-analyses.html#session-info-9"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="viz-chapter.html"><a href="viz-chapter.html"><i class="fa fa-check"></i><b>14</b> Visualization</a>
<ul>
<li class="chapter" data-level="14.1" data-path="viz-chapter.html"><a href="viz-chapter.html#pre-analysis-exploration"><i class="fa fa-check"></i><b>14.1</b> Pre-analysis exploration</a>
<ul>
<li class="chapter" data-level="14.1.1" data-path="viz-chapter.html"><a href="viz-chapter.html#accessing-row-and-column-data"><i class="fa fa-check"></i><b>14.1.1</b> Accessing row and column data</a></li>
<li class="chapter" data-level="14.1.2" data-path="viz-chapter.html"><a href="viz-chapter.html#viewing-abundance-and-prevalence-patterns"><i class="fa fa-check"></i><b>14.1.2</b> Viewing abundance and prevalence patterns</a></li>
</ul></li>
<li class="chapter" data-level="14.2" data-path="viz-chapter.html"><a href="viz-chapter.html#diversity-estimation"><i class="fa fa-check"></i><b>14.2</b> Diversity estimation</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="viz-chapter.html"><a href="viz-chapter.html#alpha-diversity-with-scatter-violin-and-box-plots"><i class="fa fa-check"></i><b>14.2.1</b> Alpha diversity with scatter, violin and box plots</a></li>
<li class="chapter" data-level="14.2.2" data-path="viz-chapter.html"><a href="viz-chapter.html#beta-diversity-with-shepard-and-coordination-plots"><i class="fa fa-check"></i><b>14.2.2</b> Beta diversity with Shepard and coordination plots</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="viz-chapter.html"><a href="viz-chapter.html#statistical-analysis"><i class="fa fa-check"></i><b>14.3</b> Statistical analysis</a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="viz-chapter.html"><a href="viz-chapter.html#heatmaps"><i class="fa fa-check"></i><b>14.3.1</b> Heatmaps</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="viz-chapter.html"><a href="viz-chapter.html#session-info-10"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="part"><span><b>III Appendix</b></span></li>
<li class="chapter" data-level="15" data-path="training.html"><a href="training.html"><i class="fa fa-check"></i><b>15</b> Training</a>
<ul>
<li class="chapter" data-level="15.1" data-path="training.html"><a href="training.html#coc"><i class="fa fa-check"></i><b>15.1</b> Code of Conduct</a></li>
<li class="chapter" data-level="15.2" data-path="training.html"><a href="training.html#checklist"><i class="fa fa-check"></i><b>15.2</b> Checklist</a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="training.html"><a href="training.html#setting-up-your-own-computer"><i class="fa fa-check"></i><b>15.2.1</b> Setting up Your own computer</a></li>
<li class="chapter" data-level="15.2.2" data-path="training.html"><a href="training.html#support-and-resources"><i class="fa fa-check"></i><b>15.2.2</b> Support and resources</a></li>
<li class="chapter" data-level="15.2.3" data-path="training.html"><a href="training.html#material"><i class="fa fa-check"></i><b>15.2.3</b> Study material</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="packages.html"><a href="packages.html#packages"><i class="fa fa-check"></i><b>15.3</b> Installing and loading the required R packages</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="resources.html"><a href="resources.html"><i class="fa fa-check"></i><b>16</b> Resources</a>
<ul>
<li class="chapter" data-level="16.1" data-path="resources.html"><a href="resources.html#data-containers-1"><i class="fa fa-check"></i><b>16.1</b> Data containers</a>
<ul>
<li class="chapter" data-level="16.1.1" data-path="resources.html"><a href="resources.html#resources-for-treesummarizedexperiment"><i class="fa fa-check"></i><b>16.1.1</b> Resources for TreeSummarizedExperiment</a></li>
<li class="chapter" data-level="16.1.2" data-path="resources.html"><a href="resources.html#other-relevant-containers"><i class="fa fa-check"></i><b>16.1.2</b> Other relevant containers</a></li>
<li class="chapter" data-level="16.1.3" data-path="resources.html"><a href="resources.html#alternative-containers-for-microbiome-data"><i class="fa fa-check"></i><b>16.1.3</b> Alternative containers for microbiome data</a></li>
<li class="chapter" data-level="16.1.4" data-path="resources.html"><a href="resources.html#resources-for-phyloseq"><i class="fa fa-check"></i><b>16.1.4</b> Resources for phyloseq</a></li>
</ul></li>
<li class="chapter" data-level="16.2" data-path="resources.html"><a href="resources.html#r-programming-resources"><i class="fa fa-check"></i><b>16.2</b> R programming resources</a>
<ul>
<li class="chapter" data-level="16.2.1" data-path="resources.html"><a href="resources.html#bioc_intro"><i class="fa fa-check"></i><b>16.2.1</b> Bioconductor Classes</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="resources.html"><a href="resources.html#reproducible-reporting-with-quarto"><i class="fa fa-check"></i><b>16.3</b> Reproducible reporting with Quarto</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="extras.html"><a href="extras.html"><i class="fa fa-check"></i><b>17</b> Extra material</a>
<ul>
<li class="chapter" data-level="17.1" data-path="extras.html"><a href="extras.html#permanova-comparison"><i class="fa fa-check"></i><b>17.1</b> PERMANOVA comparison</a></li>
<li class="chapter" data-level="17.2" data-path="extras.html"><a href="extras.html#bayesian-multinomial-logistic-normal-models"><i class="fa fa-check"></i><b>17.2</b> Bayesian Multinomial Logistic-Normal Models</a></li>
<li class="chapter" data-level="17.3" data-path="extras.html"><a href="extras.html#interactive-3d-plots"><i class="fa fa-check"></i><b>17.3</b> Interactive 3D Plots</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="acknowledgments.html"><a href="acknowledgments.html"><i class="fa fa-check"></i><b>18</b> Acknowledgments</a></li>
<li class="chapter" data-level="19" data-path="exercises.html"><a href="exercises.html"><i class="fa fa-check"></i><b>19</b> Exercises</a>
<ul>
<li class="chapter" data-level="19.1" data-path="exercises.html"><a href="exercises.html#workflows"><i class="fa fa-check"></i><b>19.1</b> Workflows</a>
<ul>
<li class="chapter" data-level="19.1.1" data-path="exercises.html"><a href="exercises.html#reproducible-reporting"><i class="fa fa-check"></i><b>19.1.1</b> Reproducible reporting</a></li>
</ul></li>
<li class="chapter" data-level="19.2" data-path="exercises.html"><a href="exercises.html#treesummarizedexperiment-basic-components"><i class="fa fa-check"></i><b>19.2</b> TreeSummarizedExperiment: basic components</a>
<ul>
<li class="chapter" data-level="19.2.1" data-path="exercises.html"><a href="exercises.html#basic-summaries"><i class="fa fa-check"></i><b>19.2.1</b> Basic summaries</a></li>
<li class="chapter" data-level="19.2.2" data-path="exercises.html"><a href="exercises.html#taxonomic-abundance-table-assay"><i class="fa fa-check"></i><b>19.2.2</b> Taxonomic abundance table (assay)</a></li>
<li class="chapter" data-level="19.2.3" data-path="exercises.html"><a href="exercises.html#sample-side-information"><i class="fa fa-check"></i><b>19.2.3</b> Sample side information</a></li>
<li class="chapter" data-level="19.2.4" data-path="exercises.html"><a href="exercises.html#feature-side-information"><i class="fa fa-check"></i><b>19.2.4</b> Feature side information</a></li>
<li class="chapter" data-level="19.2.5" data-path="exercises.html"><a href="exercises.html#other-components"><i class="fa fa-check"></i><b>19.2.5</b> Other components</a></li>
</ul></li>
<li class="chapter" data-level="19.3" data-path="exercises.html"><a href="exercises.html#treesummarizedexperiment-data-import"><i class="fa fa-check"></i><b>19.3</b> TreeSummarizedExperiment: data import</a></li>
<li class="chapter" data-level="19.4" data-path="exercises.html"><a href="exercises.html#data-manipulation"><i class="fa fa-check"></i><b>19.4</b> Data manipulation</a>
<ul>
<li class="chapter" data-level="19.4.1" data-path="exercises.html"><a href="exercises.html#subsetting-1"><i class="fa fa-check"></i><b>19.4.1</b> Subsetting</a></li>
<li class="chapter" data-level="19.4.2" data-path="exercises.html"><a href="exercises.html#library-sizes"><i class="fa fa-check"></i><b>19.4.2</b> Library sizes</a></li>
<li class="chapter" data-level="19.4.3" data-path="exercises.html"><a href="exercises.html#prevalent-and-core-taxonomic-features"><i class="fa fa-check"></i><b>19.4.3</b> Prevalent and core taxonomic features</a></li>
<li class="chapter" data-level="19.4.4" data-path="exercises.html"><a href="exercises.html#explore-the-data"><i class="fa fa-check"></i><b>19.4.4</b> Explore the data</a></li>
<li class="chapter" data-level="19.4.5" data-path="exercises.html"><a href="exercises.html#other-functions"><i class="fa fa-check"></i><b>19.4.5</b> Other functions</a></li>
</ul></li>
<li class="chapter" data-level="19.5" data-path="exercises.html"><a href="exercises.html#transformations"><i class="fa fa-check"></i><b>19.5</b> Transformations</a>
<ul>
<li class="chapter" data-level="19.5.1" data-path="exercises.html"><a href="exercises.html#transformations-1"><i class="fa fa-check"></i><b>19.5.1</b> Transformations</a></li>
</ul></li>
<li class="chapter" data-level="19.6" data-path="exercises.html"><a href="exercises.html#taxonomic-levels"><i class="fa fa-check"></i><b>19.6</b> Taxonomic levels</a>
<ul>
<li class="chapter" data-level="19.6.1" data-path="exercises.html"><a href="exercises.html#taxonomic-levels-1"><i class="fa fa-check"></i><b>19.6.1</b> Taxonomic levels</a></li>
<li class="chapter" data-level="19.6.2" data-path="exercises.html"><a href="exercises.html#alternative-experiments-altexp"><i class="fa fa-check"></i><b>19.6.2</b> Alternative experiments (altExp)</a></li>
</ul></li>
<li class="chapter" data-level="19.7" data-path="exercises.html"><a href="exercises.html#alpha-diversity"><i class="fa fa-check"></i><b>19.7</b> Alpha diversity</a>
<ul>
<li class="chapter" data-level="19.7.1" data-path="exercises.html"><a href="exercises.html#alpha-diversity-basics"><i class="fa fa-check"></i><b>19.7.1</b> Alpha diversity basics</a></li>
<li class="chapter" data-level="19.7.2" data-path="exercises.html"><a href="exercises.html#alpha-diversity-extra"><i class="fa fa-check"></i><b>19.7.2</b> Alpha diversity extra</a></li>
</ul></li>
<li class="chapter" data-level="19.8" data-path="exercises.html"><a href="exercises.html#community-composition"><i class="fa fa-check"></i><b>19.8</b> Community composition</a>
<ul>
<li class="chapter" data-level="19.8.1" data-path="exercises.html"><a href="exercises.html#beta-diversity-basics"><i class="fa fa-check"></i><b>19.8.1</b> Beta diversity basics</a></li>
<li class="chapter" data-level="19.8.2" data-path="exercises.html"><a href="exercises.html#beta-diversity-extra"><i class="fa fa-check"></i><b>19.8.2</b> Beta diversity extra</a></li>
</ul></li>
<li class="chapter" data-level="19.9" data-path="exercises.html"><a href="exercises.html#visualization-1"><i class="fa fa-check"></i><b>19.9</b> Visualization</a>
<ul>
<li class="chapter" data-level="19.9.1" data-path="exercises.html"><a href="exercises.html#multivariate-ordination"><i class="fa fa-check"></i><b>19.9.1</b> Multivariate ordination</a></li>
<li class="chapter" data-level="19.9.2" data-path="exercises.html"><a href="exercises.html#heatmap-visualization"><i class="fa fa-check"></i><b>19.9.2</b> Heatmap visualization</a></li>
</ul></li>
<li class="chapter" data-level="19.10" data-path="exercises.html"><a href="exercises.html#differential-abundance-1"><i class="fa fa-check"></i><b>19.10</b> Differential abundance</a>
<ul>
<li class="chapter" data-level="19.10.1" data-path="exercises.html"><a href="exercises.html#univariate-analyses"><i class="fa fa-check"></i><b>19.10.1</b> Univariate analyses</a></li>
<li class="chapter" data-level="19.10.2" data-path="exercises.html"><a href="exercises.html#differential-abundance-analysis-1"><i class="fa fa-check"></i><b>19.10.2</b> Differential abundance analysis</a></li>
</ul></li>
<li class="chapter" data-level="19.11" data-path="exercises.html"><a href="exercises.html#multiomics"><i class="fa fa-check"></i><b>19.11</b> Multiomics</a>
<ul>
<li class="chapter" data-level="19.11.1" data-path="exercises.html"><a href="exercises.html#introduction-to-multiassayexperiment-mae"><i class="fa fa-check"></i><b>19.11.1</b> Introduction to MultiAssayExperiment (MAE)</a></li>
<li class="chapter" data-level="19.11.2" data-path="exercises.html"><a href="exercises.html#introduction-to-multiomics"><i class="fa fa-check"></i><b>19.11.2</b> Introduction to multiomics</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="bibliography.html"><a href="bibliography.html"><i class="fa fa-check"></i>Bibliography</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Orchestrating Microbiome Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="microbiome-community" class="section level1 hasAnchor" number="9">
<h1><span class="header-section-number">Chapter 9</span> Community composition<a href="microbiome-community.html#microbiome-community" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<script>
document.addEventListener("click", function (event) {
    if (event.target.classList.contains("rebook-collapse")) {
        event.target.classList.toggle("active");
        var content = event.target.nextElementSibling;
        if (content.style.display === "block") {
            content.style.display = "none";
        } else {
            content.style.display = "block";
        }
    }
})
</script>
<style>
.rebook-collapse {
  background-color: #eee;
  color: #444;
  cursor: pointer;
  padding: 18px;
  width: 100%;
  border: none;
  text-align: left;
  outline: none;
  font-size: 15px;
}

.rebook-content {
  padding: 0 18px;
  display: none;
  overflow: hidden;
  background-color: #f1f1f1;
}
</style>
<div class="sourceCode" id="cb247"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb247-1"><a href="microbiome-community.html#cb247-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mia)</span>
<span id="cb247-2"><a href="microbiome-community.html#cb247-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;GlobalPatterns&quot;</span>, <span class="at">package=</span><span class="st">&quot;mia&quot;</span>)</span>
<span id="cb247-3"><a href="microbiome-community.html#cb247-3" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> GlobalPatterns</span></code></pre></div>
<div id="visual-composition" class="section level2 hasAnchor" number="9.1">
<h2><span class="header-section-number">9.1</span> Visualizing taxonomic composition<a href="microbiome-community.html#visual-composition" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="composition-barplot" class="section level3 hasAnchor" number="9.1.1">
<h3><span class="header-section-number">9.1.1</span> Composition barplot<a href="microbiome-community.html#composition-barplot" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>A typical way to visualize microbiome composition is by using
composition barplot. In the following, relative abundance is
calculated and top taxa are retrieved for the Phylum rank. Thereafter,
the barplot is visualized ordering rank by abundance values and
samples by “Bacteroidetes”:</p>
<div class="sourceCode" id="cb248"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb248-1"><a href="microbiome-community.html#cb248-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(miaViz)</span>
<span id="cb248-2"><a href="microbiome-community.html#cb248-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Computing relative abundance</span></span>
<span id="cb248-3"><a href="microbiome-community.html#cb248-3" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">relAbundanceCounts</span>(tse)</span>
<span id="cb248-4"><a href="microbiome-community.html#cb248-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb248-5"><a href="microbiome-community.html#cb248-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Getting top taxa on a Phylum level</span></span>
<span id="cb248-6"><a href="microbiome-community.html#cb248-6" aria-hidden="true" tabindex="-1"></a>tse_phylum <span class="ot">&lt;-</span> <span class="fu">agglomerateByRank</span>(tse, <span class="at">rank =</span><span class="st">&quot;Phylum&quot;</span>, <span class="at">onRankOnly=</span><span class="cn">TRUE</span>)</span>
<span id="cb248-7"><a href="microbiome-community.html#cb248-7" aria-hidden="true" tabindex="-1"></a>top_taxa <span class="ot">&lt;-</span> <span class="fu">getTopTaxa</span>(tse_phylum,<span class="at">top =</span> <span class="dv">5</span>, <span class="at">assay_name =</span> <span class="st">&quot;relabundance&quot;</span>)</span>
<span id="cb248-8"><a href="microbiome-community.html#cb248-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb248-9"><a href="microbiome-community.html#cb248-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Renaming the &quot;Phylum&quot; rank to keep only top taxa and the rest to &quot;Other&quot;</span></span>
<span id="cb248-10"><a href="microbiome-community.html#cb248-10" aria-hidden="true" tabindex="-1"></a>phylum_renamed <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">rowData</span>(tse)<span class="sc">$</span>Phylum,</span>
<span id="cb248-11"><a href="microbiome-community.html#cb248-11" aria-hidden="true" tabindex="-1"></a>                   <span class="cf">function</span>(x){<span class="cf">if</span> (x <span class="sc">%in%</span> top_taxa) {x} <span class="cf">else</span> {<span class="st">&quot;Other&quot;</span>}})</span>
<span id="cb248-12"><a href="microbiome-community.html#cb248-12" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(tse)<span class="sc">$</span>Phylum <span class="ot">&lt;-</span> <span class="fu">as.character</span>(phylum_renamed)</span>
<span id="cb248-13"><a href="microbiome-community.html#cb248-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb248-14"><a href="microbiome-community.html#cb248-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualizing the composition barplot, with samples order by &quot;Bacteroidetes&quot;</span></span>
<span id="cb248-15"><a href="microbiome-community.html#cb248-15" aria-hidden="true" tabindex="-1"></a><span class="fu">plotAbundance</span>(tse, <span class="at">assay_name=</span><span class="st">&quot;relabundance&quot;</span>, <span class="at">rank =</span> <span class="st">&quot;Phylum&quot;</span>,</span>
<span id="cb248-16"><a href="microbiome-community.html#cb248-16" aria-hidden="true" tabindex="-1"></a>              <span class="at">order_rank_by=</span><span class="st">&quot;abund&quot;</span>, </span>
<span id="cb248-17"><a href="microbiome-community.html#cb248-17" aria-hidden="true" tabindex="-1"></a>              <span class="at">order_sample_by =</span> <span class="st">&quot;Bacteroidetes&quot;</span>)</span></code></pre></div>
<p><img src="21_microbiome_community_files/figure-html/unnamed-chunk-1-1.png" width="672" /></p>
</div>
<div id="composition-heatmap" class="section level3 hasAnchor" number="9.1.2">
<h3><span class="header-section-number">9.1.2</span> Composition heatmap<a href="microbiome-community.html#composition-heatmap" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Community composition can be visualized with heatmap, where the
horizontal axis represents samples and the vertical axis the
taxa. Color of each intersection point represents abundance of a taxon
in a specific sample.</p>
<p>Here, abundances are first CLR (centered log-ratio) transformed to
remove compositionality bias. Then Z transformation is applied to
CLR-transformed data. This shifts all taxa to zero mean and unit
variance, allowing visual comparison between taxa that have different
absolute abundance levels. After these rough visual exploration
techniques, we can visualize the abundances at Phylum level.</p>
<div class="sourceCode" id="cb249"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb249-1"><a href="microbiome-community.html#cb249-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb249-2"><a href="microbiome-community.html#cb249-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb249-3"><a href="microbiome-community.html#cb249-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Add clr-transformation on samples</span></span>
<span id="cb249-4"><a href="microbiome-community.html#cb249-4" aria-hidden="true" tabindex="-1"></a><span class="fu">assay</span>(tse_phylum, <span class="st">&quot;pseudo&quot;</span>) <span class="ot">&lt;-</span> <span class="fu">assay</span>(tse_phylum, <span class="st">&quot;counts&quot;</span>) <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb249-5"><a href="microbiome-community.html#cb249-5" aria-hidden="true" tabindex="-1"></a>tse_phylum <span class="ot">&lt;-</span> <span class="fu">transformCounts</span>(tse_phylum, <span class="at">assay_name =</span> <span class="st">&quot;pseudo&quot;</span>,</span>
<span id="cb249-6"><a href="microbiome-community.html#cb249-6" aria-hidden="true" tabindex="-1"></a>                              <span class="at">method =</span> <span class="st">&quot;relabundance&quot;</span>)</span>
<span id="cb249-7"><a href="microbiome-community.html#cb249-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb249-8"><a href="microbiome-community.html#cb249-8" aria-hidden="true" tabindex="-1"></a>tse_phylum <span class="ot">&lt;-</span> <span class="fu">transformCounts</span>(tse_phylum,</span>
<span id="cb249-9"><a href="microbiome-community.html#cb249-9" aria-hidden="true" tabindex="-1"></a>                  <span class="at">assay_name =</span> <span class="st">&quot;relabundance&quot;</span>,</span>
<span id="cb249-10"><a href="microbiome-community.html#cb249-10" aria-hidden="true" tabindex="-1"></a>          <span class="at">method =</span> <span class="st">&quot;clr&quot;</span>)</span>
<span id="cb249-11"><a href="microbiome-community.html#cb249-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb249-12"><a href="microbiome-community.html#cb249-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Add z-transformation on features (taxa)</span></span>
<span id="cb249-13"><a href="microbiome-community.html#cb249-13" aria-hidden="true" tabindex="-1"></a>tse_phylum <span class="ot">&lt;-</span> <span class="fu">transformCounts</span>(tse_phylum, <span class="at">assay_name =</span> <span class="st">&quot;clr&quot;</span>, </span>
<span id="cb249-14"><a href="microbiome-community.html#cb249-14" aria-hidden="true" tabindex="-1"></a>                              <span class="at">MARGIN =</span> <span class="st">&quot;features&quot;</span>,</span>
<span id="cb249-15"><a href="microbiome-community.html#cb249-15" aria-hidden="true" tabindex="-1"></a>                              <span class="at">method =</span> <span class="st">&quot;z&quot;</span>, <span class="at">name =</span> <span class="st">&quot;clr_z&quot;</span>)</span></code></pre></div>
<p>Visualize as heatmap.</p>
<div class="sourceCode" id="cb250"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb250-1"><a href="microbiome-community.html#cb250-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Melt the assay for plotting purposes</span></span>
<span id="cb250-2"><a href="microbiome-community.html#cb250-2" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">meltAssay</span>(tse_phylum, <span class="at">assay_name =</span> <span class="st">&quot;clr_z&quot;</span>)</span>
<span id="cb250-3"><a href="microbiome-community.html#cb250-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb250-4"><a href="microbiome-community.html#cb250-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Determines the scaling of colours</span></span>
<span id="cb250-5"><a href="microbiome-community.html#cb250-5" aria-hidden="true" tabindex="-1"></a>maxval <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">max</span>(<span class="fu">abs</span>(df<span class="sc">$</span>clr_z)))</span>
<span id="cb250-6"><a href="microbiome-community.html#cb250-6" aria-hidden="true" tabindex="-1"></a>limits <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="sc">-</span>maxval, maxval)</span>
<span id="cb250-7"><a href="microbiome-community.html#cb250-7" aria-hidden="true" tabindex="-1"></a>breaks <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="at">from =</span> <span class="fu">min</span>(limits), <span class="at">to =</span> <span class="fu">max</span>(limits), <span class="at">by =</span> <span class="fl">0.5</span>)</span>
<span id="cb250-8"><a href="microbiome-community.html#cb250-8" aria-hidden="true" tabindex="-1"></a>colours <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;darkblue&quot;</span>, <span class="st">&quot;blue&quot;</span>, <span class="st">&quot;white&quot;</span>, <span class="st">&quot;red&quot;</span>, <span class="st">&quot;darkred&quot;</span>)</span>
<span id="cb250-9"><a href="microbiome-community.html#cb250-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb250-10"><a href="microbiome-community.html#cb250-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates a ggplot object</span></span>
<span id="cb250-11"><a href="microbiome-community.html#cb250-11" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x =</span> SampleID, <span class="at">y =</span> FeatureID, <span class="at">fill =</span> clr_z)) <span class="sc">+</span></span>
<span id="cb250-12"><a href="microbiome-community.html#cb250-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb250-13"><a href="microbiome-community.html#cb250-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradientn</span>(<span class="at">name =</span> <span class="st">&quot;CLR + Z transform&quot;</span>, </span>
<span id="cb250-14"><a href="microbiome-community.html#cb250-14" aria-hidden="true" tabindex="-1"></a>                       <span class="at">breaks =</span> breaks, <span class="at">limits =</span> limits, <span class="at">colours =</span> colours) <span class="sc">+</span> </span>
<span id="cb250-15"><a href="microbiome-community.html#cb250-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>),</span>
<span id="cb250-16"><a href="microbiome-community.html#cb250-16" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">45</span>, <span class="at">hjust=</span><span class="dv">1</span>),</span>
<span id="cb250-17"><a href="microbiome-community.html#cb250-17" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.key.size =</span> <span class="fu">unit</span>(<span class="dv">1</span>, <span class="st">&quot;cm&quot;</span>)) <span class="sc">+</span></span>
<span id="cb250-18"><a href="microbiome-community.html#cb250-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Samples&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Taxa&quot;</span>)</span></code></pre></div>
<p><img src="21_microbiome_community_files/figure-html/heatmapvisu-1.png" width="672" /></p>
<p><em>pheatmap</em> is a package that provides methods to plot clustered heatmaps.</p>
<div class="sourceCode" id="cb251"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb251-1"><a href="microbiome-community.html#cb251-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(pheatmap)){<span class="fu">install.packages</span>(<span class="st">&quot;pheatmap&quot;</span>); <span class="fu">library</span>(pheatmap)}</span>
<span id="cb251-2"><a href="microbiome-community.html#cb251-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb251-3"><a href="microbiome-community.html#cb251-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Takes subset: only samples from feces, skin, or tongue</span></span>
<span id="cb251-4"><a href="microbiome-community.html#cb251-4" aria-hidden="true" tabindex="-1"></a>tse_phylum_subset <span class="ot">&lt;-</span> tse_phylum[ , <span class="fu">colData</span>(tse_phylum)<span class="sc">$</span>SampleType <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Feces&quot;</span>, <span class="st">&quot;Skin&quot;</span>, <span class="st">&quot;Tongue&quot;</span>) ]</span>
<span id="cb251-5"><a href="microbiome-community.html#cb251-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb251-6"><a href="microbiome-community.html#cb251-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Add clr-transformation</span></span>
<span id="cb251-7"><a href="microbiome-community.html#cb251-7" aria-hidden="true" tabindex="-1"></a>tse_phylum_subset <span class="ot">&lt;-</span> <span class="fu">transformCounts</span>(tse_phylum_subset,</span>
<span id="cb251-8"><a href="microbiome-community.html#cb251-8" aria-hidden="true" tabindex="-1"></a>                         <span class="at">method =</span> <span class="st">&quot;clr&quot;</span>,</span>
<span id="cb251-9"><a href="microbiome-community.html#cb251-9" aria-hidden="true" tabindex="-1"></a>                 <span class="at">pseudocount =</span> <span class="dv">1</span>)</span>
<span id="cb251-10"><a href="microbiome-community.html#cb251-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb251-11"><a href="microbiome-community.html#cb251-11" aria-hidden="true" tabindex="-1"></a>tse_phylum_subset <span class="ot">&lt;-</span> <span class="fu">transformCounts</span>(tse_phylum_subset, <span class="at">assay_name =</span> <span class="st">&quot;clr&quot;</span>,</span>
<span id="cb251-12"><a href="microbiome-community.html#cb251-12" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">MARGIN =</span> <span class="st">&quot;features&quot;</span>, </span>
<span id="cb251-13"><a href="microbiome-community.html#cb251-13" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">method =</span> <span class="st">&quot;z&quot;</span>, <span class="at">name =</span> <span class="st">&quot;clr_z&quot;</span>)</span>
<span id="cb251-14"><a href="microbiome-community.html#cb251-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb251-15"><a href="microbiome-community.html#cb251-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Get n most abundant taxa, and subsets the data by them</span></span>
<span id="cb251-16"><a href="microbiome-community.html#cb251-16" aria-hidden="true" tabindex="-1"></a>top_taxa <span class="ot">&lt;-</span> <span class="fu">getTopTaxa</span>(tse_phylum_subset, <span class="at">top =</span> <span class="dv">20</span>)</span>
<span id="cb251-17"><a href="microbiome-community.html#cb251-17" aria-hidden="true" tabindex="-1"></a>tse_phylum_subset <span class="ot">&lt;-</span> tse_phylum_subset[top_taxa, ]</span>
<span id="cb251-18"><a href="microbiome-community.html#cb251-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb251-19"><a href="microbiome-community.html#cb251-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Gets the assay table</span></span>
<span id="cb251-20"><a href="microbiome-community.html#cb251-20" aria-hidden="true" tabindex="-1"></a>mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(tse_phylum_subset, <span class="st">&quot;clr_z&quot;</span>)</span>
<span id="cb251-21"><a href="microbiome-community.html#cb251-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb251-22"><a href="microbiome-community.html#cb251-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates the heatmap</span></span>
<span id="cb251-23"><a href="microbiome-community.html#cb251-23" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(mat)</span></code></pre></div>
<p><img src="21_microbiome_community_files/figure-html/pheatmap1-1.png" width="672" /></p>
<p>We can create clusters by hierarchical clustering and add them to the plot.</p>
<div class="sourceCode" id="cb252"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb252-1"><a href="microbiome-community.html#cb252-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(ape)){</span>
<span id="cb252-2"><a href="microbiome-community.html#cb252-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;ape&quot;</span>)</span>
<span id="cb252-3"><a href="microbiome-community.html#cb252-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(ape)</span>
<span id="cb252-4"><a href="microbiome-community.html#cb252-4" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb252-5"><a href="microbiome-community.html#cb252-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb252-6"><a href="microbiome-community.html#cb252-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Hierarchical clustering</span></span>
<span id="cb252-7"><a href="microbiome-community.html#cb252-7" aria-hidden="true" tabindex="-1"></a>taxa_hclust <span class="ot">&lt;-</span> <span class="fu">hclust</span>(<span class="fu">dist</span>(mat), <span class="at">method =</span> <span class="st">&quot;complete&quot;</span>)</span>
<span id="cb252-8"><a href="microbiome-community.html#cb252-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb252-9"><a href="microbiome-community.html#cb252-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates a phylogenetic tree</span></span>
<span id="cb252-10"><a href="microbiome-community.html#cb252-10" aria-hidden="true" tabindex="-1"></a>taxa_tree <span class="ot">&lt;-</span> <span class="fu">as.phylo</span>(taxa_hclust)</span></code></pre></div>
<div class="sourceCode" id="cb253"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb253-1"><a href="microbiome-community.html#cb253-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(ggtree)){</span>
<span id="cb253-2"><a href="microbiome-community.html#cb253-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;ggtree&quot;</span>)</span>
<span id="cb253-3"><a href="microbiome-community.html#cb253-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(ggtree)</span>
<span id="cb253-4"><a href="microbiome-community.html#cb253-4" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb253-5"><a href="microbiome-community.html#cb253-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb253-6"><a href="microbiome-community.html#cb253-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot taxa tree</span></span>
<span id="cb253-7"><a href="microbiome-community.html#cb253-7" aria-hidden="true" tabindex="-1"></a>taxa_tree <span class="ot">&lt;-</span> <span class="fu">ggtree</span>(taxa_tree) <span class="sc">+</span> </span>
<span id="cb253-8"><a href="microbiome-community.html#cb253-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.margin=</span><span class="fu">margin</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>)) <span class="co"># removes margins</span></span>
<span id="cb253-9"><a href="microbiome-community.html#cb253-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb253-10"><a href="microbiome-community.html#cb253-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Get order of taxa in plot</span></span>
<span id="cb253-11"><a href="microbiome-community.html#cb253-11" aria-hidden="true" tabindex="-1"></a>taxa_ordered <span class="ot">&lt;-</span> <span class="fu">get_taxa_name</span>(taxa_tree)</span>
<span id="cb253-12"><a href="microbiome-community.html#cb253-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb253-13"><a href="microbiome-community.html#cb253-13" aria-hidden="true" tabindex="-1"></a>taxa_tree</span></code></pre></div>
<p><img src="21_microbiome_community_files/figure-html/pheatmap3-1.png" width="672" /></p>
<p>Based on phylo tree, we decide to create three clusters.</p>
<div class="sourceCode" id="cb254"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb254-1"><a href="microbiome-community.html#cb254-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates clusters</span></span>
<span id="cb254-2"><a href="microbiome-community.html#cb254-2" aria-hidden="true" tabindex="-1"></a>taxa_clusters <span class="ot">&lt;-</span> <span class="fu">cutree</span>(<span class="at">tree =</span> taxa_hclust, <span class="at">k =</span> <span class="dv">3</span>)</span>
<span id="cb254-3"><a href="microbiome-community.html#cb254-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb254-4"><a href="microbiome-community.html#cb254-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Converts into data frame</span></span>
<span id="cb254-5"><a href="microbiome-community.html#cb254-5" aria-hidden="true" tabindex="-1"></a>taxa_clusters <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">clusters =</span> taxa_clusters)</span>
<span id="cb254-6"><a href="microbiome-community.html#cb254-6" aria-hidden="true" tabindex="-1"></a>taxa_clusters<span class="sc">$</span>clusters <span class="ot">&lt;-</span> <span class="fu">factor</span>(taxa_clusters<span class="sc">$</span>clusters)</span>
<span id="cb254-7"><a href="microbiome-community.html#cb254-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb254-8"><a href="microbiome-community.html#cb254-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Order data so that it&#39;s same as in phylo tree</span></span>
<span id="cb254-9"><a href="microbiome-community.html#cb254-9" aria-hidden="true" tabindex="-1"></a>taxa_clusters <span class="ot">&lt;-</span> taxa_clusters[taxa_ordered, , drop <span class="ot">=</span> <span class="cn">FALSE</span>] </span>
<span id="cb254-10"><a href="microbiome-community.html#cb254-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb254-11"><a href="microbiome-community.html#cb254-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Prints taxa and their clusters</span></span>
<span id="cb254-12"><a href="microbiome-community.html#cb254-12" aria-hidden="true" tabindex="-1"></a>taxa_clusters</span></code></pre></div>
<pre><code>##                  clusters
## Chloroflexi             3
## Actinobacteria          3
## Crenarchaeota           3
## Planctomycetes          3
## Gemmatimonadetes        3
## Thermi                  3
## Acidobacteria           3
## Spirochaetes            2
## Fusobacteria            2
## SR1                     2
## Cyanobacteria           2
## Proteobacteria          2
## Synergistetes           2
## Lentisphaerae           1
## Bacteroidetes           1
## Verrucomicrobia         1
## Tenericutes             1
## Firmicutes              1
## Euryarchaeota           1
## SAR406                  1</code></pre>
<div class="sourceCode" id="cb256"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb256-1"><a href="microbiome-community.html#cb256-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Adds information to rowData</span></span>
<span id="cb256-2"><a href="microbiome-community.html#cb256-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(tse_phylum_subset)<span class="sc">$</span>clusters <span class="ot">&lt;-</span> taxa_clusters[<span class="fu">order</span>(<span class="fu">match</span>(<span class="fu">rownames</span>(taxa_clusters), <span class="fu">rownames</span>(tse_phylum_subset))), ]</span>
<span id="cb256-3"><a href="microbiome-community.html#cb256-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb256-4"><a href="microbiome-community.html#cb256-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Prints taxa and their clusters</span></span>
<span id="cb256-5"><a href="microbiome-community.html#cb256-5" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(tse_phylum_subset)<span class="sc">$</span>clusters</span></code></pre></div>
<pre><code>##  [1] 1 1 2 3 2 2 1 1 1 1 3 2 3 3 3 2 2 3 3 1
## Levels: 1 2 3</code></pre>
<div class="sourceCode" id="cb258"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb258-1"><a href="microbiome-community.html#cb258-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Hierarchical clustering</span></span>
<span id="cb258-2"><a href="microbiome-community.html#cb258-2" aria-hidden="true" tabindex="-1"></a>sample_hclust <span class="ot">&lt;-</span> <span class="fu">hclust</span>(<span class="fu">dist</span>(<span class="fu">t</span>(mat)), <span class="at">method =</span> <span class="st">&quot;complete&quot;</span>)</span>
<span id="cb258-3"><a href="microbiome-community.html#cb258-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb258-4"><a href="microbiome-community.html#cb258-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates a phylogenetic tree</span></span>
<span id="cb258-5"><a href="microbiome-community.html#cb258-5" aria-hidden="true" tabindex="-1"></a>sample_tree <span class="ot">&lt;-</span> <span class="fu">as.phylo</span>(sample_hclust)</span>
<span id="cb258-6"><a href="microbiome-community.html#cb258-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb258-7"><a href="microbiome-community.html#cb258-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot sample tree</span></span>
<span id="cb258-8"><a href="microbiome-community.html#cb258-8" aria-hidden="true" tabindex="-1"></a>sample_tree <span class="ot">&lt;-</span> <span class="fu">ggtree</span>(sample_tree) <span class="sc">+</span> <span class="fu">layout_dendrogram</span>() <span class="sc">+</span> </span>
<span id="cb258-9"><a href="microbiome-community.html#cb258-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.margin=</span><span class="fu">margin</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>)) <span class="co"># removes margins</span></span>
<span id="cb258-10"><a href="microbiome-community.html#cb258-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb258-11"><a href="microbiome-community.html#cb258-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Get order of samples in plot</span></span>
<span id="cb258-12"><a href="microbiome-community.html#cb258-12" aria-hidden="true" tabindex="-1"></a>samples_ordered <span class="ot">&lt;-</span> <span class="fu">rev</span>(<span class="fu">get_taxa_name</span>(sample_tree))</span>
<span id="cb258-13"><a href="microbiome-community.html#cb258-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb258-14"><a href="microbiome-community.html#cb258-14" aria-hidden="true" tabindex="-1"></a>sample_tree</span></code></pre></div>
<p><img src="21_microbiome_community_files/figure-html/pheatmap6-1.png" width="672" /></p>
<div class="sourceCode" id="cb259"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb259-1"><a href="microbiome-community.html#cb259-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates clusters</span></span>
<span id="cb259-2"><a href="microbiome-community.html#cb259-2" aria-hidden="true" tabindex="-1"></a>sample_clusters <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">cutree</span>(<span class="at">tree =</span> sample_hclust, <span class="at">k =</span> <span class="dv">3</span>))</span>
<span id="cb259-3"><a href="microbiome-community.html#cb259-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb259-4"><a href="microbiome-community.html#cb259-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Converts into data frame</span></span>
<span id="cb259-5"><a href="microbiome-community.html#cb259-5" aria-hidden="true" tabindex="-1"></a>sample_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">clusters =</span> sample_clusters)</span>
<span id="cb259-6"><a href="microbiome-community.html#cb259-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb259-7"><a href="microbiome-community.html#cb259-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Order data so that it&#39;s same as in phylo tree</span></span>
<span id="cb259-8"><a href="microbiome-community.html#cb259-8" aria-hidden="true" tabindex="-1"></a>sample_data <span class="ot">&lt;-</span> sample_data[samples_ordered, , drop <span class="ot">=</span> <span class="cn">FALSE</span>] </span>
<span id="cb259-9"><a href="microbiome-community.html#cb259-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb259-10"><a href="microbiome-community.html#cb259-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Order data based on </span></span>
<span id="cb259-11"><a href="microbiome-community.html#cb259-11" aria-hidden="true" tabindex="-1"></a>tse_phylum_subset <span class="ot">&lt;-</span> tse_phylum_subset[ , <span class="fu">rownames</span>(sample_data)]</span>
<span id="cb259-12"><a href="microbiome-community.html#cb259-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb259-13"><a href="microbiome-community.html#cb259-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Add sample type data</span></span>
<span id="cb259-14"><a href="microbiome-community.html#cb259-14" aria-hidden="true" tabindex="-1"></a>sample_data<span class="sc">$</span>sample_types <span class="ot">&lt;-</span> <span class="fu">unfactor</span>(<span class="fu">colData</span>(tse_phylum_subset)<span class="sc">$</span>SampleType)</span>
<span id="cb259-15"><a href="microbiome-community.html#cb259-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb259-16"><a href="microbiome-community.html#cb259-16" aria-hidden="true" tabindex="-1"></a>sample_data</span></code></pre></div>
<pre><code>##         clusters sample_types
## M11Plmr        2         Skin
## M31Plmr        2         Skin
## F21Plmr        2         Skin
## M31Fcsw        1        Feces
## M11Fcsw        1        Feces
## TS28           3        Feces
## TS29           3        Feces
## M31Tong        3       Tongue
## M11Tong        3       Tongue</code></pre>
<p>Now we can create heatmap with additional annotations.</p>
<div class="sourceCode" id="cb261"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb261-1"><a href="microbiome-community.html#cb261-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Determines the scaling of colorss</span></span>
<span id="cb261-2"><a href="microbiome-community.html#cb261-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Scale colors</span></span>
<span id="cb261-3"><a href="microbiome-community.html#cb261-3" aria-hidden="true" tabindex="-1"></a>breaks <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="fu">ceiling</span>(<span class="fu">max</span>(<span class="fu">abs</span>(mat))), <span class="fu">ceiling</span>(<span class="fu">max</span>(<span class="fu">abs</span>(mat))), </span>
<span id="cb261-4"><a href="microbiome-community.html#cb261-4" aria-hidden="true" tabindex="-1"></a>              <span class="at">length.out =</span> <span class="fu">ifelse</span>( <span class="fu">max</span>(<span class="fu">abs</span>(mat))<span class="sc">&gt;</span><span class="dv">5</span>, <span class="dv">2</span><span class="sc">*</span><span class="fu">ceiling</span>(<span class="fu">max</span>(<span class="fu">abs</span>(mat))), <span class="dv">10</span> ) )</span>
<span id="cb261-5"><a href="microbiome-community.html#cb261-5" aria-hidden="true" tabindex="-1"></a>colors <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">&quot;darkblue&quot;</span>, <span class="st">&quot;blue&quot;</span>, <span class="st">&quot;white&quot;</span>, <span class="st">&quot;red&quot;</span>, <span class="st">&quot;darkred&quot;</span>))(<span class="fu">length</span>(breaks)<span class="sc">-</span><span class="dv">1</span>)</span>
<span id="cb261-6"><a href="microbiome-community.html#cb261-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb261-7"><a href="microbiome-community.html#cb261-7" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(mat, <span class="at">annotation_row =</span> taxa_clusters, </span>
<span id="cb261-8"><a href="microbiome-community.html#cb261-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_col =</span> sample_data,</span>
<span id="cb261-9"><a href="microbiome-community.html#cb261-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">breaks =</span> breaks,</span>
<span id="cb261-10"><a href="microbiome-community.html#cb261-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">color =</span> colors)</span></code></pre></div>
<p><img src="21_microbiome_community_files/figure-html/pheatmap8-1.png" width="672" /></p>
<p>In addition, there are also other packages that provide functions for
more complex heatmaps, such as
<a href="https://docs.ropensci.org/iheatmapr/articles/full_vignettes/iheatmapr.html"><em>iheatmapr</em></a>
and ComplexHeatmap <span class="citation">(<a href="#ref-ComplexHeatmap" role="doc-biblioref">Gu 2022</a>)</span>.
<a href="http://www.bioconductor.org/packages/release/bioc/vignettes/sechm/inst/doc/sechm.html">sechm</a>
package provides wrapper for <em>ComplexHeatmap</em> and its usage is
explained in chapter <a href="viz-chapter.html#viz-chapter">14</a> along with the <code>pheatmap</code>
package for clustered heatmaps.</p>
</div>
</div>
<div id="session-info-6" class="section level2 unnumbered hasAnchor">
<h2>Session Info<a href="microbiome-community.html#session-info-6" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<button class="rebook-collapse">
View session info
</button>
<div class="rebook-content">
<pre><code>R version 4.2.1 (2022-06-23)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.4 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ggtree_3.4.4                   ape_5.7                       
 [3] pheatmap_1.0.12                miaViz_1.7.4                  
 [5] ggraph_2.1.0                   ggplot2_3.4.1                 
 [7] mia_1.7.9                      MultiAssayExperiment_1.24.0   
 [9] TreeSummarizedExperiment_2.1.4 Biostrings_2.66.0             
[11] XVector_0.38.0                 SingleCellExperiment_1.20.0   
[13] SummarizedExperiment_1.28.0    Biobase_2.58.0                
[15] GenomicRanges_1.50.2           GenomeInfoDb_1.34.9           
[17] IRanges_2.32.0                 S4Vectors_0.36.1              
[19] BiocGenerics_0.44.0            MatrixGenerics_1.10.0         
[21] matrixStats_0.63.0-9003        BiocStyle_2.24.0              
[23] rebook_1.6.0                  

loaded via a namespace (and not attached):
  [1] utf8_1.2.3                  tidyselect_1.2.0           
  [3] RSQLite_2.3.0               AnnotationDbi_1.58.0       
  [5] grid_4.2.1                  TSP_1.2-3                  
  [7] BiocParallel_1.32.5         Rtsne_0.16                 
  [9] munsell_0.5.0               ScaledMatrix_1.6.0         
 [11] codetools_0.2-19            withr_2.5.0                
 [13] colorspace_2.1-0            filelock_1.0.2             
 [15] highr_0.10                  knitr_1.42                 
 [17] ca_0.71.1                   labeling_0.4.2             
 [19] miaTime_0.1.20              GenomeInfoDbData_1.2.9     
 [21] polyclip_1.10-4             bit64_4.0.5                
 [23] farver_2.1.1                vctrs_0.5.2                
 [25] treeio_1.22.0               generics_0.1.3             
 [27] xfun_0.37                   R6_2.5.1                   
 [29] doParallel_1.0.17           ggbeeswarm_0.7.1           
 [31] clue_0.3-64                 graphlayouts_0.8.4         
 [33] rsvd_1.0.5                  seriation_1.4.2            
 [35] locfit_1.5-9.7              gridGraphics_0.5-1         
 [37] bitops_1.0-7                cachem_1.0.7               
 [39] DelayedArray_0.24.0         scales_1.2.1               
 [41] SEtools_1.10.0              beeswarm_0.4.0             
 [43] gtable_0.3.1                beachmat_2.14.0            
 [45] sva_3.44.0                  tidygraph_1.2.3            
 [47] rlang_1.0.6                 genefilter_1.78.0          
 [49] GlobalOptions_0.1.2         splines_4.2.1              
 [51] lazyeval_0.2.2              BiocManager_1.30.20        
 [53] yaml_2.3.7                  reshape2_1.4.4             
 [55] tools_4.2.1                 bookdown_0.33              
 [57] ggplotify_0.1.0             decontam_1.18.0            
 [59] jquerylib_0.1.4             RColorBrewer_1.1-3         
 [61] Rcpp_1.0.10                 plyr_1.8.8                 
 [63] sparseMatrixStats_1.10.0    zlibbioc_1.44.0            
 [65] purrr_1.0.1                 RCurl_1.98-1.10            
 [67] GetoptLong_1.0.5            viridis_0.6.2              
 [69] ggrepel_0.9.3               cluster_2.1.4              
 [71] DECIPHER_2.26.0             magrittr_2.0.3             
 [73] data.table_1.14.8           openxlsx_4.2.5.2           
 [75] circlize_0.4.15             ggnewscale_0.4.8           
 [77] randomcoloR_1.1.0.1         patchwork_1.1.2            
 [79] evaluate_0.20               xtable_1.8-4               
 [81] XML_3.99-0.13               gridExtra_2.3              
 [83] shape_1.4.6                 compiler_4.2.1             
 [85] scater_1.26.1               tibble_3.2.0               
 [87] V8_4.2.2                    crayon_1.5.2               
 [89] htmltools_0.5.4             ggfun_0.0.9                
 [91] mgcv_1.8-42                 aplot_0.1.10               
 [93] tidyr_1.3.0                 geneplotter_1.74.0         
 [95] DBI_1.1.3                   tweenr_2.0.2               
 [97] ComplexHeatmap_2.12.1       MASS_7.3-58.3              
 [99] Matrix_1.5-3                permute_0.9-7              
[101] cli_3.6.0                   parallel_4.2.1             
[103] igraph_1.4.1                pkgconfig_2.0.3            
[105] dir.expiry_1.4.0            registry_0.5-1             
[107] scuttle_1.8.4               foreach_1.5.2              
[109] annotate_1.74.0             vipor_0.4.5                
[111] bslib_0.4.2                 DirichletMultinomial_1.40.0
[113] yulab.utils_0.0.6           stringr_1.5.0              
[115] digest_0.6.31               vegan_2.6-4                
[117] graph_1.74.0                rmarkdown_2.20             
[119] tidytree_0.4.2              edgeR_3.38.4               
[121] DelayedMatrixStats_1.20.0   curl_5.0.0                 
[123] rjson_0.2.21                lifecycle_1.0.3            
[125] nlme_3.1-162                jsonlite_1.8.4             
[127] BiocNeighbors_1.16.0        CodeDepends_0.6.5          
[129] viridisLite_0.4.1           limma_3.52.4               
[131] fansi_1.0.4                 pillar_1.8.1               
[133] lattice_0.20-45             KEGGREST_1.36.3            
[135] fastmap_1.1.1               httr_1.4.5                 
[137] survival_3.3-1              glue_1.6.2                 
[139] zip_2.2.2                   sechm_1.4.1                
[141] png_0.1-8                   iterators_1.0.14           
[143] bit_4.0.5                   ggforce_0.4.1              
[145] stringi_1.7.12              sass_0.4.5                 
[147] blob_1.2.3                  DESeq2_1.36.0              
[149] BiocSingular_1.14.0         memoise_2.0.1              
[151] dplyr_1.1.0                 irlba_2.3.5.1              </code></pre>
</div>

</div>
</div>
<h3>Bibliography<a href="bibliography.html#bibliography" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-ComplexHeatmap" class="csl-entry">
Gu, Zuguang. 2022. <span>“Complex Heatmap Visualization.”</span> <em>iMeta</em> 1 (3): e43. https://doi.org/<a href="https://doi.org/10.1002/imt2.43">https://doi.org/10.1002/imt2.43</a>.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="community-similarity.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="clustering.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["OMA.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
