<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 13 Multi-assay analyses | Orchestrating Microbiome Analysis</title>
  <meta name="description" content="Chapter 13 Multi-assay analyses | Orchestrating Microbiome Analysis" />
  <meta name="generator" content="bookdown 0.26 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 13 Multi-assay analyses | Orchestrating Microbiome Analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="microbiome/OMA" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 13 Multi-assay analyses | Orchestrating Microbiome Analysis" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="differential-abundance.html"/>
<link rel="next" href="resources.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<script src="libs/plotly-binding-4.10.0/plotly.js"></script>
<script src="libs/typedarray-0.1/typedarray.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="libs/plotly-htmlwidgets-css-2.5.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="libs/plotly-main-2.5.1/plotly-latest.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Orchestrating Microbiome Analysis</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="part"><span><b>I Introduction</b></span></li>
<li class="chapter" data-level="1" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="containers.html"><a href="containers.html"><i class="fa fa-check"></i><b>2</b> Microbiome Data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="containers.html"><a href="containers.html#data-science-framework"><i class="fa fa-check"></i><b>2.1</b> Data science framework</a></li>
<li class="chapter" data-level="2.2" data-path="containers.html"><a href="containers.html#data-containers"><i class="fa fa-check"></i><b>2.2</b> Data containers</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="containers.html"><a href="containers.html#assay-data"><i class="fa fa-check"></i><b>2.2.1</b> Assay data</a></li>
<li class="chapter" data-level="2.2.2" data-path="containers.html"><a href="containers.html#coldata"><i class="fa fa-check"></i><b>2.2.2</b> colData</a></li>
<li class="chapter" data-level="2.2.3" data-path="containers.html"><a href="containers.html#rowdata"><i class="fa fa-check"></i><b>2.2.3</b> rowData</a></li>
<li class="chapter" data-level="2.2.4" data-path="containers.html"><a href="containers.html#rowtree"><i class="fa fa-check"></i><b>2.2.4</b> rowTree</a></li>
<li class="chapter" data-level="2.2.5" data-path="containers.html"><a href="containers.html#alternative-experiments"><i class="fa fa-check"></i><b>2.2.5</b> Alternative experiments</a></li>
<li class="chapter" data-level="2.2.6" data-path="containers.html"><a href="containers.html#multiassayexperiments"><i class="fa fa-check"></i><b>2.2.6</b> MultiAssayExperiments</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="containers.html"><a href="containers.html#loading-experimental-microbiome-data"><i class="fa fa-check"></i><b>2.3</b> Loading experimental microbiome data</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="containers.html"><a href="containers.html#s-workflow"><i class="fa fa-check"></i><b>2.3.1</b> 16S workflow</a></li>
<li class="chapter" data-level="2.3.2" data-path="containers.html"><a href="containers.html#import-from-external-files"><i class="fa fa-check"></i><b>2.3.2</b> Import from external files</a></li>
<li class="chapter" data-level="2.3.3" data-path="containers.html"><a href="containers.html#conversions-between-data-formats-in-r"><i class="fa fa-check"></i><b>2.3.3</b> Conversions between data formats in R</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="containers.html"><a href="containers.html#example-data"><i class="fa fa-check"></i><b>2.4</b> Demonstration data</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="containers.html"><a href="containers.html#package-data"><i class="fa fa-check"></i><b>2.4.1</b> Package data</a></li>
<li class="chapter" data-level="2.4.2" data-path="containers.html"><a href="containers.html#experimenthub-data"><i class="fa fa-check"></i><b>2.4.2</b> ExperimentHub data</a></li>
<li class="chapter" data-level="2.4.3" data-path="containers.html"><a href="containers.html#other-data-sources"><i class="fa fa-check"></i><b>2.4.3</b> Other data sources</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="containers.html"><a href="containers.html#session-info"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="packages.html"><a href="packages.html"><i class="fa fa-check"></i><b>3</b> Packages</a>
<ul>
<li class="chapter" data-level="3.1" data-path="packages.html"><a href="packages.html#package-installation"><i class="fa fa-check"></i><b>3.1</b> Package installation</a></li>
<li class="chapter" data-level="3.2" data-path="packages.html"><a href="packages.html#some-available-packages"><i class="fa fa-check"></i><b>3.2</b> Some available packages</a></li>
<li class="chapter" data-level="" data-path="packages.html"><a href="packages.html#session-info-1"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="part"><span><b>II Focus Topics</b></span></li>
<li class="chapter" data-level="4" data-path="datamanipulation.html"><a href="datamanipulation.html"><i class="fa fa-check"></i><b>4</b> Data Manipulation</a>
<ul>
<li class="chapter" data-level="4.1" data-path="datamanipulation.html"><a href="datamanipulation.html#tidying-and-subsetting"><i class="fa fa-check"></i><b>4.1</b> Tidying and subsetting</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="datamanipulation.html"><a href="datamanipulation.html#tidy-data"><i class="fa fa-check"></i><b>4.1.1</b> Tidy data</a></li>
<li class="chapter" data-level="4.1.2" data-path="datamanipulation.html"><a href="datamanipulation.html#subsetting"><i class="fa fa-check"></i><b>4.1.2</b> Subsetting</a></li>
<li class="chapter" data-level="4.1.3" data-path="datamanipulation.html"><a href="datamanipulation.html#additional-functions"><i class="fa fa-check"></i><b>4.1.3</b> Additional functions</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>5</b> Exploration and quality Control</a>
<ul>
<li class="chapter" data-level="5.1" data-path="quality-control.html"><a href="quality-control.html#abundance"><i class="fa fa-check"></i><b>5.1</b> Abundance</a></li>
<li class="chapter" data-level="5.2" data-path="quality-control.html"><a href="quality-control.html#prevalence"><i class="fa fa-check"></i><b>5.2</b> Prevalence</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="quality-control.html"><a href="quality-control.html#prevalence-analysis"><i class="fa fa-check"></i><b>5.2.1</b> Prevalence analysis</a></li>
<li class="chapter" data-level="5.2.2" data-path="quality-control.html"><a href="quality-control.html#rare-taxa"><i class="fa fa-check"></i><b>5.2.2</b> Rare taxa</a></li>
<li class="chapter" data-level="5.2.3" data-path="quality-control.html"><a href="quality-control.html#plotting-prevalence"><i class="fa fa-check"></i><b>5.2.3</b> Plotting prevalence</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="quality-control.html"><a href="quality-control.html#quality-control-1"><i class="fa fa-check"></i><b>5.3</b> Quality control</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="quality-control.html"><a href="quality-control.html#top-taxa"><i class="fa fa-check"></i><b>5.3.1</b> Top taxa</a></li>
<li class="chapter" data-level="5.3.2" data-path="quality-control.html"><a href="quality-control.html#library-size-read-count"><i class="fa fa-check"></i><b>5.3.2</b> Library size / read count</a></li>
<li class="chapter" data-level="5.3.3" data-path="quality-control.html"><a href="quality-control.html#contaminant-sequences"><i class="fa fa-check"></i><b>5.3.3</b> Contaminant sequences</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="quality-control.html"><a href="quality-control.html#session-info-2"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="taxonomic-information.html"><a href="taxonomic-information.html"><i class="fa fa-check"></i><b>6</b> Taxonomic Information</a>
<ul>
<li class="chapter" data-level="6.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#assigning-taxonomic-information."><i class="fa fa-check"></i><b>6.1</b> Assigning taxonomic information.</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#dada2"><i class="fa fa-check"></i><b>6.1.1</b> dada2</a></li>
<li class="chapter" data-level="6.1.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#decipher"><i class="fa fa-check"></i><b>6.1.2</b> DECIPHER</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#functions-to-access-taxonomic-information"><i class="fa fa-check"></i><b>6.2</b> Functions to access taxonomic information</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#generate-a-taxonomic-tree-on-the-fly"><i class="fa fa-check"></i><b>6.2.1</b> Generate a taxonomic tree on the fly</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="taxonomic-information.html"><a href="taxonomic-information.html#data-agglomeration"><i class="fa fa-check"></i><b>6.3</b> Data agglomeration</a></li>
<li class="chapter" data-level="6.4" data-path="taxonomic-information.html"><a href="taxonomic-information.html#data-transformation"><i class="fa fa-check"></i><b>6.4</b> Data transformation</a></li>
<li class="chapter" data-level="6.5" data-path="taxonomic-information.html"><a href="taxonomic-information.html#pick-specific"><i class="fa fa-check"></i><b>6.5</b> Pick specific</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#abundances-of-all-taxa-in-specific-sample"><i class="fa fa-check"></i><b>6.5.1</b> Abundances of all taxa in specific sample</a></li>
<li class="chapter" data-level="6.5.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#abundances-of-specific-taxa-in-all-samples"><i class="fa fa-check"></i><b>6.5.2</b> Abundances of specific taxa in all samples</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="taxonomic-information.html"><a href="taxonomic-information.html#session-info-3"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="community-diversity.html"><a href="community-diversity.html"><i class="fa fa-check"></i><b>7</b> Community diversity</a>
<ul>
<li class="chapter" data-level="7.1" data-path="community-diversity.html"><a href="community-diversity.html#estimation"><i class="fa fa-check"></i><b>7.1</b> Estimation</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="community-diversity.html"><a href="community-diversity.html#richness"><i class="fa fa-check"></i><b>7.1.1</b> Richness</a></li>
<li class="chapter" data-level="7.1.2" data-path="community-diversity.html"><a href="community-diversity.html#diversity"><i class="fa fa-check"></i><b>7.1.2</b> Diversity</a></li>
<li class="chapter" data-level="7.1.3" data-path="community-diversity.html"><a href="community-diversity.html#faith-phylogenetic-diversity"><i class="fa fa-check"></i><b>7.1.3</b> Faith phylogenetic diversity</a></li>
<li class="chapter" data-level="7.1.4" data-path="community-diversity.html"><a href="community-diversity.html#evenness"><i class="fa fa-check"></i><b>7.1.4</b> Evenness</a></li>
<li class="chapter" data-level="7.1.5" data-path="community-diversity.html"><a href="community-diversity.html#dominance"><i class="fa fa-check"></i><b>7.1.5</b> Dominance</a></li>
<li class="chapter" data-level="7.1.6" data-path="community-diversity.html"><a href="community-diversity.html#rarity"><i class="fa fa-check"></i><b>7.1.6</b> Rarity</a></li>
<li class="chapter" data-level="7.1.7" data-path="community-diversity.html"><a href="community-diversity.html#divergence"><i class="fa fa-check"></i><b>7.1.7</b> Divergence</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="community-diversity.html"><a href="community-diversity.html#visualization"><i class="fa fa-check"></i><b>7.2</b> Visualization</a></li>
<li class="chapter" data-level="" data-path="community-diversity.html"><a href="community-diversity.html#session-info-4"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="community-similarity.html"><a href="community-similarity.html"><i class="fa fa-check"></i><b>8</b> Community similarity</a>
<ul>
<li class="chapter" data-level="8.1" data-path="community-similarity.html"><a href="community-similarity.html#explained-variance"><i class="fa fa-check"></i><b>8.1</b> Explained variance</a></li>
<li class="chapter" data-level="8.2" data-path="community-similarity.html"><a href="community-similarity.html#community-comparisons-by-beta-diversity-analysis"><i class="fa fa-check"></i><b>8.2</b> Community comparisons by beta diversity analysis</a></li>
<li class="chapter" data-level="8.3" data-path="community-similarity.html"><a href="community-similarity.html#other-ordination-methods"><i class="fa fa-check"></i><b>8.3</b> Other ordination methods</a></li>
<li class="chapter" data-level="8.4" data-path="community-similarity.html"><a href="community-similarity.html#visualizing-the-most-dominant-genus-on-pcoa"><i class="fa fa-check"></i><b>8.4</b> Visualizing the most dominant genus on PCoA</a>
<ul>
<li class="chapter" data-level="8.4.1" data-path="community-similarity.html"><a href="community-similarity.html#testing-differences-in-community-composition-between-sample-groups"><i class="fa fa-check"></i><b>8.4.1</b> Testing differences in community composition between sample groups</a></li>
<li class="chapter" data-level="8.4.2" data-path="community-similarity.html"><a href="community-similarity.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>8.4.2</b> Checking the homogeneity condition</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="community-similarity.html"><a href="community-similarity.html#further-reading"><i class="fa fa-check"></i><b>8.5</b> Further reading</a></li>
<li class="chapter" data-level="" data-path="community-similarity.html"><a href="community-similarity.html#session-info-5"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="microbiome-community.html"><a href="microbiome-community.html"><i class="fa fa-check"></i><b>9</b> Community composition</a>
<ul>
<li class="chapter" data-level="9.1" data-path="microbiome-community.html"><a href="microbiome-community.html#visualizing-taxonomic-composition"><i class="fa fa-check"></i><b>9.1</b> Visualizing taxonomic composition</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="microbiome-community.html"><a href="microbiome-community.html#composition-barplot"><i class="fa fa-check"></i><b>9.1.1</b> Composition barplot</a></li>
<li class="chapter" data-level="9.1.2" data-path="microbiome-community.html"><a href="microbiome-community.html#composition-heatmap"><i class="fa fa-check"></i><b>9.1.2</b> Composition heatmap</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="community-typing.html"><a href="community-typing.html"><i class="fa fa-check"></i><b>10</b> Community typing</a>
<ul>
<li class="chapter" data-level="10.1" data-path="community-typing.html"><a href="community-typing.html#dirichlet-multinomial-mixtures-dmm"><i class="fa fa-check"></i><b>10.1</b> Dirichlet Multinomial Mixtures (DMM)</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="community-typing.html"><a href="community-typing.html#pcoa-for-asv-level-data-with-bray-curtis-with-dmm-clusters-shown-with-colors"><i class="fa fa-check"></i><b>10.1.1</b> PCoA for ASV-level data with Bray-Curtis; with DMM clusters shown with colors</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="community-typing.html"><a href="community-typing.html#community-detection"><i class="fa fa-check"></i><b>10.2</b> Community Detection</a></li>
<li class="chapter" data-level="10.3" data-path="community-typing.html"><a href="community-typing.html#additional-community-typing"><i class="fa fa-check"></i><b>10.3</b> Additional Community Typing</a></li>
<li class="chapter" data-level="" data-path="community-typing.html"><a href="community-typing.html#session-info-6"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="biclustering.html"><a href="biclustering.html"><i class="fa fa-check"></i><b>11</b> Biclustering</a>
<ul>
<li class="chapter" data-level="11.1" data-path="biclustering.html"><a href="biclustering.html#taxa-vs-samples"><i class="fa fa-check"></i><b>11.1</b> Taxa vs samples</a></li>
<li class="chapter" data-level="11.2" data-path="biclustering.html"><a href="biclustering.html#taxa-vs-biomolecules"><i class="fa fa-check"></i><b>11.2</b> Taxa vs biomolecules</a></li>
<li class="chapter" data-level="11.3" data-path="biclustering.html"><a href="biclustering.html#taxa-vs-taxa"><i class="fa fa-check"></i><b>11.3</b> Taxa vs taxa</a></li>
<li class="chapter" data-level="" data-path="biclustering.html"><a href="biclustering.html#session-info-7"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="differential-abundance.html"><a href="differential-abundance.html"><i class="fa fa-check"></i><b>12</b> Differential abundance</a>
<ul>
<li class="chapter" data-level="12.1" data-path="differential-abundance.html"><a href="differential-abundance.html#differential-abundance-analysis"><i class="fa fa-check"></i><b>12.1</b> Differential abundance analysis</a>
<ul>
<li class="chapter" data-level="12.1.1" data-path="differential-abundance.html"><a href="differential-abundance.html#examples-and-tools"><i class="fa fa-check"></i><b>12.1.1</b> Examples and tools</a></li>
<li class="chapter" data-level="12.1.2" data-path="differential-abundance.html"><a href="differential-abundance.html#prevalence-filtering"><i class="fa fa-check"></i><b>12.1.2</b> Prevalence Filtering</a></li>
<li class="chapter" data-level="12.1.3" data-path="differential-abundance.html"><a href="differential-abundance.html#aldex2"><i class="fa fa-check"></i><b>12.1.3</b> ALDEx2</a></li>
<li class="chapter" data-level="12.1.4" data-path="differential-abundance.html"><a href="differential-abundance.html#ancom-bc"><i class="fa fa-check"></i><b>12.1.4</b> ANCOM-BC</a></li>
<li class="chapter" data-level="12.1.5" data-path="differential-abundance.html"><a href="differential-abundance.html#maaslin2"><i class="fa fa-check"></i><b>12.1.5</b> MaAsLin2</a></li>
<li class="chapter" data-level="12.1.6" data-path="differential-abundance.html"><a href="differential-abundance.html#comparison-of-the-methods"><i class="fa fa-check"></i><b>12.1.6</b> Comparison of the methods</a></li>
<li class="chapter" data-level="12.1.7" data-path="differential-abundance.html"><a href="differential-abundance.html#confounding-variables"><i class="fa fa-check"></i><b>12.1.7</b> Confounding variables</a></li>
</ul></li>
<li class="chapter" data-level="12.2" data-path="differential-abundance.html"><a href="differential-abundance.html#tree-based-methods"><i class="fa fa-check"></i><b>12.2</b> Tree-based methods</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="differential-abundance.html"><a href="differential-abundance.html#group-wise-associations-testing-based-on-balances-with-fido"><i class="fa fa-check"></i><b>12.2.1</b> Group-wise associations testing based on balances with fido</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="differential-abundance.html"><a href="differential-abundance.html#session-info-8"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="multi-assay_analyses.html"><a href="multi-assay_analyses.html"><i class="fa fa-check"></i><b>13</b> Multi-assay analyses</a>
<ul>
<li class="chapter" data-level="13.1" data-path="multi-assay_analyses.html"><a href="multi-assay_analyses.html#cross-correlation-analysis"><i class="fa fa-check"></i><b>13.1</b> Cross-correlation Analysis</a></li>
<li class="chapter" data-level="13.2" data-path="multi-assay_analyses.html"><a href="multi-assay_analyses.html#multi-omics-factor-analysis"><i class="fa fa-check"></i><b>13.2</b> Multi-Omics Factor Analysis</a></li>
<li class="chapter" data-level="" data-path="multi-assay_analyses.html"><a href="multi-assay_analyses.html#session-info-9"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="part"><span><b>III Appendix</b></span></li>
<li class="chapter" data-level="14" data-path="resources.html"><a href="resources.html"><i class="fa fa-check"></i><b>14</b> Resources</a>
<ul>
<li class="chapter" data-level="14.1" data-path="resources.html"><a href="resources.html#data-containers-1"><i class="fa fa-check"></i><b>14.1</b> Data containers</a>
<ul>
<li class="chapter" data-level="14.1.1" data-path="resources.html"><a href="resources.html#resources-for-treesummarizedexperiment"><i class="fa fa-check"></i><b>14.1.1</b> Resources for TreeSummarizedExperiment</a></li>
<li class="chapter" data-level="14.1.2" data-path="resources.html"><a href="resources.html#other-relevant-containers"><i class="fa fa-check"></i><b>14.1.2</b> Other relevant containers</a></li>
<li class="chapter" data-level="14.1.3" data-path="resources.html"><a href="resources.html#alternative-containers-for-microbiome-data"><i class="fa fa-check"></i><b>14.1.3</b> Alternative containers for microbiome data</a></li>
<li class="chapter" data-level="14.1.4" data-path="resources.html"><a href="resources.html#resources-for-phyloseq"><i class="fa fa-check"></i><b>14.1.4</b> Resources for phyloseq</a></li>
</ul></li>
<li class="chapter" data-level="14.2" data-path="resources.html"><a href="resources.html#r-programming-resources"><i class="fa fa-check"></i><b>14.2</b> R programming resources</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="resources.html"><a href="resources.html#bioc_intro"><i class="fa fa-check"></i><b>14.2.1</b> Bioconductor Classes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="extras.html"><a href="extras.html"><i class="fa fa-check"></i><b>15</b> Extra material</a>
<ul>
<li class="chapter" data-level="15.1" data-path="extras.html"><a href="extras.html#interactive-3d-plots"><i class="fa fa-check"></i><b>15.1</b> Interactive 3D Plots</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="acknowledgments.html"><a href="acknowledgments.html"><i class="fa fa-check"></i><b>16</b> Acknowledgments</a></li>
<li class="chapter" data-level="" data-path="bibliography.html"><a href="bibliography.html"><i class="fa fa-check"></i>Bibliography</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Orchestrating Microbiome Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="multi-assay_analyses" class="section level1 hasAnchor" number="13">
<h1><span class="header-section-number">Chapter 13</span> Multi-assay analyses<a href="multi-assay_analyses.html#multi-assay_analyses" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<script>
document.addEventListener("click", function (event) {
    if (event.target.classList.contains("rebook-collapse")) {
        event.target.classList.toggle("active");
        var content = event.target.nextElementSibling;
        if (content.style.display === "block") {
            content.style.display = "none";
        } else {
            content.style.display = "block";
        }
    }
})
</script>
<style>
.rebook-collapse {
  background-color: #eee;
  color: #444;
  cursor: pointer;
  padding: 18px;
  width: 100%;
  border: none;
  text-align: left;
  outline: none;
  font-size: 15px;
}

.rebook-content {
  padding: 0 18px;
  display: none;
  overflow: hidden;
  background-color: #f1f1f1;
}
</style>
<pre><code>## Loading required package: ecodist</code></pre>
<div class="sourceCode" id="cb335"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb335-1"><a href="multi-assay_analyses.html#cb335-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mia)</span></code></pre></div>
<p>Multi-omics means that we integrate data from multiple sources. For example,
we can integrate microbial abundances in the gut with biomolecular profiling data
from blood samples. This kind of integrative multi-omic approaches can support the
analysis of microbiome dysbiosis and facilitate the discovery of novel biomarkers
for health and disease.</p>
<p>With cross-correlation analysis, we can analyze how strongly and how differently
variables are associated between each other. For instance, we can analyze if
higher presence of a specific taxon equals to higher levels of a biomolecule.</p>
<p>The data containers that the <em>miaverse</em> utilizes are scalable and they can contain
different types of data in a same container. Because of that, the <em>miaverse</em> is
well-suitable for multi-assay microbiome data which incorporates different types
of complementary data sources in a single reproducible workflow.</p>
<p>Another experiment can be stored in
<a href="https://microbiome.github.io/OMA/containers.html#alternative-experiments">altExp</a>
slot of SE data container or both experiments can be stored side-by-side in
<a href="https://microbiome.github.io/OMA/containers.html#multiassayexperiments">MAE</a>
data container.</p>
<p>Different experiments are first imported into SE or TreeSE data container similarly
to the case when only one experiment is present. After that different experiments are
combined into the same data container. Result is one TreeSE object with alternative
experiment in altExp slot, or MAE object with multiple experiment in its
experiment slot.</p>
<p>As an example data, we use data from following publication: Hintikka L <em>et al.</em> (2021)
<a href="https://doi.org/10.3390/ijerph18084049">Xylo-oligosaccharides in prevention of hepatic steatosis and adipose tissue inflammation:
associating taxonomic and metabolomic patterns in fecal microbiotas with
biclustering</a>.</p>
<p>In this article, mice were fed with high-fat and low-fat diets with or without prebiotics.
The purpose of this was to study if prebiotics would reduce the negative impacts
of high-fat diet.</p>
<p>This example data can be loaded from microbiomeDataSets. The data is already in MAE
format. It includes three different experiments: microbial abundance data,
metabolite concentrations, and data about different biomarkers. Help for importing
data into SE object you can find from <a href="https://microbiome.github.io/OMA/containers.html#loading-experimental-microbiome-data">here</a>.</p>
<div class="sourceCode" id="cb336"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb336-1"><a href="multi-assay_analyses.html#cb336-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load the data</span></span>
<span id="cb336-2"><a href="multi-assay_analyses.html#cb336-2" aria-hidden="true" tabindex="-1"></a>mae <span class="ot">&lt;-</span> microbiomeDataSets<span class="sc">::</span><span class="fu">HintikkaXOData</span>()</span>
<span id="cb336-3"><a href="multi-assay_analyses.html#cb336-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb336-4"><a href="multi-assay_analyses.html#cb336-4" aria-hidden="true" tabindex="-1"></a>mae</span></code></pre></div>
<pre><code>## A MultiAssayExperiment object of 3 listed
##  experiments with user-defined names and respective classes.
##  Containing an ExperimentList class object of length 3:
##  [1] microbiota: SummarizedExperiment with 12706 rows and 40 columns
##  [2] metabolites: SummarizedExperiment with 38 rows and 40 columns
##  [3] biomarkers: SummarizedExperiment with 39 rows and 40 columns
## Functionality:
##  experiments() - obtain the ExperimentList instance
##  colData() - the primary/phenotype DataFrame
##  sampleMap() - the sample coordination DataFrame
##  `$`, `[`, `[[` - extract colData columns, subset, or experiment
##  *Format() - convert into a long or wide DataFrame
##  assays() - convert ExperimentList to a SimpleList of matrices
##  exportClass() - save data to flat files</code></pre>
<div class="sourceCode" id="cb338"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb338-1"><a href="multi-assay_analyses.html#cb338-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(stringr)){</span>
<span id="cb338-2"><a href="multi-assay_analyses.html#cb338-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;stringr&quot;</span>)</span>
<span id="cb338-3"><a href="multi-assay_analyses.html#cb338-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(stringr)</span>
<span id="cb338-4"><a href="multi-assay_analyses.html#cb338-4" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb338-5"><a href="multi-assay_analyses.html#cb338-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Drop off those bacteria that do not include information in Phylum or lower levels</span></span>
<span id="cb338-6"><a href="multi-assay_analyses.html#cb338-6" aria-hidden="true" tabindex="-1"></a>mae[[<span class="dv">1</span>]] <span class="ot">&lt;-</span> mae[[<span class="dv">1</span>]][<span class="sc">!</span><span class="fu">is.na</span>(<span class="fu">rowData</span>(mae[[<span class="dv">1</span>]])<span class="sc">$</span>Phylum), ]</span>
<span id="cb338-7"><a href="multi-assay_analyses.html#cb338-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean taxonomy data, so that names do not include additional characters</span></span>
<span id="cb338-8"><a href="multi-assay_analyses.html#cb338-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(mae[[<span class="dv">1</span>]]) <span class="ot">&lt;-</span> <span class="fu">DataFrame</span>(<span class="fu">apply</span>(<span class="fu">rowData</span>(mae[[<span class="dv">1</span>]]), <span class="dv">2</span>, </span>
<span id="cb338-9"><a href="multi-assay_analyses.html#cb338-9" aria-hidden="true" tabindex="-1"></a>                                     str_remove, <span class="at">pattern =</span> <span class="st">&quot;._[0-9]__&quot;</span>))</span>
<span id="cb338-10"><a href="multi-assay_analyses.html#cb338-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Microbiome data</span></span>
<span id="cb338-11"><a href="multi-assay_analyses.html#cb338-11" aria-hidden="true" tabindex="-1"></a>mae[[<span class="dv">1</span>]]</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 12613 40 
## metadata(0):
## assays(1): counts
## rownames(12613): GAYR01026362.62.2014 CVJT01000011.50.2173 ...
##   JRJTB:03787:02429 JRJTB:03787:02478
## rowData names(7): Phylum Class ... Species OTU
## colnames(40): C1 C2 ... C39 C40
## colData names(0):</code></pre>
<div class="sourceCode" id="cb340"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb340-1"><a href="multi-assay_analyses.html#cb340-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Metabolite data</span></span>
<span id="cb340-2"><a href="multi-assay_analyses.html#cb340-2" aria-hidden="true" tabindex="-1"></a>mae[[<span class="dv">2</span>]]</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 38 40 
## metadata(0):
## assays(1): nmr
## rownames(38): Butyrate Acetate ... Malonate 1,3-dihydroxyacetone
## rowData names(0):
## colnames(40): C1 C2 ... C39 C40
## colData names(0):</code></pre>
<div class="sourceCode" id="cb342"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb342-1"><a href="multi-assay_analyses.html#cb342-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Biomarker data</span></span>
<span id="cb342-2"><a href="multi-assay_analyses.html#cb342-2" aria-hidden="true" tabindex="-1"></a>mae[[<span class="dv">3</span>]]</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 39 40 
## metadata(0):
## assays(1): signals
## rownames(39): Triglycerides_liver CLSs_epi ... NPY_serum Glycogen_liver
## rowData names(0):
## colnames(40): C1 C2 ... C39 C40
## colData names(0):</code></pre>
<div id="cross-correlation-analysis" class="section level2 hasAnchor" number="13.1">
<h2><span class="header-section-number">13.1</span> Cross-correlation Analysis<a href="multi-assay_analyses.html#cross-correlation-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Next we can do the cross-correlation analysis.
Here we analyse if individual bacteria genera correlates
with concentrations of individual metabolites. This helps as to answer the question:
“If this bacteria is present, is this metabolite’s concentration then low or high”?</p>
<div class="sourceCode" id="cb344"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb344-1"><a href="multi-assay_analyses.html#cb344-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Agglomerate microbiome data at Genus level</span></span>
<span id="cb344-2"><a href="multi-assay_analyses.html#cb344-2" aria-hidden="true" tabindex="-1"></a>mae[[<span class="dv">1</span>]] <span class="ot">&lt;-</span> <span class="fu">agglomerateByPrevalence</span>(mae[[<span class="dv">1</span>]], <span class="at">rank =</span> <span class="st">&quot;Genus&quot;</span>)</span>
<span id="cb344-3"><a href="multi-assay_analyses.html#cb344-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Does log10 transform for microbiome data</span></span>
<span id="cb344-4"><a href="multi-assay_analyses.html#cb344-4" aria-hidden="true" tabindex="-1"></a>mae[[<span class="dv">1</span>]] <span class="ot">&lt;-</span> <span class="fu">transformSamples</span>(mae[[<span class="dv">1</span>]], <span class="at">method =</span> <span class="st">&quot;log10&quot;</span>, <span class="at">pseudocount =</span> <span class="dv">1</span>)</span>
<span id="cb344-5"><a href="multi-assay_analyses.html#cb344-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb344-6"><a href="multi-assay_analyses.html#cb344-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Give unique names so that we do not have problems when we are creating a plot</span></span>
<span id="cb344-7"><a href="multi-assay_analyses.html#cb344-7" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(mae[[<span class="dv">1</span>]]) <span class="ot">&lt;-</span> <span class="fu">getTaxonomyLabels</span>(mae[[<span class="dv">1</span>]])</span>
<span id="cb344-8"><a href="multi-assay_analyses.html#cb344-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb344-9"><a href="multi-assay_analyses.html#cb344-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Cross correlates data sets</span></span>
<span id="cb344-10"><a href="multi-assay_analyses.html#cb344-10" aria-hidden="true" tabindex="-1"></a>correlations <span class="ot">&lt;-</span> <span class="fu">testExperimentCrossCorrelation</span>(mae, </span>
<span id="cb344-11"><a href="multi-assay_analyses.html#cb344-11" aria-hidden="true" tabindex="-1"></a>                                               <span class="at">experiment1 =</span> <span class="dv">1</span>,</span>
<span id="cb344-12"><a href="multi-assay_analyses.html#cb344-12" aria-hidden="true" tabindex="-1"></a>                                               <span class="at">experiment2 =</span> <span class="dv">2</span>,</span>
<span id="cb344-13"><a href="multi-assay_analyses.html#cb344-13" aria-hidden="true" tabindex="-1"></a>                                               <span class="at">abund_values1 =</span> <span class="st">&quot;log10&quot;</span>, </span>
<span id="cb344-14"><a href="multi-assay_analyses.html#cb344-14" aria-hidden="true" tabindex="-1"></a>                                               <span class="at">abund_values2 =</span> <span class="st">&quot;nmr&quot;</span>,</span>
<span id="cb344-15"><a href="multi-assay_analyses.html#cb344-15" aria-hidden="true" tabindex="-1"></a>                                               <span class="at">method =</span> <span class="st">&quot;spearman&quot;</span>, </span>
<span id="cb344-16"><a href="multi-assay_analyses.html#cb344-16" aria-hidden="true" tabindex="-1"></a>                                               <span class="at">p_adj_threshold =</span> <span class="fl">0.05</span>,</span>
<span id="cb344-17"><a href="multi-assay_analyses.html#cb344-17" aria-hidden="true" tabindex="-1"></a>                                               <span class="at">cor_threshold =</span> <span class="dv">0</span>,</span>
<span id="cb344-18"><a href="multi-assay_analyses.html#cb344-18" aria-hidden="true" tabindex="-1"></a>                                               <span class="co"># Remove when mia is fixed</span></span>
<span id="cb344-19"><a href="multi-assay_analyses.html#cb344-19" aria-hidden="true" tabindex="-1"></a>                                               <span class="co"># mode = &quot;matrix&quot;,</span></span>
<span id="cb344-20"><a href="multi-assay_analyses.html#cb344-20" aria-hidden="true" tabindex="-1"></a>                                               <span class="co"># sort = TRUE,</span></span>
<span id="cb344-21"><a href="multi-assay_analyses.html#cb344-21" aria-hidden="true" tabindex="-1"></a>                                               <span class="at">show_warnings =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p>Creates the heatmap</p>
<div class="sourceCode" id="cb345"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb345-1"><a href="multi-assay_analyses.html#cb345-1" aria-hidden="true" tabindex="-1"></a><span class="co"># if( !require(&quot;ComplexHeatmap&quot;) ){</span></span>
<span id="cb345-2"><a href="multi-assay_analyses.html#cb345-2" aria-hidden="true" tabindex="-1"></a><span class="co">#     BiocManager::install(&quot;ComplexHeatmap&quot;)</span></span>
<span id="cb345-3"><a href="multi-assay_analyses.html#cb345-3" aria-hidden="true" tabindex="-1"></a><span class="co">#     library(&quot;ComplexHeatmap&quot;)</span></span>
<span id="cb345-4"><a href="multi-assay_analyses.html#cb345-4" aria-hidden="true" tabindex="-1"></a><span class="co"># }</span></span>
<span id="cb345-5"><a href="multi-assay_analyses.html#cb345-5" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb345-6"><a href="multi-assay_analyses.html#cb345-6" aria-hidden="true" tabindex="-1"></a><span class="co"># # Create a heatmap and store it</span></span>
<span id="cb345-7"><a href="multi-assay_analyses.html#cb345-7" aria-hidden="true" tabindex="-1"></a><span class="co"># plot &lt;- Heatmap(correlations$cor, </span></span>
<span id="cb345-8"><a href="multi-assay_analyses.html#cb345-8" aria-hidden="true" tabindex="-1"></a><span class="co">#                 # Print values to cells</span></span>
<span id="cb345-9"><a href="multi-assay_analyses.html#cb345-9" aria-hidden="true" tabindex="-1"></a><span class="co">#                 cell_fun = function(j, i, x, y, width, height, fill) {</span></span>
<span id="cb345-10"><a href="multi-assay_analyses.html#cb345-10" aria-hidden="true" tabindex="-1"></a><span class="co">#                     # If the p-value is under threshold</span></span>
<span id="cb345-11"><a href="multi-assay_analyses.html#cb345-11" aria-hidden="true" tabindex="-1"></a><span class="co">#                     if( correlations$p_adj[i, j] &lt; 0.05 )</span></span>
<span id="cb345-12"><a href="multi-assay_analyses.html#cb345-12" aria-hidden="true" tabindex="-1"></a><span class="co">#                         # Print &quot;X&quot;</span></span>
<span id="cb345-13"><a href="multi-assay_analyses.html#cb345-13" aria-hidden="true" tabindex="-1"></a><span class="co">#                         grid.text(sprintf(&quot;%s&quot;, &quot;X&quot;), x, y, gp = gpar(fontsize = 8, col = &quot;black&quot;))</span></span>
<span id="cb345-14"><a href="multi-assay_analyses.html#cb345-14" aria-hidden="true" tabindex="-1"></a><span class="co">#                     },</span></span>
<span id="cb345-15"><a href="multi-assay_analyses.html#cb345-15" aria-hidden="true" tabindex="-1"></a><span class="co">#                 heatmap_legend_param = list(title = &quot;&quot;, legend_height = unit(5, &quot;cm&quot;))</span></span>
<span id="cb345-16"><a href="multi-assay_analyses.html#cb345-16" aria-hidden="true" tabindex="-1"></a><span class="co">#                 )</span></span>
<span id="cb345-17"><a href="multi-assay_analyses.html#cb345-17" aria-hidden="true" tabindex="-1"></a><span class="co"># plot</span></span>
<span id="cb345-18"><a href="multi-assay_analyses.html#cb345-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb345-19"><a href="multi-assay_analyses.html#cb345-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb345-20"><a href="multi-assay_analyses.html#cb345-20" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb345-21"><a href="multi-assay_analyses.html#cb345-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb345-22"><a href="multi-assay_analyses.html#cb345-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Determines the scaling of colors</span></span>
<span id="cb345-23"><a href="multi-assay_analyses.html#cb345-23" aria-hidden="true" tabindex="-1"></a>limits <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">1</span>)</span>
<span id="cb345-24"><a href="multi-assay_analyses.html#cb345-24" aria-hidden="true" tabindex="-1"></a>breaks <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="at">from =</span> <span class="fu">min</span>(limits), <span class="at">to =</span> <span class="fu">max</span>(limits), <span class="at">by =</span> <span class="fl">0.2</span>)</span>
<span id="cb345-25"><a href="multi-assay_analyses.html#cb345-25" aria-hidden="true" tabindex="-1"></a>colours <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;darkblue&quot;</span>, <span class="st">&quot;blue&quot;</span>, <span class="st">&quot;white&quot;</span>, <span class="st">&quot;red&quot;</span>, <span class="st">&quot;darkred&quot;</span>)</span>
<span id="cb345-26"><a href="multi-assay_analyses.html#cb345-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb345-27"><a href="multi-assay_analyses.html#cb345-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Which observation have p-value under 0.05? --&gt; creates a subset</span></span>
<span id="cb345-28"><a href="multi-assay_analyses.html#cb345-28" aria-hidden="true" tabindex="-1"></a>cor_table_sub <span class="ot">&lt;-</span> correlations[<span class="fu">which</span>(correlations[[<span class="st">&quot;p_adj&quot;</span>]] <span class="sc">&lt;</span> <span class="fl">0.05</span>), ]</span>
<span id="cb345-29"><a href="multi-assay_analyses.html#cb345-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb345-30"><a href="multi-assay_analyses.html#cb345-30" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates a ggplot object</span></span>
<span id="cb345-31"><a href="multi-assay_analyses.html#cb345-31" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(correlations, <span class="fu">aes</span>(<span class="at">x =</span> Var1, <span class="at">y =</span> Var2, <span class="at">fill =</span> cor)) <span class="sc">+</span></span>
<span id="cb345-32"><a href="multi-assay_analyses.html#cb345-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb345-33"><a href="multi-assay_analyses.html#cb345-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb345-34"><a href="multi-assay_analyses.html#cb345-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradientn</span>(<span class="at">name =</span> <span class="st">&quot;Correlation&quot;</span>,</span>
<span id="cb345-35"><a href="multi-assay_analyses.html#cb345-35" aria-hidden="true" tabindex="-1"></a>                       <span class="at">breaks =</span> breaks, <span class="at">limits =</span> limits, <span class="at">colours =</span> colours) <span class="sc">+</span></span>
<span id="cb345-36"><a href="multi-assay_analyses.html#cb345-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb345-37"><a href="multi-assay_analyses.html#cb345-37" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Adds label to those observations that have p-value under 0.05</span></span>
<span id="cb345-38"><a href="multi-assay_analyses.html#cb345-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">data =</span> cor_table_sub, <span class="fu">aes</span>(<span class="at">x =</span> Var1, <span class="at">y =</span> Var2, <span class="at">label =</span> <span class="st">&quot;+&quot;</span>)) <span class="sc">+</span></span>
<span id="cb345-39"><a href="multi-assay_analyses.html#cb345-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb345-40"><a href="multi-assay_analyses.html#cb345-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>),</span>
<span id="cb345-41"><a href="multi-assay_analyses.html#cb345-41" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">45</span>, <span class="at">hjust=</span><span class="dv">1</span>),</span>
<span id="cb345-42"><a href="multi-assay_analyses.html#cb345-42" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.key.size =</span> <span class="fu">unit</span>(<span class="dv">1</span>, <span class="st">&quot;cm&quot;</span>)) <span class="sc">+</span></span>
<span id="cb345-43"><a href="multi-assay_analyses.html#cb345-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Taxa&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Metabolites&quot;</span>)</span></code></pre></div>
<p><img src="23_multi-assay_analyses_files/figure-html/cross-correlation6-1.png" width="960" /></p>
</div>
<div id="multi-omics-factor-analysis" class="section level2 hasAnchor" number="13.2">
<h2><span class="header-section-number">13.2</span> Multi-Omics Factor Analysis<a href="multi-assay_analyses.html#multi-omics-factor-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Multi-Omics Factor Analysis <span class="citation">(<a href="#ref-Argelaguet2018" role="doc-biblioref">Argelaguet 2018</a>)</span> (MOFA) is
an unsupervised method for integrating multi-omic data sets in a downstream analysis.
It could be seen as a generalization of principal component analysis. Yet, with the ability to infer a latent (low-dimensional) representation,
shared among the mutliple (-omics) data sets in hand.</p>
<p>We use the R <a href="https://biofam.github.io/MOFA2/index.html">MOFA2</a> package
for the analysis, and <a href="https://biofam.github.io/MOFA2/installation.html">install</a> the corresponding dependencies.</p>
<div class="sourceCode" id="cb346"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb346-1"><a href="multi-assay_analyses.html#cb346-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(MOFA2)){</span>
<span id="cb346-2"><a href="multi-assay_analyses.html#cb346-2" aria-hidden="true" tabindex="-1"></a>    BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;MOFA2&quot;</span>)</span>
<span id="cb346-3"><a href="multi-assay_analyses.html#cb346-3" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb346-4"><a href="multi-assay_analyses.html#cb346-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb346-5"><a href="multi-assay_analyses.html#cb346-5" aria-hidden="true" tabindex="-1"></a><span class="co"># For inter-operability between Python and R, and setting Python dependencies,</span></span>
<span id="cb346-6"><a href="multi-assay_analyses.html#cb346-6" aria-hidden="true" tabindex="-1"></a><span class="co"># reticulate package is needed</span></span>
<span id="cb346-7"><a href="multi-assay_analyses.html#cb346-7" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(reticulate)){</span>
<span id="cb346-8"><a href="multi-assay_analyses.html#cb346-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;reticulate&quot;</span>)</span>
<span id="cb346-9"><a href="multi-assay_analyses.html#cb346-9" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb346-10"><a href="multi-assay_analyses.html#cb346-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb346-11"><a href="multi-assay_analyses.html#cb346-11" aria-hidden="true" tabindex="-1"></a>reticulate<span class="sc">::</span><span class="fu">install_miniconda</span>(<span class="at">force =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## [1] &quot;/github/home/.local/share/r-miniconda&quot;</code></pre>
<div class="sourceCode" id="cb348"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb348-1"><a href="multi-assay_analyses.html#cb348-1" aria-hidden="true" tabindex="-1"></a>reticulate<span class="sc">::</span><span class="fu">use_miniconda</span>(<span class="at">condaenv =</span> <span class="st">&quot;env1&quot;</span>, <span class="at">required =</span> <span class="cn">FALSE</span>)</span>
<span id="cb348-2"><a href="multi-assay_analyses.html#cb348-2" aria-hidden="true" tabindex="-1"></a>reticulate<span class="sc">::</span><span class="fu">py_install</span>(<span class="at">packages =</span> <span class="fu">c</span>(<span class="st">&quot;mofapy2&quot;</span>), <span class="at">pip =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p>The <code>mae</code> object could be used straight to create the MOFA model. Yet, we transform
our assays since the model assumes normality per default. Other distributions that
can be used, include Poisson or Bernoulli.</p>
<div class="sourceCode" id="cb349"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb349-1"><a href="multi-assay_analyses.html#cb349-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(MOFA2)</span>
<span id="cb349-2"><a href="multi-assay_analyses.html#cb349-2" aria-hidden="true" tabindex="-1"></a><span class="co"># For simplicity, classify all high-fat diets as high-fat, and all the low-fat </span></span>
<span id="cb349-3"><a href="multi-assay_analyses.html#cb349-3" aria-hidden="true" tabindex="-1"></a><span class="co"># diets as low-fat diets</span></span>
<span id="cb349-4"><a href="multi-assay_analyses.html#cb349-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(mae)<span class="sc">$</span>Diet <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">colData</span>(mae)<span class="sc">$</span>Diet <span class="sc">==</span> <span class="st">&quot;High-fat&quot;</span> <span class="sc">|</span> </span>
<span id="cb349-5"><a href="multi-assay_analyses.html#cb349-5" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">colData</span>(mae)<span class="sc">$</span>Diet <span class="sc">==</span> <span class="st">&quot;High-fat + XOS&quot;</span>, </span>
<span id="cb349-6"><a href="multi-assay_analyses.html#cb349-6" aria-hidden="true" tabindex="-1"></a>                            <span class="st">&quot;High-fat&quot;</span>, <span class="st">&quot;Low-fat&quot;</span>)</span>
<span id="cb349-7"><a href="multi-assay_analyses.html#cb349-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb349-8"><a href="multi-assay_analyses.html#cb349-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Removing duplicates at the microbiome data</span></span>
<span id="cb349-9"><a href="multi-assay_analyses.html#cb349-9" aria-hidden="true" tabindex="-1"></a><span class="co"># which are also in form e.g. &quot;Ambiguous&quot; and &quot;uncultured&quot; taxa</span></span>
<span id="cb349-10"><a href="multi-assay_analyses.html#cb349-10" aria-hidden="true" tabindex="-1"></a>mae[[<span class="dv">1</span>]] <span class="ot">&lt;-</span> mae[[<span class="dv">1</span>]][<span class="sc">!</span><span class="fu">duplicated</span>(<span class="fu">rownames</span>(<span class="fu">assay</span>(mae[[<span class="dv">1</span>]]))), ]</span>
<span id="cb349-11"><a href="multi-assay_analyses.html#cb349-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb349-12"><a href="multi-assay_analyses.html#cb349-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Transforming microbiome data with rclr</span></span>
<span id="cb349-13"><a href="multi-assay_analyses.html#cb349-13" aria-hidden="true" tabindex="-1"></a>mae[[<span class="dv">1</span>]] <span class="ot">&lt;-</span> <span class="fu">transformCounts</span>(mae[[<span class="dv">1</span>]], <span class="at">abund_values =</span> <span class="st">&quot;counts&quot;</span>, <span class="at">method =</span> <span class="st">&quot;rclr&quot;</span>)</span>
<span id="cb349-14"><a href="multi-assay_analyses.html#cb349-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb349-15"><a href="multi-assay_analyses.html#cb349-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Transforming metabolomic data with log10</span></span>
<span id="cb349-16"><a href="multi-assay_analyses.html#cb349-16" aria-hidden="true" tabindex="-1"></a>mae[[<span class="dv">2</span>]] <span class="ot">&lt;-</span> <span class="fu">transformSamples</span>(mae[[<span class="dv">2</span>]], <span class="at">abund_values =</span> <span class="st">&quot;nmr&quot;</span>, <span class="at">method =</span> <span class="st">&quot;log10&quot;</span>)</span>
<span id="cb349-17"><a href="multi-assay_analyses.html#cb349-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb349-18"><a href="multi-assay_analyses.html#cb349-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Transforming biomarker data with z-transform</span></span>
<span id="cb349-19"><a href="multi-assay_analyses.html#cb349-19" aria-hidden="true" tabindex="-1"></a>mae[[<span class="dv">3</span>]] <span class="ot">&lt;-</span> <span class="fu">transformFeatures</span>(mae[[<span class="dv">3</span>]], <span class="at">abund_values =</span> <span class="st">&quot;signals&quot;</span>, <span class="at">method =</span> <span class="st">&quot;z&quot;</span>, <span class="at">pseudocount =</span> <span class="dv">1</span>)</span>
<span id="cb349-20"><a href="multi-assay_analyses.html#cb349-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb349-21"><a href="multi-assay_analyses.html#cb349-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Removing assays no longer needed</span></span>
<span id="cb349-22"><a href="multi-assay_analyses.html#cb349-22" aria-hidden="true" tabindex="-1"></a><span class="fu">assay</span>(mae[[<span class="dv">1</span>]], <span class="st">&quot;counts&quot;</span>) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb349-23"><a href="multi-assay_analyses.html#cb349-23" aria-hidden="true" tabindex="-1"></a><span class="fu">assay</span>(mae[[<span class="dv">1</span>]], <span class="st">&quot;log10&quot;</span>) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb349-24"><a href="multi-assay_analyses.html#cb349-24" aria-hidden="true" tabindex="-1"></a><span class="fu">assay</span>(mae[[<span class="dv">2</span>]], <span class="st">&quot;nmr&quot;</span>) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb349-25"><a href="multi-assay_analyses.html#cb349-25" aria-hidden="true" tabindex="-1"></a><span class="fu">assay</span>(mae[[<span class="dv">3</span>]], <span class="st">&quot;signals&quot;</span>) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb349-26"><a href="multi-assay_analyses.html#cb349-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb349-27"><a href="multi-assay_analyses.html#cb349-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Building our mofa model</span></span>
<span id="cb349-28"><a href="multi-assay_analyses.html#cb349-28" aria-hidden="true" tabindex="-1"></a>model <span class="ot">&lt;-</span> <span class="fu">create_mofa_from_MultiAssayExperiment</span>(mae,</span>
<span id="cb349-29"><a href="multi-assay_analyses.html#cb349-29" aria-hidden="true" tabindex="-1"></a>                                               <span class="at">groups =</span> <span class="st">&quot;Diet&quot;</span>, </span>
<span id="cb349-30"><a href="multi-assay_analyses.html#cb349-30" aria-hidden="true" tabindex="-1"></a>                                               <span class="at">extract_metadata =</span> <span class="cn">TRUE</span>)</span>
<span id="cb349-31"><a href="multi-assay_analyses.html#cb349-31" aria-hidden="true" tabindex="-1"></a>model</span></code></pre></div>
<pre><code>## Untrained MOFA model with the following characteristics: 
##  Number of views: 3 
##  Views names: microbiota metabolites biomarkers 
##  Number of features (per view): 136 38 39 
##  Number of groups: 2 
##  Groups names: High-fat Low-fat 
##  Number of samples (per group): 20 20 
## </code></pre>
<p>Model options could be defined as follows:</p>
<div class="sourceCode" id="cb351"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb351-1"><a href="multi-assay_analyses.html#cb351-1" aria-hidden="true" tabindex="-1"></a>model_opts <span class="ot">&lt;-</span> <span class="fu">get_default_model_options</span>(model)</span>
<span id="cb351-2"><a href="multi-assay_analyses.html#cb351-2" aria-hidden="true" tabindex="-1"></a>model_opts<span class="sc">$</span>num_factors <span class="ot">&lt;-</span> <span class="dv">15</span></span>
<span id="cb351-3"><a href="multi-assay_analyses.html#cb351-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(model_opts)</span></code></pre></div>
<pre><code>## $likelihoods
##  microbiota metabolites  biomarkers 
##  &quot;gaussian&quot;  &quot;gaussian&quot;  &quot;gaussian&quot; 
## 
## $num_factors
## [1] 15
## 
## $spikeslab_factors
## [1] FALSE
## 
## $spikeslab_weights
## [1] TRUE
## 
## $ard_factors
## [1] TRUE
## 
## $ard_weights
## [1] TRUE</code></pre>
<p>Model’s training options are defined with the following:</p>
<div class="sourceCode" id="cb353"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb353-1"><a href="multi-assay_analyses.html#cb353-1" aria-hidden="true" tabindex="-1"></a>train_opts <span class="ot">&lt;-</span> <span class="fu">get_default_training_options</span>(model)</span>
<span id="cb353-2"><a href="multi-assay_analyses.html#cb353-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(train_opts)</span></code></pre></div>
<pre><code>## $maxiter
## [1] 1000
## 
## $convergence_mode
## [1] &quot;fast&quot;
## 
## $drop_factor_threshold
## [1] -1
## 
## $verbose
## [1] FALSE
## 
## $startELBO
## [1] 1
## 
## $freqELBO
## [1] 5</code></pre>
<p>Preparing and training the model:</p>
<div class="sourceCode" id="cb355"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb355-1"><a href="multi-assay_analyses.html#cb355-1" aria-hidden="true" tabindex="-1"></a>model.prepared <span class="ot">&lt;-</span> <span class="fu">prepare_mofa</span>(</span>
<span id="cb355-2"><a href="multi-assay_analyses.html#cb355-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">object =</span> model,</span>
<span id="cb355-3"><a href="multi-assay_analyses.html#cb355-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">model_options =</span> model_opts</span>
<span id="cb355-4"><a href="multi-assay_analyses.html#cb355-4" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb355-5"><a href="multi-assay_analyses.html#cb355-5" aria-hidden="true" tabindex="-1"></a>model.trained <span class="ot">&lt;-</span> <span class="fu">run_mofa</span>(model.prepared)</span></code></pre></div>
<p>Visualizing the variance explained:</p>
<div class="sourceCode" id="cb356"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb356-1"><a href="multi-assay_analyses.html#cb356-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb356-2"><a href="multi-assay_analyses.html#cb356-2" aria-hidden="true" tabindex="-1"></a><span class="fu">wrap_plots</span>(</span>
<span id="cb356-3"><a href="multi-assay_analyses.html#cb356-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plot_variance_explained</span>(model.trained, <span class="at">x=</span><span class="st">&quot;view&quot;</span>, <span class="at">y=</span><span class="st">&quot;factor&quot;</span>, <span class="at">plot_total =</span> T),</span>
<span id="cb356-4"><a href="multi-assay_analyses.html#cb356-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">nrow =</span> <span class="dv">2</span></span>
<span id="cb356-5"><a href="multi-assay_analyses.html#cb356-5" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span> <span class="fu">plot_annotation</span>(<span class="at">title =</span> <span class="st">&quot;Variance Explained per factor and assay&quot;</span>,</span>
<span id="cb356-6"><a href="multi-assay_analyses.html#cb356-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">theme =</span> <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>)))</span></code></pre></div>
<p><img src="23_multi-assay_analyses_files/figure-html/unnamed-chunk-6-1.png" width="960" /></p>
<p>The top weights for each assay using the three first factors:</p>
<div class="sourceCode" id="cb357"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb357-1"><a href="multi-assay_analyses.html#cb357-1" aria-hidden="true" tabindex="-1"></a>plots <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">c</span>(<span class="st">&quot;microbiota&quot;</span>, <span class="st">&quot;metabolites&quot;</span>,<span class="st">&quot;biomarkers&quot;</span>), <span class="cf">function</span>(name) {</span>
<span id="cb357-2"><a href="multi-assay_analyses.html#cb357-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plot_top_weights</span>(model.trained,</span>
<span id="cb357-3"><a href="multi-assay_analyses.html#cb357-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">view =</span> name,</span>
<span id="cb357-4"><a href="multi-assay_analyses.html#cb357-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">factors =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>,</span>
<span id="cb357-5"><a href="multi-assay_analyses.html#cb357-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">nfeatures =</span> <span class="dv">10</span>) <span class="sc">+</span></span>
<span id="cb357-6"><a href="multi-assay_analyses.html#cb357-6" aria-hidden="true" tabindex="-1"></a>        <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste0</span>(<span class="st">&quot;Top weights of the &quot;</span>, name,<span class="st">&quot; assay&quot;</span>))</span>
<span id="cb357-7"><a href="multi-assay_analyses.html#cb357-7" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb357-8"><a href="multi-assay_analyses.html#cb357-8" aria-hidden="true" tabindex="-1"></a><span class="fu">wrap_plots</span>(plots, <span class="at">nrow =</span> <span class="dv">3</span>) <span class="sc">&amp;</span> <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>))</span></code></pre></div>
<p><img src="23_multi-assay_analyses_files/figure-html/unnamed-chunk-7-1.png" width="960" /></p>
<p>More tutorials and examples of using the package are found at: <a href="https://biofam.github.io/MOFA2/tutorials.html">link</a></p>
</div>
<div id="session-info-9" class="section level2 unnumbered hasAnchor">
<h2>Session Info<a href="multi-assay_analyses.html#session-info-9" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<button class="rebook-collapse">
View session info
</button>
<div class="rebook-content">
<pre><code>R version 4.1.3 (2022-03-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.4 LTS

Matrix products: default
BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] patchwork_1.1.1                reticulate_1.24               
 [3] MOFA2_1.4.0                    ggplot2_3.3.5                 
 [5] stringr_1.4.0                  microbiomeDataSets_1.1.5      
 [7] mia_1.3.19                     MultiAssayExperiment_1.20.0   
 [9] TreeSummarizedExperiment_2.1.4 Biostrings_2.62.0             
[11] XVector_0.34.0                 SingleCellExperiment_1.16.0   
[13] SummarizedExperiment_1.24.0    Biobase_2.54.0                
[15] GenomicRanges_1.46.1           GenomeInfoDb_1.30.1           
[17] IRanges_2.28.0                 S4Vectors_0.32.4              
[19] BiocGenerics_0.40.0            MatrixGenerics_1.6.0          
[21] matrixStats_0.62.0-9000        ecodist_2.0.7                 
[23] BiocStyle_2.22.0               rebook_1.4.0                  

loaded via a namespace (and not attached):
  [1] AnnotationHub_3.2.2           corrplot_0.92                
  [3] BiocFileCache_2.2.1           plyr_1.8.7                   
  [5] lazyeval_0.2.2                splines_4.1.3                
  [7] BiocParallel_1.28.3           scater_1.22.0                
  [9] digest_0.6.29                 yulab.utils_0.0.4            
 [11] htmltools_0.5.2               viridis_0.6.2                
 [13] fansi_1.0.3                   magrittr_2.0.3               
 [15] memoise_2.0.1                 ScaledMatrix_1.2.0           
 [17] cluster_2.1.3                 DECIPHER_2.22.0              
 [19] colorspace_2.0-3              blob_1.2.3                   
 [21] rappdirs_0.3.3                ggrepel_0.9.1                
 [23] xfun_0.30                     dplyr_1.0.8                  
 [25] crayon_1.5.1                  RCurl_1.98-1.6               
 [27] jsonlite_1.8.0                graph_1.72.0                 
 [29] ape_5.6-2                     glue_1.6.2                   
 [31] gtable_0.3.0                  zlibbioc_1.40.0              
 [33] DelayedArray_0.20.0           BiocSingular_1.10.0          
 [35] Rhdf5lib_1.16.0               HDF5Array_1.22.1             
 [37] scales_1.2.0                  pheatmap_1.0.12              
 [39] DBI_1.1.2                     Rcpp_1.0.8.3                 
 [41] viridisLite_0.4.0             xtable_1.8-4                 
 [43] decontam_1.14.0               tidytree_0.3.9               
 [45] bit_4.0.4                     rsvd_1.0.5                   
 [47] httr_1.4.2                    RColorBrewer_1.1-3           
 [49] dir.expiry_1.2.0              ellipsis_0.3.2               
 [51] farver_2.1.0                  pkgconfig_2.0.3              
 [53] XML_3.99-0.9                  scuttle_1.4.0                
 [55] uwot_0.1.11                   CodeDepends_0.6.5            
 [57] sass_0.4.1                    dbplyr_2.1.1                 
 [59] here_1.0.1                    utf8_1.2.2                   
 [61] labeling_0.4.2                AnnotationDbi_1.56.2         
 [63] tidyselect_1.1.2              rlang_1.0.2                  
 [65] reshape2_1.4.4                later_1.3.0                  
 [67] munsell_0.5.0                 BiocVersion_3.14.0           
 [69] tools_4.1.3                   cachem_1.0.6                 
 [71] cli_3.2.0                     DirichletMultinomial_1.36.0  
 [73] generics_0.1.2                RSQLite_2.2.12               
 [75] ExperimentHub_2.2.1           evaluate_0.15                
 [77] fastmap_1.1.0                 yaml_2.3.5                   
 [79] knitr_1.38                    bit64_4.0.5                  
 [81] purrr_0.3.4                   KEGGREST_1.34.0              
 [83] nlme_3.1-157                  sparseMatrixStats_1.6.0      
 [85] mime_0.12                     compiler_4.1.3               
 [87] png_0.1-7                     beeswarm_0.4.0               
 [89] filelock_1.0.2                curl_4.3.2                   
 [91] interactiveDisplayBase_1.32.0 treeio_1.18.1                
 [93] tibble_3.1.6                  bslib_0.3.1                  
 [95] stringi_1.7.6                 basilisk.utils_1.6.0         
 [97] highr_0.9                     forcats_0.5.1                
 [99] lattice_0.20-45               Matrix_1.4-1                 
[101] vegan_2.6-2                   permute_0.9-7                
[103] vctrs_0.4.1                   rhdf5filters_1.6.0           
[105] pillar_1.7.0                  lifecycle_1.0.1              
[107] BiocManager_1.30.16           jquerylib_0.1.4              
[109] BiocNeighbors_1.12.0          cowplot_1.1.1                
[111] bitops_1.0-7                  irlba_2.3.5                  
[113] httpuv_1.6.5                  R6_2.5.1                     
[115] bookdown_0.26                 promises_1.2.0.1             
[117] gridExtra_2.3                 vipor_0.4.5                  
[119] codetools_0.2-18              MASS_7.3-56                  
[121] assertthat_0.2.1              rhdf5_2.38.1                 
[123] rprojroot_2.0.3               withr_2.5.0                  
[125] GenomeInfoDbData_1.2.7        mgcv_1.8-40                  
[127] parallel_4.1.3                grid_4.1.3                   
[129] beachmat_2.10.0               basilisk_1.6.0               
[131] tidyr_1.2.0                   rmarkdown_2.13               
[133] DelayedMatrixStats_1.16.0     Rtsne_0.16                   
[135] shiny_1.7.1                   ggbeeswarm_0.6.0             </code></pre>
</div>

</div>
</div>



<h3>Bibliography<a href="bibliography.html#bibliography" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Argelaguet2018" class="csl-entry">
Argelaguet, Ricard et al. 2018. <span>“Multi-Omics Factor Analysis—a Framework for Unsupervised Integration of Multi-Omics Data Sets.”</span> <em>Molecular Systems Biology</em> 14 (6): e8124. <a href="https://doi.org/10.15252/msb.20178124">https://doi.org/10.15252/msb.20178124</a>.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="differential-abundance.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="resources.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["OMA.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
