<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Quality Control | Orchestrating Microbiome Analysis</title>
  <meta name="description" content="Chapter 5 Quality Control | Orchestrating Microbiome Analysis" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Quality Control | Orchestrating Microbiome Analysis" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://github.com/microbiome/OMA" />
  
  
  <meta name="github-repo" content="microbiome/OMA" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Quality Control | Orchestrating Microbiome Analysis" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="taxonomic-information.html"/>
<link rel="next" href="microbiome-exploration.html"/>
<script src="libs/header-attrs-2.10/header-attrs.js"></script>
<script src="libs/jquery-3.5.1/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<script src="libs/plotly-binding-4.9.4.1/plotly.js"></script>
<script src="libs/typedarray-0.1/typedarray.min.js"></script>
<link href="libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
<link href="libs/plotly-htmlwidgets-css-1.57.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="libs/plotly-main-1.57.1/plotly-latest.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Orchestrating Microbiome Analysis</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="" data-path="introduction-to-bioconductor-classes.html"><a href="introduction-to-bioconductor-classes.html"><i class="fa fa-check"></i>Introduction to Bioconductor Classes</a></li>
<li class="part"><span><b>I Introduction</b></span></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#community"><i class="fa fa-check"></i><b>1.1</b> Community</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#acknowledgements"><i class="fa fa-check"></i><b>1.2</b> Acknowledgements</a></li>
</ul></li>
<li class="part"><span><b>II Introduction</b></span></li>
<li class="chapter" data-level="2" data-path="data-introduction.html"><a href="data-introduction.html"><i class="fa fa-check"></i><b>2</b> Data Infrastructure</a>
<ul>
<li class="chapter" data-level="2.1" data-path="data-introduction.html"><a href="data-introduction.html#installation"><i class="fa fa-check"></i><b>2.1</b> Installation</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="data-introduction.html"><a href="data-introduction.html#packages"><i class="fa fa-check"></i><b>2.1.1</b> Packages</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="data-introduction.html"><a href="data-introduction.html#background"><i class="fa fa-check"></i><b>2.2</b> Background</a></li>
<li class="chapter" data-level="2.3" data-path="data-introduction.html"><a href="data-introduction.html#loading-experimental-microbiome-data"><i class="fa fa-check"></i><b>2.3</b> Loading experimental microbiome data</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="data-introduction.html"><a href="data-introduction.html#importing-data-from-external-files"><i class="fa fa-check"></i><b>2.3.1</b> Importing data from external files</a></li>
<li class="chapter" data-level="2.3.2" data-path="data-introduction.html"><a href="data-introduction.html#conversions-between-data-formats-in-r"><i class="fa fa-check"></i><b>2.3.2</b> Conversions between data formats in R</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="data-introduction.html"><a href="data-introduction.html#metadata"><i class="fa fa-check"></i><b>2.4</b> Metadata</a></li>
<li class="chapter" data-level="2.5" data-path="data-introduction.html"><a href="data-introduction.html#microbiome-and-tree-data-specific-aspects"><i class="fa fa-check"></i><b>2.5</b> Microbiome and tree data specific aspects</a>
<ul>
<li class="chapter" data-level="2.5.1" data-path="data-introduction.html"><a href="data-introduction.html#assays"><i class="fa fa-check"></i><b>2.5.1</b> Assays</a></li>
<li class="chapter" data-level="2.5.2" data-path="data-introduction.html"><a href="data-introduction.html#coldata"><i class="fa fa-check"></i><b>2.5.2</b> colData</a></li>
<li class="chapter" data-level="2.5.3" data-path="data-introduction.html"><a href="data-introduction.html#rowdata"><i class="fa fa-check"></i><b>2.5.3</b> rowData</a></li>
<li class="chapter" data-level="2.5.4" data-path="data-introduction.html"><a href="data-introduction.html#rowtree"><i class="fa fa-check"></i><b>2.5.4</b> rowTree</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="data-introduction.html"><a href="data-introduction.html#data-conversion"><i class="fa fa-check"></i><b>2.6</b> Data conversion</a>
<ul>
<li class="chapter" data-level="2.6.1" data-path="data-introduction.html"><a href="data-introduction.html#tidy-data"><i class="fa fa-check"></i><b>2.6.1</b> Tidy data</a></li>
</ul></li>
<li class="chapter" data-level="2.7" data-path="data-introduction.html"><a href="data-introduction.html#conclusion"><i class="fa fa-check"></i><b>2.7</b> Conclusion</a></li>
<li class="chapter" data-level="2.8" data-path="data-introduction.html"><a href="data-introduction.html#additional-reading"><i class="fa fa-check"></i><b>2.8</b> Additional Reading</a>
<ul>
<li class="chapter" data-level="2.8.1" data-path="data-introduction.html"><a href="data-introduction.html#lecture-slides"><i class="fa fa-check"></i><b>2.8.1</b> Lecture slides</a></li>
<li class="chapter" data-level="2.8.2" data-path="data-introduction.html"><a href="data-introduction.html#r-programming-resources"><i class="fa fa-check"></i><b>2.8.2</b> R programming resources</a></li>
<li class="chapter" data-level="2.8.3" data-path="data-introduction.html"><a href="data-introduction.html#resources-for-treesummarizedexperiment"><i class="fa fa-check"></i><b>2.8.3</b> Resources for TreeSummarizedExperiment</a></li>
<li class="chapter" data-level="2.8.4" data-path="data-introduction.html"><a href="data-introduction.html#resources-for-phyloseq"><i class="fa fa-check"></i><b>2.8.4</b> Resources for phyloseq</a></li>
<li class="chapter" data-level="2.8.5" data-path="data-introduction.html"><a href="data-introduction.html#further-reading"><i class="fa fa-check"></i><b>2.8.5</b> Further reading</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-introduction.html"><a href="data-introduction.html#session-info"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="part"><span><b>III Focus Topics</b></span></li>
<li class="chapter" data-level="3" data-path="overview.html"><a href="overview.html"><i class="fa fa-check"></i><b>3</b> Overview</a>
<ul>
<li class="chapter" data-level="3.1" data-path="overview.html"><a href="overview.html#introduction-1"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="overview.html"><a href="overview.html#microbiome-experiment-types"><i class="fa fa-check"></i><b>3.2</b> Microbiome experiment types</a></li>
<li class="chapter" data-level="3.3" data-path="overview.html"><a href="overview.html#processing-steps"><i class="fa fa-check"></i><b>3.3</b> Processing steps</a></li>
<li class="chapter" data-level="3.4" data-path="overview.html"><a href="overview.html#quick-start"><i class="fa fa-check"></i><b>3.4</b> Quick Start</a></li>
<li class="chapter" data-level="" data-path="overview.html"><a href="overview.html#session-info-1"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="taxonomic-information.html"><a href="taxonomic-information.html"><i class="fa fa-check"></i><b>4</b> Taxonomic Information</a>
<ul>
<li class="chapter" data-level="4.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#assigning-taxonomic-information."><i class="fa fa-check"></i><b>4.1</b> Assigning taxonomic information.</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#dada2"><i class="fa fa-check"></i><b>4.1.1</b> dada2</a></li>
<li class="chapter" data-level="4.1.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#decipher"><i class="fa fa-check"></i><b>4.1.2</b> DECIPHER</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#functions-to-access-taxonomic-information"><i class="fa fa-check"></i><b>4.2</b> Functions to access taxonomic information</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#generate-a-taxonomic-tree-on-the-fly"><i class="fa fa-check"></i><b>4.2.1</b> Generate a taxonomic tree on the fly</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="taxonomic-information.html"><a href="taxonomic-information.html#data-agglomeration"><i class="fa fa-check"></i><b>4.3</b> Data agglomeration</a></li>
<li class="chapter" data-level="4.4" data-path="taxonomic-information.html"><a href="taxonomic-information.html#data-transformation"><i class="fa fa-check"></i><b>4.4</b> Data transformation</a></li>
<li class="chapter" data-level="4.5" data-path="taxonomic-information.html"><a href="taxonomic-information.html#pick-specific"><i class="fa fa-check"></i><b>4.5</b> Pick specific</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#abundances-of-all-taxa-in-specific-sample"><i class="fa fa-check"></i><b>4.5.1</b> Abundances of all taxa in specific sample</a></li>
<li class="chapter" data-level="4.5.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#abundances-of-specific-taxa-in-all-samples"><i class="fa fa-check"></i><b>4.5.2</b> Abundances of specific taxa in all samples</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="taxonomic-information.html"><a href="taxonomic-information.html#session-info-2"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>5</b> Quality Control</a>
<ul>
<li class="chapter" data-level="5.1" data-path="quality-control.html"><a href="quality-control.html#get-top-taxa-and-taxonomy"><i class="fa fa-check"></i><b>5.1</b> Get top taxa and taxonomy</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="quality-control.html"><a href="quality-control.html#features"><i class="fa fa-check"></i><b>5.1.1</b> Features</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="quality-control.html"><a href="quality-control.html#library-size"><i class="fa fa-check"></i><b>5.2</b> Library size</a></li>
<li class="chapter" data-level="" data-path="quality-control.html"><a href="quality-control.html#session-info-3"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html"><i class="fa fa-check"></i><b>6</b> Microbiome Exploration</a>
<ul>
<li class="chapter" data-level="6.1" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#prevalence"><i class="fa fa-check"></i><b>6.1</b> Prevalence</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#prevalent-microbiota-analysis"><i class="fa fa-check"></i><b>6.1.1</b> Prevalent microbiota analysis</a></li>
<li class="chapter" data-level="6.1.2" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#plotting-prevalence"><i class="fa fa-check"></i><b>6.1.2</b> Plotting prevalence</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#abundance"><i class="fa fa-check"></i><b>6.2</b> Abundance</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="alpha-diversity.html"><a href="alpha-diversity.html"><i class="fa fa-check"></i><b>7</b> Alpha Diversity</a>
<ul>
<li class="chapter" data-level="7.1" data-path="alpha-diversity.html"><a href="alpha-diversity.html#estimating-alpha-diversity"><i class="fa fa-check"></i><b>7.1</b> Estimating alpha diversity</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="alpha-diversity.html"><a href="alpha-diversity.html#richness"><i class="fa fa-check"></i><b>7.1.1</b> Richness</a></li>
<li class="chapter" data-level="7.1.2" data-path="alpha-diversity.html"><a href="alpha-diversity.html#diversity"><i class="fa fa-check"></i><b>7.1.2</b> Diversity</a></li>
<li class="chapter" data-level="7.1.3" data-path="alpha-diversity.html"><a href="alpha-diversity.html#evenness"><i class="fa fa-check"></i><b>7.1.3</b> Evenness</a></li>
<li class="chapter" data-level="7.1.4" data-path="alpha-diversity.html"><a href="alpha-diversity.html#dominance"><i class="fa fa-check"></i><b>7.1.4</b> Dominance</a></li>
<li class="chapter" data-level="7.1.5" data-path="alpha-diversity.html"><a href="alpha-diversity.html#rarity"><i class="fa fa-check"></i><b>7.1.5</b> Rarity</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="alpha-diversity.html"><a href="alpha-diversity.html#visualize-alpha-diversities"><i class="fa fa-check"></i><b>7.2</b> Visualize alpha diversities</a></li>
<li class="chapter" data-level="" data-path="alpha-diversity.html"><a href="alpha-diversity.html#session-info-4"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="beta-diversity.html"><a href="beta-diversity.html"><i class="fa fa-check"></i><b>8</b> Beta Diversity</a>
<ul>
<li class="chapter" data-level="8.1" data-path="beta-diversity.html"><a href="beta-diversity.html#explained-variance"><i class="fa fa-check"></i><b>8.1</b> Explained variance</a></li>
<li class="chapter" data-level="8.2" data-path="beta-diversity.html"><a href="beta-diversity.html#estimating-beta-diversity"><i class="fa fa-check"></i><b>8.2</b> Estimating beta diversity</a></li>
<li class="chapter" data-level="8.3" data-path="beta-diversity.html"><a href="beta-diversity.html#other-ordination-methods"><i class="fa fa-check"></i><b>8.3</b> Other ordination methods</a></li>
<li class="chapter" data-level="8.4" data-path="beta-diversity.html"><a href="beta-diversity.html#visualizing-the-most-dominant-genus-on-pcoa"><i class="fa fa-check"></i><b>8.4</b> Visualizing the most dominant genus on PCoA</a></li>
<li class="chapter" data-level="8.5" data-path="beta-diversity.html"><a href="beta-diversity.html#community-comparisons-todo-combine-with-the-material-above-for-simplicity"><i class="fa fa-check"></i><b>8.5</b> Community comparisons [TODO combine with the material above for simplicity?]</a>
<ul>
<li class="chapter" data-level="8.5.1" data-path="beta-diversity.html"><a href="beta-diversity.html#testing-differences-in-community-composition-between-sample-groups"><i class="fa fa-check"></i><b>8.5.1</b> Testing differences in community composition between sample groups</a></li>
<li class="chapter" data-level="8.5.2" data-path="beta-diversity.html"><a href="beta-diversity.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>8.5.2</b> Checking the homogeneity condition</a></li>
</ul></li>
<li class="chapter" data-level="8.6" data-path="beta-diversity.html"><a href="beta-diversity.html#further-reading-1"><i class="fa fa-check"></i><b>8.6</b> Further reading</a></li>
<li class="chapter" data-level="" data-path="beta-diversity.html"><a href="beta-diversity.html#session-info-5"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="microbiome-community.html"><a href="microbiome-community.html"><i class="fa fa-check"></i><b>9</b> Microbiome Community</a>
<ul>
<li class="chapter" data-level="9.1" data-path="microbiome-community.html"><a href="microbiome-community.html#community-composition"><i class="fa fa-check"></i><b>9.1</b> Community composition</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="microbiome-community.html"><a href="microbiome-community.html#composition-barplot"><i class="fa fa-check"></i><b>9.1.1</b> Composition barplot</a></li>
<li class="chapter" data-level="9.1.2" data-path="microbiome-community.html"><a href="microbiome-community.html#composition-heatmap"><i class="fa fa-check"></i><b>9.1.2</b> Composition heatmap</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="microbiome-community.html"><a href="microbiome-community.html#community-typing"><i class="fa fa-check"></i><b>9.2</b> Community typing</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="microbiome-community.html"><a href="microbiome-community.html#dirichlet-multinomial-mixtures-dmm"><i class="fa fa-check"></i><b>9.2.1</b> Dirichlet Multinomial Mixtures (DMM)</a></li>
<li class="chapter" data-level="9.2.2" data-path="microbiome-community.html"><a href="microbiome-community.html#pcoa-for-asv-level-data-with-bray-curtis-with-dmm-clusters-shown-with-colors"><i class="fa fa-check"></i><b>9.2.2</b> PCoA for ASV-level data with Bray-Curtis; with DMM clusters shown with colors</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="microbiome-community.html"><a href="microbiome-community.html#community-detection"><i class="fa fa-check"></i><b>9.3</b> Community Detection</a></li>
<li class="chapter" data-level="" data-path="microbiome-community.html"><a href="microbiome-community.html#session-info-6"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="differential-abundance.html"><a href="differential-abundance.html"><i class="fa fa-check"></i><b>10</b> Differential Abundance</a>
<ul>
<li class="chapter" data-level="10.1" data-path="differential-abundance.html"><a href="differential-abundance.html#tree-based-methods"><i class="fa fa-check"></i><b>10.1</b> Tree-based methods</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="differential-abundance.html"><a href="differential-abundance.html#group-wise-associations-testing-based-on-balances"><i class="fa fa-check"></i><b>10.1.1</b> Group-wise associations testing based on balances</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="differential-abundance.html"><a href="differential-abundance.html#session-info-7"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="microbiome-timeseries.html"><a href="microbiome-timeseries.html"><i class="fa fa-check"></i><b>11</b> Microbiome Time Series</a>
<ul>
<li class="chapter" data-level="11.1" data-path="microbiome-timeseries.html"><a href="microbiome-timeseries.html#stability"><i class="fa fa-check"></i><b>11.1</b> Stability</a></li>
<li class="chapter" data-level="11.2" data-path="microbiome-timeseries.html"><a href="microbiome-timeseries.html#tipping-elements"><i class="fa fa-check"></i><b>11.2</b> Tipping elements</a></li>
<li class="chapter" data-level="" data-path="microbiome-timeseries.html"><a href="microbiome-timeseries.html#session-info-8"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="multitable.html"><a href="multitable.html"><i class="fa fa-check"></i><b>12</b> Representation of Multiple Data Tables</a>
<ul>
<li class="chapter" data-level="12.1" data-path="multitable.html"><a href="multitable.html#assay-data"><i class="fa fa-check"></i><b>12.1</b> Assay data</a></li>
<li class="chapter" data-level="12.2" data-path="multitable.html"><a href="multitable.html#alternative-experiments"><i class="fa fa-check"></i><b>12.2</b> Alternative experiments</a></li>
<li class="chapter" data-level="12.3" data-path="multitable.html"><a href="multitable.html#multiassayexperiments"><i class="fa fa-check"></i><b>12.3</b> MultiAssayExperiments</a></li>
<li class="chapter" data-level="" data-path="multitable.html"><a href="multitable.html#session-info-9"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="part"><span><b>IV Appendix</b></span></li>
<li class="chapter" data-level="13" data-path="example-data.html"><a href="example-data.html"><i class="fa fa-check"></i><b>13</b> Example Data</a>
<ul>
<li class="chapter" data-level="13.1" data-path="example-data.html"><a href="example-data.html#package-data"><i class="fa fa-check"></i><b>13.1</b> Package data</a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="example-data.html"><a href="example-data.html#globalpatterns"><i class="fa fa-check"></i><b>13.1.1</b> GlobalPatterns</a></li>
<li class="chapter" data-level="13.1.2" data-path="example-data.html"><a href="example-data.html#enterotype"><i class="fa fa-check"></i><b>13.1.2</b> Enterotype</a></li>
<li class="chapter" data-level="13.1.3" data-path="example-data.html"><a href="example-data.html#esophagus"><i class="fa fa-check"></i><b>13.1.3</b> Esophagus</a></li>
<li class="chapter" data-level="13.1.4" data-path="example-data.html"><a href="example-data.html#soilrep"><i class="fa fa-check"></i><b>13.1.4</b> Soilrep</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="example-data.html"><a href="example-data.html#session-info-10"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="part"><span><b>V Appendix</b></span></li>
<li class="chapter" data-level="14" data-path="function-ref.html"><a href="function-ref.html"><i class="fa fa-check"></i><b>14</b> Function Reference</a>
<ul>
<li class="chapter" data-level="14.1" data-path="function-ref.html"><a href="function-ref.html#objects"><i class="fa fa-check"></i><b>14.1</b> Objects</a></li>
<li class="chapter" data-level="14.2" data-path="function-ref.html"><a href="function-ref.html#reduced-dimension-pca-t-sne-umap-mds-etc"><i class="fa fa-check"></i><b>14.2</b> Reduced dimension (PCA, t-SNE, UMAP, MDS, etc)</a></li>
<li class="chapter" data-level="14.3" data-path="function-ref.html"><a href="function-ref.html#plotting"><i class="fa fa-check"></i><b>14.3</b> Plotting</a></li>
</ul></li>
<li class="part"><span><b>VI Appendix</b></span></li>
<li class="chapter" data-level="15" data-path="extras.html"><a href="extras.html"><i class="fa fa-check"></i><b>15</b> Extras</a>
<ul>
<li class="chapter" data-level="15.1" data-path="extras.html"><a href="extras.html#interactive-3d-plots"><i class="fa fa-check"></i><b>15.1</b> Interactive 3D Plots</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="bibliography.html"><a href="bibliography.html"><i class="fa fa-check"></i>Bibliography</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Orchestrating Microbiome Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="quality-control" class="section level1" number="5">
<h1><span class="header-section-number">Chapter 5</span> Quality Control</h1>
<script>
document.addEventListener("click", function (event) {
    if (event.target.classList.contains("rebook-collapse")) {
        event.target.classList.toggle("active");
        var content = event.target.nextElementSibling;
        if (content.style.display === "block") {
            content.style.display = "none";
        } else {
            content.style.display = "block";
        }
    }
})
</script>
<style>
.rebook-collapse {
  background-color: #eee;
  color: #444;
  cursor: pointer;
  padding: 18px;
  width: 100%;
  border: none;
  text-align: left;
  outline: none;
  font-size: 15px;
}

.rebook-content {
  padding: 0 18px;
  display: none;
  overflow: hidden;
  background-color: #f1f1f1;
}
</style>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="quality-control.html#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mia)</span>
<span id="cb96-2"><a href="quality-control.html#cb96-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;GlobalPatterns&quot;</span>)</span>
<span id="cb96-3"><a href="quality-control.html#cb96-3" aria-hidden="true" tabindex="-1"></a>se <span class="ot">&lt;-</span> GlobalPatterns </span></code></pre></div>
<div id="get-top-taxa-and-taxonomy" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> Get top taxa and taxonomy</h2>
<div id="features" class="section level3" number="5.1.1">
<h3><span class="header-section-number">5.1.1</span> Features</h3>
<p>The <code>getTopTaxa</code> can be used for identifying top taxa in the data.</p>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="quality-control.html#cb97-1" aria-hidden="true" tabindex="-1"></a>top_features <span class="ot">&lt;-</span> <span class="fu">getTopTaxa</span>(se, <span class="at">method=</span><span class="st">&quot;median&quot;</span>, <span class="at">top=</span><span class="dv">10</span>)</span>
<span id="cb97-2"><a href="quality-control.html#cb97-2" aria-hidden="true" tabindex="-1"></a>tax_data <span class="ot">&lt;-</span> <span class="fu">rowData</span>(se)[top_features,<span class="fu">taxonomyRanks</span>(se)]</span>
<span id="cb97-3"><a href="quality-control.html#cb97-3" aria-hidden="true" tabindex="-1"></a>tax_data</span></code></pre></div>
<pre><code>## DataFrame with 10 rows and 7 columns
##            Kingdom         Phylum               Class             Order
##        &lt;character&gt;    &lt;character&gt;         &lt;character&gt;       &lt;character&gt;
## 549656    Bacteria  Cyanobacteria         Chloroplast     Stramenopiles
## 331820    Bacteria  Bacteroidetes         Bacteroidia     Bacteroidales
## 317182    Bacteria  Cyanobacteria         Chloroplast     Stramenopiles
## 94166     Bacteria Proteobacteria Gammaproteobacteria    Pasteurellales
## 279599    Bacteria  Cyanobacteria    Nostocophycideae        Nostocales
## 158660    Bacteria  Bacteroidetes         Bacteroidia     Bacteroidales
## 329744    Bacteria Actinobacteria      Actinobacteria   Actinomycetales
## 326977    Bacteria Actinobacteria      Actinobacteria Bifidobacteriales
## 248140    Bacteria  Bacteroidetes         Bacteroidia     Bacteroidales
## 550960    Bacteria Proteobacteria Gammaproteobacteria Enterobacteriales
##                    Family           Genus                Species
##               &lt;character&gt;     &lt;character&gt;            &lt;character&gt;
## 549656                 NA              NA                     NA
## 331820     Bacteroidaceae     Bacteroides                     NA
## 317182                 NA              NA                     NA
## 94166     Pasteurellaceae     Haemophilus Haemophilusparainflu..
## 279599        Nostocaceae  Dolichospermum                     NA
## 158660     Bacteroidaceae     Bacteroides                     NA
## 329744             ACK-M1              NA                     NA
## 326977 Bifidobacteriaceae Bifidobacterium Bifidobacteriumadole..
## 248140     Bacteroidaceae     Bacteroides      Bacteroidescaccae
## 550960 Enterobacteriaceae     Providencia                     NA</code></pre>
</div>
</div>
<div id="library-size" class="section level2" number="5.2">
<h2><span class="header-section-number">5.2</span> Library size</h2>
<p>The total counts/sample can be calculated using the
<code>perCellQCMetrics</code>/<code>addPerCellQC</code> from the <code>scater</code> package. The first one
just calculates the values, whereas the latter one directly adds them to the
<code>colData</code>.</p>
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="quality-control.html#cb99-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scater)</span>
<span id="cb99-2"><a href="quality-control.html#cb99-2" aria-hidden="true" tabindex="-1"></a><span class="fu">perCellQCMetrics</span>(se)</span></code></pre></div>
<pre><code>## DataFrame with 26 rows and 3 columns
##               sum  detected     total
##         &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
## CL3        864077      6964    864077
## CC1       1135457      7679   1135457
## SV1        697509      5729    697509
## M31Fcsw   1543451      2667   1543451
## M11Fcsw   2076476      2574   2076476
## ...           ...       ...       ...
## TS28       937466      2679    937466
## TS29      1211071      2629   1211071
## Even1     1216137      4213   1216137
## Even2      971073      3130    971073
## Even3     1078241      2776   1078241</code></pre>
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb101-1"><a href="quality-control.html#cb101-1" aria-hidden="true" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">addPerCellQC</span>(se)</span>
<span id="cb101-2"><a href="quality-control.html#cb101-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(se)</span></code></pre></div>
<pre><code>## DataFrame with 26 rows and 10 columns
##         X.SampleID   Primer Final_Barcode Barcode_truncated_plus_T
##           &lt;factor&gt; &lt;factor&gt;      &lt;factor&gt;                 &lt;factor&gt;
## CL3        CL3      ILBC_01        AACGCA                   TGCGTT
## CC1        CC1      ILBC_02        AACTCG                   CGAGTT
## SV1        SV1      ILBC_03        AACTGT                   ACAGTT
## M31Fcsw    M31Fcsw  ILBC_04        AAGAGA                   TCTCTT
## M11Fcsw    M11Fcsw  ILBC_05        AAGCTG                   CAGCTT
## ...            ...      ...           ...                      ...
## TS28         TS28   ILBC_25        ACCAGA                   TCTGGT
## TS29         TS29   ILBC_26        ACCAGC                   GCTGGT
## Even1        Even1  ILBC_27        ACCGCA                   TGCGGT
## Even2        Even2  ILBC_28        ACCTCG                   CGAGGT
## Even3        Even3  ILBC_29        ACCTGT                   ACAGGT
##         Barcode_full_length SampleType
##                    &lt;factor&gt;   &lt;factor&gt;
## CL3             CTAGCGTGCGT      Soil 
## CC1             CATCGACGAGT      Soil 
## SV1             GTACGCACAGT      Soil 
## M31Fcsw         TCGACATCTCT      Feces
## M11Fcsw         CGACTGCAGCT      Feces
## ...                     ...        ...
## TS28            GCATCGTCTGG      Feces
## TS29            CTAGTCGCTGG      Feces
## Even1           TGACTCTGCGG      Mock 
## Even2           TCTGATCGAGG      Mock 
## Even3           AGAGAGACAGG      Mock 
##                                        Description       sum  detected
##                                           &lt;factor&gt; &lt;numeric&gt; &lt;numeric&gt;
## CL3     Calhoun South Carolina Pine soil, pH 4.9      864077      6964
## CC1     Cedar Creek Minnesota, grassland, pH 6.1     1135457      7679
## SV1     Sevilleta new Mexico, desert scrub, pH 8.3    697509      5729
## M31Fcsw M3, Day 1, fecal swab, whole body study      1543451      2667
## M11Fcsw M1, Day 1, fecal swab, whole body study      2076476      2574
## ...                                            ...       ...       ...
## TS28                                       Twin #1    937466      2679
## TS29                                       Twin #2   1211071      2629
## Even1                                      Even1     1216137      4213
## Even2                                      Even2      971073      3130
## Even3                                      Even3     1078241      2776
##             total
##         &lt;numeric&gt;
## CL3        864077
## CC1       1135457
## SV1        697509
## M31Fcsw   1543451
## M11Fcsw   2076476
## ...           ...
## TS28       937466
## TS29      1211071
## Even1     1216137
## Even2      971073
## Even3     1078241</code></pre>
<p>The distribution of calculated library sizes can be visualized as a
histogram (left), or by sorting the samples by library size (right).</p>
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb103-1"><a href="quality-control.html#cb103-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb103-2"><a href="quality-control.html#cb103-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-3"><a href="quality-control.html#cb103-3" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">as.data.frame</span>(<span class="fu">colData</span>(se))) <span class="sc">+</span></span>
<span id="cb103-4"><a href="quality-control.html#cb103-4" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x =</span> sum), <span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;gray&quot;</span>, <span class="at">bins =</span> <span class="dv">30</span>) <span class="sc">+</span></span>
<span id="cb103-5"><a href="quality-control.html#cb103-5" aria-hidden="true" tabindex="-1"></a>        <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Library size&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Frequency (n)&quot;</span>) <span class="sc">+</span> </span>
<span id="cb103-6"><a href="quality-control.html#cb103-6" aria-hidden="true" tabindex="-1"></a>        <span class="co"># scale_x_log10(breaks = scales::trans_breaks(&quot;log10&quot;, function(x) 10^x), </span></span>
<span id="cb103-7"><a href="quality-control.html#cb103-7" aria-hidden="true" tabindex="-1"></a>        <span class="co"># labels = scales::trans_format(&quot;log10&quot;, scales::math_format(10^.x))) +</span></span>
<span id="cb103-8"><a href="quality-control.html#cb103-8" aria-hidden="true" tabindex="-1"></a>        <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb103-9"><a href="quality-control.html#cb103-9" aria-hidden="true" tabindex="-1"></a>        <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(), <span class="co"># Removes the grid</span></span>
<span id="cb103-10"><a href="quality-control.html#cb103-10" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb103-11"><a href="quality-control.html#cb103-11" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb103-12"><a href="quality-control.html#cb103-12" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb103-13"><a href="quality-control.html#cb103-13" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">&quot;black&quot;</span>)) <span class="co"># Adds y-axis</span></span>
<span id="cb103-14"><a href="quality-control.html#cb103-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-15"><a href="quality-control.html#cb103-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb103-16"><a href="quality-control.html#cb103-16" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(se)) <span class="sc">%&gt;%</span></span>
<span id="cb103-17"><a href="quality-control.html#cb103-17" aria-hidden="true" tabindex="-1"></a>        <span class="fu">arrange</span>(sum) <span class="sc">%&gt;%</span></span>
<span id="cb103-18"><a href="quality-control.html#cb103-18" aria-hidden="true" tabindex="-1"></a>        <span class="fu">mutate</span>(<span class="at">index =</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">n</span>())</span>
<span id="cb103-19"><a href="quality-control.html#cb103-19" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">y =</span> index, <span class="at">x =</span> sum<span class="sc">/</span><span class="fl">1e6</span>)) <span class="sc">+</span></span>
<span id="cb103-20"><a href="quality-control.html#cb103-20" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_point</span>() <span class="sc">+</span>  </span>
<span id="cb103-21"><a href="quality-control.html#cb103-21" aria-hidden="true" tabindex="-1"></a>        <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Library size (million reads)&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Sample index&quot;</span>) <span class="sc">+</span>  </span>
<span id="cb103-22"><a href="quality-control.html#cb103-22" aria-hidden="true" tabindex="-1"></a>        <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb103-23"><a href="quality-control.html#cb103-23" aria-hidden="true" tabindex="-1"></a>        <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(), <span class="co"># Removes the grid</span></span>
<span id="cb103-24"><a href="quality-control.html#cb103-24" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb103-25"><a href="quality-control.html#cb103-25" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb103-26"><a href="quality-control.html#cb103-26" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb103-27"><a href="quality-control.html#cb103-27" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">&quot;black&quot;</span>)) <span class="co"># Adds y-axis</span></span>
<span id="cb103-28"><a href="quality-control.html#cb103-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-29"><a href="quality-control.html#cb103-29" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb103-30"><a href="quality-control.html#cb103-30" aria-hidden="true" tabindex="-1"></a>p1 <span class="sc">+</span> p2</span></code></pre></div>
<div class="figure"><span id="fig:plot-viz-lib-size-1"></span>
<img src="12_quality_control_files/figure-html/plot-viz-lib-size-1-1.png" alt="Library size distribution." width="768" />
<p class="caption">
Figure 5.1: Library size distribution.
</p>
</div>
<p>Library sizes - and other variables from <code>colData</code> - can be also visualized by using
specified function called <code>plotColData</code>.</p>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="quality-control.html#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb104-2"><a href="quality-control.html#cb104-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Sort samples by read count, order the factor levels, and store back to se as DataFrame</span></span>
<span id="cb104-3"><a href="quality-control.html#cb104-3" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">TODO</span><span class="co">: plotColData could include an option for sorting samples based on colData variables</span></span>
<span id="cb104-4"><a href="quality-control.html#cb104-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(se) <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(se)) <span class="sc">%&gt;%</span></span>
<span id="cb104-5"><a href="quality-control.html#cb104-5" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">arrange</span>(X.SampleID) <span class="sc">%&gt;%</span></span>
<span id="cb104-6"><a href="quality-control.html#cb104-6" aria-hidden="true" tabindex="-1"></a>             <span class="fu">mutate</span>(<span class="at">X.SampleID =</span> <span class="fu">factor</span>(X.SampleID, <span class="at">levels=</span>X.SampleID)) <span class="sc">%&gt;%</span></span>
<span id="cb104-7"><a href="quality-control.html#cb104-7" aria-hidden="true" tabindex="-1"></a>         DataFrame</span>
<span id="cb104-8"><a href="quality-control.html#cb104-8" aria-hidden="true" tabindex="-1"></a><span class="fu">plotColData</span>(se,<span class="st">&quot;sum&quot;</span>,<span class="st">&quot;X.SampleID&quot;</span>, <span class="at">colour_by =</span> <span class="st">&quot;SampleType&quot;</span>) <span class="sc">+</span> </span>
<span id="cb104-9"><a href="quality-control.html#cb104-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust=</span><span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb104-10"><a href="quality-control.html#cb104-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&quot;Library size (N)&quot;</span>, <span class="at">x =</span> <span class="st">&quot;Sample ID&quot;</span>)       </span></code></pre></div>
<div class="figure"><span id="fig:plot-viz-lib-size-2"></span>
<img src="12_quality_control_files/figure-html/plot-viz-lib-size-2-1.png" alt="Library sizes per sample." width="768" />
<p class="caption">
Figure 5.2: Library sizes per sample.
</p>
</div>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="quality-control.html#cb105-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotColData</span>(se,<span class="st">&quot;sum&quot;</span>,<span class="st">&quot;SampleType&quot;</span>, <span class="at">colour_by =</span> <span class="st">&quot;SampleType&quot;</span>) <span class="sc">+</span> </span>
<span id="cb105-2"><a href="quality-control.html#cb105-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust=</span><span class="dv">1</span>))</span></code></pre></div>
<div class="figure"><span id="fig:plot-viz-lib-size-3"></span>
<img src="12_quality_control_files/figure-html/plot-viz-lib-size-3-1.png" alt="Library sizes per sample type." width="768" />
<p class="caption">
Figure 5.3: Library sizes per sample type.
</p>
</div>
</div>
<div id="session-info-3" class="section level2 unnumbered">
<h2>Session Info</h2>
<button class="rebook-collapse">
View session info
</button>
<div class="rebook-content">
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.2 LTS

Matrix products: default
BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] patchwork_1.1.1                dplyr_1.0.7                   
 [3] scater_1.21.3                  ggplot2_3.3.5                 
 [5] scuttle_1.3.1                  mia_1.1.9                     
 [7] TreeSummarizedExperiment_2.1.3 Biostrings_2.61.2             
 [9] XVector_0.33.0                 SingleCellExperiment_1.15.1   
[11] SummarizedExperiment_1.23.1    Biobase_2.53.0                
[13] GenomicRanges_1.45.0           GenomeInfoDb_1.29.3           
[15] IRanges_2.27.0                 S4Vectors_0.31.0              
[17] BiocGenerics_0.39.1            MatrixGenerics_1.5.3          
[19] matrixStats_0.60.0             BiocStyle_2.21.3              
[21] rebook_1.3.0                  

loaded via a namespace (and not attached):
  [1] ggbeeswarm_0.6.0            colorspace_2.0-2           
  [3] ellipsis_0.3.2              BiocNeighbors_1.11.0       
  [5] farver_2.1.0                ggrepel_0.9.1              
  [7] bit64_4.0.5                 fansi_0.5.0                
  [9] decontam_1.13.0             splines_4.1.0              
 [11] codetools_0.2-18            sparseMatrixStats_1.5.2    
 [13] cachem_1.0.5                knitr_1.33                 
 [15] jsonlite_1.7.2              cluster_2.1.2              
 [17] graph_1.71.2                BiocManager_1.30.16        
 [19] compiler_4.1.0              assertthat_0.2.1           
 [21] Matrix_1.3-4                fastmap_1.1.0              
 [23] lazyeval_0.2.2              BiocSingular_1.9.1         
 [25] htmltools_0.5.1.1           tools_4.1.0                
 [27] rsvd_1.0.5                  gtable_0.3.0               
 [29] glue_1.4.2                  GenomeInfoDbData_1.2.6     
 [31] reshape2_1.4.4              Rcpp_1.0.7                 
 [33] jquerylib_0.1.4             vctrs_0.3.8                
 [35] ape_5.5                     nlme_3.1-152               
 [37] DECIPHER_2.21.0             DelayedMatrixStats_1.15.2  
 [39] xfun_0.25                   stringr_1.4.0              
 [41] beachmat_2.9.0              lifecycle_1.0.0            
 [43] irlba_2.3.3                 XML_3.99-0.6               
 [45] zlibbioc_1.39.0             MASS_7.3-54                
 [47] scales_1.1.1                parallel_4.1.0             
 [49] yaml_2.2.1                  memoise_2.0.0              
 [51] gridExtra_2.3               sass_0.4.0                 
 [53] stringi_1.7.3               RSQLite_2.2.7              
 [55] highr_0.9                   ScaledMatrix_1.1.0         
 [57] tidytree_0.3.4              permute_0.9-5              
 [59] filelock_1.0.2              BiocParallel_1.27.3        
 [61] rlang_0.4.11                pkgconfig_2.0.3            
 [63] bitops_1.0-7                evaluate_0.14              
 [65] lattice_0.20-44             purrr_0.3.4                
 [67] labeling_0.4.2              treeio_1.17.2              
 [69] CodeDepends_0.6.5           cowplot_1.1.1              
 [71] bit_4.0.4                   tidyselect_1.1.1           
 [73] plyr_1.8.6                  magrittr_2.0.1             
 [75] bookdown_0.22               R6_2.5.0                   
 [77] generics_0.1.0              DelayedArray_0.19.1        
 [79] DBI_1.1.1                   withr_2.4.2                
 [81] mgcv_1.8-36                 pillar_1.6.2               
 [83] RCurl_1.98-1.3              tibble_3.1.3               
 [85] dir.expiry_1.1.0            crayon_1.4.1               
 [87] utf8_1.2.2                  rmarkdown_2.10             
 [89] viridis_0.6.1               grid_4.1.0                 
 [91] blob_1.2.2                  vegan_2.5-7                
 [93] digest_0.6.27               tidyr_1.1.3                
 [95] munsell_0.5.0               DirichletMultinomial_1.35.0
 [97] beeswarm_0.4.0              viridisLite_0.4.0          
 [99] vipor_0.4.5                 bslib_0.2.5.1              </code></pre>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="taxonomic-information.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="microbiome-exploration.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["OMA.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
