<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Exploration and quality Control | Orchestrating Microbiome Analysis</title>
  <meta name="description" content="Chapter 5 Exploration and quality Control | Orchestrating Microbiome Analysis" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Exploration and quality Control | Orchestrating Microbiome Analysis" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://github.com/microbiome/OMA" />
  
  
  <meta name="github-repo" content="microbiome/OMA" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Exploration and quality Control | Orchestrating Microbiome Analysis" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="datamanipulation.html"/>
<link rel="next" href="taxonomic-information.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<script src="libs/plotly-binding-4.10.0/plotly.js"></script>
<script src="libs/typedarray-0.1/typedarray.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="libs/plotly-htmlwidgets-css-2.5.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="libs/plotly-main-2.5.1/plotly-latest.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Orchestrating Microbiome Analysis</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="part"><span><b>I Introduction</b></span></li>
<li class="chapter" data-level="1" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="containers.html"><a href="containers.html"><i class="fa fa-check"></i><b>2</b> Microbiome Data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="containers.html"><a href="containers.html#data-science-framework"><i class="fa fa-check"></i><b>2.1</b> Data science framework</a></li>
<li class="chapter" data-level="2.2" data-path="containers.html"><a href="containers.html#data-containers"><i class="fa fa-check"></i><b>2.2</b> Data containers</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="containers.html"><a href="containers.html#assay-data"><i class="fa fa-check"></i><b>2.2.1</b> Assay data</a></li>
<li class="chapter" data-level="2.2.2" data-path="containers.html"><a href="containers.html#coldata"><i class="fa fa-check"></i><b>2.2.2</b> colData</a></li>
<li class="chapter" data-level="2.2.3" data-path="containers.html"><a href="containers.html#rowdata"><i class="fa fa-check"></i><b>2.2.3</b> rowData</a></li>
<li class="chapter" data-level="2.2.4" data-path="containers.html"><a href="containers.html#rowtree"><i class="fa fa-check"></i><b>2.2.4</b> rowTree</a></li>
<li class="chapter" data-level="2.2.5" data-path="containers.html"><a href="containers.html#alternative-experiments"><i class="fa fa-check"></i><b>2.2.5</b> Alternative experiments</a></li>
<li class="chapter" data-level="2.2.6" data-path="containers.html"><a href="containers.html#multiassayexperiments"><i class="fa fa-check"></i><b>2.2.6</b> MultiAssayExperiments</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="containers.html"><a href="containers.html#loading-experimental-microbiome-data"><i class="fa fa-check"></i><b>2.3</b> Loading experimental microbiome data</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="containers.html"><a href="containers.html#import-from-external-files"><i class="fa fa-check"></i><b>2.3.1</b> Import from external files</a></li>
<li class="chapter" data-level="2.3.2" data-path="containers.html"><a href="containers.html#conversions-between-data-formats-in-r"><i class="fa fa-check"></i><b>2.3.2</b> Conversions between data formats in R</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="containers.html"><a href="containers.html#example-data"><i class="fa fa-check"></i><b>2.4</b> Demonstration data</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="containers.html"><a href="containers.html#package-data"><i class="fa fa-check"></i><b>2.4.1</b> Package data</a></li>
<li class="chapter" data-level="2.4.2" data-path="containers.html"><a href="containers.html#experimenthub-data"><i class="fa fa-check"></i><b>2.4.2</b> ExperimentHub data</a></li>
<li class="chapter" data-level="2.4.3" data-path="containers.html"><a href="containers.html#other-data-sources"><i class="fa fa-check"></i><b>2.4.3</b> Other data sources</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="containers.html"><a href="containers.html#session-info"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="packages.html"><a href="packages.html"><i class="fa fa-check"></i><b>3</b> Packages</a>
<ul>
<li class="chapter" data-level="3.1" data-path="packages.html"><a href="packages.html#package-installation"><i class="fa fa-check"></i><b>3.1</b> Package installation</a></li>
<li class="chapter" data-level="3.2" data-path="packages.html"><a href="packages.html#some-available-packages"><i class="fa fa-check"></i><b>3.2</b> Some available packages</a></li>
<li class="chapter" data-level="" data-path="packages.html"><a href="packages.html#session-info-1"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="part"><span><b>II Focus Topics</b></span></li>
<li class="chapter" data-level="4" data-path="datamanipulation.html"><a href="datamanipulation.html"><i class="fa fa-check"></i><b>4</b> Data Manipulation</a>
<ul>
<li class="chapter" data-level="4.1" data-path="datamanipulation.html"><a href="datamanipulation.html#tidying-and-subsetting"><i class="fa fa-check"></i><b>4.1</b> Tidying and subsetting</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="datamanipulation.html"><a href="datamanipulation.html#tidy-data"><i class="fa fa-check"></i><b>4.1.1</b> Tidy data</a></li>
<li class="chapter" data-level="4.1.2" data-path="datamanipulation.html"><a href="datamanipulation.html#subsetting"><i class="fa fa-check"></i><b>4.1.2</b> Subsetting</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>5</b> Exploration and quality Control</a>
<ul>
<li class="chapter" data-level="5.1" data-path="quality-control.html"><a href="quality-control.html#abundance"><i class="fa fa-check"></i><b>5.1</b> Abundance</a></li>
<li class="chapter" data-level="5.2" data-path="quality-control.html"><a href="quality-control.html#prevalence"><i class="fa fa-check"></i><b>5.2</b> Prevalence</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="quality-control.html"><a href="quality-control.html#prevalent-microbiota-analysis"><i class="fa fa-check"></i><b>5.2.1</b> Prevalent microbiota analysis</a></li>
<li class="chapter" data-level="5.2.2" data-path="quality-control.html"><a href="quality-control.html#plotting-prevalence"><i class="fa fa-check"></i><b>5.2.2</b> Plotting prevalence</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="quality-control.html"><a href="quality-control.html#quality-control-1"><i class="fa fa-check"></i><b>5.3</b> Quality control</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="quality-control.html"><a href="quality-control.html#top-taxa"><i class="fa fa-check"></i><b>5.3.1</b> Top taxa</a></li>
<li class="chapter" data-level="5.3.2" data-path="quality-control.html"><a href="quality-control.html#library-size"><i class="fa fa-check"></i><b>5.3.2</b> Library size</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="quality-control.html"><a href="quality-control.html#session-info-2"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="taxonomic-information.html"><a href="taxonomic-information.html"><i class="fa fa-check"></i><b>6</b> Taxonomic Information</a>
<ul>
<li class="chapter" data-level="6.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#assigning-taxonomic-information."><i class="fa fa-check"></i><b>6.1</b> Assigning taxonomic information.</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#dada2"><i class="fa fa-check"></i><b>6.1.1</b> dada2</a></li>
<li class="chapter" data-level="6.1.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#decipher"><i class="fa fa-check"></i><b>6.1.2</b> DECIPHER</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#functions-to-access-taxonomic-information"><i class="fa fa-check"></i><b>6.2</b> Functions to access taxonomic information</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#generate-a-taxonomic-tree-on-the-fly"><i class="fa fa-check"></i><b>6.2.1</b> Generate a taxonomic tree on the fly</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="taxonomic-information.html"><a href="taxonomic-information.html#data-agglomeration"><i class="fa fa-check"></i><b>6.3</b> Data agglomeration</a></li>
<li class="chapter" data-level="6.4" data-path="taxonomic-information.html"><a href="taxonomic-information.html#data-transformation"><i class="fa fa-check"></i><b>6.4</b> Data transformation</a></li>
<li class="chapter" data-level="6.5" data-path="taxonomic-information.html"><a href="taxonomic-information.html#pick-specific"><i class="fa fa-check"></i><b>6.5</b> Pick specific</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#abundances-of-all-taxa-in-specific-sample"><i class="fa fa-check"></i><b>6.5.1</b> Abundances of all taxa in specific sample</a></li>
<li class="chapter" data-level="6.5.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#abundances-of-specific-taxa-in-all-samples"><i class="fa fa-check"></i><b>6.5.2</b> Abundances of specific taxa in all samples</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="taxonomic-information.html"><a href="taxonomic-information.html#session-info-3"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="community-diversity.html"><a href="community-diversity.html"><i class="fa fa-check"></i><b>7</b> Community diversity</a>
<ul>
<li class="chapter" data-level="7.1" data-path="community-diversity.html"><a href="community-diversity.html#estimation"><i class="fa fa-check"></i><b>7.1</b> Estimation</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="community-diversity.html"><a href="community-diversity.html#richness"><i class="fa fa-check"></i><b>7.1.1</b> Richness</a></li>
<li class="chapter" data-level="7.1.2" data-path="community-diversity.html"><a href="community-diversity.html#diversity"><i class="fa fa-check"></i><b>7.1.2</b> Diversity</a></li>
<li class="chapter" data-level="7.1.3" data-path="community-diversity.html"><a href="community-diversity.html#evenness"><i class="fa fa-check"></i><b>7.1.3</b> Evenness</a></li>
<li class="chapter" data-level="7.1.4" data-path="community-diversity.html"><a href="community-diversity.html#dominance"><i class="fa fa-check"></i><b>7.1.4</b> Dominance</a></li>
<li class="chapter" data-level="7.1.5" data-path="community-diversity.html"><a href="community-diversity.html#rarity"><i class="fa fa-check"></i><b>7.1.5</b> Rarity</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="community-diversity.html"><a href="community-diversity.html#visualization"><i class="fa fa-check"></i><b>7.2</b> Visualization</a></li>
<li class="chapter" data-level="" data-path="community-diversity.html"><a href="community-diversity.html#session-info-4"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="community-similarity.html"><a href="community-similarity.html"><i class="fa fa-check"></i><b>8</b> Community similarity</a>
<ul>
<li class="chapter" data-level="8.1" data-path="community-similarity.html"><a href="community-similarity.html#explained-variance"><i class="fa fa-check"></i><b>8.1</b> Explained variance</a></li>
<li class="chapter" data-level="8.2" data-path="community-similarity.html"><a href="community-similarity.html#community-comparisons-by-beta-diversity-analysis"><i class="fa fa-check"></i><b>8.2</b> Community comparisons by beta diversity analysis</a></li>
<li class="chapter" data-level="8.3" data-path="community-similarity.html"><a href="community-similarity.html#other-ordination-methods"><i class="fa fa-check"></i><b>8.3</b> Other ordination methods</a></li>
<li class="chapter" data-level="8.4" data-path="community-similarity.html"><a href="community-similarity.html#visualizing-the-most-dominant-genus-on-pcoa"><i class="fa fa-check"></i><b>8.4</b> Visualizing the most dominant genus on PCoA</a>
<ul>
<li class="chapter" data-level="8.4.1" data-path="community-similarity.html"><a href="community-similarity.html#testing-differences-in-community-composition-between-sample-groups"><i class="fa fa-check"></i><b>8.4.1</b> Testing differences in community composition between sample groups</a></li>
<li class="chapter" data-level="8.4.2" data-path="community-similarity.html"><a href="community-similarity.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>8.4.2</b> Checking the homogeneity condition</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="community-similarity.html"><a href="community-similarity.html#further-reading"><i class="fa fa-check"></i><b>8.5</b> Further reading</a></li>
<li class="chapter" data-level="" data-path="community-similarity.html"><a href="community-similarity.html#session-info-5"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="microbiome-community.html"><a href="microbiome-community.html"><i class="fa fa-check"></i><b>9</b> Community composition</a>
<ul>
<li class="chapter" data-level="9.1" data-path="microbiome-community.html"><a href="microbiome-community.html#visualizing-taxonomic-composition"><i class="fa fa-check"></i><b>9.1</b> Visualizing taxonomic composition</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="microbiome-community.html"><a href="microbiome-community.html#composition-barplot"><i class="fa fa-check"></i><b>9.1.1</b> Composition barplot</a></li>
<li class="chapter" data-level="9.1.2" data-path="microbiome-community.html"><a href="microbiome-community.html#composition-heatmap"><i class="fa fa-check"></i><b>9.1.2</b> Composition heatmap</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="community-typing.html"><a href="community-typing.html"><i class="fa fa-check"></i><b>10</b> Community typing</a>
<ul>
<li class="chapter" data-level="10.0.1" data-path="community-typing.html"><a href="community-typing.html#dirichlet-multinomial-mixtures-dmm"><i class="fa fa-check"></i><b>10.0.1</b> Dirichlet Multinomial Mixtures (DMM)</a></li>
<li class="chapter" data-level="10.0.2" data-path="community-typing.html"><a href="community-typing.html#pcoa-for-asv-level-data-with-bray-curtis-with-dmm-clusters-shown-with-colors"><i class="fa fa-check"></i><b>10.0.2</b> PCoA for ASV-level data with Bray-Curtis; with DMM clusters shown with colors</a></li>
<li class="chapter" data-level="10.1" data-path="community-typing.html"><a href="community-typing.html#community-detection"><i class="fa fa-check"></i><b>10.1</b> Community Detection</a></li>
<li class="chapter" data-level="10.2" data-path="community-typing.html"><a href="community-typing.html#additional-community-typing"><i class="fa fa-check"></i><b>10.2</b> Additional Community Typing</a></li>
<li class="chapter" data-level="" data-path="community-typing.html"><a href="community-typing.html#session-info-6"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="differential-abundance.html"><a href="differential-abundance.html"><i class="fa fa-check"></i><b>11</b> Differential abundance</a>
<ul>
<li class="chapter" data-level="11.1" data-path="differential-abundance.html"><a href="differential-abundance.html#differential-abundance-analysis"><i class="fa fa-check"></i><b>11.1</b> Differential abundance analysis</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="differential-abundance.html"><a href="differential-abundance.html#examples-and-tools"><i class="fa fa-check"></i><b>11.1.1</b> Examples and tools</a></li>
<li class="chapter" data-level="11.1.2" data-path="differential-abundance.html"><a href="differential-abundance.html#prevalence-filtering"><i class="fa fa-check"></i><b>11.1.2</b> Prevalence Filtering</a></li>
<li class="chapter" data-level="11.1.3" data-path="differential-abundance.html"><a href="differential-abundance.html#aldex2"><i class="fa fa-check"></i><b>11.1.3</b> ALDEx2</a></li>
<li class="chapter" data-level="11.1.4" data-path="differential-abundance.html"><a href="differential-abundance.html#ancom-bc"><i class="fa fa-check"></i><b>11.1.4</b> ANCOM-BC</a></li>
<li class="chapter" data-level="11.1.5" data-path="differential-abundance.html"><a href="differential-abundance.html#maaslin2"><i class="fa fa-check"></i><b>11.1.5</b> MaAsLin2</a></li>
<li class="chapter" data-level="11.1.6" data-path="differential-abundance.html"><a href="differential-abundance.html#comparison-of-the-methods"><i class="fa fa-check"></i><b>11.1.6</b> Comparison of the methods</a></li>
<li class="chapter" data-level="11.1.7" data-path="differential-abundance.html"><a href="differential-abundance.html#confounding-variables"><i class="fa fa-check"></i><b>11.1.7</b> Confounding variables</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="differential-abundance.html"><a href="differential-abundance.html#tree-based-methods"><i class="fa fa-check"></i><b>11.2</b> Tree-based methods</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="differential-abundance.html"><a href="differential-abundance.html#group-wise-associations-testing-based-on-balances-with-fido"><i class="fa fa-check"></i><b>11.2.1</b> Group-wise associations testing based on balances with fido</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="differential-abundance.html"><a href="differential-abundance.html#session-info-7"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="multi-assay_analyses.html"><a href="multi-assay_analyses.html"><i class="fa fa-check"></i><b>12</b> Multi-assay analyses</a>
<ul>
<li class="chapter" data-level="" data-path="multi-assay_analyses.html"><a href="multi-assay_analyses.html#session-info-8"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="part"><span><b>III Appendix</b></span></li>
<li class="chapter" data-level="13" data-path="resources.html"><a href="resources.html"><i class="fa fa-check"></i><b>13</b> Resources</a>
<ul>
<li class="chapter" data-level="13.1" data-path="resources.html"><a href="resources.html#data-containers-1"><i class="fa fa-check"></i><b>13.1</b> Data containers</a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="resources.html"><a href="resources.html#resources-for-treesummarizedexperiment"><i class="fa fa-check"></i><b>13.1.1</b> Resources for TreeSummarizedExperiment</a></li>
<li class="chapter" data-level="13.1.2" data-path="resources.html"><a href="resources.html#other-relevant-containers"><i class="fa fa-check"></i><b>13.1.2</b> Other relevant containers</a></li>
<li class="chapter" data-level="13.1.3" data-path="resources.html"><a href="resources.html#alternative-containers-for-microbiome-data"><i class="fa fa-check"></i><b>13.1.3</b> Alternative containers for microbiome data</a></li>
<li class="chapter" data-level="13.1.4" data-path="resources.html"><a href="resources.html#resources-for-phyloseq"><i class="fa fa-check"></i><b>13.1.4</b> Resources for phyloseq</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="resources.html"><a href="resources.html#r-programming-resources"><i class="fa fa-check"></i><b>13.2</b> R programming resources</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="resources.html"><a href="resources.html#bioc_intro"><i class="fa fa-check"></i><b>13.2.1</b> Bioconductor Classes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="extras.html"><a href="extras.html"><i class="fa fa-check"></i><b>14</b> Extra material</a>
<ul>
<li class="chapter" data-level="14.1" data-path="extras.html"><a href="extras.html#interactive-3d-plots"><i class="fa fa-check"></i><b>14.1</b> Interactive 3D Plots</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="acknowledgments.html"><a href="acknowledgments.html"><i class="fa fa-check"></i><b>15</b> Acknowledgments</a></li>
<li class="chapter" data-level="" data-path="bibliography.html"><a href="bibliography.html"><i class="fa fa-check"></i>Bibliography</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Orchestrating Microbiome Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="quality-control" class="section level1" number="5">
<h1><span class="header-section-number">Chapter 5</span> Exploration and quality Control</h1>
<script>
document.addEventListener("click", function (event) {
    if (event.target.classList.contains("rebook-collapse")) {
        event.target.classList.toggle("active");
        var content = event.target.nextElementSibling;
        if (content.style.display === "block") {
            content.style.display = "none";
        } else {
            content.style.display = "block";
        }
    }
})
</script>
<style>
.rebook-collapse {
  background-color: #eee;
  color: #444;
  cursor: pointer;
  padding: 18px;
  width: 100%;
  border: none;
  text-align: left;
  outline: none;
  font-size: 15px;
}

.rebook-content {
  padding: 0 18px;
  display: none;
  overflow: hidden;
  background-color: #f1f1f1;
}
</style>
<p>This chapter focuses on the exploration and quality control of
microbiome data and establishes commonly used descriptors of a
microbiome. The main difference to quality control is that the
exploration assumes that technical aspects of the dataset have been
investigated to your satisfaction. Generally speaking, at this point
you should be quite certain that the dataset does not suffer from
severe technical biases, or you should at least be aware of potential
problems.</p>
<p>In reality you might need to go back and forth between QC and exploration,
since you discover through exploration of your dataset technical aspects you
need to check.</p>
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb101-1"><a href="quality-control.html#cb101-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mia)</span></code></pre></div>
<div id="abundance" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> Abundance</h2>
<p>One initial approach for exploring data is by visualizing abundance. <code>miaViz</code> offers
the function <code>plotAbundanceDensity</code> where most abundant taxa can be plotted
including several options.</p>
<p>In the following few demonstrations are shown, using the
<span class="citation">(<a href="#ref-Lahti2014" role="doc-biblioref">Lahti et al. 2014</a>)</span> dataset.</p>
<p>A Jitter plot based on relative abundance data, similar to the one presented at
<span class="citation">(<a href="#ref-Salosensaari2021" role="doc-biblioref">Salosensaari et al. 2021</a>)</span> supplementary figure 1, could be visualized as follows:</p>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="quality-control.html#cb102-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Loading data</span></span>
<span id="cb102-2"><a href="quality-control.html#cb102-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(microbiomeDataSets)</span>
<span id="cb102-3"><a href="quality-control.html#cb102-3" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">atlas1006</span>()</span>
<span id="cb102-4"><a href="quality-control.html#cb102-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-5"><a href="quality-control.html#cb102-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Counts relative abundances</span></span>
<span id="cb102-6"><a href="quality-control.html#cb102-6" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">transformSamples</span>(tse, <span class="at">method =</span> <span class="st">&quot;relabundance&quot;</span>)</span>
<span id="cb102-7"><a href="quality-control.html#cb102-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-8"><a href="quality-control.html#cb102-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(miaViz)</span>
<span id="cb102-9"><a href="quality-control.html#cb102-9" aria-hidden="true" tabindex="-1"></a><span class="fu">plotAbundanceDensity</span>(tse, <span class="at">layout =</span> <span class="st">&quot;jitter&quot;</span>, <span class="at">abund_values =</span> <span class="st">&quot;relabundance&quot;</span>,</span>
<span id="cb102-10"><a href="quality-control.html#cb102-10" aria-hidden="true" tabindex="-1"></a>                     <span class="at">n =</span> <span class="dv">40</span>, <span class="at">point_size=</span><span class="dv">1</span>, <span class="at">point_shape=</span><span class="dv">19</span>, <span class="at">point_alpha=</span><span class="fl">0.1</span>) <span class="sc">+</span> </span>
<span id="cb102-11"><a href="quality-control.html#cb102-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_log10</span>(<span class="at">label=</span>scales<span class="sc">::</span>percent)</span></code></pre></div>
<p><img src="12_quality_control_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
<p>For instance, relative abundance values for the top 5 taxa can be
visualized as a density plot over a log scaled axis, using
“nationality” as an overlaying information:</p>
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb103-1"><a href="quality-control.html#cb103-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotAbundanceDensity</span>(tse, <span class="at">layout =</span> <span class="st">&quot;density&quot;</span>, <span class="at">abund_values =</span> <span class="st">&quot;relabundance&quot;</span>,</span>
<span id="cb103-2"><a href="quality-control.html#cb103-2" aria-hidden="true" tabindex="-1"></a>                     <span class="at">n =</span> <span class="dv">5</span>, <span class="at">colour_by=</span><span class="st">&quot;nationality&quot;</span>, <span class="at">point_alpha=</span><span class="dv">1</span><span class="sc">/</span><span class="dv">10</span> ) <span class="sc">+</span></span>
<span id="cb103-3"><a href="quality-control.html#cb103-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_log10</span>()</span></code></pre></div>
<p><img src="12_quality_control_files/figure-html/unnamed-chunk-3-1.png" width="672" /></p>
</div>
<div id="prevalence" class="section level2" number="5.2">
<h2><span class="header-section-number">5.2</span> Prevalence</h2>
<p>Prevalence is a measurement which describes in how many samples certain
microbes were detected.</p>
<p>Investigating the prevalence of microbes allows you either to focus on changes,
which pertain to most of the samples, or to focus on less often found microbes,
which are nonetheless abundantly found in a small number of samples.</p>
<p>On the raw data, the population prevalence (frequency) at a 1% relative
abundance threshold (<code>detection = 1/100</code> and <code>as_relative = TRUE</code>), can look
like this. The low prevalence in this example can be explained by rather
different sample types as well as the in-depth nature of the data.</p>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="quality-control.html#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">getPrevalence</span>(tse, <span class="at">detection =</span> <span class="dv">1</span><span class="sc">/</span><span class="dv">100</span>, <span class="at">sort =</span> <span class="cn">TRUE</span>, <span class="at">as_relative =</span> <span class="cn">TRUE</span>))</span></code></pre></div>
<pre><code>## Faecalibacterium prausnitzii et rel.           Ruminococcus obeum et rel. 
##                               0.9522                               0.9140 
##   Oscillospira guillermondii et rel.        Clostridium symbiosum et rel. 
##                               0.8801                               0.8714 
##     Subdoligranulum variable at rel.     Clostridium orbiscindens et rel. 
##                               0.8358                               0.8315</code></pre>
<p>The <code>detection</code> and <code>as_relative</code> can also be used to access, how many samples
do pass a threshold for raw counts. Here the population prevalence (frequency)
at the absolute abundance threshold (<code>as_relative = FALSE</code>) at read count 1
(<code>detection = 1</code>) is accessed.</p>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="quality-control.html#cb106-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">getPrevalence</span>(tse, <span class="at">detection =</span> <span class="dv">1</span>, <span class="at">sort =</span> <span class="cn">TRUE</span>, <span class="at">abund_values =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb106-2"><a href="quality-control.html#cb106-2" aria-hidden="true" tabindex="-1"></a>                   <span class="at">as_relative =</span> <span class="cn">FALSE</span>))</span></code></pre></div>
<pre><code>##            Uncultured Mollicutes      Uncultured Clostridiales II 
##                                1                                1 
##       Uncultured Clostridiales I               Tannerella et rel. 
##                                1                                1 
##   Sutterella wadsworthia et rel. Subdoligranulum variable at rel. 
##                                1                                1</code></pre>
<p>Note that, if the output should be used for subsetting or storing the data in
the <code>rowData</code>, set <code>sort = FALSE</code>.</p>
<div id="prevalent-microbiota-analysis" class="section level3" number="5.2.1">
<h3><span class="header-section-number">5.2.1</span> Prevalent microbiota analysis</h3>
<p>To investigate the microbiome data at a selected taxonomic level, two
approaches are available.</p>
<p>First the data can be agglomerated to the taxonomic level and <code>getPrevalence</code>
be used on the result.</p>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="quality-control.html#cb108-1" aria-hidden="true" tabindex="-1"></a><span class="fu">altExp</span>(tse,<span class="st">&quot;Phylum&quot;</span>) <span class="ot">&lt;-</span> <span class="fu">agglomerateByRank</span>(tse, <span class="st">&quot;Phylum&quot;</span>)</span>
<span id="cb108-2"><a href="quality-control.html#cb108-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">getPrevalence</span>(<span class="fu">altExp</span>(tse,<span class="st">&quot;Phylum&quot;</span>), <span class="at">detection =</span> <span class="dv">1</span><span class="sc">/</span><span class="dv">100</span>, <span class="at">sort =</span> <span class="cn">TRUE</span>,</span>
<span id="cb108-3"><a href="quality-control.html#cb108-3" aria-hidden="true" tabindex="-1"></a>                   <span class="at">abund_values =</span> <span class="st">&quot;counts&quot;</span>, <span class="at">as_relative =</span> <span class="cn">TRUE</span>))</span></code></pre></div>
<pre><code>##      Firmicutes   Bacteroidetes  Actinobacteria  Proteobacteria Verrucomicrobia 
##       1.0000000       0.9852302       0.4821894       0.2988705       0.1277150 
##   Cyanobacteria 
##       0.0008688</code></pre>
<p>Alternatively, the <code>rank</code> argument can be set to perform the agglomeration on
the fly.</p>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="quality-control.html#cb110-1" aria-hidden="true" tabindex="-1"></a><span class="fu">altExp</span>(tse,<span class="st">&quot;Phylum&quot;</span>) <span class="ot">&lt;-</span> <span class="fu">agglomerateByRank</span>(tse, <span class="st">&quot;Phylum&quot;</span>)</span>
<span id="cb110-2"><a href="quality-control.html#cb110-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">getPrevalence</span>(tse, <span class="at">rank =</span> <span class="st">&quot;Phylum&quot;</span>, <span class="at">detection =</span> <span class="dv">1</span><span class="sc">/</span><span class="dv">100</span>, <span class="at">sort =</span> <span class="cn">TRUE</span>,</span>
<span id="cb110-3"><a href="quality-control.html#cb110-3" aria-hidden="true" tabindex="-1"></a>                   <span class="at">abund_values =</span> <span class="st">&quot;counts&quot;</span>, <span class="at">as_relative =</span> <span class="cn">TRUE</span>))</span></code></pre></div>
<pre><code>##      Firmicutes   Bacteroidetes  Actinobacteria  Proteobacteria Verrucomicrobia 
##       1.0000000       0.9852302       0.4821894       0.2988705       0.1277150 
##   Cyanobacteria 
##       0.0008688</code></pre>
<p>The difference in the naming scheme is that, by default, <code>na.rm = TRUE</code> is used
for agglomeration in <code>getPrevalence</code>, whereas the default for
<code>agglomerateByRank</code> is <code>FALSE</code> to prevent accidental data loss.</p>
<p>If you only need the names of the prevalent taxa, <code>getPrevalentTaxa</code> is
available. This returns the taxa that exceed the given prevalence and detection
thresholds.</p>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="quality-control.html#cb112-1" aria-hidden="true" tabindex="-1"></a><span class="fu">getPrevalentTaxa</span>(tse, <span class="at">detection =</span> <span class="dv">0</span>, <span class="at">prevalence =</span> <span class="dv">50</span><span class="sc">/</span><span class="dv">100</span>)</span>
<span id="cb112-2"><a href="quality-control.html#cb112-2" aria-hidden="true" tabindex="-1"></a>prev <span class="ot">&lt;-</span> <span class="fu">getPrevalentTaxa</span>(tse, <span class="at">detection =</span> <span class="dv">0</span>, <span class="at">prevalence =</span> <span class="dv">50</span><span class="sc">/</span><span class="dv">100</span>,</span>
<span id="cb112-3"><a href="quality-control.html#cb112-3" aria-hidden="true" tabindex="-1"></a>                         <span class="at">rank =</span> <span class="st">&quot;Phylum&quot;</span>, <span class="at">sort =</span> <span class="cn">TRUE</span>)</span>
<span id="cb112-4"><a href="quality-control.html#cb112-4" aria-hidden="true" tabindex="-1"></a>prev</span></code></pre></div>
<p>Note that the <code>detection</code> and <code>prevalence</code> thresholds are not the same, since
<code>detection</code> can be applied to relative counts or absolute counts depending on
whether <code>as_relative</code> is set <code>TRUE</code> or <code>FALSE</code></p>
<p>TODO
See also related functions for the analysis of rare and variable taxa
(rareMembers; rareAbundance; lowAbundance).</p>
</div>
<div id="plotting-prevalence" class="section level3" number="5.2.2">
<h3><span class="header-section-number">5.2.2</span> Plotting prevalence</h3>
<p>To plot the prevalence, the data is first added to the <code>rowData</code>.</p>
<div class="sourceCode" id="cb113"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb113-1"><a href="quality-control.html#cb113-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(<span class="fu">altExp</span>(tse,<span class="st">&quot;Phylum&quot;</span>))<span class="sc">$</span>prevalence <span class="ot">&lt;-</span> </span>
<span id="cb113-2"><a href="quality-control.html#cb113-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">getPrevalence</span>(<span class="fu">altExp</span>(tse,<span class="st">&quot;Phylum&quot;</span>), <span class="at">detection =</span> <span class="dv">1</span><span class="sc">/</span><span class="dv">100</span>, <span class="at">sort =</span> <span class="cn">FALSE</span>,</span>
<span id="cb113-3"><a href="quality-control.html#cb113-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">abund_values =</span> <span class="st">&quot;counts&quot;</span>, <span class="at">as_relative =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p>Then it can be plotted via the plotting functions from the <code>scater</code> package.</p>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="quality-control.html#cb114-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scater)</span>
<span id="cb114-2"><a href="quality-control.html#cb114-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotRowData</span>(<span class="fu">altExp</span>(tse,<span class="st">&quot;Phylum&quot;</span>), <span class="st">&quot;prevalence&quot;</span>, <span class="at">colour_by =</span> <span class="st">&quot;Phylum&quot;</span>)</span></code></pre></div>
<p><img src="12_quality_control_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
<p>Additionally, the prevalence can be plotted on the taxonomic tree using the
<code>miaViz</code> package.</p>
<div class="sourceCode" id="cb115"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb115-1"><a href="quality-control.html#cb115-1" aria-hidden="true" tabindex="-1"></a><span class="fu">altExps</span>(tse) <span class="ot">&lt;-</span> <span class="fu">splitByRanks</span>(tse)</span>
<span id="cb115-2"><a href="quality-control.html#cb115-2" aria-hidden="true" tabindex="-1"></a><span class="fu">altExps</span>(tse) <span class="ot">&lt;-</span></span>
<span id="cb115-3"><a href="quality-control.html#cb115-3" aria-hidden="true" tabindex="-1"></a>   <span class="fu">lapply</span>(<span class="fu">altExps</span>(tse),</span>
<span id="cb115-4"><a href="quality-control.html#cb115-4" aria-hidden="true" tabindex="-1"></a>          <span class="cf">function</span>(y){</span>
<span id="cb115-5"><a href="quality-control.html#cb115-5" aria-hidden="true" tabindex="-1"></a>              <span class="fu">rowData</span>(y)<span class="sc">$</span>prevalence <span class="ot">&lt;-</span> </span>
<span id="cb115-6"><a href="quality-control.html#cb115-6" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">getPrevalence</span>(y, <span class="at">detection =</span> <span class="dv">1</span><span class="sc">/</span><span class="dv">100</span>, <span class="at">sort =</span> <span class="cn">FALSE</span>,</span>
<span id="cb115-7"><a href="quality-control.html#cb115-7" aria-hidden="true" tabindex="-1"></a>                                <span class="at">abund_values =</span> <span class="st">&quot;counts&quot;</span>, <span class="at">as_relative =</span> <span class="cn">TRUE</span>)</span>
<span id="cb115-8"><a href="quality-control.html#cb115-8" aria-hidden="true" tabindex="-1"></a>              y</span>
<span id="cb115-9"><a href="quality-control.html#cb115-9" aria-hidden="true" tabindex="-1"></a>          })</span>
<span id="cb115-10"><a href="quality-control.html#cb115-10" aria-hidden="true" tabindex="-1"></a>top_phyla <span class="ot">&lt;-</span> <span class="fu">getTopTaxa</span>(<span class="fu">altExp</span>(tse,<span class="st">&quot;Phylum&quot;</span>),</span>
<span id="cb115-11"><a href="quality-control.html#cb115-11" aria-hidden="true" tabindex="-1"></a>                        <span class="at">method=</span><span class="st">&quot;prevalence&quot;</span>,</span>
<span id="cb115-12"><a href="quality-control.html#cb115-12" aria-hidden="true" tabindex="-1"></a>                        <span class="at">top=</span>5L,</span>
<span id="cb115-13"><a href="quality-control.html#cb115-13" aria-hidden="true" tabindex="-1"></a>                        <span class="at">abund_values=</span><span class="st">&quot;counts&quot;</span>)</span>
<span id="cb115-14"><a href="quality-control.html#cb115-14" aria-hidden="true" tabindex="-1"></a>top_phyla_mean <span class="ot">&lt;-</span> <span class="fu">getTopTaxa</span>(<span class="fu">altExp</span>(tse,<span class="st">&quot;Phylum&quot;</span>),</span>
<span id="cb115-15"><a href="quality-control.html#cb115-15" aria-hidden="true" tabindex="-1"></a>                             <span class="at">method=</span><span class="st">&quot;mean&quot;</span>,</span>
<span id="cb115-16"><a href="quality-control.html#cb115-16" aria-hidden="true" tabindex="-1"></a>                             <span class="at">top=</span>5L,</span>
<span id="cb115-17"><a href="quality-control.html#cb115-17" aria-hidden="true" tabindex="-1"></a>                             <span class="at">abund_values=</span><span class="st">&quot;counts&quot;</span>)</span>
<span id="cb115-18"><a href="quality-control.html#cb115-18" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">unsplitByRanks</span>(tse, <span class="at">ranks =</span> <span class="fu">taxonomyRanks</span>(tse)[<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>])</span>
<span id="cb115-19"><a href="quality-control.html#cb115-19" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">addTaxonomyTree</span>(x)</span></code></pre></div>
<p>After some preparation the data is assembled and can be plotted via
<code>plotRowTree</code>.</p>
<div class="sourceCode" id="cb116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb116-1"><a href="quality-control.html#cb116-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(miaViz)</span>
<span id="cb116-2"><a href="quality-control.html#cb116-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotRowTree</span>(x[<span class="fu">rowData</span>(x)<span class="sc">$</span>Phylum <span class="sc">%in%</span> top_phyla,],</span>
<span id="cb116-3"><a href="quality-control.html#cb116-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">edge_colour_by =</span> <span class="st">&quot;Phylum&quot;</span>,</span>
<span id="cb116-4"><a href="quality-control.html#cb116-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">tip_colour_by =</span> <span class="st">&quot;prevalence&quot;</span>,</span>
<span id="cb116-5"><a href="quality-control.html#cb116-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">node_colour_by =</span> <span class="st">&quot;prevalence&quot;</span>)</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:plot-prev-prev"></span>
<img src="12_quality_control_files/figure-html/plot-prev-prev-1.png" alt="Prevalence of top phyla as judged by prevalence" width="672" />
<p class="caption">
Figure 5.1: Prevalence of top phyla as judged by prevalence
</p>
</div>
<div class="sourceCode" id="cb117"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb117-1"><a href="quality-control.html#cb117-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotRowTree</span>(x[<span class="fu">rowData</span>(x)<span class="sc">$</span>Phylum <span class="sc">%in%</span> top_phyla_mean,],</span>
<span id="cb117-2"><a href="quality-control.html#cb117-2" aria-hidden="true" tabindex="-1"></a>            <span class="at">edge_colour_by =</span> <span class="st">&quot;Phylum&quot;</span>,</span>
<span id="cb117-3"><a href="quality-control.html#cb117-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">tip_colour_by =</span> <span class="st">&quot;prevalence&quot;</span>,</span>
<span id="cb117-4"><a href="quality-control.html#cb117-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">node_colour_by =</span> <span class="st">&quot;prevalence&quot;</span>)</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:plot-prev-mean"></span>
<img src="12_quality_control_files/figure-html/plot-prev-mean-1.png" alt="Prevalence of top phyla as judged by mean abundance" width="672" />
<p class="caption">
Figure 5.2: Prevalence of top phyla as judged by mean abundance
</p>
</div>
</div>
</div>
<div id="quality-control-1" class="section level2" number="5.3">
<h2><span class="header-section-number">5.3</span> Quality control</h2>
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb118-1"><a href="quality-control.html#cb118-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mia)</span>
<span id="cb118-2"><a href="quality-control.html#cb118-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;GlobalPatterns&quot;</span>, <span class="at">package=</span><span class="st">&quot;mia&quot;</span>)</span>
<span id="cb118-3"><a href="quality-control.html#cb118-3" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> GlobalPatterns </span></code></pre></div>
<div id="top-taxa" class="section level3" number="5.3.1">
<h3><span class="header-section-number">5.3.1</span> Top taxa</h3>
<p>The <code>getTopTaxa</code> can be used for identifying top taxa in the data.</p>
<div class="sourceCode" id="cb119"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb119-1"><a href="quality-control.html#cb119-1" aria-hidden="true" tabindex="-1"></a>top_features <span class="ot">&lt;-</span> <span class="fu">getTopTaxa</span>(tse, <span class="at">method=</span><span class="st">&quot;median&quot;</span>, <span class="at">top=</span><span class="dv">10</span>)</span>
<span id="cb119-2"><a href="quality-control.html#cb119-2" aria-hidden="true" tabindex="-1"></a>tax_data <span class="ot">&lt;-</span> <span class="fu">rowData</span>(tse)[top_features,<span class="fu">taxonomyRanks</span>(tse)]</span>
<span id="cb119-3"><a href="quality-control.html#cb119-3" aria-hidden="true" tabindex="-1"></a>tax_data</span></code></pre></div>
<pre><code>## DataFrame with 10 rows and 7 columns
##            Kingdom         Phylum               Class             Order
##        &lt;character&gt;    &lt;character&gt;         &lt;character&gt;       &lt;character&gt;
## 549656    Bacteria  Cyanobacteria         Chloroplast     Stramenopiles
## 331820    Bacteria  Bacteroidetes         Bacteroidia     Bacteroidales
## 317182    Bacteria  Cyanobacteria         Chloroplast     Stramenopiles
## 94166     Bacteria Proteobacteria Gammaproteobacteria    Pasteurellales
## 279599    Bacteria  Cyanobacteria    Nostocophycideae        Nostocales
## 158660    Bacteria  Bacteroidetes         Bacteroidia     Bacteroidales
## 329744    Bacteria Actinobacteria      Actinobacteria   Actinomycetales
## 326977    Bacteria Actinobacteria      Actinobacteria Bifidobacteriales
## 248140    Bacteria  Bacteroidetes         Bacteroidia     Bacteroidales
## 550960    Bacteria Proteobacteria Gammaproteobacteria Enterobacteriales
##                    Family           Genus                Species
##               &lt;character&gt;     &lt;character&gt;            &lt;character&gt;
## 549656                 NA              NA                     NA
## 331820     Bacteroidaceae     Bacteroides                     NA
## 317182                 NA              NA                     NA
## 94166     Pasteurellaceae     Haemophilus Haemophilusparainflu..
## 279599        Nostocaceae  Dolichospermum                     NA
## 158660     Bacteroidaceae     Bacteroides                     NA
## 329744             ACK-M1              NA                     NA
## 326977 Bifidobacteriaceae Bifidobacterium Bifidobacteriumadole..
## 248140     Bacteroidaceae     Bacteroides      Bacteroidescaccae
## 550960 Enterobacteriaceae     Providencia                     NA</code></pre>
</div>
<div id="library-size" class="section level3" number="5.3.2">
<h3><span class="header-section-number">5.3.2</span> Library size</h3>
<p>The total counts/sample can be calculated using the
<code>perCellQCMetrics</code>/<code>addPerCellQC</code> from the <code>scater</code> package. The former one
just calculates the values, whereas the latter one directly adds them to the
<code>colData</code>.</p>
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb121-1"><a href="quality-control.html#cb121-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scater)</span>
<span id="cb121-2"><a href="quality-control.html#cb121-2" aria-hidden="true" tabindex="-1"></a><span class="fu">perCellQCMetrics</span>(tse)</span></code></pre></div>
<pre><code>## DataFrame with 26 rows and 3 columns
##               sum  detected     total
##         &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
## CL3        864077      6964    864077
## CC1       1135457      7679   1135457
## SV1        697509      5729    697509
## M31Fcsw   1543451      2667   1543451
## M11Fcsw   2076476      2574   2076476
## ...           ...       ...       ...
## TS28       937466      2679    937466
## TS29      1211071      2629   1211071
## Even1     1216137      4213   1216137
## Even2      971073      3130    971073
## Even3     1078241      2776   1078241</code></pre>
<div class="sourceCode" id="cb123"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb123-1"><a href="quality-control.html#cb123-1" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">addPerCellQC</span>(tse)</span>
<span id="cb123-2"><a href="quality-control.html#cb123-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(tse)</span></code></pre></div>
<pre><code>## DataFrame with 26 rows and 10 columns
##         X.SampleID   Primer Final_Barcode Barcode_truncated_plus_T
##           &lt;factor&gt; &lt;factor&gt;      &lt;factor&gt;                 &lt;factor&gt;
## CL3        CL3      ILBC_01        AACGCA                   TGCGTT
## CC1        CC1      ILBC_02        AACTCG                   CGAGTT
## SV1        SV1      ILBC_03        AACTGT                   ACAGTT
## M31Fcsw    M31Fcsw  ILBC_04        AAGAGA                   TCTCTT
## M11Fcsw    M11Fcsw  ILBC_05        AAGCTG                   CAGCTT
## ...            ...      ...           ...                      ...
## TS28         TS28   ILBC_25        ACCAGA                   TCTGGT
## TS29         TS29   ILBC_26        ACCAGC                   GCTGGT
## Even1        Even1  ILBC_27        ACCGCA                   TGCGGT
## Even2        Even2  ILBC_28        ACCTCG                   CGAGGT
## Even3        Even3  ILBC_29        ACCTGT                   ACAGGT
##         Barcode_full_length SampleType
##                    &lt;factor&gt;   &lt;factor&gt;
## CL3             CTAGCGTGCGT      Soil 
## CC1             CATCGACGAGT      Soil 
## SV1             GTACGCACAGT      Soil 
## M31Fcsw         TCGACATCTCT      Feces
## M11Fcsw         CGACTGCAGCT      Feces
## ...                     ...        ...
## TS28            GCATCGTCTGG      Feces
## TS29            CTAGTCGCTGG      Feces
## Even1           TGACTCTGCGG      Mock 
## Even2           TCTGATCGAGG      Mock 
## Even3           AGAGAGACAGG      Mock 
##                                        Description       sum  detected
##                                           &lt;factor&gt; &lt;numeric&gt; &lt;numeric&gt;
## CL3     Calhoun South Carolina Pine soil, pH 4.9      864077      6964
## CC1     Cedar Creek Minnesota, grassland, pH 6.1     1135457      7679
## SV1     Sevilleta new Mexico, desert scrub, pH 8.3    697509      5729
## M31Fcsw M3, Day 1, fecal swab, whole body study      1543451      2667
## M11Fcsw M1, Day 1, fecal swab, whole body study      2076476      2574
## ...                                            ...       ...       ...
## TS28                                       Twin #1    937466      2679
## TS29                                       Twin #2   1211071      2629
## Even1                                      Even1     1216137      4213
## Even2                                      Even2      971073      3130
## Even3                                      Even3     1078241      2776
##             total
##         &lt;numeric&gt;
## CL3        864077
## CC1       1135457
## SV1        697509
## M31Fcsw   1543451
## M11Fcsw   2076476
## ...           ...
## TS28       937466
## TS29      1211071
## Even1     1216137
## Even2      971073
## Even3     1078241</code></pre>
<p>The distribution of calculated library sizes can be visualized as a
histogram (left), or by sorting the samples by library size (right).</p>
<div class="sourceCode" id="cb125"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb125-1"><a href="quality-control.html#cb125-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb125-2"><a href="quality-control.html#cb125-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-3"><a href="quality-control.html#cb125-3" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">as.data.frame</span>(<span class="fu">colData</span>(tse))) <span class="sc">+</span></span>
<span id="cb125-4"><a href="quality-control.html#cb125-4" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x =</span> sum), <span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;gray&quot;</span>, <span class="at">bins =</span> <span class="dv">30</span>) <span class="sc">+</span></span>
<span id="cb125-5"><a href="quality-control.html#cb125-5" aria-hidden="true" tabindex="-1"></a>        <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Library size&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Frequency (n)&quot;</span>) <span class="sc">+</span> </span>
<span id="cb125-6"><a href="quality-control.html#cb125-6" aria-hidden="true" tabindex="-1"></a>        <span class="co"># scale_x_log10(breaks = scales::trans_breaks(&quot;log10&quot;, function(x) 10^x), </span></span>
<span id="cb125-7"><a href="quality-control.html#cb125-7" aria-hidden="true" tabindex="-1"></a>        <span class="co"># labels = scales::trans_format(&quot;log10&quot;, scales::math_format(10^.x))) +</span></span>
<span id="cb125-8"><a href="quality-control.html#cb125-8" aria-hidden="true" tabindex="-1"></a>        <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb125-9"><a href="quality-control.html#cb125-9" aria-hidden="true" tabindex="-1"></a>        <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(), <span class="co"># Removes the grid</span></span>
<span id="cb125-10"><a href="quality-control.html#cb125-10" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb125-11"><a href="quality-control.html#cb125-11" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb125-12"><a href="quality-control.html#cb125-12" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb125-13"><a href="quality-control.html#cb125-13" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">&quot;black&quot;</span>)) <span class="co"># Adds y-axis</span></span>
<span id="cb125-14"><a href="quality-control.html#cb125-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-15"><a href="quality-control.html#cb125-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb125-16"><a href="quality-control.html#cb125-16" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(tse)) <span class="sc">%&gt;%</span></span>
<span id="cb125-17"><a href="quality-control.html#cb125-17" aria-hidden="true" tabindex="-1"></a>        <span class="fu">arrange</span>(sum) <span class="sc">%&gt;%</span></span>
<span id="cb125-18"><a href="quality-control.html#cb125-18" aria-hidden="true" tabindex="-1"></a>        <span class="fu">mutate</span>(<span class="at">index =</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">n</span>())</span>
<span id="cb125-19"><a href="quality-control.html#cb125-19" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">y =</span> index, <span class="at">x =</span> sum<span class="sc">/</span><span class="fl">1e6</span>)) <span class="sc">+</span></span>
<span id="cb125-20"><a href="quality-control.html#cb125-20" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_point</span>() <span class="sc">+</span>  </span>
<span id="cb125-21"><a href="quality-control.html#cb125-21" aria-hidden="true" tabindex="-1"></a>        <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Library size (million reads)&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Sample index&quot;</span>) <span class="sc">+</span>  </span>
<span id="cb125-22"><a href="quality-control.html#cb125-22" aria-hidden="true" tabindex="-1"></a>        <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb125-23"><a href="quality-control.html#cb125-23" aria-hidden="true" tabindex="-1"></a>        <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(), <span class="co"># Removes the grid</span></span>
<span id="cb125-24"><a href="quality-control.html#cb125-24" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb125-25"><a href="quality-control.html#cb125-25" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb125-26"><a href="quality-control.html#cb125-26" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb125-27"><a href="quality-control.html#cb125-27" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">&quot;black&quot;</span>)) <span class="co"># Adds y-axis</span></span>
<span id="cb125-28"><a href="quality-control.html#cb125-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-29"><a href="quality-control.html#cb125-29" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb125-30"><a href="quality-control.html#cb125-30" aria-hidden="true" tabindex="-1"></a>p1 <span class="sc">+</span> p2</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:plot-viz-lib-size-1"></span>
<img src="12_quality_control_files/figure-html/plot-viz-lib-size-1-1.png" alt="Library size distribution." width="768" />
<p class="caption">
Figure 5.3: Library size distribution.
</p>
</div>
<p>Library sizes - and other variables from <code>colData</code> - can be also visualized by using
specified function called <code>plotColData</code>.</p>
<div class="sourceCode" id="cb126"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb126-1"><a href="quality-control.html#cb126-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb126-2"><a href="quality-control.html#cb126-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Sort samples by read count, order the factor levels, and store back to tse as DataFrame</span></span>
<span id="cb126-3"><a href="quality-control.html#cb126-3" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">TODO</span><span class="co">: plotColData could include an option for sorting samples based on colData variables</span></span>
<span id="cb126-4"><a href="quality-control.html#cb126-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(tse) <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(tse)) <span class="sc">%&gt;%</span></span>
<span id="cb126-5"><a href="quality-control.html#cb126-5" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">arrange</span>(X.SampleID) <span class="sc">%&gt;%</span></span>
<span id="cb126-6"><a href="quality-control.html#cb126-6" aria-hidden="true" tabindex="-1"></a>             <span class="fu">mutate</span>(<span class="at">X.SampleID =</span> <span class="fu">factor</span>(X.SampleID, <span class="at">levels=</span>X.SampleID)) <span class="sc">%&gt;%</span></span>
<span id="cb126-7"><a href="quality-control.html#cb126-7" aria-hidden="true" tabindex="-1"></a>         DataFrame</span>
<span id="cb126-8"><a href="quality-control.html#cb126-8" aria-hidden="true" tabindex="-1"></a><span class="fu">plotColData</span>(tse,<span class="st">&quot;sum&quot;</span>,<span class="st">&quot;X.SampleID&quot;</span>, <span class="at">colour_by =</span> <span class="st">&quot;SampleType&quot;</span>) <span class="sc">+</span> </span>
<span id="cb126-9"><a href="quality-control.html#cb126-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust=</span><span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb126-10"><a href="quality-control.html#cb126-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&quot;Library size (N)&quot;</span>, <span class="at">x =</span> <span class="st">&quot;Sample ID&quot;</span>)       </span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:plot-viz-lib-size-2"></span>
<img src="12_quality_control_files/figure-html/plot-viz-lib-size-2-1.png" alt="Library sizes per sample." width="768" />
<p class="caption">
Figure 5.4: Library sizes per sample.
</p>
</div>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="quality-control.html#cb127-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotColData</span>(tse,<span class="st">&quot;sum&quot;</span>,<span class="st">&quot;SampleType&quot;</span>, <span class="at">colour_by =</span> <span class="st">&quot;SampleType&quot;</span>) <span class="sc">+</span> </span>
<span id="cb127-2"><a href="quality-control.html#cb127-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust=</span><span class="dv">1</span>))</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:plot-viz-lib-size-3"></span>
<img src="12_quality_control_files/figure-html/plot-viz-lib-size-3-1.png" alt="Library sizes per sample type." width="768" />
<p class="caption">
Figure 5.5: Library sizes per sample type.
</p>
</div>
</div>
</div>
<div id="session-info-2" class="section level2 unnumbered">
<h2>Session Info</h2>
<button class="rebook-collapse">
View session info
</button>
<div class="rebook-content">
<pre><code>R version 4.1.2 (2021-11-01)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.3 LTS

Matrix products: default
BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] patchwork_1.1.1                dplyr_1.0.7                   
 [3] scater_1.22.0                  scuttle_1.4.0                 
 [5] miaViz_1.3.2                   ggraph_2.0.5                  
 [7] ggplot2_3.3.5                  microbiomeDataSets_1.1.5      
 [9] mia_1.3.13                     MultiAssayExperiment_1.20.0   
[11] TreeSummarizedExperiment_2.1.4 Biostrings_2.62.0             
[13] XVector_0.34.0                 SingleCellExperiment_1.16.0   
[15] SummarizedExperiment_1.24.0    Biobase_2.54.0                
[17] GenomicRanges_1.46.1           GenomeInfoDb_1.30.0           
[19] IRanges_2.28.0                 S4Vectors_0.32.3              
[21] BiocGenerics_0.40.0            MatrixGenerics_1.6.0          
[23] matrixStats_0.61.0-9001        BiocStyle_2.22.0              
[25] rebook_1.4.0                  

loaded via a namespace (and not attached):
  [1] AnnotationHub_3.2.0           BiocFileCache_2.2.0          
  [3] igraph_1.2.11                 plyr_1.8.6                   
  [5] lazyeval_0.2.2                splines_4.1.2                
  [7] BiocParallel_1.28.3           digest_0.6.29                
  [9] yulab.utils_0.0.4             htmltools_0.5.2              
 [11] viridis_0.6.2                 fansi_0.5.0                  
 [13] magrittr_2.0.1                memoise_2.0.1                
 [15] ScaledMatrix_1.2.0            cluster_2.1.2                
 [17] DECIPHER_2.22.0               graphlayouts_0.8.0           
 [19] colorspace_2.0-2              blob_1.2.2                   
 [21] rappdirs_0.3.3                ggrepel_0.9.1                
 [23] xfun_0.29                     crayon_1.4.2                 
 [25] RCurl_1.98-1.5                jsonlite_1.7.2               
 [27] graph_1.72.0                  ape_5.6                      
 [29] glue_1.6.0                    polyclip_1.10-0              
 [31] gtable_0.3.0                  zlibbioc_1.40.0              
 [33] DelayedArray_0.20.0           BiocSingular_1.10.0          
 [35] scales_1.1.1                  DBI_1.1.2                    
 [37] Rcpp_1.0.7                    viridisLite_0.4.0            
 [39] xtable_1.8-4                  decontam_1.14.0              
 [41] gridGraphics_0.5-1            tidytree_0.3.6               
 [43] bit_4.0.4                     rsvd_1.0.5                   
 [45] httr_1.4.2                    dir.expiry_1.2.0             
 [47] ellipsis_0.3.2                farver_2.1.0                 
 [49] pkgconfig_2.0.3               XML_3.99-0.8                 
 [51] CodeDepends_0.6.5             sass_0.4.0                   
 [53] dbplyr_2.1.1                  utf8_1.2.2                   
 [55] labeling_0.4.2                ggplotify_0.1.0              
 [57] tidyselect_1.1.1              rlang_0.4.12                 
 [59] reshape2_1.4.4                later_1.3.0                  
 [61] AnnotationDbi_1.56.2          munsell_0.5.0                
 [63] BiocVersion_3.14.0            tools_4.1.2                  
 [65] cachem_1.0.6                  DirichletMultinomial_1.36.0  
 [67] generics_0.1.1                RSQLite_2.2.9                
 [69] ExperimentHub_2.2.0           evaluate_0.14                
 [71] stringr_1.4.0                 fastmap_1.1.0                
 [73] yaml_2.2.1                    ggtree_3.2.1                 
 [75] knitr_1.37                    bit64_4.0.5                  
 [77] tidygraph_1.2.0               purrr_0.3.4                  
 [79] KEGGREST_1.34.0               nlme_3.1-153                 
 [81] sparseMatrixStats_1.6.0       mime_0.12                    
 [83] aplot_0.1.1                   compiler_4.1.2               
 [85] beeswarm_0.4.0                filelock_1.0.2               
 [87] curl_4.3.2                    png_0.1-7                    
 [89] interactiveDisplayBase_1.32.0 treeio_1.18.1                
 [91] tweenr_1.0.2                  tibble_3.1.6                 
 [93] bslib_0.3.1                   stringi_1.7.6                
 [95] highr_0.9                     lattice_0.20-45              
 [97] Matrix_1.4-0                  vegan_2.5-7                  
 [99] permute_0.9-5                 vctrs_0.3.8                  
[101] pillar_1.6.4                  lifecycle_1.0.1              
[103] BiocManager_1.30.16           jquerylib_0.1.4              
[105] BiocNeighbors_1.12.0          cowplot_1.1.1                
[107] bitops_1.0-7                  irlba_2.3.5                  
[109] httpuv_1.6.5                  R6_2.5.1                     
[111] bookdown_0.24                 promises_1.2.0.1             
[113] gridExtra_2.3                 vipor_0.4.5                  
[115] codetools_0.2-18              MASS_7.3-54                  
[117] assertthat_0.2.1              withr_2.4.3                  
[119] GenomeInfoDbData_1.2.7        mgcv_1.8-38                  
[121] parallel_4.1.2                ggfun_0.0.4                  
[123] grid_4.1.2                    beachmat_2.10.0              
[125] tidyr_1.1.4                   rmarkdown_2.11               
[127] DelayedMatrixStats_1.16.0     ggnewscale_0.4.5             
[129] ggforce_0.3.3                 shiny_1.7.1                  
[131] ggbeeswarm_0.6.0             </code></pre>
</div>

</div>
</div>
<h3>Bibliography</h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Lahti2014" class="csl-entry">
Lahti, Leo, Jarkko Salojärvi, Anne Salonen, Marten Scheffer, and Willem M. de Vos. 2014. <span>“Tipping Elements in the Human Intestinal Ecosystem.”</span> <em>Nature Communications</em> 2014: 1–10. https://doi.org/<a href="https://doi.org/10.1038/ncomms5344">https://doi.org/10.1038/ncomms5344</a>.
</div>
<div id="ref-Salosensaari2021" class="csl-entry">
Salosensaari, Aaro, Ville Laitinen, Aki Havulinna, Guillaume Méric, Susan Cheng, Markus Perola, Liisa Valsta, et al. 2021. <span>“Taxonomic Signatures of Cause-Specific Mortality Risk in Human Gut Microbiome.”</span> <em>Nature Communications</em> 12: 1–8. <a href="https://www.nature.com/articles/s41467-021-22962-y">https://www.nature.com/articles/s41467-021-22962-y</a>.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="datamanipulation.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="taxonomic-information.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["OMA.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
