<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Exploration and quality Control | Orchestrating Microbiome Analysis</title>
  <meta name="description" content="Chapter 5 Exploration and quality Control | Orchestrating Microbiome Analysis" />
  <meta name="generator" content="bookdown 0.33 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Exploration and quality Control | Orchestrating Microbiome Analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="microbiome/OMA" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Exploration and quality Control | Orchestrating Microbiome Analysis" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="datamanipulation.html"/>
<link rel="next" href="taxonomic-information.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Orchestrating Microbiome Analysis</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="part"><span><b>I Introduction</b></span></li>
<li class="chapter" data-level="1" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="packages.html"><a href="packages.html"><i class="fa fa-check"></i><b>2</b> Packages</a>
<ul>
<li class="chapter" data-level="2.1" data-path="packages.html"><a href="packages.html#package-installation"><i class="fa fa-check"></i><b>2.1</b> Package installation</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="packages.html"><a href="packages.html#packages"><i class="fa fa-check"></i><b>2.1.1</b> Installing specific packages</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="packages.html"><a href="packages.html#package-ecosystem"><i class="fa fa-check"></i><b>2.2</b> Package ecosystem</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="packages.html"><a href="packages.html#mia-family-of-methods"><i class="fa fa-check"></i><b>2.2.1</b> mia family of methods</a></li>
<li class="chapter" data-level="2.2.2" data-path="packages.html"><a href="packages.html#sub-tree-methods"><i class="fa fa-check"></i><b>2.2.2</b> Tree-based methods</a></li>
<li class="chapter" data-level="2.2.3" data-path="packages.html"><a href="packages.html#sub-diff-abund"><i class="fa fa-check"></i><b>2.2.3</b> Differential abundance</a></li>
<li class="chapter" data-level="2.2.4" data-path="packages.html"><a href="packages.html#sub-manipulation"><i class="fa fa-check"></i><b>2.2.4</b> Manipulation</a></li>
<li class="chapter" data-level="2.2.5" data-path="packages.html"><a href="packages.html#further-options"><i class="fa fa-check"></i><b>2.2.5</b> Further options</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="packages.html"><a href="packages.html#session-info"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="containers.html"><a href="containers.html"><i class="fa fa-check"></i><b>3</b> Microbiome Data</a>
<ul>
<li class="chapter" data-level="3.1" data-path="containers.html"><a href="containers.html#data-science-framework"><i class="fa fa-check"></i><b>3.1</b> Data science framework</a></li>
<li class="chapter" data-level="3.2" data-path="containers.html"><a href="containers.html#data-containers"><i class="fa fa-check"></i><b>3.2</b> Data containers</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="containers.html"><a href="containers.html#assay-data"><i class="fa fa-check"></i><b>3.2.1</b> Assay data</a></li>
<li class="chapter" data-level="3.2.2" data-path="containers.html"><a href="containers.html#coldata"><i class="fa fa-check"></i><b>3.2.2</b> colData</a></li>
<li class="chapter" data-level="3.2.3" data-path="containers.html"><a href="containers.html#rowdata"><i class="fa fa-check"></i><b>3.2.3</b> rowData</a></li>
<li class="chapter" data-level="3.2.4" data-path="containers.html"><a href="containers.html#rowtree"><i class="fa fa-check"></i><b>3.2.4</b> rowTree</a></li>
<li class="chapter" data-level="3.2.5" data-path="containers.html"><a href="containers.html#alt-exp"><i class="fa fa-check"></i><b>3.2.5</b> Alternative experiments</a></li>
<li class="chapter" data-level="3.2.6" data-path="containers.html"><a href="containers.html#mae"><i class="fa fa-check"></i><b>3.2.6</b> MultiAssayExperiments</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="containers.html"><a href="containers.html#example-data"><i class="fa fa-check"></i><b>3.3</b> Demonstration data</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="containers.html"><a href="containers.html#package-data"><i class="fa fa-check"></i><b>3.3.1</b> Package data</a></li>
<li class="chapter" data-level="3.3.2" data-path="containers.html"><a href="containers.html#experimenthub-data"><i class="fa fa-check"></i><b>3.3.2</b> ExperimentHub data</a></li>
<li class="chapter" data-level="3.3.3" data-path="containers.html"><a href="containers.html#curated-metagenomic-data"><i class="fa fa-check"></i><b>3.3.3</b> Curated metagenomic data</a></li>
<li class="chapter" data-level="3.3.4" data-path="containers.html"><a href="containers.html#other-data-sources"><i class="fa fa-check"></i><b>3.3.4</b> Other data sources</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="containers.html"><a href="containers.html#loading-experimental-microbiome-data"><i class="fa fa-check"></i><b>3.4</b> Loading experimental microbiome data</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="containers.html"><a href="containers.html#s-workflow"><i class="fa fa-check"></i><b>3.4.1</b> 16S workflow</a></li>
<li class="chapter" data-level="3.4.2" data-path="containers.html"><a href="containers.html#import-from-external-files"><i class="fa fa-check"></i><b>3.4.2</b> Import from external files</a></li>
<li class="chapter" data-level="3.4.3" data-path="containers.html"><a href="containers.html#constructing-treesummarizedexperiment"><i class="fa fa-check"></i><b>3.4.3</b> Constructing TreeSummarizedExperiment</a></li>
<li class="chapter" data-level="3.4.4" data-path="containers.html"><a href="containers.html#constructing-multiassayexperiment"><i class="fa fa-check"></i><b>3.4.4</b> Constructing MultiAssayExperiment</a></li>
<li class="chapter" data-level="3.4.5" data-path="containers.html"><a href="containers.html#import-functions-for-standard-formats"><i class="fa fa-check"></i><b>3.4.5</b> Import functions for standard formats</a></li>
<li class="chapter" data-level="3.4.6" data-path="containers.html"><a href="containers.html#conversions-between-data-formats-in-r"><i class="fa fa-check"></i><b>3.4.6</b> Conversions between data formats in R</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="containers.html"><a href="containers.html#session-info-1"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="part"><span><b>II Focus Topics</b></span></li>
<li class="chapter" data-level="4" data-path="datamanipulation.html"><a href="datamanipulation.html"><i class="fa fa-check"></i><b>4</b> Data Manipulation</a>
<ul>
<li class="chapter" data-level="4.1" data-path="datamanipulation.html"><a href="datamanipulation.html#tidying-and-subsetting"><i class="fa fa-check"></i><b>4.1</b> Tidying and subsetting</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="datamanipulation.html"><a href="datamanipulation.html#tidy-data"><i class="fa fa-check"></i><b>4.1.1</b> Tidy data</a></li>
<li class="chapter" data-level="4.1.2" data-path="datamanipulation.html"><a href="datamanipulation.html#subsetting"><i class="fa fa-check"></i><b>4.1.2</b> Subsetting</a></li>
<li class="chapter" data-level="4.1.3" data-path="datamanipulation.html"><a href="datamanipulation.html#splitting"><i class="fa fa-check"></i><b>4.1.3</b> Splitting</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="datamanipulation.html"><a href="datamanipulation.html#add-or-modify-data"><i class="fa fa-check"></i><b>4.2</b> Add or modify data</a></li>
<li class="chapter" data-level="4.3" data-path="datamanipulation.html"><a href="datamanipulation.html#merge-data"><i class="fa fa-check"></i><b>4.3</b> Merge data</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="datamanipulation.html"><a href="datamanipulation.html#additional-functions"><i class="fa fa-check"></i><b>4.3.1</b> Additional functions</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>5</b> Exploration and quality Control</a>
<ul>
<li class="chapter" data-level="5.1" data-path="quality-control.html"><a href="quality-control.html#abundance"><i class="fa fa-check"></i><b>5.1</b> Abundance</a></li>
<li class="chapter" data-level="5.2" data-path="quality-control.html"><a href="quality-control.html#prevalence"><i class="fa fa-check"></i><b>5.2</b> Prevalence</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="quality-control.html"><a href="quality-control.html#prevalence-analysis"><i class="fa fa-check"></i><b>5.2.1</b> Prevalence analysis</a></li>
<li class="chapter" data-level="5.2.2" data-path="quality-control.html"><a href="quality-control.html#rare-taxa"><i class="fa fa-check"></i><b>5.2.2</b> Rare taxa</a></li>
<li class="chapter" data-level="5.2.3" data-path="quality-control.html"><a href="quality-control.html#plotting-prevalence"><i class="fa fa-check"></i><b>5.2.3</b> Plotting prevalence</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="quality-control.html"><a href="quality-control.html#qc"><i class="fa fa-check"></i><b>5.3</b> Quality control</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="quality-control.html"><a href="quality-control.html#top-taxa"><i class="fa fa-check"></i><b>5.3.1</b> Top taxa</a></li>
<li class="chapter" data-level="5.3.2" data-path="quality-control.html"><a href="quality-control.html#library-size-read-count"><i class="fa fa-check"></i><b>5.3.2</b> Library size / read count</a></li>
<li class="chapter" data-level="5.3.3" data-path="quality-control.html"><a href="quality-control.html#contaminant-sequences"><i class="fa fa-check"></i><b>5.3.3</b> Contaminant sequences</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="quality-control.html"><a href="quality-control.html#session-info-2"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="taxonomic-information.html"><a href="taxonomic-information.html"><i class="fa fa-check"></i><b>6</b> Taxonomic Information</a>
<ul>
<li class="chapter" data-level="6.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#assigning-taxonomic-information."><i class="fa fa-check"></i><b>6.1</b> Assigning taxonomic information.</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#dada2"><i class="fa fa-check"></i><b>6.1.1</b> dada2</a></li>
<li class="chapter" data-level="6.1.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#decipher"><i class="fa fa-check"></i><b>6.1.2</b> DECIPHER</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#functions-to-access-taxonomic-information"><i class="fa fa-check"></i><b>6.2</b> Functions to access taxonomic information</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#generate-a-taxonomic-tree-on-the-fly"><i class="fa fa-check"></i><b>6.2.1</b> Generate a taxonomic tree on the fly</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="taxonomic-information.html"><a href="taxonomic-information.html#data-agglomeration"><i class="fa fa-check"></i><b>6.3</b> Data agglomeration</a></li>
<li class="chapter" data-level="6.4" data-path="taxonomic-information.html"><a href="taxonomic-information.html#data-transformation"><i class="fa fa-check"></i><b>6.4</b> Data transformation</a></li>
<li class="chapter" data-level="6.5" data-path="taxonomic-information.html"><a href="taxonomic-information.html#pick-specific"><i class="fa fa-check"></i><b>6.5</b> Pick specific</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#abundances-of-all-taxa-in-specific-sample"><i class="fa fa-check"></i><b>6.5.1</b> Abundances of all taxa in specific sample</a></li>
<li class="chapter" data-level="6.5.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#abundances-of-specific-taxa-in-all-samples"><i class="fa fa-check"></i><b>6.5.2</b> Abundances of specific taxa in all samples</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="taxonomic-information.html"><a href="taxonomic-information.html#session-info-3"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="community-diversity.html"><a href="community-diversity.html"><i class="fa fa-check"></i><b>7</b> Community diversity</a>
<ul>
<li class="chapter" data-level="7.1" data-path="community-diversity.html"><a href="community-diversity.html#estimation"><i class="fa fa-check"></i><b>7.1</b> Estimation</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="community-diversity.html"><a href="community-diversity.html#richness"><i class="fa fa-check"></i><b>7.1.1</b> Richness</a></li>
<li class="chapter" data-level="7.1.2" data-path="community-diversity.html"><a href="community-diversity.html#estimate-diversity"><i class="fa fa-check"></i><b>7.1.2</b> Diversity</a></li>
<li class="chapter" data-level="7.1.3" data-path="community-diversity.html"><a href="community-diversity.html#faith-diversity"><i class="fa fa-check"></i><b>7.1.3</b> Faith phylogenetic diversity</a></li>
<li class="chapter" data-level="7.1.4" data-path="community-diversity.html"><a href="community-diversity.html#evenness"><i class="fa fa-check"></i><b>7.1.4</b> Evenness</a></li>
<li class="chapter" data-level="7.1.5" data-path="community-diversity.html"><a href="community-diversity.html#dominance"><i class="fa fa-check"></i><b>7.1.5</b> Dominance</a></li>
<li class="chapter" data-level="7.1.6" data-path="community-diversity.html"><a href="community-diversity.html#rarity"><i class="fa fa-check"></i><b>7.1.6</b> Rarity</a></li>
<li class="chapter" data-level="7.1.7" data-path="community-diversity.html"><a href="community-diversity.html#divergence"><i class="fa fa-check"></i><b>7.1.7</b> Divergence</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="community-diversity.html"><a href="community-diversity.html#visualization"><i class="fa fa-check"></i><b>7.2</b> Visualization</a></li>
<li class="chapter" data-level="" data-path="community-diversity.html"><a href="community-diversity.html#session-info-4"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="community-similarity.html"><a href="community-similarity.html"><i class="fa fa-check"></i><b>8</b> Community similarity</a>
<ul>
<li class="chapter" data-level="8.1" data-path="community-similarity.html"><a href="community-similarity.html#explained-variance"><i class="fa fa-check"></i><b>8.1</b> Explained variance</a></li>
<li class="chapter" data-level="8.2" data-path="community-similarity.html"><a href="community-similarity.html#community-comparisons-by-beta-diversity-analysis"><i class="fa fa-check"></i><b>8.2</b> Community comparisons by beta diversity analysis</a></li>
<li class="chapter" data-level="8.3" data-path="community-similarity.html"><a href="community-similarity.html#other-ordination-methods"><i class="fa fa-check"></i><b>8.3</b> Other ordination methods</a></li>
<li class="chapter" data-level="8.4" data-path="community-similarity.html"><a href="community-similarity.html#pcoa-genus"><i class="fa fa-check"></i><b>8.4</b> Visualizing the most dominant genus on PCoA</a>
<ul>
<li class="chapter" data-level="8.4.1" data-path="community-similarity.html"><a href="community-similarity.html#testing-differences-in-community-composition-between-sample-groups"><i class="fa fa-check"></i><b>8.4.1</b> Testing differences in community composition between sample groups</a></li>
<li class="chapter" data-level="8.4.2" data-path="community-similarity.html"><a href="community-similarity.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>8.4.2</b> Checking the homogeneity condition</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="community-similarity.html"><a href="community-similarity.html#further-reading"><i class="fa fa-check"></i><b>8.5</b> Further reading</a></li>
<li class="chapter" data-level="" data-path="community-similarity.html"><a href="community-similarity.html#session-info-5"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="microbiome-community.html"><a href="microbiome-community.html"><i class="fa fa-check"></i><b>9</b> Community composition</a>
<ul>
<li class="chapter" data-level="9.1" data-path="microbiome-community.html"><a href="microbiome-community.html#visual-composition"><i class="fa fa-check"></i><b>9.1</b> Visualizing taxonomic composition</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="microbiome-community.html"><a href="microbiome-community.html#composition-barplot"><i class="fa fa-check"></i><b>9.1.1</b> Composition barplot</a></li>
<li class="chapter" data-level="9.1.2" data-path="microbiome-community.html"><a href="microbiome-community.html#composition-heatmap"><i class="fa fa-check"></i><b>9.1.2</b> Composition heatmap</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="microbiome-community.html"><a href="microbiome-community.html#session-info-6"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="clustering.html"><a href="clustering.html"><i class="fa fa-check"></i><b>10</b> Community typing (clustering)</a>
<ul>
<li class="chapter" data-level="10.1" data-path="clustering.html"><a href="clustering.html#hierarchical-clustering"><i class="fa fa-check"></i><b>10.1</b> Hierarchical clustering</a></li>
<li class="chapter" data-level="10.2" data-path="clustering.html"><a href="clustering.html#k-means-clustering"><i class="fa fa-check"></i><b>10.2</b> K-means clustering</a></li>
<li class="chapter" data-level="10.3" data-path="clustering.html"><a href="clustering.html#dirichlet-multinomial-mixtures-dmm"><i class="fa fa-check"></i><b>10.3</b> Dirichlet Multinomial Mixtures (DMM)</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="clustering.html"><a href="clustering.html#pcoa-for-asv-level-data-with-bray-curtis-with-dmm-clusters-shown-with-colors"><i class="fa fa-check"></i><b>10.3.1</b> PCoA for ASV-level data with Bray-Curtis; with DMM clusters shown with colors</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="clustering.html"><a href="clustering.html#community-detection"><i class="fa fa-check"></i><b>10.4</b> Community Detection</a></li>
<li class="chapter" data-level="10.5" data-path="clustering.html"><a href="clustering.html#biclustering"><i class="fa fa-check"></i><b>10.5</b> Biclustering</a>
<ul>
<li class="chapter" data-level="10.5.1" data-path="clustering.html"><a href="clustering.html#taxa-vs-samples"><i class="fa fa-check"></i><b>10.5.1</b> Taxa vs samples</a></li>
<li class="chapter" data-level="10.5.2" data-path="clustering.html"><a href="clustering.html#taxa-vs-biomolecules"><i class="fa fa-check"></i><b>10.5.2</b> Taxa vs biomolecules</a></li>
<li class="chapter" data-level="10.5.3" data-path="clustering.html"><a href="clustering.html#taxa-vs-taxa"><i class="fa fa-check"></i><b>10.5.3</b> Taxa vs taxa</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="clustering.html"><a href="clustering.html#additional-community-typing"><i class="fa fa-check"></i><b>10.6</b> Additional Community Typing</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="differential-abundance.html"><a href="differential-abundance.html"><i class="fa fa-check"></i><b>11</b> Differential abundance</a>
<ul>
<li class="chapter" data-level="11.1" data-path="differential-abundance.html"><a href="differential-abundance.html#differential-abundance-analysis"><i class="fa fa-check"></i><b>11.1</b> Differential abundance analysis</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="differential-abundance.html"><a href="differential-abundance.html#examples-and-tools"><i class="fa fa-check"></i><b>11.1.1</b> Examples and tools</a></li>
<li class="chapter" data-level="11.1.2" data-path="differential-abundance.html"><a href="differential-abundance.html#prevalence-filtering"><i class="fa fa-check"></i><b>11.1.2</b> Prevalence Filtering</a></li>
<li class="chapter" data-level="11.1.3" data-path="differential-abundance.html"><a href="differential-abundance.html#aldex2"><i class="fa fa-check"></i><b>11.1.3</b> ALDEx2</a></li>
<li class="chapter" data-level="11.1.4" data-path="differential-abundance.html"><a href="differential-abundance.html#ancom-bc"><i class="fa fa-check"></i><b>11.1.4</b> ANCOM-BC</a></li>
<li class="chapter" data-level="11.1.5" data-path="differential-abundance.html"><a href="differential-abundance.html#maaslin2"><i class="fa fa-check"></i><b>11.1.5</b> MaAsLin2</a></li>
<li class="chapter" data-level="11.1.6" data-path="differential-abundance.html"><a href="differential-abundance.html#linda"><i class="fa fa-check"></i><b>11.1.6</b> LinDA</a></li>
<li class="chapter" data-level="11.1.7" data-path="differential-abundance.html"><a href="differential-abundance.html#comparison-of-the-methods"><i class="fa fa-check"></i><b>11.1.7</b> Comparison of the methods</a></li>
<li class="chapter" data-level="11.1.8" data-path="differential-abundance.html"><a href="differential-abundance.html#confounding-variables"><i class="fa fa-check"></i><b>11.1.8</b> Confounding variables</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="differential-abundance.html"><a href="differential-abundance.html#tree-based-methods"><i class="fa fa-check"></i><b>11.2</b> Tree-based methods</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="differential-abundance.html"><a href="differential-abundance.html#group-wise-associations-testing-based-on-balances"><i class="fa fa-check"></i><b>11.2.1</b> Group-wise associations testing based on balances</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="differential-abundance.html"><a href="differential-abundance.html#session-info-7"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="machine_learning.html"><a href="machine_learning.html"><i class="fa fa-check"></i><b>12</b> Machine learning</a>
<ul>
<li class="chapter" data-level="12.1" data-path="machine_learning.html"><a href="machine_learning.html#supervised-machine-learning"><i class="fa fa-check"></i><b>12.1</b> Supervised machine learning</a></li>
<li class="chapter" data-level="12.2" data-path="machine_learning.html"><a href="machine_learning.html#unsupervised-machine-learning"><i class="fa fa-check"></i><b>12.2</b> Unsupervised machine learning</a></li>
<li class="chapter" data-level="" data-path="machine_learning.html"><a href="machine_learning.html#session-info-8"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="multi-assay-analyses.html"><a href="multi-assay-analyses.html"><i class="fa fa-check"></i><b>13</b> Multi-assay analyses</a>
<ul>
<li class="chapter" data-level="13.1" data-path="multi-assay-analyses.html"><a href="multi-assay-analyses.html#cross-correlation-analysis"><i class="fa fa-check"></i><b>13.1</b> Cross-correlation Analysis</a></li>
<li class="chapter" data-level="13.2" data-path="multi-assay-analyses.html"><a href="multi-assay-analyses.html#multi-omics-factor-analysis"><i class="fa fa-check"></i><b>13.2</b> Multi-Omics Factor Analysis</a></li>
<li class="chapter" data-level="" data-path="multi-assay-analyses.html"><a href="multi-assay-analyses.html#session-info-9"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="viz-chapter.html"><a href="viz-chapter.html"><i class="fa fa-check"></i><b>14</b> Visualization</a>
<ul>
<li class="chapter" data-level="14.1" data-path="viz-chapter.html"><a href="viz-chapter.html#pre-analysis-exploration"><i class="fa fa-check"></i><b>14.1</b> Pre-analysis exploration</a>
<ul>
<li class="chapter" data-level="14.1.1" data-path="viz-chapter.html"><a href="viz-chapter.html#accessing-row-and-column-data"><i class="fa fa-check"></i><b>14.1.1</b> Accessing row and column data</a></li>
<li class="chapter" data-level="14.1.2" data-path="viz-chapter.html"><a href="viz-chapter.html#viewing-abundance-and-prevalence-patterns"><i class="fa fa-check"></i><b>14.1.2</b> Viewing abundance and prevalence patterns</a></li>
</ul></li>
<li class="chapter" data-level="14.2" data-path="viz-chapter.html"><a href="viz-chapter.html#diversity-estimation"><i class="fa fa-check"></i><b>14.2</b> Diversity estimation</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="viz-chapter.html"><a href="viz-chapter.html#alpha-diversity-with-scatter-violin-and-box-plots"><i class="fa fa-check"></i><b>14.2.1</b> Alpha diversity with scatter, violin and box plots</a></li>
<li class="chapter" data-level="14.2.2" data-path="viz-chapter.html"><a href="viz-chapter.html#beta-diversity-with-shepard-and-coordination-plots"><i class="fa fa-check"></i><b>14.2.2</b> Beta diversity with Shepard and coordination plots</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="viz-chapter.html"><a href="viz-chapter.html#statistical-analysis"><i class="fa fa-check"></i><b>14.3</b> Statistical analysis</a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="viz-chapter.html"><a href="viz-chapter.html#heatmaps"><i class="fa fa-check"></i><b>14.3.1</b> Heatmaps</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="viz-chapter.html"><a href="viz-chapter.html#session-info-10"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="part"><span><b>III Training</b></span></li>
<li class="chapter" data-level="15" data-path="training.html"><a href="training.html"><i class="fa fa-check"></i><b>15</b> Training</a>
<ul>
<li class="chapter" data-level="15.1" data-path="training.html"><a href="training.html#checklist"><i class="fa fa-check"></i><b>15.1</b> Checklist</a></li>
<li class="chapter" data-level="15.2" data-path="training.html"><a href="training.html#software"><i class="fa fa-check"></i><b>15.2</b> Recommended software</a></li>
<li class="chapter" data-level="15.3" data-path="training.html"><a href="training.html#material"><i class="fa fa-check"></i><b>15.3</b> Study material</a></li>
<li class="chapter" data-level="15.4" data-path="training.html"><a href="training.html#support-and-resources"><i class="fa fa-check"></i><b>15.4</b> Support and resources</a></li>
<li class="chapter" data-level="15.5" data-path="training.html"><a href="training.html#coc"><i class="fa fa-check"></i><b>15.5</b> Code of Conduct</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="resources.html"><a href="resources.html"><i class="fa fa-check"></i><b>16</b> Resources</a>
<ul>
<li class="chapter" data-level="16.1" data-path="resources.html"><a href="resources.html#data-containers-1"><i class="fa fa-check"></i><b>16.1</b> Data containers</a>
<ul>
<li class="chapter" data-level="16.1.1" data-path="resources.html"><a href="resources.html#resources-for-treesummarizedexperiment"><i class="fa fa-check"></i><b>16.1.1</b> Resources for TreeSummarizedExperiment</a></li>
<li class="chapter" data-level="16.1.2" data-path="resources.html"><a href="resources.html#other-relevant-containers"><i class="fa fa-check"></i><b>16.1.2</b> Other relevant containers</a></li>
<li class="chapter" data-level="16.1.3" data-path="resources.html"><a href="resources.html#alternative-containers-for-microbiome-data"><i class="fa fa-check"></i><b>16.1.3</b> Alternative containers for microbiome data</a></li>
<li class="chapter" data-level="16.1.4" data-path="resources.html"><a href="resources.html#resources-for-phyloseq"><i class="fa fa-check"></i><b>16.1.4</b> Resources for phyloseq</a></li>
</ul></li>
<li class="chapter" data-level="16.2" data-path="resources.html"><a href="resources.html#r-programming-resources"><i class="fa fa-check"></i><b>16.2</b> R programming resources</a>
<ul>
<li class="chapter" data-level="16.2.1" data-path="resources.html"><a href="resources.html#bioc_intro"><i class="fa fa-check"></i><b>16.2.1</b> Bioconductor Classes</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="resources.html"><a href="resources.html#quarto"><i class="fa fa-check"></i><b>16.3</b> Reproducible reporting with Quarto</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="exercises.html"><a href="exercises.html"><i class="fa fa-check"></i><b>17</b> Exercises</a>
<ul>
<li class="chapter" data-level="17.1" data-path="exercises.html"><a href="exercises.html#workflows"><i class="fa fa-check"></i><b>17.1</b> Workflows</a>
<ul>
<li class="chapter" data-level="17.1.1" data-path="exercises.html"><a href="exercises.html#reproducible-reporting-with-quarto"><i class="fa fa-check"></i><b>17.1.1</b> Reproducible reporting with Quarto</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="exercises.html"><a href="exercises.html#data-containers-treese"><i class="fa fa-check"></i><b>17.2</b> Data containers: TreeSE</a>
<ul>
<li class="chapter" data-level="17.2.1" data-path="exercises.html"><a href="exercises.html#constructing-a-data-object"><i class="fa fa-check"></i><b>17.2.1</b> Constructing a data object</a></li>
<li class="chapter" data-level="17.2.2" data-path="exercises.html"><a href="exercises.html#importing-data"><i class="fa fa-check"></i><b>17.2.2</b> Importing data</a></li>
<li class="chapter" data-level="17.2.3" data-path="exercises.html"><a href="exercises.html#basic-summaries"><i class="fa fa-check"></i><b>17.2.3</b> Basic summaries</a></li>
<li class="chapter" data-level="17.2.4" data-path="exercises.html"><a href="exercises.html#taxonomic-abundance-table-assay"><i class="fa fa-check"></i><b>17.2.4</b> Taxonomic abundance table (assay)</a></li>
<li class="chapter" data-level="17.2.5" data-path="exercises.html"><a href="exercises.html#sample-side-information"><i class="fa fa-check"></i><b>17.2.5</b> Sample side information</a></li>
<li class="chapter" data-level="17.2.6" data-path="exercises.html"><a href="exercises.html#feature-side-information"><i class="fa fa-check"></i><b>17.2.6</b> Feature side information</a></li>
<li class="chapter" data-level="17.2.7" data-path="exercises.html"><a href="exercises.html#other-elements"><i class="fa fa-check"></i><b>17.2.7</b> Other elements</a></li>
</ul></li>
<li class="chapter" data-level="17.3" data-path="exercises.html"><a href="exercises.html#data-manipulation"><i class="fa fa-check"></i><b>17.3</b> Data manipulation</a>
<ul>
<li class="chapter" data-level="17.3.1" data-path="exercises.html"><a href="exercises.html#subsetting-1"><i class="fa fa-check"></i><b>17.3.1</b> Subsetting</a></li>
<li class="chapter" data-level="17.3.2" data-path="exercises.html"><a href="exercises.html#library-sizes"><i class="fa fa-check"></i><b>17.3.2</b> Library sizes</a></li>
<li class="chapter" data-level="17.3.3" data-path="exercises.html"><a href="exercises.html#prevalent-and-core-taxonomic-features"><i class="fa fa-check"></i><b>17.3.3</b> Prevalent and core taxonomic features</a></li>
<li class="chapter" data-level="17.3.4" data-path="exercises.html"><a href="exercises.html#data-exploration"><i class="fa fa-check"></i><b>17.3.4</b> Data exploration</a></li>
<li class="chapter" data-level="17.3.5" data-path="exercises.html"><a href="exercises.html#other-functions"><i class="fa fa-check"></i><b>17.3.5</b> Other functions</a></li>
<li class="chapter" data-level="17.3.6" data-path="exercises.html"><a href="exercises.html#transformations"><i class="fa fa-check"></i><b>17.3.6</b> Transformations</a></li>
</ul></li>
<li class="chapter" data-level="17.4" data-path="exercises.html"><a href="exercises.html#abundance-tables"><i class="fa fa-check"></i><b>17.4</b> Abundance tables</a>
<ul>
<li class="chapter" data-level="17.4.1" data-path="exercises.html"><a href="exercises.html#taxonomic-levels"><i class="fa fa-check"></i><b>17.4.1</b> Taxonomic levels</a></li>
<li class="chapter" data-level="17.4.2" data-path="exercises.html"><a href="exercises.html#alternative-experiments-altexp"><i class="fa fa-check"></i><b>17.4.2</b> Alternative experiments (altExp)</a></li>
</ul></li>
<li class="chapter" data-level="17.5" data-path="exercises.html"><a href="exercises.html#community-diversity-alpha-diversity"><i class="fa fa-check"></i><b>17.5</b> Community diversity (alpha diversity)</a>
<ul>
<li class="chapter" data-level="17.5.1" data-path="exercises.html"><a href="exercises.html#alpha-diversity-basics"><i class="fa fa-check"></i><b>17.5.1</b> Alpha diversity basics</a></li>
<li class="chapter" data-level="17.5.2" data-path="exercises.html"><a href="exercises.html#alpha-diversity-extra"><i class="fa fa-check"></i><b>17.5.2</b> Alpha diversity extra</a></li>
</ul></li>
<li class="chapter" data-level="17.6" data-path="exercises.html"><a href="exercises.html#community-composition-beta-diversity"><i class="fa fa-check"></i><b>17.6</b> Community composition (beta diversity)</a>
<ul>
<li class="chapter" data-level="17.6.1" data-path="exercises.html"><a href="exercises.html#beta-diversity-basics"><i class="fa fa-check"></i><b>17.6.1</b> Beta diversity basics</a></li>
<li class="chapter" data-level="17.6.2" data-path="exercises.html"><a href="exercises.html#beta-diversity-extra"><i class="fa fa-check"></i><b>17.6.2</b> Beta diversity extra</a></li>
</ul></li>
<li class="chapter" data-level="17.7" data-path="exercises.html"><a href="exercises.html#differential-abundance-1"><i class="fa fa-check"></i><b>17.7</b> Differential abundance</a>
<ul>
<li class="chapter" data-level="17.7.1" data-path="exercises.html"><a href="exercises.html#univariate-analyses"><i class="fa fa-check"></i><b>17.7.1</b> Univariate analyses</a></li>
<li class="chapter" data-level="17.7.2" data-path="exercises.html"><a href="exercises.html#differential-abundance-analysis-1"><i class="fa fa-check"></i><b>17.7.2</b> Differential abundance analysis</a></li>
</ul></li>
<li class="chapter" data-level="17.8" data-path="exercises.html"><a href="exercises.html#visualization-1"><i class="fa fa-check"></i><b>17.8</b> Visualization</a>
<ul>
<li class="chapter" data-level="17.8.1" data-path="exercises.html"><a href="exercises.html#multivariate-ordination"><i class="fa fa-check"></i><b>17.8.1</b> Multivariate ordination</a></li>
<li class="chapter" data-level="17.8.2" data-path="exercises.html"><a href="exercises.html#heatmap-visualization"><i class="fa fa-check"></i><b>17.8.2</b> Heatmap visualization</a></li>
</ul></li>
<li class="chapter" data-level="17.9" data-path="exercises.html"><a href="exercises.html#multiomics"><i class="fa fa-check"></i><b>17.9</b> Multiomics</a>
<ul>
<li class="chapter" data-level="17.9.1" data-path="exercises.html"><a href="exercises.html#multiassayexperiment-mae-data-container"><i class="fa fa-check"></i><b>17.9.1</b> MultiAssayExperiment (MAE) data container</a></li>
<li class="chapter" data-level="17.9.2" data-path="exercises.html"><a href="exercises.html#multi-omic-data-exploration"><i class="fa fa-check"></i><b>17.9.2</b> Multi-omic data exploration</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>IV Appendix</b></span></li>
<li class="chapter" data-level="18" data-path="extras.html"><a href="extras.html"><i class="fa fa-check"></i><b>18</b> Extra material</a>
<ul>
<li class="chapter" data-level="18.1" data-path="extras.html"><a href="extras.html#permanova-comparison"><i class="fa fa-check"></i><b>18.1</b> PERMANOVA comparison</a></li>
<li class="chapter" data-level="18.2" data-path="extras.html"><a href="extras.html#bayesian-multinomial-logistic-normal-models"><i class="fa fa-check"></i><b>18.2</b> Bayesian Multinomial Logistic-Normal Models</a></li>
<li class="chapter" data-level="18.3" data-path="extras.html"><a href="extras.html#interactive-3d-plots"><i class="fa fa-check"></i><b>18.3</b> Interactive 3D Plots</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="acknowledgments.html"><a href="acknowledgments.html"><i class="fa fa-check"></i><b>19</b> Acknowledgments</a></li>
<li class="chapter" data-level="" data-path="bibliography.html"><a href="bibliography.html"><i class="fa fa-check"></i>Bibliography</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Orchestrating Microbiome Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="quality-control" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">Chapter 5</span> Exploration and quality Control<a href="quality-control.html#quality-control" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<script>
document.addEventListener("click", function (event) {
    if (event.target.classList.contains("rebook-collapse")) {
        event.target.classList.toggle("active");
        var content = event.target.nextElementSibling;
        if (content.style.display === "block") {
            content.style.display = "none";
        } else {
            content.style.display = "block";
        }
    }
})
</script>
<style>
.rebook-collapse {
  background-color: #eee;
  color: #444;
  cursor: pointer;
  padding: 18px;
  width: 100%;
  border: none;
  text-align: left;
  outline: none;
  font-size: 15px;
}

.rebook-content {
  padding: 0 18px;
  display: none;
  overflow: hidden;
  background-color: #f1f1f1;
}
</style>
<p>This chapter focuses on the quality control and exploration of
microbiome data and establishes commonly used descriptive
summaries. Familiarizing with the peculiarities of a given data set is
the essential basis for any data analysis and model building.</p>
<p>The dataset should not suffer from severe technical biases, and you
should at least be aware of potential challenges, such as outliers,
biases, unexpected patterns and so forth. Standard summaries and
visualizations can help, and the rest comes with experience. The
exploration and quality control can be iterative processes.</p>
<div class="sourceCode" id="cb136"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb136-1"><a href="quality-control.html#cb136-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mia)</span></code></pre></div>
<div id="abundance" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> Abundance<a href="quality-control.html#abundance" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Abundance visualization is an important data exploration
approach. <code>miaViz</code> offers the function <code>plotAbundanceDensity</code> to plot
the most abundant taxa with several options.</p>
<p>Next, a few demonstrations are shown, using the <span class="citation">(<a href="#ref-Lahti2014" role="doc-biblioref">L. Lahti et al. 2014</a>)</span>
dataset. A Jitter plot based on relative abundance data, similar to
the one presented at <span class="citation">(<a href="#ref-Salosensaari2021" role="doc-biblioref">Salosensaari et al. 2021</a>)</span> supplementary figure 1, could
be visualized as follows:</p>
<div class="sourceCode" id="cb137"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb137-1"><a href="quality-control.html#cb137-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load example data</span></span>
<span id="cb137-2"><a href="quality-control.html#cb137-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(miaTime)</span>
<span id="cb137-3"><a href="quality-control.html#cb137-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(miaViz)</span>
<span id="cb137-4"><a href="quality-control.html#cb137-4" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(hitchip1006)</span>
<span id="cb137-5"><a href="quality-control.html#cb137-5" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> hitchip1006</span>
<span id="cb137-6"><a href="quality-control.html#cb137-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-7"><a href="quality-control.html#cb137-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Add relative abundances</span></span>
<span id="cb137-8"><a href="quality-control.html#cb137-8" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">transformCounts</span>(tse, <span class="at">MARGIN =</span> <span class="st">&quot;samples&quot;</span>, <span class="at">method =</span> <span class="st">&quot;relabundance&quot;</span>)</span>
<span id="cb137-9"><a href="quality-control.html#cb137-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-10"><a href="quality-control.html#cb137-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Use argument names</span></span>
<span id="cb137-11"><a href="quality-control.html#cb137-11" aria-hidden="true" tabindex="-1"></a><span class="co"># assay_name / assay_name / assay_name</span></span>
<span id="cb137-12"><a href="quality-control.html#cb137-12" aria-hidden="true" tabindex="-1"></a><span class="co"># depending on the mia package version</span></span>
<span id="cb137-13"><a href="quality-control.html#cb137-13" aria-hidden="true" tabindex="-1"></a><span class="fu">plotAbundanceDensity</span>(tse, <span class="at">layout =</span> <span class="st">&quot;jitter&quot;</span>, <span class="at">assay_name =</span> <span class="st">&quot;relabundance&quot;</span>,</span>
<span id="cb137-14"><a href="quality-control.html#cb137-14" aria-hidden="true" tabindex="-1"></a>                     <span class="at">n =</span> <span class="dv">40</span>, <span class="at">point_size=</span><span class="dv">1</span>, <span class="at">point_shape=</span><span class="dv">19</span>, <span class="at">point_alpha=</span><span class="fl">0.1</span>) <span class="sc">+</span> </span>
<span id="cb137-15"><a href="quality-control.html#cb137-15" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">scale_x_log10</span>(<span class="at">label=</span>scales<span class="sc">::</span>percent)</span></code></pre></div>
<p><img src="12_quality_control_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
<p>The relative abundance values for the top-5 taxonomic features can be
visualized as a density plot over a log scaled axis, with
“nationality” indicated by colors:</p>
<div class="sourceCode" id="cb138"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb138-1"><a href="quality-control.html#cb138-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotAbundanceDensity</span>(tse, <span class="at">layout =</span> <span class="st">&quot;density&quot;</span>, <span class="at">assay_name =</span> <span class="st">&quot;relabundance&quot;</span>,</span>
<span id="cb138-2"><a href="quality-control.html#cb138-2" aria-hidden="true" tabindex="-1"></a>                     <span class="at">n =</span> <span class="dv">5</span>, <span class="at">colour_by=</span><span class="st">&quot;nationality&quot;</span>, <span class="at">point_alpha=</span><span class="dv">1</span><span class="sc">/</span><span class="dv">10</span>) <span class="sc">+</span></span>
<span id="cb138-3"><a href="quality-control.html#cb138-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_log10</span>()</span></code></pre></div>
<p><img src="12_quality_control_files/figure-html/unnamed-chunk-3-1.png" width="672" /></p>
</div>
<div id="prevalence" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Prevalence<a href="quality-control.html#prevalence" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Prevalence quantifies the frequency of samples where certain microbes
were detected (above a given detection threshold). The prevalence can
be given as sample size (N) or percentage (unit interval).</p>
<p>Investigating prevalence allows you either to focus on changes which
pertain to the majority of the samples, or identify rare microbes,
which may be <em>conditionally abundant</em> in a small number of samples.</p>
<p>The population prevalence (frequency) at a 1% relative abundance
threshold (<code>detection = 1/100</code> and <code>as_relative = TRUE</code>), can look
like this.</p>
<div class="sourceCode" id="cb139"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb139-1"><a href="quality-control.html#cb139-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">getPrevalence</span>(tse, <span class="at">detection =</span> <span class="dv">1</span><span class="sc">/</span><span class="dv">100</span>, <span class="at">sort =</span> <span class="cn">TRUE</span>, <span class="at">as_relative =</span> <span class="cn">TRUE</span>))</span></code></pre></div>
<pre><code>## Faecalibacterium prausnitzii et rel.           Ruminococcus obeum et rel. 
##                               0.9522                               0.9140 
##   Oscillospira guillermondii et rel.        Clostridium symbiosum et rel. 
##                               0.8801                               0.8714 
##     Subdoligranulum variable at rel.     Clostridium orbiscindens et rel. 
##                               0.8358                               0.8315</code></pre>
<p>The function arguments <code>detection</code> and <code>as_relative</code> can also be used
to access, how many samples do pass a threshold for raw counts. Here,
the population prevalence (frequency) at the absolute abundance
threshold (<code>as_relative = FALSE</code>) at read count 1 (<code>detection = 1</code>) is
accessed.</p>
<div class="sourceCode" id="cb141"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb141-1"><a href="quality-control.html#cb141-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">getPrevalence</span>(tse, <span class="at">detection =</span> <span class="dv">1</span>, <span class="at">sort =</span> <span class="cn">TRUE</span>, <span class="at">assay_name =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb141-2"><a href="quality-control.html#cb141-2" aria-hidden="true" tabindex="-1"></a>                   <span class="at">as_relative =</span> <span class="cn">FALSE</span>))</span></code></pre></div>
<pre><code>##            Uncultured Mollicutes      Uncultured Clostridiales II 
##                                1                                1 
##       Uncultured Clostridiales I               Tannerella et rel. 
##                                1                                1 
##   Sutterella wadsworthia et rel. Subdoligranulum variable at rel. 
##                                1                                1</code></pre>
<p>If the output should be used for subsetting or storing the data in the
<code>rowData</code>, set <code>sort = FALSE</code>.</p>
<div id="prevalence-analysis" class="section level3 hasAnchor" number="5.2.1">
<h3><span class="header-section-number">5.2.1</span> Prevalence analysis<a href="quality-control.html#prevalence-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To investigate microbiome prevalence at a selected taxonomic level, two
approaches are available.</p>
<p>First the data can be agglomerated to the taxonomic level and <code>getPrevalence</code>
applied on the resulting object.</p>
<div class="sourceCode" id="cb143"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb143-1"><a href="quality-control.html#cb143-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Agglomerate taxa abundances to Phylum level, and add the new table</span></span>
<span id="cb143-2"><a href="quality-control.html#cb143-2" aria-hidden="true" tabindex="-1"></a><span class="co"># to the altExp slot</span></span>
<span id="cb143-3"><a href="quality-control.html#cb143-3" aria-hidden="true" tabindex="-1"></a><span class="fu">altExp</span>(tse,<span class="st">&quot;Phylum&quot;</span>) <span class="ot">&lt;-</span> <span class="fu">agglomerateByRank</span>(tse, <span class="st">&quot;Phylum&quot;</span>)</span>
<span id="cb143-4"><a href="quality-control.html#cb143-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Check prevalence for the Phylum abundance table from the altExp slot</span></span>
<span id="cb143-5"><a href="quality-control.html#cb143-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">getPrevalence</span>(<span class="fu">altExp</span>(tse,<span class="st">&quot;Phylum&quot;</span>), <span class="at">detection =</span> <span class="dv">1</span><span class="sc">/</span><span class="dv">100</span>, <span class="at">sort =</span> <span class="cn">TRUE</span>,</span>
<span id="cb143-6"><a href="quality-control.html#cb143-6" aria-hidden="true" tabindex="-1"></a>                   <span class="at">assay_name =</span> <span class="st">&quot;counts&quot;</span>, <span class="at">as_relative =</span> <span class="cn">TRUE</span>))</span></code></pre></div>
<pre><code>##      Firmicutes   Bacteroidetes  Actinobacteria  Proteobacteria Verrucomicrobia 
##       1.0000000       0.9852302       0.4821894       0.2988705       0.1277150 
##   Cyanobacteria 
##       0.0008688</code></pre>
<p>Alternatively, the <code>rank</code> argument could be set to perform the
agglomeration on the fly.</p>
<div class="sourceCode" id="cb145"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb145-1"><a href="quality-control.html#cb145-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">getPrevalence</span>(tse, <span class="at">rank =</span> <span class="st">&quot;Phylum&quot;</span>, <span class="at">detection =</span> <span class="dv">1</span><span class="sc">/</span><span class="dv">100</span>, <span class="at">sort =</span> <span class="cn">TRUE</span>,</span>
<span id="cb145-2"><a href="quality-control.html#cb145-2" aria-hidden="true" tabindex="-1"></a>                   <span class="at">assay_name =</span> <span class="st">&quot;counts&quot;</span>, <span class="at">as_relative =</span> <span class="cn">TRUE</span>))</span></code></pre></div>
<pre><code>##      Firmicutes   Bacteroidetes  Actinobacteria  Proteobacteria Verrucomicrobia 
##       1.0000000       0.9852302       0.4821894       0.2988705       0.1277150 
##   Cyanobacteria 
##       0.0008688</code></pre>
<p>Note that, by default, <code>na.rm = TRUE</code> is used for agglomeration in
<code>getPrevalence</code>, whereas the default for <code>agglomerateByRank</code> is
<code>FALSE</code> to prevent accidental data loss.</p>
<p>If you only need the names of the prevalent taxa, <code>getPrevalentTaxa</code>
is available. This returns the taxa that exceed the given prevalence
and detection thresholds.</p>
<div class="sourceCode" id="cb147"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb147-1"><a href="quality-control.html#cb147-1" aria-hidden="true" tabindex="-1"></a><span class="fu">getPrevalentTaxa</span>(tse, <span class="at">detection =</span> <span class="dv">0</span>, <span class="at">prevalence =</span> <span class="dv">50</span><span class="sc">/</span><span class="dv">100</span>)</span>
<span id="cb147-2"><a href="quality-control.html#cb147-2" aria-hidden="true" tabindex="-1"></a>prev <span class="ot">&lt;-</span> <span class="fu">getPrevalentTaxa</span>(tse, <span class="at">detection =</span> <span class="dv">0</span>, <span class="at">prevalence =</span> <span class="dv">50</span><span class="sc">/</span><span class="dv">100</span>,</span>
<span id="cb147-3"><a href="quality-control.html#cb147-3" aria-hidden="true" tabindex="-1"></a>                         <span class="at">rank =</span> <span class="st">&quot;Phylum&quot;</span>, <span class="at">sort =</span> <span class="cn">TRUE</span>)</span>
<span id="cb147-4"><a href="quality-control.html#cb147-4" aria-hidden="true" tabindex="-1"></a>prev</span></code></pre></div>
<p>Note that the <code>detection</code> and <code>prevalence</code> thresholds are not the same, since
<code>detection</code> can be applied to relative counts or absolute counts depending on
whether <code>as_relative</code> is set <code>TRUE</code> or <code>FALSE</code></p>
<p>The function ‘getPrevalentAbundance’ can be used to check the total
relative abundance of the prevalent taxa (between 0 and 1).</p>
</div>
<div id="rare-taxa" class="section level3 hasAnchor" number="5.2.2">
<h3><span class="header-section-number">5.2.2</span> Rare taxa<a href="quality-control.html#rare-taxa" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Related functions are available for the analysis of rare taxa
(<code>rareMembers</code>; <code>rareAbundance</code>; <code>lowAbundance</code>, <code>getRareTaxa</code>,
<code>subsetByRareTaxa</code>).</p>
</div>
<div id="plotting-prevalence" class="section level3 hasAnchor" number="5.2.3">
<h3><span class="header-section-number">5.2.3</span> Plotting prevalence<a href="quality-control.html#plotting-prevalence" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To plot the prevalence, add the prevalence of each taxon to
<code>rowData</code>. Here, we are analysing the Phylum level abundances, which
are stored in the <code>altExp</code> slot.</p>
<div class="sourceCode" id="cb148"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb148-1"><a href="quality-control.html#cb148-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(<span class="fu">altExp</span>(tse,<span class="st">&quot;Phylum&quot;</span>))<span class="sc">$</span>prevalence <span class="ot">&lt;-</span> </span>
<span id="cb148-2"><a href="quality-control.html#cb148-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">getPrevalence</span>(<span class="fu">altExp</span>(tse,<span class="st">&quot;Phylum&quot;</span>), <span class="at">detection =</span> <span class="dv">1</span><span class="sc">/</span><span class="dv">100</span>, <span class="at">sort =</span> <span class="cn">FALSE</span>,</span>
<span id="cb148-3"><a href="quality-control.html#cb148-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">assay_name =</span> <span class="st">&quot;counts&quot;</span>, <span class="at">as_relative =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p>The prevalences can then be plotted using the plotting functions from
the <code>scater</code> package.</p>
<div class="sourceCode" id="cb149"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb149-1"><a href="quality-control.html#cb149-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scater)</span>
<span id="cb149-2"><a href="quality-control.html#cb149-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotRowData</span>(<span class="fu">altExp</span>(tse,<span class="st">&quot;Phylum&quot;</span>), <span class="st">&quot;prevalence&quot;</span>, <span class="at">colour_by =</span> <span class="st">&quot;Phylum&quot;</span>)</span></code></pre></div>
<p><img src="12_quality_control_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
<p>The prevalence can also be visualized on the taxonomic tree with the
<code>miaViz</code> package.</p>
<div class="sourceCode" id="cb150"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb150-1"><a href="quality-control.html#cb150-1" aria-hidden="true" tabindex="-1"></a><span class="fu">altExps</span>(tse) <span class="ot">&lt;-</span> <span class="fu">splitByRanks</span>(tse)</span>
<span id="cb150-2"><a href="quality-control.html#cb150-2" aria-hidden="true" tabindex="-1"></a><span class="fu">altExps</span>(tse) <span class="ot">&lt;-</span></span>
<span id="cb150-3"><a href="quality-control.html#cb150-3" aria-hidden="true" tabindex="-1"></a>   <span class="fu">lapply</span>(<span class="fu">altExps</span>(tse),</span>
<span id="cb150-4"><a href="quality-control.html#cb150-4" aria-hidden="true" tabindex="-1"></a>          <span class="cf">function</span>(y){</span>
<span id="cb150-5"><a href="quality-control.html#cb150-5" aria-hidden="true" tabindex="-1"></a>              <span class="fu">rowData</span>(y)<span class="sc">$</span>prevalence <span class="ot">&lt;-</span> </span>
<span id="cb150-6"><a href="quality-control.html#cb150-6" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">getPrevalence</span>(y, <span class="at">detection =</span> <span class="dv">1</span><span class="sc">/</span><span class="dv">100</span>, <span class="at">sort =</span> <span class="cn">FALSE</span>,</span>
<span id="cb150-7"><a href="quality-control.html#cb150-7" aria-hidden="true" tabindex="-1"></a>                                <span class="at">assay_name =</span> <span class="st">&quot;counts&quot;</span>, <span class="at">as_relative =</span> <span class="cn">TRUE</span>)</span>
<span id="cb150-8"><a href="quality-control.html#cb150-8" aria-hidden="true" tabindex="-1"></a>              y</span>
<span id="cb150-9"><a href="quality-control.html#cb150-9" aria-hidden="true" tabindex="-1"></a>          })</span>
<span id="cb150-10"><a href="quality-control.html#cb150-10" aria-hidden="true" tabindex="-1"></a>top_phyla <span class="ot">&lt;-</span> <span class="fu">getTopTaxa</span>(<span class="fu">altExp</span>(tse,<span class="st">&quot;Phylum&quot;</span>),</span>
<span id="cb150-11"><a href="quality-control.html#cb150-11" aria-hidden="true" tabindex="-1"></a>                        <span class="at">method=</span><span class="st">&quot;prevalence&quot;</span>,</span>
<span id="cb150-12"><a href="quality-control.html#cb150-12" aria-hidden="true" tabindex="-1"></a>                        <span class="at">top=</span>5L,</span>
<span id="cb150-13"><a href="quality-control.html#cb150-13" aria-hidden="true" tabindex="-1"></a>                        <span class="at">assay_name=</span><span class="st">&quot;counts&quot;</span>)</span>
<span id="cb150-14"><a href="quality-control.html#cb150-14" aria-hidden="true" tabindex="-1"></a>top_phyla_mean <span class="ot">&lt;-</span> <span class="fu">getTopTaxa</span>(<span class="fu">altExp</span>(tse,<span class="st">&quot;Phylum&quot;</span>),</span>
<span id="cb150-15"><a href="quality-control.html#cb150-15" aria-hidden="true" tabindex="-1"></a>                             <span class="at">method=</span><span class="st">&quot;mean&quot;</span>,</span>
<span id="cb150-16"><a href="quality-control.html#cb150-16" aria-hidden="true" tabindex="-1"></a>                             <span class="at">top=</span>5L,</span>
<span id="cb150-17"><a href="quality-control.html#cb150-17" aria-hidden="true" tabindex="-1"></a>                             <span class="at">assay_name=</span><span class="st">&quot;counts&quot;</span>)</span>
<span id="cb150-18"><a href="quality-control.html#cb150-18" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">unsplitByRanks</span>(tse, <span class="at">ranks =</span> <span class="fu">taxonomyRanks</span>(tse)[<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>])</span>
<span id="cb150-19"><a href="quality-control.html#cb150-19" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">addTaxonomyTree</span>(x)</span></code></pre></div>
<p>After some preparation, the data is assembled and can be plotted with
<code>plotRowTree</code>.</p>
<div class="sourceCode" id="cb151"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb151-1"><a href="quality-control.html#cb151-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(miaViz)</span>
<span id="cb151-2"><a href="quality-control.html#cb151-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotRowTree</span>(x[<span class="fu">rowData</span>(x)<span class="sc">$</span>Phylum <span class="sc">%in%</span> top_phyla,],</span>
<span id="cb151-3"><a href="quality-control.html#cb151-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">edge_colour_by =</span> <span class="st">&quot;Phylum&quot;</span>,</span>
<span id="cb151-4"><a href="quality-control.html#cb151-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">tip_colour_by =</span> <span class="st">&quot;prevalence&quot;</span>,</span>
<span id="cb151-5"><a href="quality-control.html#cb151-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">node_colour_by =</span> <span class="st">&quot;prevalence&quot;</span>)</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:plot-prev-prev"></span>
<img src="12_quality_control_files/figure-html/plot-prev-prev-1.png" alt="Prevalence of top phyla as judged by prevalence" width="672" />
<p class="caption">
Figure 5.1: Prevalence of top phyla as judged by prevalence
</p>
</div>
<div class="sourceCode" id="cb152"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb152-1"><a href="quality-control.html#cb152-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotRowTree</span>(x[<span class="fu">rowData</span>(x)<span class="sc">$</span>Phylum <span class="sc">%in%</span> top_phyla_mean,],</span>
<span id="cb152-2"><a href="quality-control.html#cb152-2" aria-hidden="true" tabindex="-1"></a>            <span class="at">edge_colour_by =</span> <span class="st">&quot;Phylum&quot;</span>,</span>
<span id="cb152-3"><a href="quality-control.html#cb152-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">tip_colour_by =</span> <span class="st">&quot;prevalence&quot;</span>,</span>
<span id="cb152-4"><a href="quality-control.html#cb152-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">node_colour_by =</span> <span class="st">&quot;prevalence&quot;</span>)</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:plot-prev-mean"></span>
<img src="12_quality_control_files/figure-html/plot-prev-mean-1.png" alt="Prevalence of top phyla as judged by mean abundance" width="672" />
<p class="caption">
Figure 5.2: Prevalence of top phyla as judged by mean abundance
</p>
</div>
</div>
</div>
<div id="qc" class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">5.3</span> Quality control<a href="quality-control.html#qc" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Next, let us load the <code>GlobalPatterns</code> data set to illustrate standard
microbiome data summaries.</p>
<div class="sourceCode" id="cb153"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb153-1"><a href="quality-control.html#cb153-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mia)</span>
<span id="cb153-2"><a href="quality-control.html#cb153-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;GlobalPatterns&quot;</span>, <span class="at">package=</span><span class="st">&quot;mia&quot;</span>)</span>
<span id="cb153-3"><a href="quality-control.html#cb153-3" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> GlobalPatterns </span></code></pre></div>
<div id="top-taxa" class="section level3 hasAnchor" number="5.3.1">
<h3><span class="header-section-number">5.3.1</span> Top taxa<a href="quality-control.html#top-taxa" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The <code>getTopTaxa</code> identifies top taxa in the data.</p>
<div class="sourceCode" id="cb154"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb154-1"><a href="quality-control.html#cb154-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Pick the top taxa</span></span>
<span id="cb154-2"><a href="quality-control.html#cb154-2" aria-hidden="true" tabindex="-1"></a>top_features <span class="ot">&lt;-</span> <span class="fu">getTopTaxa</span>(tse, <span class="at">method=</span><span class="st">&quot;median&quot;</span>, <span class="at">top=</span><span class="dv">10</span>)</span>
<span id="cb154-3"><a href="quality-control.html#cb154-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb154-4"><a href="quality-control.html#cb154-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Check the information for these</span></span>
<span id="cb154-5"><a href="quality-control.html#cb154-5" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(tse)[top_features, <span class="fu">taxonomyRanks</span>(tse)]</span></code></pre></div>
<pre><code>## DataFrame with 10 rows and 7 columns
##            Kingdom         Phylum               Class             Order
##        &lt;character&gt;    &lt;character&gt;         &lt;character&gt;       &lt;character&gt;
## 549656    Bacteria  Cyanobacteria         Chloroplast     Stramenopiles
## 331820    Bacteria  Bacteroidetes         Bacteroidia     Bacteroidales
## 317182    Bacteria  Cyanobacteria         Chloroplast     Stramenopiles
## 94166     Bacteria Proteobacteria Gammaproteobacteria    Pasteurellales
## 279599    Bacteria  Cyanobacteria    Nostocophycideae        Nostocales
## 158660    Bacteria  Bacteroidetes         Bacteroidia     Bacteroidales
## 329744    Bacteria Actinobacteria      Actinobacteria   Actinomycetales
## 326977    Bacteria Actinobacteria      Actinobacteria Bifidobacteriales
## 248140    Bacteria  Bacteroidetes         Bacteroidia     Bacteroidales
## 550960    Bacteria Proteobacteria Gammaproteobacteria Enterobacteriales
##                    Family           Genus                Species
##               &lt;character&gt;     &lt;character&gt;            &lt;character&gt;
## 549656                 NA              NA                     NA
## 331820     Bacteroidaceae     Bacteroides                     NA
## 317182                 NA              NA                     NA
## 94166     Pasteurellaceae     Haemophilus Haemophilusparainflu..
## 279599        Nostocaceae  Dolichospermum                     NA
## 158660     Bacteroidaceae     Bacteroides                     NA
## 329744             ACK-M1              NA                     NA
## 326977 Bifidobacteriaceae Bifidobacterium Bifidobacteriumadole..
## 248140     Bacteroidaceae     Bacteroides      Bacteroidescaccae
## 550960 Enterobacteriaceae     Providencia                     NA</code></pre>
</div>
<div id="library-size-read-count" class="section level3 hasAnchor" number="5.3.2">
<h3><span class="header-section-number">5.3.2</span> Library size / read count<a href="quality-control.html#library-size-read-count" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The total counts/sample can be calculated using <code>perCellQCMetrics</code>/<code>addPerCellQC</code> from the <code>scater</code> package. The former one
just calculates the values, whereas the latter one directly adds them to
<code>colData</code>.</p>
<div class="sourceCode" id="cb156"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb156-1"><a href="quality-control.html#cb156-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scater)</span>
<span id="cb156-2"><a href="quality-control.html#cb156-2" aria-hidden="true" tabindex="-1"></a><span class="fu">perCellQCMetrics</span>(tse)</span></code></pre></div>
<pre><code>## DataFrame with 26 rows and 3 columns
##               sum  detected     total
##         &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
## CL3        864077      6964    864077
## CC1       1135457      7679   1135457
## SV1        697509      5729    697509
## M31Fcsw   1543451      2667   1543451
## M11Fcsw   2076476      2574   2076476
## ...           ...       ...       ...
## TS28       937466      2679    937466
## TS29      1211071      2629   1211071
## Even1     1216137      4213   1216137
## Even2      971073      3130    971073
## Even3     1078241      2776   1078241</code></pre>
<div class="sourceCode" id="cb158"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb158-1"><a href="quality-control.html#cb158-1" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">addPerCellQC</span>(tse)</span>
<span id="cb158-2"><a href="quality-control.html#cb158-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(tse)</span></code></pre></div>
<pre><code>## DataFrame with 26 rows and 10 columns
##         X.SampleID   Primer Final_Barcode Barcode_truncated_plus_T
##           &lt;factor&gt; &lt;factor&gt;      &lt;factor&gt;                 &lt;factor&gt;
## CL3        CL3      ILBC_01        AACGCA                   TGCGTT
## CC1        CC1      ILBC_02        AACTCG                   CGAGTT
## SV1        SV1      ILBC_03        AACTGT                   ACAGTT
## M31Fcsw    M31Fcsw  ILBC_04        AAGAGA                   TCTCTT
## M11Fcsw    M11Fcsw  ILBC_05        AAGCTG                   CAGCTT
## ...            ...      ...           ...                      ...
## TS28         TS28   ILBC_25        ACCAGA                   TCTGGT
## TS29         TS29   ILBC_26        ACCAGC                   GCTGGT
## Even1        Even1  ILBC_27        ACCGCA                   TGCGGT
## Even2        Even2  ILBC_28        ACCTCG                   CGAGGT
## Even3        Even3  ILBC_29        ACCTGT                   ACAGGT
##         Barcode_full_length SampleType
##                    &lt;factor&gt;   &lt;factor&gt;
## CL3             CTAGCGTGCGT      Soil 
## CC1             CATCGACGAGT      Soil 
## SV1             GTACGCACAGT      Soil 
## M31Fcsw         TCGACATCTCT      Feces
## M11Fcsw         CGACTGCAGCT      Feces
## ...                     ...        ...
## TS28            GCATCGTCTGG      Feces
## TS29            CTAGTCGCTGG      Feces
## Even1           TGACTCTGCGG      Mock 
## Even2           TCTGATCGAGG      Mock 
## Even3           AGAGAGACAGG      Mock 
##                                        Description       sum  detected
##                                           &lt;factor&gt; &lt;numeric&gt; &lt;numeric&gt;
## CL3     Calhoun South Carolina Pine soil, pH 4.9      864077      6964
## CC1     Cedar Creek Minnesota, grassland, pH 6.1     1135457      7679
## SV1     Sevilleta new Mexico, desert scrub, pH 8.3    697509      5729
## M31Fcsw M3, Day 1, fecal swab, whole body study      1543451      2667
## M11Fcsw M1, Day 1, fecal swab, whole body study      2076476      2574
## ...                                            ...       ...       ...
## TS28                                       Twin #1    937466      2679
## TS29                                       Twin #2   1211071      2629
## Even1                                      Even1     1216137      4213
## Even2                                      Even2      971073      3130
## Even3                                      Even3     1078241      2776
##             total
##         &lt;numeric&gt;
## CL3        864077
## CC1       1135457
## SV1        697509
## M31Fcsw   1543451
## M11Fcsw   2076476
## ...           ...
## TS28       937466
## TS29      1211071
## Even1     1216137
## Even2      971073
## Even3     1078241</code></pre>
<p>The distribution of calculated library sizes can be visualized as a
histogram (left), or by sorting the samples by library size (right).</p>
<div class="sourceCode" id="cb160"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb160-1"><a href="quality-control.html#cb160-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb160-2"><a href="quality-control.html#cb160-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb160-3"><a href="quality-control.html#cb160-3" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">as.data.frame</span>(<span class="fu">colData</span>(tse))) <span class="sc">+</span></span>
<span id="cb160-4"><a href="quality-control.html#cb160-4" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x =</span> sum), <span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;gray&quot;</span>, <span class="at">bins =</span> <span class="dv">30</span>) <span class="sc">+</span></span>
<span id="cb160-5"><a href="quality-control.html#cb160-5" aria-hidden="true" tabindex="-1"></a>        <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Library size&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Frequency (n)&quot;</span>) <span class="sc">+</span> </span>
<span id="cb160-6"><a href="quality-control.html#cb160-6" aria-hidden="true" tabindex="-1"></a>        <span class="co"># scale_x_log10(breaks = scales::trans_breaks(&quot;log10&quot;, function(x) 10^x), </span></span>
<span id="cb160-7"><a href="quality-control.html#cb160-7" aria-hidden="true" tabindex="-1"></a>        <span class="co"># labels = scales::trans_format(&quot;log10&quot;, scales::math_format(10^.x))) +</span></span>
<span id="cb160-8"><a href="quality-control.html#cb160-8" aria-hidden="true" tabindex="-1"></a>        <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb160-9"><a href="quality-control.html#cb160-9" aria-hidden="true" tabindex="-1"></a>        <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(), <span class="co"># Removes the grid</span></span>
<span id="cb160-10"><a href="quality-control.html#cb160-10" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb160-11"><a href="quality-control.html#cb160-11" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb160-12"><a href="quality-control.html#cb160-12" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb160-13"><a href="quality-control.html#cb160-13" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">&quot;black&quot;</span>)) <span class="co"># Adds y-axis</span></span>
<span id="cb160-14"><a href="quality-control.html#cb160-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb160-15"><a href="quality-control.html#cb160-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb160-16"><a href="quality-control.html#cb160-16" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(tse)) <span class="sc">%&gt;%</span></span>
<span id="cb160-17"><a href="quality-control.html#cb160-17" aria-hidden="true" tabindex="-1"></a>        <span class="fu">arrange</span>(sum) <span class="sc">%&gt;%</span></span>
<span id="cb160-18"><a href="quality-control.html#cb160-18" aria-hidden="true" tabindex="-1"></a>        <span class="fu">mutate</span>(<span class="at">index =</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">n</span>())</span>
<span id="cb160-19"><a href="quality-control.html#cb160-19" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">y =</span> index, <span class="at">x =</span> sum<span class="sc">/</span><span class="fl">1e6</span>)) <span class="sc">+</span></span>
<span id="cb160-20"><a href="quality-control.html#cb160-20" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_point</span>() <span class="sc">+</span>  </span>
<span id="cb160-21"><a href="quality-control.html#cb160-21" aria-hidden="true" tabindex="-1"></a>        <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Library size (million reads)&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Sample index&quot;</span>) <span class="sc">+</span>  </span>
<span id="cb160-22"><a href="quality-control.html#cb160-22" aria-hidden="true" tabindex="-1"></a>        <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb160-23"><a href="quality-control.html#cb160-23" aria-hidden="true" tabindex="-1"></a>        <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(), <span class="co"># Removes the grid</span></span>
<span id="cb160-24"><a href="quality-control.html#cb160-24" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb160-25"><a href="quality-control.html#cb160-25" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb160-26"><a href="quality-control.html#cb160-26" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb160-27"><a href="quality-control.html#cb160-27" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">&quot;black&quot;</span>)) <span class="co"># Adds y-axis</span></span>
<span id="cb160-28"><a href="quality-control.html#cb160-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb160-29"><a href="quality-control.html#cb160-29" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb160-30"><a href="quality-control.html#cb160-30" aria-hidden="true" tabindex="-1"></a>p1 <span class="sc">+</span> p2</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:plot-viz-lib-size-1"></span>
<img src="12_quality_control_files/figure-html/plot-viz-lib-size-1-1.png" alt="Library size distribution." width="768" />
<p class="caption">
Figure 5.3: Library size distribution.
</p>
</div>
<p>Library sizes other variables from <code>colData</code> can be
visualized by using specified function called <code>plotColData</code>.</p>
<div class="sourceCode" id="cb161"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb161-1"><a href="quality-control.html#cb161-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb161-2"><a href="quality-control.html#cb161-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Sort samples by read count, order the factor levels, and store back to tse as DataFrame</span></span>
<span id="cb161-3"><a href="quality-control.html#cb161-3" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">TODO</span><span class="co">: plotColData could include an option for sorting samples based on colData variables</span></span>
<span id="cb161-4"><a href="quality-control.html#cb161-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(tse) <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(tse)) <span class="sc">%&gt;%</span></span>
<span id="cb161-5"><a href="quality-control.html#cb161-5" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">arrange</span>(X.SampleID) <span class="sc">%&gt;%</span></span>
<span id="cb161-6"><a href="quality-control.html#cb161-6" aria-hidden="true" tabindex="-1"></a>             <span class="fu">mutate</span>(<span class="at">X.SampleID =</span> <span class="fu">factor</span>(X.SampleID, <span class="at">levels=</span>X.SampleID)) <span class="sc">%&gt;%</span></span>
<span id="cb161-7"><a href="quality-control.html#cb161-7" aria-hidden="true" tabindex="-1"></a>         DataFrame</span>
<span id="cb161-8"><a href="quality-control.html#cb161-8" aria-hidden="true" tabindex="-1"></a><span class="fu">plotColData</span>(tse,<span class="st">&quot;sum&quot;</span>,<span class="st">&quot;X.SampleID&quot;</span>, <span class="at">colour_by =</span> <span class="st">&quot;SampleType&quot;</span>) <span class="sc">+</span> </span>
<span id="cb161-9"><a href="quality-control.html#cb161-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust=</span><span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb161-10"><a href="quality-control.html#cb161-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&quot;Library size (N)&quot;</span>, <span class="at">x =</span> <span class="st">&quot;Sample ID&quot;</span>)       </span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:plot-viz-lib-size-2"></span>
<img src="12_quality_control_files/figure-html/plot-viz-lib-size-2-1.png" alt="Library sizes per sample." width="768" />
<p class="caption">
Figure 5.4: Library sizes per sample.
</p>
</div>
<div class="sourceCode" id="cb162"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb162-1"><a href="quality-control.html#cb162-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotColData</span>(tse,<span class="st">&quot;sum&quot;</span>,<span class="st">&quot;SampleType&quot;</span>, <span class="at">colour_by =</span> <span class="st">&quot;SampleType&quot;</span>) <span class="sc">+</span> </span>
<span id="cb162-2"><a href="quality-control.html#cb162-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust=</span><span class="dv">1</span>))</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:plot-viz-lib-size-3"></span>
<img src="12_quality_control_files/figure-html/plot-viz-lib-size-3-1.png" alt="Library sizes per sample type." width="768" />
<p class="caption">
Figure 5.5: Library sizes per sample type.
</p>
</div>
<p>In addition, data can be rarefied with
<a href="https://microbiome.github.io/mia/reference/subsampleCounts.html">subsampleCounts</a>,
which normalises the samples to an equal number of reads. However,
this practice has been discouraged for the analysis of differentially
abundant microorganisms (see <span class="citation">(<a href="#ref-mcmurdie2014waste" role="doc-biblioref">P. J. McMurdie and Holmes 2014</a>)</span>).</p>
</div>
<div id="contaminant-sequences" class="section level3 hasAnchor" number="5.3.3">
<h3><span class="header-section-number">5.3.3</span> Contaminant sequences<a href="quality-control.html#contaminant-sequences" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Samples might be contaminated with exogenous sequences. The impact of
each contaminant can be estimated based on their frequencies and
concentrations across the samples.</p>
<p>The following <a href="https://microbiome.github.io/mia/reference/isContaminant.html">decontam
functions</a>
are based on the <span class="citation">(<a href="#ref-davis2018simple" role="doc-biblioref">Davis et al. 2018</a>)</span> and support such functionality:</p>
<ul>
<li><code>isContaminant</code>, <code>isNotContaminant</code></li>
<li><code>addContaminantQC</code>, <code>addNotContaminantQC</code></li>
</ul>
</div>
</div>
<div id="session-info-2" class="section level2 unnumbered hasAnchor">
<h2>Session Info<a href="quality-control.html#session-info-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<button class="rebook-collapse">
View session info
</button>
<div class="rebook-content">
<pre><code>R version 4.3.0 (2023-04-21)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.2 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: UTC
tzcode source: system (glibc)

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] patchwork_1.1.2                dplyr_1.1.2                   
 [3] scater_1.28.0                  scuttle_1.10.0                
 [5] miaViz_1.9.1                   ggraph_2.1.0                  
 [7] ggplot2_3.4.2                  miaTime_0.1.21                
 [9] mia_1.9.2                      MultiAssayExperiment_1.26.0   
[11] TreeSummarizedExperiment_2.1.4 Biostrings_2.68.0             
[13] XVector_0.40.0                 SingleCellExperiment_1.22.0   
[15] SummarizedExperiment_1.30.0    Biobase_2.60.0                
[17] GenomicRanges_1.52.0           GenomeInfoDb_1.36.0           
[19] IRanges_2.34.0                 S4Vectors_0.38.0              
[21] BiocGenerics_0.46.0            MatrixGenerics_1.12.0         
[23] matrixStats_0.63.0-9003        BiocStyle_2.28.0              
[25] rebook_1.9.0                  

loaded via a namespace (and not attached):
  [1] jsonlite_1.8.4              CodeDepends_0.6.5          
  [3] magrittr_2.0.3              ggbeeswarm_0.7.1           
  [5] farver_2.1.1                rmarkdown_2.21             
  [7] zlibbioc_1.46.0             vctrs_0.6.2                
  [9] memoise_2.0.1               DelayedMatrixStats_1.22.0  
 [11] RCurl_1.98-1.12             ggtree_3.8.0               
 [13] htmltools_0.5.5             BiocNeighbors_1.18.0       
 [15] gridGraphics_0.5-1          sass_0.4.5                 
 [17] bslib_0.4.2                 plyr_1.8.8                 
 [19] DECIPHER_2.28.0             cachem_1.0.7               
 [21] igraph_1.4.2                lifecycle_1.0.3            
 [23] pkgconfig_2.0.3             rsvd_1.0.5                 
 [25] Matrix_1.5-4                R6_2.5.1                   
 [27] fastmap_1.1.1               GenomeInfoDbData_1.2.10    
 [29] digest_0.6.31               aplot_0.1.10               
 [31] colorspace_2.1-0            ggnewscale_0.4.8           
 [33] irlba_2.3.5.1               RSQLite_2.3.1              
 [35] vegan_2.6-4                 beachmat_2.16.0            
 [37] labeling_0.4.2              filelock_1.0.2             
 [39] fansi_1.0.4                 polyclip_1.10-4            
 [41] mgcv_1.8-42                 compiler_4.3.0             
 [43] bit64_4.0.5                 withr_2.5.0                
 [45] BiocParallel_1.34.0         viridis_0.6.2              
 [47] DBI_1.1.3                   highr_0.10                 
 [49] ggforce_0.4.1               MASS_7.3-59                
 [51] DelayedArray_0.25.0         permute_0.9-7              
 [53] tools_4.3.0                 vipor_0.4.5                
 [55] beeswarm_0.4.0              ape_5.7-1                  
 [57] glue_1.6.2                  nlme_3.1-162               
 [59] grid_4.3.0                  cluster_2.1.4              
 [61] reshape2_1.4.4              generics_0.1.3             
 [63] gtable_0.3.3                tidyr_1.3.0                
 [65] BiocSingular_1.16.0         tidygraph_1.2.3            
 [67] ScaledMatrix_1.7.1          utf8_1.2.3                 
 [69] ggrepel_0.9.3               pillar_1.9.0               
 [71] stringr_1.5.0               yulab.utils_0.0.6          
 [73] splines_4.3.0               tweenr_2.0.2               
 [75] treeio_1.24.0               lattice_0.21-8             
 [77] bit_4.0.5                   tidyselect_1.2.0           
 [79] DirichletMultinomial_1.42.0 knitr_1.42                 
 [81] gridExtra_2.3               bookdown_0.33              
 [83] xfun_0.39                   graphlayouts_0.8.4         
 [85] stringi_1.7.12              lazyeval_0.2.2             
 [87] ggfun_0.0.9                 yaml_2.3.7                 
 [89] evaluate_0.20               codetools_0.2-19           
 [91] tibble_3.2.1                BiocManager_1.30.20        
 [93] graph_1.78.0                ggplotify_0.1.0            
 [95] cli_3.6.1                   munsell_0.5.0              
 [97] jquerylib_0.1.4             Rcpp_1.0.10                
 [99] dir.expiry_1.8.0            XML_3.99-0.14              
[101] parallel_4.3.0              blob_1.2.4                 
[103] sparseMatrixStats_1.12.0    bitops_1.0-7               
[105] decontam_1.20.0             viridisLite_0.4.1          
[107] tidytree_0.4.2              scales_1.2.1               
[109] purrr_1.0.1                 crayon_1.5.2               
[111] rlang_1.1.0                 cowplot_1.1.1              </code></pre>
</div>

</div>
</div>
<h3>Bibliography<a href="bibliography.html#bibliography" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-davis2018simple" class="csl-entry">
Davis, Nicole M, Diana M Proctor, Susan P Holmes, David A Relman, and Benjamin J Callahan. 2018. <span>“Simple Statistical Identification and Removal of Contaminant Sequences in Marker-Gene and Metagenomics Data.”</span> <em>Microbiome</em> 6 (1): 1–14.
</div>
<div id="ref-Lahti2014" class="csl-entry">
Lahti, L, JSalojärvi, A Salonen, M Scheffer, and WM de Vos. 2014. <span>“Tipping Elements in the Human Intestinal Ecosystem.”</span> <em>Nature Communications</em> 2014: 1–10. https://doi.org/<a href="https://doi.org/10.1038/ncomms5344">https://doi.org/10.1038/ncomms5344</a>.
</div>
<div id="ref-mcmurdie2014waste" class="csl-entry">
McMurdie, Paul J, and Susan Holmes. 2014. <span>“Waste Not, Want Not: Why Rarefying Microbiome Data Is Inadmissible.”</span> <em>PLoS Computational Biology</em> 10 (4): e1003531.
</div>
<div id="ref-Salosensaari2021" class="csl-entry">
Salosensaari, Aaro, Ville Laitinen, Aki Havulinna, Guillaume Méric, Susan Cheng, Markus Perola, Liisa Valsta, et al. 2021. <span>“Taxonomic Signatures of Cause-Specific Mortality Risk in Human Gut Microbiome.”</span> <em>Nature Communications</em> 12: 1–8. <a href="https://www.nature.com/articles/s41467-021-22962-y">https://www.nature.com/articles/s41467-021-22962-y</a>.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="datamanipulation.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="taxonomic-information.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["OMA.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
