<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Taxonomic information | Orchestrating Microbiome Analysis</title>
  <meta name="description" content="Chapter 3 Taxonomic information | Orchestrating Microbiome Analysis" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Taxonomic information | Orchestrating Microbiome Analysis" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://github.com/microbiome/OMA" />
  
  
  <meta name="github-repo" content="microbiome/OMA" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Taxonomic information | Orchestrating Microbiome Analysis" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="overview.html"/>
<link rel="next" href="quality-control.html"/>
<script src="libs/header-attrs-2.9/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Orchestrating Microbiome Analysis</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="part"><span><b>I Introduction</b></span></li>
<li class="chapter" data-level="1" data-path="data-introduction.html"><a href="data-introduction.html"><i class="fa fa-check"></i><b>1</b> Data Infrastructure</a>
<ul>
<li class="chapter" data-level="1.1" data-path="data-introduction.html"><a href="data-introduction.html#installation"><i class="fa fa-check"></i><b>1.1</b> Installation</a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="data-introduction.html"><a href="data-introduction.html#packages"><i class="fa fa-check"></i><b>1.1.1</b> Packages</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="data-introduction.html"><a href="data-introduction.html#background"><i class="fa fa-check"></i><b>1.2</b> Background</a></li>
<li class="chapter" data-level="1.3" data-path="data-introduction.html"><a href="data-introduction.html#loading-experimental-microbiome-data"><i class="fa fa-check"></i><b>1.3</b> Loading experimental microbiome data</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="data-introduction.html"><a href="data-introduction.html#importing-data-from-external-files"><i class="fa fa-check"></i><b>1.3.1</b> Importing data from external files</a></li>
<li class="chapter" data-level="1.3.2" data-path="data-introduction.html"><a href="data-introduction.html#conversions-between-data-formats-in-r"><i class="fa fa-check"></i><b>1.3.2</b> Conversions between data formats in R</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="data-introduction.html"><a href="data-introduction.html#metadata"><i class="fa fa-check"></i><b>1.4</b> Metadata</a></li>
<li class="chapter" data-level="1.5" data-path="data-introduction.html"><a href="data-introduction.html#microbiome-and-tree-data-specific-aspects"><i class="fa fa-check"></i><b>1.5</b> Microbiome and tree data specific aspects</a>
<ul>
<li class="chapter" data-level="1.5.1" data-path="data-introduction.html"><a href="data-introduction.html#assays"><i class="fa fa-check"></i><b>1.5.1</b> Assays</a></li>
<li class="chapter" data-level="1.5.2" data-path="data-introduction.html"><a href="data-introduction.html#coldata"><i class="fa fa-check"></i><b>1.5.2</b> colData</a></li>
<li class="chapter" data-level="1.5.3" data-path="data-introduction.html"><a href="data-introduction.html#rowdata"><i class="fa fa-check"></i><b>1.5.3</b> rowData</a></li>
<li class="chapter" data-level="1.5.4" data-path="data-introduction.html"><a href="data-introduction.html#rowtree"><i class="fa fa-check"></i><b>1.5.4</b> rowTree</a></li>
</ul></li>
<li class="chapter" data-level="1.6" data-path="data-introduction.html"><a href="data-introduction.html#data-conversion"><i class="fa fa-check"></i><b>1.6</b> Data conversion</a>
<ul>
<li class="chapter" data-level="1.6.1" data-path="data-introduction.html"><a href="data-introduction.html#tidy-data"><i class="fa fa-check"></i><b>1.6.1</b> Tidy data</a></li>
</ul></li>
<li class="chapter" data-level="1.7" data-path="data-introduction.html"><a href="data-introduction.html#conclusion"><i class="fa fa-check"></i><b>1.7</b> Conclusion</a></li>
<li class="chapter" data-level="1.8" data-path="data-introduction.html"><a href="data-introduction.html#additional-reading"><i class="fa fa-check"></i><b>1.8</b> Additional Reading</a>
<ul>
<li class="chapter" data-level="1.8.1" data-path="data-introduction.html"><a href="data-introduction.html#lecture-slides"><i class="fa fa-check"></i><b>1.8.1</b> Lecture slides</a></li>
<li class="chapter" data-level="1.8.2" data-path="data-introduction.html"><a href="data-introduction.html#r-programming-resources"><i class="fa fa-check"></i><b>1.8.2</b> R programming resources</a></li>
<li class="chapter" data-level="1.8.3" data-path="data-introduction.html"><a href="data-introduction.html#resources-for-treesummarizedexperiment"><i class="fa fa-check"></i><b>1.8.3</b> Resources for TreeSummarizedExperiment</a></li>
<li class="chapter" data-level="1.8.4" data-path="data-introduction.html"><a href="data-introduction.html#resources-for-phyloseq"><i class="fa fa-check"></i><b>1.8.4</b> Resources for phyloseq</a></li>
<li class="chapter" data-level="1.8.5" data-path="data-introduction.html"><a href="data-introduction.html#further-reading"><i class="fa fa-check"></i><b>1.8.5</b> Further reading</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-introduction.html"><a href="data-introduction.html#session-info"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="part"><span><b>II Focus Topics</b></span></li>
<li class="chapter" data-level="2" data-path="overview.html"><a href="overview.html"><i class="fa fa-check"></i><b>2</b> Overview</a>
<ul>
<li class="chapter" data-level="2.1" data-path="overview.html"><a href="overview.html#introduction"><i class="fa fa-check"></i><b>2.1</b> Introduction</a></li>
<li class="chapter" data-level="2.2" data-path="overview.html"><a href="overview.html#microbiome-experiment-types"><i class="fa fa-check"></i><b>2.2</b> Microbiome experiment types</a></li>
<li class="chapter" data-level="2.3" data-path="overview.html"><a href="overview.html#processing-steps"><i class="fa fa-check"></i><b>2.3</b> Processing steps</a></li>
<li class="chapter" data-level="2.4" data-path="overview.html"><a href="overview.html#quick-start"><i class="fa fa-check"></i><b>2.4</b> Quick Start</a></li>
<li class="chapter" data-level="" data-path="overview.html"><a href="overview.html#session-info-1"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="taxonomic-information.html"><a href="taxonomic-information.html"><i class="fa fa-check"></i><b>3</b> Taxonomic information</a>
<ul>
<li class="chapter" data-level="3.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#assigning-taxonomic-information."><i class="fa fa-check"></i><b>3.1</b> Assigning taxonomic information.</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#dada2"><i class="fa fa-check"></i><b>3.1.1</b> dada2</a></li>
<li class="chapter" data-level="3.1.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#decipher"><i class="fa fa-check"></i><b>3.1.2</b> DECIPHER</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#functions-to-access-taxonomic-information"><i class="fa fa-check"></i><b>3.2</b> Functions to access taxonomic information</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#generate-a-taxonomic-tree-on-the-fly"><i class="fa fa-check"></i><b>3.2.1</b> Generate a taxonomic tree on the fly</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="taxonomic-information.html"><a href="taxonomic-information.html#data-agglomeration"><i class="fa fa-check"></i><b>3.3</b> Data agglomeration</a></li>
<li class="chapter" data-level="3.4" data-path="taxonomic-information.html"><a href="taxonomic-information.html#pick-specific"><i class="fa fa-check"></i><b>3.4</b> Pick specific</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#abundances-of-all-taxa-in-specific-sample"><i class="fa fa-check"></i><b>3.4.1</b> Abundances of all taxa in specific sample</a></li>
<li class="chapter" data-level="3.4.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#abundances-of-specific-taxa-in-all-samples"><i class="fa fa-check"></i><b>3.4.2</b> Abundances of specific taxa in all samples</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="taxonomic-information.html"><a href="taxonomic-information.html#session-info-2"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>4</b> Quality Control</a>
<ul>
<li class="chapter" data-level="4.1" data-path="quality-control.html"><a href="quality-control.html#get-top-taxa-and-taxonomy"><i class="fa fa-check"></i><b>4.1</b> Get top taxa and taxonomy</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="quality-control.html"><a href="quality-control.html#features"><i class="fa fa-check"></i><b>4.1.1</b> Features</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="quality-control.html"><a href="quality-control.html#library-size"><i class="fa fa-check"></i><b>4.2</b> Library size</a></li>
<li class="chapter" data-level="" data-path="quality-control.html"><a href="quality-control.html#session-info-3"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html"><i class="fa fa-check"></i><b>5</b> Microbiome Exploration</a>
<ul>
<li class="chapter" data-level="5.1" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#prevalence"><i class="fa fa-check"></i><b>5.1</b> Prevalence</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#prevalent-microbiota-analysis"><i class="fa fa-check"></i><b>5.1.1</b> Prevalent microbiota analysis</a></li>
<li class="chapter" data-level="5.1.2" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#plotting-prevalence"><i class="fa fa-check"></i><b>5.1.2</b> Plotting prevalence</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#session-info-4"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="microbiome-diversity.html"><a href="microbiome-diversity.html"><i class="fa fa-check"></i><b>6</b> Microbiome Diversity</a>
<ul>
<li class="chapter" data-level="6.1" data-path="microbiome-diversity.html"><a href="microbiome-diversity.html#alpha-diversity"><i class="fa fa-check"></i><b>6.1</b> Alpha diversity</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="microbiome-diversity.html"><a href="microbiome-diversity.html#estimating-alpha-diversity"><i class="fa fa-check"></i><b>6.1.1</b> Estimating alpha diversity</a></li>
<li class="chapter" data-level="6.1.2" data-path="microbiome-diversity.html"><a href="microbiome-diversity.html#visualize-alpha-diversities"><i class="fa fa-check"></i><b>6.1.2</b> Visualize alpha diversities</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="microbiome-diversity.html"><a href="microbiome-diversity.html#beta-diversity"><i class="fa fa-check"></i><b>6.2</b> Beta diversity</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="microbiome-diversity.html"><a href="microbiome-diversity.html#explained-variance"><i class="fa fa-check"></i><b>6.2.1</b> Explained variance</a></li>
<li class="chapter" data-level="6.2.2" data-path="microbiome-diversity.html"><a href="microbiome-diversity.html#estimating-beta-diversity"><i class="fa fa-check"></i><b>6.2.2</b> Estimating beta diversity</a></li>
<li class="chapter" data-level="6.2.3" data-path="microbiome-diversity.html"><a href="microbiome-diversity.html#other-ordination-methods"><i class="fa fa-check"></i><b>6.2.3</b> Other ordination methods</a></li>
<li class="chapter" data-level="6.2.4" data-path="microbiome-diversity.html"><a href="microbiome-diversity.html#visualizing-the-most-dominant-genus-on-pcoa"><i class="fa fa-check"></i><b>6.2.4</b> Visualizing the most dominant genus on PCoA</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="microbiome-diversity.html"><a href="microbiome-diversity.html#community-comparisons-todo-combine-with-the-material-above-for-simplicity"><i class="fa fa-check"></i><b>6.3</b> Community comparisons [TODO combine with the material above for simplicity?]</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="microbiome-diversity.html"><a href="microbiome-diversity.html#testing-differences-in-community-composition-between-sample-groups"><i class="fa fa-check"></i><b>6.3.1</b> Testing differences in community composition between sample groups</a></li>
<li class="chapter" data-level="6.3.2" data-path="microbiome-diversity.html"><a href="microbiome-diversity.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>6.3.2</b> Checking the homogeneity condition</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="microbiome-diversity.html"><a href="microbiome-diversity.html#further-reading-1"><i class="fa fa-check"></i><b>6.4</b> Further reading</a></li>
<li class="chapter" data-level="" data-path="microbiome-diversity.html"><a href="microbiome-diversity.html#session-info-5"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="microbiome-community.html"><a href="microbiome-community.html"><i class="fa fa-check"></i><b>7</b> Microbiome Community</a>
<ul>
<li class="chapter" data-level="7.1" data-path="microbiome-community.html"><a href="microbiome-community.html#community-composition"><i class="fa fa-check"></i><b>7.1</b> Community composition</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="microbiome-community.html"><a href="microbiome-community.html#composition-barplot"><i class="fa fa-check"></i><b>7.1.1</b> Composition barplot</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="microbiome-community.html"><a href="microbiome-community.html#community-typing"><i class="fa fa-check"></i><b>7.2</b> Community typing</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="microbiome-community.html"><a href="microbiome-community.html#dirichlet-multinomial-mixtures-dmm"><i class="fa fa-check"></i><b>7.2.1</b> Dirichlet Multinomial Mixtures (DMM)</a></li>
<li class="chapter" data-level="7.2.2" data-path="microbiome-community.html"><a href="microbiome-community.html#pcoa-for-asv-level-data-with-bray-curtis-with-dmm-clusters-shown-with-colors"><i class="fa fa-check"></i><b>7.2.2</b> PCoA for ASV-level data with Bray-Curtis; with DMM clusters shown with colors</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="microbiome-community.html"><a href="microbiome-community.html#session-info-6"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="differential-abundance.html"><a href="differential-abundance.html"><i class="fa fa-check"></i><b>8</b> Differential abundance</a>
<ul>
<li class="chapter" data-level="" data-path="differential-abundance.html"><a href="differential-abundance.html#session-info-7"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="microbiome-timeseries.html"><a href="microbiome-timeseries.html"><i class="fa fa-check"></i><b>9</b> Microbiome time series</a>
<ul>
<li class="chapter" data-level="9.1" data-path="microbiome-timeseries.html"><a href="microbiome-timeseries.html#stability"><i class="fa fa-check"></i><b>9.1</b> Stability</a></li>
<li class="chapter" data-level="9.2" data-path="microbiome-timeseries.html"><a href="microbiome-timeseries.html#tipping-elements"><i class="fa fa-check"></i><b>9.2</b> Tipping elements</a></li>
<li class="chapter" data-level="" data-path="microbiome-timeseries.html"><a href="microbiome-timeseries.html#session-info-8"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="multitable.html"><a href="multitable.html"><i class="fa fa-check"></i><b>10</b> Representation of multiple data tables</a>
<ul>
<li class="chapter" data-level="10.1" data-path="multitable.html"><a href="multitable.html#assay-data"><i class="fa fa-check"></i><b>10.1</b> Assay data</a></li>
<li class="chapter" data-level="10.2" data-path="multitable.html"><a href="multitable.html#alternative-experiments"><i class="fa fa-check"></i><b>10.2</b> Alternative experiments</a></li>
<li class="chapter" data-level="10.3" data-path="multitable.html"><a href="multitable.html#multiassayexperiments"><i class="fa fa-check"></i><b>10.3</b> MultiAssayExperiments</a></li>
<li class="chapter" data-level="" data-path="multitable.html"><a href="multitable.html#session-info-9"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="part"><span><b>III Appendix</b></span></li>
<li class="chapter" data-level="11" data-path="example-data.html"><a href="example-data.html"><i class="fa fa-check"></i><b>11</b> Example data</a>
<ul>
<li class="chapter" data-level="11.1" data-path="example-data.html"><a href="example-data.html#package-data"><i class="fa fa-check"></i><b>11.1</b> Package data</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="example-data.html"><a href="example-data.html#globalpatterns"><i class="fa fa-check"></i><b>11.1.1</b> GlobalPatterns</a></li>
<li class="chapter" data-level="11.1.2" data-path="example-data.html"><a href="example-data.html#enterotype"><i class="fa fa-check"></i><b>11.1.2</b> Enterotype</a></li>
<li class="chapter" data-level="11.1.3" data-path="example-data.html"><a href="example-data.html#esophagus"><i class="fa fa-check"></i><b>11.1.3</b> Esophagus</a></li>
<li class="chapter" data-level="11.1.4" data-path="example-data.html"><a href="example-data.html#soilrep"><i class="fa fa-check"></i><b>11.1.4</b> Soilrep</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="example-data.html"><a href="example-data.html#session-info-10"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="part"><span><b>IV Appendix</b></span></li>
<li class="chapter" data-level="12" data-path="function-ref.html"><a href="function-ref.html"><i class="fa fa-check"></i><b>12</b> Function reference</a>
<ul>
<li class="chapter" data-level="12.1" data-path="function-ref.html"><a href="function-ref.html#objects"><i class="fa fa-check"></i><b>12.1</b> Objects</a></li>
<li class="chapter" data-level="12.2" data-path="function-ref.html"><a href="function-ref.html#reduced-dimension-pca-t-sne-umap-mds-etc"><i class="fa fa-check"></i><b>12.2</b> Reduced dimension (PCA, t-SNE, UMAP, MDS, etc)</a></li>
<li class="chapter" data-level="12.3" data-path="function-ref.html"><a href="function-ref.html#plotting"><i class="fa fa-check"></i><b>12.3</b> Plotting</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="bibliography.html"><a href="bibliography.html"><i class="fa fa-check"></i>Bibliography</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Orchestrating Microbiome Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="taxonomic-information" class="section level1" number="3">
<h1><span class="header-section-number">Chapter 3</span> Taxonomic information</h1>
<script>
document.addEventListener("click", function (event) {
    if (event.target.classList.contains("rebook-collapse")) {
        event.target.classList.toggle("active");
        var content = event.target.nextElementSibling;
        if (content.style.display === "block") {
            content.style.display = "none";
        } else {
            content.style.display = "block";
        }
    }
})
</script>
<style>
.rebook-collapse {
  background-color: #eee;
  color: #444;
  cursor: pointer;
  padding: 18px;
  width: 100%;
  border: none;
  text-align: left;
  outline: none;
  font-size: 15px;
}

.rebook-content {
  padding: 0 18px;
  display: none;
  overflow: hidden;
  background-color: #f1f1f1;
}
</style>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="taxonomic-information.html#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mia)</span>
<span id="cb45-2"><a href="taxonomic-information.html#cb45-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;GlobalPatterns&quot;</span>)</span>
<span id="cb45-3"><a href="taxonomic-information.html#cb45-3" aria-hidden="true" tabindex="-1"></a>se <span class="ot">&lt;-</span> GlobalPatterns </span></code></pre></div>
<p>Taxonomic information are a key part of analyzing microbiome data and without
it, any type of data analysis probably will not make much sense. However,
the degree of detail of taxonomic information differs depending on the dataset
and annotation data used.</p>
<p>Therefore, the mia package expects a loose assembly of taxonomic information
and assumes certain key aspects:</p>
<ul>
<li>Taxonomic information is given as character vectors or factors in the
<code>rowData</code> of an <code>SummarizedExperiment</code> object.</li>
<li>The columns containing the taxonomic information must be named <code>domain</code>,
<code>kingdom</code>, <code>phylum</code>, <code>class</code>, <code>order</code>, <code>family</code>, <code>genus</code>, <code>species</code> or with
a capital first letter.</li>
<li>the columns must be given in the order shown above</li>
<li>column can be omited, but the order must remain</li>
</ul>
<div id="assigning-taxonomic-information." class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> Assigning taxonomic information.</h2>
<p>There are a number of methods to assign taxonomic information. We like to give
a short introduction about the methods available without ranking one over the
other. This has to be your choice based on the result for the individual
dataset.</p>
<div id="dada2" class="section level3" number="3.1.1">
<h3><span class="header-section-number">3.1.1</span> dada2</h3>
<p>The dada2 package <span class="citation">(<a href="#ref-R-dada2" role="doc-biblioref">Callahan, McMurdie, and Holmes 2020</a>)</span> implements the <code>assignTaxonomy</code> function, which
takes as input the ASV sequences associated with each row of data and a training
dataset. For more information visit the
<a href="https://benjjneb.github.io/dada2/assign.html">dada2 website</a>.</p>
</div>
<div id="decipher" class="section level3" number="3.1.2">
<h3><span class="header-section-number">3.1.2</span> DECIPHER</h3>
<p>The DECIPHER package <span class="citation">(<a href="#ref-R-DECIPHER" role="doc-biblioref">Wright 2020</a>)</span> implements the <code>IDTAXA</code> algorithm to assign
either taxonomic information or function information. For <code>mia</code>
only the first option is of interest for now and more information can be
found on the <a href="http://www2.decipher.codes/Classification.html">DECIPHER website</a></p>
</div>
</div>
<div id="functions-to-access-taxonomic-information" class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> Functions to access taxonomic information</h2>
<p><code>checkTaxonomy</code> checks whether the taxonomic information is usable for <code>mia</code></p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="taxonomic-information.html#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="fu">checkTaxonomy</span>(se)</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<p>Since the <code>rowData</code> can contain other data, <code>taxonomyRanks</code> will return the
columns <code>mia</code> assumes to contain the taxonomic information.</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="taxonomic-information.html#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="fu">taxonomyRanks</span>(se)</span></code></pre></div>
<pre><code>## [1] &quot;Kingdom&quot; &quot;Phylum&quot;  &quot;Class&quot;   &quot;Order&quot;   &quot;Family&quot;  &quot;Genus&quot;   &quot;Species&quot;</code></pre>
<p>This can then be used to subset the <code>rowData</code> to columns needed.</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="taxonomic-information.html#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(se)[,<span class="fu">taxonomyRanks</span>(se)]</span></code></pre></div>
<pre><code>## DataFrame with 19216 rows and 7 columns
##            Kingdom        Phylum        Class        Order        Family
##        &lt;character&gt;   &lt;character&gt;  &lt;character&gt;  &lt;character&gt;   &lt;character&gt;
## 549322     Archaea Crenarchaeota Thermoprotei           NA            NA
## 522457     Archaea Crenarchaeota Thermoprotei           NA            NA
## 951        Archaea Crenarchaeota Thermoprotei Sulfolobales Sulfolobaceae
## 244423     Archaea Crenarchaeota        Sd-NA           NA            NA
## 586076     Archaea Crenarchaeota        Sd-NA           NA            NA
## ...            ...           ...          ...          ...           ...
## 278222    Bacteria           SR1           NA           NA            NA
## 463590    Bacteria           SR1           NA           NA            NA
## 535321    Bacteria           SR1           NA           NA            NA
## 200359    Bacteria           SR1           NA           NA            NA
## 271582    Bacteria           SR1           NA           NA            NA
##              Genus                Species
##        &lt;character&gt;            &lt;character&gt;
## 549322          NA                     NA
## 522457          NA                     NA
## 951     Sulfolobus Sulfolobusacidocalda..
## 244423          NA                     NA
## 586076          NA                     NA
## ...            ...                    ...
## 278222          NA                     NA
## 463590          NA                     NA
## 535321          NA                     NA
## 200359          NA                     NA
## 271582          NA                     NA</code></pre>
<p><code>taxonomyRankEmpty</code> checks for empty values in the given <code>rank</code> and returns a
logical vector of <code>length(x)</code>.</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="taxonomic-information.html#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(<span class="sc">!</span><span class="fu">taxonomyRankEmpty</span>(se, <span class="at">rank =</span> <span class="st">&quot;Kingdom&quot;</span>))</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="taxonomic-information.html#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">taxonomyRankEmpty</span>(se, <span class="at">rank =</span> <span class="st">&quot;Genus&quot;</span>))</span></code></pre></div>
<pre><code>## 
## FALSE  TRUE 
##  8008 11208</code></pre>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="taxonomic-information.html#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">taxonomyRankEmpty</span>(se, <span class="at">rank =</span> <span class="st">&quot;Species&quot;</span>))</span></code></pre></div>
<pre><code>## 
## FALSE  TRUE 
##  1413 17803</code></pre>
<p><code>getTaxonomyLabels</code> is a multi-purpose function, which turns taxonomic
information into a character vector of <code>length(x)</code></p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="taxonomic-information.html#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">getTaxonomyLabels</span>(se))</span></code></pre></div>
<pre><code>## [1] &quot;Class:Thermoprotei&quot;               &quot;Class:Thermoprotei_1&quot;            
## [3] &quot;Species:Sulfolobusacidocaldarius&quot; &quot;Class:Sd-NA&quot;                     
## [5] &quot;Class:Sd-NA_1&quot;                    &quot;Class:Sd-NA_2&quot;</code></pre>
<p>By default this will used the lowest non-empty information to construct a
string with the following scheme <code>level:value</code>. If all levels are the same
this part is omited, but can be added by setting <code>with_rank = TRUE</code></p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="taxonomic-information.html#cb60-1" aria-hidden="true" tabindex="-1"></a>phylum <span class="ot">&lt;-</span> <span class="sc">!</span><span class="fu">is.na</span>(<span class="fu">rowData</span>(se)<span class="sc">$</span>Phylum) <span class="sc">&amp;</span> </span>
<span id="cb60-2"><a href="taxonomic-information.html#cb60-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">vapply</span>(<span class="fu">data.frame</span>(<span class="fu">apply</span>(<span class="fu">rowData</span>(se)[,<span class="fu">taxonomyRanks</span>(se)[<span class="dv">3</span><span class="sc">:</span><span class="dv">7</span>]],1L,is.na)),all,<span class="fu">logical</span>(<span class="dv">1</span>))</span>
<span id="cb60-3"><a href="taxonomic-information.html#cb60-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">getTaxonomyLabels</span>(se[phylum,]))</span></code></pre></div>
<pre><code>## [1] &quot;Crenarchaeota&quot;    &quot;Crenarchaeota_1&quot;  &quot;Crenarchaeota_2&quot;  &quot;Actinobacteria&quot;  
## [5] &quot;Actinobacteria_1&quot; &quot;Spirochaetes&quot;</code></pre>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="taxonomic-information.html#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">getTaxonomyLabels</span>(se[phylum,], <span class="at">with_rank =</span> <span class="cn">TRUE</span>))</span></code></pre></div>
<pre><code>## [1] &quot;Phylum:Crenarchaeota&quot;    &quot;Phylum:Crenarchaeota_1&quot; 
## [3] &quot;Phylum:Crenarchaeota_2&quot;  &quot;Phylum:Actinobacteria&quot;  
## [5] &quot;Phylum:Actinobacteria_1&quot; &quot;Phylum:Spirochaetes&quot;</code></pre>
<p>By default the return value of <code>getTaxonomyLabels</code> contains only unique elements
by passing it through <code>make.unique</code>. This step can be omited by setting
<code>make_unique = FALSE</code></p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="taxonomic-information.html#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">getTaxonomyLabels</span>(se[phylum,], <span class="at">with_rank =</span> <span class="cn">TRUE</span>, <span class="at">make_unique =</span> <span class="cn">FALSE</span>))</span></code></pre></div>
<pre><code>## [1] &quot;Phylum:Crenarchaeota&quot;  &quot;Phylum:Crenarchaeota&quot;  &quot;Phylum:Crenarchaeota&quot; 
## [4] &quot;Phylum:Actinobacteria&quot; &quot;Phylum:Actinobacteria&quot; &quot;Phylum:Spirochaetes&quot;</code></pre>
<p>To apply the loop resolving function <code>resolveLoop</code> from the
<code>TreeSummarizedExperiment</code> package <span class="citation">(<a href="#ref-R-TreeSummarizedExperiment" role="doc-biblioref">Huang 2020</a>)</span> within
<code>getTaxonomyLabels</code>, set <code>resolve_loops = TRUE</code>.</p>
<div id="generate-a-taxonomic-tree-on-the-fly" class="section level3" number="3.2.1">
<h3><span class="header-section-number">3.2.1</span> Generate a taxonomic tree on the fly</h3>
<p>To create a taxonomic tree <code>taxonomyTree</code> used the information and returns a
<code>phylo</code> object. Duplicate information from the <code>rowData</code> are removed.</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="taxonomic-information.html#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="fu">taxonomyTree</span>(se)</span></code></pre></div>
<pre><code>## 
## Phylogenetic tree with 1645 tips and 1089 internal nodes.
## 
## Tip labels:
##   Species:Cenarchaeumsymbiosum, Species:pIVWA5, Species:CandidatusNitrososphaeragargensis, Species:SCA1145, Species:SCA1170, Species:Sulfolobusacidocaldarius, ...
## Node labels:
##   root:ALL, Kingdom:Archaea, Phylum:Crenarchaeota, Class:C2, Class:Sd-NA, Class:Thaumarchaeota, ...
## 
## Rooted; includes branch lengths.</code></pre>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="taxonomic-information.html#cb68-1" aria-hidden="true" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">addTaxonomyTree</span>(se)</span>
<span id="cb68-2"><a href="taxonomic-information.html#cb68-2" aria-hidden="true" tabindex="-1"></a>se</span></code></pre></div>
<pre><code>## class: TreeSummarizedExperiment 
## dim: 19216 26 
## metadata(0):
## assays(1): counts
## rownames(19216): Class:Thermoprotei Class:Thermoprotei ... Phylum:SR1
##   Phylum:SR1
## rowData names(7): Kingdom Phylum ... Genus Species
## colnames(26): CL3 CC1 ... Even2 Even3
## colData names(7): X.SampleID Primer ... SampleType Description
## reducedDimNames(0):
## mainExpName: NULL
## altExpNames(0):
## rowLinks: a LinkDataFrame (19216 rows)
## rowTree: 1 phylo tree(s) (1645 leaves)
## colLinks: NULL
## colTree: NULL</code></pre>
<p>The implementation is based on the the <code>toTree</code> function from the
<code>TreeSummarizedExperiment</code> package <span class="citation">(<a href="#ref-R-TreeSummarizedExperiment" role="doc-biblioref">Huang 2020</a>)</span>.</p>
</div>
</div>
<div id="data-agglomeration" class="section level2" number="3.3">
<h2><span class="header-section-number">3.3</span> Data agglomeration</h2>
<p>One of the main applications of taxonomic information in regards to count data
is to agglomerate count data on taxonomic levels and track the influence of
changing conditions through these levels. For this <code>mia</code> contains the
<code>agglomerateByRank</code> function. The ideal location to store the agglomerated data
is as an alternative experiment.</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="taxonomic-information.html#cb70-1" aria-hidden="true" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">relAbundanceCounts</span>(se)</span>
<span id="cb70-2"><a href="taxonomic-information.html#cb70-2" aria-hidden="true" tabindex="-1"></a><span class="fu">altExp</span>(se, <span class="st">&quot;Family&quot;</span>) <span class="ot">&lt;-</span> <span class="fu">agglomerateByRank</span>(se, <span class="at">rank =</span> <span class="st">&quot;Family&quot;</span>,</span>
<span id="cb70-3"><a href="taxonomic-information.html#cb70-3" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">agglomerateTree =</span> <span class="cn">TRUE</span>)</span>
<span id="cb70-4"><a href="taxonomic-information.html#cb70-4" aria-hidden="true" tabindex="-1"></a><span class="fu">altExp</span>(se, <span class="st">&quot;Family&quot;</span>)</span></code></pre></div>
<pre><code>## class: TreeSummarizedExperiment 
## dim: 603 26 
## metadata(0):
## assays(2): counts relabundance
## rownames(603): Class:Thermoprotei Family:Sulfolobaceae ...
##   Family:Thermodesulfobiaceae Phylum:SR1
## rowData names(7): Kingdom Phylum ... Genus Species
## colnames(26): CL3 CC1 ... Even2 Even3
## colData names(7): X.SampleID Primer ... SampleType Description
## reducedDimNames(0):
## mainExpName: NULL
## altExpNames(0):
## rowLinks: a LinkDataFrame (603 rows)
## rowTree: 1 phylo tree(s) (496 leaves)
## colLinks: NULL
## colTree: NULL</code></pre>
<p>If multiple assays (counts and relabundance) exists, both will be agglomerated.</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="taxonomic-information.html#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="fu">assayNames</span>(se)</span></code></pre></div>
<pre><code>## [1] &quot;counts&quot;       &quot;relabundance&quot;</code></pre>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="taxonomic-information.html#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="fu">assayNames</span>(<span class="fu">altExp</span>(se, <span class="st">&quot;Family&quot;</span>))</span></code></pre></div>
<pre><code>## [1] &quot;counts&quot;       &quot;relabundance&quot;</code></pre>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="taxonomic-information.html#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="fu">assay</span>(<span class="fu">altExp</span>(se, <span class="st">&quot;Family&quot;</span>), <span class="st">&quot;relabundance&quot;</span>)[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">7</span>]</span></code></pre></div>
<pre><code>##                            CL3       CC1 SV1 M31Fcsw M11Fcsw M31Plmr   M11Plmr
## Class:Thermoprotei   0.0000000 0.000e+00   0       0       0       0 0.000e+00
## Family:Sulfolobaceae 0.0000000 0.000e+00   0       0       0       0 2.305e-06
## Class:Sd-NA          0.0000000 0.000e+00   0       0       0       0 0.000e+00
## Order:NRP-J          0.0001991 2.070e-04   0       0       0       0 6.914e-06
## Family:SAGMA-X       0.0000000 6.165e-06   0       0       0       0 0.000e+00</code></pre>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="taxonomic-information.html#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="fu">assay</span>(<span class="fu">altExp</span>(se, <span class="st">&quot;Family&quot;</span>), <span class="st">&quot;counts&quot;</span>)[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">7</span>]</span></code></pre></div>
<pre><code>##                      CL3 CC1 SV1 M31Fcsw M11Fcsw M31Plmr M11Plmr
## Class:Thermoprotei     0   0   0       0       0       0       0
## Family:Sulfolobaceae   0   0   0       0       0       0       1
## Class:Sd-NA            0   0   0       0       0       0       0
## Order:NRP-J          172 235   0       0       0       0       3
## Family:SAGMA-X         0   7   0       0       0       0       0</code></pre>
<p><code>altExpNames</code> now consists of <code>Family</code> level data. This can be extended to use
any level present in Kingdom, Phylum, Class, Order, Family, Genus, Species.</p>
</div>
<div id="pick-specific" class="section level2" number="3.4">
<h2><span class="header-section-number">3.4</span> Pick specific</h2>
<p>Retrieving of specific elements are required for specific analysis. For
instance, extracting abundances for a specific taxa in all samples or all taxa
in one sample.</p>
<div id="abundances-of-all-taxa-in-specific-sample" class="section level3" number="3.4.1">
<h3><span class="header-section-number">3.4.1</span> Abundances of all taxa in specific sample</h3>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="taxonomic-information.html#cb80-1" aria-hidden="true" tabindex="-1"></a>taxa.abund.cc1 <span class="ot">&lt;-</span> <span class="fu">getAbundanceSample</span>(se, </span>
<span id="cb80-2"><a href="taxonomic-information.html#cb80-2" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">sample_id =</span> <span class="st">&quot;CC1&quot;</span>,</span>
<span id="cb80-3"><a href="taxonomic-information.html#cb80-3" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">abund_values =</span> <span class="st">&quot;counts&quot;</span>)</span>
<span id="cb80-4"><a href="taxonomic-information.html#cb80-4" aria-hidden="true" tabindex="-1"></a>taxa.abund.cc1[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>]</span></code></pre></div>
<pre><code>##               Class:Thermoprotei               Class:Thermoprotei 
##                                0                                0 
## Species:Sulfolobusacidocaldarius                      Class:Sd-NA 
##                                0                                0 
##                      Class:Sd-NA                      Class:Sd-NA 
##                                0                                0 
##                      Order:NRP-J                      Order:NRP-J 
##                                1                                0 
##                      Order:NRP-J                      Order:NRP-J 
##                              194                                5</code></pre>
</div>
<div id="abundances-of-specific-taxa-in-all-samples" class="section level3" number="3.4.2">
<h3><span class="header-section-number">3.4.2</span> Abundances of specific taxa in all samples</h3>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="taxonomic-information.html#cb82-1" aria-hidden="true" tabindex="-1"></a>taxa.abundances <span class="ot">&lt;-</span> <span class="fu">getAbundanceFeature</span>(se, </span>
<span id="cb82-2"><a href="taxonomic-information.html#cb82-2" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">feature_id =</span> <span class="st">&quot;Phylum:Bacteroidetes&quot;</span>,</span>
<span id="cb82-3"><a href="taxonomic-information.html#cb82-3" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">abund_values =</span> <span class="st">&quot;counts&quot;</span>)</span>
<span id="cb82-4"><a href="taxonomic-information.html#cb82-4" aria-hidden="true" tabindex="-1"></a>taxa.abundances[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>]</span></code></pre></div>
<pre><code>##     CL3     CC1     SV1 M31Fcsw M11Fcsw M31Plmr M11Plmr F21Plmr M31Tong M11Tong 
##       2      18       2       0       0       0       0       1       0       0</code></pre>
</div>
</div>
<div id="session-info-2" class="section level2 unnumbered">
<h2>Session Info</h2>
<button class="rebook-collapse">
View session info
</button>
<div class="rebook-content">
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.2 LTS

Matrix products: default
BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] mia_1.1.7                      TreeSummarizedExperiment_2.1.3
 [3] Biostrings_2.61.1              XVector_0.33.0                
 [5] SingleCellExperiment_1.15.1    SummarizedExperiment_1.23.1   
 [7] Biobase_2.53.0                 GenomicRanges_1.45.0          
 [9] GenomeInfoDb_1.29.3            IRanges_2.27.0                
[11] S4Vectors_0.31.0               BiocGenerics_0.39.1           
[13] MatrixGenerics_1.5.1           matrixStats_0.59.0            
[15] BiocStyle_2.21.3               rebook_1.3.0                  

loaded via a namespace (and not attached):
 [1] ggbeeswarm_0.6.0            colorspace_2.0-2           
 [3] ellipsis_0.3.2              scuttle_1.3.0              
 [5] BiocNeighbors_1.11.0        bit64_4.0.5                
 [7] fansi_0.5.0                 decontam_1.13.0            
 [9] splines_4.1.0               codetools_0.2-18           
[11] sparseMatrixStats_1.5.0     cachem_1.0.5               
[13] knitr_1.33                  scater_1.21.2              
[15] jsonlite_1.7.2              cluster_2.1.2              
[17] graph_1.71.2                BiocManager_1.30.16        
[19] compiler_4.1.0              assertthat_0.2.1           
[21] Matrix_1.3-4                fastmap_1.1.0              
[23] lazyeval_0.2.2              BiocSingular_1.9.1         
[25] htmltools_0.5.1.1           tools_4.1.0                
[27] rsvd_1.0.5                  gtable_0.3.0               
[29] glue_1.4.2                  GenomeInfoDbData_1.2.6     
[31] reshape2_1.4.4              dplyr_1.0.7                
[33] Rcpp_1.0.7                  jquerylib_0.1.4            
[35] vctrs_0.3.8                 ape_5.5                    
[37] nlme_3.1-152                DECIPHER_2.21.0            
[39] DelayedMatrixStats_1.15.0   xfun_0.24                  
[41] stringr_1.4.0               beachmat_2.9.0             
[43] lifecycle_1.0.0             irlba_2.3.3                
[45] XML_3.99-0.6                zlibbioc_1.39.0            
[47] MASS_7.3-54                 scales_1.1.1               
[49] parallel_4.1.0              yaml_2.2.1                 
[51] memoise_2.0.0               gridExtra_2.3              
[53] ggplot2_3.3.5               sass_0.4.0                 
[55] stringi_1.7.3               RSQLite_2.2.7              
[57] ScaledMatrix_1.1.0          tidytree_0.3.4             
[59] permute_0.9-5               filelock_1.0.2             
[61] BiocParallel_1.27.2         rlang_0.4.11               
[63] pkgconfig_2.0.3             bitops_1.0-7               
[65] evaluate_0.14               lattice_0.20-44            
[67] purrr_0.3.4                 treeio_1.17.2              
[69] CodeDepends_0.6.5           bit_4.0.4                  
[71] tidyselect_1.1.1            plyr_1.8.6                 
[73] magrittr_2.0.1              bookdown_0.22              
[75] R6_2.5.0                    generics_0.1.0             
[77] DelayedArray_0.19.1         DBI_1.1.1                  
[79] mgcv_1.8-36                 pillar_1.6.1               
[81] RCurl_1.98-1.3              tibble_3.1.2               
[83] dir.expiry_1.1.0            crayon_1.4.1               
[85] utf8_1.2.1                  rmarkdown_2.9              
[87] viridis_0.6.1               grid_4.1.0                 
[89] blob_1.2.1                  vegan_2.5-7                
[91] digest_0.6.27               tidyr_1.1.3                
[93] munsell_0.5.0               DirichletMultinomial_1.35.0
[95] beeswarm_0.4.0              viridisLite_0.4.0          
[97] vipor_0.4.5                 bslib_0.2.5.1              </code></pre>
</div>

</div>
</div>
<h3>Bibliography</h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-R-dada2" class="csl-entry">
Callahan, Benjamin, Paul McMurdie, and Susan Holmes. 2020. <em>Dada2: Accurate, High-Resolution Sample Inference from Amplicon Sequencing Data</em>. <a href="http://benjjneb.github.io/dada2/">http://benjjneb.github.io/dada2/</a>.
</div>
<div id="ref-R-TreeSummarizedExperiment" class="csl-entry">
Huang, Ruizhu. 2020. <em>TreeSummarizedExperiment: A S4 Class for Data with Tree Structures</em>.
</div>
<div id="ref-R-DECIPHER" class="csl-entry">
Wright, Erik. 2020. <em>DECIPHER: Tools for Curating, Analyzing, and Manipulating Biological Sequences</em>.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="overview.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="quality-control.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["OMA.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
