<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Taxonomic information | Microbiome Analysis</title>
  <meta name="description" content="Chapter 3 Taxonomic information | Microbiome Analysis" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Taxonomic information | Microbiome Analysis" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://github.com/microbiome/MiaBook" />
  
  
  <meta name="github-repo" content="microbiome/MiaBook" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Taxonomic information | Microbiome Analysis" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="overview.html"/>
<link rel="next" href="quality-control.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0/anchor-sections.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Microbiome Analysis</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="part"><span><b>I Introduction</b></span></li>
<li class="chapter" data-level="1" data-path="data-introduction.html"><a href="data-introduction.html"><i class="fa fa-check"></i><b>1</b> Data Infrastructure</a><ul>
<li class="chapter" data-level="1.1" data-path="data-introduction.html"><a href="data-introduction.html#installation"><i class="fa fa-check"></i><b>1.1</b> Installation</a><ul>
<li class="chapter" data-level="1.1.1" data-path="data-introduction.html"><a href="data-introduction.html#packages"><i class="fa fa-check"></i><b>1.1.1</b> Packages</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="data-introduction.html"><a href="data-introduction.html#background"><i class="fa fa-check"></i><b>1.2</b> Background</a></li>
<li class="chapter" data-level="1.3" data-path="data-introduction.html"><a href="data-introduction.html#loading-experimental-microbiome-data"><i class="fa fa-check"></i><b>1.3</b> Loading experimental microbiome data</a></li>
<li class="chapter" data-level="1.4" data-path="data-introduction.html"><a href="data-introduction.html#metadata"><i class="fa fa-check"></i><b>1.4</b> Metadata</a></li>
<li class="chapter" data-level="1.5" data-path="data-introduction.html"><a href="data-introduction.html#microbiome-and-tree-data-specific-aspects"><i class="fa fa-check"></i><b>1.5</b> Microbiome and tree data specific aspects</a><ul>
<li class="chapter" data-level="1.5.1" data-path="data-introduction.html"><a href="data-introduction.html#assays"><i class="fa fa-check"></i><b>1.5.1</b> Assays</a></li>
<li class="chapter" data-level="1.5.2" data-path="data-introduction.html"><a href="data-introduction.html#coldata"><i class="fa fa-check"></i><b>1.5.2</b> colData</a></li>
<li class="chapter" data-level="1.5.3" data-path="data-introduction.html"><a href="data-introduction.html#rowdata"><i class="fa fa-check"></i><b>1.5.3</b> rowData</a></li>
<li class="chapter" data-level="1.5.4" data-path="data-introduction.html"><a href="data-introduction.html#rowtree"><i class="fa fa-check"></i><b>1.5.4</b> rowTree</a></li>
</ul></li>
<li class="chapter" data-level="1.6" data-path="data-introduction.html"><a href="data-introduction.html#data-conversion"><i class="fa fa-check"></i><b>1.6</b> Data conversion</a><ul>
<li class="chapter" data-level="1.6.1" data-path="data-introduction.html"><a href="data-introduction.html#tidy-data"><i class="fa fa-check"></i><b>1.6.1</b> Tidy data</a></li>
</ul></li>
<li class="chapter" data-level="1.7" data-path="data-introduction.html"><a href="data-introduction.html#conclusion"><i class="fa fa-check"></i><b>1.7</b> Conclusion</a></li>
<li class="chapter" data-level="" data-path="data-introduction.html"><a href="data-introduction.html#session-info"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="part"><span><b>II Focus Topics</b></span></li>
<li class="chapter" data-level="2" data-path="overview.html"><a href="overview.html"><i class="fa fa-check"></i><b>2</b> Overview</a><ul>
<li class="chapter" data-level="2.1" data-path="overview.html"><a href="overview.html#introduction"><i class="fa fa-check"></i><b>2.1</b> Introduction</a></li>
<li class="chapter" data-level="2.2" data-path="overview.html"><a href="overview.html#microbiome-experiment-types"><i class="fa fa-check"></i><b>2.2</b> Microbiome experiment types</a></li>
<li class="chapter" data-level="2.3" data-path="overview.html"><a href="overview.html#processing-steps"><i class="fa fa-check"></i><b>2.3</b> Processing steps</a></li>
<li class="chapter" data-level="2.4" data-path="overview.html"><a href="overview.html#quick-start"><i class="fa fa-check"></i><b>2.4</b> Quick Start</a></li>
<li class="chapter" data-level="" data-path="overview.html"><a href="overview.html#session-info-1"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="taxonomic-information.html"><a href="taxonomic-information.html"><i class="fa fa-check"></i><b>3</b> Taxonomic information</a><ul>
<li class="chapter" data-level="3.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#data-agglomeration"><i class="fa fa-check"></i><b>3.1</b> Data agglomeration</a></li>
<li class="chapter" data-level="3.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#get-unique"><i class="fa fa-check"></i><b>3.2</b> Get unique</a></li>
<li class="chapter" data-level="3.3" data-path="taxonomic-information.html"><a href="taxonomic-information.html#pick-specific"><i class="fa fa-check"></i><b>3.3</b> Pick specific</a><ul>
<li class="chapter" data-level="3.3.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#abundances-of-all-taxa-in-specific-sample"><i class="fa fa-check"></i><b>3.3.1</b> Abundances of all taxa in specific sample</a></li>
<li class="chapter" data-level="3.3.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#abundances-of-specific-taxa-in-all-samples"><i class="fa fa-check"></i><b>3.3.2</b> Abundances of specific taxa in all samples</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="taxonomic-information.html"><a href="taxonomic-information.html#session-info-2"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>4</b> Quality Control</a><ul>
<li class="chapter" data-level="4.1" data-path="quality-control.html"><a href="quality-control.html#get-top-taxa-and-taxonomy"><i class="fa fa-check"></i><b>4.1</b> Get top taxa and taxonomy</a><ul>
<li class="chapter" data-level="4.1.1" data-path="quality-control.html"><a href="quality-control.html#features"><i class="fa fa-check"></i><b>4.1.1</b> Features</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="quality-control.html"><a href="quality-control.html#library-size"><i class="fa fa-check"></i><b>4.2</b> Library size</a></li>
<li class="chapter" data-level="" data-path="quality-control.html"><a href="quality-control.html#session-info-3"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html"><i class="fa fa-check"></i><b>5</b> Microbiome Exploration</a><ul>
<li class="chapter" data-level="5.1" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#prevalence"><i class="fa fa-check"></i><b>5.1</b> Prevalence</a><ul>
<li class="chapter" data-level="5.1.1" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#prevalent-microbiota-analysis"><i class="fa fa-check"></i><b>5.1.1</b> Prevalent microbiota analysis</a></li>
<li class="chapter" data-level="5.1.2" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#plotting-prevalence"><i class="fa fa-check"></i><b>5.1.2</b> Plotting prevalence</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="microbiome-exploration.html"><a href="microbiome-exploration.html#session-info-4"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="microbiome-diversity.html"><a href="microbiome-diversity.html"><i class="fa fa-check"></i><b>6</b> Microbiome Diversity</a><ul>
<li class="chapter" data-level="6.1" data-path="microbiome-diversity.html"><a href="microbiome-diversity.html#alpha-diversity"><i class="fa fa-check"></i><b>6.1</b> Alpha diversity</a></li>
<li class="chapter" data-level="6.2" data-path="microbiome-diversity.html"><a href="microbiome-diversity.html#beta-diversity"><i class="fa fa-check"></i><b>6.2</b> Beta diversity</a><ul>
<li class="chapter" data-level="6.2.1" data-path="microbiome-diversity.html"><a href="microbiome-diversity.html#community-comparisons"><i class="fa fa-check"></i><b>6.2.1</b> Community comparisons</a></li>
<li class="chapter" data-level="6.2.2" data-path="microbiome-diversity.html"><a href="microbiome-diversity.html#visualizing-beta-diversity"><i class="fa fa-check"></i><b>6.2.2</b> Visualizing beta diversity</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="microbiome-diversity.html"><a href="microbiome-diversity.html#session-info-5"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="microbiome-community.html"><a href="microbiome-community.html"><i class="fa fa-check"></i><b>7</b> Microbiome Community</a><ul>
<li class="chapter" data-level="7.1" data-path="microbiome-community.html"><a href="microbiome-community.html#community-composition"><i class="fa fa-check"></i><b>7.1</b> Community composition</a></li>
<li class="chapter" data-level="7.2" data-path="microbiome-community.html"><a href="microbiome-community.html#community-typing"><i class="fa fa-check"></i><b>7.2</b> Community typing</a></li>
<li class="chapter" data-level="" data-path="microbiome-community.html"><a href="microbiome-community.html#session-info-6"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="differential-abundance.html"><a href="differential-abundance.html"><i class="fa fa-check"></i><b>8</b> Differential abundance</a><ul>
<li class="chapter" data-level="" data-path="differential-abundance.html"><a href="differential-abundance.html#session-info-7"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="microbiome-timeseries.html"><a href="microbiome-timeseries.html"><i class="fa fa-check"></i><b>9</b> Microbiome time series</a><ul>
<li class="chapter" data-level="9.1" data-path="microbiome-timeseries.html"><a href="microbiome-timeseries.html#stability"><i class="fa fa-check"></i><b>9.1</b> Stability</a></li>
<li class="chapter" data-level="9.2" data-path="microbiome-timeseries.html"><a href="microbiome-timeseries.html#tipping-elements"><i class="fa fa-check"></i><b>9.2</b> Tipping elements</a></li>
<li class="chapter" data-level="" data-path="microbiome-timeseries.html"><a href="microbiome-timeseries.html#session-info-8"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="multitable.html"><a href="multitable.html"><i class="fa fa-check"></i><b>10</b> Multi-table methods</a><ul>
<li class="chapter" data-level="" data-path="multitable.html"><a href="multitable.html#session-info-9"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="part"><span><b>III Appendix</b></span></li>
<li class="chapter" data-level="11" data-path="example-data.html"><a href="example-data.html"><i class="fa fa-check"></i><b>11</b> Example data</a><ul>
<li class="chapter" data-level="11.1" data-path="example-data.html"><a href="example-data.html#package-data"><i class="fa fa-check"></i><b>11.1</b> Package data</a><ul>
<li class="chapter" data-level="11.1.1" data-path="example-data.html"><a href="example-data.html#globalpatterns"><i class="fa fa-check"></i><b>11.1.1</b> GlobalPatterns</a></li>
<li class="chapter" data-level="11.1.2" data-path="example-data.html"><a href="example-data.html#enterotype"><i class="fa fa-check"></i><b>11.1.2</b> Enterotype</a></li>
<li class="chapter" data-level="11.1.3" data-path="example-data.html"><a href="example-data.html#esophagus"><i class="fa fa-check"></i><b>11.1.3</b> Esophagus</a></li>
<li class="chapter" data-level="11.1.4" data-path="example-data.html"><a href="example-data.html#soilrep"><i class="fa fa-check"></i><b>11.1.4</b> Soilrep</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="example-data.html"><a href="example-data.html#session-info-10"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="bibliography.html"><a href="bibliography.html"><i class="fa fa-check"></i>Bibliography</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Microbiome Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="taxonomic-information" class="section level1">
<h1><span class="header-section-number">Chapter 3</span> Taxonomic information</h1>
<script>
document.addEventListener("click", function (event) {
    if (event.target.classList.contains("rebook-collapse")) {
        event.target.classList.toggle("active");
        var content = event.target.nextElementSibling;
        if (content.style.display === "block") {
            content.style.display = "none";
        } else {
            content.style.display = "block";
        }
    }
})
</script>
<style>
.rebook-collapse {
  background-color: #eee;
  color: #444;
  cursor: pointer;
  padding: 18px;
  width: 100%;
  border: none;
  text-align: left;
  outline: none;
  font-size: 15px;
}

.rebook-content {
  padding: 0 18px;
  display: none;
  overflow: hidden;
  background-color: #f1f1f1;
}
</style>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="taxonomic-information.html#cb24-1"></a><span class="kw">library</span>(mia)</span>
<span id="cb24-2"><a href="taxonomic-information.html#cb24-2"></a><span class="kw">data</span>(<span class="st">&quot;GlobalPatterns&quot;</span>)</span>
<span id="cb24-3"><a href="taxonomic-information.html#cb24-3"></a>se &lt;-<span class="st"> </span>GlobalPatterns </span></code></pre></div>
<div id="data-agglomeration" class="section level2">
<h2><span class="header-section-number">3.1</span> Data agglomeration</h2>
<p>Agglomerate at a specific taxonomic rank. If multiple assays (counts and relabundance) exists, both will be agglomerated.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="taxonomic-information.html#cb25-1"></a><span class="kw">altExp</span>(se, <span class="st">&quot;Family&quot;</span>) &lt;-<span class="st"> </span><span class="kw">agglomerateByRank</span>(se, <span class="dt">rank =</span> <span class="st">&quot;Family&quot;</span>,</span>
<span id="cb25-2"><a href="taxonomic-information.html#cb25-2"></a>                                          <span class="dt">agglomerateTree =</span> <span class="ot">TRUE</span>)</span>
<span id="cb25-3"><a href="taxonomic-information.html#cb25-3"></a></span>
<span id="cb25-4"><a href="taxonomic-information.html#cb25-4"></a>se</span></code></pre></div>
<pre><code>## class: TreeSummarizedExperiment 
## dim: 19216 26 
## metadata(0):
## assays(1): counts
## rownames(19216): 549322 522457 ... 200359 271582
## rowData names(7): Kingdom Phylum ... Genus Species
## colnames(26): CL3 CC1 ... Even2 Even3
## colData names(7): X.SampleID Primer ... SampleType Description
## reducedDimNames(0):
## altExpNames(1): Family
## rowLinks: a LinkDataFrame (19216 rows)
## rowTree: a phylo (19216 leaves)
## colLinks: NULL
## colTree: NULL</code></pre>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="taxonomic-information.html#cb27-1"></a><span class="kw">altExp</span>(se, <span class="st">&quot;Family&quot;</span>)</span></code></pre></div>
<pre><code>## class: TreeSummarizedExperiment 
## dim: 603 26 
## metadata(0):
## assays(1): counts
## rownames(603): Class:Thermoprotei Family:Sulfolobaceae ...
##   Family:Thermodesulfobiaceae Phylum:SR1
## rowData names(7): Kingdom Phylum ... Genus Species
## colnames(26): CL3 CC1 ... Even2 Even3
## colData names(7): X.SampleID Primer ... SampleType Description
## reducedDimNames(0):
## altExpNames(0):
## rowLinks: a LinkDataFrame (603 rows)
## rowTree: a phylo (603 leaves)
## colLinks: NULL
## colTree: NULL</code></pre>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="taxonomic-information.html#cb29-1"></a><span class="kw">altExp</span>(se, <span class="st">&quot;Family&quot;</span>) &lt;-<span class="st"> </span><span class="kw">relAbundanceCounts</span>(<span class="kw">altExp</span>(se, <span class="st">&quot;Family&quot;</span>))</span>
<span id="cb29-2"><a href="taxonomic-information.html#cb29-2"></a><span class="kw">assay</span>(<span class="kw">altExp</span>(se, <span class="st">&quot;Family&quot;</span>), <span class="st">&quot;relabundance&quot;</span>)[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">7</span>]</span></code></pre></div>
<pre><code>##                            CL3       CC1 SV1 M31Fcsw M11Fcsw M31Plmr   M11Plmr
## Class:Thermoprotei   0.0000000 0.000e+00   0       0       0       0 0.000e+00
## Family:Sulfolobaceae 0.0000000 0.000e+00   0       0       0       0 2.305e-06
## Class:Sd-NA          0.0000000 0.000e+00   0       0       0       0 0.000e+00
## Order:NRP-J          0.0001991 2.070e-04   0       0       0       0 6.914e-06
## Family:SAGMA-X       0.0000000 6.165e-06   0       0       0       0 0.000e+00</code></pre>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="taxonomic-information.html#cb31-1"></a><span class="kw">assay</span>(<span class="kw">altExp</span>(se, <span class="st">&quot;Family&quot;</span>), <span class="st">&quot;counts&quot;</span>)[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">7</span>]</span></code></pre></div>
<pre><code>##                      CL3 CC1 SV1 M31Fcsw M11Fcsw M31Plmr M11Plmr
## Class:Thermoprotei     0   0   0       0       0       0       0
## Family:Sulfolobaceae   0   0   0       0       0       0       1
## Class:Sd-NA            0   0   0       0       0       0       0
## Order:NRP-J          172 235   0       0       0       0       3
## Family:SAGMA-X         0   7   0       0       0       0       0</code></pre>
<p>These newly created data can also be kept in <code>altExp</code> a feature provided by
<code>SingleCellExperiment</code>.</p>
<p><code>altExpNames</code> now consists of <code>family</code> level data. This can be extended to use
any level present in Kingdom, Phylum, Class, Order, Family, Genus, Species.</p>
</div>
<div id="get-unique" class="section level2">
<h2><span class="header-section-number">3.2</span> Get unique</h2>
<p>Get which Phyla are present.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="taxonomic-information.html#cb33-1"></a><span class="kw">head</span>(<span class="kw">unique</span>(<span class="kw">rowData</span>(se)[,<span class="st">&quot;Phylum&quot;</span>]))</span></code></pre></div>
<pre><code>## [1] &quot;Crenarchaeota&quot;  &quot;Euryarchaeota&quot;  &quot;Actinobacteria&quot; &quot;Spirochaetes&quot;  
## [5] &quot;MVP-15&quot;         &quot;Proteobacteria&quot;</code></pre>
</div>
<div id="pick-specific" class="section level2">
<h2><span class="header-section-number">3.3</span> Pick specific</h2>
<p>Retrieving of specific elements are required for specific analysis. For
instance, extracting abundances for a specific taxa in all samples or all taxa
in one sample.</p>
<div id="abundances-of-all-taxa-in-specific-sample" class="section level3">
<h3><span class="header-section-number">3.3.1</span> Abundances of all taxa in specific sample</h3>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="taxonomic-information.html#cb35-1"></a>taxa.abund.cc1 &lt;-<span class="st"> </span><span class="kw">getAbundanceSample</span>(se, </span>
<span id="cb35-2"><a href="taxonomic-information.html#cb35-2"></a>                                     <span class="dt">sample_id =</span> <span class="st">&quot;CC1&quot;</span>,</span>
<span id="cb35-3"><a href="taxonomic-information.html#cb35-3"></a>                                     <span class="dt">abund_values =</span> <span class="st">&quot;counts&quot;</span>)</span>
<span id="cb35-4"><a href="taxonomic-information.html#cb35-4"></a></span>
<span id="cb35-5"><a href="taxonomic-information.html#cb35-5"></a>taxa.abund.cc1[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>]</span></code></pre></div>
<pre><code>## 549322 522457    951 244423 586076 246140 143239 244960 255340 144887 
##      0      0      0      0      0      0      1      0    194      5</code></pre>
</div>
<div id="abundances-of-specific-taxa-in-all-samples" class="section level3">
<h3><span class="header-section-number">3.3.2</span> Abundances of specific taxa in all samples</h3>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="taxonomic-information.html#cb37-1"></a>taxa.abundances &lt;-<span class="st"> </span><span class="kw">getAbundanceFeature</span>(se, </span>
<span id="cb37-2"><a href="taxonomic-information.html#cb37-2"></a>                                      <span class="dt">feature_id =</span> <span class="st">&quot;255340&quot;</span>,</span>
<span id="cb37-3"><a href="taxonomic-information.html#cb37-3"></a>                                       <span class="dt">abund_values =</span> <span class="st">&quot;counts&quot;</span>)</span>
<span id="cb37-4"><a href="taxonomic-information.html#cb37-4"></a>taxa.abundances[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>]</span></code></pre></div>
<pre><code>##     CL3     CC1     SV1 M31Fcsw M11Fcsw M31Plmr M11Plmr F21Plmr M31Tong M11Tong 
##     153     194       0       0       0       0       0       0       0       0</code></pre>
</div>
</div>
<div id="session-info-2" class="section level2 unnumbered">
<h2>Session Info</h2>
<button class="rebook-collapse">
View session info
</button>
<div class="rebook-content">
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04 LTS

Matrix products: default
BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] mia_0.98.14                      MicrobiomeExperiment_0.99.0.9014
 [3] Biostrings_2.58.0                XVector_0.30.0                  
 [5] TreeSummarizedExperiment_1.6.2   SingleCellExperiment_1.12.0     
 [7] SummarizedExperiment_1.20.0      Biobase_2.50.0                  
 [9] GenomicRanges_1.42.0             GenomeInfoDb_1.26.1             
[11] IRanges_2.24.0                   S4Vectors_0.28.0                
[13] BiocGenerics_0.36.0              MatrixGenerics_1.2.0            
[15] matrixStats_0.57.0               BiocStyle_2.18.1                
[17] rebook_1.0.0                     BiocManager_1.30.10             

loaded via a namespace (and not attached):
 [1] viridis_0.5.1             tidyr_1.1.2              
 [3] BiocSingular_1.6.0        viridisLite_0.3.0        
 [5] DelayedMatrixStats_1.12.1 scuttle_1.0.3            
 [7] vipor_0.4.5               GenomeInfoDbData_1.2.4   
 [9] yaml_2.2.1                pillar_1.4.7             
[11] lattice_0.20-41           glue_1.4.2               
[13] beachmat_2.6.2            digest_0.6.27            
[15] colorspace_2.0-0          htmltools_0.5.0          
[17] Matrix_1.2-18             XML_3.99-0.5             
[19] pkgconfig_2.0.3           bookdown_0.21            
[21] zlibbioc_1.36.0           purrr_0.3.4              
[23] scales_1.1.1              processx_3.4.5           
[25] BiocParallel_1.24.1       tibble_3.0.4             
[27] generics_0.1.0            ggplot2_3.3.2            
[29] ellipsis_0.3.1            magrittr_2.0.1           
[31] crayon_1.3.4              CodeDepends_0.6.5        
[33] evaluate_0.14             ps_1.5.0                 
[35] nlme_3.1-150              beeswarm_0.2.3           
[37] graph_1.68.0              tools_4.0.3              
[39] scater_1.18.3             lifecycle_0.2.0          
[41] stringr_1.4.0             munsell_0.5.0            
[43] DelayedArray_0.16.0       irlba_2.3.3              
[45] callr_3.5.1               compiler_4.0.3           
[47] rsvd_1.0.3                rlang_0.4.9              
[49] grid_4.0.3                RCurl_1.98-1.2           
[51] BiocNeighbors_1.8.1       bitops_1.0-6             
[53] rmarkdown_2.5             gtable_0.3.0             
[55] codetools_0.2-18          R6_2.5.0                 
[57] gridExtra_2.3             knitr_1.30               
[59] dplyr_1.0.2               ape_5.4-1                
[61] stringi_1.5.3             ggbeeswarm_0.6.0         
[63] Rcpp_1.0.5                vctrs_0.3.5              
[65] tidyselect_1.1.0          xfun_0.19                
[67] sparseMatrixStats_1.2.0  </code></pre>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="overview.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="quality-control.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["MiaBook.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
