<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 14 Visualization | Exercises</title>
  <meta name="description" content="Chapter 14 Visualization | Exercises" />
  <meta name="generator" content="bookdown 0.27 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 14 Visualization | Exercises" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="microbiome/OMA" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 14 Visualization | Exercises" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="multi-assay_analyses.html"/>
<link rel="next" href="resources.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<script src="libs/plotly-binding-4.10.0/plotly.js"></script>
<script src="libs/typedarray-0.1/typedarray.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="libs/plotly-htmlwidgets-css-2.5.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="libs/plotly-main-2.5.1/plotly-latest.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Orchestrating Microbiome Analysis</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="part"><span><b>I Introduction</b></span></li>
<li class="chapter" data-level="1" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="containers.html"><a href="containers.html"><i class="fa fa-check"></i><b>2</b> Microbiome Data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="containers.html"><a href="containers.html#data-science-framework"><i class="fa fa-check"></i><b>2.1</b> Data science framework</a></li>
<li class="chapter" data-level="2.2" data-path="containers.html"><a href="containers.html#data-containers"><i class="fa fa-check"></i><b>2.2</b> Data containers</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="containers.html"><a href="containers.html#assay-data"><i class="fa fa-check"></i><b>2.2.1</b> Assay data</a></li>
<li class="chapter" data-level="2.2.2" data-path="containers.html"><a href="containers.html#coldata"><i class="fa fa-check"></i><b>2.2.2</b> colData</a></li>
<li class="chapter" data-level="2.2.3" data-path="containers.html"><a href="containers.html#rowdata"><i class="fa fa-check"></i><b>2.2.3</b> rowData</a></li>
<li class="chapter" data-level="2.2.4" data-path="containers.html"><a href="containers.html#rowtree"><i class="fa fa-check"></i><b>2.2.4</b> rowTree</a></li>
<li class="chapter" data-level="2.2.5" data-path="containers.html"><a href="containers.html#alternative-experiments"><i class="fa fa-check"></i><b>2.2.5</b> Alternative experiments</a></li>
<li class="chapter" data-level="2.2.6" data-path="containers.html"><a href="containers.html#multiassayexperiments"><i class="fa fa-check"></i><b>2.2.6</b> MultiAssayExperiments</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="containers.html"><a href="containers.html#loading-experimental-microbiome-data"><i class="fa fa-check"></i><b>2.3</b> Loading experimental microbiome data</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="containers.html"><a href="containers.html#s-workflow"><i class="fa fa-check"></i><b>2.3.1</b> 16S workflow</a></li>
<li class="chapter" data-level="2.3.2" data-path="containers.html"><a href="containers.html#import-from-external-files"><i class="fa fa-check"></i><b>2.3.2</b> Import from external files</a></li>
<li class="chapter" data-level="2.3.3" data-path="containers.html"><a href="containers.html#conversions-between-data-formats-in-r"><i class="fa fa-check"></i><b>2.3.3</b> Conversions between data formats in R</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="containers.html"><a href="containers.html#example-data"><i class="fa fa-check"></i><b>2.4</b> Demonstration data</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="containers.html"><a href="containers.html#package-data"><i class="fa fa-check"></i><b>2.4.1</b> Package data</a></li>
<li class="chapter" data-level="2.4.2" data-path="containers.html"><a href="containers.html#experimenthub-data"><i class="fa fa-check"></i><b>2.4.2</b> ExperimentHub data</a></li>
<li class="chapter" data-level="2.4.3" data-path="containers.html"><a href="containers.html#other-data-sources"><i class="fa fa-check"></i><b>2.4.3</b> Other data sources</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="containers.html"><a href="containers.html#session-info"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="packages.html"><a href="packages.html"><i class="fa fa-check"></i><b>3</b> Packages</a>
<ul>
<li class="chapter" data-level="3.1" data-path="packages.html"><a href="packages.html#package-installation"><i class="fa fa-check"></i><b>3.1</b> Package installation</a></li>
<li class="chapter" data-level="3.2" data-path="packages.html"><a href="packages.html#some-available-packages"><i class="fa fa-check"></i><b>3.2</b> Some available packages</a></li>
<li class="chapter" data-level="" data-path="packages.html"><a href="packages.html#session-info-1"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="part"><span><b>II Focus Topics</b></span></li>
<li class="chapter" data-level="4" data-path="datamanipulation.html"><a href="datamanipulation.html"><i class="fa fa-check"></i><b>4</b> Data Manipulation</a>
<ul>
<li class="chapter" data-level="4.1" data-path="datamanipulation.html"><a href="datamanipulation.html#tidying-and-subsetting"><i class="fa fa-check"></i><b>4.1</b> Tidying and subsetting</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="datamanipulation.html"><a href="datamanipulation.html#tidy-data"><i class="fa fa-check"></i><b>4.1.1</b> Tidy data</a></li>
<li class="chapter" data-level="4.1.2" data-path="datamanipulation.html"><a href="datamanipulation.html#subsetting"><i class="fa fa-check"></i><b>4.1.2</b> Subsetting</a></li>
<li class="chapter" data-level="4.1.3" data-path="datamanipulation.html"><a href="datamanipulation.html#additional-functions"><i class="fa fa-check"></i><b>4.1.3</b> Additional functions</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>5</b> Exploration and quality Control</a>
<ul>
<li class="chapter" data-level="5.1" data-path="quality-control.html"><a href="quality-control.html#abundance"><i class="fa fa-check"></i><b>5.1</b> Abundance</a></li>
<li class="chapter" data-level="5.2" data-path="quality-control.html"><a href="quality-control.html#prevalence"><i class="fa fa-check"></i><b>5.2</b> Prevalence</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="quality-control.html"><a href="quality-control.html#prevalence-analysis"><i class="fa fa-check"></i><b>5.2.1</b> Prevalence analysis</a></li>
<li class="chapter" data-level="5.2.2" data-path="quality-control.html"><a href="quality-control.html#rare-taxa"><i class="fa fa-check"></i><b>5.2.2</b> Rare taxa</a></li>
<li class="chapter" data-level="5.2.3" data-path="quality-control.html"><a href="quality-control.html#plotting-prevalence"><i class="fa fa-check"></i><b>5.2.3</b> Plotting prevalence</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="quality-control.html"><a href="quality-control.html#qc"><i class="fa fa-check"></i><b>5.3</b> Quality control</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="quality-control.html"><a href="quality-control.html#top-taxa"><i class="fa fa-check"></i><b>5.3.1</b> Top taxa</a></li>
<li class="chapter" data-level="5.3.2" data-path="quality-control.html"><a href="quality-control.html#library-size-read-count"><i class="fa fa-check"></i><b>5.3.2</b> Library size / read count</a></li>
<li class="chapter" data-level="5.3.3" data-path="quality-control.html"><a href="quality-control.html#contaminant-sequences"><i class="fa fa-check"></i><b>5.3.3</b> Contaminant sequences</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="quality-control.html"><a href="quality-control.html#session-info-2"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="taxonomic-information.html"><a href="taxonomic-information.html"><i class="fa fa-check"></i><b>6</b> Taxonomic Information</a>
<ul>
<li class="chapter" data-level="6.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#assigning-taxonomic-information."><i class="fa fa-check"></i><b>6.1</b> Assigning taxonomic information.</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#dada2"><i class="fa fa-check"></i><b>6.1.1</b> dada2</a></li>
<li class="chapter" data-level="6.1.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#decipher"><i class="fa fa-check"></i><b>6.1.2</b> DECIPHER</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#functions-to-access-taxonomic-information"><i class="fa fa-check"></i><b>6.2</b> Functions to access taxonomic information</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#generate-a-taxonomic-tree-on-the-fly"><i class="fa fa-check"></i><b>6.2.1</b> Generate a taxonomic tree on the fly</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="taxonomic-information.html"><a href="taxonomic-information.html#data-agglomeration"><i class="fa fa-check"></i><b>6.3</b> Data agglomeration</a></li>
<li class="chapter" data-level="6.4" data-path="taxonomic-information.html"><a href="taxonomic-information.html#data-transformation"><i class="fa fa-check"></i><b>6.4</b> Data transformation</a></li>
<li class="chapter" data-level="6.5" data-path="taxonomic-information.html"><a href="taxonomic-information.html#pick-specific"><i class="fa fa-check"></i><b>6.5</b> Pick specific</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="taxonomic-information.html"><a href="taxonomic-information.html#abundances-of-all-taxa-in-specific-sample"><i class="fa fa-check"></i><b>6.5.1</b> Abundances of all taxa in specific sample</a></li>
<li class="chapter" data-level="6.5.2" data-path="taxonomic-information.html"><a href="taxonomic-information.html#abundances-of-specific-taxa-in-all-samples"><i class="fa fa-check"></i><b>6.5.2</b> Abundances of specific taxa in all samples</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="taxonomic-information.html"><a href="taxonomic-information.html#session-info-3"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="community-diversity.html"><a href="community-diversity.html"><i class="fa fa-check"></i><b>7</b> Community diversity</a>
<ul>
<li class="chapter" data-level="7.1" data-path="community-diversity.html"><a href="community-diversity.html#estimation"><i class="fa fa-check"></i><b>7.1</b> Estimation</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="community-diversity.html"><a href="community-diversity.html#richness"><i class="fa fa-check"></i><b>7.1.1</b> Richness</a></li>
<li class="chapter" data-level="7.1.2" data-path="community-diversity.html"><a href="community-diversity.html#estimate-diversity"><i class="fa fa-check"></i><b>7.1.2</b> Diversity</a></li>
<li class="chapter" data-level="7.1.3" data-path="community-diversity.html"><a href="community-diversity.html#faith-diversity"><i class="fa fa-check"></i><b>7.1.3</b> Faith phylogenetic diversity</a></li>
<li class="chapter" data-level="7.1.4" data-path="community-diversity.html"><a href="community-diversity.html#evenness"><i class="fa fa-check"></i><b>7.1.4</b> Evenness</a></li>
<li class="chapter" data-level="7.1.5" data-path="community-diversity.html"><a href="community-diversity.html#dominance"><i class="fa fa-check"></i><b>7.1.5</b> Dominance</a></li>
<li class="chapter" data-level="7.1.6" data-path="community-diversity.html"><a href="community-diversity.html#rarity"><i class="fa fa-check"></i><b>7.1.6</b> Rarity</a></li>
<li class="chapter" data-level="7.1.7" data-path="community-diversity.html"><a href="community-diversity.html#divergence"><i class="fa fa-check"></i><b>7.1.7</b> Divergence</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="community-diversity.html"><a href="community-diversity.html#visualization"><i class="fa fa-check"></i><b>7.2</b> Visualization</a></li>
<li class="chapter" data-level="" data-path="community-diversity.html"><a href="community-diversity.html#session-info-4"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="community-similarity.html"><a href="community-similarity.html"><i class="fa fa-check"></i><b>8</b> Community similarity</a>
<ul>
<li class="chapter" data-level="8.1" data-path="community-similarity.html"><a href="community-similarity.html#explained-variance"><i class="fa fa-check"></i><b>8.1</b> Explained variance</a></li>
<li class="chapter" data-level="8.2" data-path="community-similarity.html"><a href="community-similarity.html#community-comparisons-by-beta-diversity-analysis"><i class="fa fa-check"></i><b>8.2</b> Community comparisons by beta diversity analysis</a></li>
<li class="chapter" data-level="8.3" data-path="community-similarity.html"><a href="community-similarity.html#other-ordination-methods"><i class="fa fa-check"></i><b>8.3</b> Other ordination methods</a></li>
<li class="chapter" data-level="8.4" data-path="community-similarity.html"><a href="community-similarity.html#visualizing-the-most-dominant-genus-on-pcoa"><i class="fa fa-check"></i><b>8.4</b> Visualizing the most dominant genus on PCoA</a>
<ul>
<li class="chapter" data-level="8.4.1" data-path="community-similarity.html"><a href="community-similarity.html#testing-differences-in-community-composition-between-sample-groups"><i class="fa fa-check"></i><b>8.4.1</b> Testing differences in community composition between sample groups</a></li>
<li class="chapter" data-level="8.4.2" data-path="community-similarity.html"><a href="community-similarity.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>8.4.2</b> Checking the homogeneity condition</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="community-similarity.html"><a href="community-similarity.html#further-reading"><i class="fa fa-check"></i><b>8.5</b> Further reading</a></li>
<li class="chapter" data-level="" data-path="community-similarity.html"><a href="community-similarity.html#session-info-5"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="microbiome-community.html"><a href="microbiome-community.html"><i class="fa fa-check"></i><b>9</b> Community composition</a>
<ul>
<li class="chapter" data-level="9.1" data-path="microbiome-community.html"><a href="microbiome-community.html#visual-composition"><i class="fa fa-check"></i><b>9.1</b> Visualizing taxonomic composition</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="microbiome-community.html"><a href="microbiome-community.html#composition-barplot"><i class="fa fa-check"></i><b>9.1.1</b> Composition barplot</a></li>
<li class="chapter" data-level="9.1.2" data-path="microbiome-community.html"><a href="microbiome-community.html#composition-heatmap"><i class="fa fa-check"></i><b>9.1.2</b> Composition heatmap</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="community-typing.html"><a href="community-typing.html"><i class="fa fa-check"></i><b>10</b> Community typing</a>
<ul>
<li class="chapter" data-level="10.1" data-path="community-typing.html"><a href="community-typing.html#dirichlet-multinomial-mixtures-dmm"><i class="fa fa-check"></i><b>10.1</b> Dirichlet Multinomial Mixtures (DMM)</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="community-typing.html"><a href="community-typing.html#pcoa-for-asv-level-data-with-bray-curtis-with-dmm-clusters-shown-with-colors"><i class="fa fa-check"></i><b>10.1.1</b> PCoA for ASV-level data with Bray-Curtis; with DMM clusters shown with colors</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="community-typing.html"><a href="community-typing.html#community-detection"><i class="fa fa-check"></i><b>10.2</b> Community Detection</a></li>
<li class="chapter" data-level="10.3" data-path="community-typing.html"><a href="community-typing.html#additional-community-typing"><i class="fa fa-check"></i><b>10.3</b> Additional Community Typing</a></li>
<li class="chapter" data-level="" data-path="community-typing.html"><a href="community-typing.html#session-info-6"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="biclustering.html"><a href="biclustering.html"><i class="fa fa-check"></i><b>11</b> Biclustering</a>
<ul>
<li class="chapter" data-level="11.1" data-path="biclustering.html"><a href="biclustering.html#taxa-vs-samples"><i class="fa fa-check"></i><b>11.1</b> Taxa vs samples</a></li>
<li class="chapter" data-level="11.2" data-path="biclustering.html"><a href="biclustering.html#taxa-vs-biomolecules"><i class="fa fa-check"></i><b>11.2</b> Taxa vs biomolecules</a></li>
<li class="chapter" data-level="11.3" data-path="biclustering.html"><a href="biclustering.html#taxa-vs-taxa"><i class="fa fa-check"></i><b>11.3</b> Taxa vs taxa</a></li>
<li class="chapter" data-level="" data-path="biclustering.html"><a href="biclustering.html#session-info-7"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="differential-abundance.html"><a href="differential-abundance.html"><i class="fa fa-check"></i><b>12</b> Differential abundance</a>
<ul>
<li class="chapter" data-level="12.1" data-path="differential-abundance.html"><a href="differential-abundance.html#differential-abundance-analysis"><i class="fa fa-check"></i><b>12.1</b> Differential abundance analysis</a>
<ul>
<li class="chapter" data-level="12.1.1" data-path="differential-abundance.html"><a href="differential-abundance.html#examples-and-tools"><i class="fa fa-check"></i><b>12.1.1</b> Examples and tools</a></li>
<li class="chapter" data-level="12.1.2" data-path="differential-abundance.html"><a href="differential-abundance.html#prevalence-filtering"><i class="fa fa-check"></i><b>12.1.2</b> Prevalence Filtering</a></li>
<li class="chapter" data-level="12.1.3" data-path="differential-abundance.html"><a href="differential-abundance.html#aldex2"><i class="fa fa-check"></i><b>12.1.3</b> ALDEx2</a></li>
<li class="chapter" data-level="12.1.4" data-path="differential-abundance.html"><a href="differential-abundance.html#ancom-bc"><i class="fa fa-check"></i><b>12.1.4</b> ANCOM-BC</a></li>
<li class="chapter" data-level="12.1.5" data-path="differential-abundance.html"><a href="differential-abundance.html#maaslin2"><i class="fa fa-check"></i><b>12.1.5</b> MaAsLin2</a></li>
<li class="chapter" data-level="12.1.6" data-path="differential-abundance.html"><a href="differential-abundance.html#comparison-of-the-methods"><i class="fa fa-check"></i><b>12.1.6</b> Comparison of the methods</a></li>
<li class="chapter" data-level="12.1.7" data-path="differential-abundance.html"><a href="differential-abundance.html#confounding-variables"><i class="fa fa-check"></i><b>12.1.7</b> Confounding variables</a></li>
</ul></li>
<li class="chapter" data-level="12.2" data-path="differential-abundance.html"><a href="differential-abundance.html#tree-based-methods"><i class="fa fa-check"></i><b>12.2</b> Tree-based methods</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="differential-abundance.html"><a href="differential-abundance.html#group-wise-associations-testing-based-on-balances-with-fido"><i class="fa fa-check"></i><b>12.2.1</b> Group-wise associations testing based on balances with fido</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="differential-abundance.html"><a href="differential-abundance.html#session-info-8"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="multi-assay_analyses.html"><a href="multi-assay_analyses.html"><i class="fa fa-check"></i><b>13</b> Multi-assay analyses</a>
<ul>
<li class="chapter" data-level="13.1" data-path="multi-assay_analyses.html"><a href="multi-assay_analyses.html#cross-correlation-analysis"><i class="fa fa-check"></i><b>13.1</b> Cross-correlation Analysis</a></li>
<li class="chapter" data-level="13.2" data-path="multi-assay_analyses.html"><a href="multi-assay_analyses.html#multi-omics-factor-analysis"><i class="fa fa-check"></i><b>13.2</b> Multi-Omics Factor Analysis</a></li>
<li class="chapter" data-level="" data-path="multi-assay_analyses.html"><a href="multi-assay_analyses.html#session-info-9"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="viz-chapter.html"><a href="viz-chapter.html"><i class="fa fa-check"></i><b>14</b> Visualization</a>
<ul>
<li class="chapter" data-level="14.1" data-path="viz-chapter.html"><a href="viz-chapter.html#pre-analysis-exploration"><i class="fa fa-check"></i><b>14.1</b> Pre-analysis exploration</a>
<ul>
<li class="chapter" data-level="14.1.1" data-path="viz-chapter.html"><a href="viz-chapter.html#accessing-row-and-column-data"><i class="fa fa-check"></i><b>14.1.1</b> Accessing row and column data</a></li>
<li class="chapter" data-level="14.1.2" data-path="viz-chapter.html"><a href="viz-chapter.html#viewing-abundance-and-prevalence-patterns"><i class="fa fa-check"></i><b>14.1.2</b> Viewing abundance and prevalence patterns</a></li>
</ul></li>
<li class="chapter" data-level="14.2" data-path="viz-chapter.html"><a href="viz-chapter.html#diversity-estimation"><i class="fa fa-check"></i><b>14.2</b> Diversity estimation</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="viz-chapter.html"><a href="viz-chapter.html#alpha-diversity-with-scatter-violin-and-box-plots"><i class="fa fa-check"></i><b>14.2.1</b> Alpha diversity with scatter, violin and box plots</a></li>
<li class="chapter" data-level="14.2.2" data-path="viz-chapter.html"><a href="viz-chapter.html#beta-diversity-with-shepard-and-coordination-plots"><i class="fa fa-check"></i><b>14.2.2</b> Beta diversity with Shepard and coordination plots</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="viz-chapter.html"><a href="viz-chapter.html#statistical-analysis"><i class="fa fa-check"></i><b>14.3</b> Statistical analysis</a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="viz-chapter.html"><a href="viz-chapter.html#heatmaps"><i class="fa fa-check"></i><b>14.3.1</b> Heatmaps</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="viz-chapter.html"><a href="viz-chapter.html#session-info-10"><i class="fa fa-check"></i>Session Info</a></li>
</ul></li>
<li class="part"><span><b>III Appendix</b></span></li>
<li class="chapter" data-level="15" data-path="resources.html"><a href="resources.html"><i class="fa fa-check"></i><b>15</b> Resources</a>
<ul>
<li class="chapter" data-level="15.1" data-path="resources.html"><a href="resources.html#data-containers-1"><i class="fa fa-check"></i><b>15.1</b> Data containers</a>
<ul>
<li class="chapter" data-level="15.1.1" data-path="resources.html"><a href="resources.html#resources-for-treesummarizedexperiment"><i class="fa fa-check"></i><b>15.1.1</b> Resources for TreeSummarizedExperiment</a></li>
<li class="chapter" data-level="15.1.2" data-path="resources.html"><a href="resources.html#other-relevant-containers"><i class="fa fa-check"></i><b>15.1.2</b> Other relevant containers</a></li>
<li class="chapter" data-level="15.1.3" data-path="resources.html"><a href="resources.html#alternative-containers-for-microbiome-data"><i class="fa fa-check"></i><b>15.1.3</b> Alternative containers for microbiome data</a></li>
<li class="chapter" data-level="15.1.4" data-path="resources.html"><a href="resources.html#resources-for-phyloseq"><i class="fa fa-check"></i><b>15.1.4</b> Resources for phyloseq</a></li>
</ul></li>
<li class="chapter" data-level="15.2" data-path="resources.html"><a href="resources.html#r-programming-resources"><i class="fa fa-check"></i><b>15.2</b> R programming resources</a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="resources.html"><a href="resources.html#bioc_intro"><i class="fa fa-check"></i><b>15.2.1</b> Bioconductor Classes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="extras.html"><a href="extras.html"><i class="fa fa-check"></i><b>16</b> Extra material</a>
<ul>
<li class="chapter" data-level="16.1" data-path="extras.html"><a href="extras.html#interactive-3d-plots"><i class="fa fa-check"></i><b>16.1</b> Interactive 3D Plots</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="acknowledgments.html"><a href="acknowledgments.html"><i class="fa fa-check"></i><b>17</b> Acknowledgments</a>
<ul>
<li class="chapter" data-level="17.1" data-path="acknowledgments.html"><a href="acknowledgments.html#introduction-to-treese"><i class="fa fa-check"></i><b>17.1</b> Introduction to TreeSE</a></li>
<li class="chapter" data-level="17.2" data-path="acknowledgments.html"><a href="acknowledgments.html#data-exploration"><i class="fa fa-check"></i><b>17.2</b> Data exploration</a></li>
<li class="chapter" data-level="17.3" data-path="acknowledgments.html"><a href="acknowledgments.html#differential-abundance-analysis-1"><i class="fa fa-check"></i><b>17.3</b> Differential abundance analysis</a></li>
<li class="chapter" data-level="17.4" data-path="acknowledgments.html"><a href="acknowledgments.html#basics-of-community-composition"><i class="fa fa-check"></i><b>17.4</b> Basics of community composition</a></li>
<li class="chapter" data-level="17.5" data-path="acknowledgments.html"><a href="acknowledgments.html#introduction-to-mae"><i class="fa fa-check"></i><b>17.5</b> Introduction to MAE</a></li>
<li class="chapter" data-level="17.6" data-path="acknowledgments.html"><a href="acknowledgments.html#introduction-to-multiomics"><i class="fa fa-check"></i><b>17.6</b> Introduction to multiomics</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="bibliography.html"><a href="bibliography.html"><i class="fa fa-check"></i>Bibliography</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Exercises</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="viz-chapter" class="section level1 hasAnchor" number="14">
<h1><span class="header-section-number">Chapter 14</span> Visualization<a href="viz-chapter.html#viz-chapter" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<script>
document.addEventListener("click", function (event) {
    if (event.target.classList.contains("rebook-collapse")) {
        event.target.classList.toggle("active");
        var content = event.target.nextElementSibling;
        if (content.style.display === "block") {
            content.style.display = "none";
        } else {
            content.style.display = "block";
        }
    }
})
</script>
<style>
.rebook-collapse {
  background-color: #eee;
  color: #444;
  cursor: pointer;
  padding: 18px;
  width: 100%;
  border: none;
  text-align: left;
  outline: none;
  font-size: 15px;
}

.rebook-content {
  padding: 0 18px;
  display: none;
  overflow: hidden;
  background-color: #f1f1f1;
}
</style>
<p>Whether a data set contains information on a microbial community or it originates from a different source, the way that data are visualized inevitably shapes how they will be interpreted, and motivates the next steps of the analysis.</p>
<p>A large variety of graphing methods belong to microbial analysis, but only few are the choices that will return useful answers about the data. Therefore, knowledge on the available tools and their possible applications plays an important role in selecting the most suitable method for the asked question.</p>
<p>This chapter introduces the reader to a number of visualization techniques found in this book, such as:</p>
<ul>
<li>bar plots</li>
<li>box plots</li>
<li>heatmaps</li>
<li>ordination charts</li>
<li>regression charts</li>
<li>trees</li>
</ul>
<p>The toolkit which provides the essential plotting functionality includes the following packages:</p>
<ul>
<li><em>patchwork</em>, <em>cowplot</em>, <em>ggpubr</em> and <em>gridExtra</em>: plot layout and multi-panel plotting</li>
<li><em>miaViz</em>: specific visualization tools for <code>TreeSummaizedExperiment</code> objects</li>
<li><em>scater</em>: specific visualization tools for <code>SingleCellExperiment</code> objects</li>
<li><em>ggplot2</em>, <em>pheatmap</em>, <em>ggtree</em>, <em>sechm</em>: composition heatmaps</li>
<li><em>ANCOMBC</em>, <em>ALDEx2</em> and <em>Maaslin2</em>: visual differential abundance</li>
<li><em>fido</em>: tree-based methods for differential abundance</li>
<li><em>plotly</em>: animated and 3D plotting</li>
</ul>
<p>For systematic and extensive tutorials on the visual tools available in <em>mia</em>, readers can refer to the following material:</p>
<ul>
<li><a href="https://microbiome.github.io/tutorials/">microbiome tutorials</a></li>
</ul>
<div id="pre-analysis-exploration" class="section level2 hasAnchor" number="14.1">
<h2><span class="header-section-number">14.1</span> Pre-analysis exploration<a href="viz-chapter.html#pre-analysis-exploration" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="accessing-row-and-column-data" class="section level3 hasAnchor" number="14.1.1">
<h3><span class="header-section-number">14.1.1</span> Accessing row and column data<a href="viz-chapter.html#accessing-row-and-column-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><code>SCE</code> and <code>TSE</code> objects contain multiple layers of information in the form of
rows, columns and meta data. The <em>scater</em> package supports in accessing,
modifying and graphing the meta data related to features as well as samples.</p>
<div class="sourceCode" id="cb351"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb351-1"><a href="viz-chapter.html#cb351-1" aria-hidden="true" tabindex="-1"></a><span class="co"># list row meta data</span></span>
<span id="cb351-2"><a href="viz-chapter.html#cb351-2" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(<span class="fu">rowData</span>(tse))</span></code></pre></div>
<pre><code>## [1] &quot;Kingdom&quot; &quot;Phylum&quot;  &quot;Class&quot;   &quot;Order&quot;   &quot;Family&quot;  &quot;Genus&quot;   &quot;Species&quot;</code></pre>
<div class="sourceCode" id="cb353"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb353-1"><a href="viz-chapter.html#cb353-1" aria-hidden="true" tabindex="-1"></a><span class="co"># list column meta data</span></span>
<span id="cb353-2"><a href="viz-chapter.html#cb353-2" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(<span class="fu">colData</span>(tse))</span></code></pre></div>
<pre><code>## [1] &quot;X.SampleID&quot;               &quot;Primer&quot;                  
## [3] &quot;Final_Barcode&quot;            &quot;Barcode_truncated_plus_T&quot;
## [5] &quot;Barcode_full_length&quot;      &quot;SampleType&quot;              
## [7] &quot;Description&quot;</code></pre>
<p>Such meta data can be directly plotted with the functions <code>plotRowData</code> and <code>plotColData</code>.</p>
<div class="sourceCode" id="cb355"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb355-1"><a href="viz-chapter.html#cb355-1" aria-hidden="true" tabindex="-1"></a><span class="co"># obtain QC data</span></span>
<span id="cb355-2"><a href="viz-chapter.html#cb355-2" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">addPerCellQC</span>(tse)</span>
<span id="cb355-3"><a href="viz-chapter.html#cb355-3" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">addPerFeatureQC</span>(tse)</span>
<span id="cb355-4"><a href="viz-chapter.html#cb355-4" aria-hidden="true" tabindex="-1"></a><span class="co"># plot QC Mean against Species</span></span>
<span id="cb355-5"><a href="viz-chapter.html#cb355-5" aria-hidden="true" tabindex="-1"></a><span class="fu">plotRowData</span>(tse, <span class="st">&quot;mean&quot;</span>, <span class="st">&quot;Species&quot;</span>) <span class="sc">+</span></span>
<span id="cb355-6"><a href="viz-chapter.html#cb355-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb355-7"><a href="viz-chapter.html#cb355-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Species&quot;</span>, <span class="at">y =</span> <span class="st">&quot;QC Mean&quot;</span>)</span></code></pre></div>
<p><img src="19_visualization_techniques_files/figure-html/unnamed-chunk-3-1.png" width="672" /></p>
<div class="sourceCode" id="cb356"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb356-1"><a href="viz-chapter.html#cb356-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot QC Sum against Sample ID, colour-labeled by Sample Type</span></span>
<span id="cb356-2"><a href="viz-chapter.html#cb356-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotColData</span>(tse, <span class="st">&quot;sum&quot;</span>, <span class="st">&quot;X.SampleID&quot;</span>, <span class="at">colour_by =</span> <span class="st">&quot;SampleType&quot;</span>) <span class="sc">+</span></span>
<span id="cb356-3"><a href="viz-chapter.html#cb356-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb356-4"><a href="viz-chapter.html#cb356-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Sample ID&quot;</span>, <span class="at">y =</span> <span class="st">&quot;QC Sum&quot;</span>)</span></code></pre></div>
<p><img src="19_visualization_techniques_files/figure-html/unnamed-chunk-3-2.png" width="672" /></p>
<p>Alternatively, they can be converted to a <code>data.frame</code> object and passed to <code>ggplot</code>.</p>
<div class="sourceCode" id="cb357"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb357-1"><a href="viz-chapter.html#cb357-1" aria-hidden="true" tabindex="-1"></a><span class="co"># store colData into a data frame</span></span>
<span id="cb357-2"><a href="viz-chapter.html#cb357-2" aria-hidden="true" tabindex="-1"></a>coldata <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(tse))</span>
<span id="cb357-3"><a href="viz-chapter.html#cb357-3" aria-hidden="true" tabindex="-1"></a><span class="co"># plot Number of Samples against Sampling Site</span></span>
<span id="cb357-4"><a href="viz-chapter.html#cb357-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(coldata, <span class="fu">aes</span>(<span class="at">x =</span> SampleType)) <span class="sc">+</span></span>
<span id="cb357-5"><a href="viz-chapter.html#cb357-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">width =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb357-6"><a href="viz-chapter.html#cb357-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb357-7"><a href="viz-chapter.html#cb357-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Sampling Site&quot;</span>,</span>
<span id="cb357-8"><a href="viz-chapter.html#cb357-8" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">&quot;Number of Samples&quot;</span>)</span></code></pre></div>
<p><img src="19_visualization_techniques_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<p>Further methods of application can be found in the chapters <a href="quality-control.html#qc">5.3</a> and
<a href="community-diversity.html#richness">7.1.1</a> and in a few <a href="https://github.com/davismcc/scater_tutorials_open_data">external tutorials</a>
with open data. Additionally, <code>rowData</code> and <code>colData</code> allow manipulation and
subsetting of large data sets into smaller units, as explained in chapter <a href="datamanipulation.html#datamanipulation">4</a>.</p>
</div>
<div id="viewing-abundance-and-prevalence-patterns" class="section level3 hasAnchor" number="14.1.2">
<h3><span class="header-section-number">14.1.2</span> Viewing abundance and prevalence patterns<a href="viz-chapter.html#viewing-abundance-and-prevalence-patterns" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Prior-to-analysis exploration may involve questions such as how microorganisms
are distributed across samples (abundance) and what microorganisms are present
in most of the samples (prevalence). The information on abundance and prevalence
can be summarized into a <strong>jitter</strong> or <strong>density plot</strong> and a <strong>tree</strong>,
respectively, with the <em>miaViz</em> package.</p>
<p>Specifically, the functions <code>plotAbundance</code>, <code>plotAbundanceDensity</code> and
<code>plotRowTree</code> are used, and examples on their usage are discussed throughout
chapter <a href="quality-control.html#quality-control">5</a>.</p>
</div>
</div>
<div id="diversity-estimation" class="section level2 hasAnchor" number="14.2">
<h2><span class="header-section-number">14.2</span> Diversity estimation<a href="viz-chapter.html#diversity-estimation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Alpha diversity is commonly measured as one of the diversity indices explained
in chapter <a href="community-diversity.html#community-diversity">7</a>. Because the focus lies on each sample
separately, one-dimensional plots, such as <strong>scatter</strong>, <strong>violin</strong> and
<strong>box plots</strong>, are suitable.</p>
<p>Beta diversity is generally evaluated as one of the dissimilarity indices
reported in chapter <a href="community-similarity.html#community-similarity">8</a>. Unlike alpha diversity,
samples are compared collectively to estimate the heterogeneity across them,
therefore multidimensional plots, such as <strong>Shepard</strong> and <strong>ordination plots</strong>
are suitable.</p>
<table>
<colgroup>
<col width="31%" />
<col width="35%" />
<col width="33%" />
</colgroup>
<thead>
<tr class="header">
<th align="center"></th>
<th align="center">alpha diversity</th>
<th align="center">beta diversity</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">used metrics</td>
<td align="center">diversity indices</td>
<td align="center">dissimilarity indices</td>
</tr>
<tr class="even">
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
</tr>
<tr class="odd">
<td align="center">metric dimensionality</td>
<td align="center">one-dimensional</td>
<td align="center">multidimensional</td>
</tr>
<tr class="even">
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
</tr>
<tr class="odd">
<td align="center">suitable visualization</td>
<td align="center">scatter, violin, box plots</td>
<td align="center">Shepard, ordination plots</td>
</tr>
</tbody>
</table>
<p>As a conclusion, visualization techniques for alpha and beta diversity significantly differ from one another.</p>
<div id="alpha-diversity-with-scatter-violin-and-box-plots" class="section level3 hasAnchor" number="14.2.1">
<h3><span class="header-section-number">14.2.1</span> Alpha diversity with scatter, violin and box plots<a href="viz-chapter.html#alpha-diversity-with-scatter-violin-and-box-plots" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The basic method to visualize the diversity values assigned to the different
samples in a <code>TSE</code> object includes the following, where each data point represents one sample:</p>
<div class="sourceCode" id="cb358"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb358-1"><a href="viz-chapter.html#cb358-1" aria-hidden="true" tabindex="-1"></a><span class="co"># estimate shannon diversity index</span></span>
<span id="cb358-2"><a href="viz-chapter.html#cb358-2" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> mia<span class="sc">::</span><span class="fu">estimateDiversity</span>(tse, </span>
<span id="cb358-3"><a href="viz-chapter.html#cb358-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">abund_values =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb358-4"><a href="viz-chapter.html#cb358-4" aria-hidden="true" tabindex="-1"></a>                              <span class="at">index =</span> <span class="st">&quot;shannon&quot;</span>, </span>
<span id="cb358-5"><a href="viz-chapter.html#cb358-5" aria-hidden="true" tabindex="-1"></a>                              <span class="at">name =</span> <span class="st">&quot;shannon&quot;</span>)</span>
<span id="cb358-6"><a href="viz-chapter.html#cb358-6" aria-hidden="true" tabindex="-1"></a><span class="co"># plot shannon diversity index, colour-labeled by Sample Type</span></span>
<span id="cb358-7"><a href="viz-chapter.html#cb358-7" aria-hidden="true" tabindex="-1"></a><span class="fu">plotColData</span>(tse, <span class="st">&quot;shannon&quot;</span>, <span class="at">colour_by =</span> <span class="st">&quot;SampleType&quot;</span>)</span></code></pre></div>
<p><img src="19_visualization_techniques_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
<p>The several indices available for the evaluation of alpha diversity often return
slightly divergent results, which can be visually compared with a multiple violin
or box plot. For this purpose, <code>plotColData</code> (for violin plots) or <code>ggplot</code>
(for box plots) are recursively applied to a number of diversity indices with
the function <code>lapply</code> and the multi-panel plotting functionality of the
<em>patchwork</em> package is then exploited.</p>
<div class="sourceCode" id="cb359"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb359-1"><a href="viz-chapter.html#cb359-1" aria-hidden="true" tabindex="-1"></a><span class="co"># estimate faith diversity index</span></span>
<span id="cb359-2"><a href="viz-chapter.html#cb359-2" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> mia<span class="sc">::</span><span class="fu">estimateFaith</span>(tse,</span>
<span id="cb359-3"><a href="viz-chapter.html#cb359-3" aria-hidden="true" tabindex="-1"></a>                          <span class="at">abund_values =</span> <span class="st">&quot;counts&quot;</span>)</span>
<span id="cb359-4"><a href="viz-chapter.html#cb359-4" aria-hidden="true" tabindex="-1"></a><span class="co"># store colData into a data frame</span></span>
<span id="cb359-5"><a href="viz-chapter.html#cb359-5" aria-hidden="true" tabindex="-1"></a>coldata <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(tse))</span>
<span id="cb359-6"><a href="viz-chapter.html#cb359-6" aria-hidden="true" tabindex="-1"></a><span class="co"># generate plots for shannon and faith indices</span></span>
<span id="cb359-7"><a href="viz-chapter.html#cb359-7" aria-hidden="true" tabindex="-1"></a><span class="co"># and store them into a list</span></span>
<span id="cb359-8"><a href="viz-chapter.html#cb359-8" aria-hidden="true" tabindex="-1"></a>plots <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">c</span>(<span class="st">&quot;shannon&quot;</span>, <span class="st">&quot;faith&quot;</span>),</span>
<span id="cb359-9"><a href="viz-chapter.html#cb359-9" aria-hidden="true" tabindex="-1"></a>                <span class="cf">function</span>(i) <span class="fu">ggplot</span>(coldata, <span class="fu">aes_string</span>(<span class="at">y =</span> i)) <span class="sc">+</span></span>
<span id="cb359-10"><a href="viz-chapter.html#cb359-10" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb359-11"><a href="viz-chapter.html#cb359-11" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb359-12"><a href="viz-chapter.html#cb359-12" aria-hidden="true" tabindex="-1"></a>                        <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>()))</span>
<span id="cb359-13"><a href="viz-chapter.html#cb359-13" aria-hidden="true" tabindex="-1"></a><span class="co"># combine plots with patchwork</span></span>
<span id="cb359-14"><a href="viz-chapter.html#cb359-14" aria-hidden="true" tabindex="-1"></a>plots[[<span class="dv">1</span>]] <span class="sc">+</span> plots[[<span class="dv">2</span>]]</span></code></pre></div>
<p><img src="19_visualization_techniques_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
<p>The analogous output in the form of a violin plot is obtained in chapter
<a href="community-diversity.html#faith-diversity">7.1.3</a>. In addition, box plots that group samples according to
certain information, such as origin, sex, age and health condition, can be
labeled with p-values for significant differences with the package <em>ggsignif</em>
package, as shown in chapter <a href="community-diversity.html#estimate-diversity">7.1.2</a>.</p>
</div>
<div id="beta-diversity-with-shepard-and-coordination-plots" class="section level3 hasAnchor" number="14.2.2">
<h3><span class="header-section-number">14.2.2</span> Beta diversity with Shepard and coordination plots<a href="viz-chapter.html#beta-diversity-with-shepard-and-coordination-plots" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The <em>scater</em> package offers the general function <code>plotReducedDim</code>. In its basic
form, it takes a <code>TSE</code> object and the results on sample similarity stored in the
same object, which can be evaluated with the following coordination methods:</p>
<ul>
<li><code>runMDS</code></li>
<li><code>runNMDS</code></li>
<li><code>runPCA</code></li>
<li><code>runTSNE</code></li>
<li><code>runUMAP</code></li>
</ul>
<p>Since these clustering techniques allow for multiple coordinates or components,
<strong>coordination plots</strong> can also span multiple dimensions, which is explained in chapter @ref{extras}.</p>
<div class="sourceCode" id="cb360"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb360-1"><a href="viz-chapter.html#cb360-1" aria-hidden="true" tabindex="-1"></a><span class="co"># perform NMDS coordination method</span></span>
<span id="cb360-2"><a href="viz-chapter.html#cb360-2" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">runNMDS</span>(tse,</span>
<span id="cb360-3"><a href="viz-chapter.html#cb360-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">FUN =</span> vegan<span class="sc">::</span>vegdist,</span>
<span id="cb360-4"><a href="viz-chapter.html#cb360-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">name =</span> <span class="st">&quot;NMDS&quot;</span>)</span></code></pre></div>
<pre><code>## initial  value 47.733208 
## iter   5 value 33.853364
## iter  10 value 32.891200
## final  value 32.823570 
## converged</code></pre>
<div class="sourceCode" id="cb362"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb362-1"><a href="viz-chapter.html#cb362-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot results of a 2-component NMDS on tse,</span></span>
<span id="cb362-2"><a href="viz-chapter.html#cb362-2" aria-hidden="true" tabindex="-1"></a><span class="co"># coloured-scaled by shannon diversity index</span></span>
<span id="cb362-3"><a href="viz-chapter.html#cb362-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plotReducedDim</span>(tse, <span class="st">&quot;NMDS&quot;</span>, <span class="at">colour_by =</span> <span class="st">&quot;shannon&quot;</span>)</span></code></pre></div>
<p><img src="19_visualization_techniques_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
<p>Multiple combinations of coordinates or dimensions can also be integrated into a multi-panel arrangement.</p>
<div class="sourceCode" id="cb363"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb363-1"><a href="viz-chapter.html#cb363-1" aria-hidden="true" tabindex="-1"></a><span class="co"># perform MDS coordination method</span></span>
<span id="cb363-2"><a href="viz-chapter.html#cb363-2" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">runMDS</span>(tse,</span>
<span id="cb363-3"><a href="viz-chapter.html#cb363-3" aria-hidden="true" tabindex="-1"></a>              <span class="at">FUN =</span> vegan<span class="sc">::</span>vegdist,</span>
<span id="cb363-4"><a href="viz-chapter.html#cb363-4" aria-hidden="true" tabindex="-1"></a>              <span class="at">method =</span> <span class="st">&quot;bray&quot;</span>,</span>
<span id="cb363-5"><a href="viz-chapter.html#cb363-5" aria-hidden="true" tabindex="-1"></a>              <span class="at">name =</span> <span class="st">&quot;MDS&quot;</span>,</span>
<span id="cb363-6"><a href="viz-chapter.html#cb363-6" aria-hidden="true" tabindex="-1"></a>              <span class="at">exprs_values =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb363-7"><a href="viz-chapter.html#cb363-7" aria-hidden="true" tabindex="-1"></a>              <span class="at">ncomponents =</span> <span class="dv">3</span>)</span>
<span id="cb363-8"><a href="viz-chapter.html#cb363-8" aria-hidden="true" tabindex="-1"></a><span class="co"># plot results of a 3-component MDS on tse,</span></span>
<span id="cb363-9"><a href="viz-chapter.html#cb363-9" aria-hidden="true" tabindex="-1"></a><span class="co"># coloured-scaled by faith diversity index</span></span>
<span id="cb363-10"><a href="viz-chapter.html#cb363-10" aria-hidden="true" tabindex="-1"></a><span class="fu">plotReducedDim</span>(tse, <span class="st">&quot;MDS&quot;</span>, <span class="at">ncomponents =</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>), <span class="at">colour_by =</span> <span class="st">&quot;faith&quot;</span>)</span></code></pre></div>
<p><img src="19_visualization_techniques_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
<p>Similarly to iterating <code>plotColData</code> over indices of alpha diversity, <code>lapply</code>
can be used in combination with <em>patchwork</em> to recursively apply <code>plotReducedDim</code>
and visually compare results among various coordination methods.</p>
<div class="sourceCode" id="cb364"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb364-1"><a href="viz-chapter.html#cb364-1" aria-hidden="true" tabindex="-1"></a><span class="co"># generate plots for MDS and NMDS methods</span></span>
<span id="cb364-2"><a href="viz-chapter.html#cb364-2" aria-hidden="true" tabindex="-1"></a><span class="co"># and store them into a list</span></span>
<span id="cb364-3"><a href="viz-chapter.html#cb364-3" aria-hidden="true" tabindex="-1"></a>plots <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">c</span>(<span class="st">&quot;MDS&quot;</span>, <span class="st">&quot;NMDS&quot;</span>),</span>
<span id="cb364-4"><a href="viz-chapter.html#cb364-4" aria-hidden="true" tabindex="-1"></a>                plotReducedDim,</span>
<span id="cb364-5"><a href="viz-chapter.html#cb364-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">object =</span> tse,</span>
<span id="cb364-6"><a href="viz-chapter.html#cb364-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">colour_by =</span> <span class="st">&quot;shannon&quot;</span>)</span>
<span id="cb364-7"><a href="viz-chapter.html#cb364-7" aria-hidden="true" tabindex="-1"></a><span class="co"># combine plots with patchwork</span></span>
<span id="cb364-8"><a href="viz-chapter.html#cb364-8" aria-hidden="true" tabindex="-1"></a>plots[[<span class="dv">1</span>]] <span class="sc">+</span> plots[[<span class="dv">2</span>]] <span class="sc">+</span></span>
<span id="cb364-9"><a href="viz-chapter.html#cb364-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_layout</span>(<span class="at">guides =</span> <span class="st">&quot;collect&quot;</span>)</span></code></pre></div>
<p><img src="19_visualization_techniques_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
<p>For similar examples, readers are referred to chapter <a href="community-similarity.html#community-similarity">8</a>.
Further material on the graphic capabilities of <em>patchwork</em> is available in its
<a href="https://patchwork.data-imaginist.com/articles/patchwork.html">official package tutorial</a>.</p>
</div>
</div>
<div id="statistical-analysis" class="section level2 hasAnchor" number="14.3">
<h2><span class="header-section-number">14.3</span> Statistical analysis<a href="viz-chapter.html#statistical-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="heatmaps" class="section level3 hasAnchor" number="14.3.1">
<h3><span class="header-section-number">14.3.1</span> Heatmaps<a href="viz-chapter.html#heatmaps" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>As described in chapter <a href="microbiome-community.html#visual-composition">9.1</a>, bar plots and heatmaps can
offer a useful insight into the composition of a community. Simple methods involve
the functions <code>plotAbundance</code> and <code>geom_tile</code> in combination with <code>scale_fill_gradientn</code>
from the packages <em>miaViz</em> and <em>ggplot2</em>, respectively.</p>
<p>For instance, below the composition of multiple samples (x axis) is reported
in terms of relative abundances (y axis) for the top 10 taxa at the Order rank.
Bar plots and heatmaps with analogous information at the Phylum level are
available in the aforementioned chapter.</p>
<div class="sourceCode" id="cb365"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb365-1"><a href="viz-chapter.html#cb365-1" aria-hidden="true" tabindex="-1"></a><span class="co"># agglomerate tse by Order</span></span>
<span id="cb365-2"><a href="viz-chapter.html#cb365-2" aria-hidden="true" tabindex="-1"></a>tse_order <span class="ot">&lt;-</span> <span class="fu">agglomerateByRank</span>(tse,</span>
<span id="cb365-3"><a href="viz-chapter.html#cb365-3" aria-hidden="true" tabindex="-1"></a>                                <span class="at">rank =</span> <span class="st">&quot;Order&quot;</span>,</span>
<span id="cb365-4"><a href="viz-chapter.html#cb365-4" aria-hidden="true" tabindex="-1"></a>                                <span class="at">onRankOnly =</span> <span class="cn">TRUE</span>)</span>
<span id="cb365-5"><a href="viz-chapter.html#cb365-5" aria-hidden="true" tabindex="-1"></a><span class="co"># transform counts into relative abundance</span></span>
<span id="cb365-6"><a href="viz-chapter.html#cb365-6" aria-hidden="true" tabindex="-1"></a>tse_order <span class="ot">&lt;-</span> <span class="fu">transformCounts</span>(tse_order,</span>
<span id="cb365-7"><a href="viz-chapter.html#cb365-7" aria-hidden="true" tabindex="-1"></a>                              <span class="at">abund_values =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb365-8"><a href="viz-chapter.html#cb365-8" aria-hidden="true" tabindex="-1"></a>                              <span class="at">method =</span> <span class="st">&quot;relabundance&quot;</span>)</span>
<span id="cb365-9"><a href="viz-chapter.html#cb365-9" aria-hidden="true" tabindex="-1"></a><span class="co"># get top orders</span></span>
<span id="cb365-10"><a href="viz-chapter.html#cb365-10" aria-hidden="true" tabindex="-1"></a>top_taxa <span class="ot">&lt;-</span> <span class="fu">getTopTaxa</span>(tse_order,</span>
<span id="cb365-11"><a href="viz-chapter.html#cb365-11" aria-hidden="true" tabindex="-1"></a>                       <span class="at">top =</span> <span class="dv">10</span>,</span>
<span id="cb365-12"><a href="viz-chapter.html#cb365-12" aria-hidden="true" tabindex="-1"></a>                       <span class="at">abund_values =</span> <span class="st">&quot;relabundance&quot;</span>)</span>
<span id="cb365-13"><a href="viz-chapter.html#cb365-13" aria-hidden="true" tabindex="-1"></a><span class="co"># leave only names for top 10 orders and label the rest with &quot;Other&quot;</span></span>
<span id="cb365-14"><a href="viz-chapter.html#cb365-14" aria-hidden="true" tabindex="-1"></a>order_renamed <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">rowData</span>(tse_order)<span class="sc">$</span>Order,</span>
<span id="cb365-15"><a href="viz-chapter.html#cb365-15" aria-hidden="true" tabindex="-1"></a>                   <span class="cf">function</span>(x){<span class="cf">if</span> (x <span class="sc">%in%</span> top_taxa) {x} <span class="cf">else</span> {<span class="st">&quot;Other&quot;</span>}})</span>
<span id="cb365-16"><a href="viz-chapter.html#cb365-16" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(tse_order)<span class="sc">$</span>Order <span class="ot">&lt;-</span> <span class="fu">as.character</span>(order_renamed)</span>
<span id="cb365-17"><a href="viz-chapter.html#cb365-17" aria-hidden="true" tabindex="-1"></a><span class="co"># plot composition as a bar plot</span></span>
<span id="cb365-18"><a href="viz-chapter.html#cb365-18" aria-hidden="true" tabindex="-1"></a><span class="fu">plotAbundance</span>(tse_order,</span>
<span id="cb365-19"><a href="viz-chapter.html#cb365-19" aria-hidden="true" tabindex="-1"></a>              <span class="at">abund_values =</span> <span class="st">&quot;relabundance&quot;</span>,</span>
<span id="cb365-20"><a href="viz-chapter.html#cb365-20" aria-hidden="true" tabindex="-1"></a>              <span class="at">rank =</span> <span class="st">&quot;Order&quot;</span>,</span>
<span id="cb365-21"><a href="viz-chapter.html#cb365-21" aria-hidden="true" tabindex="-1"></a>              <span class="at">order_rank_by =</span> <span class="st">&quot;abund&quot;</span>,</span>
<span id="cb365-22"><a href="viz-chapter.html#cb365-22" aria-hidden="true" tabindex="-1"></a>              <span class="at">order_sample_by =</span> <span class="st">&quot;Clostridiales&quot;</span>)</span></code></pre></div>
<p><img src="19_visualization_techniques_files/figure-html/unnamed-chunk-10-1.png" width="672" /></p>
<p>For more sophisticated visualizations than those produced with <code>plotAbundance</code>
and <em>ggplot2</em>, the packages <em>pheatmap</em> and <em>sechm</em> provide methods to include
feature and sample clusters in a heatmap, along with further functionality.</p>
<div class="sourceCode" id="cb366"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb366-1"><a href="viz-chapter.html#cb366-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Agglomerate tse by phylum</span></span>
<span id="cb366-2"><a href="viz-chapter.html#cb366-2" aria-hidden="true" tabindex="-1"></a>tse_phylum <span class="ot">&lt;-</span> <span class="fu">agglomerateByRank</span>(tse,</span>
<span id="cb366-3"><a href="viz-chapter.html#cb366-3" aria-hidden="true" tabindex="-1"></a>                                <span class="at">rank =</span> <span class="st">&quot;Phylum&quot;</span>,</span>
<span id="cb366-4"><a href="viz-chapter.html#cb366-4" aria-hidden="true" tabindex="-1"></a>                                <span class="at">onRankOnly =</span> <span class="cn">TRUE</span>)</span>
<span id="cb366-5"><a href="viz-chapter.html#cb366-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Add clr-transformation on samples</span></span>
<span id="cb366-6"><a href="viz-chapter.html#cb366-6" aria-hidden="true" tabindex="-1"></a>tse_phylum <span class="ot">&lt;-</span> <span class="fu">transformSamples</span>(tse_phylum, <span class="at">method =</span> <span class="st">&quot;clr&quot;</span>, <span class="at">pseudocount =</span> <span class="dv">1</span>)</span>
<span id="cb366-7"><a href="viz-chapter.html#cb366-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Add z-transformation on features (taxa)</span></span>
<span id="cb366-8"><a href="viz-chapter.html#cb366-8" aria-hidden="true" tabindex="-1"></a>tse_phylum <span class="ot">&lt;-</span> <span class="fu">transformFeatures</span>(tse_phylum, <span class="at">abund_values =</span> <span class="st">&quot;clr&quot;</span>, </span>
<span id="cb366-9"><a href="viz-chapter.html#cb366-9" aria-hidden="true" tabindex="-1"></a>                                <span class="at">method =</span> <span class="st">&quot;z&quot;</span>, <span class="at">name =</span> <span class="st">&quot;clr_z&quot;</span>)</span>
<span id="cb366-10"><a href="viz-chapter.html#cb366-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb366-11"><a href="viz-chapter.html#cb366-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Takes subset: only samples from feces, skin, or tongue</span></span>
<span id="cb366-12"><a href="viz-chapter.html#cb366-12" aria-hidden="true" tabindex="-1"></a>tse_phylum_subset <span class="ot">&lt;-</span> tse_phylum[ , <span class="fu">colData</span>(tse_phylum)<span class="sc">$</span>SampleType <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Feces&quot;</span>, <span class="st">&quot;Skin&quot;</span>, <span class="st">&quot;Tongue&quot;</span>) ]</span>
<span id="cb366-13"><a href="viz-chapter.html#cb366-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb366-14"><a href="viz-chapter.html#cb366-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Does clr-transformation</span></span>
<span id="cb366-15"><a href="viz-chapter.html#cb366-15" aria-hidden="true" tabindex="-1"></a>tse_phylum_subset <span class="ot">&lt;-</span> <span class="fu">transformSamples</span>(tse_phylum_subset, <span class="at">method =</span> <span class="st">&quot;clr&quot;</span>, <span class="at">pseudocount =</span> <span class="dv">1</span>)</span>
<span id="cb366-16"><a href="viz-chapter.html#cb366-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Does z-transformation</span></span>
<span id="cb366-17"><a href="viz-chapter.html#cb366-17" aria-hidden="true" tabindex="-1"></a>tse_phylum_subset <span class="ot">&lt;-</span> <span class="fu">transformFeatures</span>(tse_phylum_subset, <span class="at">abund_values =</span> <span class="st">&quot;clr&quot;</span>, </span>
<span id="cb366-18"><a href="viz-chapter.html#cb366-18" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">method =</span> <span class="st">&quot;z&quot;</span>, <span class="at">name =</span> <span class="st">&quot;clr_z&quot;</span>)</span>
<span id="cb366-19"><a href="viz-chapter.html#cb366-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb366-20"><a href="viz-chapter.html#cb366-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Get n most abundant taxa, and subsets the data by them</span></span>
<span id="cb366-21"><a href="viz-chapter.html#cb366-21" aria-hidden="true" tabindex="-1"></a>top_taxa <span class="ot">&lt;-</span> <span class="fu">getTopTaxa</span>(tse_phylum_subset, <span class="at">top =</span> <span class="dv">20</span>)</span>
<span id="cb366-22"><a href="viz-chapter.html#cb366-22" aria-hidden="true" tabindex="-1"></a>tse_phylum_subset <span class="ot">&lt;-</span> tse_phylum_subset[top_taxa, ]</span>
<span id="cb366-23"><a href="viz-chapter.html#cb366-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb366-24"><a href="viz-chapter.html#cb366-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Gets the assay table</span></span>
<span id="cb366-25"><a href="viz-chapter.html#cb366-25" aria-hidden="true" tabindex="-1"></a>mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(tse_phylum_subset, <span class="st">&quot;clr_z&quot;</span>)</span>
<span id="cb366-26"><a href="viz-chapter.html#cb366-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb366-27"><a href="viz-chapter.html#cb366-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates the heatmap</span></span>
<span id="cb366-28"><a href="viz-chapter.html#cb366-28" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(mat)</span></code></pre></div>
<p><img src="19_visualization_techniques_files/figure-html/pheatmap1-1.png" width="672" /></p>
<p>We can cluster both samples and features hierarchically and add them to the
x and y axes of the heatmap, respectively.</p>
<div class="sourceCode" id="cb367"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb367-1"><a href="viz-chapter.html#cb367-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Hierarchical clustering</span></span>
<span id="cb367-2"><a href="viz-chapter.html#cb367-2" aria-hidden="true" tabindex="-1"></a>taxa_hclust <span class="ot">&lt;-</span> <span class="fu">hclust</span>(<span class="fu">dist</span>(mat), <span class="at">method =</span> <span class="st">&quot;complete&quot;</span>)</span>
<span id="cb367-3"><a href="viz-chapter.html#cb367-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb367-4"><a href="viz-chapter.html#cb367-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates a phylogenetic tree</span></span>
<span id="cb367-5"><a href="viz-chapter.html#cb367-5" aria-hidden="true" tabindex="-1"></a>taxa_tree <span class="ot">&lt;-</span> <span class="fu">as.phylo</span>(taxa_hclust)</span>
<span id="cb367-6"><a href="viz-chapter.html#cb367-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb367-7"><a href="viz-chapter.html#cb367-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot taxa tree</span></span>
<span id="cb367-8"><a href="viz-chapter.html#cb367-8" aria-hidden="true" tabindex="-1"></a>taxa_tree <span class="ot">&lt;-</span> <span class="fu">ggtree</span>(taxa_tree) <span class="sc">+</span> </span>
<span id="cb367-9"><a href="viz-chapter.html#cb367-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.margin=</span><span class="fu">margin</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>)) <span class="co"># removes margins</span></span>
<span id="cb367-10"><a href="viz-chapter.html#cb367-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb367-11"><a href="viz-chapter.html#cb367-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Get order of taxa in plot</span></span>
<span id="cb367-12"><a href="viz-chapter.html#cb367-12" aria-hidden="true" tabindex="-1"></a>taxa_ordered <span class="ot">&lt;-</span> <span class="fu">get_taxa_name</span>(taxa_tree)</span>
<span id="cb367-13"><a href="viz-chapter.html#cb367-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb367-14"><a href="viz-chapter.html#cb367-14" aria-hidden="true" tabindex="-1"></a><span class="co"># to view the tree, run</span></span>
<span id="cb367-15"><a href="viz-chapter.html#cb367-15" aria-hidden="true" tabindex="-1"></a><span class="co"># taxa_tree</span></span></code></pre></div>
<p>Based on phylo tree, we decide to create three clusters.</p>
<div class="sourceCode" id="cb368"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb368-1"><a href="viz-chapter.html#cb368-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates clusters</span></span>
<span id="cb368-2"><a href="viz-chapter.html#cb368-2" aria-hidden="true" tabindex="-1"></a>taxa_clusters <span class="ot">&lt;-</span> <span class="fu">cutree</span>(<span class="at">tree =</span> taxa_hclust, <span class="at">k =</span> <span class="dv">3</span>)</span>
<span id="cb368-3"><a href="viz-chapter.html#cb368-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb368-4"><a href="viz-chapter.html#cb368-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Converts into data frame</span></span>
<span id="cb368-5"><a href="viz-chapter.html#cb368-5" aria-hidden="true" tabindex="-1"></a>taxa_clusters <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">clusters =</span> taxa_clusters)</span>
<span id="cb368-6"><a href="viz-chapter.html#cb368-6" aria-hidden="true" tabindex="-1"></a>taxa_clusters<span class="sc">$</span>clusters <span class="ot">&lt;-</span> <span class="fu">factor</span>(taxa_clusters<span class="sc">$</span>clusters)</span>
<span id="cb368-7"><a href="viz-chapter.html#cb368-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb368-8"><a href="viz-chapter.html#cb368-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Order data so that it&#39;s same as in phylo tree</span></span>
<span id="cb368-9"><a href="viz-chapter.html#cb368-9" aria-hidden="true" tabindex="-1"></a>taxa_clusters <span class="ot">&lt;-</span> taxa_clusters[taxa_ordered, , drop <span class="ot">=</span> <span class="cn">FALSE</span>] </span>
<span id="cb368-10"><a href="viz-chapter.html#cb368-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb368-11"><a href="viz-chapter.html#cb368-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Prints taxa and their clusters</span></span>
<span id="cb368-12"><a href="viz-chapter.html#cb368-12" aria-hidden="true" tabindex="-1"></a>taxa_clusters</span></code></pre></div>
<pre><code>##                  clusters
## Chloroflexi             3
## Actinobacteria          3
## Crenarchaeota           3
## Planctomycetes          3
## Gemmatimonadetes        3
## Thermi                  3
## Acidobacteria           3
## Spirochaetes            2
## Fusobacteria            2
## SR1                     2
## Cyanobacteria           2
## Proteobacteria          2
## Synergistetes           2
## Lentisphaerae           1
## Bacteroidetes           1
## Verrucomicrobia         1
## Tenericutes             1
## Firmicutes              1
## Euryarchaeota           1
## SAR406                  1</code></pre>
<p>The information on the clusters is then added to the feature meta data.</p>
<div class="sourceCode" id="cb370"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb370-1"><a href="viz-chapter.html#cb370-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Adds information to rowData</span></span>
<span id="cb370-2"><a href="viz-chapter.html#cb370-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(tse_phylum_subset)<span class="sc">$</span>clusters <span class="ot">&lt;-</span> taxa_clusters[<span class="fu">order</span>(<span class="fu">match</span>(<span class="fu">rownames</span>(taxa_clusters), </span>
<span id="cb370-3"><a href="viz-chapter.html#cb370-3" aria-hidden="true" tabindex="-1"></a>                                                                 <span class="fu">rownames</span>(tse_phylum_subset))), ]</span>
<span id="cb370-4"><a href="viz-chapter.html#cb370-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb370-5"><a href="viz-chapter.html#cb370-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Prints taxa and their clusters</span></span>
<span id="cb370-6"><a href="viz-chapter.html#cb370-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(tse_phylum_subset)<span class="sc">$</span>clusters</span></code></pre></div>
<pre><code>##  [1] 1 1 2 3 2 2 1 1 1 1 3 2 3 3 3 2 2 3 3 1
## Levels: 1 2 3</code></pre>
<p>Similarly, samples are hierarchically grouped into clusters, the most suitable
number of clusters for the plot is selected and the new information is stored
into the sample meta data.</p>
<div class="sourceCode" id="cb372"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb372-1"><a href="viz-chapter.html#cb372-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Hierarchical clustering</span></span>
<span id="cb372-2"><a href="viz-chapter.html#cb372-2" aria-hidden="true" tabindex="-1"></a>sample_hclust <span class="ot">&lt;-</span> <span class="fu">hclust</span>(<span class="fu">dist</span>(<span class="fu">t</span>(mat)), <span class="at">method =</span> <span class="st">&quot;complete&quot;</span>)</span>
<span id="cb372-3"><a href="viz-chapter.html#cb372-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb372-4"><a href="viz-chapter.html#cb372-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates a phylogenetic tree</span></span>
<span id="cb372-5"><a href="viz-chapter.html#cb372-5" aria-hidden="true" tabindex="-1"></a>sample_tree <span class="ot">&lt;-</span> <span class="fu">as.phylo</span>(sample_hclust)</span>
<span id="cb372-6"><a href="viz-chapter.html#cb372-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb372-7"><a href="viz-chapter.html#cb372-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot sample tree</span></span>
<span id="cb372-8"><a href="viz-chapter.html#cb372-8" aria-hidden="true" tabindex="-1"></a>sample_tree <span class="ot">&lt;-</span> <span class="fu">ggtree</span>(sample_tree) <span class="sc">+</span> <span class="fu">layout_dendrogram</span>() <span class="sc">+</span> </span>
<span id="cb372-9"><a href="viz-chapter.html#cb372-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.margin=</span><span class="fu">margin</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>)) <span class="co"># removes margins</span></span>
<span id="cb372-10"><a href="viz-chapter.html#cb372-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb372-11"><a href="viz-chapter.html#cb372-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Get order of samples in plot</span></span>
<span id="cb372-12"><a href="viz-chapter.html#cb372-12" aria-hidden="true" tabindex="-1"></a>samples_ordered <span class="ot">&lt;-</span> <span class="fu">rev</span>(<span class="fu">get_taxa_name</span>(sample_tree))</span>
<span id="cb372-13"><a href="viz-chapter.html#cb372-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb372-14"><a href="viz-chapter.html#cb372-14" aria-hidden="true" tabindex="-1"></a><span class="co"># to view the tree, run</span></span>
<span id="cb372-15"><a href="viz-chapter.html#cb372-15" aria-hidden="true" tabindex="-1"></a><span class="co"># sample_tree</span></span>
<span id="cb372-16"><a href="viz-chapter.html#cb372-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb372-17"><a href="viz-chapter.html#cb372-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates clusters</span></span>
<span id="cb372-18"><a href="viz-chapter.html#cb372-18" aria-hidden="true" tabindex="-1"></a>sample_clusters <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">cutree</span>(<span class="at">tree =</span> sample_hclust, <span class="at">k =</span> <span class="dv">3</span>))</span>
<span id="cb372-19"><a href="viz-chapter.html#cb372-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb372-20"><a href="viz-chapter.html#cb372-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Converts into data frame</span></span>
<span id="cb372-21"><a href="viz-chapter.html#cb372-21" aria-hidden="true" tabindex="-1"></a>sample_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">clusters =</span> sample_clusters)</span>
<span id="cb372-22"><a href="viz-chapter.html#cb372-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb372-23"><a href="viz-chapter.html#cb372-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Order data so that it&#39;s same as in phylo tree</span></span>
<span id="cb372-24"><a href="viz-chapter.html#cb372-24" aria-hidden="true" tabindex="-1"></a>sample_data <span class="ot">&lt;-</span> sample_data[samples_ordered, , drop <span class="ot">=</span> <span class="cn">FALSE</span>] </span>
<span id="cb372-25"><a href="viz-chapter.html#cb372-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb372-26"><a href="viz-chapter.html#cb372-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Order data based on </span></span>
<span id="cb372-27"><a href="viz-chapter.html#cb372-27" aria-hidden="true" tabindex="-1"></a>tse_phylum_subset <span class="ot">&lt;-</span> tse_phylum_subset[ , <span class="fu">rownames</span>(sample_data)]</span>
<span id="cb372-28"><a href="viz-chapter.html#cb372-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb372-29"><a href="viz-chapter.html#cb372-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Add sample type data</span></span>
<span id="cb372-30"><a href="viz-chapter.html#cb372-30" aria-hidden="true" tabindex="-1"></a>sample_data<span class="sc">$</span>sample_types <span class="ot">&lt;-</span> <span class="fu">unfactor</span>(<span class="fu">colData</span>(tse_phylum_subset)<span class="sc">$</span>SampleType)</span>
<span id="cb372-31"><a href="viz-chapter.html#cb372-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb372-32"><a href="viz-chapter.html#cb372-32" aria-hidden="true" tabindex="-1"></a>sample_data</span></code></pre></div>
<pre><code>##         clusters sample_types
## M11Plmr        2         Skin
## M31Plmr        2         Skin
## F21Plmr        2         Skin
## M31Fcsw        1        Feces
## M11Fcsw        1        Feces
## TS28           3        Feces
## TS29           3        Feces
## M31Tong        3       Tongue
## M11Tong        3       Tongue</code></pre>
<p>Now we can create heatmap with additional annotations.</p>
<div class="sourceCode" id="cb374"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb374-1"><a href="viz-chapter.html#cb374-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Determines the scaling of colorss</span></span>
<span id="cb374-2"><a href="viz-chapter.html#cb374-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Scale colors</span></span>
<span id="cb374-3"><a href="viz-chapter.html#cb374-3" aria-hidden="true" tabindex="-1"></a>breaks <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="fu">ceiling</span>(<span class="fu">max</span>(<span class="fu">abs</span>(mat))), <span class="fu">ceiling</span>(<span class="fu">max</span>(<span class="fu">abs</span>(mat))), </span>
<span id="cb374-4"><a href="viz-chapter.html#cb374-4" aria-hidden="true" tabindex="-1"></a>              <span class="at">length.out =</span> <span class="fu">ifelse</span>( <span class="fu">max</span>(<span class="fu">abs</span>(mat))<span class="sc">&gt;</span><span class="dv">5</span>, <span class="dv">2</span><span class="sc">*</span><span class="fu">ceiling</span>(<span class="fu">max</span>(<span class="fu">abs</span>(mat))), <span class="dv">10</span> ) )</span>
<span id="cb374-5"><a href="viz-chapter.html#cb374-5" aria-hidden="true" tabindex="-1"></a>colors <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">&quot;darkblue&quot;</span>, <span class="st">&quot;blue&quot;</span>, <span class="st">&quot;white&quot;</span>, <span class="st">&quot;red&quot;</span>, <span class="st">&quot;darkred&quot;</span>))(<span class="fu">length</span>(breaks)<span class="sc">-</span><span class="dv">1</span>)</span>
<span id="cb374-6"><a href="viz-chapter.html#cb374-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb374-7"><a href="viz-chapter.html#cb374-7" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(mat, <span class="at">annotation_row =</span> taxa_clusters, </span>
<span id="cb374-8"><a href="viz-chapter.html#cb374-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_col =</span> sample_data,</span>
<span id="cb374-9"><a href="viz-chapter.html#cb374-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">breaks =</span> breaks,</span>
<span id="cb374-10"><a href="viz-chapter.html#cb374-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">color =</span> colors)</span></code></pre></div>
<p><img src="19_visualization_techniques_files/figure-html/pheatmap6-1.png" width="672" /></p>
<p>The package <em>sechm</em> allows for further visual capabilities and flexibility.
In this case, the clustering step is automatically performed by the plotting
function and does not need to be executed in advance.</p>
<div class="sourceCode" id="cb375"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb375-1"><a href="viz-chapter.html#cb375-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Stores annotation colros to metadata</span></span>
<span id="cb375-2"><a href="viz-chapter.html#cb375-2" aria-hidden="true" tabindex="-1"></a><span class="fu">metadata</span>(tse_phylum_subset)<span class="sc">$</span>anno_colors<span class="sc">$</span>SampleType <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">Feces =</span> <span class="st">&quot;blue&quot;</span>, </span>
<span id="cb375-3"><a href="viz-chapter.html#cb375-3" aria-hidden="true" tabindex="-1"></a>                                                        <span class="at">Skin =</span> <span class="st">&quot;red&quot;</span>, </span>
<span id="cb375-4"><a href="viz-chapter.html#cb375-4" aria-hidden="true" tabindex="-1"></a>                                                        <span class="at">Tongue =</span> <span class="st">&quot;gray&quot;</span>)</span>
<span id="cb375-5"><a href="viz-chapter.html#cb375-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb375-6"><a href="viz-chapter.html#cb375-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a plot</span></span>
<span id="cb375-7"><a href="viz-chapter.html#cb375-7" aria-hidden="true" tabindex="-1"></a><span class="fu">sechm</span>(tse_phylum_subset, </span>
<span id="cb375-8"><a href="viz-chapter.html#cb375-8" aria-hidden="true" tabindex="-1"></a>      <span class="at">features =</span> <span class="fu">rownames</span>(tse_phylum_subset), </span>
<span id="cb375-9"><a href="viz-chapter.html#cb375-9" aria-hidden="true" tabindex="-1"></a>      <span class="at">assayName =</span> <span class="st">&quot;clr&quot;</span>, </span>
<span id="cb375-10"><a href="viz-chapter.html#cb375-10" aria-hidden="true" tabindex="-1"></a>      <span class="at">do.scale =</span> <span class="cn">TRUE</span>, </span>
<span id="cb375-11"><a href="viz-chapter.html#cb375-11" aria-hidden="true" tabindex="-1"></a>      <span class="at">top_annotation =</span> <span class="fu">c</span>(<span class="st">&quot;SampleType&quot;</span>), </span>
<span id="cb375-12"><a href="viz-chapter.html#cb375-12" aria-hidden="true" tabindex="-1"></a>      <span class="at">gaps_at =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb375-13"><a href="viz-chapter.html#cb375-13" aria-hidden="true" tabindex="-1"></a>      <span class="at">cluster_cols =</span> <span class="cn">TRUE</span>, <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="19_visualization_techniques_files/figure-html/sechm-1.png" width="672" /></p>
<p>It is also possible to create an analogous heatmap by just using the <em>ggplot2</em>
package. However, a relatively long code is required to generate an identical output.</p>
<div class="sourceCode" id="cb376"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb376-1"><a href="viz-chapter.html#cb376-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Add feature names to column as a factor</span></span>
<span id="cb376-2"><a href="viz-chapter.html#cb376-2" aria-hidden="true" tabindex="-1"></a>taxa_clusters<span class="sc">$</span>Feature <span class="ot">&lt;-</span> <span class="fu">rownames</span>(taxa_clusters)</span>
<span id="cb376-3"><a href="viz-chapter.html#cb376-3" aria-hidden="true" tabindex="-1"></a>taxa_clusters<span class="sc">$</span>Feature <span class="ot">&lt;-</span> <span class="fu">factor</span>(taxa_clusters<span class="sc">$</span>Feature, <span class="at">levels =</span> taxa_clusters<span class="sc">$</span>Feature)</span>
<span id="cb376-4"><a href="viz-chapter.html#cb376-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb376-5"><a href="viz-chapter.html#cb376-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Create annotation plot</span></span>
<span id="cb376-6"><a href="viz-chapter.html#cb376-6" aria-hidden="true" tabindex="-1"></a>row_annotation <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(taxa_clusters) <span class="sc">+</span> </span>
<span id="cb376-7"><a href="viz-chapter.html#cb376-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="cn">NA</span>, <span class="at">y =</span> Feature, <span class="at">fill =</span> clusters)) <span class="sc">+</span> <span class="fu">coord_equal</span>(<span class="at">ratio =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb376-8"><a href="viz-chapter.html#cb376-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb376-9"><a href="viz-chapter.html#cb376-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x=</span><span class="fu">element_blank</span>(),</span>
<span id="cb376-10"><a href="viz-chapter.html#cb376-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y=</span><span class="fu">element_blank</span>(),</span>
<span id="cb376-11"><a href="viz-chapter.html#cb376-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.ticks.y=</span><span class="fu">element_blank</span>(),</span>
<span id="cb376-12"><a href="viz-chapter.html#cb376-12" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.y=</span><span class="fu">element_blank</span>(),</span>
<span id="cb376-13"><a href="viz-chapter.html#cb376-13" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust=</span><span class="dv">1</span>),</span>
<span id="cb376-14"><a href="viz-chapter.html#cb376-14" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.margin=</span><span class="fu">margin</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>),</span>
<span id="cb376-15"><a href="viz-chapter.html#cb376-15" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">+</span></span>
<span id="cb376-16"><a href="viz-chapter.html#cb376-16" aria-hidden="true" tabindex="-1"></a>      <span class="fu">labs</span>(<span class="at">fill =</span> <span class="st">&quot;Clusters&quot;</span>, <span class="at">x =</span> <span class="st">&quot;Clusters&quot;</span>)</span>
<span id="cb376-17"><a href="viz-chapter.html#cb376-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb376-18"><a href="viz-chapter.html#cb376-18" aria-hidden="true" tabindex="-1"></a><span class="co"># to view the notation, run</span></span>
<span id="cb376-19"><a href="viz-chapter.html#cb376-19" aria-hidden="true" tabindex="-1"></a><span class="co"># row_annotation</span></span>
<span id="cb376-20"><a href="viz-chapter.html#cb376-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb376-21"><a href="viz-chapter.html#cb376-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Add sample names to one of the columns</span></span>
<span id="cb376-22"><a href="viz-chapter.html#cb376-22" aria-hidden="true" tabindex="-1"></a>sample_data<span class="sc">$</span>sample <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">rownames</span>(sample_data), <span class="at">levels =</span> <span class="fu">rownames</span>(sample_data))</span>
<span id="cb376-23"><a href="viz-chapter.html#cb376-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb376-24"><a href="viz-chapter.html#cb376-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Create annotation plot</span></span>
<span id="cb376-25"><a href="viz-chapter.html#cb376-25" aria-hidden="true" tabindex="-1"></a>sample_types_annotation <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(sample_data) <span class="sc">+</span> <span class="fu">scale_y_discrete</span>(<span class="at">position =</span> <span class="st">&quot;right&quot;</span>, <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb376-26"><a href="viz-chapter.html#cb376-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>(<span class="fu">aes</span>(<span class="at">y =</span> <span class="cn">NA</span>, <span class="at">x =</span> sample, <span class="at">fill =</span> sample_types)) <span class="sc">+</span> <span class="fu">coord_equal</span>(<span class="at">ratio =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb376-27"><a href="viz-chapter.html#cb376-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb376-28"><a href="viz-chapter.html#cb376-28" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x=</span><span class="fu">element_blank</span>(),</span>
<span id="cb376-29"><a href="viz-chapter.html#cb376-29" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y=</span><span class="fu">element_blank</span>(),</span>
<span id="cb376-30"><a href="viz-chapter.html#cb376-30" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x=</span><span class="fu">element_blank</span>(),</span>
<span id="cb376-31"><a href="viz-chapter.html#cb376-31" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.ticks.x=</span><span class="fu">element_blank</span>(),</span>
<span id="cb376-32"><a href="viz-chapter.html#cb376-32" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.margin=</span><span class="fu">margin</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>),</span>
<span id="cb376-33"><a href="viz-chapter.html#cb376-33" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.y.right =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">0</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>)</span>
<span id="cb376-34"><a href="viz-chapter.html#cb376-34" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">+</span></span>
<span id="cb376-35"><a href="viz-chapter.html#cb376-35" aria-hidden="true" tabindex="-1"></a>      <span class="fu">labs</span>(<span class="at">fill =</span> <span class="st">&quot;Sample types&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Sample types&quot;</span>)</span>
<span id="cb376-36"><a href="viz-chapter.html#cb376-36" aria-hidden="true" tabindex="-1"></a><span class="co"># to view the notation, run</span></span>
<span id="cb376-37"><a href="viz-chapter.html#cb376-37" aria-hidden="true" tabindex="-1"></a><span class="co"># sample_types_annotation</span></span>
<span id="cb376-38"><a href="viz-chapter.html#cb376-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb376-39"><a href="viz-chapter.html#cb376-39" aria-hidden="true" tabindex="-1"></a><span class="co"># Create annotation plot</span></span>
<span id="cb376-40"><a href="viz-chapter.html#cb376-40" aria-hidden="true" tabindex="-1"></a>sample_clusters_annotation <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(sample_data) <span class="sc">+</span> <span class="fu">scale_y_discrete</span>(<span class="at">position =</span> <span class="st">&quot;right&quot;</span>, <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb376-41"><a href="viz-chapter.html#cb376-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>(<span class="fu">aes</span>(<span class="at">y =</span> <span class="cn">NA</span>, <span class="at">x =</span> sample, <span class="at">fill =</span> clusters)) <span class="sc">+</span> <span class="fu">coord_equal</span>(<span class="at">ratio =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb376-42"><a href="viz-chapter.html#cb376-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb376-43"><a href="viz-chapter.html#cb376-43" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x=</span><span class="fu">element_blank</span>(),</span>
<span id="cb376-44"><a href="viz-chapter.html#cb376-44" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y=</span><span class="fu">element_blank</span>(),</span>
<span id="cb376-45"><a href="viz-chapter.html#cb376-45" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x=</span><span class="fu">element_blank</span>(),</span>
<span id="cb376-46"><a href="viz-chapter.html#cb376-46" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.ticks.x=</span><span class="fu">element_blank</span>(),</span>
<span id="cb376-47"><a href="viz-chapter.html#cb376-47" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.margin=</span><span class="fu">margin</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>),</span>
<span id="cb376-48"><a href="viz-chapter.html#cb376-48" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.y.right =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">0</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>)</span>
<span id="cb376-49"><a href="viz-chapter.html#cb376-49" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">+</span></span>
<span id="cb376-50"><a href="viz-chapter.html#cb376-50" aria-hidden="true" tabindex="-1"></a>      <span class="fu">labs</span>(<span class="at">fill =</span> <span class="st">&quot;Clusters&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Clusters&quot;</span>)</span>
<span id="cb376-51"><a href="viz-chapter.html#cb376-51" aria-hidden="true" tabindex="-1"></a><span class="co"># to view the notation, run</span></span>
<span id="cb376-52"><a href="viz-chapter.html#cb376-52" aria-hidden="true" tabindex="-1"></a><span class="co"># sample_clusters_annotation</span></span>
<span id="cb376-53"><a href="viz-chapter.html#cb376-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb376-54"><a href="viz-chapter.html#cb376-54" aria-hidden="true" tabindex="-1"></a><span class="co"># Order data based on clusters and sample types</span></span>
<span id="cb376-55"><a href="viz-chapter.html#cb376-55" aria-hidden="true" tabindex="-1"></a>mat <span class="ot">&lt;-</span> mat[<span class="fu">unfactor</span>(taxa_clusters<span class="sc">$</span>Feature), <span class="fu">unfactor</span>(sample_data<span class="sc">$</span>sample)]</span>
<span id="cb376-56"><a href="viz-chapter.html#cb376-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb376-57"><a href="viz-chapter.html#cb376-57" aria-hidden="true" tabindex="-1"></a><span class="co"># ggplot requires data in melted format</span></span>
<span id="cb376-58"><a href="viz-chapter.html#cb376-58" aria-hidden="true" tabindex="-1"></a>melted_mat <span class="ot">&lt;-</span> <span class="fu">melt</span>(mat)</span>
<span id="cb376-59"><a href="viz-chapter.html#cb376-59" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(melted_mat) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Taxa&quot;</span>, <span class="st">&quot;Sample&quot;</span>, <span class="st">&quot;clr_z&quot;</span>)</span>
<span id="cb376-60"><a href="viz-chapter.html#cb376-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb376-61"><a href="viz-chapter.html#cb376-61" aria-hidden="true" tabindex="-1"></a><span class="co"># Determines the scaling of colorss</span></span>
<span id="cb376-62"><a href="viz-chapter.html#cb376-62" aria-hidden="true" tabindex="-1"></a>maxval <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">max</span>(<span class="fu">abs</span>(melted_mat<span class="sc">$</span>clr_z)))</span>
<span id="cb376-63"><a href="viz-chapter.html#cb376-63" aria-hidden="true" tabindex="-1"></a>limits <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="sc">-</span>maxval, maxval)</span>
<span id="cb376-64"><a href="viz-chapter.html#cb376-64" aria-hidden="true" tabindex="-1"></a>breaks <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="at">from =</span> <span class="fu">min</span>(limits), <span class="at">to =</span> <span class="fu">max</span>(limits), <span class="at">by =</span> <span class="fl">0.5</span>)</span>
<span id="cb376-65"><a href="viz-chapter.html#cb376-65" aria-hidden="true" tabindex="-1"></a>colours <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;darkblue&quot;</span>, <span class="st">&quot;blue&quot;</span>, <span class="st">&quot;white&quot;</span>, <span class="st">&quot;red&quot;</span>, <span class="st">&quot;darkred&quot;</span>)</span>
<span id="cb376-66"><a href="viz-chapter.html#cb376-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb376-67"><a href="viz-chapter.html#cb376-67" aria-hidden="true" tabindex="-1"></a>heatmap <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(melted_mat) <span class="sc">+</span> </span>
<span id="cb376-68"><a href="viz-chapter.html#cb376-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>(<span class="fu">aes</span>(<span class="at">x =</span> Sample, <span class="at">y =</span> Taxa, <span class="at">fill =</span> clr_z)) <span class="sc">+</span></span>
<span id="cb376-69"><a href="viz-chapter.html#cb376-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb376-70"><a href="viz-chapter.html#cb376-70" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.y=</span><span class="fu">element_blank</span>(),</span>
<span id="cb376-71"><a href="viz-chapter.html#cb376-71" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x=</span><span class="fu">element_blank</span>(),</span>
<span id="cb376-72"><a href="viz-chapter.html#cb376-72" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.ticks.y=</span><span class="fu">element_blank</span>(),</span>
<span id="cb376-73"><a href="viz-chapter.html#cb376-73" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust=</span><span class="dv">1</span>),</span>
<span id="cb376-74"><a href="viz-chapter.html#cb376-74" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb376-75"><a href="viz-chapter.html#cb376-75" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.margin=</span><span class="fu">margin</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>), <span class="co"># removes margins</span></span>
<span id="cb376-76"><a href="viz-chapter.html#cb376-76" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.key.height=</span> <span class="fu">unit</span>(<span class="dv">1</span>, <span class="st">&#39;cm&#39;</span>)</span>
<span id="cb376-77"><a href="viz-chapter.html#cb376-77" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb376-78"><a href="viz-chapter.html#cb376-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradientn</span>(<span class="at">name =</span> <span class="st">&quot;CLR + Z transform&quot;</span>, </span>
<span id="cb376-79"><a href="viz-chapter.html#cb376-79" aria-hidden="true" tabindex="-1"></a>                       <span class="at">breaks =</span> breaks, </span>
<span id="cb376-80"><a href="viz-chapter.html#cb376-80" aria-hidden="true" tabindex="-1"></a>                       <span class="at">limits =</span> limits, </span>
<span id="cb376-81"><a href="viz-chapter.html#cb376-81" aria-hidden="true" tabindex="-1"></a>                       <span class="at">colours =</span> colours) <span class="sc">+</span> </span>
<span id="cb376-82"><a href="viz-chapter.html#cb376-82" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="at">position =</span> <span class="st">&quot;right&quot;</span>)</span>
<span id="cb376-83"><a href="viz-chapter.html#cb376-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb376-84"><a href="viz-chapter.html#cb376-84" aria-hidden="true" tabindex="-1"></a>heatmap</span></code></pre></div>
<p><img src="19_visualization_techniques_files/figure-html/more_complex_heatmap-1.png" width="672" /></p>
<div class="sourceCode" id="cb377"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb377-1"><a href="viz-chapter.html#cb377-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create layout</span></span>
<span id="cb377-2"><a href="viz-chapter.html#cb377-2" aria-hidden="true" tabindex="-1"></a>design <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb377-3"><a href="viz-chapter.html#cb377-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">area</span>(<span class="dv">3</span>, <span class="dv">1</span>, <span class="dv">4</span>, <span class="dv">1</span>),</span>
<span id="cb377-4"><a href="viz-chapter.html#cb377-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">area</span>(<span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">1</span>, <span class="dv">3</span>),</span>
<span id="cb377-5"><a href="viz-chapter.html#cb377-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">area</span>(<span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">3</span>),</span>
<span id="cb377-6"><a href="viz-chapter.html#cb377-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">area</span>(<span class="dv">3</span>, <span class="dv">2</span>, <span class="dv">4</span>, <span class="dv">3</span>)</span>
<span id="cb377-7"><a href="viz-chapter.html#cb377-7" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb377-8"><a href="viz-chapter.html#cb377-8" aria-hidden="true" tabindex="-1"></a><span class="co"># to view the design, run</span></span>
<span id="cb377-9"><a href="viz-chapter.html#cb377-9" aria-hidden="true" tabindex="-1"></a><span class="co"># plot(design)</span></span>
<span id="cb377-10"><a href="viz-chapter.html#cb377-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb377-11"><a href="viz-chapter.html#cb377-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine plots</span></span>
<span id="cb377-12"><a href="viz-chapter.html#cb377-12" aria-hidden="true" tabindex="-1"></a>plot <span class="ot">&lt;-</span> row_annotation <span class="sc">+</span> sample_clusters_annotation <span class="sc">+</span> sample_types_annotation <span class="sc">+</span> heatmap  <span class="sc">+</span></span>
<span id="cb377-13"><a href="viz-chapter.html#cb377-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plot_layout</span>(<span class="at">design =</span> design, <span class="at">guides =</span> <span class="st">&quot;collect&quot;</span>, <span class="co"># Specify layout, collect legends</span></span>
<span id="cb377-14"><a href="viz-chapter.html#cb377-14" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb377-15"><a href="viz-chapter.html#cb377-15" aria-hidden="true" tabindex="-1"></a>                <span class="co"># Adjust widths and heights to align plots.</span></span>
<span id="cb377-16"><a href="viz-chapter.html#cb377-16" aria-hidden="true" tabindex="-1"></a>                <span class="co"># When annotation plot is larger, it might not fit into its column/row.</span></span>
<span id="cb377-17"><a href="viz-chapter.html#cb377-17" aria-hidden="true" tabindex="-1"></a>                <span class="co"># Then you need to make column/row larger.</span></span>
<span id="cb377-18"><a href="viz-chapter.html#cb377-18" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb377-19"><a href="viz-chapter.html#cb377-19" aria-hidden="true" tabindex="-1"></a>                <span class="co"># Relative widths and heights of each column and row:</span></span>
<span id="cb377-20"><a href="viz-chapter.html#cb377-20" aria-hidden="true" tabindex="-1"></a>                <span class="co"># Currently, the width of the first column is 15 % and the height of</span></span>
<span id="cb377-21"><a href="viz-chapter.html#cb377-21" aria-hidden="true" tabindex="-1"></a>                <span class="co"># first two rows are 30 % the size of others</span></span>
<span id="cb377-22"><a href="viz-chapter.html#cb377-22" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb377-23"><a href="viz-chapter.html#cb377-23" aria-hidden="true" tabindex="-1"></a>                <span class="co"># To get this work most of the times, you can adjust all sizes to be 1, i.e. equal, </span></span>
<span id="cb377-24"><a href="viz-chapter.html#cb377-24" aria-hidden="true" tabindex="-1"></a>                <span class="co"># but then the gaps between plots are larger.</span></span>
<span id="cb377-25"><a href="viz-chapter.html#cb377-25" aria-hidden="true" tabindex="-1"></a>                <span class="at">widths =</span> <span class="fu">c</span>(<span class="fl">0.15</span>, <span class="dv">1</span>, <span class="dv">1</span>),</span>
<span id="cb377-26"><a href="viz-chapter.html#cb377-26" aria-hidden="true" tabindex="-1"></a>                <span class="at">heights =</span> <span class="fu">c</span>(<span class="fl">0.3</span>, <span class="fl">0.3</span>, <span class="dv">1</span>, <span class="dv">1</span>))</span>
<span id="cb377-27"><a href="viz-chapter.html#cb377-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb377-28"><a href="viz-chapter.html#cb377-28" aria-hidden="true" tabindex="-1"></a><span class="co"># plot</span></span></code></pre></div>
<div class="sourceCode" id="cb378"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb378-1"><a href="viz-chapter.html#cb378-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create layout</span></span>
<span id="cb378-2"><a href="viz-chapter.html#cb378-2" aria-hidden="true" tabindex="-1"></a>design <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb378-3"><a href="viz-chapter.html#cb378-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">area</span>(<span class="dv">4</span>, <span class="dv">1</span>, <span class="dv">5</span>, <span class="dv">1</span>),</span>
<span id="cb378-4"><a href="viz-chapter.html#cb378-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">area</span>(<span class="dv">4</span>, <span class="dv">2</span>, <span class="dv">5</span>, <span class="dv">2</span>),</span>
<span id="cb378-5"><a href="viz-chapter.html#cb378-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">area</span>(<span class="dv">1</span>, <span class="dv">3</span>, <span class="dv">1</span>, <span class="dv">4</span>),</span>
<span id="cb378-6"><a href="viz-chapter.html#cb378-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">area</span>(<span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">2</span>, <span class="dv">4</span>),</span>
<span id="cb378-7"><a href="viz-chapter.html#cb378-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">area</span>(<span class="dv">3</span>, <span class="dv">3</span>, <span class="dv">3</span>, <span class="dv">4</span>),</span>
<span id="cb378-8"><a href="viz-chapter.html#cb378-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">area</span>(<span class="dv">4</span>, <span class="dv">3</span>, <span class="dv">5</span>, <span class="dv">4</span>)</span>
<span id="cb378-9"><a href="viz-chapter.html#cb378-9" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb378-10"><a href="viz-chapter.html#cb378-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb378-11"><a href="viz-chapter.html#cb378-11" aria-hidden="true" tabindex="-1"></a><span class="co"># to view the design, run</span></span>
<span id="cb378-12"><a href="viz-chapter.html#cb378-12" aria-hidden="true" tabindex="-1"></a><span class="co"># plot(design)</span></span>
<span id="cb378-13"><a href="viz-chapter.html#cb378-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb378-14"><a href="viz-chapter.html#cb378-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine plots</span></span>
<span id="cb378-15"><a href="viz-chapter.html#cb378-15" aria-hidden="true" tabindex="-1"></a>plot <span class="ot">&lt;-</span> taxa_tree <span class="sc">+</span> </span>
<span id="cb378-16"><a href="viz-chapter.html#cb378-16" aria-hidden="true" tabindex="-1"></a>  row_annotation <span class="sc">+</span></span>
<span id="cb378-17"><a href="viz-chapter.html#cb378-17" aria-hidden="true" tabindex="-1"></a>  sample_tree <span class="sc">+</span> </span>
<span id="cb378-18"><a href="viz-chapter.html#cb378-18" aria-hidden="true" tabindex="-1"></a>  sample_clusters_annotation <span class="sc">+</span></span>
<span id="cb378-19"><a href="viz-chapter.html#cb378-19" aria-hidden="true" tabindex="-1"></a>  sample_types_annotation <span class="sc">+</span></span>
<span id="cb378-20"><a href="viz-chapter.html#cb378-20" aria-hidden="true" tabindex="-1"></a>  heatmap <span class="sc">+</span></span>
<span id="cb378-21"><a href="viz-chapter.html#cb378-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plot_layout</span>(<span class="at">design =</span> design, <span class="at">guides =</span> <span class="st">&quot;collect&quot;</span>, <span class="co"># Specify layout, collect legends</span></span>
<span id="cb378-22"><a href="viz-chapter.html#cb378-22" aria-hidden="true" tabindex="-1"></a>                <span class="at">widths =</span> <span class="fu">c</span>(<span class="fl">0.2</span>, <span class="fl">0.15</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>),</span>
<span id="cb378-23"><a href="viz-chapter.html#cb378-23" aria-hidden="true" tabindex="-1"></a>                <span class="at">heights =</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="fl">0.15</span>, <span class="fl">0.15</span>, <span class="fl">0.25</span>, <span class="dv">1</span>, <span class="dv">1</span>))</span>
<span id="cb378-24"><a href="viz-chapter.html#cb378-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb378-25"><a href="viz-chapter.html#cb378-25" aria-hidden="true" tabindex="-1"></a>plot</span></code></pre></div>
<p><img src="19_visualization_techniques_files/figure-html/more_complex_heatmap3-1.png" width="960" /></p>
<p>Heatmaps find several other applications in biclustering and multi-assay
analyses, that are discussed in chapters <a href="biclustering.html#biclustering">11</a> and
@ref(multi-assay_analyses), where the packages <em>cobiclust</em> and <em>MOFA2</em> are of interest.</p>
</div>
</div>
<div id="session-info-10" class="section level2 unnumbered hasAnchor">
<h2>Session Info<a href="viz-chapter.html#session-info-10" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<button class="rebook-collapse">
View session info
</button>
<div class="rebook-content">
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.4 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ggtree_3.4.0                   ape_5.6-2                     
 [3] pheatmap_1.0.12                reshape2_1.4.4                
 [5] sechm_1.4.1                    miaViz_1.3.3                  
 [7] ggraph_2.0.5                   patchwork_1.1.1               
 [9] scater_1.24.0                  scuttle_1.6.2                 
[11] mia_1.3.25                     MultiAssayExperiment_1.22.0   
[13] TreeSummarizedExperiment_2.1.4 Biostrings_2.64.0             
[15] XVector_0.36.0                 SingleCellExperiment_1.18.0   
[17] SummarizedExperiment_1.26.1    Biobase_2.56.0                
[19] GenomicRanges_1.48.0           GenomeInfoDb_1.32.2           
[21] IRanges_2.30.0                 S4Vectors_0.34.0              
[23] BiocGenerics_0.42.0            MatrixGenerics_1.8.0          
[25] matrixStats_0.62.0-9000        ggplot2_3.3.6                 
[27] BiocStyle_2.24.0               rebook_1.6.0                  

loaded via a namespace (and not attached):
  [1] circlize_0.4.15             plyr_1.8.7                 
  [3] igraph_1.3.2                lazyeval_0.2.2             
  [5] splines_4.2.0               BiocParallel_1.30.3        
  [7] digest_0.6.29               foreach_1.5.2              
  [9] yulab.utils_0.0.4           htmltools_0.5.2            
 [11] viridis_0.6.2               fansi_1.0.3                
 [13] magrittr_2.0.3              memoise_2.0.1              
 [15] ScaledMatrix_1.4.0          doParallel_1.0.17          
 [17] cluster_2.1.3               DECIPHER_2.24.0            
 [19] ComplexHeatmap_2.12.0       graphlayouts_0.8.0         
 [21] colorspace_2.0-3            blob_1.2.3                 
 [23] ggrepel_0.9.1               xfun_0.31                  
 [25] dplyr_1.0.9                 crayon_1.5.1               
 [27] RCurl_1.98-1.7              jsonlite_1.8.0             
 [29] graph_1.74.0                iterators_1.0.14           
 [31] glue_1.6.2                  polyclip_1.10-0            
 [33] registry_0.5-1              gtable_0.3.0               
 [35] zlibbioc_1.42.0             V8_4.2.0                   
 [37] GetoptLong_1.0.5            DelayedArray_0.22.0        
 [39] BiocSingular_1.12.0         shape_1.4.6                
 [41] scales_1.2.0                DBI_1.1.2                  
 [43] randomcoloR_1.1.0.1         Rcpp_1.0.8.3               
 [45] viridisLite_0.4.0           clue_0.3-61                
 [47] decontam_1.16.0             gridGraphics_0.5-1         
 [49] tidytree_0.3.9              bit_4.0.4                  
 [51] rsvd_1.0.5                  RColorBrewer_1.1-3         
 [53] dir.expiry_1.4.0            ellipsis_0.3.2             
 [55] pkgconfig_2.0.3             XML_3.99-0.10              
 [57] farver_2.1.0                CodeDepends_0.6.5          
 [59] sass_0.4.1                  utf8_1.2.2                 
 [61] labeling_0.4.2              ggplotify_0.1.0            
 [63] tidyselect_1.1.2            rlang_1.0.2                
 [65] munsell_0.5.0               tools_4.2.0                
 [67] cachem_1.0.6                cli_3.3.0                  
 [69] DirichletMultinomial_1.38.0 generics_0.1.2             
 [71] RSQLite_2.2.14              evaluate_0.15              
 [73] stringr_1.4.0               fastmap_1.1.0              
 [75] yaml_2.3.5                  knitr_1.39                 
 [77] bit64_4.0.5                 tidygraph_1.2.1            
 [79] purrr_0.3.4                 nlme_3.1-158               
 [81] sparseMatrixStats_1.8.0     aplot_0.1.6                
 [83] compiler_4.2.0              curl_4.3.2                 
 [85] png_0.1-7                   beeswarm_0.4.0             
 [87] filelock_1.0.2              treeio_1.20.0              
 [89] tibble_3.1.7                tweenr_1.0.2               
 [91] bslib_0.3.1                 stringi_1.7.6              
 [93] highr_0.9                   lattice_0.20-45            
 [95] Matrix_1.4-1                vegan_2.6-2                
 [97] permute_0.9-7               vctrs_0.4.1                
 [99] pillar_1.7.0                lifecycle_1.0.1            
[101] BiocManager_1.30.18         GlobalOptions_0.1.2        
[103] jquerylib_0.1.4             BiocNeighbors_1.14.0       
[105] cowplot_1.1.1               bitops_1.0-7               
[107] irlba_2.3.5                 seriation_1.3.5            
[109] R6_2.5.1                    TSP_1.2-0                  
[111] bookdown_0.27               gridExtra_2.3              
[113] vipor_0.4.5                 codetools_0.2-18           
[115] MASS_7.3-57                 assertthat_0.2.1           
[117] rjson_0.2.21                withr_2.5.0                
[119] GenomeInfoDbData_1.2.8      mgcv_1.8-40                
[121] parallel_4.2.0              grid_4.2.0                 
[123] ggfun_0.0.6                 beachmat_2.12.0            
[125] tidyr_1.2.0                 rmarkdown_2.14             
[127] DelayedMatrixStats_1.18.0   Cairo_1.5-15               
[129] Rtsne_0.16                  ggnewscale_0.4.7           
[131] ggforce_0.3.3               ggbeeswarm_0.6.0           </code></pre>
</div>

</div>
</div>



            </section>

          </div>
        </div>
      </div>
<a href="multi-assay_analyses.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="resources.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["OMA.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
